Index: sys/dev/aic7xxx/aicasm/Makefile
===================================================================
--- sys/dev/aic7xxx/aicasm/Makefile.orig	2005-12-27 22:35:40.000000000 +0100
+++ sys/dev/aic7xxx/aicasm/Makefile	2006-06-24 17:31:56.000000000 +0200
@@ -14,7 +14,7 @@
 SRCS=	${GENHDRS} ${CSRCS} ${YSRCS} ${LSRCS}
 CLEANFILES+= ${GENHDRS} ${YSRCS:R:C/(.*)/\1.output/g}
 DPADD=	${LIBL}
-LDADD=	-ll
+LDADD=	-ll -ldb
 
 # Correct path for kernel builds
 # Don't rely on the kernel's .depend file
@@ -23,13 +23,6 @@
 DEPENDFILE=	.depend_aicasm
 .endif
 
-.if ${CC} == "icc"
-CFLAGS+=	-restrict
-NOSTDINC=	-X
-.else
-NOSTDINC=	-nostdinc
-.endif
-CFLAGS+= ${NOSTDINC} -I/usr/include -I.
 .ifdef MAKESRCPATH
 CFLAGS+= -I${MAKESRCPATH}
 .endif
Index: sys/dev/aic7xxx/aicasm/aicasm.h
===================================================================
--- sys/dev/aic7xxx/aicasm/aicasm.h.orig	2005-01-06 02:42:27.000000000 +0100
+++ sys/dev/aic7xxx/aicasm/aicasm.h	2006-06-24 17:31:56.000000000 +0200
@@ -42,8 +42,9 @@
  * $FreeBSD: src/sys/dev/aic7xxx/aicasm/aicasm.h,v 1.17 2005/01/06 01:42:27 imp Exp $
  */
 
-#ifdef __linux__
-#include "../queue.h"
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
Index: sys/dev/aic7xxx/aicasm/aicasm_gram.y
===================================================================
--- sys/dev/aic7xxx/aicasm/aicasm_gram.y.orig	2005-01-06 02:42:27.000000000 +0100
+++ sys/dev/aic7xxx/aicasm/aicasm_gram.y	2006-06-24 17:31:56.000000000 +0200
@@ -52,8 +52,9 @@
 #include <string.h>
 #include <sysexits.h>
 
-#ifdef __linux__
-#include "../queue.h"
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
Index: sys/dev/aic7xxx/aicasm/aicasm_macro_gram.y
===================================================================
--- sys/dev/aic7xxx/aicasm/aicasm_macro_gram.y.orig	2005-01-06 02:42:27.000000000 +0100
+++ sys/dev/aic7xxx/aicasm/aicasm_macro_gram.y	2006-06-24 17:31:56.000000000 +0200
@@ -52,8 +52,9 @@
 #include <string.h>
 #include <sysexits.h>
 
-#ifdef __linux__
-#include "../queue.h"
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
Index: sys/dev/aic7xxx/aicasm/aicasm_macro_scan.l
===================================================================
--- sys/dev/aic7xxx/aicasm/aicasm_macro_scan.l.orig	2005-01-06 02:42:27.000000000 +0100
+++ sys/dev/aic7xxx/aicasm/aicasm_macro_scan.l	2006-06-24 17:31:56.000000000 +0200
@@ -51,8 +51,10 @@
 #include <stdio.h>
 #include <string.h>
 #include <sysexits.h>
-#ifdef __linux__
-#include "../queue.h"
+
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
Index: sys/dev/aic7xxx/aicasm/aicasm_scan.l
===================================================================
--- sys/dev/aic7xxx/aicasm/aicasm_scan.l.orig	2005-01-06 02:42:27.000000000 +0100
+++ sys/dev/aic7xxx/aicasm/aicasm_scan.l	2006-06-24 17:31:56.000000000 +0200
@@ -51,8 +51,10 @@
 #include <stdio.h>
 #include <string.h>
 #include <sysexits.h>
-#ifdef __linux__
-#include "../queue.h"
+
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
Index: sys/dev/aic7xxx/aicasm/aicasm_symbol.c
===================================================================
--- sys/dev/aic7xxx/aicasm/aicasm_symbol.c.orig	2005-01-06 02:42:27.000000000 +0100
+++ sys/dev/aic7xxx/aicasm/aicasm_symbol.c	2006-06-24 17:31:56.000000000 +0200
@@ -44,10 +44,11 @@
 
 #include <sys/types.h>
 
-#ifdef __linux__
-#include "aicdb.h"
+#include <sys/param.h>		/* BSD */
+#ifdef BSD
+#include <db.h>			/* BSD native libdb */
 #else
-#include <db.h>
+#include <db_185.h>		/* Sleepycat 1.85 compat */
 #endif
 #include <fcntl.h>
 #include <inttypes.h>
Index: sys/dev/aic7xxx/aicasm/aicasm_symbol.h
===================================================================
--- sys/dev/aic7xxx/aicasm/aicasm_symbol.h.orig	2005-01-06 02:42:27.000000000 +0100
+++ sys/dev/aic7xxx/aicasm/aicasm_symbol.h	2006-06-24 17:31:56.000000000 +0200
@@ -42,8 +42,9 @@
  * $FreeBSD: src/sys/dev/aic7xxx/aicasm/aicasm_symbol.h,v 1.17 2005/01/06 01:42:27 imp Exp $
  */
 
-#ifdef __linux__
-#include "../queue.h"
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
