
Status: in upstream BTS

diff -ur sys/conf.old/Makefile.alpha sys/conf/Makefile.alpha
--- sys/conf.old/Makefile.alpha	2006-04-25 18:41:51.000000000 +0200
+++ sys/conf/Makefile.alpha	2006-04-25 18:51:49.000000000 +0200
@@ -28,6 +28,8 @@
 .endif
 .include "$S/conf/kern.pre.mk"
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
diff -ur sys/conf.old/Makefile.amd64 sys/conf/Makefile.amd64
--- sys/conf.old/Makefile.amd64	2006-04-25 18:41:51.000000000 +0200
+++ sys/conf/Makefile.amd64	2006-04-25 18:51:49.000000000 +0200
@@ -38,6 +38,8 @@
 
 MKMODULESENV+= MACHINE=amd64
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
diff -ur sys/conf.old/Makefile.arm sys/conf/Makefile.arm
--- sys/conf.old/Makefile.arm	2006-04-25 18:41:51.000000000 +0200
+++ sys/conf/Makefile.arm	2006-04-25 18:51:49.000000000 +0200
@@ -44,6 +44,9 @@
 .if !defined(DEBUG)
 CFLAGS += -mno-apcs-frame
 .endif
+
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
diff -ur sys/conf.old/Makefile.i386 sys/conf/Makefile.i386
--- sys/conf.old/Makefile.i386	2006-04-25 18:41:51.000000000 +0200
+++ sys/conf/Makefile.i386	2006-04-25 18:51:49.000000000 +0200
@@ -32,6 +32,8 @@
 
 MKMODULESENV+= MACHINE=i386
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
diff -ur sys/conf.old/Makefile.ia64 sys/conf/Makefile.ia64
--- sys/conf.old/Makefile.ia64	2006-04-25 18:41:51.000000000 +0200
+++ sys/conf/Makefile.ia64	2006-04-25 18:51:49.000000000 +0200
@@ -38,6 +38,8 @@
 
 ASM_CFLAGS= -x assembler-with-cpp -Wa,-x -DLOCORE ${CFLAGS}
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
diff -ur sys/conf.old/Makefile.pc98 sys/conf/Makefile.pc98
--- sys/conf.old/Makefile.pc98	2006-04-25 18:41:51.000000000 +0200
+++ sys/conf/Makefile.pc98	2006-04-25 18:51:49.000000000 +0200
@@ -32,6 +32,8 @@
 
 MKMODULESENV+=	MACHINE=pc98
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
diff -ur sys/conf.old/Makefile.powerpc sys/conf/Makefile.powerpc
--- sys/conf.old/Makefile.powerpc	2006-04-25 18:41:51.000000000 +0200
+++ sys/conf/Makefile.powerpc	2006-04-25 18:51:49.000000000 +0200
@@ -37,6 +37,8 @@
 CFLAGS+=	-fno-omit-frame-pointer
 .endif
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
diff -ur sys/conf.old/Makefile.sparc64 sys/conf/Makefile.sparc64
--- sys/conf.old/Makefile.sparc64	2006-04-25 18:41:51.000000000 +0200
+++ sys/conf/Makefile.sparc64	2006-04-25 18:51:49.000000000 +0200
@@ -32,6 +32,8 @@
 
 MDOBJS=	exception.o interrupt.o
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
diff -ur usr.sbin/config.old/mkmakefile.c usr.sbin/config/mkmakefile.c
--- usr.sbin/config.old/mkmakefile.c	2006-04-25 18:44:15.000000000 +0200
+++ usr.sbin/config/mkmakefile.c	2006-04-25 18:51:59.000000000 +0200
@@ -70,6 +70,7 @@
 static void do_rules(FILE *);
 static void do_xxfiles(char *, FILE *);
 static void do_objs(FILE *);
+static void do_without_modules(FILE *);
 static void do_before_depend(FILE *);
 static int opteq(const char *, const char *);
 static void read_files(void);
@@ -150,6 +151,8 @@
 		}
 		if (eq(line, "%BEFORE_DEPEND\n"))
 			do_before_depend(ofp);
+		else if (eq(line, "%WITHOUT_MODULES\n"))
+			do_without_modules(ofp);
 		else if (eq(line, "%OBJS\n"))
 			do_objs(ofp);
 		else if (strncmp(line, "%FILES.", 7) == 0)
@@ -618,6 +621,28 @@
 }
 
 static void
+do_without_modules(FILE *fp)
+{
+	struct device *dp;
+	int lpos, len;
+	char *cp, *sp;
+
+	fprintf(fp, "MKMODULESENV+= WITHOUT_MODULES=\"");
+	lpos = 34;
+	STAILQ_FOREACH(dp, &dtab, d_next) {
+		sp = dp->d_name;
+		cp = sp + (len = strlen(sp)) - 1;
+		if (len + lpos > 72) {
+			lpos = 8;
+			fprintf(fp, "\\\n\t");
+		}
+		fprintf(fp, "%s ", sp);
+		lpos += len + 1;
+	}
+	fprintf(fp, "\"\n");
+}
+
+static void
 do_xxfiles(char *tag, FILE *fp)
 {
 	struct file_list *tp;
