
Author: aurel32
Status: Merged in 7-CURRENT.

--- sys.orig/kern/imgact_elf.c	2005-08-17 15:29:41.000000000 +0200
+++ sys/kern/imgact_elf.c	2005-08-17 15:31:08.000000000 +0200
@@ -655,7 +655,7 @@
 	/*
 	 * Do we have a valid ELF header ?
 	 */
-	if (__elfN(check_header)(hdr) != 0 || hdr->e_type != ET_EXEC)
+	if (__elfN(check_header)(hdr) != 0 || (hdr->e_type != ET_EXEC && hdr->e_type != ET_DYN))
 		return (-1);
 
 	/*
