--- sys/sys/conf.h.old	2004-10-05 00:22:17.000000000 +0200
+++ sys/sys/conf.h	2004-10-05 00:37:00.000000000 +0200
@@ -295,15 +295,15 @@
 void devfs_destroy(dev_t dev);
 
 #define		UID_ROOT	0
-#define		UID_BIN		3
-#define		UID_UUCP	66
+#define		UID_BIN		2	/* debian bin */
+#define		UID_UUCP	0	/* linux ttyS0 */
 
 #define		GID_WHEEL	0
-#define		GID_KMEM	2
-#define		GID_OPERATOR	5
-#define		GID_BIN		7
-#define		GID_GAMES	13
-#define		GID_DIALER	68
+#define		GID_KMEM	15	/* linux kmem */
+#define		GID_OPERATOR	0	/* no such group */
+#define		GID_BIN		2	/* debian bin */
+#define		GID_GAMES	60	/* debian games */
+#define		GID_DIALER	20	/* linux ttyS0 */
 
 typedef void (*dev_clone_fn)(void *arg, char *name, int namelen, dev_t *result);
 
