--- sys/boot/forth/loader.conf~	2005-12-16 17:12:00.000000000 +0100
+++ sys/boot/forth/loader.conf	2005-12-16 17:12:58.000000000 +0100
@@ -35,7 +35,7 @@
 bitmap_load="NO"		# Set this to YES if you want splash screen!
 bitmap_name="splash.bmp"	# Set this to the name of the bmp or pcx file
 bitmap_type="splash_image_data" # and place it on the module_path
-loader_color="NO"		# Set this to YES to enable splash screen colors
+loader_color="YES"		# Set this to YES to enable splash screen colors
 
 
 ##############################################################
