Index: sys/conf/newvers.sh
===================================================================
--- sys/conf/newvers.sh.orig	2007-12-09 00:42:11.000000000 +0100
+++ sys/conf/newvers.sh	2007-12-09 00:43:16.000000000 +0100
@@ -89,7 +89,7 @@
 cat << EOF > vers.c
 $COPYRIGHT
 #define SCCSSTR "@(#)${VERSION} #${v}: ${t}"
-#define VERSTR "${VERSION} #${v}: ${t}\\n    ${u}@${h}:${d}\\n"
+#define VERSTR "#${v} ${t}\\n"
 #define RELSTR "${RELEASE}"
 
 char sccs[sizeof(SCCSSTR) > 128 ? sizeof(SCCSSTR) : 128] = SCCSSTR;
