--- sys.old/conf/newvers.sh	2005-08-21 23:01:35.000000000 -0400
+++ sys/conf/newvers.sh	2005-09-03 16:17:28.000000000 -0400
@@ -36,8 +36,8 @@
 if [ "X${BRANCH_OVERRIDE}" != "X" ]; then
 	BRANCH=${BRANCH_OVERRIDE}
 fi
-RELEASE="${REVISION}-${BRANCH}"
-VERSION="${TYPE} ${RELEASE}"
+RELEASE="${REVISION}"
+VERSION="${RELEASE}"
 
 if [ "X${PARAMFILE}" != "X" ]; then
 	RELDATE=$(awk '/__FreeBSD_version.*propagated to newvers/ {print $3}' \
@@ -89,7 +89,7 @@
 cat << EOF > vers.c
 $COPYRIGHT
 #define SCCSSTR "@(#)${VERSION} #${v}: ${t}"
-#define VERSTR "${VERSION} #${v}: ${t}\\n    ${u}@${h}:${d}\\n"
+#define VERSTR "#${v} ${t}\\n"
 #define RELSTR "${RELEASE}"
 
 char sccs[sizeof(SCCSSTR) > 128 ? sizeof(SCCSSTR) : 128] = SCCSSTR;
