--- sys/conf/newvers.sh~	2006-02-11 11:48:49.000000000 +0100
+++ sys/conf/newvers.sh	2006-02-11 12:02:45.000000000 +0100
@@ -36,8 +36,8 @@
 if [ "X${BRANCH_OVERRIDE}" != "X" ]; then
 	BRANCH=${BRANCH_OVERRIDE}
 fi
-RELEASE="${REVISION}-${BRANCH}"
-VERSION="${TYPE} ${RELEASE}"
+RELEASE="${REVISION}"
+VERSION="${RELEASE}"
 
 if [ "X${PARAMFILE}" != "X" ]; then
 	RELDATE=$(awk '/__FreeBSD_version.*propagated to newvers/ {print $3}' \
