--- sys/conf/newvers.sh~	2005-12-16 17:19:12.000000000 +0100
+++ sys/conf/newvers.sh	2005-12-16 17:21:47.000000000 +0100
@@ -36,8 +36,8 @@
 if [ "X${BRANCH_OVERRIDE}" != "X" ]; then
 	BRANCH=${BRANCH_OVERRIDE}
 fi
-RELEASE=6.0-RELEASE
-VERSION="${TYPE} ${RELEASE}"
+RELEASE=${REVISION}
+VERSION=${RELEASE}
 
 if [ "X${PARAMFILE}" != "X" ]; then
 	RELDATE=$(awk '/__FreeBSD_version.*propagated to newvers/ {print $3}' \
