--- sys/conf/newvers.sh~	2006-02-16 20:03:49.000000000 +0100
+++ sys/conf/newvers.sh	2006-02-16 20:17:04.000000000 +0100
@@ -89,7 +89,7 @@
 cat << EOF > vers.c
 $COPYRIGHT
 #define SCCSSTR "@(#)${VERSION} #${v}: ${t}"
-#define VERSTR "${VERSION} #${v}: ${t}\\n    ${u}@${h}:${d}\\n"
+#define VERSTR "#${v} ${t}\\n"
 #define RELSTR "${RELEASE}"
 
 char sccs[sizeof(SCCSSTR) > 128 ? sizeof(SCCSSTR) : 128] = SCCSSTR;
