diff -Nur sys.old/conf/files sys/conf/files
--- sys.old/conf/files	2005-09-03 20:20:59.000000000 -0400
+++ sys/conf/files	2005-09-03 20:22:50.000000000 -0400
@@ -240,7 +240,6 @@
 contrib/dev/acpica/utmisc.c	optional acpi
 contrib/dev/acpica/utobject.c	optional acpi
 contrib/dev/acpica/utxface.c	optional acpi
-contrib/dev/ath/freebsd/ah_osdep.c optional ath_hal
 contrib/ipfilter/netinet/fil.c	optional ipfilter inet
 contrib/ipfilter/netinet/ip_auth.c optional ipfilter inet
 contrib/ipfilter/netinet/ip_fil_freebsd.c optional ipfilter inet
@@ -429,11 +428,6 @@
 dev/ata/atapi-cd.c		optional atapicd
 dev/ata/atapi-fd.c		optional atapifd
 dev/ata/atapi-tape.c		optional atapist
-dev/ath/ath_rate/amrr/amrr.c 	optional ath_rate_amrr
-dev/ath/ath_rate/onoe/onoe.c 	optional ath_rate_onoe
-dev/ath/ath_rate/sample/sample.c 	optional ath_rate_sample
-dev/ath/if_ath.c		optional ath
-dev/ath/if_ath_pci.c		optional ath pci
 dev/awi/am79c930.c		optional awi
 dev/awi/awi.c			optional awi
 dev/awi/if_awi_pccard.c		optional awi pccard
diff -Nur sys.old/conf/files.i386 sys/conf/files.i386
--- sys.old/conf/files.i386	2005-09-03 20:20:56.000000000 -0400
+++ sys/conf/files.i386	2005-09-03 20:23:58.000000000 -0400
@@ -46,37 +46,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-trlld.o				optional	oltr			\
-	dependency	"$S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	no-implicit-rule
-#
-hal.o				optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/i386-elf.hal.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/ath/public/i386-elf.hal.o.uu" \
-	no-implicit-rule
-opt_ah.h			optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/i386-elf.opt_ah.h"	\
-	compile-with	"cp $S/contrib/dev/ath/public/i386-elf.opt_ah.h opt_ah.h" \
-	no-obj no-implicit-rule before-depend				\
-	clean		"opt_ah.h"
-#
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/i386-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/dev/hptmv/i386-elf.raid.o.uu" \
-	no-implicit-rule
-#
 #
 compat/linux/linux_file.c	optional compat_linux
 compat/linux/linux_getcwd.c	optional compat_linux
@@ -173,11 +142,6 @@
 dev/fdc/fdc_isa.c		optional fdc isa
 dev/fdc/fdc_pccard.c		optional fdc pccard
 dev/fe/if_fe_isa.c		optional fe isa
-dev/hptmv/entry.c		optional hptmv
-dev/hptmv/mv.c			optional hptmv
-dev/hptmv/gui_lib.c		optional hptmv
-dev/hptmv/hptproc.c		optional hptmv
-dev/hptmv/ioctl.c		optional hptmv
 dev/hwpmc/hwpmc_amd.c		optional hwpmc
 dev/hwpmc/hwpmc_pentium.c	optional hwpmc
 dev/hwpmc/hwpmc_piv.c		optional hwpmc
@@ -198,7 +162,6 @@
 dev/mem/memutil.c		optional mem
 dev/mse/mse.c			optional mse
 dev/mse/mse_isa.c		optional mse isa
-dev/nve/if_nve.c		optional nve pci
 dev/ppc/ppc.c			optional ppc
 dev/ppc/ppc_puc.c		optional ppc puc pci
 dev/random/nehemiah.c		optional random
diff -Nur sys.old/conf/kern.pre.mk sys/conf/kern.pre.mk
--- sys.old/conf/kern.pre.mk	2005-09-03 20:21:12.000000000 -0400
+++ sys/conf/kern.pre.mk	2005-09-03 20:24:14.000000000 -0400
@@ -61,9 +61,6 @@
 # ... and the same for pf
 INCLUDES+= -I$S/contrib/pf
 
-# ... and the same for Atheros HAL
-INCLUDES+= -I$S/contrib/dev/ath -I$S/contrib/dev/ath/freebsd
-
 # ... and the same for the NgATM stuff
 INCLUDES+= -I$S/contrib/ngatm
 
diff -Nur sys.old/i386/conf/GENERIC sys/i386/conf/GENERIC
--- sys.old/i386/conf/GENERIC	2005-09-03 20:21:39.000000000 -0400
+++ sys/i386/conf/GENERIC	2005-09-03 20:32:53.000000000 -0400
@@ -130,7 +130,6 @@
 device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
@@ -209,7 +208,6 @@
 device		fxp		# Intel EtherExpress PRO/100B (82557, 82558)
 device		lge		# Level 1 LXT1001 gigabit Ethernet
 device		nge		# NatSemi DP83820 gigabit Ethernet
-device		nve		# nVidia nForce MCP on-board Ethernet Networking
 device		pcn		# AMD Am79C97x PCI 10/100(precedence over 'lnc')
 device		re		# RealTek 8139C+/8169/8169S/8110S
 device		rl		# RealTek 8129/8139
diff -Nur sys.old/modules/Makefile sys/modules/Makefile
--- sys.old/modules/Makefile	2005-09-03 20:21:53.000000000 -0400
+++ sys/modules/Makefile	2005-09-03 20:25:21.000000000 -0400
@@ -25,11 +25,6 @@
 	${_arl} \
 	${_asr} \
 	ata \
-	ath \
-	${_ath_hal} \
-	ath_rate_amrr \
-	ath_rate_sample \
-	ath_rate_onoe \
 	aue \
 	${_auxio} \
 	${_awi} \
@@ -90,7 +85,6 @@
 	${_hfa} \
 	hifn \
 	hme \
-	${_hptmv} \
 	hwpmc \
 	${_i2c} \
 	${_ibcs2} \
@@ -164,14 +158,12 @@
 	nfsclient \
 	nfsserver \
 	nge \
-	${_nve} \
 	nmdm \
 	${_nsp} \
 	ntfs \
 	ntfs_iconv \
 	nullfs \
 	${_nwfs} \
-	${_oltr} \
 	${_osf1} \
 	${_padlock} \
 	patm \
@@ -318,7 +310,6 @@
 _apm=		apm
 _ar=		ar
 _arcnet=	arcnet
-_ath_hal=	ath_hal
 _awi=		awi
 _bktr=		bktr
 _coff=		coff
@@ -347,7 +338,6 @@
 _ndis=		ndis
 _nsp=		nsp
 _nwfs=		nwfs
-_oltr=		oltr
 _pccard=	pccard
 _pcfclock=	pcfclock
 _pecoff=	pecoff
@@ -387,14 +377,12 @@
 _el=		el
 _ex=		ex
 _exca=		exca
-_hptmv=		hptmv
 _ichwd=		ichwd
 _ida=		ida
 _idt=		idt
 _iir=		iir
 _ips=		ips
 _mly=		mly
-_nve=		nve
 .if !defined(NO_CRYPT) || defined(ALL_MODULES)
 .if exists(${.CURDIR}/../crypto/via)
 _padlock=	padlock
@@ -427,7 +415,6 @@
 #_acpi=		acpi		# doesn't work on amd64 yet
 _agp=		agp
 _arcmsr=	arcmsr
-_ath_hal=	ath_hal
 _ciss=		ciss
 _cpufreq=	cpufreq
 _digi=		digi
@@ -444,7 +431,6 @@
 #_lnc=		lnc
 _mly=		mly
 _ndis=		ndis
-_nve=		nve
 _safe=		safe
 _scsi_low=	scsi_low
 _smbfs=		smbfs
