Index: sys/conf/files
===================================================================
--- sys/conf/files.orig
+++ sys/conf/files
@@ -440,18 +440,6 @@
 dev/ata/atapi-cd.c		optional atapicd
 dev/ata/atapi-fd.c		optional atapifd
 dev/ata/atapi-tape.c		optional atapist
-dev/ath/ah_osdep.c optional ath_hal \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/ath_rate/amrr/amrr.c	optional ath_rate_amrr \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/ath_rate/onoe/onoe.c	optional ath_rate_onoe \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/ath_rate/sample/sample.c	optional ath_rate_sample \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/if_ath.c		optional ath \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/if_ath_pci.c		optional ath pci \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
 dev/awi/am79c930.c		optional awi
 dev/awi/awi.c			optional awi
 dev/awi/if_awi_pccard.c		optional awi pccard
Index: sys/conf/files.i386
===================================================================
--- sys/conf/files.i386.orig
+++ sys/conf/files.i386
@@ -46,42 +46,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-trlld.o				optional	oltr			\
-	dependency	"$S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	no-implicit-rule
-#
-hal.o				optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/i386-elf.hal.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/ath/public/i386-elf.hal.o.uu" \
-	no-implicit-rule
-opt_ah.h			optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/i386-elf.opt_ah.h"	\
-	compile-with	"cp $S/contrib/dev/ath/public/i386-elf.opt_ah.h opt_ah.h" \
-	no-obj no-implicit-rule before-depend				\
-	clean		"opt_ah.h"
-#
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/i386-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/dev/hptmv/i386-elf.raid.o.uu" \
-	no-implicit-rule
-#
-rr232x_lib.o			optional	rr232x			\
-	dependency	"$S/dev/rr232x/i386-elf.rr232x_lib.o.uu"	\
-	compile-with	"uudecode < $S/dev/rr232x/i386-elf.rr232x_lib.o.uu" \
-	no-implicit-rule
-#
 #
 compat/linux/linux_file.c	optional compat_linux
 compat/linux/linux_getcwd.c	optional compat_linux
@@ -185,11 +149,6 @@
 dev/fdc/fdc_isa.c		optional fdc isa
 dev/fdc/fdc_pccard.c		optional fdc pccard
 dev/fe/if_fe_isa.c		optional fe isa
-dev/hptmv/entry.c		optional hptmv
-dev/hptmv/mv.c			optional hptmv
-dev/hptmv/gui_lib.c		optional hptmv
-dev/hptmv/hptproc.c		optional hptmv
-dev/hptmv/ioctl.c		optional hptmv
 dev/hwpmc/hwpmc_amd.c		optional hwpmc
 dev/hwpmc/hwpmc_pentium.c	optional hwpmc
 dev/hwpmc/hwpmc_piv.c		optional hwpmc
@@ -224,13 +183,9 @@
 dev/mxge/mxge_eth_z8e.c		optional mxge pci
 dev/mxge/mxge_ethp_z8e.c	optional mxge pci
 net/zlib.c			optional mxge
-dev/nve/if_nve.c		optional nve pci
 dev/ppc/ppc.c			optional ppc
 dev/ppc/ppc_puc.c		optional ppc puc pci
 dev/random/nehemiah.c		optional random
-dev/rr232x/os_bsd.c		optional rr232x
-dev/rr232x/osm_bsd.c		optional rr232x
-dev/rr232x/rr232x_config.c	optional rr232x
 dev/sbni/if_sbni.c		optional sbni
 dev/sbni/if_sbni_isa.c		optional sbni isa
 dev/sbni/if_sbni_pci.c		optional sbni pci
Index: sys/conf/kern.pre.mk
===================================================================
--- sys/conf/kern.pre.mk.orig
+++ sys/conf/kern.pre.mk
@@ -58,9 +58,6 @@
 # ... and the same for pf
 INCLUDES+= -I$S/contrib/pf
 
-# ... and the same for Atheros HAL
-INCLUDES+= -I$S/dev/ath
-
 # ... and the same for the NgATM stuff
 INCLUDES+= -I$S/contrib/ngatm
 
Index: sys/i386/conf/GENERIC
===================================================================
--- sys/i386/conf/GENERIC.orig
+++ sys/i386/conf/GENERIC
@@ -117,8 +117,6 @@
 device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		rr232x		# Highpoint RocketRAID 232x
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
@@ -200,7 +198,6 @@
 device		msk		# Marvell/SysKonnect Yukon II Gigabit Ethernet
 device		nfe		# nVidia nForce MCP on-board Ethernet Networking
 device		nge		# NatSemi DP83820 gigabit Ethernet
-device		nve		# nVidia nForce MCP on-board Ethernet Networking
 device		pcn		# AMD Am79C97x PCI 10/100(precedence over 'lnc')
 device		re		# RealTek 8139C+/8169/8169S/8110S
 device		rl		# RealTek 8129/8139
Index: sys/modules/Makefile
===================================================================
--- sys/modules/Makefile.orig
+++ sys/modules/Makefile
@@ -26,11 +26,6 @@
 	${_arl} \
 	${_asr} \
 	ata \
-	ath \
-	${_ath_hal} \
-	ath_rate_amrr \
-	ath_rate_onoe \
-	ath_rate_sample \
 	aue \
 	${_auxio} \
 	${_awi} \
@@ -96,7 +91,6 @@
 	${_hfa} \
 	hifn \
 	hme \
-	${_hptmv} \
 	hwpmc \
 	${_i2c} \
 	${_ibcs2} \
@@ -185,9 +179,7 @@
 	ntfs \
 	ntfs_iconv \
 	nullfs \
-	${_nve} \
 	${_nwfs} \
-	${_oltr} \
 	${_osf1} \
 	${_padlock} \
 	patm \
@@ -343,7 +335,6 @@
 _apm=		apm
 _ar=		ar
 _arcnet=	arcnet
-_ath_hal=	ath_hal
 _awi=		awi
 _bktr=		bktr
 _cardbus=	cardbus
@@ -377,7 +368,6 @@
 _ndis=		ndis
 _nsp=		nsp
 _nwfs=		nwfs
-_oltr=		oltr
 _pccard=	pccard
 _pcfclock=	pcfclock
 _pecoff=	pecoff
@@ -415,7 +405,6 @@
 _dpt=		dpt
 _el=		el
 _ex=		ex
-_hptmv=		hptmv
 _ichwd=		ichwd
 _ida=		ida
 _idt=		idt
@@ -427,14 +416,12 @@
 _ixgb=  	ixgb
 _mly=		mly
 _mxge=		mxge
-_nve=		nve
 .if !defined(NO_CRYPT) || defined(ALL_MODULES)
 .if exists(${.CURDIR}/../crypto/via)
 _padlock=	padlock
 .endif
 .endif
 _ppc=		ppc
-_rr232x=	rr232x
 _s3=		s3
 _twa=		twa
 _vesa=		vesa
@@ -461,7 +448,6 @@
 #_acpi=		acpi		# doesn't work on amd64 yet
 _agp=		agp
 _arcmsr=	arcmsr
-_ath_hal=	ath_hal
 _ciss=		ciss
 _coretemp=	coretemp
 _cpufreq=	cpufreq
@@ -469,7 +455,6 @@
 _drm=		drm
 _em=		em
 _ext2fs=	ext2fs
-_hptmv=		hptmv
 _i2c=		i2c
 _ichwd=		ichwd
 _ida=		ida
@@ -486,7 +471,6 @@
 _ndis=		ndis
 _nve=		nve
 _ppc=		ppc
-_rr232x=	rr232x
 _safe=		safe
 _scsi_low=	scsi_low
 _smbfs=		smbfs
Index: sys/amd64/conf/GENERIC
===================================================================
--- sys/amd64/conf/GENERIC.orig
+++ sys/amd64/conf/GENERIC
@@ -116,8 +116,6 @@
 device		arcmsr		# Areca SATA II RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		rr232x		# Highpoint RocketRAID 232x
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
@@ -191,7 +189,6 @@
 device		msk		# Marvell/SysKonnect Yukon II Gigabit Ethernet
 device		nfe		# nVidia nForce MCP on-board Ethernet Networking
 device		nge		# NatSemi DP83820 gigabit Ethernet
-device		nve		# nVidia nForce MCP on-board Ethernet Networking
 device		pcn		# AMD Am79C97x PCI 10/100 (precedence over 'le')
 device		re		# RealTek 8139C+/8169/8169S/8110S
 device		rl		# RealTek 8129/8139
Index: sys/conf/files.amd64
===================================================================
--- sys/conf/files.amd64.orig
+++ sys/conf/files.amd64
@@ -47,35 +47,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-hal.o				optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/x86_64-elf.hal.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/ath/public/x86_64-elf.hal.o.uu" \
-	no-implicit-rule
-opt_ah.h			optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/x86_64-elf.opt_ah.h"	\
-	compile-with	"cp $S/contrib/dev/ath/public/x86_64-elf.opt_ah.h opt_ah.h" \
-	no-obj no-implicit-rule before-depend				\
-	clean		"opt_ah.h"
-#
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
- 	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/amd64-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/dev/hptmv/amd64-elf.raid.o.uu" \
-	no-implicit-rule
-rr232x_lib.o		optional	rr232x				\
-	dependency	"$S/dev/rr232x/amd64-elf.rr232x_lib.o.uu"	\
-	compile-with	"uudecode < $S/dev/rr232x/amd64-elf.rr232x_lib.o.uu" \
-	no-implicit-rule
 #
 amd64/acpica/OsdEnvironment.c	optional	acpi
 amd64/acpica/acpi_machdep.c	optional	acpi
@@ -172,11 +143,6 @@
 dev/fdc/fdc_acpi.c		optional	fdc
 dev/fdc/fdc_isa.c		optional	fdc isa
 dev/fdc/fdc_pccard.c		optional	fdc pccard
-dev/hptmv/entry.c		optional	hptmv
-dev/hptmv/mv.c			optional	hptmv
-dev/hptmv/gui_lib.c		optional	hptmv
-dev/hptmv/hptproc.c		optional	hptmv
-dev/hptmv/ioctl.c		optional	hptmv
 dev/hwpmc/hwpmc_amd.c		optional	hwpmc
 dev/hwpmc/hwpmc_piv.c		optional	hwpmc
 dev/hwpmc/hwpmc_x86.c		optional	hwpmc
@@ -189,10 +155,6 @@
 dev/mxge/mxge_eth_z8e.c		optional	mxge pci
 dev/mxge/mxge_ethp_z8e.c	optional	mxge pci
 net/zlib.c			optional	mxge
-dev/nve/if_nve.c		optional	nve pci
-dev/rr232x/os_bsd.c		optional	rr232x
-dev/rr232x/osm_bsd.c		optional	rr232x
-dev/rr232x/rr232x_config.c	optional	rr232x
 dev/ppc/ppc.c			optional	ppc
 dev/ppc/ppc_puc.c		optional	ppc puc
 dev/sio/sio.c			optional	sio
