
Status: Rejected in upstream (kern/77355).

--- sys/i386/i386/identcpu.c.old	2005-02-08 05:02:33.000000000 +0100
+++ sys/i386/i386/identcpu.c	2005-02-08 05:05:09.000000000 +0100
@@ -604,6 +604,7 @@
 #if defined(I486_CPU)
 	case CPUCLASS_486:
 		printf("486");
+		machine[1] = '4';
 		bzero_vector = i486_bzero;
 		break;
 #endif
@@ -614,6 +615,7 @@
 		       (intmax_t)(tsc_freq + 4999) / 1000000,
 		       (u_int)((tsc_freq + 4999) / 10000) % 100);
 		printf("586");
+		machine[1] = '5';
 		break;
 #endif
 #if defined(I686_CPU)
@@ -623,6 +625,7 @@
 		       (intmax_t)(tsc_freq + 4999) / 1000000,
 		       (u_int)((tsc_freq + 4999) / 10000) % 100);
 		printf("686");
+		machine[1] = '6';
 		break;
 #endif
 	default:
--- sys/amd64/amd64/identcpu.c~	2006-02-28 12:31:12.000000000 +0100
+++ sys/amd64/amd64/identcpu.c	2006-02-28 12:41:06.000000000 +0100
@@ -71,7 +71,7 @@
 static void print_AMD_assoc(int i);
 
 int	cpu_class;
-char machine[] = "amd64";
+char machine[] = "x86_64";
 SYSCTL_STRING(_hw, HW_MACHINE, machine, CTLFLAG_RD, 
     machine, 0, "Machine class");
 
