diff -ur sys.orig/modules/cxgb/Makefile sys/modules/cxgb/Makefile
--- sys.orig/modules/cxgb/Makefile	2007-05-30 18:42:07.000000000 +0200
+++ sys/modules/cxgb/Makefile
@@ -15,7 +15,7 @@
 .if ${MACHINE_ARCH} != "ia64"
 # ld is broken on ia64
 t3fw-4.7.0.bin: ${CXGB}/t3fw-4.7.0.bin.gz.uu
-	uudecode -p < ${CXGB}/t3fw-4.7.0.bin.gz.uu \
+	uudecode -o- < ${CXGB}/t3fw-4.7.0.bin.gz.uu \
 	| gzip -dc > ${.TARGET}
 
 FIRMWS= t3fw-4.7.0.bin:t3fw470
