--- debian/sysdeps/kfreebsd.mk~	2006-06-12 16:58:52.000000000 +0200
+++ debian/sysdeps/kfreebsd.mk	2006-06-12 16:58:52.000000000 +0200
@@ -28,6 +28,7 @@
 	ln -s $(KFREEBSD_HEADERS)/net debian/include
 	ln -s $(KFREEBSD_HEADERS)/netatalk debian/include
 	ln -s $(KFREEBSD_HEADERS)/netipx debian/include
+	ln -s $(KFREEBSD_HEADERS)/nfs debian/include
 	ln -s $(KFREEBSD_HEADERS)/osreldate.h debian/include
 	ln -s $(KFREEBSD_HEADERS)/sys debian/include
 	ln -s $(KFREEBSD_HEADERS)/vm debian/include
