Index: linuxthreads/kfreebsd/pt-sigsuspend.S
===================================================================
--- linuxthreads/kfreebsd/pt-sigsuspend.S	(revision 1659)
+++ linuxthreads/kfreebsd/pt-sigsuspend.S	(working copy)
@@ -23,3 +23,7 @@
 PSEUDO_NOERRNO(__pthread_sigsuspend, sigsuspend, 1)
 	ret_NOERRNO
 PSEUDO_END_NOERRNO(__pthread_sigsuspend)
+
+PSEUDO(__syscall_sigsuspend, sigsuspend, 1)
+	ret
+PSEUDO_END(__syscall_sigsuspend)
Index: sysdeps/kfreebsd/not-cancel.h
===================================================================
--- sysdeps/kfreebsd/not-cancel.h	(revision 1659)
+++ sysdeps/kfreebsd/not-cancel.h	(working copy)
@@ -74,3 +74,15 @@
 /* Uncancelable waitpid.  */
 # define waitpid_not_cancel(pid, stat_loc, options) \
   INLINE_SYSCALL (wait4, 4, pid, stat_loc, options, NULL)
+
+/* Uncancelable pause.  */
+# define pause_not_cancel() \
+  __pause_nocancel ()
+
+/* Uncancelable nanosleep.  */
+# define nanosleep_not_cancel(requested_time, remaining) \
+  INLINE_SYSCALL (nanosleep, 2, requested_time, remaining)
+
+/* Uncancelable sigsuspend.  */
+#define sigsuspend_not_cancel(set) \
+  INLINE_SYSCALL (sigsuspend, 1, set)
Index: sysdeps/kfreebsd/syscalls-inline.h
===================================================================
--- sysdeps/kfreebsd/syscalls-inline.h	(revision 1659)
+++ sysdeps/kfreebsd/syscalls-inline.h	(working copy)
@@ -21,9 +21,12 @@
 #define KFREEBSD_INLINE_SYSCALLS_H
 
 #include <sys/types.h>
+#define __need_sigset_t
+#include <signal.h>
 
 struct iovec;
 struct rusage;
+struct timespec;
 
 int __syscall_open(const char *path, int flags, ...);
 int __syscall_close(int fd);
@@ -35,5 +38,7 @@
 int __syscall_fcntl(int fd, int cmd, ...);
 int __syscall_fork(void);
 int __syscall_wait4(int pid, int *status, int options, struct rusage *rusage);
+int __syscall_sigsuspend (const sigset_t *set);
+int __syscall_nanosleep (const struct timespec *requested_time, struct timespec *remaining);
 
 #endif
Index: sysdeps/kfreebsd/syscalls.list
===================================================================
--- sysdeps/kfreebsd/syscalls.list	(revision 1659)
+++ sysdeps/kfreebsd/syscalls.list	(working copy)
@@ -144,6 +144,8 @@
 sys_connect	-       connect	                i:ipi	__syscall_connect
 sys_bind        -       bind                    i:ipi   __syscall_bind
 sys_sendto      -       sendto          	i:ibnibn __syscall_sendto
+sys_nanosleep	EXTRA	nanosleep		i:pp	__syscall_nanosleep
+sys_sigsuspend	EXTRA	sigsuspend		i:p	__syscall_sigsuspend
 swapcontext	-	swapcontext		i:pp	__swapcontext	swapcontext
 swapon		-	swapon			i:s	swapon
 swapoff		-	swapoff			i:s	swapoff
