--- src.orig/gnu/usr.bin/send-pr/send-pr.sh
+++ src/gnu/usr.bin/send-pr/send-pr.sh	
@@ -277,7 +277,7 @@
 # Catch some signals. ($xs kludge needed by Sun /bin/sh)
 xs=0
 trap 'rm -f $REF $TEMP; exit $xs' 0
-trap 'echo "$COMMAND: Aborting ... saving unfinished PR into /tmp/pr.$$"; rm -f $REF ; mv $TEMP /tmp/pr.$$; xs=1; exit' 1 2 3 13 15
+trap 'echo "$COMMAND: Aborting ... saving unfinished PR into /tmp/pr.$$"; rm -f $REF ; mv $TEMP /tmp/pr.$$; xs=1; exit' SIGHUP SIGINT SIGQUIT SIGPIPE SIGTERM
 
 # If they told us to use a specific file, then do so.
 if [ -n "$IN_FILE" ]; then
