
Author: rmh
Status: in BTS

--- base-config-2.69/debian/control~	2005-06-30 20:14:59.000000000 +0200
+++ base-config-2.69/debian/control	2005-07-25 21:39:32.000000000 +0200
@@ -7,8 +7,8 @@
 Standards-Version: 3.6.1.0
 
 Package: base-config
-Architecture: all
-Depends: debconf (>= 1.3.22), apt, adduser, console-data (>= 2002.12.04dbs-16), console-tools, passwd (>= 20000902-6), bsdutils (>= 1:2.11l), debianutils (>= 1.6), gettext-base
+Architecture: any
+Depends: debconf (>= 1.3.22), apt, adduser, console-data (>= 2002.12.04dbs-16) [!kfreebsd-i386 !hurd-i386], console-tools [!kfreebsd-i386 !hurd-i386], passwd (>= 20000902-6), bsdutils (>= 1:2.11l), debianutils (>= 1.6), gettext-base
 Conflicts: tasksel (<< 2.25)
 Description: Debian base system configurator
  This package handles setting up the Debian base system. It contains the
