diff -ur dpkg-1.10.27.old/archtable dpkg-1.10.27/archtable
--- dpkg-1.10.27.old/archtable	2005-02-10 16:24:17.000000000 +0100
+++ dpkg-1.10.27/archtable	2005-02-11 15:31:31.000000000 +0100
@@ -52,6 +52,9 @@
 hppa64-linux-gnu		hppa		hppa
 i386-freebsd			freebsd-i386	freebsd-i386
 i386-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i486-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i586-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i686-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
 i386-netbsdelf-gnu		netbsd-i386	netbsd-i386
 i386-knetbsd-gnu		knetbsd-i386	knetbsd-i386
 s390-linux-gnu			s390		s390
diff -ur dpkg-1.10.27.old/scripts/dpkg-gencontrol.pl dpkg-1.10.27/scripts/dpkg-gencontrol.pl
--- dpkg-1.10.27.old/scripts/dpkg-gencontrol.pl	2005-02-10 16:24:18.000000000 +0100
+++ dpkg-1.10.27/scripts/dpkg-gencontrol.pl	2005-02-11 15:31:01.000000000 +0100
@@ -242,7 +242,7 @@
     binmode(X);
     while (<X>) {
         chomp;
-        next if m/^([-+0-9a-z.]+)_[^_]+_(\w+)\.deb /
+        next if m/^([-+0-9a-z.]+)_[^_]+_([-\w]+)\.deb /
                 && ($1 eq $oppackage) && ($2 eq $arch || $2 eq 'all');
         print(Y "$_\n") || &syserr("copy old entry to new files list file");
     }
diff -ur dpkg-1.10.27.old/utils/start-stop-daemon.c dpkg-1.10.27/utils/start-stop-daemon.c
--- dpkg-1.10.27.old/utils/start-stop-daemon.c	2005-02-10 16:24:18.000000000 +0100
+++ dpkg-1.10.27/utils/start-stop-daemon.c	2005-02-11 15:31:01.000000000 +0100
@@ -22,7 +22,7 @@
 
 #include "config.h"
 
-#if defined(linux)
+#if defined(linux) || (defined(__FreeBSD_kernel__) && defined(__GLIBC__))
 #  define OSLinux
 #elif defined(__GNU__)
 #  define OSHURD
