
Status:  first hunk in BTS, second hunk is dirty

diff -ur exim4-4.50.old/OS/os.c-Linux exim4-4.50/OS/os.c-Linux
--- exim4-4.50.old/OS/os.c-Linux	2005-02-17 15:49:11.000000000 +0100
+++ exim4-4.50/OS/os.c-Linux	2005-03-22 23:47:51.000000000 +0100
@@ -20,7 +20,7 @@
 provide the function. However, when compiling os.c for utilities, we may not
 want this at all, so check that it isn't set first. */
 
-#ifndef OS_LOAD_AVERAGE
+#if !defined(OS_LOAD_AVERAGE) && defined(__linux__)
 #define OS_LOAD_AVERAGE
 
 /* Linux has 2 ways of returning load average:
diff -ur exim4-4.50.old/debian/patches/36_pcre.dpatch exim4-4.50/debian/patches/36_pcre.dpatch
--- exim4-4.50.old/debian/patches/36_pcre.dpatch	2005-03-22 23:15:58.000000000 +0100
+++ exim4-4.50/debian/patches/36_pcre.dpatch	2005-03-22 23:28:34.000000000 +0100
@@ -61,7 +61,7 @@
  	  $(CFLAGS) $(XINCLUDE) -I. ../exim_monitor/em_version.c
  	$(PURIFY) $(LNCC) -o eximon.bin em_version.o $(LFLAGS) $(XLFLAGS) \
 -	$(OBJ_MONBIN) -lXaw -lXmu -lXt -lXext -lX11 pcre/libpcre.a \
-+	$(OBJ_MONBIN) -lXaw -lXmu -lXt -lXext -lX11 -lpcre \
++	$(OBJ_MONBIN) -lXaw -lXmu -lXt -lXext -lX11 -lpcre -lXpm -lSM -lICE \
  	  $(LIBS) $(LIBS_EXIMON) $(EXTRALIBS) $(EXTRALIBS_EXIMON) -lc
  	@if [ x"$(STRIP_COMMAND)" != x"" ]; then \
  	  echo $(STRIP_COMMAND) eximon.bin; \
