
Author: Brian M. Carlson
Status: in BTS

--- ifupdown.nw.orig	2005-11-19 04:23:38 +0000
+++ ifupdown.nw	2005-11-19 04:27:52 +0000
@@ -3789,7 +3789,7 @@
 #include "archlinux.h"
 
 unsigned int mylinuxver() {
-	static int maj = -1, rev, min;
+	static int maj = -1, rev = 0, min = 0;
 
 	if (maj == -1) {
 		struct utsname u;
@@ -3797,9 +3797,13 @@
 		uname(&u);
 		maj = atoi(u.release);
 		pch = strchr(u.release, '.');
-		rev = atoi(pch+1);
-		pch = strchr(pch+1, '.');
-		min = atoi(pch+1);
+		if (pch) {
+			rev = atoi(pch+1);
+			pch = strchr(pch+1, '.');
+			if (pch) {
+				min = atoi(pch+1);
+			}
+		}
 	}
 
 	return mylinux(maj,rev,min);
