
Author: rmh
Status: in BTS

diff -ur netbase-4.22.old/debian/netbase.init netbase-4.22/debian/netbase.init
--- netbase-4.22.old/debian/netbase.init	2005-09-24 18:26:36.000000000 +0200
+++ netbase-4.22/debian/netbase.init	2005-09-25 12:33:27.000000000 +0200
@@ -60,9 +60,11 @@
 process_options() {
     [ -e /etc/network/options ] || return 0
     log_warning_msg "/etc/network/options is deprecated."
-    doopt spoofprotect yes
-    doopt syncookies no
-    doopt ip_forward no
+    if [ "`uname -s`" = "Linux" ] ; then
+        doopt spoofprotect yes
+        doopt syncookies no
+        doopt ip_forward no
+    fi
 }
 
 case "$1" in
