
Author: rmh
Status: In BTS.  We also need #333141.

--- pam-0.79.old/debian/control	2006-02-11 12:09:34.000000000 +0100
+++ pam-0.79/debian/control	2006-02-11 12:13:49.000000000 +0100
@@ -4,7 +4,7 @@
 Uploaders: Karl Ramm <kcr@debian.org>, Steve Langasek <vorlon@debian.org>
 Maintainer: Sam Hartman <hartmans@debian.org>
 Standards-Version: 3.5.8
-Build-Depends: cracklib2-dev (>= 2.7-9), bzip2, debhelper, patch, libdb4.3-dev, libcap-dev [!hurd-i386 !freebsd-i386 !netbsd-i386], libselinux1-dev [!hurd-i386 !freebsd-i386 !netbsd-i386]
+Build-Depends: cracklib2-dev (>= 2.7-9), bzip2, debhelper, patch, libdb4.3-dev, libcap-dev [!hurd-i386 !kfreebsd-i386 !kfreebsd-amd64 !netbsd-i386], libselinux1-dev [!hurd-i386 !kfreebsd-i386 !kfreebsd-amd64 !netbsd-i386]
 Build-Depends-Indep: linuxdoc-tools, linuxdoc-tools-latex,  tetex-extra, groff, opensp
 
 Package: libpam0g
