
Author: rmh
Status: in BTS

diff -Nur parted-1.6.25.1.old/debian/control parted-1.6.25.1/debian/control
--- parted-1.6.25.1.old/debian/control	2006-04-18 20:42:06.000000000 +0200
+++ parted-1.6.25.1/debian/control	2006-04-18 20:42:45.000000000 +0200
@@ -4,7 +4,7 @@
 Maintainer: Parted Maintainer Team <parted-maintainers@lists.alioth.debian.org>
 Uploaders: Sven Luther <luther@debian.org>, Otavio Salvador <otavio@debian.org>
 Standards-Version: 3.6.2
-Build-Depends: debhelper (>= 4.2.0), dpatch, libncurses-dev | libncurses5-dev, libreadline5-dev | libreadline-dev, uuid-dev, gettext, texinfo (>= 4.2), debianutils (>= 1.13.1), autoconf, automake1.8, libtool
+Build-Depends: debhelper (>= 4.2.0), dpatch, libncurses-dev | libncurses5-dev, libreadline5-dev | libreadline-dev, uuid-dev, gettext, texinfo (>= 4.2), debianutils (>= 1.13.1), autoconf, automake1.8, libtool, libgeom-dev [kfreebsd-i386 kfreebsd-amd64]
 
 Package: parted
 Architecture: any
diff -Nur parted-1.6.25.1.old/debian/control.in parted-1.6.25.1/debian/control.in
--- parted-1.6.25.1.old/debian/control.in	2006-04-18 20:42:06.000000000 +0200
+++ parted-1.6.25.1/debian/control.in	2006-04-18 20:42:45.000000000 +0200
@@ -4,7 +4,7 @@
 Maintainer: Parted Maintainer Team <parted-maintainers@lists.alioth.debian.org>
 Uploaders: Sven Luther <luther@debian.org>, Otavio Salvador <otavio@debian.org>
 Standards-Version: 3.6.2
-Build-Depends: debhelper (>= 4.2.0), dpatch, libncurses-dev | libncurses5-dev, libreadline5-dev | libreadline-dev, uuid-dev, gettext, texinfo (>= 4.2), debianutils (>= 1.13.1), autoconf, automake1.8, libtool
+Build-Depends: debhelper (>= 4.2.0), dpatch, libncurses-dev | libncurses5-dev, libreadline5-dev | libreadline-dev, uuid-dev, gettext, texinfo (>= 4.2), debianutils (>= 1.13.1), autoconf, automake1.8, libtool, libgeom-dev [kfreebsd-i386 kfreebsd-amd64]
 
 Package: parted@BINPKGVER@
 Architecture: any
