
Description: disable snmp
Status: dirty

diff -u php4-4.3.10/debian/control php4-4.3.10/debian/control
--- php4-4.3.10/debian/control
+++ php4-4.3.10/debian/control
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Adam Conrad <adconrad@0c3.net>
 Uploaders: Steve Langasek <vorlon@debian.org>, Andres Salomon <dilinger@debian.org>, Jeroen van Wolffelaar <jeroen@wolffelaar.nl>
-Build-Depends: apache-dev (>= 1.3.23), apache2-prefork-dev (>= 2.0.53-3), autoconf, automake1.4, bison, chrpath, debhelper (>= 3), flex (>= 2.5.4), freetds-dev, po-debconf, libbz2-dev (>= 1.0.0), libc-client-dev, libcurl3-dev, libdb4.2-dev, libexpat1-dev (>= 1.95.2-2.1), libfreetype6-dev, libgcrypt11-dev, libgd2-xpm-dev (>= 2.0.28-3), libjpeg62-dev, libkrb5-dev, libldap2-dev, libmcal0-dev (>= 0.6), libmhash-dev (>= 0.8.8), libmysqlclient12-dev, libncurses5-dev, libpam0g-dev, libpcre3-dev (>= 4.3-1), libpng12-dev, librecode-dev, libsablot0-dev (>= 0.96), libsnmp5-dev, libssl-dev (>= 0.9.6), libt1-dev, libtool (>= 1.4.2-4), libwrap0-dev, libxmltok1-dev, libxml2-dev (>= 2.4.14), libxslt1-dev (>= 1.0.18), libzzip-dev, re2c, unixodbc-dev, zlib1g-dev (>= 1.0.9)
+Build-Depends: apache-dev (>= 1.3.23), apache2-prefork-dev (>= 2.0.53-3), autoconf, automake1.4, bison, chrpath, debhelper (>= 3), flex (>= 2.5.4), freetds-dev, po-debconf, libbz2-dev (>= 1.0.0), libc-client-dev, libcurl3-dev, libdb4.2-dev, libexpat1-dev (>= 1.95.2-2.1), libfreetype6-dev, libgcrypt11-dev, libgd2-xpm-dev (>= 2.0.28-3), libjpeg62-dev, libkrb5-dev, libldap2-dev, libmcal0-dev (>= 0.6), libmhash-dev (>= 0.8.8), libmysqlclient12-dev, libncurses5-dev, libpam0g-dev, libpcre3-dev (>= 4.3-1), libpng12-dev, librecode-dev, libsablot0-dev (>= 0.96), libssl-dev (>= 0.9.6), libt1-dev, libtool (>= 1.4.2-4), libwrap0-dev, libxmltok1-dev, libxml2-dev (>= 2.4.14), libxslt1-dev (>= 1.0.18), libzzip-dev, re2c, unixodbc-dev, zlib1g-dev (>= 1.0.9)
 Build-Conflicts: bind-dev
 Standards-Version: 3.6.1
 
@@ -196,18 +196,6 @@
  in. The goal of the language is to allow web developers to write
  dynamically generated pages quickly.
 
-Package: php4-snmp
-Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, ${php:Depends}, php4-common (= ${Source-Version})
-Replaces: php4-cgi-snmp
-Description: SNMP module for php4
- This package provides a module for SNMP functions in PHP scripts.
- .
- PHP4 is an HTML-embedded scripting language. Much of its syntax is borrowed
- from C, Java and Perl with a couple of unique PHP-specific features thrown
- in. The goal of the language is to allow web developers to write
- dynamically generated pages quickly.
-
 Package: php4-sybase
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, ${php:Depends}, php4-common (= ${Source-Version})
diff -u php4-4.3.10/debian/rules php4-4.3.10/debian/rules
--- php4-4.3.10/debian/rules
+++ php4-4.3.10/debian/rules
@@ -181,7 +181,6 @@
 		--with-unixODBC=shared,/usr \
 		--with-recode=shared,/usr \
 		--enable-xslt=shared --with-xslt-sablot=shared,/usr \
-		--with-snmp=shared,/usr \
 		--with-sybase-ct=shared,/usr \
 		--with-ttf=shared,/usr --with-t1lib=shared,/usr
 	cd apache-build && \
