 in BTS #365211
 config.{guess,sub} update also needed

--- debian/rules	2006-04-28 16:28:55.000000000 +0200
+++ debian/rules	2006-04-28 16:44:03.000000000 +0200
@@ -53,11 +53,18 @@
 	with_pthreads = --enable-storeio=ufs,diskd,null
 	with_netfilter =
 	with_arp_acl =
+else
+ifeq ($(DEB_HOST_ARCH_OS), kfreebsd)
+	with_pthreads = --enable-async-io --with-pthreads \
+		--enable-storeio=ufs,aufs,diskd,null
+	with_netfilter =
+	with_arp_acl =
 else
 	with_pthreads = --enable-async-io --with-pthreads \
 		--enable-storeio=ufs,aufs,diskd,null
 	with_netfilter = --enable-linux-netfilter
 	with_arp_acl = --enable-arp-acl
+endif
 endif
 
 # setresuid is broken on sparc
