
Debian: in BTS
Upstream: Fixed in CVS as of 2006-06-03

diff -ur -x .pc -x configure -x config.guess -x config.sub -x stampdir xorg-server-1.0.2.old/configure.ac xorg-server-1.0.2/configure.ac
--- xorg-server-1.0.2.old/configure.ac	2006-04-19 15:48:17.000000000 +0200
+++ xorg-server-1.0.2/configure.ac	2006-04-19 14:53:03.000000000 +0200
@@ -224,7 +224,10 @@
 dnl it would be nice to autodetect these *CONS_SUPPORTs
 case $host_os in
   *freebsd*)
-	AC_DEFINE(CSRG_BASED, 1, [System is BSD-like])
+	case $host_os in
+		kfreebsd*-gnu) ;;
+		*) AC_DEFINE(CSRG_BASED, 1, [System is BSD-like]) ;;
+	esac
 	AC_DEFINE(PCCONS_SUPPORT, 1, [System has PC console])
 	AC_DEFINE(PCVT_SUPPORT, 1, [System has PCVT console])
 	AC_DEFINE(SYSCONS_SUPPORT, 1, [System has syscons console])
@@ -928,7 +931,7 @@
 			;;
 		esac
 		;;
-	  freebsd*)
+	  freebsd* | kfreebsd*-gnu)
 	  	XORG_OS="freebsd"
 		XORG_OS_SUBDIR="bsd"
 		XORG_OS_KBD="BSD"
@@ -1026,7 +1029,7 @@
 		;;
 	  powerpc*)
 		case $host_os in
-		  linux*|freebsd*|netbsd*|openbsd*)
+		  linux*|freebsd*|netbsd*|openbsd*|kfreebsd*-gnu)
 			;;
 		  *)
 			xorg_bus_ppcpci="yes"
@@ -1039,7 +1042,7 @@
 		;;
 	  x86_64*|amd64*)
 		case $host_os in
-		  freebsd*)
+		  freebsd*|kfreebsd*-gnu)
 			# FreeBSD uses the system pci interface
 			;;
 		  *)
@@ -1264,7 +1267,7 @@
 	cygwin*) ;;
 	solaris*) ;;
         darwin*) ;;
-	*bsd*) ;;
+	freebsd*|netbsd*|openbsd*) ;;
 	*) 
 		AC_DEFINE(_POSIX_SOURCE, 1, [POSIX-compliant source])
 		AC_DEFINE(_XOPEN_SOURCE, 500, [X/Open-compliant source])
--- xorg-server-1.0.2/hw/xfree86/os-support/bus/Pci.h.old	2006-04-19 16:14:19.000000000 +0200
+++ xorg-server-1.0.2/hw/xfree86/os-support/bus/Pci.h	2006-04-19 16:14:30.000000000 +0200
@@ -334,7 +334,7 @@
 #  define ARCH_PCI_PCI_BRIDGE sparcPciPciBridge
 # endif
 #elif defined(__amd64__) || defined(__amd64)
-# if defined(__FreeBSD__)
+# if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #  define ARCH_PCI_INIT freebsdPciInit
 # else
 #  define ARCH_PCI_INIT ix86PciInit
