--- glibc-2.3/elf/dl-load.c	2002-09-12 22:08:11.000000000 +0000
+++ glibc-2.3.new/elf/dl-load.c	2003-05-10 16:00:59.000000000 +0000
@@ -1353,6 +1353,8 @@
 	  lose (errval, fd, name, NULL, NULL, errstring);
 	}
 
+/* (c) ugly hack by Robert Millan (shoot me) */
+#if 0
       /* See whether the ELF header is what we expect.  */
       if (__builtin_expect (! VALID_ELF_HEADER (ehdr->e_ident, expected,
 						EI_PAD), 0))
@@ -1399,6 +1401,7 @@
 
 	  goto call_lose;
 	}
+#endif /* if 0 */
 
       if (__builtin_expect (ehdr->e_version, EV_CURRENT) != EV_CURRENT)
 	{
