--- glibc-2.3/include/features.h~	2002-01-03 01:42:44.000000000 +0100
+++ glibc-2.3/include/features.h	2004-01-16 19:42:38.000000000 +0100
@@ -257,7 +257,9 @@
 
 /* Major and minor version number of the GNU C library package.  Use
    these macros to test for features in specific releases.  */
+#ifndef __GLIBC__
 #define	__GLIBC__	2
+#endif
 #define	__GLIBC_MINOR__	3
 
 /* Convenience macros to test the versions of glibc and gcc.
