--- glibc-2.3/sysdeps/generic/bits/libc-tsd.h~	2004-01-17 19:36:56.000000000 +0100
+++ glibc-2.3/sysdeps/generic/bits/libc-tsd.h	2004-01-17 19:38:58.000000000 +0100
@@ -60,7 +60,7 @@
 #else
 # define __libc_tsd_define(CLASS, KEY)	CLASS void *__libc_tsd_##KEY##_data;
 
-# define __libc_tsd_address(KEY)	(&__libc_tsd_##KEY)
+# define __libc_tsd_address(KEY)	(&__libc_tsd_##KEY##_data)
 # define __libc_tsd_get(KEY)		(__libc_tsd_##KEY##_data)
 # define __libc_tsd_set(KEY, VALUE)	(__libc_tsd_##KEY##_data = (VALUE))
 #endif
