--- glibc-2.3/shlib-versions~	2004-01-18 20:29:25.000000000 +0100
+++ glibc-2.3/shlib-versions	2004-01-18 20:29:39.000000000 +0100
@@ -51,7 +51,7 @@
 sparc64-.*-linux.*	libc=6			GLIBC_2.2
 hppa.*-.*-.*		libc=6			GLIBC_2.2
 .*-.*-linux.*		libc=6
-.*-.*-kfreebsd.*-gnu	libc=1
+.*-.*-kfreebsd.*-gnu	libc=0.1
 
 # libmachuser.so.1 corresponds to mach/*.defs as of Utah's UK22 release.
 .*-.*-gnu-gnu.*		libmachuser=1
