--- glibc-2.3/resolv/res_send.c~	2004-04-04 23:37:49.000000000 +0200
+++ glibc-2.3/resolv/res_send.c	2004-04-04 23:42:16.000000000 +0200
@@ -631,32 +631,40 @@
 				act = (*statp->rhook)(nsap, buf, buflen,
 						      ans, anssiz, &resplen);
 #endif
+				fprintf (stderr, "rmh:1\n");
 				switch (act) {
 				case res_goahead:
 				case res_done:
 					done = 1;
+					fprintf (stderr, "rmh:2\n");
 					break;
 				case res_nextns:
+					fprintf (stderr, "rmh:3\n");
 					res_nclose(statp);
 					goto next_ns;
 				case res_modified:
+					fprintf (stderr, "rmh:4\n");
 					/* give the hook another try */
 					if (++loops < 42) /*doug adams*/
 						break;
 					/*FALLTHROUGH*/
 				case res_error:
+					fprintf (stderr, "rmh:5\n");
 					/*FALLTHROUGH*/
 				default:
+					fprintf (stderr, "rmh:6\n");
 					return (-1);
 				}
 			} while (!done);
 
 		}
+		fprintf (stderr, "rmh:yeah\n");
 		return (resplen);
  next_ns: ;
 	   } /*foreach ns*/
 	} /*foreach retry*/
 	res_nclose(statp);
+	fprintf (stderr, "rmh:imfucked\n");
 	if (!v_circuit) {
 		if (!gotsomewhere)
 			__set_errno (ECONNREFUSED);	/* no nameservers found */
