diff -Nur glibc-2.3.old/linuxthreads/Makefile glibc-2.3/linuxthreads/Makefile
--- glibc-2.3.old/linuxthreads/Makefile	2002-09-09 07:18:20.000000000 +0200
+++ glibc-2.3/linuxthreads/Makefile	2004-09-01 14:48:46.000000000 +0200
@@ -58,7 +58,7 @@
 CFLAGS-pt-initfini.s = -g0 -fPIC -fno-inline-functions
 endif
 
-librt-tests = ex10 ex11
+librt-tests = # ex10 ex11
 tests = ex1 ex2 ex3 ex4 ex5 ex6 ex7 ex8 ex9 $(librt-tests) ex12 ex13 joinrace \
 	tststack $(tests-nodelete-$(have-z-nodelete)) ecmutex ex14 ex15 ex16 \
 	ex17 ex18 tst-cancel tst-context bug-sleep
diff -Nur glibc-2.3.old/linuxthreads/condvar.c glibc-2.3/linuxthreads/condvar.c
--- glibc-2.3.old/linuxthreads/condvar.c	2001-04-10 18:37:53.000000000 +0200
+++ glibc-2.3/linuxthreads/condvar.c	2004-09-01 14:48:46.000000000 +0200
@@ -96,7 +96,9 @@
     __pthread_do_exit(PTHREAD_CANCELED, CURRENT_STACK_FRAME);
   }
 
+  //__pthread_message("pcw before mutex unlock\n");
   pthread_mutex_unlock(mutex);
+  //__pthread_message("pcw after mutex unlock\n");
 
   spurious_wakeup_count = 0;
   while (1)
@@ -107,6 +109,7 @@
 	      || THREAD_GETMEM(self, p_cancelstate) != PTHREAD_CANCEL_ENABLE))
 	{
 	  /* Count resumes that don't belong to us. */
+	  //__pthread_message("pcw ignoring spurious wakeup\n");
 	  spurious_wakeup_count++;
 	  continue;
 	}
@@ -171,7 +174,10 @@
     __pthread_do_exit(PTHREAD_CANCELED, CURRENT_STACK_FRAME);
   }
 
+  //__pthread_message("status (%d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
+  //__pthread_message("pctw before mutex unlock\n");
   pthread_mutex_unlock(mutex);
+  //__pthread_message("pctw after mutex unlock\n");
 
   spurious_wakeup_count = 0;
   while (1)
@@ -201,6 +207,7 @@
 	      || THREAD_GETMEM(self, p_cancelstate) != PTHREAD_CANCEL_ENABLE))
 	{
 	  /* Count resumes that don't belong to us. */
+	  //__pthread_message("pctw ignoring spurious wakeup\n");
 	  spurious_wakeup_count++;
 	  continue;
 	}
diff -Nur glibc-2.3.old/linuxthreads/lockfile.c glibc-2.3/linuxthreads/lockfile.c
--- glibc-2.3.old/linuxthreads/lockfile.c	2002-08-27 06:52:34.000000000 +0200
+++ glibc-2.3/linuxthreads/lockfile.c	2004-09-01 14:48:46.000000000 +0200
@@ -1,5 +1,5 @@
 /* lockfile - Handle locking and unlocking of stream.
-   Copyright (C) 1996, 1998, 2000 Free Software Foundation, Inc.
+   Copyright (C) 1996, 1998, 2000, 2002 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -78,22 +78,27 @@
 void
 __flockfilelist(void)
 {
+#if 0
 #ifdef USE_IN_LIBIO
   _IO_list_lock();
 #endif
+#endif
 }
 
 void
 __funlockfilelist(void)
 {
+#if 0
 #ifdef USE_IN_LIBIO
   _IO_list_unlock();
 #endif
+#endif
 }
 
 void
 __fresetlockfiles (void)
 {
+#if 0
 #ifdef USE_IN_LIBIO
   _IO_ITER i;
 
@@ -103,10 +108,17 @@
   __pthread_mutexattr_settype (&attr, PTHREAD_MUTEX_RECURSIVE_NP);
 
   for (i = _IO_iter_begin(); i != _IO_iter_end(); i = _IO_iter_next(i))
-    __pthread_mutex_init (_IO_iter_file(i)->_lock, &attr);
+    {
+      _IO_lock_t *lock = _IO_iter_file(i)->_lock;
+#if 0
+      if (lock != NULL)
+#endif
+	__pthread_mutex_init (lock, &attr);
+    }
 
   __pthread_mutexattr_destroy (&attr);
 
   _IO_list_resetlock();
 #endif
+#endif
 }
diff -Nur glibc-2.3.old/linuxthreads/pthread.c glibc-2.3/linuxthreads/pthread.c
--- glibc-2.3.old/linuxthreads/pthread.c	2002-09-29 19:45:47.000000000 +0200
+++ glibc-2.3/linuxthreads/pthread.c	2004-09-01 14:48:46.000000000 +0200
@@ -14,6 +14,7 @@
 /* GNU Library General Public License for more details.                 */
 
 /* Thread creation, initialization, and basic low-level routines */
+#define DEBUG 1
 
 #include <errno.h>
 #include <stddef.h>
@@ -30,6 +31,7 @@
 #include "internals.h"
 #include "spinlock.h"
 #include "restart.h"
+#include "pthreadsig.h"
 #include "smp.h"
 #include <ldsodefs.h>
 #include <tls.h>
@@ -261,8 +263,8 @@
 #ifndef __SIGRTMIN
 static int current_rtmin = -1;
 static int current_rtmax = -1;
-int __pthread_sig_restart = SIGUSR1;
-int __pthread_sig_cancel = SIGUSR2;
+int __pthread_sig_restart = PTHREAD_SIG_RESTART;
+int __pthread_sig_cancel = PTHREAD_SIG_CANCEL;
 int __pthread_sig_debug;
 #else
 static int current_rtmin;
@@ -273,8 +275,8 @@
 int __pthread_sig_cancel = __SIGRTMIN + 1;
 int __pthread_sig_debug = __SIGRTMIN + 2;
 #else
-int __pthread_sig_restart = SIGUSR1;
-int __pthread_sig_cancel = SIGUSR2;
+int __pthread_sig_restart = PTHREAD_SIG_RESTART;
+int __pthread_sig_cancel = PTHREAD_SIG_CANCEL;
 int __pthread_sig_debug;
 #endif
 
@@ -293,8 +295,8 @@
       current_rtmin = -1;
       current_rtmax = -1;
 # if __SIGRTMAX - __SIGRTMIN >= 3
-      __pthread_sig_restart = SIGUSR1;
-      __pthread_sig_cancel = SIGUSR2;
+      __pthread_sig_restart = PTHREAD_SIG_RESTART;
+      __pthread_sig_cancel = PTHREAD_SIG_CANCEL;
       __pthread_sig_debug = 0;
 # endif
     }
@@ -919,7 +921,11 @@
     request.req_args.exit.code = retcode;
     TEMP_FAILURE_RETRY(__libc_write(__pthread_manager_request,
 				    (char *) &request, sizeof(request)));
+#if 0 /* The manager thread is already dead at this point.  Attempting
+	 to wait on the __pthread_sig_cancel signal from it means waiting
+	 for Godot.  */
     suspend(self);
+#endif
     /* Main thread should accumulate times for thread manager and its
        children, so that timings for main thread account for all threads. */
     if (self == __pthread_main_thread)
@@ -959,6 +965,7 @@
 {
   pthread_descr self = thread_self();
   THREAD_SETMEM(self, p_signal, sig);
+  //__pthread_message("received restart signal\n");
   if (THREAD_GETMEM(self, p_signal_jmp) != NULL)
     siglongjmp(*THREAD_GETMEM(self, p_signal_jmp), 1);
 }
@@ -1129,7 +1136,7 @@
 void __pthread_wait_for_restart_signal(pthread_descr self)
 {
   sigset_t mask;
-
+  //__pthread_message("going sigsuspend\n");
   sigprocmask(SIG_SETMASK, NULL, &mask); /* Get current signal mask */
   sigdelset(&mask, __pthread_sig_restart); /* Unblock the restart signal */
   THREAD_SETMEM(self, p_signal, 0);
@@ -1150,12 +1157,18 @@
 
 void __pthread_restart_old(pthread_descr th)
 {
-  if (atomic_increment(&th->p_resume_count) == -1)
+  //__pthread_message("status (%d p_resume_count = %ld)\n",thread_self()->p_pid,thread_self()->p_resume_count.p_count);
+  if (atomic_increment(&th->p_resume_count) == -1) {
+    //__pthread_message("sending restart signal to %d\n",th->p_pid);
     kill(th->p_pid, __pthread_sig_restart);
+  } else {
+    //__pthread_message("NOT sending restart signal to %d (%d p_resume_count = %ld)\n",th->p_pid,th->p_pid,th->p_resume_count.p_count);
+  }
 }
 
 void __pthread_suspend_old(pthread_descr self)
 {
+  //__pthread_message("suspend_old (%d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
   if (atomic_decrement(&self->p_resume_count) <= 0)
     __pthread_wait_for_restart_signal(self);
 }
@@ -1167,6 +1180,7 @@
   int was_signalled = 0;
   sigjmp_buf jmpbuf;
 
+  //__pthread_message("timedsuspend_old (%d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
   if (atomic_decrement(&self->p_resume_count) == 0) {
     /* Set up a longjmp handler for the restart signal, unblock
        the signal and sleep. */
@@ -1205,8 +1219,22 @@
       was_signalled = 1;
     }
     THREAD_SETMEM(self, p_signal_jmp, NULL);
+
+    if (!was_signalled) {
+      /* woke spontaneously but did not consume restart---caller must resolve */
+      //__pthread_message("timedsuspend_old returns 0 (was_signalled = 0, %d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
+      return 0;
+    }
+    /* woken due to restart signal */
+    //__pthread_message("timedsuspend_old returns 1 (was_signalled = 1, %d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
+    return 1;
+  } else {
+    /* consumed queued restart signal */
+    //__pthread_message("timedsuspend_old returns 1 (was_signalled = 0, %d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
+    return 1;
   }
 
+#if 0
   /* Now was_signalled is true if we exited the above code
      due to the delivery of a restart signal.  In that case,
      we know we have been dequeued and resumed and that the
@@ -1228,13 +1256,17 @@
       __pthread_wait_for_restart_signal(self);
       atomic_decrement(&self->p_resume_count); /* should be zero now! */
       /* woke spontaneously and consumed restart signal */
+      //__pthread_message("timedsuspend_old returns 1 (was_signalled = 0, %d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
       return 1;
     }
     /* woke spontaneously but did not consume restart---caller must resolve */
+    //__pthread_message("timedsuspend_old returns 0 (was_signalled = 0, %d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
     return 0;
   }
   /* woken due to restart signal */
+  //__pthread_message("timedsuspend_old returns 1 (was_signalled = 1, %d p_resume_count = %ld)\n",self->p_pid,self->p_resume_count.p_count);
   return 1;
+#endif
 }
 #endif /* __ASSUME_REALTIME_SIGNALS */
 
diff -Nur glibc-2.3.old/linuxthreads/shlib-versions glibc-2.3/linuxthreads/shlib-versions
--- glibc-2.3.old/linuxthreads/shlib-versions	2002-09-05 12:14:21.000000000 +0200
+++ glibc-2.3/linuxthreads/shlib-versions	2004-09-01 14:48:46.000000000 +0200
@@ -8,4 +8,5 @@
 cris-.*-linux.*		libpthread=0		GLIBC_2.2
 x86_64-.*-linux.*	libpthread=0		GLIBC_2.2.5
 powerpc64-.*-linux.*	libpthread=0		GLIBC_2.3
-.*-.*-linux.*		libpthread=0
+.*-.*-kfreebsd.*-gnu.*	libpthread=0		GLIBC_2.3
+.*-.*-.*		libpthread=0
diff -Nur glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/Implies glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/Implies
--- glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/Implies	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/Implies	2002-07-07 20:32:13.000000000 +0200
@@ -0,0 +1 @@
+pthread
diff -Nur glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/local_lim.h glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/local_lim.h
--- glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/local_lim.h	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/local_lim.h	2002-07-09 01:20:13.000000000 +0200
@@ -0,0 +1,48 @@
+/* Minimum guaranteed maximum values for system limits.  FreeBSD version.
+   Copyright (C) 1993-1998, 2000, 2002 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+#include <sys/syslimits.h>
+
+/* The number of data keys per process.  */
+#define _POSIX_THREAD_KEYS_MAX	128
+/* This is the value this implementation supports.  */
+#define PTHREAD_KEYS_MAX	1024
+
+/* Controlling the iterations of destructors for thread-specific data.  */
+#define _POSIX_THREAD_DESTRUCTOR_ITERATIONS	4
+/* Number of iterations this implementation does.  */
+#define PTHREAD_DESTRUCTOR_ITERATIONS	_POSIX_THREAD_DESTRUCTOR_ITERATIONS
+
+/* The number of threads per process.  */
+#define _POSIX_THREAD_THREADS_MAX	64
+/* This is the value this implementation supports.  */
+#define PTHREAD_THREADS_MAX	1024
+
+/* Maximum amount by which a process can descrease its asynchronous I/O
+   priority level.  */
+#define AIO_PRIO_DELTA_MAX	20
+
+/* Minimum size for a thread.  We are free to choose a reasonable value.  */
+#define PTHREAD_STACK_MIN	16384
+
+/* Maximum number of POSIX timers available.  */
+#define TIMER_MAX	256
+
+/* Maximum number of timer expiration overruns.  */
+#define DELAYTIMER_MAX	2147483647
diff -Nur glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/posix_opt.h glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/posix_opt.h
--- glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/posix_opt.h	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/posix_opt.h	2002-09-15 20:51:42.000000000 +0200
@@ -0,0 +1,133 @@
+/* Define POSIX options for FreeBSD.
+   Copyright (C) 1996-1997, 1999, 2000, 2002 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+/*
+ * Never include this file directly; use <unistd.h> instead.
+ */
+
+#ifndef	_BITS_POSIX_OPT_H
+#define	_BITS_POSIX_OPT_H	1
+
+/* Job control is supported.  */
+#define	_POSIX_JOB_CONTROL	1
+
+/* Processes have a saved set-user-ID and a saved set-group-ID.  */
+#define	_POSIX_SAVED_IDS	1
+
+/* Priority scheduling is supported.  */
+#define	_POSIX_PRIORITY_SCHEDULING	1
+
+/* Synchronizing file data is supported.  */
+#define	_POSIX_SYNCHRONIZED_IO	1
+
+/* The fsync function is present.  */
+#define	_POSIX_FSYNC	1
+
+/* Mapping of files to memory is supported.  */
+#define	_POSIX_MAPPED_FILES	1
+
+/* Locking of all memory is supported.  */
+#define	_POSIX_MEMLOCK	1
+
+/* Locking of ranges of memory is supported.  */
+#define	_POSIX_MEMLOCK_RANGE	1
+
+/* Setting of memory protections is supported.  */
+#define	_POSIX_MEMORY_PROTECTION	1
+
+/* Implementation supports `poll' function.  */
+#define	_POSIX_POLL	1
+
+/* Implementation supports `select' and `pselect' functions.  */
+#define	_POSIX_SELECT	1
+
+/* XPG4.2 shared memory is supported.  */
+#define	_XOPEN_SHM	1
+
+/* Tell we have POSIX threads.  */
+#define _POSIX_THREADS	1
+
+/* We have the reentrant functions described in POSIX.  */
+#define _POSIX_REENTRANT_FUNCTIONS	1
+#define _POSIX_THREAD_SAFE_FUNCTIONS	1
+
+/* We provide priority scheduling for threads.  */
+#define _POSIX_THREAD_PRIORITY_SCHEDULING	1
+
+/* We support user-defined stack sizes.  */
+#define _POSIX_THREAD_ATTR_STACKSIZE	1
+
+/* We support user-defined stacks.  */
+#define _POSIX_THREAD_ATTR_STACKADDR	1
+
+/* We support POSIX.1b semaphores, but only the non-shared form for now.  */
+#define _POSIX_SEMAPHORES	1
+
+/* We support asynchronous I/O.  */
+#define _POSIX_ASYNCHRONOUS_IO	1
+#define _POSIX_ASYNC_IO		1
+/* Alternative name for Unix98.  */
+#define _LFS_ASYNCHRONOUS_IO	1
+
+/* The LFS support in asynchronous I/O is also available.  */
+#define _LFS64_ASYNCHRONOUS_IO	1
+
+/* The rest of the LFS is also available.  */
+#define _LFS_LARGEFILE		1
+#define _LFS64_LARGEFILE	1
+#define _LFS64_STDIO		1
+
+/* X/Open realtime support is available.  */
+#define _XOPEN_REALTIME	1
+
+/* X/Open realtime thread support is available.  */
+#define _XOPEN_REALTIME_THREADS	1
+
+/* Only root can change owner of file.  */
+#define	_POSIX_CHOWN_RESTRICTED	1
+
+/* `c_cc' member of 'struct termios' structure can be disabled by
+   using the value _POSIX_VDISABLE.  */
+#define	_POSIX_VDISABLE	'\0'
+
+/* POSIX timers are available.  */
+#define _POSIX_TIMERS 1
+
+/* Reader/Writer locks are available.  */
+#define _POSIX_READER_WRITER_LOCKS	200912L
+
+/* GNU libc provides regular expression handling.  */
+#define _POSIX_REGEXP	1
+
+/* We have a POSIX shell.  */
+#define _POSIX_SHELL	1
+
+/* We support the Timeouts option.  */
+#define _POSIX_TIMEOUTS	200912L
+
+/* We support spinlocks.  */
+#define _POSIX_SPIN_LOCKS	200912L
+
+/* The `spawn' function family is supported.  */
+#define _POSIX_SPAWN	200912L
+
+/* The barrier functions are available.  */
+#define _POSIX_BARRIERS	200912L
+
+#endif /* bits/posix_opt.h */
diff -Nur glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/sigthread.h glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/sigthread.h
--- glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/sigthread.h	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/sigthread.h	2002-07-12 12:03:22.000000000 +0200
@@ -0,0 +1,38 @@
+/* Signal handling function for threaded programs.
+   Copyright (C) 1998, 1999, 2000, 2002 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+#ifndef _BITS_SIGTHREAD_H
+#define _BITS_SIGTHREAD_H	1
+
+#if !defined _SIGNAL_H && !defined _PTHREAD_H
+# error "Never include this file directly.  Use <pthread.h> instead"
+#endif
+
+/* Functions for handling signals. */
+
+/* Modify the signal mask for the calling thread.  The arguments have
+   the same meaning as for sigprocmask(2). */
+extern int pthread_sigmask (int __how,
+			    __const __sigset_t *__restrict __newmask,
+			    __sigset_t *__restrict __oldmask) __THROW;
+
+/* Send signal SIGNO to the given thread. */
+extern int pthread_kill (pthread_t __threadid, int __signo) __THROW;
+
+#endif	/* bits/sigthread.h */
diff -Nur glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/i386/bits/posix_opt.h glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/i386/bits/posix_opt.h
--- glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/i386/bits/posix_opt.h	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/i386/bits/posix_opt.h	2002-09-15 20:51:47.000000000 +0200
@@ -0,0 +1,139 @@
+/* Define POSIX options for FreeBSD.
+   Copyright (C) 1996-1997, 1999, 2000, 2002 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+/*
+ * Never include this file directly; use <unistd.h> instead.
+ */
+
+#ifndef	_BITS_POSIX_OPT_H
+#define	_BITS_POSIX_OPT_H	1
+
+/* Job control is supported.  */
+#define	_POSIX_JOB_CONTROL	1
+
+/* Processes have a saved set-user-ID and a saved set-group-ID.  */
+#define	_POSIX_SAVED_IDS	1
+
+/* Priority scheduling is supported.  */
+#define	_POSIX_PRIORITY_SCHEDULING	1
+
+/* Synchronizing file data is supported.  */
+#define	_POSIX_SYNCHRONIZED_IO	1
+
+/* The fsync function is present.  */
+#define	_POSIX_FSYNC	1
+
+/* Mapping of files to memory is supported.  */
+#define	_POSIX_MAPPED_FILES	1
+
+/* Locking of all memory is supported.  */
+#define	_POSIX_MEMLOCK	1
+
+/* Locking of ranges of memory is supported.  */
+#define	_POSIX_MEMLOCK_RANGE	1
+
+/* Setting of memory protections is supported.  */
+#define	_POSIX_MEMORY_PROTECTION	1
+
+/* Implementation supports `poll' function.  */
+#define	_POSIX_POLL	1
+
+/* Implementation supports `select' and `pselect' functions.  */
+#define	_POSIX_SELECT	1
+
+/* XPG4.2 shared memory is supported.  */
+#define	_XOPEN_SHM	1
+
+/* Tell we have POSIX threads.  */
+#define _POSIX_THREADS	1
+
+/* We have the reentrant functions described in POSIX.  */
+#define _POSIX_REENTRANT_FUNCTIONS	1
+#define _POSIX_THREAD_SAFE_FUNCTIONS	1
+
+/* We provide priority scheduling for threads.  */
+#define _POSIX_THREAD_PRIORITY_SCHEDULING	1
+
+/* We support user-defined stack sizes.  */
+#define _POSIX_THREAD_ATTR_STACKSIZE	1
+
+/* We support user-defined stacks.  */
+#define _POSIX_THREAD_ATTR_STACKADDR	1
+
+/* We support POSIX.1b semaphores, but only the non-shared form for now.  */
+#define _POSIX_SEMAPHORES	1
+
+/* We support asynchronous I/O.  */
+#define _POSIX_ASYNCHRONOUS_IO	1
+#define _POSIX_ASYNC_IO		1
+/* Alternative name for Unix98.  */
+#define _LFS_ASYNCHRONOUS_IO	1
+
+/* The LFS support in asynchronous I/O is also available.  */
+#define _LFS64_ASYNCHRONOUS_IO	1
+
+/* The rest of the LFS is also available.  */
+#define _LFS_LARGEFILE		1
+#define _LFS64_LARGEFILE	1
+#define _LFS64_STDIO		1
+
+/* X/Open realtime support is available.  */
+#define _XOPEN_REALTIME	1
+
+/* X/Open realtime thread support is available.  */
+#define _XOPEN_REALTIME_THREADS	1
+
+/* Only root can change owner of file.  */
+#define	_POSIX_CHOWN_RESTRICTED	1
+
+/* `c_cc' member of 'struct termios' structure can be disabled by
+   using the value _POSIX_VDISABLE.  */
+#define	_POSIX_VDISABLE	'\0'
+
+/* POSIX timers are available.  */
+#define _POSIX_TIMERS 1
+
+/* CPU-time clocks supported.  */
+#define _POSIX_CPUTIME	200912L
+
+/* We support the clock also in threads.  */
+#define _POSIX_THREAD_CPUTIME	200912L
+
+/* Reader/Writer locks are available.  */
+#define _POSIX_READER_WRITER_LOCKS	200912L
+
+/* GNU libc provides regular expression handling.  */
+#define _POSIX_REGEXP	1
+
+/* We have a POSIX shell.  */
+#define _POSIX_SHELL	1
+
+/* We support the Timeouts option.  */
+#define _POSIX_TIMEOUTS	200912L
+
+/* We support spinlocks.  */
+#define _POSIX_SPIN_LOCKS	200912L
+
+/* The `spawn' function family is supported.  */
+#define _POSIX_SPAWN	200912L
+
+/* The barrier functions are available.  */
+#define _POSIX_BARRIERS	200912L
+
+#endif /* bits/posix_opt.h */
diff -Nur glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/pthreadsig.h glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/pthreadsig.h
--- glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/pthreadsig.h	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/pthreadsig.h	2002-07-24 00:00:42.000000000 +0200
@@ -0,0 +1,23 @@
+/* Copyright (C) 2002 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public License as
+   published by the Free Software Foundation; either version 2 of the
+   License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with the GNU C Library; see the file COPYING.LIB.  If not,
+   write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+   Boston, MA 02111-1307, USA.  */
+
+/* On FreeBSD, only SIGUSR1 can be used as the thread termination notification.
+   This is hardwired in the kernel.  */
+
+#define PTHREAD_SIG_RESTART	SIGUSR2
+#define PTHREAD_SIG_CANCEL	SIGUSR1
diff -Nur glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/smp.h glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/smp.h
--- glibc-2.3.old/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/smp.h	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.3/linuxthreads/sysdeps/unix/bsd/bsd4.4/kfreebsd/smp.h	2002-09-17 02:51:12.000000000 +0200
@@ -0,0 +1,36 @@
+/* Determine whether the host has multiple processors.  FreeBSD version.
+   Copyright (C) 1996, 2002 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public License as
+   published by the Free Software Foundation; either version 2 of the
+   License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with the GNU C Library; see the file COPYING.LIB.  If not,
+   write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+   Boston, MA 02111-1307, USA.  */
+
+#include <sys/sysctl.h>
+
+/* Test whether the machine has more than one processor.  */
+static inline int
+is_smp_system (void)
+{
+  /* Fetch sysctl value of "hw.ncpu".  */
+  int request[] = { CTL_HW, HW_NCPU };
+  int result;
+  size_t result_len = sizeof (result);
+
+  if (__sysctl (request, 2, &result, &result_len, NULL, 0) < 0)
+    /* Dummy result.  */
+    return 0;
+
+  return (result > 1);
+}
diff -Nur glibc-2.3.old/linuxthreads/sysdeps/unix/sysv/linux/pthreadsig.h glibc-2.3/linuxthreads/sysdeps/unix/sysv/linux/pthreadsig.h
--- glibc-2.3.old/linuxthreads/sysdeps/unix/sysv/linux/pthreadsig.h	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.3/linuxthreads/sysdeps/unix/sysv/linux/pthreadsig.h	2002-07-23 23:59:27.000000000 +0200
@@ -0,0 +1,20 @@
+/* Copyright (C) 2002 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public License as
+   published by the Free Software Foundation; either version 2 of the
+   License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with the GNU C Library; see the file COPYING.LIB.  If not,
+   write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+   Boston, MA 02111-1307, USA.  */
+
+#define PTHREAD_SIG_RESTART	SIGUSR1
+#define PTHREAD_SIG_CANCEL	SIGUSR2
--- glibc-2.3/linuxthreads_db/td_thr_getgregs.c~	2002-07-16 05:24:27.000000000 +0200
+++ glibc-2.3/linuxthreads_db/td_thr_getgregs.c	2004-09-17 01:19:13.000000000 +0200
@@ -31,7 +31,7 @@
   if (th->th_unique == NULL)
     {
       /* No data yet.  */
-      memset (gregs, '\0', sizeof (prgregset_t));
+      memset (& gregs, '\0', sizeof (prgregset_t));
       return TD_OK;
     }
 
@@ -42,7 +42,7 @@
 
   /* If the thread already terminated we return all zeroes.  */
   if (pds.p_terminated)
-    memset (gregs, '\0', sizeof (prgregset_t));
+    memset (& gregs, '\0', sizeof (prgregset_t));
   /* Otherwise get the register content through the callback.  */
   else
     {
