--- glibc-2.3/posix/glob.h.orig     2005-04-18 20:48:11.000000000 +0200
+++ glibc-2.3/posix/glob.h 2005-10-27 10:18:52.000000000 +0200
@@ -25,7 +25,7 @@
 
 #undef __ptr_t
 #if defined __cplusplus || (defined __STDC__ && __STDC__) || defined WINDOWS32
-# if !defined __GLIBC__ || !defined __P
+# if !defined __GLIBC__ || !defined __P || !defined __PMT
 #  undef __P
 #  undef __PMT
 #  define __P(protos)  protos

