cvs server: Diffing .
Index: ipx_addr.c
===================================================================
RCS file: /home/ncvs/src/lib/libipx/ipx_addr.c,v
retrieving revision 1.8
diff -u -r1.8 ipx_addr.c
--- ipx_addr.c	25 Aug 2002 13:06:32 -0000	1.8
+++ ipx_addr.c	4 Sep 2004 06:28:10 -0000
@@ -35,12 +35,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libipx/ipx_addr.c,v 1.8 2002/08/25 13:06:32 charnier Exp $");
-
-#if defined(LIBC_SCCS) && !defined(lint)
-static char sccsid[] = "@(#)ipx_addr.c";
-#endif /* LIBC_SCCS and not lint */
-
 #include <sys/param.h>
 #include <arpa/inet.h>
 #include <netipx/ipx.h>
Index: ipx_ntoa.c
===================================================================
RCS file: /home/ncvs/src/lib/libipx/ipx_ntoa.c,v
retrieving revision 1.5
diff -u -r1.5 ipx_ntoa.c
--- ipx_ntoa.c	21 Mar 2002 23:38:00 -0000	1.5
+++ ipx_ntoa.c	4 Sep 2004 06:28:11 -0000
@@ -32,12 +32,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libipx/ipx_ntoa.c,v 1.5 2002/03/21 23:38:00 obrien Exp $");
-
-#if defined(LIBC_SCCS) && !defined(lint)
-static char sccsid[] = "@(#)ipx_ntoa.c";
-#endif /* LIBC_SCCS and not lint */
-
 #include <sys/param.h>
 #include <arpa/inet.h>
 #include <netipx/ipx.h>
