diff -ur arts-1.2.3.old/flow/bufferqueue.h arts-1.2.3/flow/bufferqueue.h
--- arts-1.2.3.old/flow/bufferqueue.h	2002-03-08 21:30:19.000000000 +0100
+++ arts-1.2.3/flow/bufferqueue.h	2004-08-05 03:28:22.000000000 +0200
@@ -10,6 +10,10 @@
 #ifndef _BUFFERQUEUE_H
 #define _BUFFERQUEUE_H
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 #include "thread.h"
 
 #define _DEFAULT_CHUNK_SIZE 4096
@@ -67,27 +71,35 @@
 	ByteBuffer bufs[_MAX_CHUNKS];
 	int rp;
 	int wp;
+#ifdef HAVE_SEMAPHORE_H
 	Arts::Semaphore* sema_produced;
 	Arts::Semaphore* sema_consumed;
+#endif
 
 	void semaReinit() {
+#ifdef HAVE_SEMAPHORE_H
 		delete sema_consumed;
 		delete sema_produced;
 		sema_consumed = new Arts::Semaphore(0, _MAX_CHUNKS);
 		sema_produced = new Arts::Semaphore(0, 0);
+#endif
 	}
 
 
 public:
 	BufferQueue() {
+#ifdef HAVE_SEMAPHORE_H
 		rp = wp = 0;
 		sema_consumed = new Arts::Semaphore(0, _MAX_CHUNKS);
 		sema_produced = new Arts::Semaphore(0, 0);
+#endif
 	}
 
 	~BufferQueue() {
+#ifdef HAVE_SEMAPHORE_H
 		delete sema_consumed;
 		delete sema_produced;
+#endif
 	}
 
 	void write(void* data, int len);
@@ -97,9 +109,30 @@
 	ByteBuffer* waitProduced();
 	void consumed();
 
-	bool isEmpty() const       { return sema_produced->getValue() == 0; }
-	int bufferedChunks() const { return sema_produced->getValue(); }
-	int freeChunks() const     { return sema_consumed->getValue(); }
+	bool isEmpty() const
+	  {
+#ifdef HAVE_SEMAPHORE_H
+	    return sema_produced->getValue() == 0;
+#else
+	    return 0;
+#endif
+	  }
+	int bufferedChunks() const
+	  {
+#ifdef HAVE_SEMAPHORE_H
+	    return sema_produced->getValue();
+#else
+	    return 0;
+#endif
+	  }
+	int freeChunks() const
+	  {
+#ifdef HAVE_SEMAPHORE_H
+	    return sema_consumed->getValue();
+#else
+	    return 0;
+#endif
+	  }
 	int maxChunks() const      { return _MAX_CHUNKS; }
 	int chunkSize() const      { return bufs[0].maxSize(); }
 	void clear()               { rp = wp = 0; semaReinit(); }
@@ -113,34 +146,48 @@
 
 inline void BufferQueue::write(void* data, int len)
 {
+#ifdef HAVE_SEMAPHORE_H
 	sema_consumed->wait();
 	bufs[wp].put(data, len);
 	++wp %= _MAX_CHUNKS;
 	sema_produced->post();
+#endif
 }
 
 inline ByteBuffer* BufferQueue::waitConsumed()
 {
+#ifdef HAVE_SEMAPHORE_H
 	sema_consumed->wait();
 	return &bufs[wp];
+#else
+	return 0;
+#endif
 }
 
 inline void BufferQueue::produced()
 {
+#ifdef HAVE_SEMAPHORE_H
 	++wp %= _MAX_CHUNKS;
 	sema_produced->post();
+#endif
 }
 
 inline ByteBuffer* BufferQueue::waitProduced()
 {
+#ifdef HAVE_SEMAPHORE_H
 	sema_produced->wait();
 	return &bufs[rp];
+#else
+	return 0;
+#endif
 }
 
 inline void BufferQueue::consumed()
 {
+#ifdef HAVE_SEMAPHORE_H
 	++rp %=_MAX_CHUNKS;
 	sema_consumed->post();
+#endif
 }
 
 }
diff -ur arts-1.2.3.old/mcop/thread.h arts-1.2.3/mcop/thread.h
--- arts-1.2.3.old/mcop/thread.h	2003-10-13 21:59:41.000000000 +0200
+++ arts-1.2.3/mcop/thread.h	2004-08-05 03:28:22.000000000 +0200
@@ -23,6 +23,10 @@
 #ifndef ARTS_MCOP_THREAD_H
 #define ARTS_MCOP_THREAD_H
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 /*
  * BC - Status (2002-03-08): SystemThreads, Thread, Mutex, ThreadCondition,
  * Semaphore
@@ -78,7 +82,9 @@
 	virtual Mutex_impl *createRecMutex_impl() = 0;
 	virtual Thread_impl *createThread_impl(Thread *thread) = 0;
 	virtual ThreadCondition_impl *createThreadCondition_impl() = 0;
+#ifdef HAVE_SEMAPHORE_H
 	virtual Semaphore_impl *createSemaphore_impl(int, int) = 0;
+#endif
 	virtual ~SystemThreads();
 
 	/**
@@ -335,6 +341,7 @@
 	}
 };
 
+#ifdef HAVE_SEMAPHORE_H
 class Semaphore {
 private:
 	Semaphore_impl *impl;
@@ -367,6 +374,7 @@
 		return impl->getValue();
 	}
 };
+#endif /* HAVE_SEMAPHORE_H */
 
 }
 
