
Status: in BTS

--- d-shlibs-0.23/d-devlibdeps	2004-08-16 17:24:42.000000000 +0200
+++ d-shlibs-0.23+kbsd/d-devlibdeps	2004-08-27 00:06:16.000000000 +0200
@@ -51,9 +51,11 @@
 	-e 's/libdl2.1-dev//' \
 	-e 's/libm6-dev//' \
 	-e 's/libm6.1-dev//' \
+	-e 's/libm1-dev//' \
 	-e 's/libnsl1-dev//' \
 	-e 's/libnsl1\.1-dev//' \
 	-e 's/libpthread0-dev//' \
+	-e 's/libpthread.*-dev/libpthread-dev/' \
 	-e 's/libglib-1.2-0-dev/libglib1.2-dev/' \
 	-e 's/libgthread-1.2-0-dev/libglib1.2-dev/' \
 	-e 's/libgmodule-1.2-0-dev/libglib1.2-dev/' \
