diff -ur debootstrap-0.2.41.old/Makefile debootstrap-0.2.41/Makefile
--- debootstrap-0.2.41.old/Makefile	2004-07-29 20:37:29.000000000 +0200
+++ debootstrap-0.2.41/Makefile	2004-08-09 02:23:21.000000000 +0200
@@ -13,7 +13,7 @@
         $(call setarchdevs,powerpc,hdc hdd fd0 fd1 isdn-io m68k-mice) \
         $(call setarchdevs,ia64,ida fd0 fd1 ataraid cciss)
 
-all: pkgdetails devices.tar.gz debootstrap-arch
+all: pkgdetails debootstrap-arch
 clean:
 	rm -f pkgdetails pkgdetails.o devices.tar.gz debootstrap-arch
 	rm -rf dev
@@ -33,7 +33,7 @@
 	install -o root -g root -m 0644 sid.buildd $(DSDIR)/scripts/
 	install -o root -g root -m 0644 functions $(DSDIR)/
 	install -o root -g root -m 0755 pkgdetails $(DSDIR)/
-	install -o root -g root -m 0644 devices.tar.gz $(DSDIR)/
+	-install -o root -g root -m 0644 devices.tar.gz $(DSDIR)/
 	install -o root -g root -m 0644 debootstrap-arch $(DSDIR)/arch
 	install -o root -g root -m 0755 debootstrap.8 $(DESTDIR)/usr/share/man/man8/
 	install -o root -g root -m 0755 debootstrap $(DESTDIR)/usr/sbin/
diff -ur debootstrap-0.2.41.old/debian/changelog debootstrap-0.2.41/debian/changelog
--- debootstrap-0.2.41.old/debian/changelog	2004-07-31 06:30:05.000000000 +0200
+++ debootstrap-0.2.41/debian/changelog	2004-08-09 02:21:34.000000000 +0200
@@ -1,3 +1,9 @@
+debootstrap (0.2.41-0.2+kbsd) unreleased; urgency=low
+
+  * foo
+
+ -- Robert Millan <rmh@debian.org>  Mon,  9 Aug 2004 02:20:38 +0200
+
 debootstrap (0.2.41-0.2) unstable; urgency=low
 
   * Non-maintainer upload
diff -ur debootstrap-0.2.41.old/debian/control debootstrap-0.2.41/debian/control
--- debootstrap-0.2.41.old/debian/control	2004-05-03 23:15:38.000000000 +0200
+++ debootstrap-0.2.41/debian/control	2004-08-09 02:38:12.000000000 +0200
@@ -3,7 +3,7 @@
 Priority: required
 Maintainer: Anthony Towns <ajt@debian.org>
 Uploaders: J.H.M. Dassen (Ray) <jdassen@debian.org>
-Build-Depends: debhelper (>> 2.0.0), makedev (>= 2.3.1-69)
+Build-Depends: debhelper (>> 2.0.0), makedev (>= 2.3.1-69) [alpha arm hppa i386 ia64 m68k mips mipsel powerpc s390 s390x sh3 sh3eb sh4 sh4eb sparc sparc64 amd64], type-handling (>= 0.2.1), dpkg-dev
 Standards-Version: 3.6.1
 
 Package: debootstrap
diff -ur debootstrap-0.2.41.old/debian/control.in debootstrap-0.2.41/debian/control.in
--- debootstrap-0.2.41.old/debian/control.in	2004-08-09 01:45:29.000000000 +0200
+++ debootstrap-0.2.41/debian/control.in	2004-08-09 01:58:22.000000000 +0200
@@ -3,7 +3,7 @@
 Priority: required
 Maintainer: Anthony Towns <ajt@debian.org>
 Uploaders: J.H.M. Dassen (Ray) <jdassen@debian.org>
-Build-Depends: debhelper (>> 2.0.0), makedev (>= 2.3.1-69)
+Build-Depends: debhelper (>> 2.0.0), makedev (>= 2.3.1-69) [@makedev@], type-handling (>= 0.2.1), dpkg-dev
 Standards-Version: 3.6.1
 
 Package: debootstrap
diff -ur debootstrap-0.2.41.old/debian/rules debootstrap-0.2.41/debian/rules
--- debootstrap-0.2.41.old/debian/rules	2004-03-06 11:54:28.000000000 +0100
+++ debootstrap-0.2.41/debian/rules	2004-08-09 01:56:58.000000000 +0200
@@ -20,6 +20,10 @@
 
 	-$(MAKE) clean
 
+	cat debian/control.in \
+	| sed "s/@makedev@/`type-handling any linux-gnu`/g" \
+	> debian/control
+
 	dh_clean
 
 build: configure
diff -ur debootstrap-0.2.41.old/sid debootstrap-0.2.41/sid
--- debootstrap-0.2.41.old/sid	2004-07-31 05:53:03.000000000 +0200
+++ debootstrap-0.2.41/sid	2004-08-09 02:20:13.000000000 +0200
@@ -3,10 +3,8 @@
 
 work_out_debs () {
 
-    required="base-files base-passwd bash bsdutils coreutils libacl1 libattr1 debconf debconf-i18n liblocale-gettext-perl libtext-wrapi18n-perl libtext-charwidth-perl debianutils diff dpkg dselect libblkid1 e2fsprogs e2fslibs libcomerr2 libss2 libuuid1 findutils grep gzip hostname libcap1 libc6 libdb1-compat libdb3 libncurses5 libnewt0.51 libpam-modules libpam-runtime libpam0g libperl5.8 libpopt0 login makedev mawk modutils mount ncurses-base ncurses-bin passwd perl-base procps sed slang1a-utf8 initscripts sysvinit sysv-rc tar util-linux whiptail libgcc1 gcc-3.3-base libstdc++5"
-
-    base="adduser apt apt-utils libdb4.2 at base-config aptitude libsigc++-1.2-5c102 bsdmainutils console-common console-tools libconsole console-data cpio cron dhcp-client ed exim4 exim4-base exim4-config exim4-daemon-light libgnutls11 libgcrypt11 libgnutls10 libgcrypt7 libgpg-error0 libopencdk8 libtasn1-2 fdutils gettext-base groff-base ifupdown info klogd libssl0.9.7 liblzo1 zlib1g liblockfile1 libpcre3 libwrap0 logrotate mailx man-db libgdbm3 manpages nano net-tools netbase netkit-inetd iputils-ping nvi ppp pppconfig pppoe pppoeconf libpcap0.7 sysklogd tasksel libtextwrap1 tcpd telnet libtext-iconv-perl wget $additional" 
-
+    required="base-files base-passwd bash bsdutils coreutils debconf debconf-i18n liblocale-gettext-perl libtext-wrapi18n-perl libtext-charwidth-perl debianutils diff dpkg dselect libblkid1 e2fsprogs e2fslibs libcomerr2 libss2 libuuid1 findutils grep gzip hostname libdb1-compat libdb3 libncurses5 libnewt0.51 libpam-modules libpam-runtime libpam0g libperl5.8 libpopt0 login mawk ncurses-base ncurses-bin passwd perl-base sed slang1a-utf8 initscripts sysvinit sysv-rc tar util-linux whiptail libgcc1 gcc-3.3-base libstdc++5"
+    base="adduser apt apt-utils libdb4.2 at base-config aptitude libsigc++-1.2-5c102 bsdmainutils cpio cron ed exim4 exim4-base exim4-config exim4-daemon-light libgnutls11 libgcrypt11 libgnutls10 libgcrypt7 libgpg-error0 libopencdk8 libtasn1-2 fdutils gettext-base groff-base ifupdown info libssl0.9.7 liblzo1 zlib1g liblockfile1 libpcre3 libwrap0 logrotate mailx man-db libgdbm3 manpages nano netbase nvi libpcap0.7 tasksel libtextwrap1 tcpd telnet libtext-iconv-perl wget $additional"
     without_package () {
         echo "$2" | tr ' ' '\n' | grep -v "^$1$" | tr '\n' ' '
     }
@@ -14,14 +12,26 @@
         echo "$3" | tr ' ' '\n' | sed "s/^$1$/$2/" | tr '\n' ' '
     }
 
-    LIBC6=libc6
-    IPFWTOOL=iptables
+    CPU=`dpkg-architecture -f -a$ARCH -qDEB_HOST_GNU_CPU`
+    SYSTEM=`dpkg-architecture -f -a$ARCH -qDEB_HOST_GNU_SYSTEM`
+
+    case $SYSTEM in
+        linux)
+            required="$required libacl1 libattr1 libcap1 makedev modutils mount procps"
+            base="$base console-common console-tools libconsole console-data dhcp-client klogd net-tools netkit-inetd iputils-ping ppp pppconfig pppoe pppoeconf sysklogd"
+            case $CPU in
+                alpha|ia64)
+                    LIBC=libc6.1
+                    ;;
+                *)
+                    LIBC=libc6
+                    ;;
+            esac
+            IPFWTOOL="iptables"
 
-    case $ARCH in
+    case $CPU in
         "alpha")
-            required="$(subst_package "libc6" "libc6.1" "$required")"
             base="$base pciutils"
-            LIBC6="libc6.1"
 	    IPFWTOOL="ipchains"
             ;;
         "arm")
@@ -34,13 +44,11 @@
 	    IPFWTOOL="ipchains iptables"
             ;;
         "ia64")
-            required="$(subst_package "libc6" "libc6.1" "$required") libreadline4"
+            required="$required libreadline4"
             base="$base elilo efibootmgr dosfstools libparted1.6-0 parted"
-            LIBC6="libc6.1"
             ;;
         "m68k")
             required="$required libreadline4"
-            required="$required libreadline4"
             base="$base atari-bootstrap atari-fdisk amiga-fdisk eject mac-fdisk pmac-fdisk-cross vmelilo"
 	    IPFWTOOL="ipchains"
             ;;
@@ -80,7 +88,7 @@
 	    required="$required libgcc1"
 	    required="$(without_package "libstdc++2.10-glibc2.2" "$required")"
 	    ;;
-        "amd64")
+        "x86_64")
 	    required="$required libtext-iconv-perl zlib1g"
             base="$base pciutils psmisc"
 	    IPFWTOOL="iptables"
@@ -90,6 +98,28 @@
             ;;
     esac
 
+
+            ;;
+        gnu|k*bsd-gnu)
+            base="$base inetutils-syslogd inetutils-tools inetutils-inetd inetutils-ping"
+            case $SYSTEM in
+                gnu)
+                    required="$required gnumach hurd"
+                    LIBC=libc0.3
+                    ;;
+                kfreebsd-gnu)
+                    required="$required freebsd-utils kfreebsd5 kfreebsd-loader libpthread2 ufsutils"
+                    LIBC=libc0.1
+                    ;;
+                knetbsd-gnu)
+                    required="$required libpthread2 ufsutils"
+                    LIBC=libc0.1
+                    ;;
+            esac
+            ;;
+    esac
+
+    required="$required $LIBC"
     base="$base $IPFWTOOL"
 
     all_debs="$required $base"
@@ -174,7 +204,7 @@
     fi
 
     p; progress $baseprog $bases INSTBASE "Installing base system" #4
-    x_core_install $LIBC6
+    x_core_install $LIBC
 
     p; progress $baseprog $bases INSTBASE "Installing base system" #5
     x_core_install perl-base
