
Status: in BTS

--- dpkg-1.10.22/utils/start-stop-daemon.c	2004-06-01 23:18:16.000000000 +0200
+++ dpkg-1.10.22+kbsd/utils/start-stop-daemon.c	2004-06-04 00:44:35.000000000 +0200
@@ -36,6 +36,10 @@
 #  define OSFreeBSD
 #elif defined(__NetBSD__)
 #  define OSNetBSD
+#elif defined(__GLIBC__) && defined(__FreeBSD_kernel__)
+#  define OSGNU_kFreeBSD
+#elif defined(__GLIBC__) && defined(__NetBSD_kernel__)
+#  define OSGNU_kNetBSD
 #else
 #  error Unknown architecture - cannot build start-stop-daemon
 #endif
@@ -725,7 +729,7 @@
 /* WTA: this  needs to be an autoconf check for /proc/pid existance.
  */
 
-#if defined(OSLinux) || defined (OSsunos) || defined(OSfreebsd)
+#if defined(OSLinux) || defined (OSsunos) || defined(OSfreebsd) || defined(OSGNU_kFreeBSD) || defined(OSGNU_kNetBSD)
 static void
 do_procinit(void)
 {
--- dpkg-1.10.23+kbsd/archtable~	2004-07-19 20:56:49.000000000 +0200
+++ dpkg-1.10.23+kbsd/archtable	2004-08-02 18:05:29.000000000 +0200
@@ -52,8 +52,14 @@
 hppa64-linux-gnu		hppa		hppa
 i386-freebsd			freebsd-i386	freebsd-i386
 i386-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i486-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i586-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i686-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
 i386-netbsdelf-gnu		netbsd-i386	netbsd-i386
 i386-knetbsd-gnu		knetbsd-i386	knetbsd-i386
+i486-knetbsd-gnu		knetbsd-i386	knetbsd-i386
+i586-knetbsd-gnu		knetbsd-i386	knetbsd-i386
+i686-knetbsd-gnu		knetbsd-i386	knetbsd-i386
 s390-linux-gnu			s390		s390
 s390-ibm-linux-gnu		s390		s390
 s390-unknown-linux-gnu		s390		s390
