
Status: in BTS

diff -ur dpkg-1.10.23.old/archtable dpkg-1.10.23+kbsd/archtable
--- dpkg-1.10.23.old/archtable	2004-07-19 20:56:49.000000000 +0200
+++ dpkg-1.10.23+kbsd/archtable	2004-08-17 12:55:17.000000000 +0200
@@ -52,8 +52,14 @@
 hppa64-linux-gnu		hppa		hppa
 i386-freebsd			freebsd-i386	freebsd-i386
 i386-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i486-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i586-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
+i686-kfreebsd-gnu		kfreebsd-i386	kfreebsd-i386
 i386-netbsdelf-gnu		netbsd-i386	netbsd-i386
 i386-knetbsd-gnu		knetbsd-i386	knetbsd-i386
+i486-knetbsd-gnu		knetbsd-i386	knetbsd-i386
+i586-knetbsd-gnu		knetbsd-i386	knetbsd-i386
+i686-knetbsd-gnu		knetbsd-i386	knetbsd-i386
 s390-linux-gnu			s390		s390
 s390-ibm-linux-gnu		s390		s390
 s390-unknown-linux-gnu		s390		s390
diff -ur dpkg-1.10.23.old/configure.in dpkg-1.10.23+kbsd/configure.in
--- dpkg-1.10.23.old/configure.in	2004-07-19 20:58:18.000000000 +0200
+++ dpkg-1.10.23+kbsd/configure.in	2004-08-17 12:48:29.000000000 +0200
@@ -173,7 +173,7 @@
 AC_CHECK_TYPE(ptrdiff_t,int)
 AC_CHECK_FUNCS(unsetenv alphasort scandir strerror strsignal strtoul setsid)
 AC_CHECK_FUNCS(vsnprintf lchown snprintf)
-AC_CHECK_HEADERS(sys/cdefs.h syslog.h stddef.h)
+AC_CHECK_HEADERS(sys/cdefs.h syslog.h stddef.h kvm.h)
 AC_CHECK_HEADERS(error.h locale.h)
 AC_DECL_SYS_SIGLIST
 AC_CHECK_LIB(ihash, ihash_create, SSD_LIBS="-lihash $SSD_LIBS")
diff -ur dpkg-1.10.23.old/debian/control dpkg-1.10.23+kbsd/debian/control
--- dpkg-1.10.23.old/debian/control	2004-07-19 20:56:44.000000000 +0200
+++ dpkg-1.10.23+kbsd/debian/control	2004-08-17 12:48:12.000000000 +0200
@@ -6,7 +6,7 @@
 Origin: debian
 Bugs: debbugs://bugs.debian.org
 Standards-Version: 3.5.8
-Build-Depends: debiandoc-sgml, sgml-base (>= 1.9.1), sgmltools-lite, libncurses-dev, gettext (>= 0.12.1-3), zlib1g-dev (>= 1:1.1.3-19.1), autotools-dev
+Build-Depends: debiandoc-sgml, sgml-base (>= 1.9.1), sgmltools-lite, libncurses-dev, gettext (>= 0.12.1-3), zlib1g-dev (>= 1:1.1.3-19.1), autotools-dev, libkvm-dev [kfreebsd-i386 knetbsd-i386]
 
 Package: dpkg
 Architecture: any
diff -ur dpkg-1.10.23.old/utils/start-stop-daemon.c dpkg-1.10.23+kbsd/utils/start-stop-daemon.c
--- dpkg-1.10.23.old/utils/start-stop-daemon.c	2004-07-19 20:56:49.000000000 +0200
+++ dpkg-1.10.23+kbsd/utils/start-stop-daemon.c	2004-08-17 12:48:46.000000000 +0200
@@ -36,6 +36,8 @@
 #  define OSFreeBSD
 #elif defined(__NetBSD__)
 #  define OSNetBSD
+#elif defined(__GLIBC__)
+#  define OSGLIBC
 #else
 #  error Unknown architecture - cannot build start-stop-daemon
 #endif
@@ -47,18 +49,21 @@
 #  include <ps.h>
 #endif
 
-#if defined(OSOpenBSD) || defined(OSFreeBSD) || defined(OSNetBSD)
+#if  defined(OSOpenBSD) || defined(OSFreeBSD) || defined(OSNetBSD)
 #include <sys/param.h>
-#include <sys/user.h>
 #include <sys/proc.h>
 #include <sys/stat.h>
-#include <sys/sysctl.h>
 #include <sys/types.h>
- 
+
 #include <err.h>
-#include <kvm.h>
 #include <limits.h>
 #endif
+ 
+#ifdef HAVE_KVM_H
+#include <kvm.h>
+#include <sys/sysctl.h>
+#include <sys/user.h>
+#endif
 
 #if defined(OShpux)
 #include <sys/param.h>
@@ -204,7 +209,7 @@
 	va_start(arglist, format);
 	vfprintf(stderr, format, arglist);
 	va_end(arglist);
-	putc('\n', stderr);
+	fprintf(stderr, " (%s)\n", strerror (errno));
 	exit(2);
 }
 
@@ -756,7 +761,7 @@
 /* WTA: this  needs to be an autoconf check for /proc/pid existance.
  */
 
-#if defined(OSLinux) || defined (OSsunos) || defined(OSfreebsd)
+#if defined(OSLinux) || defined (OSsunos)
 static void
 do_procinit(void)
 {
@@ -802,7 +807,7 @@
 #endif /* OSHURD */
 
 
-#if defined(OSOpenBSD) || defined(OSFreeBSD) || defined(OSNetBSD)
+#ifdef HAVE_KVM_H
 static int
 pid_is_cmd(pid_t pid, const char *name)
 {
