
Status: dirty (libc headers shouldn't use extra types)

diff -ur e2fsprogs-1.35.old/lib/uuid/gen_uuid.c e2fsprogs-1.35/lib/uuid/gen_uuid.c
--- e2fsprogs-1.35.old/lib/uuid/gen_uuid.c	2004-08-09 23:00:16.000000000 +0200
+++ e2fsprogs-1.35/lib/uuid/gen_uuid.c	2004-08-09 22:40:46.000000000 +0200
@@ -37,6 +37,7 @@
  * gcc-wall wall mode
  */
 #define _SVID_SOURCE
+#define _BSD_SOURCE
 
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
