diff -ur firebird2-1.5.1.old/configure.in firebird2-1.5.1/configure.in
--- firebird2-1.5.1.old/configure.in	2004-08-08 19:07:43.000000000 +0200
+++ firebird2-1.5.1/configure.in	2004-08-08 19:13:26.000000000 +0200
@@ -56,7 +56,7 @@
     SHRLIB_EXT=so
     ;;
 
-  x86_64*-*-linux-*)
+  x86_64*-*-linux* | x86_64*-*-gnu* | x86_64*-*-k*bsd*-gnu)
     MAKEFILE_PREFIX=linux_amd64
     INSTALL_PREFIX=linux
     PLATFORM=LINUX
@@ -68,7 +68,7 @@
     CPU_TYPE=amd64
     ;;
 
-  sparc*-*-linux-*)
+  sparc*-*-linux* | sparc*-*-gnu* | sparc*-*-k*bsd*-gnu)
     MAKEFILE_PREFIX=linux_sparc32
     INSTALL_PREFIX=linux
     PLATFORM=LINUX
@@ -78,7 +78,7 @@
     SHRLIB_EXT=so
     ;;
 
-  *-*-linux*)
+  *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu)
     MAKEFILE_PREFIX=linux
     PLATFORM=LINUX
     AC_DEFINE(LINUX)
diff -ur firebird2-1.5.1.old/debian/control.in firebird2-1.5.1/debian/control.in
--- firebird2-1.5.1.old/debian/control.in	2004-08-08 19:08:21.000000000 +0200
+++ firebird2-1.5.1/debian/control.in	2004-08-08 19:10:26.000000000 +0200
@@ -3,11 +3,11 @@
 Priority: optional
 Maintainer: Debian Firebird Group <pkg-firebird-general@lists.alioth.debian.org>
 Uploaders: Daniel Urban <daniel@sente.pl>
-Build-Depends: debhelper (>> 3.0.0), docbook-to-man, autoconf, libtool, libncurses5-dev
+Build-Depends: debhelper (>> 3.0.0), docbook-to-man, autoconf, libtool, libncurses5-dev, type-handling (>= 0.2.1)
 Standards-Version: 3.5.2
 
 Package: firebird2-super-server
-Architecture: i386
+Architecture: @arches@
 Depends: ${shlibs:Depends}, firebird2-server-common
 Provides: firebird2-server
 Conflicts: firebird2-server, firebird-server
@@ -30,7 +30,7 @@
  License v.1.0 on 25 July, 2000. 
 	   
 Package: firebird2-classic-server
-Architecture: i386
+Architecture: @arches@
 Depends: ${shlibs:Depends}, firebird2-server-common
 Provides: firebird2-server
 Conflicts: firebird2-server, firebird-server
@@ -53,7 +53,7 @@
  License v.1.0 on 25 July, 2000. 
 
 Package: libfirebird2-super
-Architecture: i386
+Architecture: @arches@
 Depends: ${shlibs:Depends}
 Provides: libfirebird2, libfirebird
 Conflicts: libfirebird2, libfirebird
@@ -61,7 +61,7 @@
  Please see the firebird2-super-server package for a full description.
 
 Package: libfirebird2-classic
-Architecture: i386
+Architecture: @arches@
 Depends: ${shlibs:Depends}
 Provides: libfirebird2, libfirebird
 Conflicts: libfirebird2, libfirebird
@@ -69,7 +69,7 @@
  Please see the firebird2-classic-server package for a full description.
 
 Package: firebird2-server-common
-Architecture: i386
+Architecture: @arches@
 Depends: ${shlibs:Depends}
 Recommends: firebird2-server
 Conflicts: firebird-common
@@ -92,7 +92,7 @@
  License v.1.0 on 25 July, 2000.
 
 Package: firebird2-utils-super
-Architecture: i386
+Architecture: @arches@
 Depends: ${shlibs:Depends}, firebird2-server-common
 Conflicts: firebird-utils
 Description: Utilities for Firebird - a RDBMS based on InterBase 6.0 code
@@ -105,7 +105,7 @@
  about Firebird in general. 
 
 Package: firebird2-utils-classic
-Architecture: i386
+Architecture: @arches@
 Depends: ${shlibs:Depends}, firebird2-server-common
 Conflicts: firebird-utils
 Description: Utilities for Firebird - a RDBMS based on InterBase 6.0 code
@@ -118,7 +118,7 @@
  about Firebird in general. 
 
 Package: firebird2-dev
-Architecture: i386
+Architecture: @arches@
 Depends: libfirebird2
 Suggests: firebird2-examples
 Conflicts: firebird-dev
@@ -129,7 +129,7 @@
  about Firebird in general.
 
 Package: firebird2-examples
-Architecture: i386
+Architecture: @arches@
 Description: Examples for Firebird - a RDBMS based on InterBase 6.0 code
  The examples included in this package are provided as samples and are also
  used by Firebird's testing suite (TCS). The examples are in the form 
diff -ur firebird2-1.5.1.old/debian/rules firebird2-1.5.1/debian/rules
--- firebird2-1.5.1.old/debian/rules	2004-08-08 19:07:43.000000000 +0200
+++ firebird2-1.5.1/debian/rules	2004-08-08 19:26:50.000000000 +0200
@@ -77,7 +77,7 @@
 build-super-server: configure-super
 	echo Starting build-super-server
 	dh_testdir
-	$(MAKE) > make-super.log 2>&1
+	$(MAKE)
 
 	# moving to SuperServer target
 	rm -rf ./debian/firebird2-super
@@ -87,7 +87,7 @@
 build-classic-server: configure-classic
 	echo Starting build-classic-server
 	dh_testdir
-	$(MAKE) > make-classic.log 2>&1
+	$(MAKE)
 
 	# moving to ClassicServer target
 	rm -rf ./debian/firebird2-classic
@@ -119,7 +119,10 @@
 	    src/extern/editline/config.log \
 	    src/extern/editline/config.h \
 	    src/extern/editline/config.status 
-	    
+
+	cat debian/control.in \
+	| sed "s/@arches@/`type-handling i386 any`/g" \
+	> debian/control
 	
 	#chmod -f 644 src/*.fdb
 	rm -f src/*.fdb src/*.lnk src/indicator.* src/Makefile
diff -ur firebird2-1.5.1.old/src/jrd/gds.cpp firebird2-1.5.1/src/jrd/gds.cpp
--- firebird2-1.5.1.old/src/jrd/gds.cpp	2004-02-06 05:56:28.000000000 +0100
+++ firebird2-1.5.1/src/jrd/gds.cpp	2004-08-09 03:10:02.000000000 +0200
@@ -966,14 +966,7 @@
 		break;
 
 	case gds_arg_unix:
-		if (code > 0 && code < sys_nerr && (p = (TEXT*)sys_errlist[code]))
-			strcpy(s, p);
-		else if (code == 60)
-			strcpy(s, "connection timed out");
-		else if (code == 61)
-			strcpy(s, "connection refused");
-		else
-			sprintf(s, "unknown unix error %ld", code);	/* TXNN */
+		sprintf(s, "%s", strerror (code));
 		break;
 
 	case gds_arg_dos:
diff -ur firebird2-1.5.1.old/src/lock/lock.cpp firebird2-1.5.1/src/lock/lock.cpp
--- firebird2-1.5.1.old/src/lock/lock.cpp	2004-03-29 06:35:49.000000000 +0200
+++ firebird2-1.5.1/src/lock/lock.cpp	2004-08-08 19:48:21.000000000 +0200
@@ -2061,13 +2061,13 @@
 		}
 	}
 
-	sprintf(s, "Fatal lock manager error: %s, errno: %d", string, ERRNO);
+	sprintf(s, "Fatal lock manager error: %s, %s", string, strerror (ERRNO));
 	gds__log(s);
 	ib_fprintf(ib_stderr, "%s\n", s);
 
 #if !(defined WIN_NT)
 	if (errno > 0)
-		ib_fprintf(ib_stderr, "--%s\n", sys_errlist[errno]);
+		ib_fprintf(ib_stderr, "--%s\n", strerror (errno));
 #endif
 
 #ifdef DEV_BUILD
