
Status: in BTS

diff -Nur freeglut-2.2.0.old/configure.in freeglut-2.2.0/configure.in
--- freeglut-2.2.0.old/configure.in	2004-03-30 10:37:27.000000000 +0200
+++ freeglut-2.2.0/configure.in	2004-08-31 21:40:40.000000000 +0200
@@ -6,6 +6,7 @@
 
 dnl Checks for programs.
 AC_PROG_CC
+AC_PROG_CXX
 AC_C_CONST
 AC_EXEEXT
 AC_LIBTOOL_DLOPEN
diff -Nur freeglut-2.2.0.old/progs/demos/CallbackMaker/Makefile.am freeglut-2.2.0/progs/demos/CallbackMaker/Makefile.am
--- freeglut-2.2.0.old/progs/demos/CallbackMaker/Makefile.am	2004-03-30 10:37:27.000000000 +0200
+++ freeglut-2.2.0/progs/demos/CallbackMaker/Makefile.am	2004-08-31 21:40:40.000000000 +0200
@@ -1,5 +1,5 @@
 EXTRA_DIST = CallbackMaker.c CallbackMaker.dsp
 noinst_PROGRAMS = CallbackMaker
-CallbackMaker_SOURCES = CallbackMaker.c
+CallbackMaker_SOURCES = CallbackMaker.c empty.cc
 CallbackMaker_LDFLAGS = -export-dynamic ../../../src/lib@LIBRARY@.la
 CallbackMaker_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
diff -Nur freeglut-2.2.0.old/progs/demos/CallbackMaker/empty.cc freeglut-2.2.0/progs/demos/CallbackMaker/empty.cc
--- freeglut-2.2.0.old/progs/demos/CallbackMaker/empty.cc	1970-01-01 01:00:00.000000000 +0100
+++ freeglut-2.2.0/progs/demos/CallbackMaker/empty.cc	2004-08-31 21:40:40.000000000 +0200
@@ -0,0 +1 @@
+
diff -Nur freeglut-2.2.0.old/progs/demos/Fractals/Makefile.am freeglut-2.2.0/progs/demos/Fractals/Makefile.am
--- freeglut-2.2.0.old/progs/demos/Fractals/Makefile.am	2004-03-30 10:37:27.000000000 +0200
+++ freeglut-2.2.0/progs/demos/Fractals/Makefile.am	2004-08-31 21:40:40.000000000 +0200
@@ -1,5 +1,5 @@
 EXTRA_DIST = Fractals.dsp fractals.c fractals.dat
 noinst_PROGRAMS = fractals
-fractals_SOURCES = fractals.c
+fractals_SOURCES = fractals.c empty.cc
 fractals_LDFLAGS = -export-dynamic ../../../src/lib@LIBRARY@.la
 fractals_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
diff -Nur freeglut-2.2.0.old/progs/demos/Fractals/empty.cc freeglut-2.2.0/progs/demos/Fractals/empty.cc
--- freeglut-2.2.0.old/progs/demos/Fractals/empty.cc	1970-01-01 01:00:00.000000000 +0100
+++ freeglut-2.2.0/progs/demos/Fractals/empty.cc	2004-08-31 21:40:40.000000000 +0200
@@ -0,0 +1 @@
+
diff -Nur freeglut-2.2.0.old/progs/demos/Fractals_random/Makefile.am freeglut-2.2.0/progs/demos/Fractals_random/Makefile.am
--- freeglut-2.2.0.old/progs/demos/Fractals_random/Makefile.am	2004-03-30 10:37:27.000000000 +0200
+++ freeglut-2.2.0/progs/demos/Fractals_random/Makefile.am	2004-08-31 21:40:40.000000000 +0200
@@ -1,6 +1,6 @@
 EXTRA_DIST = Fractals_random.dsp fractals.dat fractals_random.c
 
 noinst_PROGRAMS = fractals_random
-fractals_random_SOURCES = fractals_random.c
+fractals_random_SOURCES = fractals_random.c empty.cc
 fractals_random_LDFLAGS = -export-dynamic ../../../src/lib@LIBRARY@.la
 fractals_random_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
diff -Nur freeglut-2.2.0.old/progs/demos/Fractals_random/empty.cc freeglut-2.2.0/progs/demos/Fractals_random/empty.cc
--- freeglut-2.2.0.old/progs/demos/Fractals_random/empty.cc	1970-01-01 01:00:00.000000000 +0100
+++ freeglut-2.2.0/progs/demos/Fractals_random/empty.cc	2004-08-31 21:40:40.000000000 +0200
@@ -0,0 +1 @@
+
diff -Nur freeglut-2.2.0.old/progs/demos/Lorenz/Makefile.am freeglut-2.2.0/progs/demos/Lorenz/Makefile.am
--- freeglut-2.2.0.old/progs/demos/Lorenz/Makefile.am	2004-03-30 10:37:27.000000000 +0200
+++ freeglut-2.2.0/progs/demos/Lorenz/Makefile.am	2004-08-31 21:40:40.000000000 +0200
@@ -1,6 +1,6 @@
 EXTRA_DIST = lorenz.c lorenz.dsp
 
 noinst_PROGRAMS = lorenz
-lorenz_SOURCES = lorenz.c
+lorenz_SOURCES = lorenz.c empty.cc
 lorenz_LDFLAGS = -export-dynamic ../../../src/lib@LIBRARY@.la
 lorenz_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
diff -Nur freeglut-2.2.0.old/progs/demos/Lorenz/empty.cc freeglut-2.2.0/progs/demos/Lorenz/empty.cc
--- freeglut-2.2.0.old/progs/demos/Lorenz/empty.cc	1970-01-01 01:00:00.000000000 +0100
+++ freeglut-2.2.0/progs/demos/Lorenz/empty.cc	2004-08-31 21:40:40.000000000 +0200
@@ -0,0 +1 @@
+
diff -Nur freeglut-2.2.0.old/progs/demos/One/Makefile.am freeglut-2.2.0/progs/demos/One/Makefile.am
--- freeglut-2.2.0.old/progs/demos/One/Makefile.am	2004-03-30 10:37:27.000000000 +0200
+++ freeglut-2.2.0/progs/demos/One/Makefile.am	2004-08-31 21:40:40.000000000 +0200
@@ -1,5 +1,5 @@
 EXTRA_DIST = one.c one.dsp
 noinst_PROGRAMS = one
-one_SOURCES = one.c
+one_SOURCES = one.c empty.cc
 one_LDFLAGS = -export-dynamic ../../../src/lib@LIBRARY@.la
 one_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
diff -Nur freeglut-2.2.0.old/progs/demos/One/empty.cc freeglut-2.2.0/progs/demos/One/empty.cc
--- freeglut-2.2.0.old/progs/demos/One/empty.cc	1970-01-01 01:00:00.000000000 +0100
+++ freeglut-2.2.0/progs/demos/One/empty.cc	2004-08-31 21:40:40.000000000 +0200
@@ -0,0 +1 @@
+
diff -Nur freeglut-2.2.0.old/progs/demos/shapes/Makefile.am freeglut-2.2.0/progs/demos/shapes/Makefile.am
--- freeglut-2.2.0.old/progs/demos/shapes/Makefile.am	2004-03-30 10:37:27.000000000 +0200
+++ freeglut-2.2.0/progs/demos/shapes/Makefile.am	2004-08-31 21:40:40.000000000 +0200
@@ -1,5 +1,5 @@
 EXTRA_DIST = shapes.c shapes.dsp
 noinst_PROGRAMS = shapes
-shapes_SOURCES = shapes.c
+shapes_SOURCES = shapes.c empty.cc
 shapes_LDFLAGS = -export-dynamic ../../../src/lib@LIBRARY@.la
 shapes_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
diff -Nur freeglut-2.2.0.old/progs/demos/shapes/empty.cc freeglut-2.2.0/progs/demos/shapes/empty.cc
--- freeglut-2.2.0.old/progs/demos/shapes/empty.cc	1970-01-01 01:00:00.000000000 +0100
+++ freeglut-2.2.0/progs/demos/shapes/empty.cc	2004-08-31 21:40:40.000000000 +0200
@@ -0,0 +1 @@
+
