
Status: libtool bits are merged upstream. rest is pending

diff -ur gdb-6.1.old/gdb/configure.tgt gdb-6.1/gdb/configure.tgt
--- gdb-6.1.old/gdb/configure.tgt	2004-06-03 23:56:45.000000000 +0200
+++ gdb-6.1/gdb/configure.tgt	2004-06-04 00:03:11.000000000 +0200
@@ -37,8 +37,10 @@
 
 case "${target}" in
 
-x86_64-*-freebsd*)	gdb_target=fbsd64 ;;
-*-*-freebsd*)		gdb_target=fbsd	;;
+x86_64-*-freebsd* | x86_64-*-kfreebsd*-gnu)
+			gdb_target=fbsd64 ;;
+*-*-freebsd* | *-*-kfreebsd*-gnu)
+			gdb_target=fbsd	;;
 
 alpha*-*-osf*)		gdb_target=alpha-osf1 ;;
 alpha*-*-linux*)	gdb_target=alpha-linux ;;
@@ -224,6 +226,10 @@
 
 case "${target}" in
 *-*-linux*)	gdb_osabi=GDB_OSABI_LINUX ;;
+*-*-kfreebsd*-gnu)
+		gdb_osabi=GDB_OSABI_FREEBSD_ELF ;;
+*-*-knetbsd*-gnu)
+		gdb_osabi=GDB_OSABI_NETBSD_ELF ;;
 *-*-gnu*)	gdb_osabi=GDB_OSABI_HURD ;;
 *-*-nto*)	gdb_osabi=GDB_OSABI_QNXNTO ;;
 *-*-solaris*)	gdb_osabi=GDB_OSABI_SOLARIS ;;
diff -ur gdb-6.1.old/gdb/i386bsd-nat.c gdb-6.1/gdb/i386bsd-nat.c
--- gdb-6.1.old/gdb/i386bsd-nat.c	2003-09-28 15:35:44.000000000 +0200
+++ gdb-6.1/gdb/i386bsd-nat.c	2004-06-04 00:03:11.000000000 +0200
@@ -30,6 +30,16 @@
 #include <machine/reg.h>
 #include <machine/frame.h>
 
+#if defined(__FreeBSD__) && ! defined(__FreeBSD_kernel__)
+# define __FreeBSD_kernel__ __FreeBSD__
+#endif
+#ifdef __FreeBSD_kernel__
+# include <osreldate.h>
+# ifndef __FreeBSD_kernel_version
+#  define __FreeBSD_kernel_version __FreeBSD_version
+# endif
+#endif
+
 #ifdef HAVE_SYS_PROCFS_H
 #include <sys/procfs.h>
 #endif
@@ -392,9 +402,9 @@
      system header files and sysctl(3) to get at the relevant
      information.  */
 
-#if defined (__FreeBSD_version) && __FreeBSD_version >= 400011
+#if defined (__FreeBSD_kernel_version) && __FreeBSD_kernel_version >= 400011
 #define SC_REG_OFFSET i386fbsd4_sc_reg_offset
-#elif defined (__FreeBSD_version) && __FreeBSD_version >= 300005
+#elif defined (__FreeBSD_kernel_version) && __FreeBSD_kernel_version >= 300005
 #define SC_REG_OFFSET i386fbsd_sc_reg_offset
 #elif defined (NetBSD) || defined (__NetBSD_Version__)
 #define SC_REG_OFFSET i386nbsd_sc_reg_offset
diff -ur gdb-6.1.old/libtool.m4 gdb-6.1/libtool.m4
--- gdb-6.1.old/libtool.m4	2003-04-11 05:58:39.000000000 +0200
+++ gdb-6.1/libtool.m4	2004-06-04 00:03:11.000000000 +0200
@@ -577,7 +577,7 @@
   esac
   ;;
 
-freebsd* )
+freebsd* | kfreebsd*-gnu)
   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
     case $host_cpu in
     i*86 )
@@ -645,7 +645,7 @@
   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
   ;;
 
-netbsd*)
+netbsd* | knetbsd*-gnu)
   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
     [lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$']
   else
diff -ur gdb-6.1.old/ltcf-c.sh gdb-6.1/ltcf-c.sh
--- gdb-6.1.old/ltcf-c.sh	2002-12-01 13:01:26.000000000 +0100
+++ gdb-6.1/ltcf-c.sh	2004-06-04 00:03:11.000000000 +0200
@@ -185,7 +185,7 @@
     whole_archive_flag_spec='-all_load $convenience'
     ;;
 
-  netbsd*)
+  netbsd* | knetbsd*-gnu)
     if echo __ELF__ | $CC -E - | grep __ELF__ >/dev/null; then
       archive_cmds='$LD -Bshareable $libobjs $deplibs $linker_flags -o $lib'
       wlarc=
@@ -409,7 +409,7 @@
     ;;
 
   # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
-  freebsd*)
+  freebsd* | kfreebsd*-gnu)
     archive_cmds='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
     hardcode_libdir_flag_spec='-R$libdir'
     hardcode_direct=yes
@@ -456,7 +456,7 @@
     link_all_deplibs=yes
     ;;
 
-  netbsd*)
+  netbsd* | knetbsd*-gnu)
     if echo __ELF__ | $CC -E - | grep __ELF__ >/dev/null; then
       archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'  # a.out
     else
diff -ur gdb-6.1.old/ltcf-cxx.sh gdb-6.1/ltcf-cxx.sh
--- gdb-6.1.old/ltcf-cxx.sh	2003-02-20 02:12:28.000000000 +0100
+++ gdb-6.1/ltcf-cxx.sh	2004-06-04 00:03:11.000000000 +0200
@@ -244,7 +244,7 @@
     # C++ shared libraries reported to be fairly broken before switch to ELF
     ld_shlibs=no
     ;;
-  freebsd*)
+  freebsd* | kfreebsd*-gnu)
     # FreeBSD 3 and later use GNU C++ and GNU ld with standard ELF
     # conventions
     ld_shlibs=yes
@@ -404,7 +404,7 @@
         ;;
     esac
     ;;
-  netbsd*)
+  netbsd* | knetbsd*-gnu)
     # NetBSD uses g++ - do we need to do anything?
     ;;
   osf3*)
@@ -759,7 +759,7 @@
           ;;
       esac
       ;;
-    freebsd*)
+    freebsd* | kfreebsd*-gnu)
       # FreeBSD uses GNU C++
       ;;
     gnu*)
diff -ur gdb-6.1.old/ltcf-gcj.sh gdb-6.1/ltcf-gcj.sh
--- gdb-6.1.old/ltcf-gcj.sh	2003-02-20 00:51:28.000000000 +0100
+++ gdb-6.1/ltcf-gcj.sh	2004-06-04 00:03:11.000000000 +0200
@@ -178,7 +178,7 @@
       $CC $output_objdir/$soname-exp '$lt_cv_cc_dll_switch' -Wl,-e,'$dll_entry' -o $output_objdir/$soname '$ltdll_obj'$libobjs $deplibs $compiler_flags'
     ;;
 
-  netbsd*)
+  netbsd* | knetbsd*-gnu)
     if echo __ELF__ | $CC -E - | grep __ELF__ >/dev/null; then
       archive_cmds='$LD -Bshareable $libobjs $deplibs $linker_flags -o $lib'
       wlarc=
@@ -402,7 +402,7 @@
     ;;
 
   # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
-  freebsd*)
+  freebsd* | kfreebsd*-gnu)
     archive_cmds='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
     hardcode_libdir_flag_spec='-R$libdir'
     hardcode_direct=yes
@@ -433,7 +433,7 @@
     link_all_deplibs=yes
     ;;
 
-  netbsd*)
+  netbsd* | knetbsd*-gnu)
     if echo __ELF__ | $CC -E - | grep __ELF__ >/dev/null; then
       archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'  # a.out
     else
diff -ur gdb-6.1.old/ltconfig gdb-6.1/ltconfig
--- gdb-6.1.old/ltconfig	2003-10-04 06:54:47.000000000 +0200
+++ gdb-6.1/ltconfig	2004-06-04 00:03:11.000000000 +0200
@@ -1164,6 +1164,17 @@
   hardcode_into_libs=yes
   ;;
 
+kfreebsd*-gnu | knetbsd*-gnu)
+  version_type=linux
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='${libname}${release}.so$versuffix ${libname}${release}.so${major} ${libname}.so'
+  soname_spec='${libname}${release}.so$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=yes
+  hardcode_into_libs=yes
+  ;;
+
 hpux9* | hpux10* | hpux11*)
   # Give a soname corresponding to the major version so that dld.sl refuses to
   # link against other versions.
