
Status: in BTS

diff -Nur gimp-print-4.2.7.old/debian/rules gimp-print-4.2.7/debian/rules
--- gimp-print-4.2.7.old/debian/rules	2004-08-05 18:37:37.000000000 +0200
+++ gimp-print-4.2.7/debian/rules	2004-08-05 17:05:54.000000000 +0200
@@ -10,9 +10,15 @@
 
 # These are used for cross-compiling and for saving the configure script
 # from having to guess our platform (since we know it already)
-DEB_HOST_GNU_TYPE       ?= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
-DEB_BUILD_GNU_TYPE      ?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
-
+DEB_HOST_GNU_TYPE	?= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
+DEB_BUILD_GNU_TYPE	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
+DEB_BUILD_GNU_CPU	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_CPU)
+DEB_BUILD_GNU_SYSTEM	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_SYSTEM)
+
+ifneq (, $(filter $(DEB_BUILD_GNU_SYSTEM), kfreebsd-gnu knetbsd-gnu))
+DEB_BUILD_GNU_TYPE	= $(DEB_BUILD_GNU_CPU)-gnu
+DEB_HOST_GNU_TYPE	= $(DEB_HOST_GNU_CPU)-gnu
+endif
 
 ifneq (,$(findstring debug,$(DEB_BUILD_OPTIONS)))
 	CFLAGS += -g
