
Status: in BTS.

diff -ur gnustep-base-1.10.1.old/debian/control.m4 gnustep-base-1.10.1/debian/control.m4
--- gnustep-base-1.10.1.old/debian/control.m4	2004-12-20 02:21:39.000000000 +0100
+++ gnustep-base-1.10.1/debian/control.m4	2004-12-20 02:26:06.000000000 +0100
@@ -8,7 +8,7 @@
 Maintainer: Debian GNUstep maintainers <pkg-gnustep-maintainers@lists.alioth.debian.org>
 Uploaders: Matthias Klose <doko@debian.org>, Eric Heintzmann <eric@gnustep.fr.st>
 Standards-Version: 3.6.1.1
-Build-Depends: debhelper (>= 4), gobjc (>= V_OBJC), gnustep-make (>= V_MAKE), libffcall1-dev (>= 1.10) [FFCALL_ARCHS], libffi2-dev [FFI_ARCHS], libxml2-dev, libxslt1-dev, libssl-dev, zlib1g-dev, m4
+Build-Depends: debhelper (>= 4), gobjc (>= V_OBJC), gnustep-make (>= V_MAKE), libffcall1-dev (>= 1.10) [FFCALL_ARCHS], libffi2-dev [FFI_ARCHS], libxml2-dev, libxslt1-dev, libssl-dev, zlib1g-dev, m4, libkvm-dev [KVM_ARCHS], type-handling (>= 0.2.1)
 Build-Depends-Indep: texi2html, texinfo, tetex-bin
 
 Package: gnustep-base-common
diff -ur gnustep-base-1.10.1.old/debian/rules gnustep-base-1.10.1/debian/rules
--- gnustep-base-1.10.1.old/debian/rules	2004-12-20 02:21:39.000000000 +0100
+++ gnustep-base-1.10.1/debian/rules	2004-12-20 02:25:06.000000000 +0100
@@ -28,7 +28,7 @@
 # from having to guess our platform (since we know it already)
 DEB_BUILD_GNU_TYPE	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
 DEB_HOST_GNU_TYPE	?= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
-DEB_HOST_ARCH		?= $(shell dpkg-architecture -qDEB_HOST_ARCH)
+DEB_HOST_GNU_CPU	?= $(shell dpkg-architecture -qDEB_HOST_GNU_CPU)
 
 # make some files executable
 dummy := $(shell chmod +x configure)
@@ -58,15 +58,15 @@
 # -----------------------------------------------------------------------------
 # FFI lib settings
 
-FFCALL_ARCHS	= alpha amd64 arm hppa i386 ia64 m68k mips powerpc s390 sparc
-FFI_ARCHS	= mipsel
+FFCALL_CPUS	= alpha x86_64 arm hppa i386 ia64 m68k mips powerpc s390 sparc
+FFI_CPUS	= mipsel
 
-ifeq ($(DEB_HOST_ARCH), $(findstring $(DEB_HOST_ARCH),$(FFI_ARCHS)))
+ifeq ($(DEB_HOST_GNU_CPU), $(findstring $(DEB_HOST_GNU_CPU),$(FFI_CPUS)))
   FFI_LIB = ffi
   FFI_CONFIG = --enable-libffi
   FFI_DEVPKG = libffi2-dev
 else
-  ifeq ($(DEB_HOST_ARCH), $(findstring $(DEB_HOST_ARCH),$(FFCALL_ARCHS)))
+  ifeq ($(DEB_HOST_GNU_CPU), $(findstring $(DEB_HOST_GNU_CPU),$(FFCALL_CPUS)))
     FFI_LIB = ffcall
     FFI_CONFIG = --enable-ffcall
     FFI_DEVPKG = libffcall1-dev
@@ -130,8 +130,9 @@
 		-DV_MAKE='$(v_make)' \
 		-DSOV_BASE='$(sov_base)' \
 		-DWITH_DEBUG \
-		-DFFCALL_ARCHS='$(FFCALL_ARCHS)' \
-		-DFFI_ARCHS='$(FFI_ARCHS)' \
+		-DFFCALL_ARCHS="$(shell type-handling `echo $(FFCALL_CPUS) | tr ' ' ,` any)" \
+		-DFFI_ARCHS="$(shell type-handling `echo $(FFI_CPUS) | tr ' ' ,` any)" \
+		-DKVM_ARCHS="$(shell type-handling any kfreebsd-gnu,knetbsd-gnu)" \
 		-DFFI_DEVPKG='$(FFI_DEVPKG)' \
 		debian/control.m4 | uniq > debian/control.tmp
 
