
Status: in BTS

diff -x aclocal.m4 -x config.guess -x config.sub -x configure -x makefile.mingw -x gtkglarea.spec -x install-sh -x ltmain.sh -x missing -x mkinstalldirs -x Makefile.in -Nur gtkgl2-1.99.0.old/examples/Makefile.am gtkgl2-1.99.0/examples/Makefile.am
--- gtkgl2-1.99.0.old/examples/Makefile.am	2002-03-13 03:09:09.000000000 +0100
+++ gtkgl2-1.99.0/examples/Makefile.am	2004-09-02 09:43:38.000000000 +0200
@@ -7,9 +7,9 @@
 INCLUDES = -I$(top_srcdir)
 LDADD = $(top_builddir)/gtkgl/libgtkgl-2.0.la $(GTK_LIBS) $(GL_LIBS)
 
-simple_SOURCES   = simple.c
-zktor_SOURCES    = zktor.c
-glpixmap_SOURCES = glpixmap.c
-viewlw_SOURCES   = viewlw.c lw.h lw.c trackball.h trackball.c
-gdkfont_SOURCES  = gdkfont.c
-gtkglarea_demo_SOURCES = gtkglarea_demo.c
+simple_SOURCES   = simple.c empty.cc
+zktor_SOURCES    = zktor.c empty.cc
+glpixmap_SOURCES = glpixmap.c empty.cc
+viewlw_SOURCES   = viewlw.c lw.h lw.c trackball.h trackball.c empty.cc
+gdkfont_SOURCES  = gdkfont.c empty.cc
+gtkglarea_demo_SOURCES = gtkglarea_demo.c empty.cc
diff -x aclocal.m4 -x config.guess -x config.sub -x configure -x makefile.mingw -x gtkglarea.spec -x install-sh -x ltmain.sh -x missing -x mkinstalldirs -x Makefile.in -Nur gtkgl2-1.99.0.old/examples/empty.cc gtkgl2-1.99.0/examples/empty.cc
--- gtkgl2-1.99.0.old/examples/empty.cc	1970-01-01 01:00:00.000000000 +0100
+++ gtkgl2-1.99.0/examples/empty.cc	2004-09-02 09:34:31.000000000 +0200
@@ -0,0 +1 @@
+
