
Status: dirty

diff -ur libgnomeui-2.6.1.1.old/debian/control.in libgnomeui-2.6.1.1/debian/control.in
--- libgnomeui-2.6.1.1.old/debian/control.in	2004-09-20 21:53:26.000000000 +0200
+++ libgnomeui-2.6.1.1/debian/control.in	2004-09-20 21:31:11.000000000 +0200
@@ -15,21 +15,6 @@
  This package contains the shared library for the base GNOME library
  functions (User Interface functions).
 
-Package: libgnomeui-dbg
-Architecture: any
-Section: libdevel
-Depends: libgnomeui-0 (= ${Source-Version})
-Description: The GNOME 2 libraries (User Interface) and debugging symbols
- This package contains the shared library for the base GNOME library
- functions (User Interface functions).
- .
- This package contains unstripped shared libraries. It is provided primarily
- to provide a backtrace with names in a debugger, this makes it somewhat
- easier to interpret core dumps. The libraries are installed in
- /usr/lib/debug and can be used by placing that directory in
- LD_LIBRARY_PATH.
- Most people will not need this package.
-
 Package: libgnomeui-dev
 Architecture: any
 Section: libdevel
diff -ur libgnomeui-2.6.1.1.old/debian/rules libgnomeui-2.6.1.1/debian/rules
--- libgnomeui-2.6.1.1.old/debian/rules	2004-09-20 21:53:26.000000000 +0200
+++ libgnomeui-2.6.1.1/debian/rules	2004-09-20 21:31:16.000000000 +0200
@@ -17,7 +17,6 @@
 # Speed up loading.
 LDFLAGS += -Wl,-O1
 
-DEB_DH_STRIP_ARGS := --keep-debug
 DEB_CONFIGURE_EXTRA_FLAGS := --libexecdir=\$${prefix}/lib/${version} LDFLAGS="$(LDFLAGS)"
 
 clean::
