
Status: in BTS.

diff -Nur perl-5.8.4.old/debian/rules perl-5.8.4/debian/rules
--- perl-5.8.4.old/debian/rules	2004-05-09 16:46:58.000000000 +0200
+++ perl-5.8.4/debian/rules	2004-05-09 16:55:43.000000000 +0200
@@ -6,6 +6,8 @@
 
 export SHELL = /bin/sh
 
+DEB_BUILD_GNU_SYSTEM ?= $(shell dpkg-architecture -qDEB_BUILD_GNU_SYSTEM)
+
 fullversion := $(shell /bin/bash debian/config.debian --full-version)
 version     := $(shell /bin/bash debian/config.debian --version)
 installtype := $(shell /bin/bash debian/config.debian --install-type)
@@ -43,7 +45,12 @@
 	rm -f libperl.so* # must be built last
 	-$(MAKE) distclean
 	$(SHELL) debian/config.debian --static
-	$(MAKE) perl $(test_target)
+	$(MAKE) perl
+ifeq ($(DEB_BUILD_GNU_SYSTEM),linux)
+	$(MAKE) $(test_target)
+else
+	-$(MAKE) $(test_target)
+endif
 	mv libperl.a libperl-static
 	mv perl perl.static
 
@@ -63,7 +70,12 @@
 	$(MAKE) SHRPLDFLAGS='$$(LDDLFLAGS) -Wl,-soname,libperl.so.$(version)' $@
 	ln -s libperl.so.$(fullversion) libperl.so.$(version)
 	ln -s libperl.so.$(version) libperl.so
-	$(MAKE) all $(test_target) || { rm -f libperl.so*; exit 1; }
+	$(MAKE) all || { rm -f libperl.so*; exit 1; }
+ifeq ($(DEB_BUILD_GNU_SYSTEM),linux)
+	$(MAKE) $(test_target) || { rm -f libperl.so*; exit 1; }
+else
+	-$(MAKE) $(test_target)
+endif
 
 clean:
 	$(checkdir)
