
Description: Disable tests.
Status: Won't merge. Needs proper fixing.

diff -ur python2.3-2.3.4.old/debian/rules python2.3-2.3.4/debian/rules
--- python2.3-2.3.4.old/debian/rules	2004-08-08 08:51:52.000000000 +0200
+++ python2.3-2.3.4/debian/rules	2004-08-08 08:39:52.000000000 +0200
@@ -134,10 +134,6 @@
 	touch stamp-configure-static
 
 stamp-check:
-	@echo "BEGIN test"
-	-$(MAKE) -C $(buildd_static) test TESTOPTS="-l -unetwork,bsddb" 2>&1 \
-	    | tee debian/test_results
-	@echo "END test"
 	touch stamp-check
 
 stamp-doc-html:
@@ -389,8 +385,7 @@
 	dh_installdocs -p$(p_base) \
 		README Misc/NEWS Misc/ACKS Misc/HISTORY \
 		debian/sample.postinst debian/sample.prerm \
-		debian/README.maintainers \
-		debian/test_results
+		debian/README.maintainers
 	dh_installdocs --all -N$(p_base) debian/README.Debian
 
 	: # IDLE has its own changelogs, docs...
