diff -ur samba-3.0.9.old/debian/control.in samba-3.0.9/debian/control.in
--- samba-3.0.9.old/debian/control.in	2004-12-09 17:22:38.000000000 +0100
+++ samba-3.0.9/debian/control.in	2004-12-11 01:23:02.000000000 +0100
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Eloy A. Paris <peloy@debian.org>
 Uploaders: Steve Langasek <vorlon@debian.org>
-Build-Depends: debhelper (>= 4.1.13), libpam0g-dev, libreadline4-dev, libcupsys2-dev (>=1.1.20final+cvs20040330-4), autoconf, libacl1-dev (>= 2.2.11-1), libacl1 (>= 2.2.11-1), libkrb5-dev, libldap2-dev, po-debconf, python2.3-dev, binutils (>= 2.14.90.0.7)
+Build-Depends: debhelper (>= 4.1.13), libpam0g-dev, libreadline4-dev, libcupsys2-dev (>=1.1.20final+cvs20040330-4), autoconf, libacl1-dev (>= 2.2.11-1) [@libacl@], libkrb5-dev, libldap2-dev, po-debconf, python2.3-dev, binutils (>= 2.14.90.0.7), type-handling (>= 0.2.1)
 Standards-Version: 3.5.10
 
 Package: samba
@@ -100,7 +100,7 @@
 Package: smbfs
 Section: otherosfs
 Priority: optional
-Architecture: any
+Architecture: @smbfs@
 Depends: netbase (>= 2.02), samba-common (= ${Source-Version}), ${shlibs:Depends}
 Suggests: smbclient
 Replaces: smbfsx
diff -ur samba-3.0.9.old/debian/rules samba-3.0.9/debian/rules
--- samba-3.0.9.old/debian/rules	2004-12-09 17:22:26.000000000 +0100
+++ samba-3.0.9/debian/rules	2004-12-11 01:23:02.000000000 +0100
@@ -19,9 +19,13 @@
 # cross-building, etc.
 DEB_HOST_GNU_TYPE	:= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
 DEB_BUILD_GNU_TYPE	:= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
+DEB_HOST_GNU_SYSTEM	:= $(shell dpkg-architecture -qDEB_HOST_GNU_SYSTEM)
+DEB_BUILD_GNU_SYSTEM	:= $(shell dpkg-architecture -qDEB_BUILD_GNU_SYSTEM)
 
 export DEB_HOST_GNU_TYPE
 export DEB_BUILD_GNU_TYPE
+export DEB_HOST_GNU_SYSTEM
+export DEB_BUILD_GNU_SYSTEM
 
 
 # Support the DEB_BUILD_OPTIONS variable
@@ -41,26 +45,7 @@
 
 DESTDIR=`pwd`/debian/tmp
 
-patch: patch-stamp
-patch-stamp:
-	dh_testdir
-	if [ ! -f patch-stamp ]; then /bin/sh debian/scripts/patch-source; fi
-	touch patch-stamp
-
-unpatch:
-	dh_testdir
-	if [ -f patch-stamp ]; then /bin/sh debian/scripts/unpatch-source; fi
-	rm -f patch-stamp
-
-configure: patch-stamp configure-stamp
-configure-stamp:
-	dh_testdir
-
-	if [ -f debian/config.cache ]; then \
-		cp -f debian/config.cache source/config.cache; \
-	fi
-
-	[ -f source/Makefile ] || (cd source && CFLAGS="$(CFLAGS)" ./configure \
+conf_args= \
 		--host=$(DEB_HOST_GNU_TYPE) \
 		--build=$(DEB_BUILD_GNU_TYPE) \
 		--cache-file=./config.cache \
@@ -74,7 +59,6 @@
 		--with-piddir=/var/run/samba \
 		--localstatedir=/var \
 		--with-netatalk \
-		--with-smbmount \
 		--with-pam \
 		--with-syslog \
 		--with-utmp \
@@ -84,11 +68,43 @@
 		--with-winbind \
 		--with-msdfs \
 		--with-automount \
-		--with-acl-support \
 		--with-tdbsam \
 		--with-ldap \
+		--with-python=python2.3
+
+ifeq ($(DEB_HOST_GNU_SYSTEM),linux)
+conf_args += \
+		--with-smbmount \
 		--with-quotas \
-		--with-python=python2.3)
+		--with-acl-support
+mount_cifs	= yes
+smbfs		= yes
+else
+conf_args +=	--without-quotas
+mount_cifs	= no
+smbfs		= no
+endif
+
+patch: patch-stamp
+patch-stamp:
+	dh_testdir
+	if [ ! -f patch-stamp ]; then /bin/sh debian/scripts/patch-source; fi
+	touch patch-stamp
+
+unpatch:
+	dh_testdir
+	if [ -f patch-stamp ]; then /bin/sh debian/scripts/unpatch-source; fi
+	rm -f patch-stamp
+
+configure: patch-stamp configure-stamp
+configure-stamp:
+	dh_testdir
+
+	if [ -f debian/config.cache ]; then \
+		cp -f debian/config.cache source/config.cache; \
+	fi
+
+	[ -f source/Makefile ] || (cd source && CFLAGS="$(CFLAGS)" ./configure $(conf_args))
 
 	touch configure-stamp
 
@@ -98,7 +114,9 @@
 
 	$(MAKE) -C source headers
 	$(MAKE) -C source all nsswitch/libnss_wins.so python_ext
+ifeq ($(mount_cifs),yes)
 	$(MAKE) -C source client/mount.cifs
+endif
 
 	touch build-stamp
 
@@ -117,6 +135,11 @@
 		source/bin/libsmbclient.a source/bin/mount.cifs \
 		source/include/stamp-h
 
+	sed \
+	-e "s/@libacl@/`type-handling any linux-gnu`/g" \
+	-e "s/@smbfs@/`type-handling any linux-gnu`/g" \
+	< debian/control.in > debian/control
+
 	dh_clean
 
 install: DH_OPTIONS=
@@ -167,6 +190,7 @@
 	#	right location in the libpam-smbpass package.
 	install -m 0644 source/bin/pam_smbpass.so $(DESTDIR)/lib/security/
 
+ifeq ($(smbfs),yes)
 	# Create the symlinks that will allow us to do "mount -t smbfs ..."
 	# and "mount -t smb ...". Note that the source/script/installbin.sh
 	# tries to create the first symlink, but we have commented
@@ -176,10 +200,13 @@
 	ln -s /usr/bin/smbmount $(DESTDIR)/sbin/mount.smb
 	ln -s smbmount.8 $(DESTDIR)/usr/share/man/man8/mount.smb.8
 	ln -s smbmount.8 $(DESTDIR)/usr/share/man/man8/mount.smbfs.8
+endif
 
+ifeq ($(mount_cifs),yes)
 	# Install mount.cifs and its man page
 	install -m 0755 source/client/mount.cifs $(DESTDIR)/sbin/
 	install -m 0644 docs/manpages/mount.cifs.8 $(DESTDIR)/usr/share/man/man8/
+endif
 
 	# For CUPS to support printing to samba printers, it's necessary
 	#	to make the following symlink (according to
@@ -215,11 +242,11 @@
 
 	# Install the Python modules
 	#
-	# Hmmm... need to figure this out. We have lib.linux-i686-2.2
-	# and lib.linux-i686-2.3 directories. Using only the stuff from
+	# Hmmm... need to figure this out. We have lib.*-i686-2.2
+	# and lib.*-i686-2.3 directories. Using only the stuff from
 	# the 2.3 directory for now. peloy.-
 	#cp source/build/lib.*/samba/*.so $(DESTDIR)/usr/lib/python2.3/site-packages/
-	cp source/build/lib.linux-*-2.3/samba/*.so $(DESTDIR)/usr/lib/python2.3/site-packages/samba/
+	cp source/build/lib.*-*-2.3/samba/*.so $(DESTDIR)/usr/lib/python2.3/site-packages/samba/
 	cp source/python/samba/* $(DESTDIR)/usr/lib/python2.3/site-packages/samba/
 
 	dh_movefiles
@@ -258,7 +285,7 @@
 
 # Build architecture-dependent files here.
 # Pass -a to all debhelper commands in this target to reduce clutter.
-binary-arch: DH_OPTIONS=-a
+binary-arch: DH_OPTIONS=-s
 binary-arch: build install
 	dh_testdir
 	dh_testroot
@@ -297,6 +324,7 @@
 	# Why this is executable, I have NO idea...
 	chmod a-x debian/libsmbclient-dev/usr/include/libsmbclient.h
 
+ifeq ($(smbfs),yes)
 	# You may want to make some executables suid here.
 	# The smbmnt and smbumount binaries should be setuid-root. This
 	#	has security implications because these programs haven't had
@@ -304,6 +332,7 @@
 	#	the setuid bit set. In fact, it is a security hole.
 	chmod u+s debian/smbfs/usr/bin/smbmnt
 	chmod u+s debian/smbfs/usr/bin/smbumount
+endif
 
 	# Set some reasonable default perms for the samba logdir.
 	chmod 0750 debian/samba/var/log/samba/
diff -ur samba-3.0.9.old/source/configure.in samba-3.0.9/source/configure.in
--- samba-3.0.9.old/source/configure.in	2004-11-16 04:03:30.000000000 +0100
+++ samba-3.0.9/source/configure.in	2004-12-11 01:24:06.000000000 +0100
@@ -549,49 +549,13 @@
       fi
     ;;
 
-# Tests for linux LFS support. Need kernel 2.4 and glibc2.2 or greater support.
+# Systems with LFS support.
 #
-    *linux*)
+    linux*-gnu* | gnu* | k*bsd*-gnu)
         AC_MSG_CHECKING([for LFS support])
         old_CPPFLAGS="$CPPFLAGS"
         CPPFLAGS="-D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE $CPPFLAGS"
-       AC_TRY_RUN([
-#include <unistd.h>
-#include <sys/utsname.h>
-#include <string.h>
-#include <stdlib.h>
-main() {
-#if _LFS64_LARGEFILE == 1
-       struct utsname uts;
-       char *release;
-       int major, minor;
-
-       /* Ensure this is glibc 2.2 or higher */
-#if defined(__GLIBC__) && defined(__GLIBC_MINOR__)
-       int libc_major = __GLIBC__;
-       int libc_minor = __GLIBC_MINOR__;
-
-       if (libc_major < 2)
-              exit(1);
-       if (libc_minor < 2)
-              exit(1);
-#endif
-
-       /* Ensure this is kernel 2.4 or higher */
-
-       uname(&uts);
-       release = strdup(uts.release);
-       major = atoi(strsep(&release, "."));
-       minor = atoi(strsep(&release, "."));
-
-       if (major > 2 || (major == 2 && minor > 3))
-               exit(0);
-       exit(1);
-#else
-       exit(1);
-#endif
-}
-], [LINUX_LFS_SUPPORT=yes], [LINUX_LFS_SUPPORT=no], [LINUX_LFS_SUPPORT=cross])
+        LINUX_LFS_SUPPORT=yes
 	CPPFLAGS="$old_CPPFLAGS"
 	if test x$LINUX_LFS_SUPPORT = xyes ; then
 		CPPFLAGS="-D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE $CPPFLAGS"
@@ -1015,7 +979,7 @@
 # 
 #
 case "$host_os" in
-    *linux*)
+    linux*-gnu* | gnu* | k*bsd*-gnu)
        # glibc <= 2.3.2 has a broken getgrouplist
        AC_TRY_RUN([
 #include <unistd.h>
@@ -1147,7 +1111,10 @@
 
   # and these are for particular systems
   case "$host_os" in
-		*linux*)   AC_DEFINE(LINUX,1,[Whether the host os is linux])
+		linux*-gnu* | gnu* | k*bsd*-gnu)
+			case "$host_os" in linux*)
+				AC_DEFINE(LINUX,1,[Whether the host os is linux]) ;;
+			esac
 			BLDSHARED="true"
 			if test "${ac_cv_gnu_ld_no_default_allow_shlib_undefined}" = "yes"; then
 				LDSHFLAGS="-shared -Wl,-Bsymbolic -Wl,--allow-shlib-undefined" 
@@ -4021,7 +3988,7 @@
 	AC_MSG_RESULT(yes);
 
 	case "$host_os" in
-	*linux*)
+	linux*-gnu* | gnu* | k*bsd*-gnu)
 		AC_CACHE_CHECK([for linux sendfile64 support],samba_cv_HAVE_SENDFILE64,[
 		AC_TRY_LINK([#include <sys/sendfile.h>],
 [\
@@ -4255,10 +4222,10 @@
 WINBIND_NSS_LDSHFLAGS=$LDSHFLAGS
 
 case "$host_os" in
-	*linux*)
+	linux*-gnu* | gnu* | k*bsd*-gnu)
 		WINBIND_NSS_EXTRA_OBJS="nsswitch/winbind_nss_linux.o"
 		;;
-	*freebsd5*)
+	freebsd5*)
 		# FreeBSD winbind client is implemented as a wrapper around
 		# the Linux version.
 		WINBIND_NSS_EXTRA_OBJS="nsswitch/winbind_nss_freebsd.o \
