Index: kfreebsd-kernel-di-i386/debian/control
===================================================================
--- kfreebsd-kernel-di-i386/debian/control	(revision 0)
+++ kfreebsd-kernel-di-i386/debian/control	(revision 0)
@@ -0,0 +1,163 @@
+Source: kfreebsd-kernel-di-i386
+Section: debian-installer
+Priority: optional
+Maintainer: Debian Install System Team <debian-boot@lists.debian.org>
+Uploaders: Joey Hess <joeyh@debian.org>, Frans Pop <fjp@debian.org>
+Build-Depends: kernel-wedge (>= 2.20___________________________________________XXXXXXXXXXXX_FIXME), kfreebsd-image-6.0-1-486 [kfreebsd-i386]
+
+Package: kernel-image-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: kernel-image, ufs-modules
+Depends: acpi-modules-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: extra
+Description: Linux kernel binary image for the Debian installer
+ This package contains the Linux kernel image for the Debian installer
+ boot images. It does _not_ provide a usable kernel for your full
+ Debian system.
+
+Package: nic-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: nic-modules
+Depends: kernel-image-6.0-1-486-di, nic-shared-modules-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: standard
+Description: Common NIC drivers
+ This package contains common NIC drivers for the Linux kernel.
+
+Package: nic-shared-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: nic-shared-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: standard
+Description: Shared NIC drivers
+ This package contains NIC drivers needed by combinations of nic-modules,
+ nic-extra-modules, and nic-pcmcia-modules.
+
+Package: ppp-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: ppp-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: optional
+Description: PPP drivers
+ This package contains PPP drivers for the Linux kernel.
+
+Package: ide-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: ide-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: standard
+Description: IDE drivers
+ This package contains IDE drivers for the Linux kernel.
+
+Package: plip-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: plip-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: optional
+Description: PLIP drivers
+ This package contains PLIP drivers for the Linux kernel.
+
+Package: floppy-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: floppy-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: standard
+Description: Floppy driver
+ This package contains the floppy driver for the Linux kernel.
+
+Package: loop-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: loop-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: standard
+Description: Loopback filesystem support
+ This package contains loopback filesystem support for the Linux kernel.
+
+Package: ipv6-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: ipv6-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: extra
+Description: IPv6 driver
+ This package contains the IPv6 driver for the Linux kernel.
+
+Package: ntfs-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: ntfs-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: extra
+Description: NTFS filesystem support
+ This package contains the NTFS file system module for the Linux kernel.
+
+Package: reiserfs-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: reiserfs-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: standard
+Description: Reiser filesystem support
+ This package contains the Reiser filesystem module for the Linux kernel.
+
+Package: fat-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: fat-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: standard
+Description: FAT filesystem support
+ This package contains the FAT and VFAT filesystem modules for the Linux kernel.
+
+Package: mouse-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: mouse-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: extra
+Description: Mouse support
+ This package contains mouse drivers for the Linux kernel.
+
+Package: acpi-modules-6.0-1-486-di
+XC-Package-Type: udeb
+Provides: acpi-modules
+Depends: kernel-image-6.0-1-486-di
+Architecture: kfreebsd-i386
+XB-Kernel-Version: 6.0-1-486
+Section: debian-installer
+Priority: extra
+Description: ACPI support modules
+ This package contains linux kernel modules for ACPI.
Index: kfreebsd-kernel-di-i386/debian/changelog
===================================================================
--- kfreebsd-kernel-di-i386/debian/changelog	(revision 0)
+++ kfreebsd-kernel-di-i386/debian/changelog	(revision 0)
@@ -0,0 +1,5 @@
+kfreebsd-kernel-di-i386 (0.1) UNRELEASED; urgency=low
+
+  * First upload.
+
+ -- Robert Millan <rmh@aybabtu.com>  Thu, 27 Apr 2006 20:50:53 +0200
Index: kfreebsd-kernel-di-i386/debian/copyright
===================================================================
--- kfreebsd-kernel-di-i386/debian/copyright	(revision 0)
+++ kfreebsd-kernel-di-i386/debian/copyright	(revision 0)
@@ -0,0 +1,5 @@
+This package is Copyright (c) 2001-2002 Herbert Xu <herbert@debian.org>
+and Copyright 2001, 2003 by Joey Hess <joeyh@debian.org>.
+
+It is licensed under the terms of the GNU GPL, see
+/usr/share/common-licenses/GPL
Index: kfreebsd-kernel-di-i386/debian/rules
===================================================================
--- kfreebsd-kernel-di-i386/debian/rules	(revision 0)
+++ kfreebsd-kernel-di-i386/debian/rules	(revision 0)
@@ -0,0 +1,2 @@
+#!/usr/bin/make -f
+include /usr/share/kernel-wedge/generic-rules

Property changes on: kfreebsd-kernel-di-i386/debian/rules
___________________________________________________________________
Name: svn:executable
   + *

Index: kfreebsd-kernel-di-i386/debian/control.stub
===================================================================
--- kfreebsd-kernel-di-i386/debian/control.stub	(revision 0)
+++ kfreebsd-kernel-di-i386/debian/control.stub	(revision 0)
@@ -0,0 +1,6 @@
+Source: kfreebsd-kernel-di-i386
+Section: debian-installer
+Priority: optional
+Maintainer: Debian Install System Team <debian-boot@lists.debian.org>
+Uploaders: Joey Hess <joeyh@debian.org>, Frans Pop <fjp@debian.org>
+Build-Depends: kernel-wedge (>= 2.20___________________________________________XXXXXXXXXXXX_FIXME)
Index: kfreebsd-kernel-di-i386/package-list
===================================================================
--- kfreebsd-kernel-di-i386/package-list	(revision 0)
+++ kfreebsd-kernel-di-i386/package-list	(revision 0)
@@ -0,0 +1,29 @@
+# This file is used to build up the control file. The kernel version and
+# "-di" are appended to the package names. Section can be left out. So can
+# architecture, which is derived from the files in the modules directory.
+# It overwrites specifications from /usr/share/kernel-wedge/package-list.
+#
+Package: kernel-image
+Depends: acpi-modules
+Provides: ufs-modules
+
+Package: fat-modules
+Priority: standard
+
+Package: nic-modules
+Depends: kernel-image, nic-shared-modules, firmware-modules, firewire-core-modules
+
+Package: scsi-modules
+Depends: kernel-image, isa-pnp-modules, scsi-core-modules, parport-modules, cdrom-core-modules, firmware-modules
+
+Package: nic-extra-modules
+Depends: kernel-image, isa-pnp-modules, nic-shared-modules, firmware-modules, crc-modules
+
+Package: nic-usb-modules
+Depends: kernel-image, nic-shared-modules, usb-modules, firmware-modules
+
+Package: ppp-modules
+Depends: serial-modules, kernel-image, crc-modules
+
+Package: pcmcia-modules
+Depends: kernel-image, isa-pnp-modules, firmware-modules
Index: kfreebsd-kernel-di-i386/kernel-versions
===================================================================
--- kfreebsd-kernel-di-i386/kernel-versions	(revision 0)
+++ kfreebsd-kernel-di-i386/kernel-versions	(revision 0)
@@ -0,0 +1,2 @@
+# arch		version		flavour		installedname	suffix	build-depends
+kfreebsd-i386	6.0-1		486		6.0-1-486	-	kfreebsd-image-6.0-1-486
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/floppy-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/floppy-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/floppy-modules	(revision 0)
@@ -0,0 +1,2 @@
+#include <kfreebsd/floppy-modules>
+
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/loop-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/loop-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/loop-modules	(revision 0)
@@ -0,0 +1,2 @@
+#include <kfreebsd/loop-modules>
+
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/nic-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/nic-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/nic-modules	(revision 0)
@@ -0,0 +1 @@
+#include <kfreebsd/nic-modules>
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/fat-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/fat-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/fat-modules	(revision 0)
@@ -0,0 +1,2 @@
+#include <kfreebsd/fat-modules>
+
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/kernel-image
===================================================================
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ntfs-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ntfs-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ntfs-modules	(revision 0)
@@ -0,0 +1,2 @@
+ntfs
+ntfs_iconv
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/acpi-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/acpi-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/acpi-modules	(revision 0)
@@ -0,0 +1,2 @@
+#include <kfreebsd/acpi-modules>
+
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/nic-shared-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/nic-shared-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/nic-shared-modules	(revision 0)
@@ -0,0 +1 @@
+miibus ?
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ppp-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ppp-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ppp-modules	(revision 0)
@@ -0,0 +1,2 @@
+#include <kfreebsd/ppp-modules>
+
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ide-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ide-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ide-modules	(revision 0)
@@ -0,0 +1 @@
+#include <kfreebsd/ide-modules>
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/reiserfs-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/reiserfs-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/reiserfs-modules	(revision 0)
@@ -0,0 +1,2 @@
+#include <kfreebsd/reiserfs-modules>
+
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ipv6-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ipv6-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/ipv6-modules	(revision 0)
@@ -0,0 +1,2 @@
+#include <kfreebsd/ipv6-modules>
+
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/plip-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/plip-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/plip-modules	(revision 0)
@@ -0,0 +1,2 @@
+#include <kfreebsd/plip-modules>
+
Index: kfreebsd-kernel-di-i386/modules/kfreebsd-i386/mouse-modules
===================================================================
--- kfreebsd-kernel-di-i386/modules/kfreebsd-i386/mouse-modules	(revision 0)
+++ kfreebsd-kernel-di-i386/modules/kfreebsd-i386/mouse-modules	(revision 0)
@@ -0,0 +1,2 @@
+psm ?
+ums
