diff -Nur sys/osreldate.h sys/osreldate.h
--- src/sys/osreldate.h
+++ src/sys/osreldate.h
@@ -0,0 +1,15 @@
+
+/* kFreeBSD 8.0 headers */
+
+#include <sys/cdefs.h>
+#ifndef _SYS_CDEFS_H_
+#define _SYS_CDEFS_H_
+#define __XSTRING(x)    __STRING(x)     /* expand x, then stringify */
+#define __aligned(x)    __attribute__((__aligned__(x)))
+#define __packed    	 __attribute__((__packed__))
+#endif
+
+#undef __FreeBSD_kernel__
+#define __FreeBSD_kernel__ 8
+#define __FreeBSD_kernel_version 801000
+
