--- src/sys/net80211/ieee80211.h.orig	2006-04-30 04:21:55.000000000 +0200
+++ src/sys/net80211/ieee80211.h	2006-04-30 13:28:41.000000000 +0200
@@ -31,6 +31,7 @@
  *
  * $FreeBSD: src/sys/net80211/ieee80211.h,v 1.5 2004/04/05 17:47:40 sam Exp $
  */
+#include <stdint.h>
 #ifndef _NET80211_IEEE80211_H_
 #define _NET80211_IEEE80211_H_

--- src/sys/net/if_media.h~	2006-08-25 22:53:05.000000000 +0200
+++ src/sys/net/if_media.h	2006-08-25 22:53:05.000000000 +0200
@@ -38,6 +38,8 @@
 #ifndef _NET_IF_MEDIA_H_
 #define _NET_IF_MEDIA_H_
 
+#include <stdint.h>
+
 /*
  * Prototypes and definitions for BSD/OS-compatible network interface
  * media selection.
