--- sys/sys/exec.h~	2006-05-05 11:35:50.000000000 +0200
+++ sys/sys/exec.h	2006-05-05 11:35:50.000000000 +0200
@@ -56,7 +56,10 @@
 /*
  * Address of ps_strings structure (in user space).
  */
+/* Applications should use the kern.ps_strings sysctl instead of PS_STRINGS, so ... */
+#if 0    
 #define	PS_STRINGS	(USRSTACK - sizeof(struct ps_strings))
+#endif
 #define SPARE_USRSPACE	4096
 
 struct image_params;
