diff -ur sys.orig/sys/tty.h sys/sys/tty.h
--- sys.orig/sys/tty.h	2004-07-15 22:47:41.000000000 +0200
+++ sys/sys/tty.h	2004-11-30 01:13:49.000000000 +0100
@@ -48,6 +48,8 @@
 
 #include <sys/termios.h>
 #include <sys/queue.h>
+#include <sys/types.h>
+#include <sys/ioctl.h> /* winsize struct */
 #include <sys/selinfo.h>
 #include <sys/_lock.h>
 #include <sys/_mutex.h>
@@ -96,8 +98,6 @@
 	struct	pgrp *t_pgrp;		/* Foreground process group. */
 	struct	session *t_session;	/* Enclosing session. */
 	struct  sigio *t_sigio;		/* Information for async I/O. */
-	struct	selinfo t_rsel;		/* Tty read/oob select. */
-	struct	selinfo t_wsel;		/* Tty write select. */
 	struct	termios t_termios;	/* Termios state. */
 	struct	winsize t_winsize;	/* Window size. */
 	void	*t_sc;			/* driver private softc pointer. */
