Index: kfreebsd-source-8.0/sys/sys/tty.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/sys/tty.h	2009-11-12 15:49:03.000000000 +0000
+++ kfreebsd-source-8.0/sys/sys/tty.h	2009-11-12 16:28:07.000000000 +0000
@@ -34,6 +34,8 @@
 
 #include <sys/param.h>
 #include <sys/queue.h>
+#include <sys/types.h>
+#include <sys/ioctl.h> /* winsize struct */
 #include <sys/lock.h>
 #include <sys/mutex.h>
 #include <sys/condvar.h>
