--- sys/vm.old/vm_param.h	2005-09-04 12:22:36.000000000 -0400
+++ sys/vm/vm_param.h	2005-09-04 14:13:45.000000000 -0400
@@ -67,6 +67,7 @@
 #ifndef	_VM_PARAM_
 #define	_VM_PARAM_
 
+#include <sys/types.h>
 #include <machine/vmparam.h>
 
 /*
@@ -106,7 +107,7 @@
  */
 #define XSWDEV_VERSION	1
 struct xswdev {
-	u_int	xsw_version;
+	unsigned int	xsw_version;
 	dev_t	xsw_dev;
 	int	xsw_flags;
 	int	xsw_nblks;
