--- sys/sys/param.h~	2004-12-02 01:04:55.000000000 +0100
+++ sys/sys/param.h	2004-12-02 02:59:31.000000000 +0100
@@ -56,8 +56,7 @@
  *
  * scheme is:  <major><two digit minor><0 if release branch, otherwise 1>xx
  */
-#undef __FreeBSD_version
-#define __FreeBSD_version 503001	/* Master, propagated to newvers */
+#include <osreldate.h>
 
 /* Some inet code expects that this file defines the 'u_int32_t' type.  */
 #include <sys/types.h>
