--- a/sys/i386/include/cpufunc.h
+++ b/sys/i386/include/cpufunc.h
@@ -42,6 +42,8 @@
 
 #include <sys/types.h>
 
+#include <stdint.h>
+
 #ifndef _SYS_CDEFS_H_
 #error this file needs sys/cdefs.h as a prerequisite
 #endif
@@ -697,11 +699,6 @@
 void	enable_intr(void);
 void	halt(void);
 void	ia32_pause(void);
-unsigned char	inb(unsigned int port);
-unsigned int	inl(unsigned int port);
-void	insb(unsigned int port, void *addr, size_t count);
-void	insl(unsigned int port, void *addr, size_t count);
-void	insw(unsigned int port, void *addr, size_t count);
 register_t	intr_disable(void);
 void	intr_restore(register_t ef);
 void	invd(void);
@@ -724,12 +721,6 @@
 void	load_fs(unsigned short sel);
 void	load_gs(unsigned short sel);
 void	ltr(unsigned short sel);
-void	outb(unsigned int port, unsigned char data);
-void	outl(unsigned int port, unsigned int data);
-void	outsb(unsigned int port, const void *addr, size_t count);
-void	outsl(unsigned int port, const void *addr, size_t count);
-void	outsw(unsigned int port, const void *addr, size_t count);
-void	outw(unsigned int port, unsigned short data);
 unsigned int	rcr0(void);
 unsigned int	rcr2(void);
 unsigned int	rcr3(void);
