Index: kfreebsd-source-8.0/sys/cam/scsi/scsi_all.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/cam/scsi/scsi_all.h	2009-11-12 15:48:27.000000000 +0000
+++ kfreebsd-source-8.0/sys/cam/scsi/scsi_all.h	2009-11-12 16:33:38.000000000 +0000
@@ -25,6 +25,7 @@
 #define _SCSI_SCSI_ALL_H 1
 
 #include <sys/cdefs.h>
+#include <stdint.h>
 
 #ifdef _KERNEL
 /*
Index: kfreebsd-source-8.0/sys/i386/include/md_var.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/i386/include/md_var.h	2009-11-12 15:48:49.000000000 +0000
+++ kfreebsd-source-8.0/sys/i386/include/md_var.h	2009-11-12 16:33:38.000000000 +0000
@@ -32,6 +32,8 @@
 #ifndef _MACHINE_MD_VAR_H_
 #define	_MACHINE_MD_VAR_H_
 
+#include <sys/_types.h>
+
 /*
  * Miscellaneous machine-dependent declarations.
  */
Index: kfreebsd-source-8.0/sys/amd64/include/md_var.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/amd64/include/md_var.h	2009-11-12 15:48:23.000000000 +0000
+++ kfreebsd-source-8.0/sys/amd64/include/md_var.h	2009-11-12 16:33:38.000000000 +0000
@@ -32,6 +32,8 @@
 #ifndef _MACHINE_MD_VAR_H_
 #define	_MACHINE_MD_VAR_H_
 
+#include <sys/_types.h>
+
 /*
  * Miscellaneous machine-dependent declarations.
  */
Index: kfreebsd-source-8.0/sys/sys/_mutex.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/sys/_mutex.h	2009-11-12 15:49:02.000000000 +0000
+++ kfreebsd-source-8.0/sys/sys/_mutex.h	2009-11-12 16:33:38.000000000 +0000
@@ -31,6 +31,8 @@
 #ifndef _SYS__MUTEX_H_
 #define	_SYS__MUTEX_H_
 
+#include <stdint.h>
+
 /*
  * Sleep/spin mutex.
  */
Index: kfreebsd-source-8.0/sys/sys/_timeval.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/sys/_timeval.h	2009-11-12 15:49:02.000000000 +0000
+++ kfreebsd-source-8.0/sys/sys/_timeval.h	2009-11-12 16:33:38.000000000 +0000
@@ -30,6 +30,7 @@
 #define _SYS__TIMEVAL_H_
 
 #include <sys/_types.h>
+#include <sys/time.h>
 
 #ifndef _SUSECONDS_T_DECLARED
 typedef	__suseconds_t	suseconds_t;
@@ -41,12 +42,4 @@
 #define	_TIME_T_DECLARED
 #endif
 
-/*
- * Structure returned by gettimeofday(2) system call, and used in other calls.
- */
-struct timeval {
-	time_t		tv_sec;		/* seconds */
-	suseconds_t	tv_usec;	/* and microseconds */
-};
-
 #endif /* !_SYS__TIMEVAL_H_ */
Index: kfreebsd-source-8.0/sys/sys/bio.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/sys/bio.h	2009-11-12 15:49:02.000000000 +0000
+++ kfreebsd-source-8.0/sys/sys/bio.h	2009-11-12 16:33:38.000000000 +0000
@@ -39,6 +39,15 @@
 #define	_SYS_BIO_H_
 
 #include <sys/queue.h>
+#include <stdint.h>
+/* FIXME: this struct is copied from BSDish sys/time.h.  It should be somewhere else. (maybe Glibc?) */
+#ifndef bintime
+#define bintime bintime
+struct bintime {
+        time_t  sec;
+        uint64_t frac;
+};
+#endif
 
 struct disk;
 struct bio;
Index: kfreebsd-source-8.0/sys/sys/devicestat.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/sys/devicestat.h	2009-11-12 15:49:02.000000000 +0000
+++ kfreebsd-source-8.0/sys/sys/devicestat.h	2009-11-12 16:33:38.000000000 +0000
@@ -33,6 +33,9 @@
 
 #include <sys/queue.h>
 #include <sys/time.h>
+#include <sys/_types.h>
+#include <sys/resource.h>
+#include <stdint.h>
 
 /*
  * XXX: Should really be SPECNAMELEN
Index: kfreebsd-source-8.0/sys/sys/sbuf.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/sys/sbuf.h	2009-11-12 15:49:03.000000000 +0000
+++ kfreebsd-source-8.0/sys/sys/sbuf.h	2009-11-12 16:34:29.000000000 +0000
@@ -64,10 +64,8 @@
 int		 sbuf_bcpy(struct sbuf *, const void *, size_t);
 int		 sbuf_cat(struct sbuf *, const char *);
 int		 sbuf_cpy(struct sbuf *, const char *);
-int		 sbuf_printf(struct sbuf *, const char *, ...)
-	__printflike(2, 3);
-int		 sbuf_vprintf(struct sbuf *, const char *, __va_list)
-	__printflike(2, 0);
+int		 sbuf_printf(struct sbuf *, const char *, ...);
+int		 sbuf_vprintf(struct sbuf *, const char *, va_list);
 int		 sbuf_putc(struct sbuf *, int);
 int		 sbuf_trim(struct sbuf *);
 int		 sbuf_overflowed(struct sbuf *);
Index: kfreebsd-source-8.0/sys/sys/timespec.h
===================================================================
--- kfreebsd-source-8.0.orig/sys/sys/timespec.h	2009-11-12 15:49:03.000000000 +0000
+++ kfreebsd-source-8.0/sys/sys/timespec.h	2009-11-12 16:33:38.000000000 +0000
@@ -43,10 +43,7 @@
 #define	_TIME_T_DECLARED
 #endif
 
-struct timespec {
-	time_t	tv_sec;		/* seconds */
-	long	tv_nsec;	/* and nanoseconds */
-};
+#include <sys/time.h>
 
 #if __BSD_VISIBLE
 #define	TIMEVAL_TO_TIMESPEC(tv, ts)					\
