--- sys/sys/param.h~	2005-08-22 23:18:42.000000000 +0200
+++ sys/sys/param.h	2005-08-23 19:08:26.000000000 +0200
@@ -42,10 +42,6 @@
 #define NULL 0
 #endif
 
-#define	BSD	199506		/* System version (year & month). */
-#define BSD4_3	1
-#define BSD4_4	1
-
 /* 
  * __FreeBSD_version numbers are documented in the Porter's Handbook.
  * If you bump the version for any reason, you should update the documentation
