--- sys/sys/_types.h.orig	2005-10-11 01:30:09.000000000 +0200
+++ sys/sys/_types.h	2005-12-13 20:13:11.000000000 +0100
@@ -82,4 +82,59 @@
 
 typedef	__uint32_t	__fixpt_t;	/* fixed point number */
 
+#ifndef _CLOCK_T_DECLARED
+typedef	__clock_t	clock_t;
+#define	_CLOCK_T_DECLARED
+#endif
+
+#ifndef _FFLAGS_T_DECLARED
+typedef	__fflags_t	fflags_t;	/* file flags */
+#define	_FFLAGS_T_DECLARED
+#endif
+
+typedef	__fixpt_t	fixpt_t;	/* fixed point number */
+
+#ifndef _IN_ADDR_T_DECLARED
+typedef	__uint32_t	in_addr_t;	/* base type for internet address */
+#define	_IN_ADDR_T_DECLARED
+#endif
+
+#ifndef _IN_PORT_T_DECLARED
+typedef	__uint16_t	in_port_t;
+#define	_IN_PORT_T_DECLARED
+#endif
+
+#ifndef _LWPID_T_DECLARED
+typedef	__lwpid_t	lwpid_t;	/* Thread ID (a.k.a. LWP) */
+#define	_LWPID_T_DECLARED
+#endif
+
+#ifndef _NLINK_T_DECLARED
+typedef	__nlink_t	nlink_t;	/* link count */
+#define	_NLINK_T_DECLARED
+#endif
+
+#ifndef _RLIM_T_DECLARED
+typedef	__rlim_t	rlim_t;		/* resource limit */
+#define	_RLIM_T_DECLARED
+#endif
+
+typedef	__segsz_t	segsz_t;	/* segment size (in pages) */
+
+typedef	__u_register_t	u_register_t;
+
+typedef	__vm_offset_t	vm_offset_t;
+typedef	__vm_ooffset_t	vm_ooffset_t;
+typedef	__vm_paddr_t	vm_paddr_t;
+typedef	__vm_pindex_t	vm_pindex_t;
+typedef	__vm_size_t	vm_size_t;
+
+#ifndef bintime 
+#define bintime bintime 
+struct bintime { 
+        time_t  sec; 
+        __uint64_t frac; 
+}; 
+#endif 
+                
 #endif /* !_SYS__TYPES_H_ */
--- sys/sys/proc.h.orig	2005-10-11 01:30:09.000000000 +0200
+++ sys/sys/proc.h	2005-12-13 20:09:19.000000000 +0100
@@ -38,6 +38,7 @@
 #ifndef _SYS_PROC_H_
 #define	_SYS_PROC_H_
 
+#include <sys/_types.h> 
 #include <sys/callout.h>		/* For struct callout. */
 #include <sys/event.h>			/* For struct klist. */
 #ifndef _KERNEL
@@ -61,11 +62,6 @@
 
 #include <machine/proc.h>		/* Machine-dependent proc substruct. */
 
-#ifdef __FreeBSD__
-#include <sys/types.h>			/* vm_offset_t */
-#else
-#include <machine/_types.h>		/* __vm_offset_t */
-#endif
 
 /*
  * One structure allocated per session.
