--- sys/amd64/include/param.h	2005-11-19 18:53:09.000000000 +0100
+++ sys/amd64/include/param.h	2006-01-16 13:14:18.000000000 +0100
@@ -71,6 +71,10 @@
 #define	_MACHINE_ARCH	amd64
 #endif
 
+#ifndef EXEC_PAGESIZE
+#define EXEC_PAGESIZE   4096
+#endif
+
 #ifndef _NO_NAMESPACE_POLLUTION
 
 #ifndef _MACHINE_PARAM_H_
--- sys/i386/include/param.h	2005-11-19 19:23:51.000000000 +0100
+++ sys/i386/include/param.h	2006-01-16 13:16:22.000000000 +0100
@@ -56,6 +56,10 @@
 #define	_MACHINE_ARCH	i386
 #endif
 
+#ifndef EXEC_PAGESIZE
+#define EXEC_PAGESIZE   4096
+#endif
+
 #ifndef _NO_NAMESPACE_POLLUTION
 
 #ifndef _MACHINE_PARAM_H_
