---
 sys/sys/iconv.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/sys/sys/iconv.h
+++ b/sys/sys/iconv.h
@@ -94,7 +94,7 @@
 #define	ENCODING_UNICODE	"UTF-16BE"
 #define	KICONV_VENDOR_MICSFT	1	/* Microsoft Vendor Code for quirk */
 
-int   kiconv_add_xlat_table(const char *, const char *, const u_char *);
+int   kiconv_add_xlat_table(const char *, const char *, const unsigned char *);
 int   kiconv_add_xlat16_cspair(const char *, const char *, int);
 int   kiconv_add_xlat16_cspairs(const char *, const char *);
 int   kiconv_add_xlat16_table(const char *, const char *, const void *, int);
