--- sys/sys/ktrace.h	2006-06-25 11:40:55.000000000 +0200
+++ sys/sys/ktrace.h	2006-11-20 10:32:42.000000000 +0100
@@ -33,6 +33,19 @@
 #ifndef _SYS_KTRACE_H_
 #define _SYS_KTRACE_H_
 
+#include <features.h>
+
+#include <stdint.h>
+
+/* Define MAXCOMLEN.  */
+#include <sys/param.h>
+
+/* Define enum uio_rw  */
+#include <sys/uio.h>
+
+/* Define __sighandler_t, __sigset_t.  */
+#include <signal.h>
+
 /*
  * operations to ktrace system call  (KTROP(op))
  */
