--- sys/sys/conf.h~	2006-12-04 09:29:27.000000000 +0100
+++ sys/sys/conf.h	2006-12-04 09:29:27.000000000 +0100
@@ -43,6 +43,7 @@
 #ifdef _KERNEL
 #include <sys/eventhandler.h>
 #else
+#include <sys/param.h>		/* SPECNAMELEN */
 #include <sys/queue.h>
 #endif
 
@@ -255,11 +255,11 @@
 void	dev_rel(struct cdev *dev);
 void	dev_strategy(struct cdev *dev, struct buf *bp);
 struct cdev *make_dev(struct cdevsw *_devsw, int _minor, uid_t _uid, gid_t _gid,
-		int _perms, const char *_fmt, ...) __printflike(6, 7);
+		int _perms, const char *_fmt, ...);
 struct cdev *make_dev_cred(struct cdevsw *_devsw, int _minor,
 		struct ucred *_cr, uid_t _uid, gid_t _gid, int _perms,
-		const char *_fmt, ...) __printflike(7, 8);
-struct cdev *make_dev_alias(struct cdev *_pdev, const char *_fmt, ...) __printflike(2, 3);
+		const char *_fmt, ...);
+struct cdev *make_dev_alias(struct cdev *_pdev, const char *_fmt, ...);
 int	dev2unit(struct cdev *_dev);
 void	dev_lock(void);
 void	dev_unlock(void);
