---
 sys/contrib/pf/net/pfvar.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/sys/contrib/pf/net/pfvar.h
+++ b/sys/contrib/pf/net/pfvar.h
@@ -39,13 +39,14 @@
 #include <sys/tree.h>
 #if 1
 #include <sys/lock.h>
-#include <sys/sx.h>
 #else
 #include <sys/rwlock.h>
 #endif
 
+#include <net/if.h>
 #include <net/radix.h>
 #include <net/route.h>
+#ifdef _KERNEL
 #if 1
 #include <net/if_clone.h>
 #include <net/pf_mtag.h>
@@ -53,6 +54,7 @@
 #else
 #include <netinet/ip_ipsp.h>
 #endif
+#endif
 
 #if 1
 #include <netinet/in.h>
