--- sys/contrib/pf/net/pfvar.h	2007-01-11 16:26:28.000000000 +0100
+++ sys/contrib/pf/net/pfvar.h	2007-01-11 16:26:28.000000000 +0100
@@ -39,14 +39,17 @@
 #include <sys/queue.h>
 #include <sys/tree.h>
 
+#include <net/if.h>
 #include <net/radix.h>
 #include <net/route.h>
+#ifdef _KERNEL
 #ifdef __FreeBSD_kernel__
 #include <net/if_clone.h>
 #include <vm/uma.h>
 #else
 
 #endif
+#endif
 
 #ifdef __FreeBSD_kernel__
 #include <netinet/in.h>
