--- a/sys/contrib/pf/net/pfvar.h
+++ b/sys/contrib/pf/net/pfvar.h
@@ -40,13 +40,14 @@
 #include <sys/tree.h>
 #if 1
 #include <sys/lock.h>
-#include <sys/sx.h>
 #else
 #include <sys/rwlock.h>
 #endif
 
+#include <net/if.h>
 #include <net/radix.h>
 #include <net/route.h>
+#ifdef _KERNEL
 #if 1
 #include <net/if_clone.h>
 #include <net/pf_mtag.h>
@@ -54,6 +55,7 @@
 #else
 
 #endif
+#endif
 
 #if 1
 #include <netinet/in.h>
