Index: sys/contrib/pf/net/pfvar.h
===================================================================
--- sys/contrib/pf/net/pfvar.h.orig
+++ sys/contrib/pf/net/pfvar.h
@@ -45,8 +45,10 @@
 #include <sys/rwlock.h>
 #endif
 
+#include <net/if.h>
 #include <net/radix.h>
 #include <net/route.h>
+#ifdef _KERNEL
 #ifdef __FreeBSD_kernel__
 #include <net/if_clone.h>
 #include <net/pf_mtag.h>
@@ -54,6 +56,7 @@
 #else
 
 #endif
+#endif
 
 #ifdef __FreeBSD_kernel__
 #include <netinet/in.h>
