Index: sys/ufs/ufs/quota.h
===================================================================
--- sys/ufs/ufs/quota.h.orig
+++ sys/ufs/ufs/quota.h
@@ -205,7 +205,6 @@
 int	quotaon(struct thread *td, struct mount *, int, void *);
 int	setquota(struct thread *, struct mount *, u_long, int, void *);
 int	setuse(struct thread *, struct mount *, u_long, int, void *);
-vfs_quotactl_t ufs_quotactl;
 
 #else /* !_KERNEL */
 
