--- src/sys/sys/pcpu.h~
+++ src/sys/sys/pcpu.h
@@ -41,6 +41,8 @@
 #include <sys/vmmeter.h>
 #include <sys/resource.h>
 #include <machine/pcpu.h>
+#include <sys/_types.h>
+#include <stdint.h>
 
 struct pcb;
 struct thread;
@@ -57,7 +59,7 @@
 	struct thread	*pc_fpcurthread;	/* Fp state owner */
 	struct thread	*pc_deadthread;		/* Zombie thread or NULL */
 	struct pcb	*pc_curpcb;		/* Current pcb */
-	uint64_t	pc_switchtime;
+	u_int64_t	pc_switchtime;
 	int		pc_switchticks;
 	u_int		pc_cpuid;		/* This cpu number */
 	cpumask_t	pc_cpumask;		/* This cpu mask */
