---
 sys/sys/pcpu.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/sys/sys/pcpu.h
+++ b/sys/sys/pcpu.h
@@ -42,6 +42,8 @@
 #include <sys/vmmeter.h>
 #include <sys/resource.h>
 #include <machine/pcpu.h>
+#include <sys/_types.h>
+#include <stdint.h>
 
 #define	DPCPU_SETNAME		"set_pcpu"
 #define	DPCPU_SYMPREFIX		"pcpu_entry_"
@@ -160,7 +162,7 @@
 	struct thread	*pc_fpcurthread;	/* Fp state owner */
 	struct thread	*pc_deadthread;		/* Zombie thread or NULL */
 	struct pcb	*pc_curpcb;		/* Current pcb */
-	uint64_t	pc_switchtime;		/* cpu_ticks() at last csw */
+	u_int64_t	pc_switchtime;		/* cpu_ticks() at last csw */
 	int		pc_switchticks;		/* `ticks' at last csw */
 	u_int		pc_cpuid;		/* This cpu number */
 	STAILQ_ENTRY(pcpu) pc_allcpu;
