---
 sys/sys/ucred.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/sys/sys/ucred.h
+++ b/sys/sys/ucred.h
@@ -48,7 +48,7 @@
  */
 #if defined(_KERNEL) || defined(_WANT_UCRED)
 struct ucred {
-	u_int	cr_ref;			/* reference count */
+	__u_int	cr_ref;			/* reference count */
 #define	cr_startcopy cr_uid
 	uid_t	cr_uid;			/* effective user id */
 	uid_t	cr_ruid;		/* real user id */
@@ -60,7 +60,7 @@
 	struct uidinfo	*cr_ruidinfo;	/* per ruid resource consumption */
 	struct prison	*cr_prison;	/* jail(2) */
 	struct loginclass	*cr_loginclass; /* login class */
-	u_int		cr_flags;	/* credential flags */
+	__u_int		cr_flags;	/* credential flags */
 	void 		*cr_pspare2[2];	/* general use 2 */
 #define	cr_endcopy	cr_label
 	struct label	*cr_label;	/* MAC label */
@@ -83,7 +83,7 @@
  * This is the external representation of struct ucred.
  */
 struct xucred {
-	u_int	cr_version;		/* structure layout version */
+	__u_int	cr_version;		/* structure layout version */
 	uid_t	cr_uid;			/* effective user id */
 	short	cr_ngroups;		/* number of groups */
 	gid_t	cr_groups[XU_NGROUPS];	/* groups */
