
Backward compatibility for CTLTYPE_QUAD removal in r217616.

--- a/sys/sys/sysctl.h
+++ b/sys/sys/sysctl.h
@@ -165,6 +165,7 @@
 #define	CTLTYPE_INT	2	/* name describes an integer */
 #define	CTLTYPE_STRING	3	/* name describes a string */
 #define	CTLTYPE_S64	4	/* name describes a signed 64-bit number */
+#define CTLTYPE_QUAD	CTLTYPE_S64
 #define	CTLTYPE_OPAQUE	5	/* name describes a structure */
 #define	CTLTYPE_STRUCT	CTLTYPE_OPAQUE	/* name describes a structure */
 #define	CTLTYPE_UINT	6	/* name describes an unsigned integer */
