
In upstream HEAD (SVN rev 233096)

--- a/sys/netinet/in_pcb.h
+++ b/sys/netinet/in_pcb.h
@@ -364,6 +364,7 @@
 	void 			*ipi_pspare[2];
 };
 
+#ifdef _KERNEL
 /*
  * Connection groups hold sets of connections that have similar CPU/thread
  * affinity.  Each connection belongs to exactly one connection group.
@@ -406,7 +407,6 @@
 #define	INP_WLOCK_ASSERT(inp)	rw_assert(&(inp)->inp_lock, RA_WLOCKED)
 #define	INP_UNLOCK_ASSERT(inp)	rw_assert(&(inp)->inp_lock, RA_UNLOCKED)
 
-#ifdef _KERNEL
 /*
  * These locking functions are for inpcb consumers outside of sys/netinet,
  * more specifically, they were added for the benefit of TOE drivers. The
