---
 usr.sbin/moused/moused.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/usr.sbin/moused/moused.c
+++ b/usr.sbin/moused/moused.c
@@ -45,10 +45,10 @@
  **/
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include <sys/param.h>
 #include <sys/consio.h>
+#include <sys/ttycom.h>
 #include <sys/mouse.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
@@ -533,8 +533,7 @@ static void	hup(int sig);
 static void	cleanup(int sig);
 static void	pause_mouse(int sig);
 static void	usage(void);
-static void	log_or_warn(int log_pri, int errnum, const char *fmt, ...)
-		    __printflike(3, 4);
+static void	log_or_warn(int log_pri, int errnum, const char *fmt, ...);
 
 static int	r_identify(void);
 static const char *r_if(int type);
