Author: aurel32
Status: in BTS

diff -u acpica-unix-20050930/debian/control acpica-unix-20050930/debian/control
--- acpica-unix-20050930/debian/control
+++ acpica-unix-20050930/debian/control
@@ -6,7 +6,7 @@
 Standards-Version: 3.6.2
 
 Package: iasl
-Architecture: i386 ia64 amd64
+Architecture: i386 ia64 amd64 kfreebsd-i386
 Depends: ${shlibs:Depends}
 Description: Intel ASL compiler/decompiler
  iASL compiles ASL (ACPI Source Language) into AML (ACPI Machine
only in patch2:
unchanged:
--- acpica-unix-20050930.orig/include/platform/acenv.h
+++ acpica-unix-20050930/include/platform/acenv.h
@@ -211,7 +211,7 @@
 #elif defined(MSDOS)        /* Must appear after WIN32 and WIN64 check */
 #include "acdos16.h"
 
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include "acfreebsd.h"
 
 #elif defined(__NetBSD__)
