Author: aurel32
Status: in BTS

diff -u adplay-1.4/debian/control adplay-1.4/debian/control
--- adplay-1.4/debian/control
+++ adplay-1.4/debian/control
@@ -2,7 +2,7 @@
 Section: sound
 Priority: optional
 Maintainer: Daniel Baumann <daniel.baumann@panthera-systems.net>
-Build-Depends: debhelper (>= 4.0.0), libadplug-dev, libbinio-dev, libesd0-dev, libsdl-sound1.2-dev, libasound2-dev
+Build-Depends: debhelper (>= 4.0.0), libadplug-dev, libbinio-dev, libesd0-dev, libsdl-sound1.2-dev, libasound2-dev [!kfreebsd-i386]
 Standards-Version: 3.6.2
 
 Package: adplay

