Author: aurel32
Status: in BTS

diff -u allegro4.2-4.2.0/debian/rules allegro4.2-4.2.0/debian/rules
--- allegro4.2-4.2.0/debian/rules
+++ allegro4.2-4.2.0/debian/rules
@@ -10,7 +10,7 @@
 fullver=4.2.0
 
 # SVGAlib is only for x86
-ifeq ($(shell dpkg-architecture -qDEB_BUILD_ARCH_CPU),i386)
+ifeq ($(shell dpkg-architecture -qDEB_BUILD_ARCH),i386)
 ARCH_CONFIG_FLAGS = 
 ARCH_PACKAGES = liballegro$(ver)-plugin-svgalib
 else
diff -u allegro4.2-4.2.0/debian/control allegro4.2-4.2.0/debian/control
--- allegro4.2-4.2.0/debian/control
+++ allegro4.2-4.2.0/debian/control
@@ -2,7 +2,7 @@
 Section: devel
 Priority: optional
 Maintainer: Sam Hocevar (Debian packages) <sam+deb@zoy.org>
-Build-Depends: debhelper (>= 4.0), texinfo, libesd0-dev, libxext-dev, libx11-dev, libxpm-dev, libxt-dev, svgalibg1-dev [i386], libasound2-dev (>= 0.9), libartsc0-dev, libjack0.100.0-dev, libxxf86dga-dev, libxxf86vm-dev
+Build-Depends: debhelper (>= 4.0), texinfo, libesd0-dev, libxext-dev, libx11-dev, libxpm-dev, libxt-dev, svgalibg1-dev [i386], libasound2-dev (>= 0.9) [!kfreebsd-i386], libartsc0-dev, libjack0.100.0-dev, libxxf86dga-dev, libxxf86vm-dev
 Standards-Version: 3.6.2.1
 
 Package: liballegro4.2
