diff -Nur apt-0.6.38/methods/http.cc apt-0.6.38+kbsd/methods/http.cc
--- apt-0.6.38/methods/http.cc	2005-05-24 23:33:10.000000000 +0200
+++ apt-0.6.38+kbsd/methods/http.cc	2005-07-10 23:56:06.000000000 +0200
@@ -787,7 +787,8 @@
 {
    if (File != 0)
    {
-      SetNonBlock(File->Fd(),false);
+      if (File->Name().compare("/dev/null"))
+         SetNonBlock(File->Fd(),false);
       if (Srv->In.WriteSpace() == false)
 	 return true;
       
@@ -815,7 +816,8 @@
    // Dump the buffer to the file
    if (Srv->State == ServerState::Data)
    {
-      SetNonBlock(File->Fd(),false);
+      if (File->Name().compare("/dev/null"))
+         SetNonBlock(File->Fd(),false);
       while (Srv->In.WriteSpace() == true)
       {
 	 if (Srv->In.Write(File->Fd()) == false)
