diff -u asis-3.15p/debian/rules asis-3.15p/debian/rules
--- asis-3.15p/debian/rules
+++ asis-3.15p/debian/rules
@@ -9,8 +9,8 @@
 # This has to be exported to make some magic below work.
 export DH_OPTIONS
 
-ARCH = $(shell dpkg-architecture -qDEB_HOST_ARCH)
-OS = $(ARCH)-linux
+ARCH = $(shell dpkg-architecture -qDEB_HOST_ARCH_CPU)
+OS = $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
 
 VERSION := $(shell dpkg-parsechangelog | grep "^Version: " | sed 's/^Version: \(.*\)-\(.*\)/\1/')
 REVISION := $(shell dpkg-parsechangelog | grep "^Version: " | sed 's/^Version: \(.*\)-\(.*\)/\2/')
diff -u asis-3.15p/debian/control asis-3.15p/debian/control
--- asis-3.15p/debian/control
+++ asis-3.15p/debian/control
@@ -7,7 +7,7 @@
 Build-Depends: gnat (>= 3.15p-17), debhelper (>= 4.1), libgnatvsn-dev (>= 3.15p-17)
 
 Package: libasis-3.15p-1-dev
-Architecture: i386 powerpc sparc
+Architecture: i386 powerpc sparc kfreebsd-i386
 Section: libdevel
 Conflicts: libasis-3.14p-1-dev
 Replaces: libasis-3.14p-1-dev
@@ -20,7 +20,7 @@
  This package lets you build ASIS applications with GNAT.
 
 Package: asis-programs
-Architecture: i386 powerpc sparc
+Architecture: i386 powerpc sparc kfreebsd-i386
 Section: devel
 Depends: ${shlibs:Depends}
 Description: Ada Semantic Interface Specification (ASIS) example programs
@@ -32,7 +32,7 @@
  gnatstub) which are useful to understand or optimize Ada programs.
 
 Package: libasis-3.15p-1
-Architecture: i386 powerpc sparc
+Architecture: i386 powerpc sparc kfreebsd-i386
 Section: libs
 Depends: ${shlibs:Depends}
 Description: Ada Semantic Interface Specification (ASIS) runtime library
