diff -Nru /tmp/NknjgWwSPm/avifile-0.7.44.20051021/debian/control /tmp/aXXmQEgmMf/avifile-0.7.44.20051021/debian/control
--- /tmp/NknjgWwSPm/avifile-0.7.44.20051021/debian/control	2005-10-21 14:21:48.000000000 +0200
+++ /tmp/aXXmQEgmMf/avifile-0.7.44.20051021/debian/control	2005-10-23 22:56:15.000000000 +0200
@@ -70,7 +70,7 @@
  Homepage: http://avifile.sourceforge.net
 
 Package: avifile-win32-plugin
-Architecture: i386 amd64
+Architecture: kfreebsd-i386 i386 amd64
 Section: contrib/libs
 Depends: ${shlibs:Depends}
 Description: Win32 audio/video plugin for libavifile
diff -Nru /tmp/NknjgWwSPm/avifile-0.7.44.20051021/debian/rules /tmp/aXXmQEgmMf/avifile-0.7.44.20051021/debian/rules
--- /tmp/NknjgWwSPm/avifile-0.7.44.20051021/debian/rules	2005-10-21 14:22:50.000000000 +0200
+++ /tmp/aXXmQEgmMf/avifile-0.7.44.20051021/debian/rules	2005-10-23 21:46:32.000000000 +0200
@@ -201,7 +201,7 @@
 #        -ptmp
 #DH2	@dh_movefiles
 	dh_compress
-	@chmod +x debian/$(player)/usr/share/$(aviversion)/scripts/*
+#	@chmod +x debian/$(player)/usr/share/$(aviversion)/scripts/*
 	dh_fixperms -Xkv4lsetup -Xavicap-boot -Xavicap-setnvram -Xavicap-shutdown
 	dh_strip
 
diff -Nru /tmp/NknjgWwSPm/avifile-0.7.44.20051021/samples/Makefile.am /tmp/aXXmQEgmMf/avifile-0.7.44.20051021/samples/Makefile.am
--- /tmp/NknjgWwSPm/avifile-0.7.44.20051021/samples/Makefile.am	2002-11-14 19:51:00.000000000 +0100
+++ /tmp/aXXmQEgmMf/avifile-0.7.44.20051021/samples/Makefile.am	2005-10-23 20:35:03.000000000 +0200
@@ -1,2 +1,2 @@
-SUBDIRS = artsplug misc qtvidcap qtrecompress mjpeg_plugin
+SUBDIRS = artsplug misc qtrecompress mjpeg_plugin
 MAINTAINERCLEANFILES = Makefile.in
