diff -ur avifile-0.7.38.20030710.old/debian/rules avifile-0.7.38.20030710/debian/rules
--- avifile-0.7.38.20030710.old/debian/rules	2003-07-10 15:14:07.000000000 +0200
+++ avifile-0.7.38.20030710/debian/rules	2004-08-23 04:02:52.000000000 +0200
@@ -93,7 +93,7 @@
 #        rm config.cache
 #	CFLAGS="-g -O2 -march=i586" CXXFLAGS="-g -O2 -march=i586"
 # if you wish to build packege without AC3 support: --disable-ffmpeg-ac3
-	./configure --prefix=/usr --disable-lame --enable-lame-bin --enable-quiet --enable-release --disable-dependency-tracking --host=$(DEB_HOST_GNU_TYPE) --build=$(DEB_BUILD_GNU_TYPE)
+	./configure --prefix=/usr --disable-lame --enable-lame-bin --enable-quiet --enable-release --disable-dependency-tracking --host=i386-gnu --build=i386-gnu
 	touch configure-stamp
 
 
@@ -196,7 +196,6 @@
 #        -ptmp
 #DH2	@dh_movefiles
 	dh_compress
-	@chmod +x debian/$(player)/usr/share/$(aviversion)/scripts/*
 	dh_fixperms -Xkv4lsetup -Xavicap-boot -Xavicap-setnvram -Xavicap-shutdown
 	dh_strip
 
diff -ur avifile-0.7.38.20030710.old/samples/Makefile.am avifile-0.7.38.20030710/samples/Makefile.am
--- avifile-0.7.38.20030710.old/samples/Makefile.am	2002-11-14 19:51:00.000000000 +0100
+++ avifile-0.7.38.20030710/samples/Makefile.am	2004-08-23 03:34:58.000000000 +0200
@@ -1,2 +1,2 @@
-SUBDIRS = artsplug misc qtvidcap qtrecompress mjpeg_plugin
+SUBDIRS = artsplug misc qtrecompress mjpeg_plugin
 MAINTAINERCLEANFILES = Makefile.in
