
Status: sent upstream

diff -ur awstats-6.1.old/wwwroot/cgi-bin/lib/operating_systems.pm awstats-6.1/wwwroot/cgi-bin/lib/operating_systems.pm
--- awstats-6.1.old/wwwroot/cgi-bin/lib/operating_systems.pm	2004-01-21 00:07:02.000000000 +0100
+++ awstats-6.1/wwwroot/cgi-bin/lib/operating_systems.pm	2004-07-21 19:08:54.000000000 +0200
@@ -45,6 +45,7 @@
 'freebsd',
 'openbsd',
 'gnu',
+'gnu/(.*)',
 'unix','x11',
 # Other famous OS
 'beos',
@@ -93,6 +94,7 @@
 'freebsd','freebsd',
 'openbsd','openbsd',
 'gnu','gnu',
+'gnu/(.*)','gnu',
 'unix','unix','x11','unix',
 # Other famous OS
 'beos','beos',
