diff -u bitscope-1.1/debian/control bitscope-1.1/debian/control
--- bitscope-1.1/debian/control
+++ bitscope-1.1/debian/control
@@ -2,7 +2,7 @@
 Section: sound
 Priority: optional
 Maintainer: Robert Jordens <jordens@debian.org>
-Build-Depends: debhelper (>> 4.1.0), cdbs, dh-buildinfo, libasound2-dev, libjack0.100.0-dev, libgtk2.0-dev, docbook-to-man
+Build-Depends: debhelper (>> 4.1.0), cdbs, dh-buildinfo, libasound2-dev [!kfreebsd-i386], libjack0.100.0-dev, libgtk2.0-dev, docbook-to-man
 Standards-Version: 3.6.1
 
 Package: bitscope

