Author: aurel32
Status: in BTS

diff -u brightside-1.3.2/debian/control brightside-1.3.2/debian/control
--- brightside-1.3.2/debian/control
+++ brightside-1.3.2/debian/control
@@ -2,7 +2,7 @@
 Section: gnome
 Priority: optional
 Maintainer: Ari Pollak <ari@debian.org>
-Build-Depends: debhelper (>= 4.0.0), libgnomeui-dev, libglade2-dev, libwnck-dev (>= 2.10.0), libasound2-dev, gconf2, libxml-parser-perl
+Build-Depends: debhelper (>= 4.0.0), libgnomeui-dev, libglade2-dev, libwnck-dev (>= 2.10.0), libasound2-dev [!kfreebsd-i386], gconf2, libxml-parser-perl
 Standards-Version: 3.6.1
 
 Package: brightside
