Status: in BTS
Author: aurel32

--- cl-uffi-1.5.7.orig/tests/make.sh
+++ cl-uffi-1.5.7/tests/make.sh
@@ -3,6 +3,7 @@
 case "`uname`" in
     Linux) os_linux=1 ;;
     FreeBSD) os_freebsd=1 ;;
+    GNU/kFreeBSD) os_gnukfreebsd=1;;
     Darwin) os_darwin=1 ;;
     SunOS) os_sunos=1 ;;
     AIX) os_aix=1 ;;
@@ -18,6 +19,10 @@
     gcc -fPIC -DPIC -c $SOURCE -o $OBJECT
     gcc -shared $OBJECT -o $SHARED_LIB
 
+elif [ "$os_gnukfreebsd" ]; then
+    gcc -fPIC -DPIC -c $SOURCE -o $OBJECT
+    gcc -shared $OBJECT -o $SHARED_LIB
+
 elif [ "$os_darwin" ]; then
     cc -dynamic -c $SOURCE -o $OBJECT
     ld -bundle /usr/lib/bundle1.o -flat_namespace -undefined suppress -o $BASE.dylib $OBJECT
