Author: aurel32
Status: in BTS

diff -u darkice-0.15/debian/control darkice-0.15/debian/control
--- darkice-0.15/debian/control
+++ darkice-0.15/debian/control
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Jochen Friedrich <jochen@scram.de>
 Standards-Version: 3.6.2
-Build-Depends: cdbs, debhelper (>=4.1.0), libvorbis-dev, libasound2-dev, libjack0.100.0-dev
+Build-Depends: cdbs, debhelper (>=4.1.0), libvorbis-dev, libasound2-dev [!kfreebsd-i386], libjack0.100.0-dev
 
 Package: darkice
 Architecture: any
