Author: aurel32
Status: in BTS

diff -u denemo-0.7.2b+0.7.3beta2/debian/control denemo-0.7.2b+0.7.3beta2/debian/control
--- denemo-0.7.2b+0.7.3beta2/debian/control
+++ denemo-0.7.2b+0.7.3beta2/debian/control
@@ -2,7 +2,7 @@
 Section: sound
 Priority: optional
 Maintainer: Guenter Geiger (Debian/GNU) <geiger@debian.org>
-Build-Depends: autoconf, automake1.7, libtool, bison, debhelper (>= 3.0.0), cdbs, flex, gettext, libgtk1.2-dev, libglib2.0-dev, libgtk2.0-dev, libxml2-dev, libasound2-dev
+Build-Depends: autoconf, automake1.7, libtool, bison, debhelper (>= 3.0.0), cdbs, flex, gettext, libgtk1.2-dev, libglib2.0-dev, libgtk2.0-dev, libxml2-dev, libasound2-dev [!kfreebsd-i386]
 Standards-Version: 3.6.1
 
 Package: denemo
