diff -u directfb-0.9.25.1/debian/patches/30_linux_types.patch directfb-0.9.25.1/debian/patches/30_linux_types.patch
--- directfb-0.9.25.1/debian/patches/30_linux_types.patch
+++ directfb-0.9.25.1/debian/patches/30_linux_types.patch
@@ -5,20 +5,15 @@
-@@ -1,9 +1,10 @@
+@@ -1,6 +1,9 @@
  #ifndef __DFB_TYPES_H__
  #define __DFB_TYPES_H__
  
--#include <sys/types.h>
++#ifdef __linux__
 +#include <asm/types.h>
++#else
+ #include <sys/types.h>
  
  #ifndef USE_KOS
-+#if 0
+@@ -30,3 +33,4 @@
+ #endif
  
- #define __u8  u_int8_t
- #define __u16 u_int16_t
-@@ -14,6 +15,7 @@
- #define __s16 int16_t
- #define __s32 int32_t
- #define __s64 int64_t
+ #endif
 +#endif
- 
- #else
- 
diff -u directfb-0.9.25.1/debian/rules directfb-0.9.25.1/debian/rules
--- directfb-0.9.25.1/debian/rules
+++ directfb-0.9.25.1/debian/rules
@@ -79,6 +79,7 @@
 	-$(MAKE) distclean
 	-rm -rf $(BUILD_DIST)
 	-rm -rf $(BUILD_UDEB)
+	-rm -f debian/*.install.kfreebsd-amd64
 	
 	dh_clean
 
@@ -99,7 +100,11 @@
 binary-arch: install
 	dh_testdir
 	dh_testroot
-	
+
+	for i in $(package_lib) $(package_udeb) libdirectfb-dev; do \
+		cp debian/$${i}.install.kfreebsd-i386 debian/$$i.install.kfreebsd-amd64; \
+	done
+
 	dh_install --sourcedir=$(DESTDIR_UDEB) -p$(package_udeb)
 	dh_install --sourcedir=$(DESTDIR_DIST) -p$(package_lib) -Xsdl -Xx11 -Xvnc
 	dh_install --sourcedir=$(DESTDIR_DIST) -a -N$(package_udeb) -N$(package_lib)
only in patch2:
unchanged:
--- directfb-0.9.25.1.orig/debian/libdirectfb-0.9-25-udeb.install.kfreebsd-i386
+++ directfb-0.9.25.1/debian/libdirectfb-0.9-25-udeb.install.kfreebsd-i386
@@ -0,0 +1,6 @@
+usr/lib/lib*.so.*
+usr/lib/directfb-*/interfaces/IDirectFBFont/lib*_default.so
+usr/lib/directfb-*/interfaces/IDirectFBFont/lib*_ft2.so
+usr/lib/directfb-*/interfaces/IDirectFBImageProvider/lib*_png.so
+usr/lib/directfb-*/wm/libdirectfbwm_*.so
+usr/share/directfb*/
only in patch2:
unchanged:
--- directfb-0.9.25.1.orig/debian/libdirectfb-0.9-25.install.kfreebsd-i386
+++ directfb-0.9.25.1/debian/libdirectfb-0.9-25.install.kfreebsd-i386
@@ -0,0 +1,6 @@
+usr/lib/lib*.so.*
+usr/lib/directfb-*/interfaces/IDirectFBFont/lib*_default.so
+usr/lib/directfb-*/interfaces/IDirectFBImageProvider/lib*_gif.so
+usr/lib/directfb-*/interfaces/IDirectFBImageProvider/lib*_mpeg2.so
+usr/lib/directfb-*/wm/libdirectfbwm_*.so
+usr/share/directfb-*/
only in patch2:
unchanged:
--- directfb-0.9.25.1.orig/debian/libdirectfb-dev.install.kfreebsd-i386
+++ directfb-0.9.25.1/debian/libdirectfb-dev.install.kfreebsd-i386
@@ -0,0 +1,13 @@
+usr/include/
+usr/lib/lib*.a
+usr/lib/lib*.so
+usr/lib/lib*.la
+usr/bin/directfb-*
+usr/lib/pkgconfig/
+usr/share/man/man1/directfb-*
+usr/lib/directfb-*/interfaces/IDirectFBFont/lib*.a
+usr/lib/directfb-*/interfaces/IDirectFBFont/lib*.la
+usr/lib/directfb-*/interfaces/IDirectFBImageProvider/lib*.a
+usr/lib/directfb-*/interfaces/IDirectFBImageProvider/lib*.la
+usr/lib/directfb-*/wm/libdirectfbwm_*.a
+usr/lib/directfb-*/wm/libdirectfbwm_*.la
