
Status: sent to dev@porting.openoffice.org

--- dmake-4.3.old/configure.in	2005-02-28 15:30:17.000000000 +0100
+++ dmake-4.3/configure.in	2005-02-28 16:50:31.000000000 +0100
@@ -84,7 +84,7 @@
 	"SunOS")
 	  OS_VERSION=solaris  
 	  ;;
-	"Linux")
+	Linux|GNU|GNU/*)
 	  OS_VERSION=linux
 	  ;;
 	"CYGWIN")
@@ -116,9 +116,6 @@
 	"IRIX64")
 	  OS_VERSION=sysvr4
 	  ;;
-	"GNU")
-	  OS_VERSION=linux
-	  ;;
 	"AIX")
 	  OS_VERSION=sysvr4
 	  ;;
