
Status: sent to dev@porting.openoffice.org

diff -ur old/dmake-4.3/configure dmake-4.3/configure
--- old/dmake-4.3/configure	2004-10-12 20:47:46.000000000 +0200
+++ dmake-4.3/configure	2005-05-05 18:49:10.000000000 +0200
@@ -6596,7 +6596,7 @@
 	"SunOS")
 	  OS_VERSION=solaris
 	  ;;
-	"Linux")
+	*)
 	  OS_VERSION=linux
 	  ;;
 	"CYGWIN")
diff -ur old/dmake-4.3/configure.in dmake-4.3/configure.in
--- old/dmake-4.3/configure.in	2004-10-12 21:07:26.000000000 +0200
+++ dmake-4.3/configure.in	2005-05-05 18:49:12.000000000 +0200
@@ -84,7 +84,7 @@
 	"SunOS")
 	  OS_VERSION=solaris  
 	  ;;
-	"Linux")
+	Linux|GNU|GNU/*)
 	  OS_VERSION=linux
 	  ;;
 	"CYGWIN")
@@ -116,9 +116,6 @@
 	"IRIX64")
 	  OS_VERSION=sysvr4
 	  ;;
-	"GNU")
-	  OS_VERSION=linux
-	  ;;
 	"AIX")
 	  OS_VERSION=sysvr4
 	  ;;
