Author: aurel32
Status: in BTS 

diff -u drift-2.1.2/debian/control drift-2.1.2/debian/control
--- drift-2.1.2/debian/control
+++ drift-2.1.2/debian/control
@@ -2,7 +2,7 @@
 Section: devel
 Priority: optional
 Maintainer: Arjan Oosting <arjanoosting@home.nl>
-Build-Depends: cdbs (>= 0.4.23-1.1), build-essential, autotools-dev, debhelper (>= 4.1.0), ghc6 [alpha amd64 hppa i386 ia64 m68k powerpc s390 sparc] | nhc98 [arm hppa i386 mips mipsel powerpc s390], tetex-bin, texinfo, texi2html
+Build-Depends: cdbs (>= 0.4.23-1.1), build-essential, autotools-dev, debhelper (>= 4.1.0), ghc6 [alpha amd64 hppa i386 kfreebsd-i386 ia64 m68k powerpc s390 sparc] | nhc98 [arm hppa i386 kfreebsd-i386 mips mipsel powerpc s390], tetex-bin, texinfo, texi2html
 Standards-Version: 3.6.2
 
 Package: drift

