Author: aurel32
Status: in BTS

diff -u ecasound2.2-2.4.3/debian/control ecasound2.2-2.4.3/debian/control
--- ecasound2.2-2.4.3/debian/control
+++ ecasound2.2-2.4.3/debian/control
@@ -3,7 +3,7 @@
 Priority: extra
 Maintainer: Junichi Uekawa <dancer@debian.org>
 Standards-Version: 3.6.2
-Build-Depends: libsndfile1-dev (>= 1.0.5), libaudiofile-dev, libasound2-dev, debhelper (>> 4.1.1), libwrap-dev, libncurses5-dev, python2.3-dev, ladspa-sdk (>= 1.0.LGPL-3), libtool, m4, libreadline5-dev, libjack0.100.0-dev, libsamplerate0-dev, dpatch, hevea, doxygen, tetex-bin, tetex-extra, autoconf, automake1.9, python, yodl, ruby1.8, ruby, python-docutils, g++ (>= 4:4.0.0)
+Build-Depends: libsndfile1-dev (>= 1.0.5), libaudiofile-dev, libasound2-dev [!kfreebsd-i386], debhelper (>> 4.1.1), libwrap-dev, libncurses5-dev, python2.3-dev, ladspa-sdk (>= 1.0.LGPL-3), libtool, m4, libreadline5-dev, libjack0.100.0-dev, libsamplerate0-dev, dpatch, hevea, doxygen, tetex-bin, tetex-extra, autoconf, automake1.9, python, yodl, ruby1.8, ruby, python-docutils, g++ (>= 4:4.0.0)
 Build-Conflicts: libecasound7, libkvutils2, libecasound7c102, libkvutils2c102
 
 Package: ecasound
@@ -27,7 +27,7 @@
 Package: libecasound2.2-dev
 Architecture: any
 Section: libdevel
-Depends: libkvutils2.2-dev, libaudiofile-dev, libasound2-dev, libjack0.100.0-dev, libsamplerate0-dev, libsndfile1-dev
+Depends: libkvutils2.2-dev, libaudiofile-dev, ${libasound2-dev}, libjack0.100.0-dev, libsamplerate0-dev, libsndfile1-dev
 Conflicts: libecasound-development
 Provides: libecasound-development
 Description: development files for ecasound
diff -u ecasound2.2-2.4.3/debian/rules ecasound2.2-2.4.3/debian/rules
--- ecasound2.2-2.4.3/debian/rules
+++ ecasound2.2-2.4.3/debian/rules
@@ -5,6 +5,8 @@
 # Uncomment this to turn on verbose mode.
 #export DH_VERBOSE=1
 
+DEB_BUILD_ARCH ?= $(shell dpkg-architecture -qDEB_BUILD_ARCH)
+
 include /usr/share/dpatch/dpatch.make
 
 #specify the changeable names here...
@@ -154,7 +156,11 @@
 	dh_shlibdeps -a
 	dh_compress -a
 	dh_fixperms -a
+ifeq (linux,$(DEB_HOST_ARCH_OS))
+	dh_gencontrol -a -- -Vlibasound2-dev="libasound2-dev"
+else
 	dh_gencontrol -a
+endif
 	dh_installdeb -a
 	dh_md5sums -a
 	dh_builddeb -a
