Author: aurel32
Status: in BTS

diff -u erlang-10.b.7/erts/configure.in erlang-10.b.7/erts/configure.in
--- erlang-10.b.7/erts/configure.in
+++ erlang-10.b.7/erts/configure.in
@@ -69,7 +69,7 @@
 AC_CONFIG_HEADER($host/config.h:config.h.in include/internal/$host/ethread_header_config.h:include/internal/ethread_header_config.h.in)
 
 case $host_os in
-	linux*)  AC_DEFINE(_GNU_SOURCE, 1, \
+	*gnu*)  AC_DEFINE(_GNU_SOURCE, 1, \
 [Define if you want to turn on some gnu features.]) ;;
 	*) ;;
 esac
@@ -464,7 +464,7 @@
 	ENABLE_ALLOC_TYPE_VARS="$ENABLE_ALLOC_TYPE_VARS threads"
 	case $host_os in
 	    solaris*) AC_DEFINE(ENABLE_CHILD_WAITER_THREAD) ;;
-	    linux*) AC_DEFINE(USE_RECURSIVE_MALLOC_MUTEX) ;;
+	    *gnu*) AC_DEFINE(USE_RECURSIVE_MALLOC_MUTEX) ;;
 	    *) ;;
 	esac
 fi

