diff -u fcron-3.0.0/debian/control fcron-3.0.0/debian/control
--- fcron-3.0.0/debian/control
+++ fcron-3.0.0/debian/control
@@ -3,7 +3,7 @@
 Priority: extra
 Maintainer: Henrique de Moraes Holschuh <hmh@debian.org>
 Uploaders: Russell Coker <russell@coker.com.au>
-Build-Depends: debhelper (>= 4.1.16), libpam0g-dev, libselinux1-dev (>= 1.2-1.1)
+Build-Depends: debhelper (>= 4.1.16), libpam0g-dev, libselinux1-dev (>= 1.2-1.1) [!kfreebsd-i386 !hurd-i386]
 Standards-Version: 3.6.2
 
 Package: fcron
diff -u fcron-3.0.0/debian/rules fcron-3.0.0/debian/rules
--- fcron-3.0.0/debian/rules
+++ fcron-3.0.0/debian/rules
@@ -9,6 +9,8 @@
 # Uncomment this to turn on verbose mode.
 #export DH_VERBOSE=1
 
+DEB_HOST_ARCH_OS := $(shell dpkg-architecture -qDEB_HOST_ARCH_OS)
+
 export DH_ALWAYS_EXCLUDE=CVS:.svn
 
 DEBUGFLAGS=-g
@@ -21,6 +23,12 @@
 	DEBUGFLAGS += -Werror
 endif
 
+ifeq (linux,$(DEB_HOST_ARCH_OS))
+	SELINUX = --with-selinux=yes
+else
+	SELINUX = --with-selinux=no
+endif
+	
 configure-stamp: configure
 	dh_testdir
 	./configure CFLAGS="-Wall -pipe $(DEBUGFLAGS)" $(CONFFLAGS) \
@@ -30,7 +38,7 @@
 		--with-spooldir=/var/spool/fcron \
 		--with-etcdir=/etc \
 		--with-sendmail=/usr/sbin/sendmail \
-		--with-selinux=yes
+		$(SELINUX)
 
 	touch configure-stamp
 
