
  Status: dirty hack

diff -ur firebird2-1.5.1.old/src/jrd/isc_sync.cpp firebird2-1.5.1/src/jrd/isc_sync.cpp
--- firebird2-1.5.1.old/src/jrd/isc_sync.cpp	2004-04-06 09:51:12.000000000 +0200
+++ firebird2-1.5.1/src/jrd/isc_sync.cpp	2004-08-12 17:59:06.000000000 +0200
@@ -4278,8 +4290,8 @@
 		}
 		
 		// Try to create new semaphore set
-		semid = semget(key, semaphores, IPC_CREAT | IPC_EXCL | PRIV);
-		if (semid != -1) 
+		semid = semget(key, 1, IPC_CREAT | IPC_EXCL | SEM_PERM);
+		if (semid != -1)
 			return semid;
 		else
 			if (errno != EEXIST) {
