Status: in BTS
Author: aurel32

diff -u freesweep-0.88/debian/rules freesweep-0.88/debian/rules
--- freesweep-0.88/debian/rules
+++ freesweep-0.88/debian/rules
@@ -20,9 +20,10 @@
 build-stamp:
 	dh_testdir
 
+	cp /usr/share/misc/config.{sub,guess} .
 	$(MAKE) config.h.in
 	CFLAGS="$(CFLAGS)" ./configure --prefix=/usr --with-prefsdir=/etc \
-		--with-scoresdir=/var/games --disable-debug-log
+		--with-scoresdir=/var/games --disable-debug-log --with-ncurses=/usr
 	$(MAKE)
 
 	touch build-stamp
@@ -31,6 +32,7 @@
 	dh_testdir
 	dh_testroot
 	rm -f build-stamp
+	rm -f config.{guess,sub}
 
 	-$(MAKE) clean
 
diff -u freesweep-0.88/debian/control freesweep-0.88/debian/control
--- freesweep-0.88/debian/control
+++ freesweep-0.88/debian/control
@@ -3,7 +3,7 @@
 Section: games
 Priority: optional
 Standards-Version: 3.5.5
-Build-Depends: debhelper (>= 3.0.0), libncurses5-dev, autoconf
+Build-Depends: debhelper (>= 3.0.0), libncurses5-dev, autoconf, autotools-dev
 
 Package: freesweep
 Architecture: any
