diff -Nru gcc-defaults-1.50.orig/debian/rules gcc-defaults-1.50/debian/rules
--- gcc-defaults-1.50.orig/debian/rules	2007-01-02 13:42:03.000000000 +0100
+++ gcc-defaults-1.50/debian/rules	2007-01-06 14:37:56.000000000 +0100
@@ -162,6 +162,11 @@
     OS_NAME	:= kFreeBSD
 endif
 
+ifeq ($(DEB_HOST_ARCH),kfreebsd-amd64)
+    OS_NAME	:= kFreeBSD
+    no_packages += gnat
+endif
+
 ifeq ($(DEB_HOST_ARCH),m32r)
     no_packages += gnat
 endif
