diff -u gcc-3.4-3.4.4/debian/rules.d/binary-gcc-cross.mk gcc-3.4-3.4.4/debian/rules.d/binary-gcc-cross.mk
--- gcc-3.4-3.4.4/debian/rules.d/binary-gcc-cross.mk
+++ gcc-3.4-3.4.4/debian/rules.d/binary-gcc-cross.mk
@@ -42,6 +42,10 @@
     files_gcc += $(gcc_lib_dir)/include/{,e,p,x}mmintrin.h
 endif
 
+ifeq ($(DEB_TARGET_ARCH),kfreebsd-i386)
+    files_gcc += $(gcc_lib_dir)/include/{,e,p,x}mmintrin.h
+endif
+
 ifeq ($(DEB_TARGET_ARCH),m68k)
     files_gcc += $(gcc_lib_dir)/include/math-68881.h
 endif
diff -u gcc-3.4-3.4.4/debian/rules.d/binary-gcc.mk gcc-3.4-3.4.4/debian/rules.d/binary-gcc.mk
--- gcc-3.4-3.4.4/debian/rules.d/binary-gcc.mk
+++ gcc-3.4-3.4.4/debian/rules.d/binary-gcc.mk
@@ -52,6 +52,10 @@
     files_gcc += $(gcc_lib_dir)/include/{,e,p,x}mmintrin.h
 endif
 
+ifeq ($(DEB_HOST_ARCH),kfreebsd-i386)
+    files_gcc += $(gcc_lib_dir)/include/{,e,p,x}mmintrin.h
+endif
+
 ifeq ($(DEB_HOST_ARCH),hurd-i386)
     files_gcc += $(gcc_lib_dir)/include/{,e,p,x}mmintrin.h
 endif
diff -u gcc-3.4-3.4.4/debian/patches/kbsd-gnu.dpatch gcc-3.4-3.4.4/debian/patches/kbsd-gnu.dpatch
--- gcc-3.4-3.4.4/debian/patches/kbsd-gnu.dpatch
+++ gcc-3.4-3.4.4/debian/patches/kbsd-gnu.dpatch
@@ -15,9 +15,6 @@
 case "$1" in
     -patch)
         patch $pdir -f --no-backup-if-mismatch -p1 < $0
-	for i in libffi libf2c libjava/libltdl libobjc libstdc++-v3 zlib; do
-		(set -x ; cd ${dir}/${i} ; autoreconf --force)
-	done
         ;;
     -unpatch)
         patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
@@ -30,9 +27,54 @@
 
 # append the patch here and adjust the -p? flag in the patch calls.
 
-diff -Nur src.old/gcc/config/i386/kfreebsd-gnu.h src/gcc/config/i386/kfreebsd-gnu.h
---- src.old/gcc/config/i386/kfreebsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
-+++ src/gcc/config/i386/kfreebsd-gnu.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.4.old/boehm-gc/configure gcc-3.4.4/boehm-gc/configure
+--- gcc-3.4.4.old/boehm-gc/configure	2005-05-19 11:58:00.000000000 +0200
++++ gcc-3.4.4/boehm-gc/configure	2005-06-23 16:09:42.000000000 +0200
+@@ -1879,7 +1879,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd* )
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -1947,7 +1947,7 @@
+   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
+   ;;
+ 
+-netbsd*)
++netbsd* | knetbsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$'
+   else
+diff -Nur gcc-3.4.4.old/configure gcc-3.4.4/configure
+--- gcc-3.4.4.old/configure	2005-03-08 18:31:40.000000000 +0100
++++ gcc-3.4.4/configure	2005-06-23 16:09:40.000000000 +0200
+@@ -1935,7 +1935,7 @@
+   powerpc-*-netware*)
+     target_makefile_frag="config/mt-netware"
+     ;;
+-  *-*-linux*)
++  *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu)
+     target_makefile_frag="config/mt-linux"
+     ;;
+   *-*-aix4.[3456789]* | *-*-aix[56789].*)
+diff -Nur gcc-3.4.4.old/configure.in gcc-3.4.4/configure.in
+--- gcc-3.4.4.old/configure.in	2005-03-08 18:31:40.000000000 +0100
++++ gcc-3.4.4/configure.in	2005-06-23 16:09:40.000000000 +0200
+@@ -1174,7 +1174,7 @@
+   powerpc-*-netware*)
+     target_makefile_frag="config/mt-netware"
+     ;;
+-  *-*-linux*)
++  *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu)
+     target_makefile_frag="config/mt-linux"
+     ;;
+   *-*-aix4.[[3456789]]* | *-*-aix[[56789]].*)
+diff -Nur gcc-3.4.4.old/gcc/config/i386/kfreebsd-gnu.h gcc-3.4.4/gcc/config/i386/kfreebsd-gnu.h
+--- gcc-3.4.4.old/gcc/config/i386/kfreebsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
++++ gcc-3.4.4/gcc/config/i386/kfreebsd-gnu.h	2005-06-23 16:09:40.000000000 +0200
 @@ -0,0 +1,26 @@
 +/* Definitions for Intel 386 running kFreeBSD-based GNU systems with ELF format
 +   Copyright (C) 2004
@@ -60,9 +102,9 @@
 +#define LINK_EMULATION "elf_i386_fbsd"
 +#undef REG_NAME
 +#define REG_NAME(reg) sc_ ## reg
-diff -Nur src.old/gcc/config/i386/knetbsd-gnu.h src/gcc/config/i386/knetbsd-gnu.h
---- src.old/gcc/config/i386/knetbsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
-+++ src/gcc/config/i386/knetbsd-gnu.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.4.old/gcc/config/i386/knetbsd-gnu.h gcc-3.4.4/gcc/config/i386/knetbsd-gnu.h
+--- gcc-3.4.4.old/gcc/config/i386/knetbsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
++++ gcc-3.4.4/gcc/config/i386/knetbsd-gnu.h	2005-06-23 16:09:41.000000000 +0200
 @@ -0,0 +1,24 @@
 +/* Definitions for Intel 386 running kNetBSD-based GNU systems with ELF format
 +   Copyright (C) 2004
@@ -88,9 +130,9 @@
 +
 +#undef REG_NAME
 +#define REG_NAME(reg) sc_ ## reg
-diff -Nur src.old/gcc/config/i386/linux.h src/gcc/config/i386/linux.h
---- src.old/gcc/config/i386/linux.h	2003-11-29 04:08:10.000000000 +0100
-+++ src/gcc/config/i386/linux.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.4.old/gcc/config/i386/linux.h gcc-3.4.4/gcc/config/i386/linux.h
+--- gcc-3.4.4.old/gcc/config/i386/linux.h	2003-11-29 04:08:10.000000000 +0100
++++ gcc-3.4.4/gcc/config/i386/linux.h	2005-06-23 16:09:41.000000000 +0200
 @@ -108,24 +108,30 @@
  
  /* If ELF is the default format, we should not use /lib/elf.  */
@@ -181,9 +223,9 @@
      (FS)->retaddr_column = 8;						\
      goto SUCCESS;							\
    } while (0)
-diff -Nur src.old/gcc/config/kfreebsd-gnu.h src/gcc/config/kfreebsd-gnu.h
---- src.old/gcc/config/kfreebsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
-+++ src/gcc/config/kfreebsd-gnu.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.4.old/gcc/config/kfreebsd-gnu.h gcc-3.4.4/gcc/config/kfreebsd-gnu.h
+--- gcc-3.4.4.old/gcc/config/kfreebsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
++++ gcc-3.4.4/gcc/config/kfreebsd-gnu.h	2005-06-23 16:09:41.000000000 +0200
 @@ -0,0 +1,36 @@
 +/* Definitions for kFreeBSD-based GNU systems with ELF format
 +   Copyright (C) 2004
@@ -221,9 +263,9 @@
 +
 +#undef DYNAMIC_LINKER
 +#define DYNAMIC_LINKER "/lib/ld.so.1"
-diff -Nur src.old/gcc/config/knetbsd-gnu.h src/gcc/config/knetbsd-gnu.h
---- src.old/gcc/config/knetbsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
-+++ src/gcc/config/knetbsd-gnu.h	2004-05-22 02:37:45.000000000 +0200
+diff -Nur gcc-3.4.4.old/gcc/config/knetbsd-gnu.h gcc-3.4.4/gcc/config/knetbsd-gnu.h
+--- gcc-3.4.4.old/gcc/config/knetbsd-gnu.h	1970-01-01 01:00:00.000000000 +0100
++++ gcc-3.4.4/gcc/config/knetbsd-gnu.h	2005-06-23 16:09:41.000000000 +0200
 @@ -0,0 +1,36 @@
 +/* Definitions for kNetBSD-based GNU systems with ELF format
 +   Copyright (C) 2004
@@ -261,29 +303,10 @@
 +
 +#undef DYNAMIC_LINKER
 +#define DYNAMIC_LINKER "/lib/ld.so.1"
-diff -Nur src.old/gcc/config/linux.h src/gcc/config/linux.h
---- src.old/gcc/config/linux.h	2003-11-29 04:08:10.000000000 +0100
-+++ src/gcc/config/linux.h	2004-05-22 02:37:45.000000000 +0200
-@@ -97,6 +97,7 @@
-        %{!p:%{!pg:%{!g*:-lc} %{g*:-lg}}}}"
- #endif
- 
-+#ifndef LINUX_TARGET_OS_CPP_BUILTINS
- #define LINUX_TARGET_OS_CPP_BUILTINS()				\
-     do {							\
- 	builtin_define ("__gnu_linux__");			\
-@@ -106,6 +107,7 @@
- 	builtin_assert ("system=unix");				\
- 	builtin_assert ("system=posix");			\
-     } while (0)
-+#endif
- 
- #if !defined(USE_GNULIBC_1) && defined(HAVE_LD_EH_FRAME_HDR)
- #define LINK_EH_SPEC "%{!static:--eh-frame-hdr} "
-diff -Nur src.old/gcc/config.gcc src/gcc/config.gcc
---- src.old/gcc/config.gcc	2004-05-22 01:09:22.000000000 +0200
-+++ src/gcc/config.gcc	2004-05-22 02:41:54.000000000 +0200
-@@ -428,21 +428,10 @@
+diff -Nur gcc-3.4.4.old/gcc/config.gcc gcc-3.4.4/gcc/config.gcc
+--- gcc-3.4.4.old/gcc/config.gcc	2005-04-25 06:47:59.000000000 +0200
++++ gcc-3.4.4/gcc/config.gcc	2005-06-23 16:09:41.000000000 +0200
+@@ -434,21 +434,10 @@
    esac
    fbsd_tm_file="${fbsd_tm_file} freebsd-spec.h freebsd.h"
    ;;
@@ -306,7 +329,7 @@
    # Must come before *-*-gnu* (because of *-*-linux-gnu* systems).
    extra_parts="crtbegin.o crtbeginS.o crtbeginT.o crtend.o crtendS.o"
    gas=yes
-@@ -988,23 +977,21 @@
+@@ -998,23 +987,21 @@
  		thread_file='single'
  	fi
  	;;
@@ -336,9 +359,230 @@
  i[34567]86-*-gnu*)
  	;;
  i[34567]86-pc-msdosdjgpp*)
-diff -Nur src.old/libtool.m4 src/libtool.m4
---- src.old/libtool.m4	2003-11-19 06:29:32.000000000 +0100
-+++ src/libtool.m4	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.4.old/libf2c/configure gcc-3.4.4/libf2c/configure
+--- gcc-3.4.4.old/libf2c/configure	2004-05-18 11:08:39.000000000 +0200
++++ gcc-3.4.4/libf2c/configure	2005-06-23 16:09:41.000000000 +0200
+@@ -1549,7 +1549,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd* )
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -1617,7 +1617,7 @@
+   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
+   ;;
+ 
+-netbsd*)
++netbsd* | knetbsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$'
+   else
+diff -Nur gcc-3.4.4.old/libffi/configure gcc-3.4.4/libffi/configure
+--- gcc-3.4.4.old/libffi/configure	2004-05-18 11:08:39.000000000 +0200
++++ gcc-3.4.4/libffi/configure	2005-06-23 16:09:41.000000000 +0200
+@@ -1264,7 +1264,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd* )
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -1332,7 +1332,7 @@
+   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
+   ;;
+ 
+-netbsd*)
++netbsd* | knetbsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$'
+   else
+@@ -2818,16 +2818,16 @@
+ i*86-*-linux*) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-solaris*) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-beos*) TARGET=X86; TARGETDIR=x86;;
+-i*86-*-freebsd*) TARGET=X86; TARGETDIR=x86;;
+-i*86-*-netbsdelf*) TARGET=X86; TARGETDIR=x86;;
++i*86-*-freebsd* | i*86-*-kfreebsd*-gnu) TARGET=X86; TARGETDIR=x86;;
++i*86-*-netbsdelf* | i*86-*-knetbsd*-gnu) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-win32*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ i*86-*-cygwin*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ i*86-*-mingw*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ sparc-sun-4*) TARGET=SPARC; TARGETDIR=sparc;;
+ sparc*-sun-*) TARGET=SPARC; TARGETDIR=sparc;;
+-sparc-*-linux* | sparc-*-netbsdelf*) TARGET=SPARC; TARGETDIR=sparc;;
+-sparc64-*-linux* | sparc64-*-netbsd*) TARGET=SPARC; TARGETDIR=sparc;;
+-alpha*-*-linux* | alpha*-*-osf* | alpha*-*-freebsd* | alpha*-*-netbsd*) TARGET=ALPHA; TARGETDIR=alpha;;
++sparc-*-linux* | sparc-*-netbsdelf* | sparc-*-knetbsd*-gnu) TARGET=SPARC; TARGETDIR=sparc;;
++sparc64-*-linux* | sparc64-*-netbsd* | sparc64-*-knetbsd*-gnu) TARGET=SPARC; TARGETDIR=sparc;;
++alpha*-*-linux* | alpha*-*-osf* | alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu | alpha*-*-netbsd* | alpha*-*-knetbsd*-gnu) TARGET=ALPHA; TARGETDIR=alpha;;
+ ia64*-*-*) TARGET=IA64; TARGETDIR=ia64;;
+ m68k-*-linux*) TARGET=M68K; TARGETDIR=m68k;;
+ mips64*-*);;
+@@ -2839,7 +2839,7 @@
+ powerpc-*-aix*) TARGET=POWERPC_AIX; TARGETDIR=powerpc;;
+ rs6000-*-aix*) TARGET=POWERPC_AIX; TARGETDIR=powerpc;;
+ arm*-*-linux-*) TARGET=ARM; TARGETDIR=arm;;
+-arm*-*-netbsdelf*) TARGET=ARM; TARGETDIR=arm;;
++arm*-*-netbsdelf* | i*86-*-knetbsd*-gnu) TARGET=ARM; TARGETDIR=arm;;
+ s390-*-linux-*) TARGET=S390; TARGETDIR=s390;;
+ s390x-*-linux-*) TARGET=S390; TARGETDIR=s390;;
+ x86_64-*-linux*) TARGET=X86_64; TARGETDIR=x86;;
+diff -Nur gcc-3.4.4.old/libffi/configure.in gcc-3.4.4/libffi/configure.in
+--- gcc-3.4.4.old/libffi/configure.in	2004-04-27 07:10:19.000000000 +0200
++++ gcc-3.4.4/libffi/configure.in	2005-06-23 16:09:41.000000000 +0200
+@@ -60,16 +60,16 @@
+ i*86-*-linux*) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-solaris*) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-beos*) TARGET=X86; TARGETDIR=x86;;
+-i*86-*-freebsd*) TARGET=X86; TARGETDIR=x86;;
+-i*86-*-netbsdelf*) TARGET=X86; TARGETDIR=x86;;
++i*86-*-freebsd* | i*86-*-kfreebsd*-gnu) TARGET=X86; TARGETDIR=x86;;
++i*86-*-netbsdelf* | i*86-*-knetbsd*-gnu) TARGET=X86; TARGETDIR=x86;;
+ i*86-*-win32*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ i*86-*-cygwin*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ i*86-*-mingw*) TARGET=X86_WIN32; TARGETDIR=x86;;
+ sparc-sun-4*) TARGET=SPARC; TARGETDIR=sparc;;
+ sparc*-sun-*) TARGET=SPARC; TARGETDIR=sparc;;
+-sparc-*-linux* | sparc-*-netbsdelf*) TARGET=SPARC; TARGETDIR=sparc;;
+-sparc64-*-linux* | sparc64-*-netbsd*) TARGET=SPARC; TARGETDIR=sparc;;
+-alpha*-*-linux* | alpha*-*-osf* | alpha*-*-freebsd* | alpha*-*-netbsd*) TARGET=ALPHA; TARGETDIR=alpha;;
++sparc-*-linux* | sparc-*-netbsdelf* | sparc-*-knetbsd*-gnu) TARGET=SPARC; TARGETDIR=sparc;;
++sparc64-*-linux* | sparc64-*-netbsd* | sparc64-*-knetbsd*-gnu) TARGET=SPARC; TARGETDIR=sparc;;
++alpha*-*-linux* | alpha*-*-osf* | alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu | alpha*-*-netbsd* | alpha*-*-knetbsd*-gnu) TARGET=ALPHA; TARGETDIR=alpha;;
+ ia64*-*-*) TARGET=IA64; TARGETDIR=ia64;;
+ m68k-*-linux*) TARGET=M68K; TARGETDIR=m68k;;
+ mips64*-*);;
+@@ -81,7 +81,7 @@
+ powerpc-*-aix*) TARGET=POWERPC_AIX; TARGETDIR=powerpc;;
+ rs6000-*-aix*) TARGET=POWERPC_AIX; TARGETDIR=powerpc;;
+ arm*-*-linux-*) TARGET=ARM; TARGETDIR=arm;;
+-arm*-*-netbsdelf*) TARGET=ARM; TARGETDIR=arm;;
++arm*-*-netbsdelf* | i*86-*-knetbsd*-gnu) TARGET=ARM; TARGETDIR=arm;;
+ s390-*-linux-*) TARGET=S390; TARGETDIR=s390;;
+ s390x-*-linux-*) TARGET=S390; TARGETDIR=s390;;
+ x86_64-*-linux*) TARGET=X86_64; TARGETDIR=x86;;
+diff -Nur gcc-3.4.4.old/libjava/configure gcc-3.4.4/libjava/configure
+--- gcc-3.4.4.old/libjava/configure	2005-05-19 11:58:00.000000000 +0200
++++ gcc-3.4.4/libjava/configure	2005-06-23 16:09:41.000000000 +0200
+@@ -1975,7 +1975,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd* )
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -2043,7 +2043,7 @@
+   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
+   ;;
+ 
+-netbsd*)
++netbsd* | knetbsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$'
+   else
+diff -Nur gcc-3.4.4.old/libjava/libltdl/configure gcc-3.4.4/libjava/libltdl/configure
+--- gcc-3.4.4.old/libjava/libltdl/configure	2005-05-19 11:58:00.000000000 +0200
++++ gcc-3.4.4/libjava/libltdl/configure	2005-06-23 16:09:42.000000000 +0200
+@@ -3558,7 +3558,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd*)
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -3629,7 +3629,7 @@
+   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
+   ;;
+ 
+-netbsd*)
++netbsd* | knetbsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     lt_cv_deplibs_check_method='match_pattern /lib[^/]+(\.so\.[0-9]+\.[0-9]+|_pic\.a)$'
+   else
+@@ -7343,7 +7343,7 @@
+       ;;
+ 
+     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-    freebsd*)
++    freebsd* | kfreebsd*-gnu)
+       archive_cmds='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
+       hardcode_libdir_flag_spec='-R$libdir'
+       hardcode_direct=yes
+@@ -7976,6 +7976,20 @@
+   dynamic_linker=no
+   ;;
+ 
++kfreebsd*)
++  version_type=linux
++  need_lib_prefix=no
++  need_version=no
++  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
++  soname_spec='${libname}${release}${shared_ext}$major'
++  finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
++  shlibpath_var=LD_LIBRARY_PATH
++  shlibpath_overrides_runpath=no
++  hardcode_into_libs=yes
++  dynamic_linker='GNU ld.so'
++  ;;
++
++
+ freebsd*)
+   objformat=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
+   version_type=freebsd-$objformat
+diff -Nur gcc-3.4.4.old/libobjc/configure gcc-3.4.4/libobjc/configure
+--- gcc-3.4.4.old/libobjc/configure	2004-05-18 11:08:42.000000000 +0200
++++ gcc-3.4.4/libobjc/configure	2005-06-23 16:09:41.000000000 +0200
+@@ -1554,7 +1554,7 @@
+   esac
+   ;;
+ 
+-freebsd* )
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -1622,7 +1622,7 @@
+   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
+   ;;
+ 
+-netbsd*)
++netbsd* | knetbsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$'
+   else
+diff -Nur gcc-3.4.4.old/libstdc++-v3/configure gcc-3.4.4/libstdc++-v3/configure
+--- gcc-3.4.4.old/libstdc++-v3/configure	2005-02-01 07:29:28.000000000 +0100
++++ gcc-3.4.4/libstdc++-v3/configure	2005-06-23 16:09:41.000000000 +0200
+@@ -3944,7 +3944,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd* )
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -4012,7 +4012,7 @@
+   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
+   ;;
+ 
+-netbsd*)
++netbsd* | knetbsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$'
+   else
+diff -Nur gcc-3.4.4.old/libtool.m4 gcc-3.4.4/libtool.m4
+--- gcc-3.4.4.old/libtool.m4	2004-05-18 11:08:37.000000000 +0200
++++ gcc-3.4.4/libtool.m4	2005-06-23 16:09:41.000000000 +0200
 @@ -621,7 +621,7 @@
    lt_cv_deplibs_check_method=pass_all
    ;;
@@ -357,9 +601,9 @@
    if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
      [lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$']
    else
-diff -Nur src.old/ltcf-c.sh src/ltcf-c.sh
---- src.old/ltcf-c.sh	2003-11-19 06:29:32.000000000 +0100
-+++ src/ltcf-c.sh	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.4.old/ltcf-c.sh gcc-3.4.4/ltcf-c.sh
+--- gcc-3.4.4.old/ltcf-c.sh	2003-11-19 06:29:32.000000000 +0100
++++ gcc-3.4.4/ltcf-c.sh	2005-06-23 16:09:41.000000000 +0200
 @@ -175,7 +175,7 @@
        $CC $output_objdir/$soname-exp '$lt_cv_cc_dll_switch' -Wl,-e,'$dll_entry' -o $output_objdir/$soname '$ltdll_obj'$libobjs $deplibs $compiler_flags'
      ;;
@@ -387,9 +631,9 @@
      if echo __ELF__ | $CC -E - | grep __ELF__ >/dev/null; then
        archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'  # a.out
      else
-diff -Nur src.old/ltcf-cxx.sh src/ltcf-cxx.sh
---- src.old/ltcf-cxx.sh	2003-11-19 06:29:32.000000000 +0100
-+++ src/ltcf-cxx.sh	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.4.old/ltcf-cxx.sh gcc-3.4.4/ltcf-cxx.sh
+--- gcc-3.4.4.old/ltcf-cxx.sh	2003-11-19 06:29:32.000000000 +0100
++++ gcc-3.4.4/ltcf-cxx.sh	2005-06-23 16:09:41.000000000 +0200
 @@ -289,7 +289,7 @@
      # C++ shared libraries reported to be fairly broken before switch to ELF
      ld_shlibs=no
@@ -417,9 +661,9 @@
        # FreeBSD uses GNU C++
        ;;
      gnu*)
-diff -Nur src.old/ltcf-gcj.sh src/ltcf-gcj.sh
---- src.old/ltcf-gcj.sh	2003-11-19 06:29:32.000000000 +0100
-+++ src/ltcf-gcj.sh	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.4.old/ltcf-gcj.sh gcc-3.4.4/ltcf-gcj.sh
+--- gcc-3.4.4.old/ltcf-gcj.sh	2003-11-19 06:29:32.000000000 +0100
++++ gcc-3.4.4/ltcf-gcj.sh	2005-06-23 16:09:41.000000000 +0200
 @@ -178,7 +178,7 @@
        $CC $output_objdir/$soname-exp '$lt_cv_cc_dll_switch' -Wl,-e,'$dll_entry' -o $output_objdir/$soname '$ltdll_obj'$libobjs $deplibs $compiler_flags'
      ;;
@@ -447,9 +691,9 @@
      if echo __ELF__ | $CC -E - | grep __ELF__ >/dev/null; then
        archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'  # a.out
      else
-diff -Nur src.old/ltconfig src/ltconfig
---- src.old/ltconfig	2004-03-05 22:05:41.000000000 +0100
-+++ src/ltconfig	2004-05-22 02:37:46.000000000 +0200
+diff -Nur gcc-3.4.4.old/ltconfig gcc-3.4.4/ltconfig
+--- gcc-3.4.4.old/ltconfig	2004-03-05 22:05:41.000000000 +0100
++++ gcc-3.4.4/ltconfig	2005-06-23 16:09:41.000000000 +0200
 @@ -1168,6 +1168,17 @@
    hardcode_into_libs=yes
    ;;
@@ -470,0 +715,21 @@
+diff -Nur gcc-3.4.4.old/zlib/configure gcc-3.4.4/zlib/configure
+--- gcc-3.4.4.old/zlib/configure	2004-05-18 11:09:14.000000000 +0200
++++ gcc-3.4.4/zlib/configure	2005-06-23 16:09:41.000000000 +0200
+@@ -1521,7 +1521,7 @@
+   esac
+   ;;
+ 
+-freebsd* )
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -1589,7 +1589,7 @@
+   lt_cv_file_magic_test_file=`echo /lib/libc.so* /lib/libc-*.so`
+   ;;
+ 
+-netbsd*)
++netbsd* | knetbsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     lt_cv_deplibs_check_method='match_pattern /lib[^/\.]+\.so\.[0-9]+\.[0-9]+$'
+   else
diff -u gcc-3.4-3.4.4/debian/changelog gcc-3.4-3.4.4/debian/changelog
--- gcc-3.4-3.4.4/debian/changelog
+++ gcc-3.4-3.4.4/debian/changelog
@@ -1,3 +1,9 @@
+gcc-3.4 (3.4.4-5+kbsd) unreleased; urgency=low
+
+  * Added support for GNU/kFreeBSD.
+
+ -- Aurelien Jarno <aurel32@debian.org>  Mon, 11 Jul 2005 18:31:47 +0300
+
 gcc-3.4 (3.4.4-5) unstable; urgency=low
 
   * Updated to gcc-3.4 CVS 20050706.
diff -u gcc-3.4-3.4.4/debian/rules.defs gcc-3.4-3.4.4/debian/rules.defs
--- gcc-3.4-3.4.4/debian/rules.defs
+++ gcc-3.4-3.4.4/debian/rules.defs
@@ -399,7 +399,7 @@
 
 # disable ObjC garbage collection library (needs libgc)
 libgc_no_cpus := avr
-libgc_no_systems := kfreebsd-gnu knetbsd-gnu
+libgc_no_systems := knetbsd-gnu
 ifeq ($(DEB_TARGET_GNU_CPU),$(findstring $(DEB_TARGET_GNU_CPU),$(libgc_no_cpus)))
   with_objc_gc := disabled for architecture $(DEB_TARGET_GNU_CPU)
 endif
@@ -522,7 +522,7 @@
 ifdef DEB_CROSS
   with_check := disabled for cross compiler package
 endif
-check_no_systems := hurd-i386 kfreebsd-gnu knetbsd-gnu
+check_no_systems := hurd-i386 knetbsd-gnu
 ifeq ($(DEB_TARGET_GNU_SYSTEM), $(findstring $(DEB_TARGET_GNU_SYSTEM),$(check_no_systems)))
   with_check := disabled for $(DEB_TARGET_GNU_SYSTEM)
 endif
@@ -625,7 +625,7 @@
 
 # GNU locales
 force_gnu_locales := yes
-locale_no_systems := kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
+locale_no_systems := knetbsd-gnu netbsd-elf-gnu
 ifeq ($(DEB_TARGET_GNU_SYSTEM), $(findstring $(DEB_TARGET_GNU_SYSTEM),$(locale_no_systems)))
   force_gnu_locales := disabled for $(DEB_TARGET_GNU_SYSTEM)
 endif
