--- gcc-defaults-1.17.old/debian/rules	2004-07-13 07:43:06.000000000 +0200
+++ gcc-defaults-1.17/debian/rules	2004-07-23 14:07:53.000000000 +0200
@@ -75,9 +75,28 @@
 REQV_GPC	= $(REQV_GPC_33)
 
 EPOCH=4
-OS_NAME	:= Linux
+
 packages := chill c++ fortran java objc pascal
 
+# System specific, but CPU independant
+ifeq ($(DEB_HOST_GNU_SYSTEM),linux)
+    OS_NAME	:= Linux
+endif
+ifeq ($(DEB_HOST_GNU_SYSTEM),gnu)
+    OS_NAME	:= Hurd
+endif
+ifeq ($(DEB_HOST_GNU_SYSTEM),kfreebsd-gnu)
+    OS_NAME	:= kFreeBSD
+endif
+ifeq ($(DEB_HOST_GNU_SYSTEM),knetbsd-gnu)
+    OS_NAME	:= kNetBSD
+endif
+ifeq ($(DEB_HOST_GNU_SYSTEN),netbsd-elf-gnu)
+    OS_NAME	:= NetBSD
+endif
+
+# System and CPU specific
+
 ifeq ($(DEB_HOST_GNU_TYPE),alpha-linux)
 endif
 
@@ -112,14 +131,6 @@
     packages := chill c++ fortran objc pascal
 endif
 
-ifeq ($(DEB_HOST_GNU_TYPE),i386-gnu)
-    OS_NAME	:= Hurd
-endif
-
-ifeq ($(DEB_HOST_ARCH),netbsd-i386)
-    OS_NAME	:= NetBSD
-endif
-
 # derived version number (without release)
 V_CPP	:= $(shell echo $(CV_CPP)   | sed 's/-[^-]*$$//')
 V_GCC	:= $(shell echo $(CV_GCC)   | sed 's/-[^-]*$$//')
