diff -u gcc-4.1-4.1.2/debian/patches/kbsd-gnu.dpatch gcc-4.1-4.1.2/debian/patches/kbsd-gnu.dpatch
--- gcc-4.1-4.1.2/debian/patches/kbsd-gnu.dpatch
+++ gcc-4.1-4.1.2/debian/patches/kbsd-gnu.dpatch
@@ -79,8 +79,8 @@
 +  { "dynamic_linker64", DYNAMIC_LINKER64 }
 +
  #undef	LINK_SPEC
--#define LINK_SPEC "%{!m32:-m elf_x86_64} %{m32:-m elf_i386} \
-+#define LINK_SPEC "%{!m32:-m %(link_emulation64)} %{m32:-m %(link_emulation)} \
+-#define LINK_SPEC "%{!m32:-m elf_x86_64} %{m32:-m elf_i386} --hash-style=both \
++#define LINK_SPEC "%{!m32:-m %(link_emulation64)} %{m32:-m %(link_emulation)} --hash-style=both \
    %{shared:-shared} \
    %{!shared: \
      %{!static: \
diff -u gcc-4.1-4.1.2/debian/patches/kbsd-gnu-java.dpatch gcc-4.1-4.1.2/debian/patches/kbsd-gnu-java.dpatch
--- gcc-4.1-4.1.2/debian/patches/kbsd-gnu-java.dpatch
+++ gcc-4.1-4.1.2/debian/patches/kbsd-gnu-java.dpatch
@@ -226,18 +226,6 @@
  	    /* SAVE_CALL_CHAIN is supported if the code is compiled to save	*/
  	    /* frame pointers by default, i.e. no -fomit-frame-pointer flag.	*/
  # define CAN_SAVE_CALL_ARGS
-diff -ur gcj-4.1-4.1.0/src/libjava/configure gcj-4.1-4.1.0/src/libjava/configure
---- gcj-4.1-4.1.0/src/libjava/configure	2006-05-12 13:35:06.000000000 +0200
-+++ gcj-4.1-4.1.0/src/libjava/configure	2006-05-12 11:00:31.000000000 +0200
-@@ -4351,7 +4341,7 @@
-   ;;
- 
- # This must be Linux ELF.
--linux*)
-+linux*|k*bsd*-gnu)
-   version_type=linux
-   need_lib_prefix=no
-   need_version=no
 diff -ur gcj-4.1-4.1.0/src/libjava/libltdl/acinclude.m4 gcj-4.1-4.1.0/src/libjava/libltdl/acinclude.m4
 --- gcj-4.1-4.1.0/src/libjava/libltdl/acinclude.m4	2006-05-12 13:35:07.000000000 +0200
 +++ gcj-4.1-4.1.0/src/libjava/libltdl/acinclude.m4	2006-05-12 11:07:03.000000000 +0200
@@ -289,30 +277,6 @@
    version_type=linux
    need_lib_prefix=no
    need_version=no
-diff -ur gcj-4.1-4.1.0/src/libjava/shlibpath.m4 gcj-4.1-4.1.0/src/libjava/shlibpath.m4
---- gcj-4.1-4.1.0/src/libjava/shlibpath.m4	2006-01-07 19:14:24.000000000 +0100
-+++ gcj-4.1-4.1.0/src/libjava/shlibpath.m4	2006-05-12 08:36:28.000000000 +0200
-@@ -361,7 +361,7 @@
-   ;;
- 
- # This must be Linux ELF.
--linux*)
-+linux*|k*bsd*-gnu)
-   version_type=linux
-   need_lib_prefix=no
-   need_version=no
-diff -ur gcj-4.1-4.1.1/src/libjava/configure.host gcj-4.1-4.1.1/src/libjava/configure.host  
---- gcj-4.1-4.1.1/src/libjava/configure.host	2006-09-07 01:00:37.000000000 +0200
-+++ gcj-4.1-4.1.1/src/libjava/configure.host	2006-09-07 01:00:37.000000000 +0200
-@@ -326,7 +326,7 @@
- esac
- 
- case "${host}" in
--  *linux*)
-+  *linux*|*-k*bsd*-gnu)
-     use_libgcj_bc=yes
-   ;;
-   *)
 diff -ur gcj-4.1-4.1.0/src/boehm-gc/os_dep.c.orig gcj-4.1-4.1.0/src/boehm-gc/os_dep.c
 --- gcj-4.1-4.1.0/src/boehm-gc/os_dep.c.orig	2006-02-28 09:39:11.000000000 +0100
 +++ gcj-4.1-4.1.0/src/boehm-gc/os_dep.c	2006-05-12 08:34:57.000000000 +0200
