diff -u gdm-2.6.0.8/debian/control gdm-2.6.0.8/debian/control
--- gdm-2.6.0.8/debian/control
+++ gdm-2.6.0.8/debian/control
@@ -2,7 +2,7 @@
 Section: gnome
 Priority: optional
 Maintainer: Ryan Murray <rmurray@debian.org>
-Build-Depends: libpam0g-dev, libgnomeui-dev (>= 1.96.0), librsvg2-dev, libglade2-dev, libwrap0-dev, debhelper, gettext, intltool, scrollkeeper, libselinux1-dev, libattr1-dev, xlibs-static-dev, libxt-dev
+Build-Depends: libpam0g-dev, libgnomeui-dev (>= 1.96.0), librsvg2-dev, libglade2-dev, libwrap0-dev, debhelper, gettext, intltool, scrollkeeper, libselinux1-dev [!kfreebsd-i386 !hurd-i386], libattr1-dev [!kfreebsd-i386 !hurd-i386], xlibs-static-dev, libxt-dev
 Standards-Version: 3.6.1
 
 Package: gdm
diff -u gdm-2.6.0.8/debian/rules gdm-2.6.0.8/debian/rules
--- gdm-2.6.0.8/debian/rules
+++ gdm-2.6.0.8/debian/rules
@@ -8,10 +8,24 @@
 # This is the debhelper compatability version to use.
 export DH_COMPAT=4

+export DEB_HOST_ARCH_OS ?= $(shell dpkg-architecture -qDEB_HOST_ARCH_OS 2>/dev/null)
+# Take account of old dpkg-architecture output.
+ifeq ($(DEB_HOST_ARCH_OS),)
+  DEB_HOST_ARCH_OS := $(subst -gnu,,$(shell dpkg-architecture -qDEB_HOST_GNU_SYSTEM))
+  ifeq ($(DEB_HOST_ARCH_OS),gnu)
+    DEB_HOST_ARCH_OS := hurd
+  endif
+endif 
+
+conf_args = --disable-dependency-tracking --with-tags= --prefix=/usr --libexecdir=\$${prefix}/lib --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info --sysconfdir=/etc --libexecdir=\$${prefix}/lib/gdm --localstatedir=/var/lib
+ifeq ($(DEB_HOST_ARCH_OS),linux)
+conf_args += --with-selinux
+endif
+
 configure: configure-stamp
 configure-stamp:
 	dh_testdir
-	./configure --disable-dependency-tracking --with-tags= --prefix=/usr --libexecdir=\$${prefix}/lib --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info --sysconfdir=/etc --libexecdir=\$${prefix}/lib/gdm --localstatedir=/var/lib --with-selinux
+	./configure $(conf_args)
 
 	touch configure-stamp
 
