Author: aurel32
Status: in BTS

--- gimp-gap-2.0.2.orig/libwavplayclient/wpc_msg.c
+++ gimp-gap-2.0.2/libwavplayclient/wpc_msg.c
@@ -13,7 +13,12 @@
 #include <sys/msg.h>
 #include <sys/ioctl.h>
 #include <assert.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 #include <wavplay.h>
 #include <wavfile.h>
 
--- gimp-gap-2.0.2.orig/libwavplayclient/client.c
+++ gimp-gap-2.0.2/libwavplayclient/client.c
@@ -58,7 +58,12 @@
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <assert.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 
 #include <glib.h>
 
--- gimp-gap-2.0.2.orig/libwavplayclient/msg.c
+++ gimp-gap-2.0.2/libwavplayclient/msg.c
@@ -55,7 +55,12 @@
 #include <sys/msg.h>
 #include <sys/ioctl.h>
 #include <assert.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 #include "wavplay.h"
 
 /*
