diff -u gimp-2.2.8/debian/rules gimp-2.2.8/debian/rules
--- gimp-2.2.8/debian/rules
+++ gimp-2.2.8/debian/rules
@@ -32,7 +32,7 @@
 
 # This was added in 2.2.8-3, and can be removed whenever configure gets fixed
 # to add these flags automatically
-ifeq ("$(shell dpkg-architecture -qDEB_BUILD_ARCH)","i386")
+ifeq ("$(shell dpkg-architecture -qDEB_BUILD_ARCH_CPU)","i386")
 	CFLAGS += -mmmx -msse
 endif
 
