
Status: In BTS (#217316)

--- debian/control~	2004-06-06 04:45:01.000000000 +0200
+++ debian/control	2004-06-06 05:30:31.000000000 +0200
@@ -7,7 +7,7 @@
 
 Package: gnupg
 Architecture: any
-Depends: ${shlibs:Depends}, makedev (>= 2.3.1-13) | devfsd | hurd
+Depends: ${shlibs:Depends}, makedev (>= 2.3.1-13) | devfsd | not+linux-gnu
 Suggests: gnupg-doc, xloadimage
 Conflicts: gpg-rsa, gpg-rsaref, suidmanager (<< 0.50), gpg-idea (<= 2.2)
 Replaces: gpg-rsa, gpg-rsaref
