
Status:  dirty hack obtained from upstream ML

diff -ur gnustep-back-0.9.4.old/Source/x11/XWindowBuffer.m gnustep-back-0.9.4/Source/x11/XWindowBuffer.m
--- gnustep-back-0.9.4.old/Source/x11/XWindowBuffer.m	2003-06-12 19:23:58.000000000 +0200
+++ gnustep-back-0.9.4/Source/x11/XWindowBuffer.m	2004-12-20 18:34:25.000000000 +0100
@@ -178,12 +178,9 @@
 	wi->ximage->bytes_per_line * wi->ximage->height,
 	IPC_CREAT | 0700);
 
-      if (wi->shminfo.shmid == -1)
-	{
 	  NSLog(@"Warning: shmget() failed: %m." WARN);
 	  XDestroyImage(wi->ximage);
 	  goto no_xshm;
-	}
 
       wi->shminfo.shmaddr = wi->ximage->data = shmat(wi->shminfo.shmid, 0, 0);
       if ((int)wi->shminfo.shmaddr == -1)
