diff -ur gnustep-gui-0.9.3.old/Tools/gsnd/portaudio/pa_unix_oss/pa_unix_oss.c gnustep-gui-0.9.3/Tools/gsnd/portaudio/pa_unix_oss/pa_unix_oss.c
--- gnustep-gui-0.9.3.old/Tools/gsnd/portaudio/pa_unix_oss/pa_unix_oss.c	2003-07-20 04:08:22.000000000 +0200
+++ gnustep-gui-0.9.3/Tools/gsnd/portaudio/pa_unix_oss/pa_unix_oss.c	2004-08-25 02:29:18.000000000 +0200
@@ -109,19 +109,15 @@
 #include <pthread.h>
 #include <errno.h>
 
-#ifdef __linux__
+#if defined(__linux__)
 #include <linux/soundcard.h>
-#else
-#ifdef __FreeBSD__
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <sys/soundcard.h>
-#else
-#ifdef __OpenBSD__
+#elif defined(__OpenBSD__)
 #include <soundcard.h>
-#else
+#else
 #include <machine/soundcard.h> /* JH20010905 */
 #endif
-#endif
-#endif
 
 #include "portaudio.h"
 #include "pa_host.h"
