Author: aurel32
Status: in BTS

diff -u goldedplus-1.1.4.7+1.1.5.20051016/debian/rules goldedplus-1.1.4.7+1.1.5.20051016/debian/rules
--- goldedplus-1.1.4.7+1.1.5.20051016/debian/rules
+++ goldedplus-1.1.4.7+1.1.5.20051016/debian/rules
@@ -22,6 +22,7 @@
 	cat build/Makefile debian/patches/Makefile.addon > build/Makefile.Debian
 	patch -p0 < debian/patches/compilefix-20051016snapshot.patch
 	patch -p0 < debian/patches/compilefix-20051016snapshot-noni386.patch
+	patch -p0 < debian/patches/gnu-kfreebsd.patch
 	# Build it
 	$(MAKE) -C build -f Makefile.Debian
 	touch build-stamp
--- goldedplus-1.1.4.7+1.1.5.20051016.orig/debian/patches/gnu-kfreebsd.patch
+++ goldedplus-1.1.4.7+1.1.5.20051016/debian/patches/gnu-kfreebsd.patch
@@ -0,0 +1,12 @@
+--- build/goldlib/gall/gcmpall.h.orig	2005-10-23 15:24:16.000000000 +0200
++++ build/goldlib/gall/gcmpall.h	2005-10-23 15:24:33.000000000 +0200
+@@ -68,6 +68,9 @@
+   #if defined(__FreeBSD__)
+     #define __UNIX__ __FreeBSD__
+   #endif
++  #if defined(__FreeBSD_kernel__)
++    #define __UNIX__ __FreeBSD_kernel__
++  #endif
+   #if defined(__OpenBSD__)
+     #define __UNIX__ __OpenBSD__
+   #endif
