Author: aurel32
Status: in BTS

diff -u gpsk31-0.2.4a/src/psk31-transmitter.C gpsk31-0.2.4a/src/psk31-transmitter.C
--- gpsk31-0.2.4a/src/psk31-transmitter.C
+++ gpsk31-0.2.4a/src/psk31-transmitter.C
@@ -11,7 +11,12 @@
 #include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 #include <sys/ioctl.h>
 #include "psk31-coder.h"
 #include "psk31-transmitter.h"
diff -u gpsk31-0.2.4a/src/main.C gpsk31-0.2.4a/src/main.C
--- gpsk31-0.2.4a/src/main.C
+++ gpsk31-0.2.4a/src/main.C
@@ -29,7 +29,12 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <stdio.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <fcntl.h>
--- gpsk31-0.2.4a.orig/src/spectrum.C
+++ gpsk31-0.2.4a/src/spectrum.C
@@ -30,7 +30,12 @@
 #include <stdlib.h>
 #include <sys/ioctl.h>
 #include <fcntl.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 
 #include <gtk/gtk.h>
 #include <stdio.h>
