diff -u gst-plugins-base0.10-0.10.10/debian/gstreamer-plugins-base.install gst-plugins-base0.10-0.10.10/debian/gstreamer-plugins-base.install
--- gst-plugins-base0.10-0.10.10/debian/gstreamer-plugins-base.install
+++ gst-plugins-base0.10-0.10.10/debian/gstreamer-plugins-base.install
@@ -14,7 +14,6 @@
 debian/tmp/usr/lib/gstreamer-@GST_ABI@/libgsttcp.so
 debian/tmp/usr/lib/gstreamer-@GST_ABI@/libgsttheora.so
 debian/tmp/usr/lib/gstreamer-@GST_ABI@/libgsttypefindfunctions.so
-debian/tmp/usr/lib/gstreamer-@GST_ABI@/libgstvideo4linux.so
 debian/tmp/usr/lib/gstreamer-@GST_ABI@/libgstvideorate.so
 debian/tmp/usr/lib/gstreamer-@GST_ABI@/libgstvideoscale.so
 debian/tmp/usr/lib/gstreamer-@GST_ABI@/libgstvideotestsrc.so
