Author: aurel32
Status: in BTS

diff -u gtkguitune-0.7/debian/control gtkguitune-0.7/debian/control
--- gtkguitune-0.7/debian/control
+++ gtkguitune-0.7/debian/control
@@ -2,7 +2,7 @@
 Section: x11
 Priority: optional
 Maintainer: Antonin Kral <A.Kral@sh.cvut.cz>
-Build-Depends: debhelper (>> 4.0.0), libgtkmm-dev, libgtk1.2-dev
+Build-Depends: debhelper (>> 4.0.0), libgtkmm-dev, libgtk1.2-dev, autotools-dev
 Standards-Version: 3.6.2
 
 Package: gtkguitune
only in patch2:
unchanged:
--- gtkguitune-0.7.orig/main.cc
+++ gtkguitune-0.7/main.cc
@@ -24,7 +24,12 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 
 #include <gtk/gtk.h>
 #include <gtk--/window.h>
only in patch2:
unchanged:
--- gtkguitune-0.7.orig/guitune.cc
+++ gtkguitune-0.7/guitune.cc
@@ -24,7 +24,12 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 
 #include <gtk--/main.h>
 #include <gtk--/box.h>
