--- gtkterm-0.99.4.orig/src/widgets.c
+++ gtkterm-0.99.4/src/widgets.c
@@ -42,7 +42,14 @@
 #include <gtk/gtk.h>
 #include <gdk/gdk.h>
 #include <gdk/gdkkeysyms.h>
-#include <asm/termios.h>       /* For control signals */
+#if defined (__linux__)
+#  include <asm/termios.h>       /* For control signals */
+#endif
+#if defined (__FreeBSD__) || defined (__FreeBSD_kernel__) \
+     || defined (__NetBSD__) || defined (__NetBSD_kernel__) \
+     || defined (__OpenBSD__) || defined (__OpenBSD_kernel__) 
+#  include <sys/ttycom.h>        /* For control signals */
+#endif
 #include <vte/vte.h>
 #include <stdio.h>
 #include <string.h>
