Author: aurel32
Status: in BTS

diff -u jamin-0.95.0/debian/control jamin-0.95.0/debian/control
--- jamin-0.95.0/debian/control
+++ jamin-0.95.0/debian/control
@@ -2,7 +2,7 @@
 Section: sound
 Priority: optional
 Maintainer: Robert Jordens <jordens@debian.org>
-Build-Depends: debhelper (>> 4.1.0), cdbs, dh-buildinfo, libasound2-dev, libjack0.100.0-dev, libxml2-dev, libgtk2.0-dev, ladspa-sdk, swh-plugins (>= 0.4.6), fftw3-dev, liblo0-dev, libxml-parser-perl, perl-base, indent, docbook-to-man
+Build-Depends: debhelper (>> 4.1.0), cdbs, dh-buildinfo, libasound2-dev [!kfreebsd-i386], libjack0.100.0-dev, libxml2-dev, libgtk2.0-dev, ladspa-sdk, swh-plugins (>= 0.4.6), fftw3-dev, liblo0-dev, libxml-parser-perl, perl-base, indent, docbook-to-man
 Standards-Version: 3.6.1
 
 Package: jamin
