Author: aurel32
Status: in BTS

diff -u jamvm-1.3.3/debian/control jamvm-1.3.3/debian/control
--- jamvm-1.3.3/debian/control
+++ jamvm-1.3.3/debian/control
@@ -7,7 +7,7 @@
 Standards-Version: 3.6.2
 
 Package: jamvm
-Architecture: amd64 arm hurd-i386 i386 powerpc
+Architecture: amd64 arm hurd-i386 i386 kfreebsd-i386 powerpc
 Depends: ${shlibs:Depends}, java-common, classpath (>= 2:0.18), classpath-common (>= 2:0.18)
 Provides: java-virtual-machine, java1-runtime
 Description: virtual machine which conforms to JVM specification 2
only in patch2:
unchanged:
--- jamvm-1.3.3.orig/configure
+++ jamvm-1.3.3/configure
@@ -1864,6 +1864,7 @@
 
 
 case "$host" in
+i[456]86-*-kfreebsd*-gnu*) host_cpu=i386 host_os=linux ;;
 i[456]86-*-linux*) host_cpu=i386 host_os=linux ;;
 i[456]86-*-openbsd*) host_cpu=i386 host_os=linux libdl_needed=no ;;
 x86_64-*-linux*) host_os=linux ;;
only in patch2:
unchanged:
--- jamvm-1.3.3.orig/configure.ac
+++ jamvm-1.3.3/configure.ac
@@ -55,6 +55,7 @@
 AC_CANONICAL_HOST
 
 case "$host" in
+i[[456]]86-*-kfreebsd*-gnu*) host_cpu=i386 host_os=linux ;;
 i[[456]]86-*-linux*) host_cpu=i386 host_os=linux ;;
 i[[456]]86-*-openbsd*) host_cpu=i386 host_os=linux libdl_needed=no ;;
 x86_64-*-linux*) host_os=linux ;;
