
Status: in BTS #339937

diff -u kdebase-3.5.4/debian/control kdebase-3.5.4/debian/control
--- kdebase-3.5.4/debian/control
+++ kdebase-3.5.4/debian/control
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Debian Qt/KDE Maintainers <debian-qt-kde@lists.debian.org>
 Uploaders: Isaac Clerencia <isaac@debian.org>, Pierre Habouzit <madcoder@debian.org>, Christopher Martin <chrsmrtn@debian.org>, Adeodato Simó <dato@net.com.org.es>, Riku Voipio <riku.voipio@iki.fi>, Luk Claes <luk@debian.org>
-Build-Depends: cdbs (>= 0.4.39-0.1), debhelper (>= 5.0.31), autotools-dev, gawk, gettext, kdelibs4-dev (>= 4:3.5.4), libdbus-qt-1-dev (>= 0.60), libldap2-dev, libhal-dev (>= 0.5.5.1-2), libhal-storage-dev (>= 0.5.5.1-2), libncurses5-dev, libpam0g-dev, libpopt-dev, libraw1394-dev, libsensors-dev, libsmbclient-dev, libusb-dev, libxtst-dev, libfontenc-dev, libxss-dev, libxxf86misc-dev, libxcomposite-dev, libxdamage-dev, libxkbfile-dev, xutils, sharutils, texinfo, libavahi-compat-libdnssd-dev (>= 0.4)
+Build-Depends: cdbs (>= 0.4.39-0.1), debhelper (>= 5.0.31), autotools-dev, gawk, gettext, kdelibs4-dev (>= 4:3.5.4), libdbus-qt-1-dev (>= 0.60), libldap2-dev, libhal-dev (>= 0.5.5.1-2) [!kfreebsd-i386 !kfreebsd-amd64 !hurd-i386], libhal-storage-dev (>= 0.5.5.1-2) [!kfreebsd-i386 !kfreebsd-amd64 !hurd-i386], libncurses5-dev, libpam0g-dev, libpopt-dev, libraw1394-dev [!kfreebsd-i386 !kfreebsd-amd64 !hurd-i386], libsensors-dev [!kfreebsd-i386 !kfreebsd-amd64 !hurd-i386], libsmbclient-dev, libusb-dev, libxtst-dev, libfontenc-dev, libxss-dev, libxxf86misc-dev, libxcomposite-dev, libxdamage-dev, libxkbfile-dev, xutils, sharutils, texinfo, libavahi-compat-libdnssd-dev (>= 0.4)
 Build-Depends-Indep: doxygen, qt3-doc, graphviz, gsfonts-x11, kdelibs4-doc
 Build-Conflicts: nvidia-glx
 Standards-Version: 3.7.2
diff -u kdebase-3.5.4/debian/rules kdebase-3.5.4/debian/rules
--- kdebase-3.5.4/debian/rules
+++ kdebase-3.5.4/debian/rules
@@ -33,11 +33,23 @@
 #shlibs_ver = 4:3.5.0-1
 #$(foreach p,$(PACKAGES_WITH_LIBS),$(eval DEB_DH_MAKESHLIBS_ARGS_$(p) := -V'$(p) (>= $(shlibs_ver))'))
 
+
+common-install-prehook-arch::
+	grep -v -e joystick -e view1394  debian/kcontrol.install > debian/kcontrol.install.kfreebsd-i386
+	grep -v -e joystick -e view1394  debian/kcontrol.install > debian/kcontrol.install.kfreebsd-amd64
+	grep -v -e media_propsdlgplugin  debian/kdebase-kio-plugins.install > debian/kdebase-kio-plugins.install.kfreebsd-i386
+	grep -v -e media_propsdlgplugin  debian/kdebase-kio-plugins.install > debian/kdebase-kio-plugins.install.kfreebsd-amd64
+
+clean::
+	rm -f debian/kcontrol.install.kfreebsd* debian/kdebase-kio-plugins.install.kfreebs*
+
 binary-install/kdebase-bin::
 	install -p -D -m644 debian/kcheckpass.pam debian/kdebase-bin/etc/pam.d/kcheckpass
 	install -p -D -m644 debian/kscreensaver.pam debian/kdebase-bin/etc/pam.d/kscreensaver
 
 binary-install/kdebase-data::
+# oui.db is not really arch independent, it is linux specific
+	install -p -D -m644 kcontrol/view1394/oui.db debian/kdebase-data/usr/share/apps/kcmview1394/oui.db
 	install -p -D -m644 debian/debian-kde_default.png debian/kdebase-data/usr/share/wallpapers/debian-kde_default.png
 	install -p -D -m644 debian/debian-kde_default.png.desktop debian/kdebase-data/usr/share/wallpapers/debian-kde_default.png.desktop
 	
diff -u kdebase-3.5.4/debian/kdebase-data.install kdebase-3.5.4/debian/kdebase-data.install
--- kdebase-3.5.4/debian/kdebase-data.install
+++ kdebase-3.5.4/debian/kdebase-data.install
@@ -31,7 +31,6 @@
 debian/tmp/usr/share/apps/kcmkeys/win4.kksrc
 debian/tmp/usr/share/apps/kcmkeys/wm3.kksrc
 debian/tmp/usr/share/apps/kcmlocale/pics/background.png
-debian/tmp/usr/share/apps/kcmview1394/oui.db
 debian/tmp/usr/share/apps/kconf_update/convertShortcuts.pl
 debian/tmp/usr/share/apps/kconf_update/kaccel.upd
 debian/tmp/usr/share/apps/kconf_update/kcmdisplayrc.upd
only in patch2:
unchanged:
--- kdebase-3.5.4.orig/debian/patches/27_kfreebsd_sysguard.diff
+++ kdebase-3.5.4/debian/patches/27_kfreebsd_sysguard.diff
@@ -0,0 +1,30 @@
+--- kde/configure.in	2005-08-29 01:43:19.000000000 +0200
++++ kde.patched/configure.in	2005-08-29 01:43:55.000000000 +0200
+@@ -1816,6 +1816,7 @@
+ case "$host" in 
+   *-*-linux*) ksysguardd_compile=yes; UNAME='Linux' ;;
+   *-*-freebsd*) ksysguardd_compile=yes; UNAME='FreeBSD' ;;
++  *-*-kfreebsd*-gnu) ksysguardd_compile=yes; UNAME='Linux' ;;
+   *-*-netbsd*) ksysguardd_compile=yes; UNAME='NetBSD' ;;
+   *-*-solaris*) ksysguardd_compile=yes; UNAME='Solaris' ;;
+   *-*-osf*) ksysguardd_compile=yes; UNAME='Tru64' ;;
+--- kde/configure	2005-08-29 01:42:54.000000000 +0200
++++ kde.patched/configure	2005-08-29 01:44:36.000000000 +0200
+@@ -52915,6 +52915,7 @@
+ case "$host" in
+   *-*-linux*) ksysguardd_compile=yes; UNAME='Linux' ;;
+   *-*-freebsd*) ksysguardd_compile=yes; UNAME='FreeBSD' ;;
++  *-*-kfreebsd*-gnu) ksysguardd_compile=yes; UNAME='Linux' ;;
+   *-*-netbsd*) ksysguardd_compile=yes; UNAME='NetBSD' ;;
+   *-*-solaris*) ksysguardd_compile=yes; UNAME='Solaris' ;;
+   *-*-osf*) ksysguardd_compile=yes; UNAME='Tru64' ;;
+--- kde/ksysguard/configure.in.in	2005-08-29 09:59:21.000000000 +0200
++++ kde.patched/ksysguard/configure.in.in	2005-08-29 10:00:05.000000000 +0200
+@@ -3,6 +3,7 @@
+ case "$host" in 
+   *-*-linux*) ksysguardd_compile=yes; UNAME='Linux' ;;
+   *-*-freebsd*) ksysguardd_compile=yes; UNAME='FreeBSD' ;;
++  *-*-kfreebsd*-gnu) ksysguardd_compile=yes; UNAME='Linux' ;;
+   *-*-netbsd*) ksysguardd_compile=yes; UNAME='NetBSD' ;;
+   *-*-solaris*) ksysguardd_compile=yes; UNAME='Solaris' ;;
+   *-*-osf*) ksysguardd_compile=yes; UNAME='Tru64' ;;
