diff -u kdepim-3.4.2/debian/control kdepim-3.4.2/debian/control
--- kdepim-3.4.2/debian/control
+++ kdepim-3.4.2/debian/control
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Debian Qt/KDE Maintainers <debian-qt-kde@lists.debian.org>
 Uploaders: Daniel Schepler <schepler@debian.org>
-Build-Depends: autoconf, automake1.9, bison, byacc, bzip2, cdbs (>= 0.4.27), debhelper (>= 4.2.21), flex, gawk, gettext, groff-base, kdelibs4-dev (>= 4:3.4.2), libbluetooth1-dev, libgnokii2-dev (>= 0.6.5), libgpgme11-dev, liblockdev1-dev, libmal-dev, libpisock-dev, libreadline5-dev, libsasl2-dev, libtool (>= 1.5), libxpm-dev, libxss-dev, sharutils, texinfo
+Build-Depends: autoconf, automake1.9, bison, byacc, bzip2, cdbs (>= 0.4.27), debhelper (>= 4.2.21), flex, gawk, gettext, groff-base, kdelibs4-dev (>= 4:3.4.2), libbluetooth1-dev [!kfreebsd-i386], libgnokii2-dev (>= 0.6.5), libgpgme11-dev, liblockdev1-dev, libmal-dev, libpisock-dev, libreadline5-dev, libsasl2-dev, libtool (>= 1.5), libxpm-dev, libxss-dev, sharutils, texinfo
 Build-Depends-Indep: doxygen, qt3-doc, kdelibs4-doc
 Standards-Version: 3.6.2
 
