Author: aurel32
Status: in BTS

--- kwirelessmonitor-0.5.91.orig/src/wm_bsd.cpp
+++ kwirelessmonitor-0.5.91/src/wm_bsd.cpp
@@ -7,7 +7,7 @@
  *                                                                         *
  ***************************************************************************/
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
