--- libdvdread-0.9.4.orig/dvdread/dvd_reader.c
+++ libdvdread-0.9.4/dvdread/dvd_reader.c
@@ -32,7 +32,7 @@
 #include <limits.h>
 #include <dirent.h>
  
-#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__bsdi__)|| defined(__DARWIN__)
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__bsdi__)|| defined(__DARWIN__) || defined(__FreeBSD_kernel__)
 #define SYS_BSD 1
 #endif
 
only in patch2:
unchanged:
--- libdvdread-0.9.4.orig/dvdread/bswap.h
+++ libdvdread-0.9.4/dvdread/bswap.h
@@ -35,7 +35,7 @@
 #include <sys/param.h>
 #endif
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__GLIBC__)
 #include <byteswap.h>
 #define B2N_16(x) x = bswap_16(x)
 #define B2N_32(x) x = bswap_32(x)
