Author: aurel32
Status: in BTS

diff -u libflash-0.4.13/debian/control libflash-0.4.13/debian/control
--- libflash-0.4.13/debian/control
+++ libflash-0.4.13/debian/control
@@ -2,7 +2,7 @@
 Priority: optional
 Section: libs
 Maintainer: Nobuhiro Iwamatsu <hemamu@t-base.ne.jp>
-Build-Depends: debhelper (>> 4.0.0), zlib1g-dev, libjpeg62-dev, txt2man, libx11-dev, libxt-dev, libxext-dev, libmad0-dev, libasound2-dev
+Build-Depends: debhelper (>> 4.0.0), zlib1g-dev, libjpeg62-dev, txt2man, libx11-dev, libxt-dev, libxext-dev, libmad0-dev, libasound2-dev [!kfreebsd-i386]
 Standards-Version: 3.6.1
 
 Package: libflash0

