diff -u libmng-1.0.8/libmng_types.h libmng-1.0.8/libmng_types.h
--- libmng-1.0.8/libmng_types.h
+++ libmng-1.0.8/libmng_types.h
@@ -167,7 +167,7 @@
 #ifndef ZLIB_DLL
 #undef FAR
 #endif
-#if defined(WIN32) || defined(linux)   /* different header locations */
+#if defined(WIN32) || defined(linux) || defined(__FreeBSD_kernel__)  /* different header locations */
 #include "lcms.h"
 #else
 #include "lcms/lcms.h"
