Author: aurel32
Status: ugly
Note: libtool update is also needed

--- libmal-0.40.orig/malsync/mal/common/AGNet.h
+++ libmal-0.40/malsync/mal/common/AGNet.h
@@ -61,7 +61,7 @@
 #               include <sys/filio.h>
 #               include <arpa/inet.h>
 #           else
-#               if (defined(BSD) && BSD >= 199306) || defined(_HPUX_SOURCE)
+#               if (defined(BSD) && BSD >= 199306) || defined(_HPUX_SOURCE) || defined(__FreeBSD_kernel__)
 #                   include <sys/ioctl.h>
 #                   include <arpa/inet.h>
 #               else
