
Status: in BTS and upstream BTS.

--- libpcap-2005.02.06.old/configure.in	2004-12-18 00:25:36.000000000 +0100
+++ libpcap-2005.02.06/configure.in	2005-02-06 11:19:10.000000000 +0100
@@ -70,7 +70,7 @@
 AC_MSG_CHECKING(packet capture type)
 if test ! -z "$with_pcap" ; then
 	V_PCAP="$withval"
-elif test -r /dev/bpf0 ; then
+elif test -r /dev/bpf0 || test -r /usr/include/net/bpf.h ; then
 	V_PCAP=bpf
 elif test -r /usr/include/net/pfilt.h ; then
 	V_PCAP=pf
