Author: aurel32
Status: don't merge. temporary changes until libpcap transition begins

diff -u libpcap0.8-0.9.5/debian/control libpcap0.8-0.9.5/debian/control
--- libpcap0.8-0.9.5/debian/control
+++ libpcap0.8-0.9.5/debian/control
@@ -5,11 +5,19 @@
 Build-Depends: debhelper (>= 5), flex, bison, dpatch, perl
 Standards-Version: 3.7.2
 
+Package: libpcap-dev
+Section: oldlibs
+Architecture: all
+Depends: libpcap0.8-dev
+Priority: extra
+Description: Development library for libpcap (transitional package)
+ Empty package to facilitate upgrades, can be safely removed.
+
 Package: libpcap0.8-dev
 Section: libdevel
 Architecture: any
 Depends: libpcap0.8 (= ${Source-Version}), libc6-dev
-Conflicts: libpcap-dev, libpcap0.7-dev
+Conflicts: libpcap-dev (<< 0.9), libpcap0.7-dev
 Description: Development library and header files for libpcap 0.8
  Headers, static libraries, and documentation for the libpcap library.
  .
diff -u libpcap0.8-0.9.5/debian/changelog libpcap0.8-0.9.5/debian/changelog
--- libpcap0.8-0.9.5/debian/changelog
+++ libpcap0.8-0.9.5/debian/changelog
@@ -1,3 +1,9 @@
+libpcap0.8 (0.9.5-1+kbsd) unreleased; urgency=low
+
+  * Added a libpcap-dev package as libpcap0.7 is fucked on kfreebsd-i386.
+
+ -- Aurelien Jarno <aurel32@debian.org>  Wed, 12 Oct 2005 20:50:30 +0200
+
 libpcap0.8 (0.9.5-1) unstable; urgency=low
 
   * New upstream release: no API changes, no ABI changes.
