Author: aurel32
Status: in BTS

diff -u libtool-1.5.22/debian/control libtool-1.5.22/debian/control
--- libtool-1.5.22/debian/control
+++ libtool-1.5.22/debian/control
@@ -1,5 +1,5 @@
 Source: libtool
-Build-Depends: debhelper (>= 4.0), texi2html, texinfo, file, g77 | fortran77-compiler, gcj [!mips !mipsel !netbsd-i386 !hurd-i386 !kfreebsd-i386], automake1.9, autoconf, autotools-dev, dpatch
+Build-Depends: debhelper (>= 4.0), texi2html, texinfo, file, g77 | fortran77-compiler, gcj [!mips !mipsel !netbsd-i386 !hurd-i386 !kfreebsd-i386 !kfreebsd-amd64], automake1.9, autoconf, autotools-dev, dpatch
 Section: devel
 Priority: optional
 Maintainer: Kurt Roeckx <kurt@roeckx.be>
diff -u libtool-1.5.22/debian/patches/00list libtool-1.5.22/debian/patches/00list
--- libtool-1.5.22/debian/patches/00list
+++ libtool-1.5.22/debian/patches/00list
@@ -6 +6 @@
-
+anon_versioning_kfreebsd.dpatch
--- libtool-1.5.22.orig/debian/patches/anon_versioning_kfreebsd.dpatch
+++ libtool-1.5.22/debian/patches/anon_versioning_kfreebsd.dpatch
@@ -0,0 +1,33 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## anon_versioning_kfreebsd.dpatch by Aurelien Jarno <aurel32@debian.org>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: No description.
+
+@DPATCH@
+--- ./libtool.m4.orig	2006-02-28 13:43:10.000000000 +0100
++++ ./libtool.m4	2006-03-07 22:05:08.000000000 +0100
+@@ -5589,6 +5589,23 @@
+       fi
+       ;;
+ 
++    kfreebsd*-gnu)
++      if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
++      	tmp_addflag=
++	_LT_AC_TAGVAR(archive_cmds, $1)='$CC -shared'"$tmp_addflag"' $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
++
++	if test $supports_anon_versioning = yes; then
++	  _LT_AC_TAGVAR(archive_expsym_cmds, $1)='$echo "{ global:" > $output_objdir/$libname.ver~
++  cat $export_symbols | sed -e "s/\(.*\)/\1;/" >> $output_objdir/$libname.ver~
++  $echo "local: *; };" >> $output_objdir/$libname.ver~
++	  $CC -shared'"$tmp_addflag"' $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname ${wl}-version-script ${wl}$output_objdir/$libname.ver -o $lib'
++	fi
++	_LT_AC_TAGVAR(link_all_deplibs, $1)=no
++      else
++	_LT_AC_TAGVAR(ld_shlibs, $1)=no
++      fi
++      ;;
++
+     solaris*)
+       if $LD -v 2>&1 | grep 'BFD 2\.8' > /dev/null; then
+ 	_LT_AC_TAGVAR(ld_shlibs, $1)=no
