Status: in BTS
Author: aurel32

diff -u lush-1.1/debian/control lush-1.1/debian/control
--- lush-1.1/debian/control
+++ lush-1.1/debian/control
@@ -7,7 +7,7 @@
 
 Package: lush
 Section: devel
-Architecture: alpha amd64 arm hppa i386 m68k mips mipsel powerpc ppc64 s390 sparc
+Architecture: alpha amd64 arm hppa i386 m68k mips mipsel powerpc ppc64 s390 sparc kfreebsd-i386
 Depends: lush-library (= ${Source-Version})
 Description: Lisp Universal Shell Executable
  Lush is a programming language and environment that is based on the Lisp
diff -u lush-1.1/debian/patches/00list lush-1.1/debian/patches/00list
--- lush-1.1/debian/patches/00list
+++ lush-1.1/debian/patches/00list
@@ -4,0 +5 @@
+05-kfreebsd-gnu
--- lush-1.1.orig/debian/patches/05-kfreebsd-gnu.dpatch
+++ lush-1.1/debian/patches/05-kfreebsd-gnu.dpatch
@@ -0,0 +1,25 @@
+#!/bin/sh /usr/share/dpatch/dpatch-run
+## 05-kfreebsd-gnu.dpatch by Aurelien Jarno <aurel32@debian.org>
+##
+## DP: Fixes FTBFS on GNU/kFreeBSD
+
+@DPATCH@
+
+--- lush-1.1.orig/src/unix.c
++++ lush-1.1/src/unix.c
+@@ -631,6 +631,7 @@
+ #ifndef BROKEN_SETSIG
+ #ifdef I_SETSIG
+ #ifdef I_GETSIG
++#ifdef SIGPOLL
+           flag = 0;
+           ioctl(fd, I_GETSIG, &flag);
+           if (ioctl(fd,I_SETSIG, flag|S_INPUT) != -1)
+@@ -642,6 +643,7 @@
+             }
+ #endif
+ #endif
++#endif
+ #endif /* !BROKEN_SETSIG */
+ #ifndef BROKEN_TIMER
+ #ifdef ITIMER_REAL
