Author: aurel32
Status: in BTS

diff -u moc-2.3.1/debian/control moc-2.3.1/debian/control
--- moc-2.3.1/debian/control
+++ moc-2.3.1/debian/control
@@ -5,7 +5,7 @@
 Uploaders: Bartosz Fenski <fenio@debian.org>
 Standards-Version: 3.6.2.1
 Build-Conflicts-Indep: libcurl3-gssapi
-Build-Depends: debhelper (>> 4.0.0), libncurses5-dev | libncursesw5-dev, libasound2-dev, libvorbis-dev (>= 1.1.0), libmad0-dev, libid3tag0-dev, zlib1g-dev, libsndfile1-dev, libflac-dev (>= 1.1.1), libogg-dev (>=1.1.2), dpatch (>= 2.0.10), libsamplerate0-dev, libspeex-dev, libcurl3-dev (>=7.14.0-3), libmpcdec-dev, libtagc0-dev, libjack0.100.0-dev
+Build-Depends: debhelper (>> 4.0.0), libncurses5-dev | libncursesw5-dev, libasound2-dev [!kfreebsd-i386], libvorbis-dev (>= 1.1.0), libmad0-dev, libid3tag0-dev, zlib1g-dev, libsndfile1-dev, libflac-dev (>= 1.1.1), libogg-dev (>=1.1.2), dpatch (>= 2.0.10), libsamplerate0-dev, libspeex-dev, libcurl3-dev (>=7.14.0-3), libmpcdec-dev, libtagc0-dev, libjack0.100.0-dev
 
 Package: moc
 Architecture: any
