diff -Nur mozilla-firefox-1.0+dfsg.1.old/debian/control.in mozilla-firefox-1.0+dfsg.1/debian/control.in
--- mozilla-firefox-1.0+dfsg.1.old/debian/control.in	2005-01-18 04:02:27.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/debian/control.in	2005-01-18 04:02:55.000000000 +0100
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Eric Dorland <eric@debian.org>
 Uploaders: Mike Hommey <mh@glandium.org>
-Build-Depends: debhelper (>= 4.0), libx11-dev, libxp-dev, libxt-dev, libgtk2.0-dev, zlib1g-dev, liborbit2-dev, libidl-dev (>= 0.8.0), zip, libxft2-dev, libpng12-dev, libmng-dev, libjpeg62-dev, libxrender-dev, libkrb5-dev, libgnome2-dev, libgconf2-dev, libgnomevfs2-dev, patch, sharutils, m4, gcc-3.4 [amd64], g++-3.4 [amd64]
+Build-Depends: debhelper (>= 4.0), libx11-dev, libxp-dev, libxt-dev, libgtk2.0-dev, zlib1g-dev, liborbit2-dev, libidl-dev (>= 0.8.0), zip, libxft2-dev, libpng12-dev, libmng-dev, libjpeg62-dev, libxrender-dev, libkrb5-dev, libgnome2-dev, libgconf2-dev, libgnomevfs2-dev, patch, sharutils, m4, gcc-3.4 [@x86_64@], g++-3.4 [@x86_64@], autotools-dev, autoconf2.13, type-handling (>= 0.2.1)
 Standards-Version: 3.6.1
 
 Package: mozilla-firefox
diff -Nur mozilla-firefox-1.0+dfsg.1.old/debian/rules mozilla-firefox-1.0+dfsg.1/debian/rules
--- mozilla-firefox-1.0+dfsg.1.old/debian/rules	2005-01-18 03:08:12.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/debian/rules	2005-01-18 04:03:57.000000000 +0100
@@ -7,6 +7,8 @@
 
 SHELL=/bin/bash # I use bashisms
 
+DEB_HOST_GNU_CPU	?= $(shell dpkg-architecture -qDEB_HOST_GNU_CPU)
+
 # build phoenix
 export MOZ_PHOENIX=1
 export MOZ_FIREBIRD=1
@@ -28,15 +30,15 @@
 else
 	OPTFLAGS = -O2
 
-ifeq ($(DEB_BUILD_ARCH),ia64)
+ifeq ($(DEB_HOST_GNU_CPU),ia64)
 	OPTFLAGS=-O
 endif
 
-ifeq ($(DEB_BUILD_ARCH),powerpc)
+ifeq ($(DEB_HOST_GNU_CPU),powerpc)
 	OPTFLAGS=-O
 endif
 
-ifeq ($(DEB_BUILD_ARCH),arm)
+ifeq ($(DEB_HOST_GNU_CPU),arm)
 	OPTFLAGS=-O
 endif
 
@@ -49,7 +51,7 @@
 	DEBUG_FLAG = --enable-debug
 endif
 
-ifeq ($(DEB_BUILD_ARCH),amd64)
+ifeq ($(DEB_HOST_GNU_CPU),x86_64)
 CC=gcc-3.4
 CXX=g++-3.4
 export CC CXX
@@ -94,6 +96,9 @@
 configure-stamp:
 	dh_testdir
 
+	for i in `find . -name config.guess` ; do cp /usr/share/misc/config.guess $$i ; done
+	for i in `find . -name config.sub` ; do cp /usr/share/misc/config.sub $$i ; done
+	for i in . nsprpub directory/c-sdk ; do cd $(CURDIR)/$$i && autoconf2.13 ; done
 	./configure --prefix=/usr --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info $(CONFIGURE_OPTIONS)
 
 	touch configure-stamp
@@ -116,6 +121,9 @@
 
 	-$(MAKE) distclean
 
+	sed -e "s/@x86_64@/`type-handling x86_64 any`/g" \
+	< debian/control.in > debian/control
+
 	dh_clean
 
 install: build
diff -Nur mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/config/nsinstall.c mozilla-firefox-1.0+dfsg.1/directory/c-sdk/config/nsinstall.c
--- mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/config/nsinstall.c	2002-03-27 11:06:18.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/directory/c-sdk/config/nsinstall.c	2005-01-18 03:49:24.000000000 +0100
@@ -64,7 +64,7 @@
 #if defined(AIX) || defined(BSDI) || defined(HPUX) || defined(LINUX) \
     || defined(SUNOS4) || defined(SCO) || defined(UNIXWARE) \
     || defined(RHAPSODY) || defined(NEXTSTEP) || defined(QNX) \
-    || defined(BEOS) || defined(VMS) || defined(DARWIN)
+    || defined(BEOS) || defined(VMS) || defined(DARWIN) || defined(GLIBC)
 #undef HAVE_LCHOWN
 #endif
 
@@ -106,7 +106,7 @@
 }
 #endif /* NEXTSTEP */
 
-#ifdef LINUX
+#if defined(LINUX) || defined(GLIBC)
 #include <getopt.h>
 #endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/configure.in mozilla-firefox-1.0+dfsg.1/directory/c-sdk/configure.in
--- mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/configure.in	2004-09-14 22:59:40.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/directory/c-sdk/configure.in	2005-01-18 03:49:24.000000000 +0100
@@ -1123,7 +1123,7 @@
 	esac
     ;;
 
-*-linux*)
+*-linux* | *-k*bsd*-gnu | *-gnu*)
     if test -z "$USE_NSPR_THREADS"; then
         USE_PTHREADS=1
         IMPL_STRATEGY=_PTH
diff -Nur mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/include/portable.h mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/include/portable.h
--- mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/include/portable.h	2003-09-17 05:18:10.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/include/portable.h	2005-01-18 03:49:24.000000000 +0100
@@ -116,7 +116,7 @@
  * some systems don't have the BSD re_comp and re_exec routines
  */
 #ifndef NEED_BSDREGEX
-#if ( defined( SYSV ) || defined( NETBSD ) || defined( freebsd ) || defined( linux ) || defined( DARWIN )) && !defined(sgi)
+#if ( defined( SYSV ) || defined( NETBSD ) || defined( freebsd ) || defined( linux ) || defined(__GNU__) || defined(__GLIBC__) || defined( DARWIN )) && !defined(sgi)
 #define NEED_BSDREGEX
 #endif
 #endif
@@ -145,7 +145,7 @@
  * Is snprintf() part of the standard C runtime library?
  */
 #if !defined(HAVE_SNPRINTF)
-#if defined(SOLARIS) || defined(LINUX) || defined(HPUX)
+#if defined(SOLARIS) || defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) || defined(HPUX)
 #define HAVE_SNPRINTF
 #endif
 #endif
@@ -165,7 +165,7 @@
  * for select()
  */
 #if !defined(WINSOCK) && !defined(_WINDOWS) && !defined(macintosh) && !defined(XP_OS2)
-#if defined(hpux) || defined(LINUX) || defined(SUNOS4) || defined(XP_BEOS)
+#if defined(hpux) || defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) || defined(SUNOS4) || defined(XP_BEOS)
 #include <sys/time.h>
 #else
 #include <sys/select.h>
@@ -252,15 +252,14 @@
 #define HAVE_GETGRNAM_R
 #endif
 
-#if defined(SNI) || defined(LINUX1_2)
+#if defined(SNI) || defined(LINUX1_2) || defined(__GNU__) || defined(__GLIBC__)
 int strcasecmp(const char *, const char *);
 #ifdef SNI
 int strncasecmp(const char *, const char *, int);
-#endif /* SNI */
-#ifdef LINUX1_2
+#else /* SNI */
 int strncasecmp(const char *, const char *, size_t);
-#endif /* LINUX1_2 */
-#endif /* SNI || LINUX1_2 */
+#endif
+#endif /* SNI || LINUX1_2 || __GNU__ || __GLIBC__ */
 
 #if defined(_WINDOWS) || defined(macintosh) || defined(XP_OS2) || defined(DARWIN)
 #define GETHOSTBYNAME( n, r, b, l, e )  gethostbyname( n )
@@ -276,7 +275,7 @@
     defined(UNIXWARE) || defined(SUNOS4) || defined(SNI) || defined(BSDI) || \
     defined(NCR) || defined(OSF1) || defined(NEC) || defined(VMS) || \
     ( defined(HPUX10) && !defined(_REENTRANT)) || defined(HPUX11) || \
-    defined(UnixWare) || defined(LINUX) || defined(NETBSD) || \
+    defined(UnixWare) || defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) || defined(NETBSD) || \
     defined(FREEBSD) || defined(OPENBSD) || \
     (defined(AIX) && !defined(USE_REENTRANT_LIBC))
 #define GETHOSTBYNAME( n, r, b, l, e )  gethostbyname( n )
@@ -300,7 +299,7 @@
 #define GETHOSTBYNAME( n, r, b, l, e )  gethostbyname_r( n, r, b, l, e )
 #endif
 #if defined(HPUX9) || defined(LINUX1_2) || defined(LINUX2_0) || \
-    defined(LINUX2_1) || defined(SUNOS4) || defined(SNI) || \
+    defined(LINUX2_1) || defined(__GNU__) || defined(__GLIBC__) || defined(SUNOS4) || defined(SNI) || \
     defined(SCOOS) || defined(BSDI) || defined(NCR) || \
     defined(NEC) || ( defined(HPUX10) && !defined(_REENTRANT)) || \
     (defined(AIX) && !defined(USE_REENTRANT_LIBC))
@@ -320,7 +319,7 @@
 #endif
 #if defined(hpux9) || defined(LINUX1_2) || defined(SUNOS4) || defined(SNI) || \
     defined(SCOOS) || defined(BSDI) || defined(NCR) || defined(VMS) || \
-    defined(NEC) || defined(LINUX) || (defined(AIX) && !defined(USE_REENTRANT_LIBC))
+    defined(NEC) || defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) || (defined(AIX) && !defined(USE_REENTRANT_LIBC))
 #define STRTOK( s1, s2, l )		strtok( s1, s2 )
 #else
 #define HAVE_STRTOK_R
diff -Nur mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/libraries/libldap/Makefile.in mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/libraries/libldap/Makefile.in
--- mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/libraries/libldap/Makefile.in	2004-03-10 19:47:25.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/libraries/libldap/Makefile.in	2005-01-18 03:52:16.000000000 +0100
@@ -168,7 +168,7 @@
 EXTRA_LIBS = -L$(dist_libdir) -l$(LBER_LIBNAME) $(OS_LIBS) -lc
 endif
 
-ifeq ($(OS_ARCH), Linux)
+ifneq (,$(filter $(OS_ARCH),Linux GNU GNU_kFreeBSD GNU_kNetBSD))
 EXTRA_LIBS = -L$(dist_libdir) -l$(LBER_LIBNAME)
 endif
 
@@ -286,7 +286,7 @@
 ifdef SO_FILES_TO_REMOVE
 	-$(RM) $(SO_FILES_TO_REMOVE)
 endif
-ifneq (,$(filter AIX Linux HP-UX Darwin BeOS QNX NetBSD OSF1 OpenBSD, $(OS_ARCH)))
+ifneq (,$(filter $(OS_ARCH),AIX Linux GNU GNU_kFreeBSD GNU_kNetBSD HP-UX Darwin BeOS QNX NetBSD OSF1 OpenBSD))
 	$(LINK_LIB) ../liblber/$(OBJDIR_NAME)/*.a
 else
 ifeq ($(OS_ARCH),OS2)
diff -Nur mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/libraries/libldap/request.c mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/libraries/libldap/request.c
--- mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/libraries/libldap/request.c	2002-03-26 22:53:46.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/libraries/libldap/request.c	2005-01-18 03:49:24.000000000 +0100
@@ -627,7 +627,7 @@
 	LDAPConn	*lc;
 	char        msg[256];
 /* CTIME for this platform doesn't use this. */
-#if !defined(SUNOS4) && !defined(_WIN32) && !defined(LINUX) && !defined(macintosh)
+#if !defined(SUNOS4) && !defined(_WIN32) && !defined(GLIBC) && !defined(macintosh)
 	char		buf[26];
 #endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/libraries/libldap/tmplout.c mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/libraries/libldap/tmplout.c
--- mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/libraries/libldap/tmplout.c	2002-03-26 22:53:52.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/libraries/libldap/tmplout.c	2005-01-18 03:49:24.000000000 +0100
@@ -903,7 +903,8 @@
     time_t		gmttime;
 /* CTIME for this platform doesn't use this. */
 #if !defined(SUNOS4) && !defined(BSDI) && !defined(LINUX1_2) && \
-    !defined(SNI) && !defined(_WIN32) && !defined(macintosh) && !defined(LINUX)
+    !defined(SNI) && !defined(_WIN32) && !defined(macintosh) && !defined(GLIBC)
+
     char		buf[26];
 #endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/libraries/libprldap/ldappr-error.c mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/libraries/libprldap/ldappr-error.c
--- mozilla-firefox-1.0+dfsg.1.old/directory/c-sdk/ldap/libraries/libprldap/ldappr-error.c	2003-03-25 14:57:38.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/directory/c-sdk/ldap/libraries/libprldap/ldappr-error.c	2005-01-18 03:49:24.000000000 +0100
@@ -218,7 +218,7 @@
 
 #if defined(__hpux) || defined(_AIX) || defined(OSF1) || defined(DARWIN) || \
   defined(BEOS) || defined(FREEBSD) || defined(BSDI) || defined(VMS) || \
-  defined(OPENBSD) || defined(NETBSD)
+  defined(OPENBSD) || defined(NETBSD) || defined(__FreeBSD_kernel__)
 #define EDEADLOCK       -1
 #endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/config/nsinstall.c mozilla-firefox-1.0+dfsg.1/nsprpub/config/nsinstall.c
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/config/nsinstall.c	2002-12-12 01:29:08.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/config/nsinstall.c	2005-01-18 03:48:49.000000000 +0100
@@ -95,7 +95,7 @@
 }
 #endif /* NEXTSTEP */
 
-#ifdef LINUX
+#if defined(LINUX) || defined(GLIBC)
 #include <getopt.h>
 #endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/configure.in mozilla-firefox-1.0+dfsg.1/nsprpub/configure.in
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/configure.in	2004-09-14 22:59:41.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/configure.in	2005-01-18 03:48:49.000000000 +0100
@@ -919,6 +919,7 @@
     fi
     AC_DEFINE(XP_UNIX)
     AC_DEFINE(FREEBSD)
+    AC_DEFINE(__FreeBSD_kernel__)
     AC_DEFINE(HAVE_BSD_FLOCK)
     AC_DEFINE(HAVE_SOCKLEN_T)
     CFLAGS="$CFLAGS $(DSO_CFLAGS) -ansi -Wall"
@@ -1174,7 +1175,7 @@
 	esac
     ;;
 
-*-linux*)
+*-linux* | *-gnu* | *-k*bsd*-gnu)
     if test -z "$USE_NSPR_THREADS"; then
         USE_PTHREADS=1
         IMPL_STRATEGY=_PTH
@@ -1185,7 +1186,10 @@
     AC_DEFINE(_SVID_SOURCE)
     AC_DEFINE(_LARGEFILE64_SOURCE)
     AC_DEFINE(HAVE_FCNTL_FILE_LOCKING)
-    AC_DEFINE(LINUX)
+    case "$target_os" in
+      linux*) AC_DEFINE(LINUX) ;;
+      gnu* | k*bsd*-gnu) AC_DEFINE(GLIBC) ;;
+    esac
     CFLAGS="$CFLAGS -ansi -Wall"
     CXXFLAGS="$CXXFLAGS -ansi -Wall"
     MDCPUCFG_H=_linux.cfg
@@ -1416,6 +1420,7 @@
 *-netbsd*)
     AC_DEFINE(XP_UNIX)
     AC_DEFINE(NETBSD)
+    AC_DEFINE(__NetBSD_kernel__)
     AC_DEFINE(HAVE_BSD_FLOCK)
     USE_NSPR_THREADS=1
     MDCPUCFG_H=_netbsd.cfg
@@ -2215,7 +2220,7 @@
             _PTHREAD_LDFLAGS=-pthread
         fi
         ;;
-    *-linux*)
+    *-linux* | *-gnu* | *-k*bsd*-gnu)
         AC_DEFINE(_REENTRANT)
         ;;
     esac
@@ -2299,7 +2304,7 @@
         fi
     fi
     ;;
-*-linux*)
+*-linux* | *-gnu* | *-k*bsd*-gnu)
     if test -n "$USE_NSPR_THREADS"; then
         AC_DEFINE(_PR_LOCAL_THREADS_ONLY)
     fi
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/include/md/_linux.cfg mozilla-firefox-1.0+dfsg.1/nsprpub/pr/include/md/_linux.cfg
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/include/md/_linux.cfg	2004-03-09 04:18:20.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/include/md/_linux.cfg	2005-01-18 03:48:49.000000000 +0100
@@ -39,10 +39,14 @@
 #define XP_UNIX
 #endif
 
-#ifndef LINUX
+#if !defined(LINUX) && defined(__linux__)
 #define LINUX
 #endif
 
+#if !defined(GLIBC) && (defined(__GNU__) || defined(__GLIBC__))
+#define GLIBC
+#endif
+
 #define PR_AF_INET6 10  /* same as AF_INET6 */
 
 #ifdef __powerpc__
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/include/md/_pth.h mozilla-firefox-1.0+dfsg.1/nsprpub/pr/include/md/_pth.h
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/include/md/_pth.h	2003-09-16 22:44:04.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/include/md/_pth.h	2005-01-18 03:48:50.000000000 +0100
@@ -136,10 +136,14 @@
 	(!memcmp(&(t), &pt_zero_tid, sizeof(pthread_t)))
 #define _PT_PTHREAD_COPY_THR_HANDLE(st, dt)   (dt) = (st)
 #elif defined(IRIX) || defined(OSF1) || defined(AIX) || defined(SOLARIS) \
-	|| defined(HPUX) || defined(LINUX) || defined(FREEBSD) \
+	|| defined(HPUX) || defined(LINUX) || defined(GLIBC) || defined(FREEBSD) \
 	|| defined(NETBSD) || defined(OPENBSD) || defined(BSDI) \
 	|| defined(VMS) || defined(NTO) || defined(DARWIN) \
 	|| defined(UNIXWARE)
+#ifdef __GNU__
+# error Using Hurd pthreads?
+/* Hurd pthreads don't have an invalid value for pthread_t. -- rmh */
+#endif
 #define _PT_PTHREAD_INVALIDATE_THR_HANDLE(t)  (t) = 0
 #define _PT_PTHREAD_THR_HANDLE_IS_INVALID(t)  (t) == 0
 #define _PT_PTHREAD_COPY_THR_HANDLE(st, dt)   (dt) = (st)
@@ -228,7 +232,7 @@
 #define PT_PRIO_MAX            sched_get_priority_max(SCHED_OTHER)
 #endif /* defined(_PR_DCETHREADS) */
 
-#elif defined(LINUX) || defined(FREEBSD)
+#elif defined(LINUX) || defined(GLIBC) || defined(FREEBSD)
 #define PT_PRIO_MIN            sched_get_priority_min(SCHED_OTHER)
 #define PT_PRIO_MAX            sched_get_priority_max(SCHED_OTHER)
 #elif defined(NTO)
@@ -283,7 +287,7 @@
 		onemillisec.tv_nsec = 1000000L;			\
         nanosleep(&onemillisec,NULL);			\
     PR_END_MACRO
-#elif defined(HPUX) || defined(LINUX) || defined(SOLARIS) \
+#elif defined(HPUX) || defined(LINUX) || defined(GLIBC) || defined(SOLARIS) \
 	|| defined(FREEBSD) || defined(NETBSD) || defined(OPENBSD) \
 	|| defined(BSDI) || defined(NTO) || defined(DARWIN) \
 	|| defined(UNIXWARE)
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/include/md/_unixos.h mozilla-firefox-1.0+dfsg.1/nsprpub/pr/include/md/_unixos.h
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/include/md/_unixos.h	2002-03-31 19:17:38.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/include/md/_unixos.h	2005-01-18 03:48:50.000000000 +0100
@@ -43,7 +43,7 @@
  * Linux: FD_SETSIZE is defined in /usr/include/sys/select.h and should
  * not be redefined.
  */
-#if !defined(LINUX) && !defined(DARWIN) && !defined(NEXTSTEP)
+#if !defined(LINUX) && !defined(GLIBC) && !defined(DARWIN) && !defined(NEXTSTEP)
 #ifndef FD_SETSIZE
 #define FD_SETSIZE  4096
 #endif
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/include/md/prosdep.h mozilla-firefox-1.0+dfsg.1/nsprpub/pr/include/md/prosdep.h
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/include/md/prosdep.h	2001-10-26 06:54:01.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/include/md/prosdep.h	2005-01-18 03:48:50.000000000 +0100
@@ -87,7 +87,7 @@
 #elif defined(IRIX)
 #include "md/_irix.h"
 
-#elif defined(LINUX)
+#elif defined(LINUX) || defined(GLIBC)
 #include "md/_linux.h"
 
 #elif defined(OSF1)
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/md/prosdep.c mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/md/prosdep.c
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/md/prosdep.c	2001-10-26 06:54:05.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/md/prosdep.c	2005-01-18 03:48:50.000000000 +0100
@@ -60,7 +60,7 @@
 
     /* Get page size */
 #ifdef XP_UNIX
-#if defined SUNOS4 || defined LINUX || defined BSDI || defined AIX \
+#if defined SUNOS4 || defined LINUX || defined GLIBC || defined BSDI || defined AIX \
         || defined FREEBSD || defined NETBSD || defined OPENBSD \
         || defined DARWIN || defined NEXTSTEP
     _pr_pageSize = getpagesize();
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/md/unix/unix.c mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/md/unix/unix.c
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/md/unix/unix.c	2002-12-12 01:19:57.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/md/unix/unix.c	2005-01-18 03:48:50.000000000 +0100
@@ -65,7 +65,7 @@
  * PRInt32* pointer to a _PRSockLen_t* pointer.
  */
 #if defined(HAVE_SOCKLEN_T) \
-    || (defined(LINUX) && defined(__GLIBC__) && __GLIBC__ >= 2)
+    || (defined(__GLIBC__) && __GLIBC__ >= 2)
 #define _PRSockLen_t socklen_t
 #elif defined(IRIX) || defined(HPUX) || defined(OSF1) || defined(SOLARIS) \
     || defined(AIX4_1) || defined(LINUX) || defined(SONY) \
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/md/unix/uxproces.c mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/md/unix/uxproces.c
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/md/unix/uxproces.c	2002-03-15 00:20:54.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/md/unix/uxproces.c	2005-01-18 03:48:50.000000000 +0100
@@ -89,7 +89,7 @@
  * that can share the virtual address space and file descriptors.
  */
 #if (defined(IRIX) && !defined(_PR_PTHREADS)) \
-        || (defined(LINUX) && defined(_PR_PTHREADS))
+        || ((defined(LINUX) || defined(GLIBC)) && defined(_PR_PTHREADS))
 #define _PR_SHARE_CLONES
 #endif
 
@@ -103,7 +103,7 @@
  */
 
 #if defined(_PR_GLOBAL_THREADS_ONLY) \
-	|| (defined(_PR_PTHREADS) && !defined(LINUX))
+	|| (defined(_PR_PTHREADS) && !defined(LINUX) && !defined(GLIBC))
 #define _PR_NATIVE_THREADS
 #endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/md/unix/uxrng.c mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/md/unix/uxrng.c
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/md/unix/uxrng.c	2002-06-14 05:22:14.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/md/unix/uxrng.c	2005-01-18 03:48:50.000000000 +0100
@@ -135,7 +135,7 @@
     return 0;
 }
 
-#elif (defined(LINUX) || defined(FREEBSD) || defined(NETBSD) || defined(OPENBSD))
+#elif (defined(LINUX) || defined(__FreeBSD_kernel__) || defined(__NetBSD_kernel__) || defined(OPENBSD))
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/misc/prnetdb.c mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/misc/prnetdb.c
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/misc/prnetdb.c	2004-09-02 01:07:28.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/misc/prnetdb.c	2005-01-18 03:48:50.000000000 +0100
@@ -105,7 +105,7 @@
 #define _PR_HAVE_GETPROTO_R_INT
 #endif
 
-#if (defined(LINUX) && defined(__GLIBC__) && __GLIBC__ >= 2)
+#if (defined(__GLIBC__) && __GLIBC__ >= 2)
 #define _PR_HAVE_GETPROTO_R
 #define _PR_HAVE_5_ARG_GETPROTO_R
 #endif
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/pthreads/ptio.c mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/pthreads/ptio.c
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/src/pthreads/ptio.c	2004-02-04 02:31:33.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/src/pthreads/ptio.c	2005-01-18 03:48:50.000000000 +0100
@@ -203,7 +203,7 @@
 #if defined(SOLARIS)
 #define _PRSockOptVal_t char *
 #elif defined(IRIX) || defined(OSF1) || defined(AIX) || defined(HPUX) \
-    || defined(LINUX) || defined(FREEBSD) || defined(BSDI) || defined(VMS) \
+    || defined(LINUX) || defined(GLIBC) || defined(FREEBSD) || defined(BSDI) || defined(VMS) \
     || defined(NTO) || defined(OPENBSD) || defined(DARWIN) \
     || defined(UNIXWARE) || defined(NETBSD)
 #define _PRSockOptVal_t void *
@@ -217,7 +217,7 @@
 #define _PRSelectFdSetArg_t void *
 #elif defined(IRIX) || (defined(AIX) && !defined(AIX4_1)) \
     || defined(OSF1) || defined(SOLARIS) \
-    || defined(HPUX10_30) || defined(HPUX11) || defined(LINUX) \
+    || defined(HPUX10_30) || defined(HPUX11) || defined(LINUX) || defined(GLIBC) \
     || defined(FREEBSD) || defined(NETBSD) || defined(OPENBSD) \
     || defined(BSDI) || defined(VMS) || defined(NTO) || defined(DARWIN) \
     || defined(UNIXWARE)
@@ -301,7 +301,7 @@
  * most current systems.
  */
 #if defined(HAVE_SOCKLEN_T) \
-    || (defined(LINUX) && defined(__GLIBC__) && __GLIBC__ >= 2)
+    || (defined(__GLIBC__) && __GLIBC__ >= 2)
 typedef socklen_t pt_SockLen;
 #elif (defined(AIX) && !defined(AIX4_1)) \
     || defined(VMS)
@@ -357,14 +357,14 @@
     int nbytes_to_send;                     /* size of header and file */
 #endif  /* SOLARIS */
 
-#ifdef LINUX
+#if defined(LINUX) || defined(GLIBC)
     /*
      * For sendfile()
      */
     int in_fd;                              /* descriptor of file to send */
     off_t offset;
     size_t count;
-#endif  /* LINUX */
+#endif  /* LINUX || GLIBC */
  
     PRIntervalTime timeout;                 /* client (relative) timeout */
 
@@ -3255,7 +3255,7 @@
 };
 
 #if defined(HPUX) || defined(OSF1) || defined(SOLARIS) || defined (IRIX) \
-    || defined(AIX) || defined(LINUX) || defined(FREEBSD) || defined(NETBSD) \
+    || defined(AIX) || defined(LINUX) || defined(GLIBC) || defined(FREEBSD) || defined(NETBSD) \
     || defined(OPENBSD) || defined(BSDI) || defined(VMS) || defined(NTO) \
     || defined(DARWIN) || defined(UNIXWARE)
 #define _PR_FCNTL_FLAGS O_NONBLOCK
@@ -4747,7 +4747,7 @@
 
 #include <sys/types.h>
 #include <sys/time.h>
-#if !defined(SUNOS4) && !defined(HPUX) && !defined(LINUX)
+#if !defined(SUNOS4) && !defined(HPUX) && !defined(LINUX) && !defined(GLIBC)
 #include <sys/select.h>
 #endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/tests/Makefile.in mozilla-firefox-1.0+dfsg.1/nsprpub/pr/tests/Makefile.in
--- mozilla-firefox-1.0+dfsg.1.old/nsprpub/pr/tests/Makefile.in	2004-03-23 01:40:17.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/nsprpub/pr/tests/Makefile.in	2005-01-18 03:48:50.000000000 +0100
@@ -372,8 +372,8 @@
     EXTRA_LIBS = -lsocket -lnsl -lgen -lresolv
 endif
 
-ifeq ($(OS_ARCH), Linux)
-    ifeq ($(OS_RELEASE), 1.2)
+ifneq (,$(filter $(OS_ARCH),Linux GNU GNU_kFreeBSD GNU_kNetBSD))
+    ifeq ($(OS_ARCH)-$(OS_RELEASE), Linux-1.2)
         EXTRA_LIBS = -ldl
     else
         LDOPTS += -Xlinker -rpath $(ABSOLUTE_LIB_DIR)
@@ -544,7 +544,7 @@
 endif
 endif
 
-ifeq ($(OS_TARGET),Linux)
+ifneq (,$(filter $(OS_TARGET),Linux GNU GNU_kFreeBSD GNU_kNetBSD))
 ECHO = /bin/echo
 endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/coreconf/GNU.mk mozilla-firefox-1.0+dfsg.1/security/coreconf/GNU.mk
--- mozilla-firefox-1.0+dfsg.1.old/security/coreconf/GNU.mk	1970-01-01 01:00:00.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/security/coreconf/GNU.mk	2005-01-18 03:49:24.000000000 +0100
@@ -0,0 +1,37 @@
+#
+# The contents of this file are subject to the Mozilla Public
+# License Version 1.1 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of
+# the License at http://www.mozilla.org/MPL/
+# 
+# Software distributed under the License is distributed on an "AS
+# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# rights and limitations under the License.
+# 
+# The Original Code is the Netscape security libraries.
+# 
+# The Initial Developer of the Original Code is Netscape
+# Communications Corporation.  Portions created by Netscape are 
+# Copyright (C) 1994-2000 Netscape Communications Corporation.  All
+# Rights Reserved.
+# 
+# Contributor(s):
+# 
+# Alternatively, the contents of this file may be used under the
+# terms of the GNU General Public License Version 2 or later (the
+# "GPL"), in which case the provisions of the GPL are applicable 
+# instead of those above.  If you wish to allow use of your 
+# version of this file only under the terms of the GPL and not to
+# allow others to use your version of this file under the MPL,
+# indicate your decision by deleting the provisions above and
+# replace them with the notice and other provisions required by
+# the GPL.  If you do not delete the provisions above, a recipient
+# may use your version of this file under either the MPL or the
+# GPL.
+#
+# Config stuff for GNU
+#
+
+KERNEL=gnumach
+include $(CORE_DEPTH)/coreconf/Linux.mk
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/coreconf/GNU_kFreeBSD.mk mozilla-firefox-1.0+dfsg.1/security/coreconf/GNU_kFreeBSD.mk
--- mozilla-firefox-1.0+dfsg.1.old/security/coreconf/GNU_kFreeBSD.mk	1970-01-01 01:00:00.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/security/coreconf/GNU_kFreeBSD.mk	2005-01-18 03:49:24.000000000 +0100
@@ -0,0 +1,37 @@
+#
+# The contents of this file are subject to the Mozilla Public
+# License Version 1.1 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of
+# the License at http://www.mozilla.org/MPL/
+# 
+# Software distributed under the License is distributed on an "AS
+# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# rights and limitations under the License.
+# 
+# The Original Code is the Netscape security libraries.
+# 
+# The Initial Developer of the Original Code is Netscape
+# Communications Corporation.  Portions created by Netscape are 
+# Copyright (C) 1994-2000 Netscape Communications Corporation.  All
+# Rights Reserved.
+# 
+# Contributor(s):
+# 
+# Alternatively, the contents of this file may be used under the
+# terms of the GNU General Public License Version 2 or later (the
+# "GPL"), in which case the provisions of the GPL are applicable 
+# instead of those above.  If you wish to allow use of your 
+# version of this file only under the terms of the GPL and not to
+# allow others to use your version of this file under the MPL,
+# indicate your decision by deleting the provisions above and
+# replace them with the notice and other provisions required by
+# the GPL.  If you do not delete the provisions above, a recipient
+# may use your version of this file under either the MPL or the
+# GPL.
+#
+# Config stuff for GNU/kFreeBSD
+#
+
+KERNEL=kfreebsd
+include $(CORE_DEPTH)/coreconf/Linux.mk
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/coreconf/GNU_kNetBSD.mk mozilla-firefox-1.0+dfsg.1/security/coreconf/GNU_kNetBSD.mk
--- mozilla-firefox-1.0+dfsg.1.old/security/coreconf/GNU_kNetBSD.mk	1970-01-01 01:00:00.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/security/coreconf/GNU_kNetBSD.mk	2005-01-18 03:49:24.000000000 +0100
@@ -0,0 +1,37 @@
+#
+# The contents of this file are subject to the Mozilla Public
+# License Version 1.1 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of
+# the License at http://www.mozilla.org/MPL/
+# 
+# Software distributed under the License is distributed on an "AS
+# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# rights and limitations under the License.
+# 
+# The Original Code is the Netscape security libraries.
+# 
+# The Initial Developer of the Original Code is Netscape
+# Communications Corporation.  Portions created by Netscape are 
+# Copyright (C) 1994-2000 Netscape Communications Corporation.  All
+# Rights Reserved.
+# 
+# Contributor(s):
+# 
+# Alternatively, the contents of this file may be used under the
+# terms of the GNU General Public License Version 2 or later (the
+# "GPL"), in which case the provisions of the GPL are applicable 
+# instead of those above.  If you wish to allow use of your 
+# version of this file only under the terms of the GPL and not to
+# allow others to use your version of this file under the MPL,
+# indicate your decision by deleting the provisions above and
+# replace them with the notice and other provisions required by
+# the GPL.  If you do not delete the provisions above, a recipient
+# may use your version of this file under either the MPL or the
+# GPL.
+#
+# Config stuff for GNU/kNetBSD
+#
+
+KERNEL=knetbsd
+include $(CORE_DEPTH)/coreconf/Linux.mk
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/coreconf/Linux.mk mozilla-firefox-1.0+dfsg.1/security/coreconf/Linux.mk
--- mozilla-firefox-1.0+dfsg.1.old/security/coreconf/Linux.mk	2004-02-11 03:33:51.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/security/coreconf/Linux.mk	2005-01-18 03:51:16.000000000 +0100
@@ -30,7 +30,7 @@
 # may use your version of this file under either the MPL or the
 # GPL.
 #
-# Config stuff for Linux
+# Config stuff for GNU and variants
 #
 
 include $(CORE_DEPTH)/coreconf/UNIX.mk
@@ -51,58 +51,58 @@
 DEFAULT_COMPILER = gcc
 
 ifeq ($(OS_TEST),m68k)
-	OS_REL_CFLAGS	= -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS	= -D_XOPEN_SOURCE
 	CPU_ARCH	= m68k
 else		
 ifeq ($(OS_TEST),ppc)
-	OS_REL_CFLAGS	= -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS	= -D_XOPEN_SOURCE
 	CPU_ARCH	= ppc
 else
 ifeq ($(OS_TEST),alpha)
-        OS_REL_CFLAGS   = -D_ALPHA_ -DLINUX1_2 -D_XOPEN_SOURCE
+        OS_REL_CFLAGS   = -D_ALPHA_ -D_XOPEN_SOURCE
 	CPU_ARCH	= alpha
 else
 ifeq ($(OS_TEST),ia64)
-	OS_REL_CFLAGS	= -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS	= -D_XOPEN_SOURCE
 	CPU_ARCH	= ia64
 else
 ifeq ($(OS_TEST),x86_64)
-	OS_REL_CFLAGS	= -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS	= -D_XOPEN_SOURCE
 	CPU_ARCH	= x86_64
 else
 ifeq ($(OS_TEST),sparc)
-	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS   = -D_XOPEN_SOURCE
 	CPU_ARCH        = sparc
 else
 ifeq ($(OS_TEST),sparc64)
-	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS   = -D_XOPEN_SOURCE
 	CPU_ARCH        = sparc
 else
 ifeq (,$(filter-out arm% sa110,$(OS_TEST)))
-	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS   = -D_XOPEN_SOURCE
 	CPU_ARCH        = arm
 else
 ifeq ($(OS_TEST),parisc)
-	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS   = -D_XOPEN_SOURCE
 	CPU_ARCH        = hppa
 else
 ifeq ($(OS_TEST),parisc64)
-	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS   = -D_XOPEN_SOURCE
 	CPU_ARCH        = hppa
 else
 ifeq ($(OS_TEST),s390)
-	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS   = -D_XOPEN_SOURCE
 	CPU_ARCH        = s390
 else
 ifeq ($(OS_TEST),s390x)
-	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS   = -D_XOPEN_SOURCE
 	CPU_ARCH        = s390x
 else
 ifeq ($(OS_TEST),mips)
-	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS   = -D_XOPEN_SOURCE
 	CPU_ARCH        = mips
 else
-	OS_REL_CFLAGS	= -DLINUX1_2 -Di386 -D_XOPEN_SOURCE
+	OS_REL_CFLAGS	= -Di386 -D_XOPEN_SOURCE
 	CPU_ARCH	= x86
 endif
 endif
@@ -118,11 +118,20 @@
 endif
 endif
 
+isgnu	= nope
+ifeq ($(KERNEL),linux)
+OS_REL_CFLAGS	= -DLINUX1_2
+ifeq ($(OS_RELEASE),2.0)
+	OS_REL_CFLAGS	+= -DLINUX2_0
+	isgnu		= yes
+endif
+else
+isgnu		= yes
+endif
 
 LIBC_TAG		= _glibc
 
-ifeq ($(OS_RELEASE),2.0)
-	OS_REL_CFLAGS	+= -DLINUX2_0
+ifeq ($(isgnu), yes)
 	MKSHLIB		= $(CC) -shared -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so)
 	ifdef BUILD_OPT
 		OPTIMIZER	= -O2
@@ -138,20 +147,27 @@
 OS_PTHREAD = -lpthread 
 endif
 
-OS_CFLAGS		= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -ansi -Wall -pipe -DLINUX -Dlinux -D_POSIX_SOURCE -D_BSD_SOURCE -DHAVE_STRERROR
+OS_CFLAGS		= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -ansi -Wall -pipe -D_POSIX_SOURCE -D_BSD_SOURCE -DHAVE_STRERROR
 OS_LIBS			= -L/lib $(OS_PTHREAD) -ldl -lc
+ifeq ($(KERNEL),linux)
+OS_CFLAGS += -DLINUX -Dlinux
+endif
 
 ifdef USE_PTHREADS
 	DEFINES		+= -D_REENTRANT
 endif
 
+ifeq ($(KERNEL),linux)
 ARCH			= linux
+else
+ARCH			= gnu
+endif
 
 DSO_CFLAGS		= -fPIC
 DSO_LDOPTS		= -shared
 DSO_LDFLAGS		=
 
-# INCLUDES += -I/usr/include -Y/usr/include/linux
+# INCLUDES += -I/usr/include
 G++INCLUDES		= -I/usr/include/g++
 
 #
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/coreconf/config.mk mozilla-firefox-1.0+dfsg.1/security/coreconf/config.mk
--- mozilla-firefox-1.0+dfsg.1.old/security/coreconf/config.mk	2004-01-28 01:01:56.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/security/coreconf/config.mk	2005-01-18 03:49:24.000000000 +0100
@@ -59,7 +59,7 @@
 #######################################################################
 
 TARGET_OSES = FreeBSD BSD_OS NetBSD OpenUNIX OS2 QNX Darwin BeOS OpenBSD \
-              OpenVMS AIX
+              OpenVMS AIX GNU GNU_kFreeBSD GNU_kNetBSD
 
 ifeq (,$(filter-out $(TARGET_OSES),$(OS_TARGET)))
 include $(CORE_DEPTH)/coreconf/$(OS_TARGET).mk
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/nss/cmd/platlibs.mk mozilla-firefox-1.0+dfsg.1/security/nss/cmd/platlibs.mk
--- mozilla-firefox-1.0+dfsg.1.old/security/nss/cmd/platlibs.mk	2004-10-15 23:13:50.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/security/nss/cmd/platlibs.mk	2005-01-18 03:49:24.000000000 +0100
@@ -182,7 +182,7 @@
 # If GNU ld is used, we must use the -rpath-link option to tell
 # the linker where to find libsoftokn3.so, an implicit dependency
 # of libnss3.so.
-ifeq (,$(filter-out BSD_OS FreeBSD Linux NetBSD, $(OS_ARCH)))
+ifeq (,$(filter-out BSD_OS FreeBSD Linux GNU GNU_kFreeBSD GNU_kNetBSD NetBSD, $(OS_ARCH)))
 EXTRA_SHARED_LIBS += -Wl,-rpath-link,$(DIST)/lib
 endif
 
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/freebl/ecl/Makefile mozilla-firefox-1.0+dfsg.1/security/nss/lib/freebl/ecl/Makefile
--- mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/freebl/ecl/Makefile	2003-10-17 15:45:34.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/security/nss/lib/freebl/ecl/Makefile	2005-01-18 03:49:24.000000000 +0100
@@ -65,7 +65,7 @@
 ifeq ($(TARGET),v8SOLARIS)
 ECL_USE_FP=1
 else
-ifeq ($(TARGET),x86LINUX)
+ifeq (,)
 ECL_USE_FP=1
 endif
 endif
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/freebl/mpi/target.mk mozilla-firefox-1.0+dfsg.1/security/nss/lib/freebl/mpi/target.mk
--- mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/freebl/mpi/target.mk	2003-10-17 15:45:36.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/security/nss/lib/freebl/mpi/target.mk	2005-01-18 03:49:24.000000000 +0100
@@ -203,13 +203,13 @@
 export OBJECT_MODE
 endif
 
-ifeq ($(TARGET),x86LINUX)
+ifeq (,)
 #Linux
 AS_OBJS = mpi_x86.o
 MPICMN += -DMP_ASSEMBLY_MULTIPLY -DMP_ASSEMBLY_SQUARE -DMP_ASSEMBLY_DIV_2DX1D
 MPICMN += -DMP_MONT_USE_MP_MUL
-CFLAGS= -O2 -fPIC -DLINUX1_2 -Di386 -D_XOPEN_SOURCE -DLINUX2_1 -ansi -Wall \
- -pipe -DLINUX -Dlinux -D_POSIX_SOURCE -D_BSD_SOURCE -DHAVE_STRERROR \
+CFLAGS= -O2 -fPIC -Di386 -D_XOPEN_SOURCE -ansi -Wall \
+ -pipe -D_POSIX_SOURCE -D_BSD_SOURCE -DHAVE_STRERROR \
  -DXP_UNIX -UDEBUG -DNDEBUG -D_REENTRANT $(MPICMN)
 #CFLAGS= -g -fPIC -DLINUX1_2 -Di386 -D_XOPEN_SOURCE -DLINUX2_1 -ansi -Wall \
  -pipe -DLINUX -Dlinux -D_POSIX_SOURCE -D_BSD_SOURCE -DHAVE_STRERROR \
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/freebl/unix_rand.c mozilla-firefox-1.0+dfsg.1/security/nss/lib/freebl/unix_rand.c
--- mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/freebl/unix_rand.c	2005-01-18 03:08:11.000000000 +0100
+++ mozilla-firefox-1.0+dfsg.1/security/nss/lib/freebl/unix_rand.c	2005-01-18 03:49:24.000000000 +0100
@@ -78,7 +78,8 @@
 }
 
 #if defined(SCO) || defined(UNIXWARE) || defined(BSDI) || defined(FREEBSD) \
-    || defined(NETBSD) || defined(NTO) || defined(DARWIN) || defined(OPENBSD)
+    || defined(NETBSD) || defined(NTO) || defined(DARWIN) || defined(OPENBSD) \
+    || defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)
 #include <sys/times.h>
 
 #define getdtablesize() sysconf(_SC_OPEN_MAX)
@@ -273,7 +274,7 @@
 }
 #endif /* IBM R2 */
 
-#if defined(LINUX)
+#if defined(LINUX) && !defined(GLIBC)
 #include <linux/kernel.h>
 
 static size_t
@@ -804,7 +805,7 @@
 #ifdef DO_PS
 For now it is considered that it is too expensive to run the ps command
 for the small amount of entropy it provides.
-#if defined(__sun) && (!defined(__svr4) && !defined(SVR4)) || defined(bsdi) || defined(LINUX)
+#if defined(__sun) && (!defined(__svr4) && !defined(SVR4)) || defined(bsdi) || defined(LINUX) || defined(__FreeBSD_kernel__)
     static char ps_cmd[] = "ps aux";
 #else
     static char ps_cmd[] = "ps -el";
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/ssl/sslmutex.c mozilla-firefox-1.0+dfsg.1/security/nss/lib/ssl/sslmutex.c
--- mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/ssl/sslmutex.c	2003-08-29 00:23:59.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/security/nss/lib/ssl/sslmutex.c	2005-01-18 03:49:24.000000000 +0100
@@ -87,7 +87,7 @@
     return SECSuccess;
 }
 
-#if defined(LINUX) || defined(AIX) || defined(VMS) || defined(BEOS) || defined(BSDI) || defined(NETBSD) || defined(OPENBSD)
+#if defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) || defined(AIX) || defined(VMS) || defined(BEOS) || defined(BSDI) || defined(NETBSD) || defined(OPENBSD)
 
 #include <unistd.h>
 #include <fcntl.h>
diff -Nur mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/ssl/sslmutex.h mozilla-firefox-1.0+dfsg.1/security/nss/lib/ssl/sslmutex.h
--- mozilla-firefox-1.0+dfsg.1.old/security/nss/lib/ssl/sslmutex.h	2002-05-18 05:24:17.000000000 +0200
+++ mozilla-firefox-1.0+dfsg.1/security/nss/lib/ssl/sslmutex.h	2005-01-18 03:49:24.000000000 +0100
@@ -77,7 +77,7 @@
 
 typedef int    sslPID;
 
-#elif defined(LINUX) || defined(AIX) || defined(VMS) || defined(BEOS) || defined(BSDI) || defined(NETBSD) || defined(OPENBSD)
+#elif defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) || defined(AIX) || defined(VMS) || defined(BEOS) || defined(BSDI) || defined(NETBSD) || defined(OPENBSD)
 
 #include <sys/types.h>
 #include "prtypes.h"
