diff -u mutt-1.5.8/debian/rules mutt-1.5.8/debian/rules
--- mutt-1.5.8/debian/rules
+++ mutt-1.5.8/debian/rules
@@ -44,6 +44,12 @@
 		$B/stamp-h.in $B/config.h.in
 	touch $@
 
+	# update config.guess and config.sub
+	-test -r /usr/share/misc/config.sub && \
+	cp -f /usr/share/misc/config.sub $B/config.sub
+	-test -r /usr/share/misc/config.guess && \
+	cp -f /usr/share/misc/config.guess $B/config.guess
+
 configure: $(STAMP_DIR)/configure
 $(STAMP_DIR)/configure: $(STAMP_DIR)/source.make
 	CC=cc ac_cv_path_ISPELL=ispell \
diff -u mutt-1.5.8/debian/control mutt-1.5.8/debian/control
--- mutt-1.5.8/debian/control
+++ mutt-1.5.8/debian/control
@@ -3,7 +3,7 @@
 Priority: standard
 Maintainer: Marco d'Itri <md@linux.it>
 Uploaders: Marco d'Itri <md@linux.it>, Adeodato Simó <asp16@alu.ua.es>
-Build-Depends: debhelper (>= 4), linuxdoc-tools-text, gettext, groff, libncurses5-dev, libsasl2-dev, libgnutls11-dev, libidn11-dev, zlib1g-dev, libncursesw5-dev, libdb4.3-dev
+Build-Depends: debhelper (>= 4), linuxdoc-tools-text, gettext, groff, libncurses5-dev, libsasl2-dev, libgnutls11-dev, libidn11-dev, zlib1g-dev, libncursesw5-dev, libdb4.3-dev, autotools-dev
 Standards-Version: 3.6.1.1
 
 Package: mutt
