diff -ur nessus-libraries-2.0.12.old/debian/rules nessus-libraries-2.0.12/debian/rules
--- nessus-libraries-2.0.12.old/debian/rules	2004-08-11 15:03:32.000000000 +0200
+++ nessus-libraries-2.0.12/debian/rules	2004-08-11 15:22:08.000000000 +0200
@@ -15,6 +15,14 @@
 CFLAGS += -O2
 endif
 
+DEB_BUILD_GNU_CPU	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_CPU)
+DEB_BUILD_GNU_SYSTEM	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_SYSTEM)
+DEB_BUILD_GNU_TYPE	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
+
+ifneq (, $(filter $(DEB_BUILD_GNU_SYSTEM), kfreebsd-gnu knetbsd-gnu))
+DEB_BUILD_GNU_TYPE = $(DEB_BUILD_GNU_CPU)-gnu
+endif
+
 build: build-stamp
 build-stamp: config.status
 	dh_testdir
@@ -25,7 +33,7 @@
 	CFLAGS="$(CFLAGS)" ./configure --prefix=/usr --sysconfdir=/etc \
 	--localstatedir=/var  --mandir='$${prefix}/share/man' \
 	--enable-static --enable-gmp --enable-zlib \
-	--enable-cipher --with-ssl
+	--enable-cipher --with-ssl $(DEB_BUILD_GNU_TYPE)
 # --disable-nessuspcap
 # --enable-debug
 
