Author: aurel32
Status: in BTS

diff -u oaf-0.6.10/debian/patches/002_libtool_fix oaf-0.6.10/debian/patches/002_libtool_fix
--- oaf-0.6.10/debian/patches/002_libtool_fix
+++ oaf-0.6.10/debian/patches/002_libtool_fix
@@ -1,5 +1,14 @@
---- oaf-0.6.10.orig/configure   2002-04-16 03:27:55.000000000 -0700
-+++ oaf-0.6.10/configure        2005-10-02 21:58:36.000000000 -0700
+--- oaf-0.6.10.orig/configure	2002-04-16 12:27:55.000000000 +0200
++++ oaf-0.6.10/configure	2005-10-03 16:56:07.000000000 +0200
+@@ -2478,7 +2478,7 @@
+   esac
+   ;;
+ 
+-freebsd*)
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
 @@ -2525,16 +2525,8 @@
    lt_cv_deplibs_check_method=pass_all
    ;;
@@ -21,0 +31,28 @@
+@@ -4410,7 +4402,7 @@
+     ;;
+ 
+   # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-  freebsd*)
++  freebsd* | kfreebsd*-gnu)
+     archive_cmds='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
+     hardcode_libdir_flag_spec='-R$libdir'
+     hardcode_direct=yes
+@@ -4813,6 +4805,18 @@
+   dynamic_linker=no
+   ;;
+ 
++kfreebsd*-gnu)
++  version_type=linux
++  need_lib_prefix=no
++  need_version=no
++  library_names_spec='${libname}${release}.so$versuffix ${libname}${release}.so$major $libname.so'
++  soname_spec='${libname}${release}.so$major'
++  shlibpath_var=LD_LIBRARY_PATH
++  shlibpath_overrides_runpath=no
++  hardcode_into_libs=yes
++  dynamic_linker='GNU ld.so'
++  ;;
++
+ freebsd*)
+   objformat=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
+   version_type=freebsd-$objformat
