Author: aurel32
Status: in BTS

diff -u ocaml-3.09.0/debian/patches/00list ocaml-3.09.0/debian/patches/00list
--- ocaml-3.09.0/debian/patches/00list
+++ ocaml-3.09.0/debian/patches/00list
@@ -3,0 +4 @@
+kbsd-gnu
diff -u ocaml-3.09.0/debian/control ocaml-3.09.0/debian/control
--- ocaml-3.09.0/debian/control
+++ ocaml-3.09.0/debian/control
@@ -95,7 +95,7 @@
  you do not require any graphical capilities for your runtime.
 
 Package: ocaml-native-compilers
-Architecture: alpha amd64 arm hppa i386 ia64 powerpc sparc
+Architecture: alpha amd64 arm hppa hurd-i386 i386 ia64 kfreebsd-i386 powerpc sparc
 Depends: ocaml-nox (= ${Source-Version}), ocaml-nox-${F:OCamlABI}, gcc, binutils, ${shlibs:Depends}
 Provides: ocaml-best-compilers
 Description: Native code compilers of the ocaml suite (the .opt ones)
--- ocaml-3.09.0.orig/debian/patches/kbsd-gnu.dpatch
+++ ocaml-3.09.0/debian/patches/kbsd-gnu.dpatch
@@ -0,0 +1,32 @@
+#! /bin/sh -e 
+## kbsd-gnu.dpatch by Aurelien Jarno <aurel32@debian.org>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Port to GNU/k*BSD
+
+if [ $# -ne 1 ]; then
+    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
+    exit 1
+fi
+case "$1" in
+    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
+    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;	
+    *)
+	echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
+	exit 1;;
+esac
+
+exit 0
+@DPATCH@
+--- ocaml-3.09.0/configure	2005-11-10 00:50:29.000000000 +0100
++++ ocaml-3.09.0/configure	2005-11-10 00:53:37.000000000 +0100
+@@ -480,7 +480,7 @@
+ 
+ if test $withsharedlibs = "yes"; then
+   case "$host" in
+-    *-*-linux-gnu|*-*-linux|*-*-freebsd[3-9]*)
++    *-*-linux-gnu|*-*-linux|*-*-k*bsd*-gnu|*-*-gnu*|*-*-freebsd[3-9]*)
+       sharedcccompopts="-fPIC"
+       mksharedlib="$bytecc -shared -o"
+       bytecclinkopts="$bytecclinkopts -Wl,-E"
+
