Author: aurel32
Status: in BTS

diff -u openal-0.2005080600/debian/control openal-0.2005080600/debian/control
--- openal-0.2005080600/debian/control
+++ openal-0.2005080600/debian/control
@@ -2,7 +2,7 @@
 Section: libs
 Priority: optional
 Maintainer: Dan Helfman <witten@debian.org>
-Build-Depends: debhelper (>= 4.0.1), texinfo, libsdl1.2-dev (>= 1.2.7), libesd0-dev, libsmpeg-dev, libvorbis-dev (>= 1.0.1), libasound2-dev (>= 1.0.5), libarts1-dev (>= 1.3.0)
+Build-Depends: debhelper (>= 4.0.1), texinfo, libsdl1.2-dev (>= 1.2.7), libesd0-dev, libsmpeg-dev, libvorbis-dev (>= 1.0.1), libasound2-dev [!kfreebsd-i386] (>= 1.0.5), libarts1-dev (>= 1.3.0)
 Standards-Version: 3.6.2.1
 
 Package: libopenal-dev
