
Status: in BTS

diff -ur openldap2-2.1.30.old/debian/rules openldap2-2.1.30/debian/rules
--- openldap2-2.1.30.old/debian/rules	2004-08-18 19:54:14.000000000 +0200
+++ openldap2-2.1.30/debian/rules	2004-08-18 18:44:52.000000000 +0200
@@ -18,7 +18,18 @@
 INSTALL_PROGRAM += -s
 endif
 
-CONFIG	      = $(shell grep -v "^\#" debian/configure.options)
+DEB_BUILD_GNU_TYPE	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
+DEB_BUILD_GNU_CPU	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_CPU)
+DEB_BUILD_GNU_SYSTEM	?= $(shell dpkg-architecture -qDEB_BUILD_GNU_SYSTEM)
+
+ifeq ($(DEB_BUILD_GNU_SYSTEM),kfreebsd-gnu)
+DEB_BUILD_GNU_TYPE	= $(DEB_BUILD_GNU_CPU)-gnu
+endif
+ifeq ($(DEB_BUILD_GNU_SYSTEM),knetbsd-gnu)
+DEB_BUILD_GNU_TYPE	= $(DEB_BUILD_GNU_CPU)-gnu
+endif
+
+CONFIG	      = $(shell grep -v "^\#" debian/configure.options) $(DEB_BUILD_GNU_TYPE)
 
 rootdir		:= $(shell pwd)
 installdir	:= $(rootdir)/debian/install
diff -ur openldap2-2.1.30.old/libraries/libldap/Makefile.in openldap2-2.1.30/libraries/libldap/Makefile.in
--- openldap2-2.1.30.old/libraries/libldap/Makefile.in	2004-08-18 19:54:14.000000000 +0200
+++ openldap2-2.1.30/libraries/libldap/Makefile.in	2004-08-18 19:25:25.000000000 +0200
@@ -37,7 +37,7 @@
 XLIBS = $(LIBRARY) $(LDAP_LIBLBER_LA) $(LDAP_LIBLUTIL_A)
 XXLIBS = $(SECURITY_LIBS) $(LUTIL_LIBS)
 NT_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
-UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
+UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS) $(LTHREAD_LIBS)
 
 apitest:	$(XLIBS) apitest.o
 	$(LTLINK) -o $@ apitest.o $(LIBS)
