Author: aurel32
Status: in BTS

diff -u openssh-krb5-3.8.1p1/configure.ac openssh-krb5-3.8.1p1/configure.ac
--- openssh-krb5-3.8.1p1/configure.ac
+++ openssh-krb5-3.8.1p1/configure.ac
@@ -258,6 +258,14 @@
 	AC_DEFINE(WITH_ABBREV_NO_TTY)
 	AC_DEFINE(LOCKED_PASSWD_STRING, "*LK*")
 	;;
+*-*-k*bsd*-gnu)
+	check_for_libcrypt_later=1
+	AC_DEFINE(PAM_TTY_KLUDGE)
+	AC_DEFINE(LOCKED_PASSWD_PREFIX, "!")
+	AC_DEFINE(SPT_TYPE,SPT_REUSEARGV)
+	AC_DEFINE(_PATH_BTMP, "/var/log/btmp", [log for bad login attempts])
+	AC_DEFINE(USE_BTMP, 1, [Use btmp to log bad logins])
+	;;
 *-*-linux*)
 	no_dev_ptmx=1
 	check_for_libcrypt_later=1
--- openssh-krb5-3.8.1p1.orig/configure
+++ openssh-krb5-3.8.1p1/configure
@@ -4054,6 +4054,29 @@
 EOF
 
 	;;
+*-*-k*bsd*-gnu)
+	check_for_libcrypt_later=1
+	cat >>confdefs.h <<\_ACEOF
+#define PAM_TTY_KLUDGE 1
+_ACEOF
+
+	cat >>confdefs.h <<\_ACEOF
+#define LOCKED_PASSWD_PREFIX "!"
+_ACEOF
+
+	cat >>confdefs.h <<\_ACEOF
+#define SPT_TYPE SPT_REUSEARGV
+_ACEOF
+
+
+	cat >>confdefs.h <<\_ACEOF
+#define _PATH_BTMP "/var/log/btmp"
+_ACEOF
+
+	cat >>confdefs.h <<\_ACEOF
+#define USE_BTMP 1
+_ACEOF
+	;;
 *-*-linux*)
 	no_dev_ptmx=1
 	check_for_libcrypt_later=1
