Author: aurel32
Status: in BTS

diff -u pilot-link-0.11.8/debian/rules pilot-link-0.11.8/debian/rules
--- pilot-link-0.11.8/debian/rules
+++ pilot-link-0.11.8/debian/rules
@@ -44,6 +44,7 @@
 build: build-stamp
 build-stamp: patch-stamp
 	dh_testdir
+	cp -f /usr/share/misc/config.{guess,sub} `pwd`
 	# Add here commands to compile the package.
 	CC="$(CC)" ./configure --prefix=/usr \
 		--mandir=$(pwd)/debian/tmp/usr/share/man \
@@ -65,6 +66,7 @@
 	dh_testdir
 	dh_testroot
 	rm -f build-stamp pre-binary-stamp
+	rm -f config.{guess,sub}
 	# Add here commands to clean up after the build process.
 	#-(cd tools/Perl5; make realclean)
 #	(cd tools/Python; make distclean)
diff -u pilot-link-0.11.8/debian/patches/00list pilot-link-0.11.8/debian/patches/00list
--- pilot-link-0.11.8/debian/patches/00list
+++ pilot-link-0.11.8/debian/patches/00list
@@ -17,0 +18 @@
+18_kfreebsd
--- pilot-link-0.11.8.orig/debian/patches/18_kfreebsd.dpatch
+++ pilot-link-0.11.8/debian/patches/18_kfreebsd.dpatch
@@ -0,0 +1,105 @@
+#!/bin/sh -e
+## patch by Aurelien Jarno <aurel32@debian.org>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: No description
+
+if [ $# -ne 1 ]; then
+    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
+    exit 1
+fi
+
+[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
+patch_opts="${patch_opts:--f --no-backup-if-mismatch}"
+
+case "$1" in
+       -patch) patch $patch_opts -p0 < $0;;
+       -unpatch) patch $patch_opts -p0 -R < $0;;
+        *)
+                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
+                exit 1;;
+esac
+
+exit 0
+
+@DPATCH@
+--- libpisock/freebsdusb.c	2005-09-02 22:27:03.000000000 +0200
++++ libpisock/freebsdusb.c	2005-09-02 22:27:23.000000000 +0200
+@@ -50,8 +50,9 @@
+ # define O_NONBLOCK 0
+ #endif
+ 
+-#if defined(__FreeBSD__)
++#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+ /* freebsd usb header */
++#include <sys/ioctl.h>
+ #include <dev/usb/usb.h>
+ #define MAX_BUF 256
+ #endif
+--- aclocal.m4	2005-09-02 22:32:13.000000000 +0200
++++ aclocal.m4	2005-09-02 22:34:16.000000000 +0200
+@@ -1727,7 +1727,7 @@
+     ;;
+ 
+   # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-  freebsd*)
++  freebsd* | kfreebsd*-gnu)
+     archive_cmds='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
+     hardcode_libdir_flag_spec='-R$libdir'
+     hardcode_direct=yes
+@@ -2182,7 +2182,7 @@
+   dynamic_linker=no
+   ;;
+ 
+-freebsd*-gnu*)
++kfreebsd*-gnu)
+   version_type=linux
+   need_lib_prefix=no
+   need_version=no
+@@ -2191,7 +2191,7 @@
+   shlibpath_var=LD_LIBRARY_PATH
+   shlibpath_overrides_runpath=no
+   hardcode_into_libs=yes
+-  dynamic_linker='GNU/FreeBSD ld.so'
++  dynamic_linker='ld.so'
+   ;;
+ 
+ freebsd*)
+@@ -3444,7 +3444,7 @@
+   esac
+   ;;
+ 
+-freebsd*)
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+--- configure	2005-09-02 22:36:38.000000000 +0200
++++ configure	2003-08-08 23:35:36.000000000 +0200
+@@ -3259,7 +3259,7 @@
+   esac
+   ;;
+ 
+-freebsd*)
++freebsd* | kfreebsd*-gnu)
+   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -5673,7 +5673,7 @@
+     ;;
+ 
+   # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-  freebsd*)
++  freebsd* | kfreebsd*-gnu)
+     archive_cmds='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
+     hardcode_libdir_flag_spec='-R$libdir'
+     hardcode_direct=yes
+@@ -6135,7 +6135,7 @@
+   dynamic_linker=no
+   ;;
+ 
+-freebsd*-gnu*)
++kfreebsd*-gnu*)
+   version_type=linux
+   need_lib_prefix=no
+   need_version=no
