diff -u pycurl-7.13.1/debian/rules pycurl-7.13.1/debian/rules
--- pycurl-7.13.1/debian/rules
+++ pycurl-7.13.1/debian/rules
@@ -35,9 +35,9 @@
 	dh_testroot
 	dh_clean -k
 	dh_installdirs
-	install -m 644 build/lib.linux-*-2.2/pycurl.so debian/python2.2-pycurl/usr/lib/python2.2/site-packages
-	install -m 644 build/lib.linux-*-2.3/pycurl.so debian/python2.3-pycurl/usr/lib/python2.3/site-packages
-	install -m 644 build/lib.linux-*-2.4/pycurl.so debian/python2.4-pycurl/usr/lib/python2.4/site-packages
+	install -m 644 build/lib.*-*-2.2/pycurl.so debian/python2.2-pycurl/usr/lib/python2.2/site-packages
+	install -m 644 build/lib.*-*-2.3/pycurl.so debian/python2.3-pycurl/usr/lib/python2.3/site-packages
+	install -m 644 build/lib.*-*-2.4/pycurl.so debian/python2.4-pycurl/usr/lib/python2.4/site-packages
 
 # Build architecture-independent files here.
 binary-indep: DH_OPTIONS=-i
