--- python2.3-2.3.4.old/debian/rules	2004-06-10 20:00:05.000000000 +0200
+++ python2.3-2.3.4/debian/rules	2004-06-10 18:59:06.000000000 +0200
@@ -109,6 +110,13 @@
 		--enable-unicode=ucs4 \
 		--without-cxx
 
+ifeq ($(DEB_BUILD_GNU_SYSTEM),kfreebsd-gnu)
+common_configure_args += --without-threads
+endif
+ifeq ($(DEB_BUILD_GNU_SYSTEM),knetbsd-gnu)
+common_configure_args += --without-threads
+endif
+
 stamp-configure-shared: patch-stamp
 	mkdir -p $(buildd_shared)
 	cd $(buildd_shared) && \
