Author: aurel32
Status: in BTS

diff -u qjackctl-0.2.18/debian/control qjackctl-0.2.18/debian/control
--- qjackctl-0.2.18/debian/control
+++ qjackctl-0.2.18/debian/control
@@ -2,7 +2,7 @@
 Section: sound
 Priority: optional
 Maintainer: Guenter Geiger (Debian/GNU) <geiger@debian.org>
-Build-Depends: debhelper (>> 4.0.0), cdbs, libqt3-mt-dev, libjack0.100.0-dev, libasound2-dev
+Build-Depends: debhelper (>> 4.0.0), cdbs, libqt3-mt-dev, libjack0.100.0-dev, libasound2-dev [!kfreebsd-i386]
 Standards-Version: 3.6.2
 
 Package: qjackctl
