diff -u roy-1.0.8/debian/control roy-1.0.8/debian/control
--- roy-1.0.8/debian/control
+++ roy-1.0.8/debian/control
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Brian Bassett <brianb@debian.org>
 Standards-Version: 3.6.1
-Build-Depends: debhelper (>= 4), zlib1g-dev, libc6-prof [!alpha !ia64 !hurd-i386], libc6.1-prof [alpha ia64], libc0.3-prof [hurd-i386]
+Build-Depends: debhelper (>= 4), zlib1g-dev, libc6-prof [!alpha !ia64 !hurd-i386 !kfreebsd-i386], libc6.1-prof [alpha ia64], libc0.3-prof [hurd-i386], libc0.1-prof [kfreebsd-i386]
 
 Package: libroy-dev
 Section: libdevel
