
Date: Mon, 20 Nov 2006 15:57:40 +0100 (CET)
From: Petr Salinger <Petr.Salinger@seznam.cz>
To: rpm-devel@lists.dulug.duke.edu

--- rpm-4.4.1/rpmio/rpmdav.c~	2006-11-20 12:16:47.000000000 +0100
+++ rpm-4.4.1/rpmio/rpmdav.c	2006-11-20 12:16:47.000000000 +0100
@@ -1406,7 +1406,7 @@
     dp->d_reclen = 0;		/* W2DO? */
 
 #if !defined(hpux) && !defined(sun)
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) && !defined(__FreeBSD_kernel__)
     dp->d_off = 0;		/* W2DO? */
 #endif
 /*@-boundsread@*/
@@ -1530,7 +1530,7 @@
     dp->d_reclen = 0;		/* W2DO? */
 
 #if !defined(hpux) && !defined(sun)
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) && !defined(__FreeBSD_kernel__)
     dp->d_off = 0;		/* W2DO? */
 #endif
 /*@-boundsread@*/
