Author: aurel32
Status: in BTS

diff -u sooperlooper-1.0.5/debian/control sooperlooper-1.0.5/debian/control
--- sooperlooper-1.0.5/debian/control
+++ sooperlooper-1.0.5/debian/control
@@ -2,7 +2,7 @@
 Section: sound
 Priority: optional
 Maintainer: Guenter Geiger (Debian/GNU) <geiger@debian.org>
-Build-Depends: debhelper (>= 4.0.0), cdbs, libasound2-dev, libjack0.100.0-dev, liblo-dev, libsigc++-1.2-dev, libwxgtk2.4-dev, libxml2-dev, libsndfile1-dev, libncurses5-dev, libsamplerate-dev
+Build-Depends: debhelper (>= 4.0.0), cdbs, libasound2-dev [!kfreebsd-i386], libjack0.100.0-dev, liblo-dev, libsigc++-1.2-dev, libwxgtk2.4-dev, libxml2-dev, libsndfile1-dev, libncurses5-dev, libsamplerate-dev
 Standards-Version: 3.6.1.1
 
 Package: sooperlooper
