Author: aurel32
Status: in BTS

--- stopmotion-0.3.2.orig/src/technical/audio/ossdriver.cpp
+++ stopmotion-0.3.2/src/technical/audio/ossdriver.cpp
@@ -22,7 +22,12 @@
 #include "src/foundation/logger.h"
 
 #include <sys/ioctl.h>
+#if defined(__linux__)
 #include <linux/soundcard.h>
+#endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/soundcard.h>
+#endif
 #include <fcntl.h>
 #include <errno.h>
 
