Status: in BTS
Author: aurel32

diff -u subversion-1.2.3dfsg1/debian/control subversion-1.2.3dfsg1/debian/control
--- subversion-1.2.3dfsg1/debian/control
+++ subversion-1.2.3dfsg1/debian/control
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Guilherme de S. Pastore <guilherme.pastore@terra.com.br>
 Uploaders: Al Stone <ahs3@debian.org>, David Kimdon <dwhedon@debian.org>, Troy Heber <troy.heber@hp.com>, Peter Samuelson <peter@p12n.org>
-Build-Depends: debhelper, libneon24-dev, apache2-threaded-dev, libapr0-dev (>= 2.0.55-3), libdb4.3-dev, libtool, bison, patch, python, time, python2.3-dev, autotools-dev, autoconf, swig, perl, cdbs, libperl-dev, kaffe-dev [!arm !armeb !mips !mipsel], kaffe-pthreads [!arm !armeb !mips !mipsel]
+Build-Depends: debhelper, libneon24-dev, apache2-threaded-dev, libapr0-dev (>= 2.0.55-3), libdb4.3-dev, libtool, bison, patch, python, time, python2.3-dev, autotools-dev, autoconf, swig, perl, cdbs, libperl-dev, kaffe-dev [!arm !armeb !mips !mipsel !kfreebsd-i386], kaffe-pthreads [!arm !armeb !mips !mipsel !kfreebsd-i386]
 Build-Conflicts: libsvn0 (<< 1.2)
 Standards-Version: 3.6.2.1
 
diff -u subversion-1.2.3dfsg1/debian/rules subversion-1.2.3dfsg1/debian/rules
--- subversion-1.2.3dfsg1/debian/rules
+++ subversion-1.2.3dfsg1/debian/rules
@@ -5,7 +5,7 @@
 # it must be manually kept in sync with the libsvn-javahl Architecture:
 # line in debian/control.
 
-DISABLE_JAVAHL_ARCHS := arm armeb mips mipsel
+DISABLE_JAVAHL_ARCHS := arm armeb mips mipsel kfreebsd-i386
 
 
 

