
Status: dirty

diff -ur swfdec0.3-0.3.2.old/debian/control swfdec0.3-0.3.2/debian/control
--- swfdec0.3-0.3.2.old/debian/control	2004-12-17 21:07:04.000000000 +0100
+++ swfdec0.3-0.3.2/debian/control	2004-12-17 21:05:19.000000000 +0100
@@ -1,7 +1,7 @@
 Source: swfdec0.3
 Priority: optional
 Maintainer: David Schleef <ds@schleef.org>
-Build-Depends: cdbs, debhelper (>> 3.0.0), libgtk2.0-dev (>> 2.1.2), mozilla-dev, libart-2.0-dev, zlib1g-dev, libmad0-dev, libsdl1.2-dev, libxt-dev, liboil0.2-dev
+Build-Depends: cdbs, debhelper (>> 3.0.0), libgtk2.0-dev (>> 2.1.2), libart-2.0-dev, zlib1g-dev, libmad0-dev, libsdl1.2-dev, libxt-dev, liboil0.2-dev
 Standards-Version: 3.5.6
 
 Package: libswfdec0.3-dev
diff -ur swfdec0.3-0.3.2.old/debian/rules swfdec0.3-0.3.2/debian/rules
--- swfdec0.3-0.3.2.old/debian/rules	2004-12-17 21:07:04.000000000 +0100
+++ swfdec0.3-0.3.2/debian/rules	2004-12-17 21:05:45.000000000 +0100
@@ -4,9 +4,3 @@
 include /usr/share/cdbs/1/class/autotools.mk
 
 DEB_CONFIGURE_EXTRA_FLAGS := --enable-static
-
-common-install-arch::
-	mkdir -p debian/tmp/usr/lib/mozilla-snapshot/plugins
-	cd debian/tmp/usr/lib/mozilla-snapshot/plugins && ln -s ../../mozilla/plugins/libmozswfdec.so .
-	mkdir -p debian/tmp/usr/lib/mozilla-firefox/plugins
-	cd debian/tmp/usr/lib/mozilla-firefox/plugins && ln -s ../../mozilla/plugins/libmozswfdec.so .
diff -ur swfdec0.3-0.3.2.old/debian/swf-player.install swfdec0.3-0.3.2/debian/swf-player.install
--- swfdec0.3-0.3.2.old/debian/swf-player.install	2004-12-17 21:07:04.000000000 +0100
+++ swfdec0.3-0.3.2/debian/swf-player.install	2004-12-17 21:05:01.000000000 +0100
@@ -1,7 +1,4 @@
 debian/tmp/usr/bin/*
 debian/tmp/usr/share/man/*
-debian/tmp/usr/lib/mozilla/plugins/*.so
 debian/tmp/usr/lib/gtk-2.0/*/loaders/*.so
 debian/tmp/usr/lib/gtk-2.0/*/loaders/*.la
-debian/tmp/usr/lib/mozilla-snapshot/plugins/*.so
-debian/tmp/usr/lib/mozilla-firefox/plugins/*.so
