
Author: rmh
Status: in BTS

diff -ur tofrodos-1.7.6.old/src/config.h tofrodos-1.7.6/src/config.h
--- tofrodos-1.7.6.old/src/config.h     2005-03-15 16:01:02.000000000 +0100
+++ tofrodos-1.7.6/src/config.h 2006-02-04 20:19:32.000000000 +0100
@@ -95,7 +95,7 @@
 #endif

 /* define the systems */
-#if defined(__linux__) /* (predefined) */
+#if defined(__linux__) || defined(__GLIBC__)   /* (predefined) */
 #if !defined(LINUX)
 #define        LINUX
 #endif

