Author: aurel32
Status: in BTS

diff -u twinkle-0.3.2/debian/control twinkle-0.3.2/debian/control
--- twinkle-0.3.2/debian/control
+++ twinkle-0.3.2/debian/control
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Debian VoIP Team <pkg-voip-maintainers@lists.alioth.debian.org>
 Uploaders: Kilian Krause <kilian@debian.org>, Jose Carlos Garcia Sogo <jsogo@debian.org>, Mark Purcell <msp@debian.org>, Santiago Garcia Mantinan <manty@debian.org>
-Build-Depends: debhelper (>= 4.0.0), autotools-dev, libcommoncpp2-dev (>=1.3.10-4), libccrtp-dev (>=1.3.4), help2man, libqt3-mt-dev (>= 3:3.3.4-4), qt3-apps-dev, automake1.9, libxml2-dev, cdbs, kdelibs-bin, kdelibs4-dev, libasound2-dev, libxml2-dev
+Build-Depends: debhelper (>= 4.0.0), autotools-dev, libcommoncpp2-dev (>=1.3.10-4), libccrtp-dev (>=1.3.4), help2man, libqt3-mt-dev (>= 3:3.3.4-4), qt3-apps-dev, automake1.9, libxml2-dev, cdbs, kdelibs-bin, kdelibs4-dev, libasound2-dev [!kfreebsd-i386], libxml2-dev
 Standards-Version: 3.6.1
 
 Package: twinkle

