diff -ur twisted-1.3.0.old/debian/control twisted-1.3.0/debian/control
--- twisted-1.3.0.old/debian/control	2004-08-17 22:59:14.000000000 +0200
+++ twisted-1.3.0/debian/control	2004-08-17 22:54:53.000000000 +0200
@@ -2,7 +2,7 @@
 Section: python
 Priority: extra
 Maintainer: Matthias Klose <doko@debian.org>
-Build-Depends: python2.2-dev, python2.3-dev, fortune-mod, python
+Build-Depends: python2.3-dev, fortune-mod, python
 Standards-Version: 3.6.1
 
 Package: python-twisted
@@ -16,31 +16,6 @@
  .
  It replaces the "twisted" package, which was non-policy compliant.
 
-Package: python2.2-twisted
-Architecture: all
-Depends: python2.2, python2.2-twisted-bin
-Suggests: python2.2-tk, python2.2-gtk2, python2.2-glade2, python2.3-qt3c102, twisted-doc, python2.2-twisted-conch
-Replaces: python2.1-twisted
-Recommends: python2.2-pyopenssl, python2.2-pam
-Conflicts: twisted (<< 0.99.0)
-Description: Event-based framework for internet applications
- It includes a web server, a telnet server, a multiplayer RPG engine, a 
- generic client and server for remote object access, and APIs for creating 
- new protocols.
- . 
- This version is usable with python2.2
-
-Package: python2.2-twisted-bin
-Architecture: any
-Depends: ${shlibs:Depends}, python2.2
-Conflicts: python2.2-twisted (<<0.18.1)
-Description: Event-based framework for internet applications
- It includes a web server, a telnet server, a multiplayer RPG engine, a 
- generic client and server for remote object access, and APIs for creating 
- new protocols.
- . 
- This package contains the architectures specific files for python2.2-twisted
-
 Package: python2.3-twisted
 Architecture: all
 Depends: python2.3, python2.3-twisted-bin
@@ -99,16 +74,6 @@
  Quotes about programming, politics and anything else. Gathered by the
  Twisted development team from IRC and e-mail.
 
-Package: python2.2-twisted-conch
-Architecture: all
-Depends: python2.2, python2.2-twisted, python2.2-crypto
-Conflicts: python2.2-twisted (<=1.0.0-2)
-Description: The Twisted SSH Implementation (for Python 2.2)
- A client/server implementation of the SSH protocol, using the
- twisted framework
- . 
- This version is usable with python2.2
-
 Package: python2.3-twisted-conch
 Architecture: all
 Depends: python2.3, python2.3-twisted, python2.3-crypto
