Author: aurel32
Status: workaround until libtool.m4 get fixed

diff -u unixodbc-2.2.11/aclocal.m4 unixodbc-2.2.11/aclocal.m4
--- unixodbc-2.2.11/aclocal.m4
+++ unixodbc-2.2.11/aclocal.m4
@@ -9003,7 +9003,7 @@
       _LT_AC_TAGVAR(hardcode_shlibpath_var, $1)=no
       ;;
 
-  linux*)
+  linux* | kfreebsd*-gnu)
     if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
         tmp_archive_cmds='$CC -shared $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
 	_LT_AC_TAGVAR(archive_cmds, $1)="$tmp_archive_cmds"
diff -u unixodbc-2.2.11/configure unixodbc-2.2.11/configure
--- unixodbc-2.2.11/configure
+++ unixodbc-2.2.11/configure
@@ -7499,7 +7499,7 @@
       hardcode_shlibpath_var=no
       ;;
 
-  linux*)
+  linux* | kfreebsd*-gnu)
     if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
         tmp_archive_cmds='$CC -shared $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
 	archive_cmds="$tmp_archive_cmds"
diff -u unixodbc-2.2.11/libltdl/aclocal.m4 unixodbc-2.2.11/libltdl/aclocal.m4
--- unixodbc-2.2.11/libltdl/aclocal.m4
+++ unixodbc-2.2.11/libltdl/aclocal.m4
@@ -6042,7 +6042,7 @@
       _LT_AC_TAGVAR(hardcode_shlibpath_var, $1)=no
       ;;
 
-  linux*)
+  linux* | kfreebsd*-gnu)
     if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
         tmp_archive_cmds='$CC -shared $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
 	_LT_AC_TAGVAR(archive_cmds, $1)="$tmp_archive_cmds"
diff -u unixodbc-2.2.11/libltdl/configure unixodbc-2.2.11/libltdl/configure
--- unixodbc-2.2.11/libltdl/configure
+++ unixodbc-2.2.11/libltdl/configure
@@ -7189,7 +7189,7 @@
       hardcode_shlibpath_var=no
       ;;
 
-  linux*)
+  linux* | kfreebsd-*gnu)
     if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
         tmp_archive_cmds='$CC -shared $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
 	archive_cmds="$tmp_archive_cmds"
