
Author: rmh
Status: don't send untill wildcards are allowed

diff -ur wine-0.9.6.old/debian/control wine-0.9.6/debian/control
--- wine-0.9.6.old/debian/control	2006-02-02 09:33:15.000000000 +0100
+++ wine-0.9.6/debian/control	2006-02-02 12:03:32.000000000 +0100
@@ -13,15 +13,15 @@
  libglu1-xorg | xlibmesa-glu-dev | xlibmesa-dev (<= 4.2.1-4) | libglu-dev,
  freeglut3-dev | libglut-dev | glutg3-dev, libxmu-dev | xlibs-dev (<< 4.3),
  libncurses5-dev, libcupsys2-dev, libjpeg62-dev | libjpeg-dev, libungif4-dev,
- libfreetype6-dev, libasound2-dev, libjack0.100.0-dev | libjack0.80.0-dev | libjack-dev,
+ libfreetype6-dev, libasound2-dev [linux-any], libjack0.100.0-dev | libjack0.80.0-dev | libjack-dev,
  libartsc0-dev | libarts-dev, libaudio-dev | nas-dev, libesd0-dev,
  libsane-dev, libusb-dev, libicu34-dev | libicu28-dev | libicu21-dev | libicu-dev,
- libfontconfig1-dev, libssl-dev, libcapi20-dev (>= 1:3.3.0.20041024-2),
+ libfontconfig1-dev, libssl-dev, libcapi20-dev (>= 1:3.3.0.20041024-2) [linux-any],
  liblcms1-dev, libldap2-dev, libxml2-dev, libxslt1-dev, fontforge
 Standards-Version: 3.6.0
 
 Package: wine
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: ${debconf-depends}, libwine (= ${Source-Version}), xbase-clients (>= 4.0) | xcontrib
 Suggests: wine-doc, wine-utils, msttcorefonts, binfmt-support
 Conflicts: binfmt-support (<< 1.1.2)
@@ -36,7 +36,7 @@
  Wine is often updated.
 
 Package: libwine-dev
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Section: libdevel
 Depends: libwine (= ${Source-Version}), libc6-dev
 Replaces: libwine (<< 0.0.20010216)
@@ -52,7 +52,7 @@
 
 Package: libwine
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: ${debconf-depends}, ${shlibs:Depends}${freetype}
 Replaces: libwine0.0.971116, wine (<< 0.0.20040213)
 Conflicts: libwine0.0.971116
@@ -67,7 +67,7 @@
 
 Package: libwine-alsa
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: linux-i386 linux-powerpc linux-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Description: Windows API Implementation (ALSA Sound Module)
  This is a BETA release of Wine, the free MS-Windows API implementation.
@@ -79,7 +79,7 @@
 
 Package: libwine-arts
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Replaces: libwine (<< 0.0.20020710)
 Description: Windows API Implementation (aRts Sound Module)
@@ -92,7 +92,7 @@
 
 Package: libwine-capi
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: linux-i386 linux-powerpc linux-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Description: Windows API Implementation (ISDN Module)
  This is a BETA release of Wine, the free MS-Windows API implementation.
@@ -105,7 +105,7 @@
 
 Package: libwine-cms
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Replaces: libwine (<< 0.0.20050830)
 Description: Windows API Implementation (Color Management Module)
@@ -119,7 +119,7 @@
 
 Package: libwine-esd
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Description: Windows API Implementation (EsounD Sound Module)
  This is a BETA release of Wine, the free MS-Windows API implementation.
@@ -131,7 +131,7 @@
 
 Package: libwine-gl
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Description: Windows API Implementation (OpenGL Module)
  This is a BETA release of Wine, the free MS-Windows API implementation.
@@ -144,7 +144,7 @@
 
 Package: libwine-jack
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}${jack}
 Description: Windows API Implementation (JACK Sound Module)
  This is a BETA release of Wine, the free MS-Windows API implementation.
@@ -156,7 +156,7 @@
 
 Package: libwine-ldap
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Description: Windows API Implementation (LDAP Module)
  This is a BETA release of Wine, the free MS-Windows API implementation.
@@ -169,7 +169,7 @@
 
 Package: libwine-nas
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Replaces: libwine (<< 0.0.20020710)
 Description: Windows API Implementation (NAS Sound Module)
@@ -182,7 +182,7 @@
 
 Package: libwine-print
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}${freetype}${cupsys}
 Replaces: libwine (<< 0.0.20020710)
 Description: Windows API Implementation (Printing Module)
@@ -196,7 +196,7 @@
 
 Package: libwine-twain
 Section: libs
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), ${shlibs:Depends}
 Replaces: libwine (<< 0.0.20020710)
 Description: Windows API Implementation (Scanner Module)
@@ -209,7 +209,7 @@
  Wine is often updated.
 
 Package: wine-utils
-Architecture: i386 hurd-i386 freebsd-i386 netbsd-i386 powerpc hurd-powerpc freebsd-powerpc netbsd-powerpc sparc hurd-sparc freebsd-sparc netbsd-sparc
+Architecture: any-i386 any-powerpc any-sparc
 Depends: libwine (= ${Source-Version}), wine
 Replaces: libwine-dev (<< 0.0.20010216)
 Description: Windows API Implementation (Utilities)
diff -ur wine-0.9.6.old/debian/rules wine-0.9.6/debian/rules
--- wine-0.9.6.old/debian/rules	2006-02-02 09:33:15.000000000 +0100
+++ wine-0.9.6/debian/rules	2006-02-02 13:53:29.000000000 +0100
@@ -22,6 +22,8 @@
 MINDEBCONFVER := 1.2.0
 endif
 
+DEB_HOST_GNU_SYSTEM ?= $(shell dpkg-architecture -qDEB_HOST_GNU_SYSTEM)
+
 configure: configure-stamp
 configure-stamp:
 	dh_testdir
@@ -101,9 +103,9 @@
 	dh_testdir
 	dh_testroot
 	# clean up first in case the package maintainer is experimenting again
-	dh_clean -a -k
+	dh_clean -s -k
 	# create installation directories
-	dh_installdirs -a
+	dh_installdirs -s
 
 	# install wine, libwine, and tools
 	$(MAKE) install prefix=`pwd`/debian/tmp/usr libdir=`pwd`/debian/tmp/usr/lib
@@ -147,7 +149,7 @@
 
 	# distribute the files in debian/tmp into debian/<packagename>
 	# according to the <packagename>.files files
-	dh_movefiles -a
+	dh_movefiles -s
 
 	# distribute files we want that weren't in debian/tmp
 	cp dlls/twain/README debian/libwine-twain/usr/share/doc/libwine-twain
@@ -165,9 +167,11 @@
 	cp tools/winedump/README debian/wine-utils/usr/share/doc/wine-utils/README.winedump
 
 	# split up libwine
+ifeq ($(DEB_HOST_GNU_SYSTEM), linux-gnu)
 	mv debian/libwine/usr/lib/wine/winealsa* debian/libwine-alsa/usr/lib/wine
-	mv debian/libwine/usr/lib/wine/winearts* debian/libwine-arts/usr/lib/wine
 	mv debian/libwine/usr/lib/wine/capi*     debian/libwine-capi/usr/lib/wine
+endif
+	mv debian/libwine/usr/lib/wine/winearts* debian/libwine-arts/usr/lib/wine
 	mv debian/libwine/usr/lib/wine/mscms*    debian/libwine-cms/usr/lib/wine
 	mv debian/libwine/usr/lib/wine/wineesd*  debian/libwine-esd/usr/lib/wine
 	mv debian/libwine/usr/lib/wine/d3d8*     debian/libwine-gl/usr/lib/wine
@@ -235,22 +239,22 @@
 	po2debconf -e utf8 debian/libwine.templates.master > debian/libwine.templates
 endif
 
-	dh_installdebconf -a
-	dh_installdocs -a
-#	dh_installmenu -a
-	dh_installmime -a
+	dh_installdebconf -s
+	dh_installdocs -s
+#	dh_installmenu -s
+	dh_installmime -s
 
 	# FIXME: concatenate wineinstall script
 
-	dh_undocumented -a
-	dh_installchangelogs -a ChangeLog
-	dh_link -a
-	dh_strip -a
-	dh_compress -a
-	dh_fixperms -a
-	dh_makeshlibs -a
-	dh_installdeb -a
-	dh_shlibdeps -a -ldlls:libs:
+	dh_undocumented -s
+	dh_installchangelogs -s ChangeLog
+	dh_link -s
+	dh_strip -s
+	dh_compress -s
+	dh_fixperms -s
+	dh_makeshlibs -s
+	dh_installdeb -s
+	dh_shlibdeps -s -ldlls:libs:
 
 	# if the distro we're compiling for has freetype, depend on it
 	(dpkg -s libfreetype6-dev >/dev/null && \
@@ -283,9 +287,9 @@
 	 echo "icu=, icu" >> debian/libwine.substvars) || \
 	true
 
-	dh_gencontrol -a -- -V'debconf-depends=debconf (>= $(MINDEBCONFVER)) | debconf-2.0'
-	dh_md5sums -a
-	dh_builddeb -a
+	dh_gencontrol -s -- -V'debconf-depends=debconf (>= $(MINDEBCONFVER)) | debconf-2.0'
+	dh_md5sums -s
+	dh_builddeb -s
 
 binary: binary-arch
 .PHONY: build-indep build-arch build clean binary-indep binary-arch binary \
