Author: aurel32
Status: in BTS

--- wxwidgets2.6-2.6.1.2/aclocal.m4	2005-08-25 14:52:03.000000000 +0200
+++ wxwidgets2.6-2.6.1.2+kbsd/aclocal.m4	2005-09-08 12:00:00.000000000 +0200
@@ -1935,7 +1935,7 @@
       ;;
       
       powerpc-apple-macos* | \
-      *-*-freebsd* | *-*-openbsd* | *-*-netbsd* | \
+      *-*-freebsd* | *-*-openbsd* | *-*-netbsd* | *-*-k*bsd*-gnu | \
       *-*-sunos4* | \
       *-*-osf* | \
       *-*-dgux5* | \
@@ -1983,7 +1983,7 @@
     SONAME_FLAG=
 
     case "${BAKEFILE_HOST}" in
-      *-*-linux* | *-*-freebsd* )
+      *-*-linux* | *-*-freebsd* | *-*-k*bsd*-gnu)
         SONAME_FLAG="-Wl,-soname,"
         USE_SOVERSION=1
         USE_SOVERLINUX=1
--- wxwidgets2.6-2.6.1.2/configure	2005-08-25 14:52:03.000000000 +0200
+++ wxwidgets2.6-2.6.1.2+kbsd/configure	2005-09-08 12:00:02.000000000 +0200
@@ -42761,7 +42761,7 @@
       ;;
 
       powerpc-apple-macos* | \
-      *-*-freebsd* | *-*-openbsd* | *-*-netbsd* | \
+      *-*-freebsd* | *-*-openbsd* | *-*-netbsd* | *-*-k*bsd*-gnu | \
       *-*-sunos4* | \
       *-*-osf* | \
       *-*-dgux5* | \
@@ -42801,7 +42801,7 @@
     SONAME_FLAG=
 
     case "${BAKEFILE_HOST}" in
-      *-*-linux* | *-*-freebsd* )
+      *-*-linux* | *-*-freebsd* | *-*-k*bsd*-gnu)
         SONAME_FLAG="-Wl,-soname,"
         USE_SOVERSION=1
         USE_SOVERLINUX=1
