Author: aurel32
Status: in BTS

diff -u x86info-1.15/debian/control x86info-1.15/debian/control
--- x86info-1.15/debian/control
+++ x86info-1.15/debian/control
@@ -6,7 +6,7 @@
 Standards-Version: 3.6.2
 
 Package: x86info
-Architecture: i386 amd64
+Architecture: i386 kfreebsd-i386 amd64
 Depends: ${shlibs:Depends}, ${misc:Depends}, makedev (>= 2.3.1-53)
 Description: Display diagnostic information about i386 compatible CPUs
  x86info displays diagnostic information about the CPUs fitted to i386
--- x86info-1.15.orig/bench/benchmarks.c
+++ x86info-1.15/bench/benchmarks.c
@@ -1,7 +1,8 @@
 #include <time.h>
 #include <sys/time.h>
-#include <asm/unistd.h>
+#include <sys/unistd.h>
 #include <sys/stat.h>
+#include <sys/syscall.h>
 #include <stdio.h>
 #include "../x86info.h"
 #include "bench.h"
@@ -13,7 +14,7 @@
 	if (show_bench != 1)
 		return;
 
-	TIME(asm volatile("int $0x80" :"=a" (ret) :"0" (__NR_getppid)), "int 0x80");
+	TIME(asm volatile("int $0x80" :"=a" (ret) :"0" (SYS_getppid)), "int 0x80");
 	TIME(asm volatile("cpuid": : :"ax", "dx", "cx", "bx"), "cpuid");
 
 	TIME(asm volatile("addl $1,0(%esp)"), "addl");
--- x86info-1.15.orig/x86info.h
+++ x86info-1.15/x86info.h
@@ -1,10 +1,10 @@
 #ifndef _X86INFO_H
 #define _X86INFO_H
 
-#include <linux/types.h>
-typedef unsigned char u8;
-typedef unsigned short u16;
-typedef unsigned int u32;
+#include <sys/types.h>
+typedef u_int8_t u8;
+typedef u_int16_t u16;
+typedef u_int32_t u32;
 
 #define VENDOR_AMD	1
 #define VENDOR_CENTAUR 2
