Author: aurel32
Status: in BTS

diff -u xconq-7.4.1/debian/control xconq-7.4.1/debian/control
--- xconq-7.4.1/debian/control
+++ xconq-7.4.1/debian/control
@@ -2,7 +2,7 @@
 Section: games
 Priority: optional
 Maintainer: Timo Benk <t_benk@web.de>
-Build-Depends: debhelper (>> 3.0.0), tk8.3-dev, libxaw7-dev (>= 4.1.0), texinfo (>= 4.0), tetex-bin (>= 1.0.7)
+Build-Depends: debhelper (>> 3.0.0), tk8.3-dev, libxaw7-dev (>= 4.1.0), texinfo (>= 4.0), tetex-bin (>= 1.0.7), autotools-dev
 Standards-Version: 3.5.2
 
 Package: xconq
diff -u xconq-7.4.1/debian/rules xconq-7.4.1/debian/rules
--- xconq-7.4.1/debian/rules
+++ xconq-7.4.1/debian/rules
@@ -6,6 +6,7 @@
 
 $(STAMP_DIR)/source.make:
 	$(MAKE) -f debian/sys-build.mk source.make
+	cp -f /usr/share/misc/config.{guess,sub} $(BUILD_DIR)
 
 configure: $(STAMP_DIR)/configure
 $(STAMP_DIR)/configure: $(STAMP_DIR)/source.make
