
Status: in BTS #366133

only in patch2:
unchanged:
--- xemacs21-21.4.19.orig/configure.in
+++ xemacs21-21.4.19/configure.in
@@ -1105,6 +1105,7 @@
 dnl Straightforward OS determination
 case "$canonical" in
   *-*-linux*    ) opsys=linux ;;
+  *-*-kfreebsd* ) opsys=linux ;;
   *-*-netbsd*   ) opsys=netbsd ;;
   *-*-openbsd*	) opsys=openbsd ;;
   *-*-nextstep* ) opsys=nextstep ;;
diff -u xemacs21-21.4.19/debian/control.in xemacs21-21.4.19/debian/control.in
--- xemacs21-21.4.19/debian/control.in
+++ xemacs21-21.4.19/debian/control.in
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: OHURA Makoto <ohura@debian.org>
 Standards-Version: 3.6.2
-Build-Depends: texinfo, libcanna1g-dev, libwnn6-dev, libjpeg62-dev, libncurses5-dev, libpng3-dev, libtiff4-dev, zlib1g-dev, texi2html, debhelper (>> 4), libldap2-dev, libdb3-dev, libpam0g-dev, libcompfaceg1-dev, libx11-dev, libxau-dev, libxext-dev, libxmu-dev, libxpm-dev, libxt-dev, x-dev, xlibs-data, libxaw7-dev, autoconf2.13, libgtk1.2-dev, libgnome-dev, libart-dev, libesd0-dev, dpatch (>= 1.11), libgpmg1-dev
+Build-Depends: texinfo, libcanna1g-dev, libwnn6-dev, libjpeg62-dev, libncurses5-dev, libpng3-dev, libtiff4-dev, zlib1g-dev, texi2html, debhelper (>> 4), libldap2-dev, libdb3-dev, libpam0g-dev, libcompfaceg1-dev, libx11-dev, libxau-dev, libxext-dev, libxmu-dev, libxpm-dev, libxt-dev, x-dev, xlibs-data, libxaw7-dev, autoconf2.13, libgtk1.2-dev, libgnome-dev, libart-dev, libesd0-dev, dpatch (>= 1.11), libgpmg1-dev [!hurd-i386 !kfreebsd-i386 !kfreebsd-amd64]
 
 Package: xemacs@MAJVERSION@@SUBNAME@
 Architecture: all
diff -u xemacs21-21.4.19/debian/rules xemacs21-21.4.19/debian/rules
--- xemacs21-21.4.19/debian/rules
+++ xemacs21-21.4.19/debian/rules
@@ -22,10 +22,10 @@
 # any way to do this automatically since we'd like this to be
 # easily used with no changes on other architectures, but still
 # want to be able to specify.  Does this work OK?
-machine_type = $(shell dpkg --print-architecture)
+machine_type = $(shell dpkg-architecture -qDEB_HOST_GNU_CPU)
 prefix=usr
 
-debianconfig = ${machine_type}-debian-linux
+debianconfig = $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
 tmp_file_loc = debian/tmp
 
 var_file = var_file.pl
