Author: aurel32
Status: in BTS

--- xpaint-2.7.6.orig/misc.h
+++ xpaint-2.7.6/misc.h
@@ -24,7 +24,7 @@
 long random(void);
 #endif
 
-#if !defined(__VMS) & !defined(linux) & !defined(__EMX__) & !defined(__FreeBSD__) & !defined(__CYGWIN__)
+#if !defined(__VMS) & !defined(linux) & !defined(__EMX__) & !defined(__FreeBSD__) & !defined(__CYGWIN__) & !defined(__GLIBC__)
 #if defined(BSD4_4) || defined(HPArchitecture) || defined(SGIArchitecture) || defined(_AIX) || defined(_SCO_DS)
 void srandom(unsigned int);
 #else
