--- xsim-0.3.9.4.orig/configure
+++ xsim-0.3.9.4/configure
@@ -1400,6 +1400,11 @@
     dllib="-ldl"
     pthreadlib="-lpthread"
     ;;
+  XGNU/kFreeBSD)
+    os_type="LINUX"
+    dllib="-ldl"
+    pthreadlib="-lpthread"
+    ;;
   XFreeBSD)
     os_type="FREEBSD"
     dllib="-lc"
only in patch2:
unchanged:
--- xsim-0.3.9.4.orig/configure.in
+++ xsim-0.3.9.4/configure.in
@@ -36,6 +36,11 @@
     dllib="-ldl"
     pthreadlib="-lpthread"
     ;;
+  XGNU/kFreeBSD)
+    os_type="LINUX"
+    dllib="-ldl"
+    pthreadlib="-lpthread"
+    ;;
   XFreeBSD)
     os_type="FREEBSD"
     dllib="-lc"
