diff -uwrN /tmp/zsh-4.2.4/debian/control zsh-4.2.4/debian/control
--- /tmp/zsh-4.2.4/debian/control	2005-03-18 16:31:51.000000000 +0100
+++ zsh-4.2.4/debian/control	2005-03-18 15:13:35.000000000 +0100
@@ -1,7 +1,7 @@
 Source: zsh
 Section: shells
 Priority: optional
-Build-Depends: texinfo, groff-base, libncurses5-dev, texi2html, libcap-dev [!hurd-i386 !freebsd-i386], bsdmainutils, po-debconf, libpcre3-dev
+Build-Depends: texinfo, groff-base, libncurses5-dev, texi2html, libcap-dev [!hurd-i386 !kfreebsd-i386], bsdmainutils, po-debconf, libpcre3-dev
 Maintainer: Clint Adams <schizo@debian.org>
 Standards-Version: 3.6.1
 
diff -uwrN /tmp/zsh-4.2.4/debian/rules zsh-4.2.4/debian/rules
--- /tmp/zsh-4.2.4/debian/rules	2005-03-18 16:31:51.000000000 +0100
+++ zsh-4.2.4/debian/rules	2005-03-18 19:37:14.000000000 +0100
@@ -212,7 +212,7 @@
 	gzip -9f debian/tmp-static/usr/share/doc/${package}-static/changelog.Debian
 	cd debian/tmp-static && find * -type f  ! -regex '^DEBIAN/.*' -print0 | xargs -r0 md5sum > DEBIAN/md5sums
 	cp /lib/libnss_files.so.* debian/tmp-static
-	chmod 755 debian/tmp-static/libnss_files.so.*
+	#chmod 755 debian/tmp-static/libnss_files.so.*
 	dpkg-shlibdeps -Tdebian/${package}-static.substvars -dDepends debian/tmp-static/libnss_files.so.*
 	rm debian/tmp-static/libnss_files.so.*
 	dpkg-gencontrol -ldebian/changelog -isp -p${package}-static -Tdebian/${package}-static.substvars -Pdebian/tmp-static
--- /tmp/zsh-4.2.4/configure	2005-03-18 16:31:48.000000000 +0100
+++ zsh-4.2.4/configure	2005-03-18 19:36:35.000000000 +0100
@@ -15039,7 +15039,7 @@
     esac
   fi
   case "$host_os" in
-    freebsd*|linux*|irix*|osf*|gnu*) DLLDFLAGS="${DLLDFLAGS=-shared}" ;;
+    freebsd*|linux*|irix*|osf*|gnu*|kfreebsd*-gnu) DLLDFLAGS="${DLLDFLAGS=-shared}" ;;
     sunos*)       DLLDFLAGS="${DLLDFLAGS=-assert nodefinitions}" ;;
     sysv4*|esix*) DLLDFLAGS="${DLLDFLAGS=-G $ldflags}" ;;
     netbsd*)      DLLDFLAGS="${DLLDFLAGS=${DLLDARG}-x -shared --whole-archive}" ;;
@@ -15063,7 +15063,7 @@
   esac
   case "$host" in
     *-hpux*)  EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-Wl,-E}" ;;
-    *-freebsd3-9*|*-linux*|gnu*) EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-rdynamic}" ;;
+    *-freebsd3-9*|*-linux*|gnu*|k*bsd*gnu) EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-rdynamic}" ;;
     *openbsd*)
        if test $zsh_cv_sys_elf = yes; then
 	 EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-Wl,-E}"
--- /tmp/zsh-4.2.4/configure.ac	2005-03-18 16:31:47.000000000 +0100
+++ zsh-4.2.4/configure.ac	2005-03-18 18:15:19.000000000 +0100
@@ -2061,7 +2061,7 @@
     esac
   fi
   case "$host_os" in
-    freebsd*|linux*|irix*|osf*|gnu*) DLLDFLAGS="${DLLDFLAGS=-shared}" ;;
+    freebsd*|linux*|irix*|osf*|gnu*|k*bsd*-gnu) DLLDFLAGS="${DLLDFLAGS=-shared}" ;;
     sunos*)       DLLDFLAGS="${DLLDFLAGS=-assert nodefinitions}" ;;
     sysv4*|esix*) DLLDFLAGS="${DLLDFLAGS=-G $ldflags}" ;;
     netbsd*)      DLLDFLAGS="${DLLDFLAGS=${DLLDARG}-x -shared --whole-archive}" ;;
@@ -2085,7 +2085,7 @@
   esac
   case "$host" in
     *-hpux*)  EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-Wl,-E}" ;;
-    *-freebsd[3-9]*|*-linux*|gnu*) EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-rdynamic}" ;;
+    *-freebsd[3-9]*|*-linux*|gnu*|k*bsd*gnu) EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-rdynamic}" ;;
     *openbsd*)
        if test $zsh_cv_sys_elf = yes; then
 	 EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-Wl,-E}"
