Index: extras/drm/shared/drm.h
===================================================================
RCS file: /cvs/xorg/xc/extras/drm/shared/drm.h,v
retrieving revision 1.3
diff -u -r1.3 drm.h
--- extras/drm/shared/drm.h	15 Jun 2005 19:05:06 -0000	1.3
+++ extras/drm/shared/drm.h	18 Jul 2005 19:17:24 -0000
@@ -71,15 +71,15 @@
 #define DRM_IOC_WRITE		_IOC_WRITE
 #define DRM_IOC_READWRITE	_IOC_READ|_IOC_WRITE
 #define DRM_IOC(dir, group, nr, size) _IOC(dir, group, nr, size)
-#elif defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
-#if defined(__FreeBSD__) && defined(IN_MODULE)
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
+#if (defined(__FreeBSD__) || defined(__FreeBSD_kernel__)) && defined(IN_MODULE)
 /* Prevent name collision when including sys/ioccom.h */
 #undef ioctl
 #include <sys/ioccom.h>
 #define ioctl(a,b,c)		xf86ioctl(a,b,c)
 #else
 #include <sys/ioccom.h>
-#endif /* __FreeBSD__ && xf86ioctl */
+#endif /* __FreeBSD_kernel__ && xf86ioctl */
 #define DRM_IOCTL_NR(n)		((n) & 0xff)
 #define DRM_IOC_VOID		IOC_VOID
 #define DRM_IOC_READ		IOC_OUT
Index: lib/GL/dri/drm/Imakefile
===================================================================
RCS file: /cvs/xorg/xc/lib/GL/dri/drm/Imakefile,v
retrieving revision 1.4
diff -u -r1.4 Imakefile
--- lib/GL/dri/drm/Imakefile	27 Aug 2004 03:22:12 -0000	1.4
+++ lib/GL/dri/drm/Imakefile	18 Jul 2005 19:17:33 -0000
@@ -25,7 +25,7 @@
 OS_SUBDIR2 = linux
 OS_SUBDIR3 = shared
 #endif
-#if defined(FreeBSDArchitecture) || defined(BSDOSArchitecture) || \
+#if defined(KFreeBSDArchitecture) || defined(BSDOSArchitecture) || \
     defined(NetBSDArchitecture) || defined(OpenBSDArchitecture)
 OS_SUBDIR = bsd
 OS_SUBDIR2 = linux
