Index: lib/xtrans/Xtranssock.c
===================================================================
RCS file: /cvs/xorg/xc/lib/xtrans/Xtranssock.c,v
retrieving revision 1.7
diff -u -r1.7 Xtranssock.c
--- lib/xtrans/Xtranssock.c	3 Jul 2005 07:01:01 -0000	1.7
+++ lib/xtrans/Xtranssock.c	18 Jul 2005 19:17:35 -0000
@@ -88,7 +88,7 @@
 #endif 
 
 #ifndef NO_TCP_H
-#if defined(__osf__) || defined(linux) || defined(AIXV5)
+#if defined(__osf__) || defined(linux) || defined(__GLIBC__) || defined(AIXV5)
 #include <sys/param.h>
 #endif /* osf */
 #if defined(__NetBSD__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__DragonFly__)
