Index: xc/config/cf/linux.cf
===================================================================
--- xc/config/cf/linux.cf.orig	2006-01-10 09:26:06.000000000 +0100
+++ xc/config/cf/linux.cf	2006-01-10 09:26:07.000000000 +0100
@@ -259,6 +259,7 @@
 /* Wraphelp.c magic is supplied via a custom patch. */
 #  define HasXdmAuth			YES
 
+#  ifdef LinuxArchitecture
 /* Glide2 only works on i386. */
 #  ifdef i386Architecture
 #   define HasGlide2			YES
@@ -271,6 +272,7 @@
 #   define Glide3IncDir		/usr/include/glide3
 #  endif /* i386Architecture || AlphaArchitecture || ia64Architecture ||
             AMD64Architecture */
+#  endif /* LinuxArchitecture */
 
 /* Enable extended instruction set support. */
 #  ifdef i386Architecture
