diff -Naur -x .svn -x CVS posixtestsuite.orig/locate-test posixtestsuite/locate-test
--- posixtestsuite.orig/locate-test	2005-03-14 15:53:50.000000000 +0200
+++ posixtestsuite/locate-test	2005-11-29 03:48:39.000000000 +0200
@@ -60,19 +60,19 @@
           shift;
           ;;
       "--fmake")
-          find functional/ -type f -maxdepth 2 -mindepth 2 -name "Makefile" -exec dirname '{}' ';'
+          find functional/ -maxdepth 2 -mindepth 2 -type f -name "Makefile" -exec dirname '{}' ';'
           exit 0;
           ;;
       "--frun")
-          find functional/ -type f -maxdepth 2 -mindepth 2 -name "run.sh" -exec dirname '{}' ';' 
+          find functional/ -maxdepth 2 -mindepth 2 -type f -name "run.sh" -exec dirname '{}' ';' 
           exit 0;
           ;;
       "--smake")
-          find stress/ -type f -maxdepth 2 -mindepth 2 -name "Makefile" -exec dirname '{}' ';'
+          find stress/ -maxdepth 2 -mindepth 2 -type f -name "Makefile" -exec dirname '{}' ';'
           exit 0;
           ;;
       "--srun")
-          find stress/ -type f -maxdepth 2 -mindepth 2 -name "run.sh" -exec dirname '{}' ';'
+          find stress/ -maxdepth 2 -mindepth 2 -type f -name "run.sh" -exec dirname '{}' ';'
           exit 0;
           ;;
       "--help")
