diff -ur posixtestsuite-1.4.3.old/LDFLAGS posixtestsuite-1.4.3/LDFLAGS
--- posixtestsuite-1.4.3.old/LDFLAGS	2004-07-06 00:17:13.000000000 +0200
+++ posixtestsuite-1.4.3/LDFLAGS	2004-09-01 14:33:58.000000000 +0200
@@ -3,6 +3,5 @@
 #
 #Place the -l, -L, or -D piece before the item to link.
 #Lines beginning with # are comments.
-#
-#For example,
-#-lpthread -D_GNU_SOURCE
+
+-lpthread -D_GNU_SOURCE -lrt
