--- a/sbin/ifconfig/Makefile
+++ b/sbin/ifconfig/Makefile
@@ -27,20 +27,17 @@
 .endif
 
 SRCS+=	ifclone.c		# clone device support
-SRCS+=	ifmac.c			# MAC support
 SRCS+=	ifmedia.c		# SIOC[GS]IFMEDIA support
-SRCS+=	iffib.c			# non-default FIB support
 SRCS+=	ifvlan.c		# SIOC[GS]ETVLAN support
 SRCS+=	ifgre.c			# GRE keys etc
 SRCS+=	ifgif.c			# GIF reversed header workaround
 
 SRCS+=	ifieee80211.c regdomain.c # SIOC[GS]IEEE80211 support
 DPADD+=	${LIBBSDXML} ${LIBSBUF}
-LDADD+=	-lbsdxml -lsbuf
+LDADD+=	-lexpat -lsbuf
 
 SRCS+=	ifcarp.c		# SIOC[GS]VH support
 SRCS+=	ifgroup.c		# ...
-SRCS+=	ifpfsync.c		# pfsync(4) support
 
 SRCS+=	ifbridge.c		# bridge support
 SRCS+=	iflagg.c		# lagg support
@@ -65,6 +62,7 @@
 MAN=	ifconfig.8
 
 CFLAGS+= -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wnested-externs
+LDADD+= -lbsd ../../lib/linkaddr.o
 WARNS?=	2
 
 .include <bsd.prog.mk>
--- a/sbin/ifconfig/ifconfig.c
+++ b/sbin/ifconfig/ifconfig.c
@@ -48,6 +48,10 @@
 #include <sys/module.h>
 #include <sys/linker.h>
 
+#include <sys/types.h>
+#include <sys/sockio.h>
+#include <netinet/ether.h>
+
 #include <net/ethernet.h>
 #include <net/if.h>
 #include <net/if_var.h>
@@ -158,7 +162,7 @@
 	all = downonly = uponly = namesonly = noload = verbose = 0;
 
 	/* Parse leading line options */
-	strlcpy(options, "adklmnuv", sizeof(options));
+	strlcpy(options, "+adklmnuv", sizeof(options));
 	for (p = opts; p != NULL; p = p->next)
 		strlcat(options, p->opt, sizeof(options));
 	while ((c = getopt(argc, argv, options)) != -1) {
@@ -953,7 +957,7 @@
 	putchar('\n');
 
 	for (;;) {
-		if ((descr = reallocf(descr, descrlen)) != NULL) {
+		if ((descr = realloc(descr, descrlen)) != NULL) {
 			ifr.ifr_buffer.buffer = descr;
 			ifr.ifr_buffer.length = descrlen;
 			if (ioctl(s, SIOCGIFDESCR, &ifr) == 0) {
--- a/sbin/ifconfig/af_inet6.c
+++ b/sbin/ifconfig/af_inet6.c
@@ -42,7 +42,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <ifaddrs.h>
 
 #include <arpa/inet.h>
 
@@ -53,6 +52,8 @@
 #include <netdb.h>
 
 #include <netinet6/nd6.h>	/* Define ND6_INFINITE_LIFETIME */
+#include <time.h>
+#include <ifaddrs.h>
 
 #include "ifconfig.h"
 
--- a/sbin/ifconfig/af_link.c
+++ b/sbin/ifconfig/af_link.c
@@ -46,6 +46,8 @@
 #include <net/if_dl.h>
 #include <net/if_types.h>
 #include <net/ethernet.h>
+#include <netinet/ether.h>
+#include "../../lib/linkaddr.h"
 
 #include "ifconfig.h"
 
--- a/sbin/ifconfig/ifieee80211.c
+++ b/sbin/ifconfig/ifieee80211.c
@@ -83,6 +83,9 @@
 #include <fcntl.h>
 #include <inttypes.h>
 #include <stdio.h>
+#include <stdint.h>
+#include <netinet/ether.h>
+#include "../../lib/linkaddr.h"
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
--- a/sbin/ifconfig/af_inet.c
+++ b/sbin/ifconfig/af_inet.c
@@ -43,13 +43,13 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <ifaddrs.h>
 
 #include <netinet/in.h>
 #include <net/if_var.h>		/* for struct ifaddr */
 #include <netinet/in_var.h>
 #include <arpa/inet.h>
 #include <netdb.h>
+#include <ifaddrs.h>
 
 #include "ifconfig.h"
 
--- a/sbin/ifconfig/af_ipx.c
+++ b/sbin/ifconfig/af_ipx.c
@@ -41,9 +41,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <ifaddrs.h>
 
 #include <net/if_var.h>
+#include <ifaddrs.h>
 #define IPTUNNEL
 #include <netipx/ipx.h>
 #include <netipx/ipx_if.h>
