--- a/sbin/camcontrol/camcontrol.c
+++ b/sbin/camcontrol/camcontrol.c
@@ -30,7 +30,7 @@
 __FBSDID("$FreeBSD$");
 
 #include <sys/ioctl.h>
-#include <sys/stdint.h>
+#include <stdint.h>
 #include <sys/types.h>
 #include <sys/endian.h>
 #include <sys/sbuf.h>
@@ -132,12 +132,12 @@
 };
 
 #ifndef MINIMALISTIC
-static const char scsicmd_opts[] = "a:c:dfi:o:r";
-static const char readdefect_opts[] = "f:GP";
-static const char negotiate_opts[] = "acD:M:O:qR:T:UW:";
-static const char smprg_opts[] = "l";
-static const char smppc_opts[] = "a:A:d:lm:M:o:p:s:S:T:";
-static const char smpphylist_opts[] = "lq";
+static const char scsicmd_opts[] = "+a:c:dfi:o:r";
+static const char readdefect_opts[] = "+f:GP";
+static const char negotiate_opts[] = "+acD:M:O:qR:T:UW:";
+static const char smprg_opts[] = "+l";
+static const char smppc_opts[] = "+a:A:d:lm:M:o:p:s:S:T:";
+static const char smpphylist_opts[] = "+lq";
 #endif
 
 struct camcontrol_opts option_table[] = {
@@ -6036,7 +6036,6 @@
 	 * the arguments again.
 	 */
 	optind = optstart;
-	optreset = 1;
 
 	switch(cmdlist) {
 #ifndef MINIMALISTIC
--- a/sbin/camcontrol/util.c
+++ b/sbin/camcontrol/util.c
@@ -46,7 +46,7 @@
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
-#include <sys/stdint.h>
+#include <stdint.h>
 #include <sys/types.h>
 
 #include <stdlib.h>
--- a/sbin/camcontrol/Makefile
+++ b/sbin/camcontrol/Makefile
@@ -12,7 +12,7 @@
 WARNS?= 3
 .endif
 DPADD=	${LIBCAM} ${LIBSBUF} ${LIBUTIL}
-LDADD=	-lcam -lsbuf -lutil
+LDADD=	-lcam -lsbuf -lutil -lbsd
 MAN=	camcontrol.8
 
 .include <bsd.prog.mk>
