diff -ur src.old/sys/boot/i386/boot2/Makefile src/sys/boot/i386/boot2/Makefile
--- src.old/sys/boot/i386/boot2/Makefile	2003-06-26 05:51:57.000000000 +0200
+++ src/sys/boot/i386/boot2/Makefile	2004-05-20 00:30:30.000000000 +0200
@@ -36,7 +36,6 @@
 
 CFLAGS=	-elf -ffreestanding -Os -fno-builtin \
 	-fno-guess-branch-probability -fomit-frame-pointer\
-	-mno-align-long-strings \
 	-mrtd \
 	-D${BOOT2_UFS} \
 	-I${.CURDIR}/../../common \
@@ -80,7 +79,7 @@
 		-o boot2.ld -P 1 boot2.bin
 	@ls -l boot2.ld | awk '{ x = 7680 - $$5; \
 		print x " bytes available"; if (x < 0) exit 1 }'
-	dd if=boot2.ld of=${.TARGET} obs=7680 conv=osync 2>/dev/null
+	dd if=boot2.ld of=${.TARGET} obs=7680 conv=sync 2>/dev/null
 
 boot2.ldr:
 	dd if=/dev/zero of=${.TARGET} bs=276 count=1 2>/dev/null
diff -ur src.old/sys/boot/i386/loader/Makefile src/sys/boot/i386/loader/Makefile
--- src.old/sys/boot/i386/loader/Makefile	2003-06-26 05:51:57.000000000 +0200
+++ src/sys/boot/i386/loader/Makefile	2004-05-20 00:31:02.000000000 +0200
@@ -48,7 +48,7 @@
 CLEANFILES+=	vers.c vers.o ${PROG}.list ${PROG}.bin ${PROG}.sym ${PROG}.help
 
 CFLAGS+=	-Wall
-LDFLAGS=	-nostdlib -static -Ttext 0x0
+LDFLAGS=	-nostdlib -static -Ttext=0x0
 
 # i386 standalone support library
 LIBI386=	${.OBJDIR}/../libi386/libi386.a
diff -ur src.old/sys/boot/i386/pxeldr/Makefile src/sys/boot/i386/pxeldr/Makefile
--- src.old/sys/boot/i386/pxeldr/Makefile	2003-09-03 10:12:20.000000000 +0200
+++ src/sys/boot/i386/pxeldr/Makefile	2004-05-20 00:31:18.000000000 +0200
@@ -40,7 +40,7 @@
 
 ${BOOT}: ${LDR} ${LOADER}
 	cat ${LDR} ${LOADER} > ${.TARGET}.tmp
-	dd if=${.TARGET}.tmp of=${.TARGET} obs=2k conv=osync
+	dd if=${.TARGET}.tmp of=${.TARGET} obs=2k conv=sync
 	rm ${.TARGET}.tmp
 
 ${LOADER}: ${LOADERBIN} ${BTXLDR} ${BTXKERN} ${BTXCRT}
diff -ur src.old/sys/conf/kern.mk src/sys/conf/kern.mk
--- src.old/sys/conf/kern.mk	2003-11-05 00:29:17.000000000 +0100
+++ src/sys/conf/kern.mk	2004-05-20 00:32:02.000000000 +0200
@@ -8,7 +8,7 @@
 # also pop up, but are easier to fix.
 CWARNFLAGS?=	-Wall -Wredundant-decls -Wnested-externs -Wstrict-prototypes \
 		-Wmissing-prototypes -Wpointer-arith -Winline -Wcast-qual \
-		-fformat-extensions -std=c99
+		-std=c99
 #
 # The following flags are next up for working on:
 #	-W
@@ -23,7 +23,7 @@
 # cache tag lines)
 #
 .if ${MACHINE_ARCH} == "i386"
-CFLAGS+=	-mno-align-long-strings -mpreferred-stack-boundary=2
+CFLAGS+=	-mpreferred-stack-boundary=2
 INLINE_LIMIT?=	15000
 .endif
 
diff -ur src.old/sys/conf/kern.post.mk src/sys/conf/kern.post.mk
--- src.old/sys/conf/kern.post.mk	2003-11-25 05:12:43.000000000 +0100
+++ src/sys/conf/kern.post.mk	2004-05-20 00:32:55.000000000 +0200
@@ -178,11 +178,9 @@
 .if exists(${DESTDIR}${KODIR})
 	-thiskernel=`sysctl -n kern.bootfile` ; \
 	if [ "`dirname "$$thiskernel"`" != ${DESTDIR}${KODIR} ] ; then \
-		chflags -R noschg ${DESTDIR}${KODIR} ; \
 		rm -rf ${DESTDIR}${KODIR} ; \
 	else \
 		if [ -d ${DESTDIR}${KODIR}.old ] ; then \
-			chflags -R noschg ${DESTDIR}${KODIR}.old ; \
 			rm -rf ${DESTDIR}${KODIR}.old ; \
 		fi ; \
 		mv ${DESTDIR}${KODIR} ${DESTDIR}${KODIR}.old ; \
@@ -191,17 +189,17 @@
 .endif
 	mkdir -p ${DESTDIR}${KODIR}
 .if defined(DEBUG) && defined(INSTALL_DEBUG)
-	${INSTALL} -p -m 555 -o root -g wheel ${FULLKERNEL} ${DESTDIR}${KODIR}
+	${INSTALL} -p -m 555 -o root -g root ${FULLKERNEL} ${DESTDIR}${KODIR}
 .else
-	${INSTALL} -p -m 555 -o root -g wheel ${KERNEL_KO} ${DESTDIR}${KODIR}
+	${INSTALL} -p -m 555 -o root -g root ${KERNEL_KO} ${DESTDIR}${KODIR}
 .endif
 
 kernel-reinstall:
 	@-chflags -R noschg ${DESTDIR}${KODIR}
 .if defined(DEBUG) && defined(INSTALL_DEBUG)
-	${INSTALL} -p -m 555 -o root -g wheel ${FULLKERNEL} ${DESTDIR}${KODIR}
+	${INSTALL} -p -m 555 -o root -g root ${FULLKERNEL} ${DESTDIR}${KODIR}
 .else
-	${INSTALL} -p -m 555 -o root -g wheel ${KERNEL_KO} ${DESTDIR}${KODIR}
+	${INSTALL} -p -m 555 -o root -g root ${KERNEL_KO} ${DESTDIR}${KODIR}
 .endif
 
 config.o env.o hints.o majors.o vers.o vnode_if.o:
