--- src/sys/i386/conf/GENERIC~	2004-05-19 01:49:11.000000000 +0200
+++ src/sys/i386/conf/GENERIC	2004-05-19 02:04:20.000000000 +0200
@@ -41,6 +43,7 @@
 options 	NFSSERVER		#Network Filesystem Server
 options 	NFS_ROOT		#NFS usable as /, requires NFSCLIENT
 options 	MSDOSFS			#MSDOS Filesystem
+options		EXT2FS			#Ext2 Filesystem
 options 	CD9660			#ISO 9660 Filesystem
 options 	PROCFS			#Process filesystem (requires PSEUDOFS)
 options 	PSEUDOFS		#Pseudo-filesystem framework
