diff -ur src.old/sys/conf/files.i386 src/sys/conf/files.i386
--- src.old/sys/conf/files.i386	2004-05-09 04:15:55.000000000 +0200
+++ src/sys/conf/files.i386	2004-05-09 04:43:57.000000000 +0200
@@ -51,11 +51,6 @@
 	compile-with	"uudecode < $S/contrib/dev/fla/i386/msysosak.o.uu" \
 	no-implicit-rule
 #
-trlld.o				optional	oltr			\
-	dependency	"$S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	no-implicit-rule
-#
 #
 compat/linux/linux_file.c	optional	compat_linux
 compat/linux/linux_getcwd.c	optional	compat_linux
@@ -88,10 +83,6 @@
 compat/svr4/svr4_termios.c		optional	compat_svr4
 compat/svr4/svr4_ttold.c		optional	compat_svr4
 contrib/dev/fla/fla.c		optional	fla
-contrib/dev/oltr/if_oltr.c	optional	oltr
-contrib/dev/oltr/trlldbm.c	optional	oltr
-contrib/dev/oltr/trlldhm.c	optional	oltr
-contrib/dev/oltr/trlldmac.c	optional	oltr
 bf_enc.o			optional	ipsec ipsec_esp		\
 	dependency	"$S/crypto/blowfish/arch/i386/bf_enc.S $S/crypto/blowfish/arch/i386/bf_enc_586.S $S/crypto/blowfish/arch/i386/bf_enc_686.S"		\
 	compile-with	"${CC} -c -I$S/crypto/blowfish/arch/i386 ${ASM_CFLAGS} ${WERROR} ${.IMPSRC}"	\
diff -ur src.old/sys/conf/files.pc98 src/sys/conf/files.pc98
--- src.old/sys/conf/files.pc98	2004-05-09 04:15:55.000000000 +0200
+++ src/sys/conf/files.pc98	2004-05-09 04:44:17.000000000 +0200
@@ -38,11 +38,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-trlld.o				optional	oltr			\
-	dependency	"$S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	no-implicit-rule
-#
 #
 compat/linux/linux_file.c	optional	compat_linux
 compat/linux/linux_getcwd.c	optional	compat_linux
@@ -74,10 +69,6 @@
 compat/svr4/svr4_sysvec.c		optional	compat_svr4
 compat/svr4/svr4_termios.c		optional	compat_svr4
 compat/svr4/svr4_ttold.c		optional	compat_svr4
-contrib/dev/oltr/if_oltr.c	optional	oltr
-contrib/dev/oltr/trlldbm.c	optional	oltr
-contrib/dev/oltr/trlldhm.c	optional	oltr
-contrib/dev/oltr/trlldmac.c	optional	oltr
 bf_enc.o			optional	ipsec ipsec_esp		\
 	dependency	"$S/crypto/blowfish/arch/i386/bf_enc.S $S/crypto/blowfish/arch/i386/bf_enc_586.S $S/crypto/blowfish/arch/i386/bf_enc_686.S"		\
 	compile-with	"${CC} -c -I$S/crypto/blowfish/arch/i386 ${ASM_CFLAGS} ${WERROR} ${.IMPSRC}"	\
diff -ur src.old/sys/modules/Makefile src/sys/modules/Makefile
--- src.old/sys/modules/Makefile	2004-05-09 04:15:55.000000000 +0200
+++ src/sys/modules/Makefile	2004-05-09 04:52:39.000000000 +0200
@@ -210,7 +210,6 @@
 	netgraph \
 	nsp \
 	nwfs \
-	oltr \
 	pccard \
 	pcfclock \
 	pecoff \
