diff -ur src/sys/cam/scsi.old/scsi_low.c src/sys/cam/scsi/scsi_low.c
--- src/sys/cam/scsi.old/scsi_low.c	2004-05-22 02:38:09.000000000 +0200
+++ src/sys/cam/scsi/scsi_low.c	2004-05-22 03:45:21.000000000 +0200
@@ -18,7 +18,7 @@
 #define	SCSI_LOW_TARGET_OPEN
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__
+#ifdef	__FreeBSD_kernel__
 #define	SCSI_LOW_FLAGS_QUIRKS_OK
 #endif	/* __FreeBSD__ */
 
@@ -73,7 +73,7 @@
 #include <sys/kernel.h>
 
 #ifdef __FreeBSD_kernel__
-#if __FreeBSD_version >= 500001
+#if 1
 #include <sys/bio.h>
 #else
 #include <machine/clock.h>
Only in src/sys/cam/scsi/: scsi_low.c~
diff -ur src/sys/cam/scsi.old/scsi_low.h src/sys/cam/scsi/scsi_low.h
--- src/sys/cam/scsi.old/scsi_low.h	2004-05-22 02:38:11.000000000 +0200
+++ src/sys/cam/scsi/scsi_low.h	2004-05-22 03:46:26.000000000 +0200
@@ -53,7 +53,7 @@
 #define	SCSI_LOW_INTERFACE_XS
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__
+#ifdef	__FreeBSD_kernel__
 #define	SCSI_LOW_INTERFACE_CAM
 #define	CAM
 #endif	/* __FreeBSD__ */
Only in src/sys/cam/scsi/: scsi_low.h~
diff -ur src/sys/cam/scsi.old/scsi_low_pisa.c src/sys/cam/scsi/scsi_low_pisa.c
--- src/sys/cam/scsi.old/scsi_low_pisa.c	2003-06-10 20:14:05.000000000 +0200
+++ src/sys/cam/scsi/scsi_low_pisa.c	2004-05-22 03:46:45.000000000 +0200
@@ -117,11 +117,11 @@
 }
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__ 
+#ifdef	__FreeBSD_kernel__
 #include <sys/param.h>
 #include <sys/systm.h>
 #include <sys/kernel.h>
-#if __FreeBSD_version >= 500001
+#if 1
 #include <sys/bio.h>
 #endif
 #include <sys/buf.h>
Only in src/sys/cam/scsi/: scsi_low_pisa.c~
diff -ur src/sys/cam/scsi.old/scsi_low_pisa.h src/sys/cam/scsi/scsi_low_pisa.h
--- src/sys/cam/scsi.old/scsi_low_pisa.h	2002-03-20 09:55:22.000000000 +0100
+++ src/sys/cam/scsi/scsi_low_pisa.h	2004-05-22 03:46:59.000000000 +0200
@@ -40,7 +40,7 @@
 int scsi_low_notify_pisa(pisa_device_handle_t, pisa_event_t);
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__
+#ifdef	__FreeBSD_kernel__
 int scsi_low_activate_pisa(struct scsi_low_softc *, int);
 int scsi_low_deactivate_pisa(struct scsi_low_softc *);
 #endif	/* __FreeBSD__ */
Only in src/sys/cam/scsi/: scsi_low_pisa.h~
