diff -ur src/sys/dev/ncv.old/ncr53c500.c src/sys/dev/ncv/ncr53c500.c
--- src/sys/dev/ncv.old/ncr53c500.c	2004-05-22 02:38:10.000000000 +0200
+++ src/sys/dev/ncv/ncr53c500.c	2004-05-22 03:06:19.000000000 +0200
@@ -43,7 +43,7 @@
 #include <sys/param.h>
 #include <sys/systm.h>
 #include <sys/kernel.h>
-#if defined(__FreeBSD_kernel__) && __FreeBSD_version >= 500001
+#if defined(__FreeBSD_kernel__)
 #include <sys/bio.h>
 #endif	/* __FreeBSD__ */
 #include <sys/buf.h>
Only in src/sys/dev/ncv/: ncr53c500.c~
diff -ur src/sys/dev/ncv.old/ncr53c500var.h src/sys/dev/ncv/ncr53c500var.h
--- src/sys/dev/ncv.old/ncr53c500var.h	2002-06-01 01:39:04.000000000 +0200
+++ src/sys/dev/ncv/ncr53c500var.h	2004-05-22 03:24:00.000000000 +0200
@@ -50,7 +50,7 @@
 	void *sc_ih;
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__
+#ifdef	__FreeBSD_kernel__
 	bus_space_tag_t sc_iot;
 	bus_space_tag_t sc_memt;
 	bus_space_handle_t sc_ioh;
Only in src/sys/dev/ncv/: ncr53c500var.h~
