diff -ur src.old/sys/conf/kmod.mk src/sys/conf/kmod.mk
--- src.old/sys/conf/kmod.mk	2003-11-19 06:08:26.000000000 +0100
+++ src/sys/conf/kmod.mk	2004-05-22 03:57:42.000000000 +0200
@@ -146,7 +146,7 @@
 .endif
 	awk -f ${SYSDIR}/conf/kmod_syms.awk ${.TARGET} \
 		${.OBJDIR}/export_syms | \
-	xargs -J% ${OBJCOPY} % ${.TARGET}
+	xargs -i% ${OBJCOPY} % ${.TARGET}
 .endif
 .endif
 
diff -ur src.old/sys/contrib/dev/acpica/acpica_prep.sh src/sys/contrib/dev/acpica/acpica_prep.sh
--- src.old/sys/contrib/dev/acpica/acpica_prep.sh	2003-04-29 20:48:17.000000000 +0200
+++ src/sys/contrib/dev/acpica/acpica_prep.sh	2004-05-22 03:58:14.000000000 +0200
@@ -48,7 +48,7 @@
 
 # move files to destination
 echo copy
-find ${wrk} -type f | xargs -J % mv % ${dst}
+find ${wrk} -type f | xargs -i % mv % ${dst}
 mv CHANGES.txt ${dst}
 
 # update src/headers for appropriate paths
