diff -ur src/sys/dev/raidframe.old/rf_copyback.c src/sys/dev/raidframe/rf_copyback.c
--- src/sys/dev/raidframe.old/rf_copyback.c	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_copyback.c	2004-05-22 04:36:45.000000000 +0200
@@ -45,7 +45,7 @@
 #if defined(__FreeBSD_kernel__)
 #include <sys/types.h>
 #include <sys/systm.h>
-#if __FreeBSD_version > 500005
+#if 1
 #include <sys/bio.h>
 #endif
 #endif
diff -ur src/sys/dev/raidframe.old/rf_dag.h src/sys/dev/raidframe/rf_dag.h
--- src/sys/dev/raidframe.old/rf_dag.h	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_dag.h	2004-05-22 04:37:02.000000000 +0200
@@ -49,7 +49,7 @@
 #define RF_INTR_CONTEXT     1	/* we were invoked from interrupt context */
 #define RF_MAX_ANTECEDENTS 20	/* max num of antecedents a node may posses */
 
-#if defined(__FreeBSD_kernel__) && __FreeBSD_version > 500005
+#if defined(__FreeBSD_kernel__)
 #include <sys/bio.h>
 #endif
 #include <sys/buf.h>
diff -ur src/sys/dev/raidframe.old/rf_diskqueue.c src/sys/dev/raidframe/rf_diskqueue.c
--- src/sys/dev/raidframe.old/rf_diskqueue.c	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_diskqueue.c	2004-05-22 04:37:22.000000000 +0200
@@ -146,7 +146,7 @@
 #define RF_DQD_INC       16
 #define RF_DQD_INITIAL   64
 
-#if defined(__FreeBSD_kernel__) && __FreeBSD_version > 500005
+#if defined(__FreeBSD_kernel__)
 #include <sys/bio.h>
 #endif
 
diff -ur src/sys/dev/raidframe.old/rf_driver.c src/sys/dev/raidframe/rf_driver.c
--- src/sys/dev/raidframe.old/rf_driver.c	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_driver.c	2004-05-22 04:37:37.000000000 +0200
@@ -123,7 +123,7 @@
 #include <dev/raidframe/rf_shutdown.h>
 #include <dev/raidframe/rf_kintf.h>
 
-#if defined(__FreeBSD_kernel__) && __FreeBSD_version > 500005
+#if defined(__FreeBSD_kernel__)
 #include <sys/bio.h>
 #endif
 
@@ -157,7 +157,7 @@
 	RF_LTSLEEP(&((_raid_)->accesses_suspended), PRIBIO, \
 		"raidframe quiesce", 0, &((_raid_)->access_suspend_mutex))
 
-#if defined(__FreeBSD_kernel__) && __FreeBSD_version > 500005
+#if defined(__FreeBSD_kernel__)
 #define IO_BUF_ERR(bp, err) { \
 	bp->bio_flags |= BIO_ERROR; \
 	bp->bio_resid = bp->bio_bcount; \
diff -ur src/sys/dev/raidframe.old/rf_kintf.h src/sys/dev/raidframe/rf_kintf.h
--- src/sys/dev/raidframe.old/rf_kintf.h	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_kintf.h	2004-05-22 04:37:49.000000000 +0200
@@ -41,7 +41,7 @@
 #define RF_LTSLEEP(cond, pri, text, time, mutex)	\
 		ltsleep(cond, pri, text, time, mutex)
 #elif defined(__FreeBSD_kernel__)
-#if __FreeBSD_version > 500005
+#if 1
 #define RF_LTSLEEP(cond, pri, text, time, mutex) \
 		msleep(cond, mutex, pri, text, time);
 #else
diff -ur src/sys/dev/raidframe.old/rf_reconstruct.c src/sys/dev/raidframe/rf_reconstruct.c
--- src/sys/dev/raidframe.old/rf_reconstruct.c	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_reconstruct.c	2004-05-22 04:38:08.000000000 +0200
@@ -39,7 +39,7 @@
 #include <sys/time.h>
 #if defined(__FreeBSD_kernel__)
 #include <sys/systm.h>
-#if __FreeBSD_version > 500005
+#if 1
 #include <sys/bio.h>
 #endif
 #endif
diff -ur src/sys/dev/raidframe.old/rf_stripelocks.h src/sys/dev/raidframe/rf_stripelocks.h
--- src/sys/dev/raidframe.old/rf_stripelocks.h	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_stripelocks.h	2004-05-22 04:38:25.000000000 +0200
@@ -47,7 +47,7 @@
 
 #if defined(__FreeBSD_kernel__)
 #include <sys/types.h>
-#if __FreeBSD_version > 500005
+#if 1
 #include <sys/bio.h>
 #endif
 #if _KERNEL
diff -ur src/sys/dev/raidframe.old/rf_threadstuff.c src/sys/dev/raidframe/rf_threadstuff.c
--- src/sys/dev/raidframe.old/rf_threadstuff.c	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_threadstuff.c	2004-05-22 04:38:36.000000000 +0200
@@ -172,7 +172,7 @@
 /*
  * Kernel
  */
-#if defined(__FreeBSD_kernel__) && __FreeBSD_version > 500005
+#if defined(__FreeBSD_kernel__)
 int 
 rf_mutex_init(m, s)
 decl_simple_lock_data(, *m)
diff -ur src/sys/dev/raidframe.old/rf_threadstuff.h src/sys/dev/raidframe/rf_threadstuff.h
--- src/sys/dev/raidframe.old/rf_threadstuff.h	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_threadstuff.h	2004-05-22 04:38:56.000000000 +0200
@@ -66,7 +66,7 @@
     RF_ThreadGroup_t * g, char *file, int line);
 
 #include <sys/lock.h>
-#if defined(__FreeBSD__ ) && __FreeBSD_version > 500005
+#if defined(__FreeBSD_kernel__)
 #include <sys/mutex.h>
 #define decl_simple_lock_data(a,b) a struct mtx b;
 #define simple_lock_addr(a) ((struct mtx *)&(a))
@@ -124,7 +124,7 @@
 #define RF_THREAD_EXIT(ret)	\
 	kthread_exit(ret)
 #elif defined(__FreeBSD_kernel__)
-#if __FreeBSD_version > 500005
+#if 1
 #define RF_CREATE_THREAD(_handle_, _func_, _arg_, _name_) \
 	kthread_create((void (*)(void *))(_func_), (void *)(_arg_), \
 	    (struct proc **)&(_handle_), 0, 4, _name_)
@@ -210,7 +210,7 @@
 }
 #endif
 
-#if defined(__FreeBSD_kernel__) && __FreeBSD_version > 500005
+#if defined(__FreeBSD_kernel__)
 int     rf_mutex_init(struct mtx *, const char *);
 int     rf_mutex_destroy(struct mtx *);
 int	_rf_create_managed_mutex(RF_ShutdownList_t **, struct mtx *,
diff -ur src/sys/dev/raidframe.old/rf_types.h src/sys/dev/raidframe/rf_types.h
--- src/sys/dev/raidframe.old/rf_types.h	2004-05-22 04:03:24.000000000 +0200
+++ src/sys/dev/raidframe/rf_types.h	2004-05-22 04:39:10.000000000 +0200
@@ -239,7 +239,7 @@
 typedef union RF_GenericParam_u RF_DagParam_t;
 typedef union RF_GenericParam_u RF_CBParam_t;
 
-#if defined(__FreeBSD_kernel__) && __FreeBSD_version > 500005
+#if defined(__FreeBSD_kernel__)
 typedef struct bio	*RF_Buf_t;
 #else
 typedef struct buf	*RF_Buf_t;
