diff -ur src/sys/netgraph.old/atm/atmpif/ng_atmpif_harp.c src/sys/netgraph/atm/atmpif/ng_atmpif_harp.c
--- src/sys/netgraph.old/atm/atmpif/ng_atmpif_harp.c	2003-08-11 10:40:01.000000000 +0200
+++ src/sys/netgraph/atm/atmpif/ng_atmpif_harp.c	2004-05-22 04:55:31.000000000 +0200
@@ -200,7 +200,7 @@
 	    "%s", "Virt. ATM 1.0");
 	snprintf(vup->vu_config.ac_firm_vers,
 	    sizeof(vup->vu_config.ac_firm_vers),
-	    "%d", __FreeBSD__);
+	    "%d", 5);
 
 	/*
 	 * Set the interface capabilities
