
Author: rmh
Status: in BTS

diff -ur portage-2.1_pre4.old/pym/portage_data.py portage-2.1_pre4/pym/portage_data.py
--- portage-2.1_pre4.old/pym/portage_data.py	2006-01-28 00:49:48 +0000
+++ portage-2.1_pre4/pym/portage_data.py	2006-02-06 11:53:46 +0000
@@ -11,7 +11,7 @@
 ostype=os.uname()[0]
 
 lchown = None
-if ostype=="Linux" or ostype.lower().endswith("gnu"):
+if ostype=="Linux" or ostype.lower().startswith("gnu"):
 	userland="GNU"
 	os.environ["XARGS"]="xargs -r"
 elif ostype == "Darwin":
