
Author: rmh
Status: in BTS

--- /usr/lib/portage/pym/portage_data.py.old	2006-02-04 21:37:42 +0000
+++ /usr/lib/portage/pym/portage_data.py	2006-02-04 21:37:52 +0000
@@ -11,7 +11,7 @@
 ostype=os.uname()[0]
 
 lchown = None
-if ostype=="Linux" or ostype.lower().endswith("gnu"):
+if ostype=="Linux" or ostype.lower().startswith("gnu"):
 	userland="GNU"
 	os.environ["XARGS"]="xargs -r"
 elif ostype == "Darwin":
