--- build-tree.orig/apache2/srclib/apr/build/apr_hints.m4	2005-08-01 21:43:33.923988884 +0200
+++ build-tree/apache2/srclib/apr/build/apr_hints.m4	2005-08-01 21:44:16.534822122 +0200
@@ -101,6 +101,9 @@
     *-hp-hpux*)
 	APR_ADDTO(CPPFLAGS, [-DHPUX -D_REENTRANT])
 	;;
+    *-k*bsd*-gnu)
+    	APR_ADDTO(CPPFLAGS, [-D_REENTRANT])
+    	;;
     *-linux-*)
         case `uname -r` in
 	    2.* )  APR_ADDTO(CPPFLAGS, [-DLINUX=2])
