
Status: sent to nutmeg (I'm banned in upstream ml)

diff -ur exim-snapshot.old/OS/os.h-Linux exim-snapshot/OS/os.h-Linux
--- exim-snapshot.old/OS/os.h-Linux	2005-01-14 15:19:48.000000000 +0100
+++ exim-snapshot/OS/os.h-Linux	2005-02-07 21:45:22.000000000 +0100
@@ -5,11 +5,24 @@
 #define CRYPT_H
 #define GLIBC_IP_OPTIONS
 #define HAVE_MMAP
+#define HAVE_BSD_GETLOADAVG
 #define HAVE_SYS_STATVFS_H
 #define NO_IP_VAR_H
-#define SIOCGIFCONF_GIVES_ADDR
 #define SIG_IGN_WORKS
 
+#define F_FREESP     O_TRUNC
+typedef struct flock flock_t;
+
+#define os_strsignal strsignal
+#define OS_STRSIGNAL
+
+#if defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__NetBSD_ker$
+#define SIOCGIFCONF_GIVES_ADDR
+#define HAVE_SYS_MOUNT_H
+#endif
+
+#if defined(__linux__)
+
 /* Some versions of Linux need explicit sync-ing of directories as well as
 files. This setting requests that. If the directory is on NFS, it may not
 be possible to sync it - in that case, Exim now should ignore the error. But
@@ -22,12 +35,6 @@
 /* Other OS have "const" in here */
 #define ICONV_ARG2_TYPE char **
 
-#define F_FREESP     O_TRUNC
-typedef struct flock flock_t;
-
-#define os_strsignal strsignal
-#define OS_STRSIGNAL
-
 #define os_find_running_interfaces os_find_running_interfaces_linux
 
 /* Need a prototype for the Linux-specific function. The structure hasn't
@@ -36,4 +43,6 @@
 struct ip_address_item;
 extern struct ip_address_item *os_find_running_interfaces_linux(void);
 
+#endif /* __linux__ */
+
 /* End */
diff -ur exim-snapshot.old/scripts/os-type exim-snapshot/scripts/os-type
--- exim-snapshot.old/scripts/os-type	2005-01-14 15:19:48.000000000 +0100
+++ exim-snapshot/scripts/os-type	2005-02-07 21:03:10.000000000 +0100
@@ -21,6 +21,9 @@
 
 case "$os" in '') os=`uname -s`;; esac
 
+# Identify Glibc systems under different names.
+case "$os" in GNU|GNU/*|Linux) os=Linux;; esac
+
 # It is believed that all systems respond to uname -s, but just in case
 # there is one that doesn't, use the shell's $OSTYPE variable. It is known
 # to be unhelpful for some systems (under IRIX is it "irix" and under BSDI
