
Status: in BTS and upstream BTS

--- pearpc-0.3.1.old/configure.in	2004-09-05 18:24:36.000000000 +0200
+++ pearpc-0.3.1/configure.in	2005-02-05 05:02:14.000000000 +0100
@@ -61,7 +61,7 @@
 NASM_OPTS="-f elf"
 
 case "$target_os" in
-linux*) 
+linux* | gnu* | k*bsd*-gnu)
 	OSAPI_DIR=posix
 	if test "x$enable_ui" = "xno"; then
 		enable_ui="x11"
@@ -69,7 +69,7 @@
 	AC_MSG_RESULT([*** LINUX, building POSIX version. we need pthread.])
 	PPC_LDADD="$PPC_LDADD -lpthread -lrt"
 ;;
-*freebsd*) 
+freebsd*) 
 	OSAPI_DIR=posix
 	if test "x$enable_ui" = "xno"; then
 		enable_ui="x11"
