diff -ur src.old/sys/conf/files src/sys/conf/files
--- src.old/sys/conf/files	2004-09-22 21:23:37.000000000 +0200
+++ src/sys/conf/files	2004-11-16 18:37:25.000000000 +0100
@@ -219,7 +219,6 @@
 contrib/dev/acpica/utmisc.c	 optional acpi
 contrib/dev/acpica/utobject.c	 optional acpi
 contrib/dev/acpica/utxface.c	 optional acpi
-contrib/dev/ath/freebsd/ah_osdep.c	optional ath_hal
 contrib/ipfilter/netinet/fil.c		optional ipfilter inet
 contrib/ipfilter/netinet/ip_auth.c	optional ipfilter inet
 contrib/ipfilter/netinet/ip_fil.c	optional ipfilter inet
@@ -357,9 +356,6 @@
 dev/ata/atapi-fd.c	optional atapifd
 dev/ata/atapi-tape.c	optional atapist
 dev/ata/atapi-cam.c	optional atapicam
-dev/ath/if_ath.c	optional ath
-dev/ath/if_ath_pci.c	optional ath pci
-dev/ath/if_ath_pci.c	optional ath card
 dev/awi/am79c930.c	optional awi
 dev/awi/awi.c		optional awi
 dev/awi/if_awi_pccard.c	optional awi card
diff -ur src.old/sys/conf/files.i386 src/sys/conf/files.i386
--- src.old/sys/conf/files.i386	2004-10-24 20:02:52.000000000 +0200
+++ src/sys/conf/files.i386	2004-11-16 18:37:25.000000000 +0100
@@ -46,20 +46,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-trlld.o				optional	oltr			\
-	dependency	"$S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	no-implicit-rule
-#
-hal.o				optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/freebsd/i386-elf.hal.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/ath/freebsd/i386-elf.hal.o.uu" \
-	no-implicit-rule
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/contrib/dev/hptmv/i386-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/hptmv/i386-elf.raid.o.uu" \
-	no-implicit-rule
 #
 #
 compat/linux/linux_file.c	optional	compat_linux
@@ -101,11 +87,6 @@
 contrib/dev/oltr/trlldbm.c	optional	oltr
 contrib/dev/oltr/trlldhm.c	optional	oltr
 contrib/dev/oltr/trlldmac.c	optional	oltr
-contrib/dev/hptmv/gui_lib.c	optional	hptmv
-contrib/dev/hptmv/hptproc.c	optional	hptmv
-contrib/dev/hptmv/ioctl.c	optional	hptmv
-dev/hptmv/entry.c		optional	hptmv
-dev/hptmv/mv.c			optional	hptmv
 bf_enc.o			optional	ipsec ipsec_esp		\
 	dependency	"$S/crypto/blowfish/arch/i386/bf_enc.S $S/crypto/blowfish/arch/i386/bf_enc_586.S $S/crypto/blowfish/arch/i386/bf_enc_686.S"		\
 	compile-with	"${CC} -c -I$S/crypto/blowfish/arch/i386 ${ASM_CFLAGS} ${WERROR} ${.IMPSRC}"	\
diff -ur src.old/sys/conf/kern.pre.mk src/sys/conf/kern.pre.mk
--- src.old/sys/conf/kern.pre.mk	2004-08-13 16:30:26.000000000 +0200
+++ src/sys/conf/kern.pre.mk	2004-11-16 18:37:25.000000000 +0100
@@ -60,9 +60,6 @@
 # ... and the same for pf
 INCLUDES+= -I$S/contrib/pf
 
-# ... and the same for Atheros HAL
-INCLUDES+= -I$S/contrib/dev/ath -I$S/contrib/dev/ath/freebsd
-
 # ... and the same for the NgATM stuff
 INCLUDES+= -I$S/contrib/ngatm
 
diff -ur src.old/sys/i386/conf/GENERIC src/sys/i386/conf/GENERIC
--- src.old/sys/i386/conf/GENERIC	2004-11-16 18:26:16.000000000 +0100
+++ src/sys/i386/conf/GENERIC	2004-11-16 18:37:25.000000000 +0100
@@ -113,7 +113,6 @@
 device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
diff -ur src.old/sys/modules/Makefile src/sys/modules/Makefile
--- src.old/sys/modules/Makefile	2004-10-24 20:02:52.000000000 +0200
+++ src/sys/modules/Makefile	2004-11-16 18:37:25.000000000 +0100
@@ -24,8 +24,6 @@
 	${_arcnet} \
 	${_arl} \
 	${_asr} \
-	ath \
-	${_ath_hal} \
 	aue \
 	${_awi} \
 	axe \
@@ -83,7 +81,6 @@
 	${_hfa} \
 	hifn \
 	hme \
-	${_hptmv} \
 	${_i2c} \
 	${_ibcs2} \
 	${_ichwd} \
@@ -154,7 +151,6 @@
 	ntfs_iconv \
 	nullfs \
 	${_nwfs} \
-	${_oltr} \
 	${_osf1} \
 	patm \
 	${_pccard} \
@@ -287,7 +283,6 @@
 _apm=		apm
 _ar=		ar
 _arcnet=	arcnet
-_ath_hal=	ath_hal
 _awi=		awi
 _bktr=		bktr
 _coff=		coff
@@ -312,7 +307,6 @@
 _ndis=		ndis
 _nsp=		nsp
 _nwfs=		nwfs
-_oltr=		oltr
 _pccard=	pccard
 _pcfclock=	pcfclock
 _pecoff=	pecoff
@@ -347,7 +341,6 @@
 _el=		el
 _ex=		ex
 _exca=		exca
-_hptmv=		hptmv
 _ichwd=		ichwd
 _ida=		ida
 _idt=		idt
