--- src/sys/i386/conf/GENERIC.old	2004-07-06 01:49:21.000000000 +0200
+++ src/sys/i386/conf/GENERIC	2004-07-06 01:54:15.000000000 +0200
@@ -41,6 +41,7 @@
 options 	NFSSERVER		#Network Filesystem Server
 options 	NFS_ROOT		#NFS usable as /, requires NFSCLIENT
 options 	MSDOSFS			#MSDOS Filesystem
+options		EXT2FS			#Ext2 Filesystem
 options 	CD9660			#ISO 9660 Filesystem
 options 	PROCFS			#Process filesystem (requires PSEUDOFS)
 options 	PSEUDOFS		#Pseudo-filesystem framework
