diff -ur sys.old/conf/kmod.mk sys/conf/kmod.mk
--- sys.old/conf/kmod.mk	2004-08-15 01:53:04.000000000 +0200
+++ sys/conf/kmod.mk	2004-11-16 18:22:07.000000000 +0100
@@ -171,7 +171,7 @@
 .endif
 	awk -f ${SYSDIR}/conf/kmod_syms.awk ${.TARGET} \
 		${.OBJDIR}/export_syms | \
-	xargs -J% ${OBJCOPY} % ${.TARGET}
+	xargs -i% ${OBJCOPY} % ${.TARGET}
 .endif
 .endif
 .if !defined(DEBUG_FLAGS) && ${MACHINE_ARCH} == amd64
diff -ur sys.old/contrib/dev/acpica/acpica_prep.sh sys/contrib/dev/acpica/acpica_prep.sh
--- sys.old/contrib/dev/acpica/acpica_prep.sh	2004-04-14 04:03:33.000000000 +0200
+++ sys/contrib/dev/acpica/acpica_prep.sh	2004-11-16 18:22:34.000000000 +0100
@@ -52,12 +52,12 @@
 
 echo copying full dirs
 for i in ${fulldirs}; do
-    find ${wrk} -name ${i} -type d | xargs -J % mv % ${dst}
+    find ${wrk} -name ${i} -type d | xargs -i % mv % ${dst}
 done
 
 # move files to destination
 echo copying flat dirs
-find ${wrk} -type f | xargs -J % mv % ${dst}
+find ${wrk} -type f | xargs -i % mv % ${dst}
 mv CHANGES.txt ${dst}
 
 # update src/headers for appropriate paths
