diff -ur src.old/sys/i386/conf/GENERIC src/sys/i386/conf/GENERIC
--- src.old/sys/i386/conf/GENERIC	2004-10-24 20:02:52.000000000 +0200
+++ src/sys/i386/conf/GENERIC	2004-11-16 18:24:18.000000000 +0100
@@ -39,6 +39,7 @@
 options 	NFSSERVER		# Network Filesystem Server
 options 	NFS_ROOT		# NFS usable as /, requires NFSCLIENT
 options 	MSDOSFS			# MSDOS Filesystem
+options 	EXT2FS			# Ext2 Filesystem
 options 	CD9660			# ISO 9660 Filesystem
 options 	PROCFS			# Process filesystem (requires PSEUDOFS)
 options 	PSEUDOFS		# Pseudo-filesystem framework
