--- sys.orig/i386/conf/GENERIC	2005-08-01 16:45:29.000000000 +0200
+++ sys/i386/conf/GENERIC	2005-08-01 16:46:08.000000000 +0200
@@ -75,6 +75,8 @@
 device		atapicd		# ATAPI CDROM drives
 device		atapifd		# ATAPI floppy drives
 device		atapist		# ATAPI tape drives
+device		atapicam	# emulate ATAPI devices as SCSI ditto via CAM
+				# needs CAM to be present (scbus & pass)
 options 	ATA_STATIC_ID	# Static device numbering
 
 # SCSI Controllers
@@ -166,7 +166,7 @@
 device		ppc
 device		ppbus		# Parallel port bus (required)
 device		lpt		# Printer
-device		plip		# TCP/IP over parallel
+#device		plip		# TCP/IP over parallel
 device		ppi		# Parallel port interface device
 #device		vpo		# Requires scbus and da
 
@@ -234,9 +234,9 @@
 device		io		# I/O device
 device		random		# Entropy device
 device		ether		# Ethernet support
-device		sl		# Kernel SLIP
-device		ppp		# Kernel PPP
-device		tun		# Packet tunnel.
+#device		sl		# Kernel SLIP
+#device		ppp		# Kernel PPP
+#device		tun		# Packet tunnel.
 device		pty		# Pseudo-ttys (telnet etc)
 device		md		# Memory "disks"
 device		gif		# IPv6 and IPv4 tunneling
@@ -257,8 +257,8 @@
 device		ulpt		# Printer
 device		umass		# Disks/Mass storage - Requires scbus and da
 device		ums		# Mouse
-device		urio		# Diamond Rio 500 MP3 player
-device		uscanner	# Scanners
+#device		urio		# Diamond Rio 500 MP3 player
+#device		uscanner	# Scanners
 # USB Ethernet, requires mii
 device		aue		# ADMtek USB Ethernet
 device		axe		# ASIX Electronics USB Ethernet
