--- sys.old/conf/newvers.sh	2005-07-11 13:18:53.000000000 -0400
+++ sys/conf/newvers.sh	2005-09-02 20:38:11.000000000 -0400
@@ -33,7 +33,7 @@
 TYPE="FreeBSD"
 REVISION="5.4"
 BRANCH="STABLE"
-RELEASE="${REVISION}-${BRANCH}"
+RELEASE="${REVISION}"
 VERSION="${TYPE} ${RELEASE}"
 
 if [ "X${PARAMFILE}" != "X" ]; then
@@ -86,7 +86,7 @@
 cat << EOF > vers.c
 $COPYRIGHT
 char sccs[] =  "@(#)${VERSION} #${v}: ${t}";
-char version[] = "${VERSION} #${v}: ${t}\\n    ${u}@${h}:${d}\\n";
+char version[] = "#${v} ${t}\\n";
 char ostype[] = "${TYPE}";
 char osrelease[] = "${RELEASE}";
 int osreldate = ${RELDATE};
