
Co-authored by Nali Toja

--- a/sys/conf/kern.mk
+++ b/sys/conf/kern.mk
@@ -1,11 +1,18 @@
 # $FreeBSD$
 
+FREEBSD_CC!=   ${CC} -dM -E -</dev/null | awk '/__FreeBSD_cc_version/ { print $$3 }' || true
+
 #
 # Warning flags for compiling the kernel and components of the kernel:
 #
+.if ${FREEBSD_CC}
+# FreeBSD extensions
+format_extensions=     -fformat-extensions
+no_align_long_strings= -mno-align-long-strings
+.endif
 CWARNFLAGS?=	-Wall -Wredundant-decls -Wnested-externs -Wstrict-prototypes \
 		-Wmissing-prototypes -Wpointer-arith -Winline -Wcast-qual \
-		-Wundef -Wno-pointer-sign -fformat-extensions \
+		-Wundef -Wno-pointer-sign ${format_extensions} \
 		-Wmissing-include-dirs -fdiagnostics-show-option \
 		${CWARNEXTRA}
 #
@@ -51,7 +58,7 @@
 #
 .if ${MACHINE_CPUARCH} == "i386"
 .if ${MK_CLANG_IS_CC} == "no" && ${CC:T:Mclang} != "clang"
-CFLAGS+=	-mno-align-long-strings -mpreferred-stack-boundary=2 -mno-sse
+CFLAGS+=	${no_align_long_strings} -mpreferred-stack-boundary=2 -mno-sse
 .else
 CFLAGS+=	-mno-aes -mno-avx
 .endif
