--- a/sys/contrib/ipfilter/netinet/ip_compat.h
+++ b/sys/contrib/ipfilter/netinet/ip_compat.h
@@ -985,10 +985,10 @@
 typedef struct mbuf mb_t;
 # endif /* _KERNEL */
 
-# if __FreeBSD__ < 3
+# if __FreeBSD_version < 300000
 #  include <machine/spl.h>
 # else
-#  if __FreeBSD__ == 3
+#  if __FreeBSD_version < 400000
 #   if defined(IPFILTER_LKM) && !defined(ACTUALLY_LKM_NOT_KERNEL)
 #    define	ACTUALLY_LKM_NOT_KERNEL
 #   endif
--- a/sys/contrib/ipfilter/netinet/ip_proxy.c
+++ b/sys/contrib/ipfilter/netinet/ip_proxy.c
@@ -63,7 +63,7 @@
 # include <sys/stream.h>
 # include <sys/kmem.h>
 #endif
-#if __FreeBSD__ > 2
+#if __FreeBSD_version >= 300000
 # include <sys/queue.h>
 #endif
 #include <net/if.h>
--- a/sys/dev/bktr/bktr_reg.h
+++ b/sys/dev/bktr/bktr_reg.h
@@ -717,13 +717,13 @@
 /* ioctl_cmd_t int on old versions, u_long on new versions */
 /***********************************************************/
 
-#if (__FreeBSD__ == 2)
-typedef int ioctl_cmd_t;
-#endif
-
 #if defined(__FreeBSD__)
+#if __FreeBSD_version >= 200000 && __FreeBSD_version < 300000
+typedef int ioctl_cmd_t;
+#else
 typedef u_long ioctl_cmd_t;
 #endif
+#endif
 
 #if defined(__NetBSD__) || defined(__OpenBSD__)
 typedef u_long ioctl_cmd_t;
--- a/sys/kern/uipc_socket.c
+++ b/sys/kern/uipc_socket.c
@@ -2828,7 +2828,7 @@
 	return (error);
 }
 
-/* XXX; prepare mbuf for (__FreeBSD__ < 3) routines. */
+/* XXX; prepare mbuf for (__FreeBSD_version < 300000) routines. */
 int
 soopt_getm(struct sockopt *sopt, struct mbuf **mp)
 {
@@ -2877,7 +2877,7 @@
 	return (0);
 }
 
-/* XXX; copyin sopt data into mbuf chain for (__FreeBSD__ < 3) routines. */
+/* XXX; copyin sopt data into mbuf chain for (__FreeBSD_version < 300000) routines. */
 int
 soopt_mcopyin(struct sockopt *sopt, struct mbuf *m)
 {
@@ -2906,7 +2906,7 @@
 	return (0);
 }
 
-/* XXX; copyout mbuf chain data into soopt for (__FreeBSD__ < 3) routines. */
+/* XXX; copyout mbuf chain data into soopt for (__FreeBSD_version < 300000) routines. */
 int
 soopt_mcopyout(struct sockopt *sopt, struct mbuf *m)
 {
--- a/sys/net/if_spppfr.c
+++ b/sys/net/if_spppfr.c
@@ -25,7 +25,7 @@
 
 #include <sys/param.h>
 
-#if defined(__FreeBSD__) && __FreeBSD__ >= 3
+#if defined(__FreeBSD__) && __FreeBSD_version >= 300000
 #include "opt_inet.h"
 #include "opt_inet6.h"
 #include "opt_ipx.h"
@@ -45,7 +45,7 @@
 #include <sys/sockio.h>
 #include <sys/socket.h>
 #include <sys/syslog.h>
-#if defined(__FreeBSD__) && __FreeBSD__ >= 3
+#if defined(__FreeBSD__) && __FreeBSD_version >= 300000
 #include <sys/random.h>
 #endif
 #include <sys/malloc.h>
@@ -149,7 +149,7 @@
 	unsigned short  ptarget2;
 } __packed;
 
-#if defined(__FreeBSD__) && __FreeBSD__ >= 3 && __FreeBSD_version < 501113
+#if defined(__FreeBSD__) && __FreeBSD_version >= 300000 && __FreeBSD_version < 501113
 #define	SPP_FMT		"%s%d: "
 #define	SPP_ARGS(ifp)	(ifp)->if_name, (ifp)->if_unit
 #else
