
Fix for kern/159663, simpler version obtained from PC-BSD.

--- a/sys/fs/nullfs/null_subr.c
+++ b/sys/fs/nullfs/null_subr.c
@@ -236,6 +236,8 @@
 	xp->null_vnode = vp;
 	xp->null_lowervp = lowervp;
 	vp->v_type = lowervp->v_type;
+        if (vp->v_type == VSOCK || vp->v_type == VFIFO)
+            vp->v_un = lowervp->v_un;
 	vp->v_data = xp;
 	vp->v_vnlock = lowervp->v_vnlock;
 	if (vp->v_vnlock == NULL)
