--- a/sys/i386/conf/GENERIC
+++ b/sys/i386/conf/GENERIC
@@ -147,8 +147,6 @@
 device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		hptrr		# Highpoint RocketRAID 17xx, 22xx, 23xx, 25xx
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
--- a/sys/modules/Makefile
+++ b/sys/modules/Makefile
@@ -109,10 +109,7 @@
 	hatm \
 	hifn \
 	hme \
-	${_hpt27xx} \
 	${_hptiop} \
-	${_hptmv} \
-	${_hptrr} \
 	hwpmc \
 	${_i2c} \
 	${_ibcs2} \
@@ -231,7 +228,6 @@
 	ntfs \
 	ntfs_iconv \
 	nullfs \
-	${_nve} \
 	${_nvram} \
 	${_nwfs} \
 	${_nxge} \
--- a/sys/amd64/conf/GENERIC
+++ b/sys/amd64/conf/GENERIC
@@ -141,8 +141,6 @@
 #device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		hptrr		# Highpoint RocketRAID 17xx, 22xx, 23xx, 25xx
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
--- a/sys/conf/files.amd64
+++ b/sys/conf/files.amd64
@@ -47,30 +47,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hpt27xx_lib.o			optional	hpt27xx			\
-	dependency	"$S/dev/hpt27xx/amd64-elf.hpt27xx_lib.o.uu"	\
-	compile-with	"uudecode < $S/dev/hpt27xx/amd64-elf.hpt27xx_lib.o.uu" \
-	no-implicit-rule
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/amd64-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/dev/hptmv/amd64-elf.raid.o.uu" \
-	no-implicit-rule
-hptrr_lib.o			optional	hptrr			\
-	dependency	"$S/dev/hptrr/amd64-elf.hptrr_lib.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptrr/amd64-elf.hptrr_lib.o.uu" \
-	no-implicit-rule
-#
 amd64/acpica/acpi_machdep.c	optional	acpi
 amd64/acpica/acpi_switch.S	optional	acpi
 acpi_wakecode.o			optional	acpi			\
@@ -191,17 +167,6 @@
 dev/fdc/fdc_acpi.c		optional	fdc
 dev/fdc/fdc_isa.c		optional	fdc isa
 dev/fdc/fdc_pccard.c		optional	fdc pccard
-dev/hpt27xx/os_bsd.c		optional	hpt27xx
-dev/hpt27xx/osm_bsd.c		optional	hpt27xx
-dev/hpt27xx/hpt27xx_config.c	optional	hpt27xx
-dev/hptmv/entry.c		optional	hptmv
-dev/hptmv/mv.c			optional	hptmv
-dev/hptmv/gui_lib.c		optional	hptmv
-dev/hptmv/hptproc.c		optional	hptmv
-dev/hptmv/ioctl.c		optional	hptmv
-dev/hptrr/hptrr_os_bsd.c	optional	hptrr
-dev/hptrr/hptrr_osm_bsd.c	optional	hptrr
-dev/hptrr/hptrr_config.c	optional	hptrr
 dev/hwpmc/hwpmc_amd.c		optional	hwpmc
 dev/hwpmc/hwpmc_intel.c		optional	hwpmc
 dev/hwpmc/hwpmc_core.c		optional	hwpmc
@@ -213,7 +178,6 @@
 dev/lindev/full.c		optional	lindev
 dev/lindev/lindev.c		optional	lindev
 dev/nfe/if_nfe.c		optional	nfe pci
-dev/nve/if_nve.c		optional	nve pci
 dev/nvram/nvram.c		optional	nvram isa
 dev/random/nehemiah.c		optional	random
 dev/qlxgb/qla_dbg.c		optional	qlxgb pci
--- a/sys/conf/files.i386
+++ b/sys/conf/files.i386
@@ -46,31 +46,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci				\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hpt27xx_lib.o			optional	hpt27xx			\
-	dependency	"$S/dev/hpt27xx/i386-elf.hpt27xx_lib.o.uu"	\
-	compile-with	"uudecode < $S/dev/hpt27xx/i386-elf.hpt27xx_lib.o.uu" \
-	no-implicit-rule
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/i386-elf.raid.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptmv/i386-elf.raid.o.uu"	\
-	no-implicit-rule
-#
-hptrr_lib.o			optional	hptrr			\
-	dependency	"$S/dev/hptrr/i386-elf.hptrr_lib.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptrr/i386-elf.hptrr_lib.o.uu" \
-	no-implicit-rule
-#
 compat/linprocfs/linprocfs.c	optional linprocfs
 compat/linsysfs/linsysfs.c	optional linsysfs
 compat/linux/linux_emul.c	optional compat_linux
@@ -146,18 +121,9 @@
 dev/atkbdc/atkbdc_isa.c		optional atkbdc isa
 dev/atkbdc/atkbdc_subr.c	optional atkbdc
 dev/atkbdc/psm.c		optional psm atkbdc
-dev/ce/ceddk.c			optional ce
-dev/ce/if_ce.c			optional ce
-dev/ce/tau32-ddk.c		optional ce \
-	compile-with "${NORMAL_C} ${NO_WCONSTANT_CONVERSION}"
 dev/cm/if_cm_isa.c		optional cm isa
 dev/coretemp/coretemp.c		optional coretemp
-dev/cp/cpddk.c			optional cp
-dev/cp/if_cp.c			optional cp
 dev/cpuctl/cpuctl.c		optional cpuctl
-dev/ctau/ctau.c			optional ctau
-dev/ctau/ctddk.c		optional ctau
-dev/ctau/if_ct.c		optional ctau
 dev/cx/csigma.c			optional cx
 dev/cx/cxddk.c			optional cx
 dev/cx/if_cx.c			optional cx
@@ -179,17 +145,6 @@
 dev/glxiic/glxiic.c		optional glxiic
 dev/glxsb/glxsb.c		optional glxsb
 dev/glxsb/glxsb_hash.c		optional glxsb
-dev/hpt27xx/os_bsd.c		optional hpt27xx
-dev/hpt27xx/osm_bsd.c		optional hpt27xx
-dev/hpt27xx/hpt27xx_config.c	optional hpt27xx
-dev/hptmv/entry.c		optional hptmv
-dev/hptmv/mv.c			optional hptmv
-dev/hptmv/gui_lib.c		optional hptmv
-dev/hptmv/hptproc.c		optional hptmv
-dev/hptmv/ioctl.c		optional hptmv
-dev/hptrr/hptrr_os_bsd.c	optional hptrr
-dev/hptrr/hptrr_osm_bsd.c	optional hptrr
-dev/hptrr/hptrr_config.c	optional hptrr
 dev/hwpmc/hwpmc_amd.c		optional hwpmc
 dev/hwpmc/hwpmc_intel.c		optional hwpmc
 dev/hwpmc/hwpmc_core.c		optional hwpmc
@@ -222,7 +177,6 @@
 dev/mse/mse.c			optional mse
 dev/mse/mse_isa.c		optional mse isa
 dev/nfe/if_nfe.c		optional nfe pci
-dev/nve/if_nve.c		optional nve pci
 dev/nvram/nvram.c		optional nvram isa
 dev/pcf/pcf_isa.c		optional pcf
 dev/random/nehemiah.c		optional random
