
Status: Rejected in upstream (kern/77355).

--- a/sys/i386/i386/identcpu.c
+++ b/sys/i386/i386/identcpu.c
@@ -647,6 +647,7 @@
 #if defined(I486_CPU)
 	case CPUCLASS_486:
 		printf("486");
+		machine[1] = '4';
 		break;
 #endif
 #if defined(I586_CPU)
@@ -658,6 +659,7 @@
 			       (u_int)((tsc_freq + 4999) / 10000) % 100);
 		}
 		printf("586");
+		machine[1] = '5';
 		break;
 #endif
 #if defined(I686_CPU)
@@ -669,6 +671,7 @@
 			       (u_int)((tsc_freq + 4999) / 10000) % 100);
 		}
 		printf("686");
+		machine[1] = '6';
 		break;
 #endif
 	default:
--- a/sys/amd64/amd64/identcpu.c
+++ b/sys/amd64/amd64/identcpu.c
@@ -85,7 +85,7 @@
 sysctl_hw_machine(SYSCTL_HANDLER_ARGS)
 {
 #ifdef SCTL_MASK32
-	static const char machine32[] = "i386";
+	static const char machine32[] = "i686";
 #endif
 	int error;
 
