
freebsd-buildutils 9.0 doesn't define MK_CTF.

--- a/sys/conf/kern.post.mk
+++ b/sys/conf/kern.post.mk
@@ -115,7 +115,7 @@
 	@rm -f ${.TARGET}
 	@echo linking ${.TARGET}
 	${SYSTEM_LD}
-.if ${MK_CTF} != "no"
+.if 0
 	${CTFMERGE} ${CTFFLAGS} -o ${.TARGET} ${SYSTEM_OBJS} vers.o
 .endif
 .if !defined(DEBUG)
--- a/sys/conf/kern.pre.mk
+++ b/sys/conf/kern.pre.mk
@@ -142,7 +142,7 @@
 ZFS_C=		${CC} -c ${ZFS_CFLAGS} ${WERROR} ${PROF} ${.IMPSRC}
 ZFS_S=		${CC} -c ${ZFS_ASM_CFLAGS} ${WERROR} ${.IMPSRC}
 
-.if ${MK_CTF} != "no"
+.if 0
 NORMAL_CTFCONVERT=	${CTFCONVERT} ${CTFFLAGS} ${.TARGET}
 .elif ${MAKE_VERSION} >= 5201111300
 NORMAL_CTFCONVERT=
