
See #594940  

--- a/sys/conf/files
+++ b/sys/conf/files
@@ -488,10 +488,6 @@
 dev/advansys/advansys.c		optional adv
 dev/advansys/advlib.c		optional adv
 dev/advansys/advmcode.c		optional adv
-dev/advansys/adw_pci.c		optional adw pci
-dev/advansys/adwcam.c		optional adw
-dev/advansys/adwlib.c		optional adw
-dev/advansys/adwmcode.c		optional adw
 dev/ae/if_ae.c			optional ae pci
 dev/age/if_age.c		optional age pci
 dev/agp/agp.c			optional agp pci
@@ -850,7 +846,6 @@
 dev/ath/ath_dfs/null/dfs_null.c	optional	ath \
 	compile-with "${NORMAL_C} -I$S/dev/ath"
 # 
-dev/bce/if_bce.c		optional bce
 dev/bfe/if_bfe.c		optional bfe
 dev/bge/if_bge.c		optional bge
 dev/bktr/bktr_audio.c		optional bktr pci
@@ -888,34 +883,6 @@
 dev/cs/if_cs.c			optional cs
 dev/cs/if_cs_isa.c		optional cs isa
 dev/cs/if_cs_pccard.c		optional cs pccard
-dev/cxgb/cxgb_main.c		optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/cxgb_offload.c		optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/cxgb_sge.c		optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_mc5.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_vsc7323.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_vsc8211.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_ael1002.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_aq100x.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_mv88e1xxx.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_xgmac.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_t3_hw.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/common/cxgb_tn1010.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"	
-dev/cxgb/sys/uipc_mvec.c	optional cxgb pci \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
-dev/cxgb/cxgb_t3fw.c		optional cxgb cxgb_t3fw \
-	compile-with "${NORMAL_C} -I$S/dev/cxgb"
 dev/cxgbe/t4_main.c		optional cxgbe pci \
 	compile-with "${NORMAL_C} -I$S/dev/cxgbe"
 dev/cxgbe/t4_sge.c		optional cxgbe pci \
@@ -934,13 +901,6 @@
 dev/dcons/dcons_crom.c		optional dcons_crom
 dev/dcons/dcons_os.c		optional dcons
 dev/de/if_de.c			optional de pci
-dev/digi/CX.c			optional digi_CX
-dev/digi/CX_PCI.c		optional digi_CX_PCI
-dev/digi/EPCX.c			optional digi_EPCX
-dev/digi/EPCX_PCI.c		optional digi_EPCX_PCI
-dev/digi/Xe.c			optional digi_Xe
-dev/digi/Xem.c			optional digi_Xem
-dev/digi/Xr.c			optional digi_Xr
 dev/digi/digi.c			optional digi
 dev/digi/digi_isa.c		optional digi isa
 dev/digi/digi_pci.c		optional digi pci
@@ -976,26 +936,6 @@
 dev/drm/mach64_drv.c		optional mach64drm
 dev/drm/mach64_irq.c		optional mach64drm
 dev/drm/mach64_state.c		optional mach64drm
-dev/drm/mga_dma.c		optional mgadrm
-dev/drm/mga_drv.c		optional mgadrm
-dev/drm/mga_irq.c		optional mgadrm
-dev/drm/mga_state.c		optional mgadrm \
-	compile-with "${NORMAL_C} -finline-limit=13500"
-dev/drm/mga_warp.c		optional mgadrm
-dev/drm/r128_cce.c		optional r128drm
-dev/drm/r128_drv.c		optional r128drm
-dev/drm/r128_irq.c		optional r128drm
-dev/drm/r128_state.c		optional r128drm \
-	compile-with "${NORMAL_C} -finline-limit=13500"
-dev/drm/r300_cmdbuf.c		optional radeondrm
-dev/drm/r600_blit.c		optional radeondrm
-dev/drm/r600_cp.c		optional radeondrm
-dev/drm/radeon_cp.c		optional radeondrm
-dev/drm/radeon_cs.c		optional radeondrm
-dev/drm/radeon_drv.c		optional radeondrm
-dev/drm/radeon_irq.c		optional radeondrm
-dev/drm/radeon_mem.c		optional radeondrm
-dev/drm/radeon_state.c		optional radeondrm
 dev/drm/savage_bci.c		optional savagedrm
 dev/drm/savage_drv.c		optional savagedrm
 dev/drm/savage_state.c		optional savagedrm
@@ -1091,8 +1031,6 @@
 dev/firewire/sbp_targ.c		optional sbp_targ
 dev/flash/at45d.c		optional at45d
 dev/flash/mx25l.c		optional mx25l
-dev/fxp/if_fxp.c		optional fxp
-dev/fxp/inphy.c			optional fxp
 dev/gem/if_gem.c		optional gem
 dev/gem/if_gem_pci.c		optional gem pci
 dev/gem/if_gem_sbus.c		optional gem sbus
@@ -1149,49 +1087,6 @@
 dev/ips/ips_disk.c		optional ips
 dev/ips/ips_ioctl.c		optional ips
 dev/ips/ips_pci.c		optional ips pci
-dev/ipw/if_ipw.c		optional ipw
-ipwbssfw.c			optional ipwbssfw | ipwfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk ipw_bss.fw:ipw_bss:130 -lintel_ipw -mipw_bss -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"ipwbssfw.c"
-ipw_bss.fwo			optional ipwbssfw | ipwfw		\
-	dependency	"ipw_bss.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} ipw_bss.fw" \
-	no-implicit-rule						\
-	clean		"ipw_bss.fwo"
-ipw_bss.fw			optional ipwbssfw | ipwfw		\
-	dependency	"$S/contrib/dev/ipw/ipw2100-1.3.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/ipw/ipw2100-1.3.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"ipw_bss.fw"
-ipwibssfw.c			optional ipwibssfw | ipwfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk ipw_ibss.fw:ipw_ibss:130 -lintel_ipw -mipw_ibss -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"ipwibssfw.c"
-ipw_ibss.fwo			optional ipwibssfw | ipwfw		\
-	dependency	"ipw_ibss.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} ipw_ibss.fw" \
-	no-implicit-rule						\
-	clean		"ipw_ibss.fwo"
-ipw_ibss.fw			optional ipwibssfw | ipwfw		\
-	dependency	"$S/contrib/dev/ipw/ipw2100-1.3-i.fw.uu"	\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/ipw/ipw2100-1.3-i.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"ipw_ibss.fw"
-ipwmonitorfw.c			optional ipwmonitorfw | ipwfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk ipw_monitor.fw:ipw_monitor:130 -lintel_ipw -mipw_monitor -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"ipwmonitorfw.c"
-ipw_monitor.fwo			optional ipwmonitorfw | ipwfw		\
-	dependency	"ipw_monitor.fw"				\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} ipw_monitor.fw" \
-	no-implicit-rule						\
-	clean		"ipw_monitor.fwo"
-ipw_monitor.fw			optional ipwmonitorfw | ipwfw		\
-	dependency	"$S/contrib/dev/ipw/ipw2100-1.3-p.fw.uu"	\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/ipw/ipw2100-1.3-p.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"ipw_monitor.fw"
 dev/iscsi/initiator/iscsi.c	optional iscsi_initiator scbus
 dev/iscsi/initiator/iscsi_subr.c	optional iscsi_initiator scbus
 dev/iscsi/initiator/isc_cam.c	optional iscsi_initiator scbus
@@ -1204,163 +1099,6 @@
 dev/isp/isp_pci.c		optional isp pci
 dev/isp/isp_sbus.c		optional isp sbus
 dev/isp/isp_target.c		optional isp
-dev/ispfw/ispfw.c		optional ispfw
-dev/iwi/if_iwi.c		optional iwi
-iwibssfw.c			optional iwibssfw | iwifw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwi_bss.fw:iwi_bss:300 -lintel_iwi -miwi_bss -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwibssfw.c"
-iwi_bss.fwo			optional iwibssfw | iwifw		\
-	dependency	"iwi_bss.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwi_bss.fw" \
-	no-implicit-rule						\
-	clean		"iwi_bss.fwo"
-iwi_bss.fw			optional iwibssfw | iwifw		\
-	dependency	"$S/contrib/dev/iwi/ipw2200-bss.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwi/ipw2200-bss.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwi_bss.fw"
-iwiibssfw.c			optional iwiibssfw | iwifw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwi_ibss.fw:iwi_ibss:300 -lintel_iwi -miwi_ibss -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwiibssfw.c"
-iwi_ibss.fwo			optional iwiibssfw | iwifw		\
-	dependency	"iwi_ibss.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwi_ibss.fw" \
-	no-implicit-rule						\
-	clean		"iwi_ibss.fwo"
-iwi_ibss.fw			optional iwiibssfw | iwifw		\
-	dependency	"$S/contrib/dev/iwi/ipw2200-ibss.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwi/ipw2200-ibss.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwi_ibss.fw"
-iwimonitorfw.c			optional iwimonitorfw | iwifw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwi_monitor.fw:iwi_monitor:300 -lintel_iwi -miwi_monitor -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwimonitorfw.c"
-iwi_monitor.fwo			optional iwimonitorfw | iwifw		\
-	dependency	"iwi_monitor.fw"				\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwi_monitor.fw" \
-	no-implicit-rule						\
-	clean		"iwi_monitor.fwo"
-iwi_monitor.fw			optional iwimonitorfw | iwifw		\
-	dependency	"$S/contrib/dev/iwi/ipw2200-sniffer.fw.uu"	\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwi/ipw2200-sniffer.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwi_monitor.fw"
-dev/iwn/if_iwn.c		optional iwn
-iwn1000fw.c			optional iwn1000fw | iwnfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwn1000.fw:iwn1000fw -miwn1000fw -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwn1000fw.c"
-iwn1000fw.fwo			optional iwn1000fw | iwnfw		\
-	dependency	"iwn1000.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwn1000.fw" \
-	no-implicit-rule						\
-	clean		"iwn1000fw.fwo"
-iwn1000.fw			optional iwn1000fw | iwnfw		\
-	dependency	"$S/contrib/dev/iwn/iwlwifi-1000-39.31.5.1.fw.uu" \
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwn/iwlwifi-1000-39.31.5.1.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwn1000.fw"
-iwn4965fw.c			optional iwn4965fw | iwnfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwn4965.fw:iwn4965fw -miwn4965fw -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwn4965fw.c"
-iwn4965fw.fwo			optional iwn4965fw | iwnfw		\
-	dependency	"iwn4965.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwn4965.fw" \
-	no-implicit-rule						\
-	clean		"iwn4965fw.fwo"
-iwn4965.fw			optional iwn4965fw | iwnfw		\
-	dependency	"$S/contrib/dev/iwn/iwlwifi-4965-228.61.2.24.fw.uu" \
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwn/iwlwifi-4965-228.61.2.24.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwn4965.fw"
-iwn5000fw.c			optional iwn5000fw | iwnfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwn5000.fw:iwn5000fw -miwn5000fw -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwn5000fw.c"
-iwn5000fw.fwo		optional iwn5000fw | iwnfw			\
-	dependency	"iwn5000.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwn5000.fw" \
-	no-implicit-rule						\
-	clean		"iwn5000fw.fwo"
-iwn5000.fw			optional iwn5000fw | iwnfw		\
-	dependency	"$S/contrib/dev/iwn/iwlwifi-5000-8.83.5.1.fw.uu" \
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwn/iwlwifi-5000-8.83.5.1.fw.uu"	\
-	no-obj no-implicit-rule						\
-	clean		"iwn5000.fw"
-iwn5150fw.c			optional iwn5150fw | iwnfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwn5150.fw:iwn5150fw -miwn5150fw -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwn5150fw.c"
-iwn5150fw.fwo			optional iwn5150fw | iwnfw		\
-	dependency	"iwn5150.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwn5150.fw" \
-	no-implicit-rule						\
-	clean		"iwn5150fw.fwo"
-iwn5150.fw			optional iwn5150fw | iwnfw		\
-	dependency	"$S/contrib/dev/iwn/iwlwifi-5150-8.24.2.2.fw.uu"\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwn/iwlwifi-5150-8.24.2.2.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwn5150.fw"
-iwn6000fw.c			optional iwn6000fw | iwnfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwn6000.fw:iwn6000fw -miwn6000fw -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwn6000fw.c"
-iwn6000fw.fwo			optional iwn6000fw | iwnfw		\
-	dependency	"iwn6000.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwn6000.fw" \
-	no-implicit-rule						\
-	clean		"iwn6000fw.fwo"
-iwn6000.fw			optional iwn6000fw | iwnfw		\
-	dependency	"$S/contrib/dev/iwn/iwlwifi-6000-9.221.4.1.fw.uu" \
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwn/iwlwifi-6000-9.221.4.1.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwn6000.fw"
-iwn6000g2afw.c			optional iwn6000g2afw | iwnfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwn6000g2a.fw:iwn6000g2afw -miwn6000g2afw -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwn6000g2afw.c"
-iwn6000g2afw.fwo		optional iwn6000g2afw | iwnfw		\
-	dependency	"iwn6000g2a.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwn6000g2a.fw" \
-	no-implicit-rule						\
-	clean		"iwn6000g2afw.fwo"
-iwn6000g2a.fw			optional iwn6000g2afw | iwnfw		\
-	dependency	"$S/contrib/dev/iwn/iwlwifi-6000g2a-17.168.5.2.fw.uu" \
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwn/iwlwifi-6000g2a-17.168.5.2.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwn6000g2a.fw"
-iwn6000g2bfw.c			optional iwn6000g2bfw | iwnfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwn6000g2b.fw:iwn6000g2bfw -miwn6000g2bfw -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwn6000g2bfw.c"
-iwn6000g2bfw.fwo		optional iwn6000g2bfw | iwnfw		\
-	dependency	"iwn6000g2b.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwn6000g2b.fw" \
-	no-implicit-rule						\
-	clean		"iwn6000g2bfw.fwo"
-iwn6000g2b.fw			optional iwn6000g2bfw | iwnfw		\
-	dependency	"$S/contrib/dev/iwn/iwlwifi-6000g2b-17.168.5.2.fw.uu" \
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwn/iwlwifi-6000g2b-17.168.5.2.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwn6000g2b.fw"
-iwn6050fw.c			optional iwn6050fw | iwnfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk iwn6050.fw:iwn6050fw -miwn6050fw -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"iwn6050fw.c"
-iwn6050fw.fwo			optional iwn6050fw | iwnfw		\
-	dependency	"iwn6050.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} iwn6050.fw" \
-	no-implicit-rule						\
-	clean		"iwn6050fw.fwo"
-iwn6050.fw			optional iwn6050fw | iwnfw		\
-	dependency	"$S/contrib/dev/iwn/iwlwifi-6050-41.28.5.1.fw.uu" \
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/iwn/iwlwifi-6050-41.28.5.1.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"iwn6050.fw"
 dev/ixgb/if_ixgb.c		optional ixgb
 dev/ixgb/ixgb_ee.c		optional ixgb
 dev/ixgb/ixgb_hw.c		optional ixgb
@@ -1469,33 +1207,6 @@
 dev/mvs/mvs.c			optional mvs
 dev/mvs/mvs_if.m		optional mvs
 dev/mvs/mvs_pci.c		optional mvs pci
-dev/mwl/if_mwl.c		optional mwl
-dev/mwl/if_mwl_pci.c		optional mwl pci
-dev/mwl/mwlhal.c		optional mwl
-mwlfw.c				optional mwlfw				\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk mw88W8363.fw:mw88W8363fw mwlboot.fw:mwlboot -mmwl -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"mwlfw.c"
-mw88W8363.fwo		optional mwlfw					\
-	dependency	"mw88W8363.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} mw88W8363.fw" \
-	no-implicit-rule						\
-	clean		"mw88W8363.fwo"
-mw88W8363.fw		optional mwlfw					\
-	dependency	"$S/contrib/dev/mwl/mw88W8363.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/mwl/mw88W8363.fw.uu"	\
-	no-obj no-implicit-rule						\
-	clean		"mw88W8363.fw"
-mwlboot.fwo		optional mwlfw					\
-	dependency	"mwlboot.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} mwlboot.fw" \
-	no-implicit-rule						\
-	clean		"mwlboot.fwo"
-mwlboot.fw		optional mwlfw					\
-	dependency	"$S/contrib/dev/mwl/mwlboot.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/mwl/mwlboot.fw.uu"	\
-	no-obj no-implicit-rule						\
-	clean		"mwlboot.fw"
 dev/mxge/if_mxge.c		optional mxge pci
 dev/mxge/mxge_lro.c		optional mxge pci
 dev/mxge/mxge_eth_z8e.c		optional mxge pci
@@ -1585,65 +1296,6 @@
 dev/puc/puc_pci.c		optional puc pci
 dev/puc/pucdata.c		optional puc pci
 dev/quicc/quicc_core.c		optional quicc
-dev/ral/rt2560.c		optional ral
-dev/ral/rt2661.c		optional ral
-dev/ral/if_ral_pci.c		optional ral pci
-rt2561fw.c			optional rt2561fw | ralfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk rt2561.fw:rt2561fw -mrt2561 -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"rt2561fw.c"
-rt2561fw.fwo			optional rt2561fw | ralfw		\
-	dependency	"rt2561.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} rt2561.fw" \
-	no-implicit-rule						\
-	clean		"rt2561fw.fwo"
-rt2561.fw			optional rt2561fw | ralfw		\
-	dependency	"$S/contrib/dev/ral/rt2561.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/ral/rt2561.fw.uu" \
-	no-obj no-implicit-rule						\
-	clean		"rt2561.fw"
-rt2561sfw.c			optional rt2561sfw | ralfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk rt2561s.fw:rt2561sfw -mrt2561s -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"rt2561sfw.c"
-rt2561sfw.fwo			optional rt2561sfw | ralfw		\
-	dependency	"rt2561s.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} rt2561s.fw" \
-	no-implicit-rule						\
-	clean		"rt2561sfw.fwo"
-rt2561s.fw			optional rt2561sfw | ralfw		\
-	dependency	"$S/contrib/dev/ral/rt2561s.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/ral/rt2561s.fw.uu"	\
-	no-obj no-implicit-rule						\
-	clean		"rt2561s.fw"
-rt2661fw.c			optional rt2661fw | ralfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk rt2661.fw:rt2661fw -mrt2661 -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"rt2661fw.c"
-rt2661fw.fwo			optional rt2661fw | ralfw		\
-	dependency	"rt2661.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} rt2661.fw" \
-	no-implicit-rule						\
-	clean		"rt2661fw.fwo"
-rt2661.fw			optional rt2661fw | ralfw		\
-	dependency	"$S/contrib/dev/ral/rt2661.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/ral/rt2661.fw.uu"	\
-	no-obj no-implicit-rule						\
-	clean		"rt2661.fw"
-rt2860fw.c			optional rt2860fw | ralfw		\
-	compile-with	"${AWK} -f $S/tools/fw_stub.awk rt2860.fw:rt2860fw -mrt2860 -c${.TARGET}" \
-	no-implicit-rule before-depend local				\
-	clean		"rt2860fw.c"
-rt2860fw.fwo			optional rt2860fw | ralfw		\
-	dependency	"rt2860.fw"					\
-	compile-with	"${LD} -b binary -d -warn-common -r -d -o ${.TARGET} rt2860.fw" \
-	no-implicit-rule						\
-	clean		"rt2860fw.fwo"
-rt2860.fw			optional rt2860fw | ralfw		\
-	dependency	"$S/contrib/dev/ral/rt2860.fw.uu"		\
-	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/ral/rt2860.fw.uu"	\
-	no-obj no-implicit-rule						\
-	clean		"rt2860.fw"
 dev/random/harvest.c		standard
 dev/random/hash.c		optional random
 dev/random/probe.c		optional random
@@ -1668,14 +1320,7 @@
 dev/scd/scd.c			optional scd isa
 dev/scd/scd_isa.c		optional scd isa
 dev/sdhci/sdhci.c		optional sdhci pci
-dev/sf/if_sf.c			optional sf pci
 dev/sge/if_sge.c		optional sge pci
-dev/si/si.c			optional si
-dev/si/si2_z280.c		optional si
-dev/si/si3_t225.c		optional si
-dev/si/si_eisa.c		optional si eisa
-dev/si/si_isa.c			optional si isa
-dev/si/si_pci.c			optional si pci
 dev/siba/siba_bwn.c		optional siba_bwn pci
 dev/siba/siba_core.c		optional siba_bwn pci
 dev/siis/siis.c			optional siis pci
@@ -1686,9 +1331,6 @@
 dev/smbus/smbus.c		optional smbus
 dev/smbus/smbus_if.m		optional smbus
 dev/smc/if_smc.c		optional smc
-dev/sn/if_sn.c			optional sn
-dev/sn/if_sn_isa.c		optional sn isa
-dev/sn/if_sn_pccard.c		optional sn pccard
 dev/snp/snp.c			optional snp
 dev/sound/clone.c		optional sound
 dev/sound/unit.c		optional sound
@@ -1704,10 +1346,6 @@
 dev/sound/pci/atiixp.c		optional snd_atiixp pci
 dev/sound/pci/cmi.c		optional snd_cmi pci
 dev/sound/pci/cs4281.c		optional snd_cs4281 pci
-dev/sound/pci/csa.c		optional snd_csa pci \
-	warning "kernel contains GPL contaminated csaimg.h header"
-dev/sound/pci/csapcm.c		optional snd_csa pci
-dev/sound/pci/ds1.c		optional snd_ds1 pci
 dev/sound/pci/emu10k1.c		optional snd_emu10k1 pci \
 	dependency "emu10k1-alsa%diked.h" \
 	warning "kernel contains GPL contaminated emu10k1 headers"
@@ -1730,8 +1368,6 @@
 dev/sound/pci/fm801.c		optional snd_fm801 pci
 dev/sound/pci/ich.c		optional snd_ich pci
 dev/sound/pci/maestro.c		optional snd_maestro pci
-dev/sound/pci/maestro3.c	optional snd_maestro3 pci \
-	warning "kernel contains GPL contaminated maestro3 headers"
 dev/sound/pci/neomagic.c	optional snd_neomagic pci
 dev/sound/pci/solo.c		optional snd_solo pci
 dev/sound/pci/spicds.c		optional snd_spicds pci
@@ -1811,7 +1447,6 @@
 dev/syscons/warp/warp_saver.c	optional warp_saver
 dev/tdfx/tdfx_linux.c		optional tdfx_linux tdfx compat_linux
 dev/tdfx/tdfx_pci.c		optional tdfx pci
-dev/ti/if_ti.c			optional ti pci
 dev/tl/if_tl.c			optional tl pci
 dev/trm/trm.c			optional trm
 dev/twa/tw_cl_init.c		optional twa \
@@ -1834,7 +1469,6 @@
 dev/tws/tws_services.c		optional tws
 dev/tws/tws_user.c		optional tws
 dev/tx/if_tx.c			optional tx
-dev/txp/if_txp.c		optional txp
 dev/uart/uart_bus_acpi.c	optional uart acpi
 #dev/uart/uart_bus_cbus.c	optional uart cbus
 dev/uart/uart_bus_ebus.c	optional uart ebus
@@ -1914,19 +1548,17 @@
 dev/usb/net/if_cdce.c		optional cdce
 dev/usb/net/if_cue.c		optional cue
 dev/usb/net/if_ipheth.c		optional ipheth
-dev/usb/net/if_kue.c		optional kue
 dev/usb/net/if_mos.c		optional mos
 dev/usb/net/if_rue.c		optional rue
 dev/usb/net/if_udav.c		optional udav
 dev/usb/net/if_usie.c		optional usie
 dev/usb/net/ruephy.c		optional rue
-dev/usb/net/usb_ethernet.c	optional aue | axe | cdce | cue | kue | mos | \
+dev/usb/net/usb_ethernet.c	optional aue | axe | cdce | cue | mos | \
 					 rue | udav
 dev/usb/net/uhso.c		optional uhso
 #
 # USB WLAN drivers
 #
-dev/usb/wlan/if_rum.c		optional rum
 dev/usb/wlan/if_run.c		optional run
 runfw.c				optional runfw							\
 	compile-with	"${AWK} -f $S/tools/fw_stub.awk runfw:runfw -mrunfw -c${.TARGET}"	\
@@ -1942,11 +1574,9 @@
 	compile-with	"uudecode -o ${.TARGET} $S/contrib/dev/run/rt2870.fw.uu"		\
 	no-obj no-implicit-rule									\
 	clean		"runfw"
-dev/usb/wlan/if_uath.c		optional uath
 dev/usb/wlan/if_upgt.c		optional upgt
 dev/usb/wlan/if_ural.c		optional ural
 dev/usb/wlan/if_urtw.c		optional urtw
-dev/usb/wlan/if_zyd.c		optional zyd
 #
 # USB serial and parallel port drivers
 #
--- a/sys/conf/files.i386
+++ b/sys/conf/files.i386
@@ -121,17 +121,9 @@
 dev/atkbdc/atkbdc_isa.c		optional atkbdc isa
 dev/atkbdc/atkbdc_subr.c	optional atkbdc
 dev/atkbdc/psm.c		optional psm atkbdc
-dev/ce/ceddk.c			optional ce
-dev/ce/if_ce.c			optional ce
-dev/ce/tau32-ddk.c		optional ce
 dev/cm/if_cm_isa.c		optional cm isa
 dev/coretemp/coretemp.c		optional coretemp
-dev/cp/cpddk.c			optional cp
-dev/cp/if_cp.c			optional cp
 dev/cpuctl/cpuctl.c		optional cpuctl
-dev/ctau/ctau.c			optional ctau
-dev/ctau/ctddk.c		optional ctau
-dev/ctau/if_ct.c		optional ctau
 dev/cx/csigma.c			optional cx
 dev/cx/cxddk.c			optional cx
 dev/cx/if_cx.c			optional cx
--- a/sys/modules/drm/Makefile
+++ b/sys/modules/drm/Makefile
@@ -4,9 +4,6 @@
 	drm \
 	i915 \
 	mach64 \
-	mga \
-	r128 \
-	radeon \
 	savage \
 	sis \
 	tdfx \
--- a/sys/modules/Makefile
+++ b/sys/modules/Makefile
@@ -37,7 +37,6 @@
 	ath \
 	ath_pci \
 	${_auxio} \
-	bce \
 	bfe \
 	bge \
 	${_bxe} \
@@ -56,7 +55,6 @@
 	cc \
 	cd9660 \
 	cd9660_iconv \
-	${_ce} \
 	${_ciss} \
 	${_cm} \
 	${_cmx} \
@@ -64,14 +62,11 @@
 	coda5 \
 	${_coff} \
 	${_coretemp} \
-	${_cp} \
 	${_cpuctl} \
 	${_cpufreq} \
 	${_crypto} \
 	${_cryptodev} \
 	${_cs} \
-	${_ctau} \
-	${_cxgb} \
 	cxgbe \
 	${_cyclic} \
 	dc \
@@ -93,13 +88,11 @@
 	${_ex} \
 	${_exca} \
 	${_ext2fs} \
-	fatm \
 	fdc \
 	fdescfs \
 	${_fe} \
 	firewire \
 	firmware \
-	fxp \
 	gem \
 	geom \
 	${_glxiic} \
@@ -140,14 +133,10 @@
 	ip_mroute_mod \
 	${_ips} \
 	${_ipw} \
-	${_ipwfw} \
 	iscsi \
 	isp \
-	ispfw \
 	${_iwi} \
-	${_iwifw} \
 	${_iwn} \
-	${_iwnfw} \
 	${_ixgb} \
 	${_ixgbe} \
 	jme \
@@ -202,7 +191,6 @@
 	${_mthca} \
 	mvs \
 	mwl \
-	mwlfw \
 	mxge \
 	my \
 	${_ncp} \
@@ -250,14 +238,12 @@
 	pty  \
 	puc \
 	ral \
-	ralfw \
 	${_random} \
 	rc4 \
 	${_rdma} \
 	re \
 	reiserfs \
 	rl \
-	runfw \
 	${_s3} \
 	${_safe} \
 	${_sbni} \
@@ -267,7 +253,6 @@
 	sdhci \
 	sem \
 	send \
-	sf \
 	sge \
 	siba_bwn \
 	siftr \
@@ -275,7 +260,6 @@
 	sis \
 	sk \
 	${_smbfs} \
-	sn \
 	${_snc} \
 	snp \
 	${_sound} \
@@ -290,7 +274,6 @@
 	${_sym} \
 	${_syscons} \
 	sysvipc \
-	ti \
 	tl \
 	tmpfs \
 	${_tpm} \
@@ -299,7 +282,6 @@
 	twe \
 	tws \
 	tx \
-	txp \
 	uart \
 	ubsec \
 	udf \
@@ -327,7 +309,6 @@
 	wlan_wep \
 	wlan_xauth \
 	${_wpi} \
-	${_wpifw} \
 	${_x86bios} \
 	${_xe} \
 	xfs \
@@ -399,9 +380,7 @@
 _bxe=		bxe
 _cardbus=	cardbus
 _cbb=		cbb
-_ce=		ce
 _coff=		coff
-_cp=		cp
 _cpuctl=	cpuctl
 _cpufreq=	cpufreq
 _cs=		cs
@@ -500,11 +479,8 @@
 _ipmi=		ipmi
 _ips=		ips
 _ipw=		ipw
-_ipwfw=		ipwfw
 _iwi=		iwi
-_iwifw=		iwifw
 _iwn=		iwn
-_iwnfw=		iwnfw
 _ixgb=		ixgb
 _ixgbe=		ixgbe
 _mly=		mly
@@ -513,7 +489,6 @@
 _nxge=		nxge
 _tpm=		tpm
 _wpi=		wpi
-_wpifw=		wpifw
 .if ${MK_CRYPT} != "no" || defined(ALL_MODULES)
 _padlock=	padlock
 .endif
@@ -574,11 +549,8 @@
 _ipmi=		ipmi
 _ips=		ips
 _ipw=		ipw
-_ipwfw=		ipwfw
 _iwi=		iwi
-_iwifw=		iwifw
 _iwn=		iwn
-_iwnfw=		iwnfw
 _ixgb=		ixgb
 _ixgbe=		ixgbe
 _lindev=	lindev
@@ -619,7 +591,6 @@
 _x86bios=	x86bios
 _wi=		wi
 _wpi=		wpi
-_wpifw=		wpifw
 .if ${MK_ZFS} != "no" || defined(ALL_MODULES)
 _zfs=		zfs
 .endif
--- a/sys/modules/sound/driver/Makefile
+++ b/sys/modules/sound/driver/Makefile
@@ -1,7 +1,7 @@
 # $FreeBSD$
 
-SUBDIR=		ad1816 als4000 atiixp cs4281 csa ds1 emu10k1 emu10kx
-SUBDIR+=	envy24 envy24ht es137x ess fm801 hda ich maestro maestro3
+SUBDIR=		ad1816 als4000 atiixp cs4281 emu10k1 emu10kx
+SUBDIR+=	envy24 envy24ht es137x ess fm801 hda ich maestro
 SUBDIR+=	neomagic sb16 sb8 sbc solo spicds t4dwave via8233
 SUBDIR+=	via82c686 vibes driver uaudio
 
--- a/sys/modules/usb/Makefile
+++ b/sys/modules/usb/Makefile
@@ -27,11 +27,11 @@
 
 SUBDIR = usb
 SUBDIR += ehci musb ohci uhci xhci uss820dci ${_at91dci} ${_atmegadci}
-SUBDIR += rum run uath upgt usie ural zyd ${_urtw}
+SUBDIR += run upgt usie ural ${_urtw}
 SUBDIR += atp uhid ukbd ums udbp ufm uep
 SUBDIR += ucom u3g uark ubsa ubser uchcom ucycom ufoma uftdi ugensa uipaq ulpt \
 	  umct umcs umodem umoscom uplcom uslcom uvisor uvscom
-SUBDIR += uether aue axe cdce cue kue mos rue udav uhso ipheth
+SUBDIR += uether aue axe cdce cue mos rue udav uhso ipheth
 SUBDIR += usfs umass urio
 SUBDIR += quirk template
 
