--- a/sys/amd64/conf/GENERIC
+++ b/sys/amd64/conf/GENERIC
@@ -18,10 +18,12 @@
 #
 # $FreeBSD$
 
-cpu		HAMMER
-ident		GENERIC
+#cpu		HAMMER
+#ident		GENERIC
 
-makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
+hints          "GENERIC.hints"         # Default places to look for devices.
+
+#makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
 
 options 	SCHED_ULE		# ULE scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
@@ -49,7 +51,7 @@
 options 	COMPAT_FREEBSD5		# Compatible with FreeBSD5
 options 	COMPAT_FREEBSD6		# Compatible with FreeBSD6
 options 	COMPAT_FREEBSD7		# Compatible with FreeBSD7
-options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
+options 	SCSI_DELAY=2000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	STACK			# stack(9) support
 options 	SYSVSHM			# SYSV-style shared memory
@@ -78,7 +80,7 @@
 options 	KDB_TRACE		# Print a stack trace for a panic.
 
 # Make an SMP-capable kernel by default
-options 	SMP			# Symmetric MultiProcessor Kernel
+#options 	SMP			# Symmetric MultiProcessor Kernel
 
 # CPU frequency control
 device		cpufreq
@@ -87,9 +89,6 @@
 device		acpi
 device		pci
 
-# Floppy drives
-device		fdc
-
 # ATA controllers
 device		ahci		# AHCI-compatible SATA controllers
 device		ata		# Legacy ATA/SATA controllers
@@ -185,10 +184,6 @@
 # Parallel port
 device		ppc
 device		ppbus		# Parallel port bus (required)
-device		lpt		# Printer
-device		plip		# TCP/IP over parallel
-device		ppi		# Parallel port interface device
-#device		vpo		# Requires scbus and da
 
 device		puc		# Multi I/O cards and multi-channel UARTs
 
@@ -279,11 +274,8 @@
 device		random		# Entropy device
 device		ether		# Ethernet support
 device		vlan		# 802.1Q VLAN support
-device		tun		# Packet tunnel.
 device		pty		# BSD-style compatibility pseudo ttys
 device		md		# Memory "disks"
-device		gif		# IPv6 and IPv4 tunneling
-device		faith		# IPv6-to-IPv4 relaying (translation)
 device		firmware	# firmware assist module
 
 # The `bpf' device enables the Berkeley Packet Filter.
@@ -298,39 +290,9 @@
 device		ehci		# EHCI PCI->USB interface (USB 2.0)
 device		xhci		# XHCI PCI->USB interface (USB 3.0)
 device		usb		# USB Bus (required)
-#device		udbp		# USB Double Bulk Pipe devices (needs netgraph)
-device		uhid		# "Human Interface Devices"
 device		ukbd		# Keyboard
-device		ulpt		# Printer
 device		umass		# Disks/Mass storage - Requires scbus and da
 device		ums		# Mouse
-device		urio		# Diamond Rio 500 MP3 player
-# USB Serial devices
-device		u3g		# USB-based 3G modems (Option, Huawei, Sierra)
-device		uark		# Technologies ARK3116 based serial adapters
-device		ubsa		# Belkin F5U103 and compatible serial adapters
-device		uftdi		# For FTDI usb serial adapters
-device		uipaq		# Some WinCE based devices
-device		uplcom		# Prolific PL-2303 serial adapters
-device		uslcom		# SI Labs CP2101/CP2102 serial adapters
-device		uvisor		# Visor and Palm devices
-device		uvscom		# USB serial support for DDI pocket's PHS
-# USB Ethernet, requires miibus
-device		aue		# ADMtek USB Ethernet
-device		axe		# ASIX Electronics USB Ethernet
-device		cdce		# Generic USB over Ethernet
-device		cue		# CATC USB Ethernet
-device		kue		# Kawasaki LSI USB Ethernet
-device		rue		# RealTek RTL8150 USB Ethernet
-device		udav		# Davicom DM9601E USB
-# USB Wireless
-device		rum		# Ralink Technology RT2501USB wireless NICs
-device		run		# Ralink Technology RT2700/RT2800/RT3000 NICs.
-device		uath		# Atheros AR5523 wireless NICs
-device		upgt		# Conexant/Intersil PrismGT wireless NICs.
-device		ural		# Ralink Technology RT2500USB wireless NICs
-device		urtw		# Realtek RTL8187B/L wireless NICs
-device		zyd		# ZyDAS zd1211/zd1211b wireless NICs
 
 # FireWire support
 device		firewire	# FireWire bus code
@@ -341,14 +303,6 @@
 device		dcons		# Dumb console driver
 device		dcons_crom	# Configuration ROM for dcons
 
-# Sound support
-device		sound		# Generic sound driver (required)
-device		snd_es137x	# Ensoniq AudioPCI ES137x
-device		snd_hda		# Intel High Definition Audio
-device		snd_ich		# Intel, NVidia and other ICH AC'97 Audio
-device		snd_uaudio	# USB Audio
-device		snd_via8233	# VIA VT8233x Audio
-
 # MMC/SD
 device		mmc		# MMC/SD bus
 device		mmcsd		# MMC/SD memory card
--- a/sys/i386/conf/GENERIC
+++ b/sys/i386/conf/GENERIC
@@ -18,12 +18,14 @@
 #
 # $FreeBSD$
 
-cpu		I486_CPU
-cpu		I586_CPU
-cpu		I686_CPU
-ident		GENERIC
+#cpu		I486_CPU
+#cpu		I586_CPU
+#cpu		I686_CPU
+#ident		GENERIC
 
-makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
+hints          "GENERIC.hints"         # Default places to look for devices.
+
+#makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
 
 options 	SCHED_ULE		# ULE scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
@@ -43,6 +45,8 @@
 options 	MSDOSFS			# MSDOS Filesystem
 options 	CD9660			# ISO 9660 Filesystem
 options 	PROCFS			# Process filesystem (requires PSEUDOFS)
+options		LINPROCFS
+options		COMPAT_LINUX		# Needed by LINPROCFS
 options 	PSEUDOFS		# Pseudo-filesystem framework
 options 	GEOM_PART_GPT		# GUID Partition Tables.
 options 	GEOM_LABEL		# Provides labelization
@@ -50,7 +54,7 @@
 options 	COMPAT_FREEBSD5		# Compatible with FreeBSD5
 options 	COMPAT_FREEBSD6		# Compatible with FreeBSD6
 options 	COMPAT_FREEBSD7		# Compatible with FreeBSD7
-options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
+options 	SCSI_DELAY=2000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	STACK			# stack(9) support
 options 	SYSVSHM			# SYSV-style shared memory
@@ -78,7 +82,7 @@
 options 	KDB_TRACE		# Print a stack trace for a panic.
 
 # To make an SMP kernel, the next two lines are needed
-options 	SMP			# Symmetric MultiProcessor Kernel
+#options 	SMP			# Symmetric MultiProcessor Kernel
 device		apic			# I/O APIC
 
 # CPU frequency control
@@ -89,9 +93,6 @@
 device		eisa
 device		pci
 
-# Floppy drives
-device		fdc
-
 # ATA controllers
 device		ahci		# AHCI-compatible SATA controllers
 device		ata		# Legacy ATA/SATA controllers
@@ -195,10 +196,6 @@
 # Parallel port
 device		ppc
 device		ppbus		# Parallel port bus (required)
-device		lpt		# Printer
-device		plip		# TCP/IP over parallel
-device		ppi		# Parallel port interface device
-#device		vpo		# Requires scbus and da
 
 device		puc		# Multi I/O cards and multi-channel UARTs
 
@@ -292,11 +289,8 @@
 device		random		# Entropy device
 device		ether		# Ethernet support
 device		vlan		# 802.1Q VLAN support
-device		tun		# Packet tunnel.
 device		pty		# BSD-style compatibility pseudo ttys
 device		md		# Memory "disks"
-device		gif		# IPv6 and IPv4 tunneling
-device		faith		# IPv6-to-IPv4 relaying (translation)
 device		firmware	# firmware assist module
 
 # The `bpf' device enables the Berkeley Packet Filter.
@@ -311,39 +305,9 @@
 device		ehci		# EHCI PCI->USB interface (USB 2.0)
 device		xhci		# XHCI PCI->USB interface (USB 3.0)
 device		usb		# USB Bus (required)
-#device		udbp		# USB Double Bulk Pipe devices (needs netgraph)
-device		uhid		# "Human Interface Devices"
 device		ukbd		# Keyboard
-device		ulpt		# Printer
 device		umass		# Disks/Mass storage - Requires scbus and da
 device		ums		# Mouse
-device		urio		# Diamond Rio 500 MP3 player
-# USB Serial devices
-device		u3g		# USB-based 3G modems (Option, Huawei, Sierra)
-device		uark		# Technologies ARK3116 based serial adapters
-device		ubsa		# Belkin F5U103 and compatible serial adapters
-device		uftdi		# For FTDI usb serial adapters
-device		uipaq		# Some WinCE based devices
-device		uplcom		# Prolific PL-2303 serial adapters
-device		uslcom		# SI Labs CP2101/CP2102 serial adapters
-device		uvisor		# Visor and Palm devices
-device		uvscom		# USB serial support for DDI pocket's PHS
-# USB Ethernet, requires miibus
-device		aue		# ADMtek USB Ethernet
-device		axe		# ASIX Electronics USB Ethernet
-device		cdce		# Generic USB over Ethernet
-device		cue		# CATC USB Ethernet
-device		kue		# Kawasaki LSI USB Ethernet
-device		rue		# RealTek RTL8150 USB Ethernet
-device		udav		# Davicom DM9601E USB
-# USB Wireless
-device		rum		# Ralink Technology RT2501USB wireless NICs
-device		run		# Ralink Technology RT2700/RT2800/RT3000 NICs.
-device		uath		# Atheros AR5523 wireless NICs
-device		upgt		# Conexant/Intersil PrismGT wireless NICs.
-device		ural		# Ralink Technology RT2500USB wireless NICs
-device		urtw		# Realtek RTL8187B/L wireless NICs
-device		zyd		# ZyDAS zd1211/zd1211b wireless NICs
 
 # FireWire support
 device		firewire	# FireWire bus code
@@ -354,15 +318,15 @@
 device		dcons		# Dumb console driver
 device		dcons_crom	# Configuration ROM for dcons
 
-# Sound support
-device		sound		# Generic sound driver (required)
-device		snd_es137x	# Ensoniq AudioPCI ES137x
-device		snd_hda		# Intel High Definition Audio
-device		snd_ich		# Intel, NVidia and other ICH AC'97 Audio
-device		snd_uaudio	# USB Audio
-device		snd_via8233	# VIA VT8233x Audio
-
 # MMC/SD
 device		mmc		# MMC/SD bus
 device		mmcsd		# MMC/SD memory card
 device		sdhci		# Generic PCI SD Host Controller
+
+# Alternate queueing
+options		ALTQ
+options		ALTQ_CBQ	# Class Bases Queuing (CBQ)
+options		ALTQ_RED	# Random Early Detection (RED)
+options		ALTQ_RIO	# RED In/Out
+options		ALTQ_HFSC	# Hierarchical Packet Scheduler (HFSC)
+options		ALTQ_PRIQ	# Priority Queuing (PRIQ)
--- /dev/null
+++ b/sys/conf/DEBIAN
@@ -0,0 +1,25 @@
+# Common options to all Debian GNU/kFreeBSD kernels
+
+options		LINPROCFS
+options         LINSYSFS
+options		FDESCFS
+options		TMPFS
+options         QUOTA
+options		KTRACE
+
+# Network options
+device		carp
+device		pf
+device		pfsync
+device		pflog
+
+# Alternate queueing
+options		ALTQ
+options		ALTQ_CBQ	# Class Bases Queuing (CBQ)
+options		ALTQ_RED	# Random Early Detection (RED)
+options		ALTQ_RIO	# RED In/Out
+options		ALTQ_HFSC	# Hierarchical Packet Scheduler (HFSC)
+options		ALTQ_PRIQ	# Priority Queuing (PRIQ)
+
+# Disable binary blobs
+include		WITHOUT_SOURCELESS
