--- a/sys/amd64/conf/GENERIC
+++ b/sys/amd64/conf/GENERIC
@@ -18,10 +18,12 @@
 #
 # $FreeBSD$
 
-cpu		HAMMER
-ident		GENERIC
+#cpu		HAMMER
+#ident		GENERIC
 
-makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
+hints          "GENERIC.hints"         # Default places to look for devices.
+
+#makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
 
 options 	SCHED_ULE		# ULE scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
@@ -49,7 +51,7 @@
 options 	COMPAT_FREEBSD5		# Compatible with FreeBSD5
 options 	COMPAT_FREEBSD6		# Compatible with FreeBSD6
 options 	COMPAT_FREEBSD7		# Compatible with FreeBSD7
-options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
+options 	SCSI_DELAY=2000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	STACK			# stack(9) support
 options 	SYSVSHM			# SYSV-style shared memory
@@ -96,9 +98,6 @@
 device		acpi
 device		pci
 
-# Floppy drives
-device		fdc
-
 # ATA controllers
 device		ahci		# AHCI-compatible SATA controllers
 device		ata		# Legacy ATA/SATA controllers
@@ -195,10 +194,6 @@
 # Parallel port
 device		ppc
 device		ppbus		# Parallel port bus (required)
-device		lpt		# Printer
-device		plip		# TCP/IP over parallel
-device		ppi		# Parallel port interface device
-#device		vpo		# Requires scbus and da
 
 device		puc		# Multi I/O cards and multi-channel UARTs
 
@@ -292,10 +287,7 @@
 device		random		# Entropy device
 device		ether		# Ethernet support
 device		vlan		# 802.1Q VLAN support
-device		tun		# Packet tunnel.
 device		md		# Memory "disks"
-device		gif		# IPv6 and IPv4 tunneling
-device		faith		# IPv6-to-IPv4 relaying (translation)
 device		firmware	# firmware assist module
 
 # The `bpf' device enables the Berkeley Packet Filter.
--- a/sys/i386/conf/GENERIC
+++ b/sys/i386/conf/GENERIC
@@ -18,12 +18,14 @@
 #
 # $FreeBSD$
 
-cpu		I486_CPU
-cpu		I586_CPU
-cpu		I686_CPU
-ident		GENERIC
+#cpu		I486_CPU
+#cpu		I586_CPU
+#cpu		I686_CPU
+#ident		GENERIC
 
-makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
+hints          "GENERIC.hints"         # Default places to look for devices.
+
+#makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
 
 options 	SCHED_ULE		# ULE scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
@@ -43,6 +45,8 @@
 options 	MSDOSFS			# MSDOS Filesystem
 options 	CD9660			# ISO 9660 Filesystem
 options 	PROCFS			# Process filesystem (requires PSEUDOFS)
+options		LINPROCFS
+options		COMPAT_LINUX		# Needed by LINPROCFS
 options 	PSEUDOFS		# Pseudo-filesystem framework
 options 	GEOM_PART_GPT		# GUID Partition Tables.
 options 	GEOM_LABEL		# Provides labelization
@@ -50,7 +54,7 @@
 options 	COMPAT_FREEBSD5		# Compatible with FreeBSD5
 options 	COMPAT_FREEBSD6		# Compatible with FreeBSD6
 options 	COMPAT_FREEBSD7		# Compatible with FreeBSD7
-options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
+options 	SCSI_DELAY=2000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	STACK			# stack(9) support
 options 	SYSVSHM			# SYSV-style shared memory
@@ -98,9 +102,6 @@
 device		eisa
 device		pci
 
-# Floppy drives
-device		fdc
-
 # ATA controllers
 device		ahci		# AHCI-compatible SATA controllers
 device		ata		# Legacy ATA/SATA controllers
@@ -205,10 +206,6 @@
 # Parallel port
 device		ppc
 device		ppbus		# Parallel port bus (required)
-device		lpt		# Printer
-device		plip		# TCP/IP over parallel
-device		ppi		# Parallel port interface device
-#device		vpo		# Requires scbus and da
 
 device		puc		# Multi I/O cards and multi-channel UARTs
 
@@ -305,10 +302,7 @@
 device		random		# Entropy device
 device		ether		# Ethernet support
 device		vlan		# 802.1Q VLAN support
-device		tun		# Packet tunnel.
 device		md		# Memory "disks"
-device		gif		# IPv6 and IPv4 tunneling
-device		faith		# IPv6-to-IPv4 relaying (translation)
 device		firmware	# firmware assist module
 
 # The `bpf' device enables the Berkeley Packet Filter.
--- /dev/null
+++ b/sys/conf/DEBIAN
@@ -0,0 +1,52 @@
+# Common options to all Debian GNU/kFreeBSD kernels
+
+options		LINPROCFS
+options         LINSYSFS
+options		FDESCFS
+options		TMPFS
+options         QUOTA
+options		KTRACE
+
+# Network options
+device		carp
+device		pf
+device		pfsync
+device		pflog
+
+# pty(4) was removed from GENERIC in 10-CURRENT (see rev 233271),
+# but unfortunetely glibc doesn't support pts(4) yet (see #667448).
+device		pty
+
+# Full debugger support is disabled by upstream when a branch is close to
+# release.  Unfortunately this can trigger the appearance of heisenbugs
+# which were previously hidden (we've been hit by this already, see #651624).
+#
+# This disables full debugger support in advance, so that the build
+# configuration which is going to be used for release is tested during the
+# whole release cycle.  If you need a debug kernel, simply remove/comment
+# the following lines:
+
+nooption 	DDB			# Support DDB.
+nooption 	GDB			# Support remote GDB.
+nooption 	DEADLKRES		# Enable the deadlock resolver
+nooption 	INVARIANTS		# Enable calls of extra sanity checking
+nooption 	INVARIANT_SUPPORT	# Extra sanity checks of internal structures, required by INVARIANTS
+nooption 	WITNESS			# Enable checks to detect deadlocks and cycles
+nooption 	WITNESS_SKIPSPIN	# Don't run witness on spinlocks for speed
+nooption 	MALLOC_DEBUG_MAXZONES	# Separate malloc(9) zones
+
+# Usually when full debugger is enabled, minimum debugger support is disabled.
+# Bring it back:
+
+options 	KDB_TRACE		# Print a stack trace for a panic.
+
+# Alternate queueing
+options		ALTQ
+options		ALTQ_CBQ	# Class Bases Queuing (CBQ)
+options		ALTQ_RED	# Random Early Detection (RED)
+options		ALTQ_RIO	# RED In/Out
+options		ALTQ_HFSC	# Hierarchical Packet Scheduler (HFSC)
+options		ALTQ_PRIQ	# Priority Queuing (PRIQ)
+
+# Disable binary blobs
+include		WITHOUT_SOURCELESS
