
See #594940  

--- a/sys/modules/drm/Makefile
+++ b/sys/modules/drm/Makefile
@@ -4,9 +4,6 @@
 	drm \
 	i915 \
 	mach64 \
-	mga \
-	r128 \
-	radeon \
 	savage \
 	sis \
 	tdfx \
--- a/sys/modules/Makefile
+++ b/sys/modules/Makefile
@@ -36,7 +36,6 @@
 	ath \
 	ath_pci \
 	${_auxio} \
-	bce \
 	bfe \
 	bge \
 	${_bxe} \
@@ -65,14 +64,11 @@
 	coda5 \
 	${_coff} \
 	${_coretemp} \
-	${_cp} \
 	${_cpuctl} \
 	${_cpufreq} \
 	${_crypto} \
 	${_cryptodev} \
 	${_cs} \
-	${_ctau} \
-	${_cxgb} \
 	cxgbe \
 	${_cyclic} \
 	dc \
@@ -95,13 +91,11 @@
 	${_ex} \
 	${_exca} \
 	${_ext2fs} \
-	fatm \
 	fdc \
 	fdescfs \
 	${_fe} \
 	firewire \
 	firmware \
-	fxp \
 	gem \
 	geom \
 	${_glxiic} \
@@ -141,14 +135,10 @@
 	ip_mroute_mod \
 	${_ips} \
 	${_ipw} \
-	${_ipwfw} \
 	iscsi \
 	isp \
-	ispfw \
 	${_iwi} \
-	${_iwifw} \
 	${_iwn} \
-	${_iwnfw} \
 	${_ixgb} \
 	${_ixgbe} \
 	jme \
@@ -203,7 +193,6 @@
 	${_mthca} \
 	mvs \
 	mwl \
-	mwlfw \
 	mxge \
 	my \
 	${_ncp} \
@@ -252,14 +241,12 @@
 	puc \
 	${_qlxgb} \
 	ral \
-	ralfw \
 	${_random} \
 	rc4 \
 	${_rdma} \
 	re \
 	reiserfs \
 	rl \
-	runfw \
 	${_s3} \
 	${_safe} \
 	${_sbni} \
@@ -269,7 +256,6 @@
 	sdhci \
 	sem \
 	send \
-	sf \
 	${_sfxge} \
 	sge \
 	siba_bwn \
@@ -278,7 +264,6 @@
 	sis \
 	sk \
 	${_smbfs} \
-	sn \
 	${_snc} \
 	snp \
 	${_sound} \
@@ -293,7 +278,6 @@
 	${_sym} \
 	${_syscons} \
 	sysvipc \
-	ti \
 	tl \
 	tmpfs \
 	${_tpm} \
@@ -302,7 +286,6 @@
 	twe \
 	tws \
 	tx \
-	txp \
 	uart \
 	ubsec \
 	udf \
@@ -332,7 +315,6 @@
 	wlan_wep \
 	wlan_xauth \
 	${_wpi} \
-	${_wpifw} \
 	${_x86bios} \
 	${_xe} \
 	xfs \
@@ -409,9 +391,7 @@
 _bxe=		bxe
 _cardbus=	cardbus
 _cbb=		cbb
-_ce=		ce
 _coff=		coff
-_cp=		cp
 _cpuctl=	cpuctl
 _cpufreq=	cpufreq
 _cs=		cs
@@ -513,11 +493,8 @@
 _ipmi=		ipmi
 _ips=		ips
 _ipw=		ipw
-_ipwfw=		ipwfw
 _iwi=		iwi
-_iwifw=		iwifw
 _iwn=		iwn
-_iwnfw=		iwnfw
 _ixgb=		ixgb
 _ixgbe=		ixgbe
 _mly=		mly
@@ -528,7 +505,6 @@
 _tpm=		tpm
 _viawd=		viawd
 _wpi=		wpi
-_wpifw=		wpifw
 .if ${MK_CRYPT} != "no" || defined(ALL_MODULES)
 _padlock=	padlock
 .endif
@@ -593,11 +569,8 @@
 _ipmi=		ipmi
 _ips=		ips
 _ipw=		ipw
-_ipwfw=		ipwfw
 _iwi=		iwi
-_iwifw=		iwifw
 _iwn=		iwn
-_iwnfw=		iwnfw
 _ixgb=		ixgb
 _ixgbe=		ixgbe
 _lindev=	lindev
@@ -643,7 +616,6 @@
 _x86bios=	x86bios
 _wi=		wi
 _wpi=		wpi
-_wpifw=		wpifw
 .if ${MK_ZFS} != "no" || defined(ALL_MODULES)
 _zfs=		zfs
 .endif
--- a/sys/modules/sound/driver/Makefile
+++ b/sys/modules/sound/driver/Makefile
@@ -1,7 +1,7 @@
 # $FreeBSD$
 
-SUBDIR=		ad1816 als4000 atiixp cs4281 csa ds1 emu10k1 emu10kx
-SUBDIR+=	envy24 envy24ht es137x ess fm801 hda ich maestro maestro3
+SUBDIR=		ad1816 als4000 atiixp cs4281 emu10k1 emu10kx
+SUBDIR+=	envy24 envy24ht es137x ess fm801 hda ich maestro
 SUBDIR+=	neomagic sb16 sb8 sbc solo spicds t4dwave via8233
 SUBDIR+=	via82c686 vibes driver uaudio
 
--- a/sys/modules/usb/Makefile
+++ b/sys/modules/usb/Makefile
@@ -27,11 +27,11 @@
 
 SUBDIR = usb
 SUBDIR += ehci musb ohci uhci xhci uss820dci ${_at91dci} ${_atmegadci} ${_avr32dci}
-SUBDIR += rum run uath upgt usie ural zyd ${_urtw}
+SUBDIR += run upgt usie ural ${_urtw}
 SUBDIR += atp uhid ukbd ums udbp ufm uep
 SUBDIR += ucom u3g uark ubsa ubser uchcom ucycom ufoma uftdi ugensa uipaq ulpt \
 	  umct umcs umodem umoscom uplcom uslcom uvisor uvscom
-SUBDIR += uether aue axe cdce cue kue mos rue udav uhso ipheth
+SUBDIR += uether aue axe cdce cue mos rue udav uhso ipheth
 SUBDIR += usfs umass urio
 SUBDIR += quirk template
 
