diff -urN nscd/mem.c.old nscd/mem.c
--- nscd/mem.c.old	2004-09-13 07:56:03.000000000 +0200
+++ nscd/mem.c	2005-12-19 10:49:44.000000000 +0100
@@ -493,7 +493,8 @@
 	  size_t newtotal = (sizeof (struct database_pers_head)
 			     + db->head->module * sizeof (ref_t)
 			     + new_data_size);
-
+#include <_G_config.h>
+#ifdef _G_HAVE_MREMAP
 	  if ((!db->mmap_used || ftruncate (db->wr_fd, newtotal) != 0)
 	      /* Try to resize the mapping.  Note: no MREMAP_MAYMOVE.  */
 	      && mremap (db->head, oldtotal, newtotal, 0) == 0)
@@ -502,6 +503,7 @@
 	      tried_resize = true;
 	      goto retry;
 	    }
+#endif	    
 	}
 
       if (! db->last_alloc_failed)
