--- include/sys/wait.h~	2005-12-28 09:31:34.000000000 +0100
+++ include/sys/wait.h	2005-12-28 09:31:34.000000000 +0100
@@ -13,6 +13,5 @@
 extern __pid_t __wait3 (__WAIT_STATUS __stat_loc,
 			int __options, struct rusage * __usage);
 extern __pid_t __wait4 (__pid_t __pid, __WAIT_STATUS __stat_loc,
-			int __options, struct rusage *__usage)
-			attribute_hidden;
+			int __options, struct rusage *__usage);
 #endif

