*** orig/alpha/machine/proc.h	Thu Aug  3 23:10:58 2000
--- import/alpha/machine/proc.h	Fri Sep  6 01:34:23 2002
***************
*** 1,3 ****
--- 1,21 ----
+ /* Copyright (C) 2002 Free Software Foundation, Inc.
+    This file is part of the GNU C Library.
+ 
+    The GNU C Library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+  
+    The GNU C Library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+ 
+    You should have received a copy of the GNU Lesser General Public
+    License along with the GNU C Library; if not, write to the Free
+    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+    02111-1307 USA.  */
+ 
  /* $FreeBSD: src/sys/alpha/include/proc.h,v 1.7.2.1 2000/08/03 21:10:58 peter Exp $ */
  /* From: NetBSD: proc.h,v 1.3 1997/04/06 08:47:36 cgd Exp */
  
***************
*** 28,33 ****
--- 46,57 ----
   * rights to redistribute these changes.
   */
  
+ #ifndef _MACHINE_PROC_H_
+ #define	_MACHINE_PROC_H_
+ 
+ #include <sys/types.h>
+ #include <machine/types.h>
+ 
  /*
   * Machine-dependent part of the proc struct for the Alpha.
   */
***************
*** 55,57 ****
--- 79,83 ----
  #define MDP_UAC_SIGBUS	0x0040		/* Deliver SIGBUS upon
  					   unaligned access */
  #define MDP_UAC_MASK	(MDP_UAC_NOPRINT | MDP_UAC_NOFIX | MDP_UAC_SIGBUS)
+ 
+ #endif /* !_MACHINE_PROC_H_ */
