--- sysdeps/generic/bits/types.h	2004-09-26 10:47:37.000000000 +0200
+++ sysdeps/generic/bits/types.h	2005-08-23 17:35:07.000000000 +0200
@@ -145,6 +145,7 @@
 __STD_TYPE __PID_T_TYPE __pid_t;	/* Type of process identifications.  */
 __STD_TYPE __FSID_T_TYPE __fsid_t;	/* Type of file system IDs.  */
 __STD_TYPE __CLOCK_T_TYPE __clock_t;	/* Type of CPU usage counts.  */
+__STD_TYPE __CLOCK_T_TYPE __kernel_time_t; /* Type of CPU usage counts.  */
 __STD_TYPE __RLIM_T_TYPE __rlim_t;	/* Type for resource measurement.  */
 __STD_TYPE __RLIM64_T_TYPE __rlim64_t;	/* Type for resource measurement (LFS).  */
 __STD_TYPE __ID_T_TYPE __id_t;		/* General type for IDs.  */

