diff -ur src.old/sys/conf/kern.mk src/sys/conf/kern.mk
--- src.old/sys/conf/kern.mk	2004-05-14 15:35:46.000000000 +0200
+++ src/sys/conf/kern.mk	2004-11-08 23:54:04.000000000 +0100
@@ -12,7 +12,7 @@
 .else
 CWARNFLAGS?=	-Wall -Wredundant-decls -Wnested-externs -Wstrict-prototypes \
 		-Wmissing-prototypes -Wpointer-arith -Winline -Wcast-qual \
-		-fformat-extensions -std=c99
+		-std=c99
 .endif
 #
 # The following flags are next up for working on:
@@ -28,7 +28,7 @@
 # cache tag lines)
 #
 .if ${MACHINE_ARCH} == "i386" && ${CC} != "icc"
-CFLAGS+=	-mno-align-long-strings -mpreferred-stack-boundary=2
+CFLAGS+=	-mpreferred-stack-boundary=2
 INLINE_LIMIT?=	8000
 .endif
 
diff -ur src.old/sys/conf/kern.post.mk src/sys/conf/kern.post.mk
--- src.old/sys/conf/kern.post.mk	2004-09-23 10:34:50.000000000 +0200
+++ src/sys/conf/kern.post.mk	2004-11-08 22:26:14.000000000 +0100
@@ -184,11 +184,9 @@
 .if exists(${DESTDIR}${KODIR})
 	-thiskernel=`sysctl -n kern.bootfile` ; \
 	if [ "`dirname "$$thiskernel"`" != ${DESTDIR}${KODIR} ] ; then \
-		chflags -R noschg ${DESTDIR}${KODIR} ; \
 		rm -rf ${DESTDIR}${KODIR} ; \
 	else \
 		if [ -d ${DESTDIR}${KODIR}.old ] ; then \
-			chflags -R noschg ${DESTDIR}${KODIR}.old ; \
 			rm -rf ${DESTDIR}${KODIR}.old ; \
 		fi ; \
 		mv ${DESTDIR}${KODIR} ${DESTDIR}${KODIR}.old ; \
@@ -197,17 +195,17 @@
 .endif
 	mkdir -p ${DESTDIR}${KODIR}
 .if defined(DEBUG) && defined(INSTALL_DEBUG)
-	${INSTALL} -p -m 555 -o root -g wheel ${FULLKERNEL} ${DESTDIR}${KODIR}
+	${INSTALL} -p -m 555 -o root -g root ${FULLKERNEL} ${DESTDIR}${KODIR}
 .else
-	${INSTALL} -p -m 555 -o root -g wheel ${KERNEL_KO} ${DESTDIR}${KODIR}
+	${INSTALL} -p -m 555 -o root -g root ${KERNEL_KO} ${DESTDIR}${KODIR}
 .endif
 
 kernel-reinstall:
 	@-chflags -R noschg ${DESTDIR}${KODIR}
 .if defined(DEBUG) && defined(INSTALL_DEBUG)
-	${INSTALL} -p -m 555 -o root -g wheel ${FULLKERNEL} ${DESTDIR}${KODIR}
+	${INSTALL} -p -m 555 -o root -g root ${FULLKERNEL} ${DESTDIR}${KODIR}
 .else
-	${INSTALL} -p -m 555 -o root -g wheel ${KERNEL_KO} ${DESTDIR}${KODIR}
+	${INSTALL} -p -m 555 -o root -g root ${KERNEL_KO} ${DESTDIR}${KODIR}
 .endif
 
 config.o env.o hints.o majors.o vers.o vnode_if.o:
