diff -ur src.old/sys/sys/conf.h src/sys/sys/conf.h
--- src.old/sys/sys/conf.h	2004-11-08 21:11:48.000000000 +0100
+++ src/sys/sys/conf.h	2004-11-09 00:52:56.000000000 +0100
@@ -273,15 +273,15 @@
 void devfs_destroy(struct cdev *dev);
 
 #define		UID_ROOT	0
-#define		UID_BIN		3
-#define		UID_UUCP	66
+#define		UID_BIN		2	/* debian bin */
+#define		UID_UUCP	0	/* linux ttyS0 */
 
 #define		GID_WHEEL	0
-#define		GID_KMEM	2
-#define		GID_OPERATOR	5
-#define		GID_BIN		7
-#define		GID_GAMES	13
-#define		GID_DIALER	68
+#define		GID_KMEM	15	/* linux kmem */
+#define		GID_OPERATOR	0	/* no such group */
+#define		GID_BIN		2	/* debian bin */
+#define		GID_GAMES	60	/* debian games */
+#define		GID_DIALER	20	/* linux ttyS0 */
 
 typedef void (*dev_clone_fn)(void *arg, char *name, int namelen, struct cdev **result);
 
Only in src/sys/sys: conf.h.orig
