diff -ur src.old/sys/sys/syslimits.h src/sys/sys/syslimits.h
--- src.old/sys/sys/syslimits.h	2004-04-07 06:19:49.000000000 +0200
+++ src/sys/sys/syslimits.h	2004-11-09 02:38:57.000000000 +0100
@@ -43,7 +43,7 @@
  * Do not add any new variables here.  (See the comment at the end of
  * the file for why.)
  */
-#define	ARG_MAX			65536	/* max bytes for an exec function */
+#define	ARG_MAX		       131072	/* max bytes for an exec function */
 #ifndef CHILD_MAX
 #define	CHILD_MAX		   40	/* max simultaneous processes */
 #endif
