diff -ur src.old/sys/conf/kmod.mk src/sys/conf/kmod.mk
--- src.old/sys/conf/kmod.mk	2004-10-24 10:26:48.000000000 +0200
+++ src/sys/conf/kmod.mk	2004-11-09 00:46:12.000000000 +0100
@@ -170,7 +170,7 @@
 	grep -v '^#' < ${EXPORT_SYMS} > export_syms
 .endif
 	awk -f ${SYSDIR}/conf/kmod_syms.awk ${.TARGET} \
-	    export_syms | xargs -J% ${OBJCOPY} % ${.TARGET}
+	    export_syms | xargs -i% ${OBJCOPY} % ${.TARGET}
 .endif
 .endif
 .if !defined(DEBUG_FLAGS) && ${MACHINE_ARCH} == amd64
diff -ur src.old/sys/contrib/dev/acpica/acpica_prep.sh src/sys/contrib/dev/acpica/acpica_prep.sh
--- src.old/sys/contrib/dev/acpica/acpica_prep.sh	2004-04-14 04:03:33.000000000 +0200
+++ src/sys/contrib/dev/acpica/acpica_prep.sh	2004-11-09 00:46:36.000000000 +0100
@@ -52,12 +52,12 @@
 
 echo copying full dirs
 for i in ${fulldirs}; do
-    find ${wrk} -name ${i} -type d | xargs -J % mv % ${dst}
+    find ${wrk} -name ${i} -type d | xargs -i % mv % ${dst}
 done
 
 # move files to destination
 echo copying flat dirs
-find ${wrk} -type f | xargs -J % mv % ${dst}
+find ${wrk} -type f | xargs -i % mv % ${dst}
 mv CHANGES.txt ${dst}
 
 # update src/headers for appropriate paths
