--- sys/boot/forth/beastie.4th.old	2006-04-21 22:04:10.000000000 +0200
+++ sys/boot/forth/beastie.4th	2006-04-21 22:04:16.000000000 +0200
@@ -49,49 +49,47 @@
 
 46 constant dot
 
-\ The BSD Daemon.  He is 19 rows high and 34 columns wide
+\ Debian swirl
 : technicolor-beastie ( x y -- )
-2dup at-xy ."               [1;31m,        ," 1+
-2dup at-xy ."              /(        )`" 1+
-2dup at-xy ."              \ \___   / |" 1+
-2dup at-xy ."              /- [37m_[31m  `-/  '" 1+
-2dup at-xy ."             ([37m/\/ \[31m \   /\" 1+
-2dup at-xy ."             [37m/ /   |[31m `    \" 1+
-2dup at-xy ."             [34mO O   [37m) [31m/    |" 1+
-2dup at-xy ."             [37m`-^--'[31m`<     '" 1+
-2dup at-xy ."            (_.)  _  )   /" 1+
-2dup at-xy ."             `.___/`    /       " 1+
-2dup at-xy ."               `-----' /" 1+
-2dup at-xy ."  [33m<----.[31m     __ / __   \" 1+
-2dup at-xy ."  [33m<----|====[31mO)))[33m==[31m) \) /[33m====|" 1+
-2dup at-xy ."  [33m<----'[31m    `--' `.__,' \" 1+
-2dup at-xy ."               |        |" 1+
-2dup at-xy ."                \       /       /\" 1+
-2dup at-xy ."           [36m______[31m( (_  / \______/" 1+
-2dup at-xy ."         [36m,'  ,-----'   |" 1+
-at-xy ."         `--{__________) [0m"
+2dup at-xy ." " 1+
+2dup at-xy ." " 1+
+2dup at-xy ."           [31m_sudZUZ#Z#XZo=_[0m" 1+
+2dup at-xy ."        [31m_jmZZ2!!~---~!!X##wa[0m" 1+
+2dup at-xy ."     [31m.<wdP~~[0m            [31m-!YZL,[0m   [0;37;0m" 1+
+2dup at-xy ."    [31m.mX2'[0m       [31m_%aaa__[0m     [31mXZ[.[0m [0;37;0m" 1+
+2dup at-xy ."    [31moZ[[0m      [31m_jdXY!~?S#wa[0m   [31m]Xb;[0m [0;37;0m" 1+
+2dup at-xy ."   [31m_#e'[0m     [31m.]X2([0m     [31m~Xw|[0m  [31m)XXc[0m [0;37;0m" 1+
+2dup at-xy ."  [31m.2Z`[0m      [31m]X[.[0m       [31mxY|[0m  [31m]oZ([0m [0;37;0m" 1+
+2dup at-xy ."  [31m.2#;[0m      [31m)3k;[0m     [31m_s!~[0m   [31mjXf`[0m [0;37;0m" 1+
+2dup at-xy ."   [31m1Z>[0m      [31m-]Xb/[0m    [31m~[0m    [31m__#2([0m  [0;37;0m" 1+
+2dup at-xy ."   [31m-Zo;[0m       [31m+!4ZwaaaauZZXY'[0m    [0;37;0m" 1+
+2dup at-xy ."    [31m*#[,[0m        [31m~-?!!!!!!-~[0m      [0;37;0m" 1+
+2dup at-xy ."     [31mXUb;.[0m                       [0;37;0m" 1+
+2dup at-xy ."      [31m)YXL,,[0m                     [0;37;0m" 1+
+2dup at-xy ."        [31m+3#bc,[0m                   [0;37;0m" 1+
+2dup at-xy ."          [31m-)SSL,,[0m                [0;37;0m" 1+
+     at-xy ."             [31m~~~~~[0m               [0;37;0m"
 ;
 
 : boring-beastie ( x y -- )
-	2dup at-xy ."              ,        ," 1+
-	2dup at-xy ."             /(        )`" 1+
-	2dup at-xy ."             \ \___   / |" 1+
-	2dup at-xy ."             /- _  `-/  '" 1+
-	2dup at-xy ."            (/\/ \ \   /\" 1+
-	2dup at-xy ."            / /   | `    \" 1+
-	2dup at-xy ."            O O   ) /    |" 1+
-	2dup at-xy ."            `-^--'`<     '" 1+
-	2dup at-xy ."           (_.)  _  )   /" 1+
-	2dup at-xy ."            `.___/`    /" 1+
-	2dup at-xy ."              `-----' /" 1+
-	2dup at-xy ." <----.     __ / __   \" 1+
-	2dup at-xy ." <----|====O)))==) \) /====" 1+
-	2dup at-xy ." <----'    `--' `.__,' \" 1+
-	2dup at-xy ."              |        |" 1+
-	2dup at-xy ."               \       /       /\" 1+
-	2dup at-xy ."          ______( (_  / \______/" 1+
-	2dup at-xy ."        ,'  ,-----'   |" 1+
-	     at-xy ."        `--{__________)"
+2dup at-xy ." " 1+
+2dup at-xy ." " 1+
+2dup at-xy ."           _sudZUZ#Z#XZo=_" 1+
+2dup at-xy ."        _jmZZ2!!~---~!!X##wa" 1+
+2dup at-xy ."     .<wdP~~            -!YZL," 1+
+2dup at-xy ."    .mX2'       _%aaa__     XZ[." 1+
+2dup at-xy ."    oZ[      _jdXY!~?S#wa   ]Xb;" 1+
+2dup at-xy ."   _#e'     .]X2(     ~Xw|  )XXc" 1+
+2dup at-xy ."  ..2Z`      ]X[.       xY|  ]oZ(" 1+
+2dup at-xy ."  ..2#;      )3k;     _s!~   jXf`" 1+
+2dup at-xy ."   1Z>      -]Xb/    ~    __#2(" 1+
+2dup at-xy ."   -Zo;       +!4ZwaaaauZZXY'" 1+
+2dup at-xy ."    *#[,        ~-?!!!!!!-~" 1+
+2dup at-xy ."     XUb;." 1+
+2dup at-xy ."      )YXL,," 1+
+2dup at-xy ."        +3#bc," 1+
+2dup at-xy ."          -)SSL,," 1+
+     at-xy ."             ~~~~~"
 ;
 
 : print-beastie ( x y -- )
@@ -146,11 +144,11 @@
 	clear
 	46 4 print-beastie
 	42 20 2 2 box
-	13 6 at-xy ." Welcome to FreeBSD!"
-	printmenuitem ."  Boot FreeBSD [default]" bootkey !
+	7 6 at-xy ." Welcome to Debian GNU/kFreeBSD!"
+	printmenuitem ."  Boot Debian [default]" bootkey !
 	s" arch-i386" environment? if
 		drop
-		printmenuitem ."  Boot FreeBSD with ACPI " bootacpikey !
+		printmenuitem ."  Boot Debian with ACPI " bootacpikey !
 		acpienabled? if
 			." disabled"
 		else
@@ -159,13 +157,13 @@
 	else
 		-2 bootacpikey !
 	then
-	printmenuitem ."  Boot FreeBSD in Safe Mode" bootsafekey !
-	printmenuitem ."  Boot FreeBSD in single user mode" bootsinglekey !
-	printmenuitem ."  Boot FreeBSD with verbose logging" bootverbosekey !
+	printmenuitem ."  Boot Debian in Safe Mode" bootsafekey !
+	printmenuitem ."  Boot Debian in single user mode" bootsinglekey !
+	printmenuitem ."  Boot Debian with verbose logging" bootverbosekey !
 	printmenuitem ."  Escape to loader prompt" escapekey !
 	s" arch-i386" environment? if
 		drop
-		printmenuitem ."  Boot FreeBSD with USB keyboard" bootusbkey !
+		printmenuitem ."  Boot Debian with USB keyboard" bootusbkey !
 	else
 		-2 bootusbkey !
 	then
