diff -ur kfreebsd-headers-4.6/import.old/dev/asr/i2oadptr.h kfreebsd-headers-4.6/import/dev/asr/i2oadptr.h
--- kfreebsd-headers-4.6/import.old/dev/asr/i2oadptr.h	2004-05-31 03:25:19.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/i2oadptr.h	2004-05-31 04:06:24.000000000 +0200
@@ -81,7 +81,7 @@
 #if !defined(I2O_ADPTR_HDR)
 #define I2O_ADPTR_HDR
 
-#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD__))
+#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD_kernel__))
 # if (KERN_VERSION < 3)
 #  include "i386/pci/i2omsg.h"
 # else
diff -ur kfreebsd-headers-4.6/import.old/dev/asr/i2obscsi.h kfreebsd-headers-4.6/import/dev/asr/i2obscsi.h
--- kfreebsd-headers-4.6/import.old/dev/asr/i2obscsi.h	2004-05-31 03:25:19.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/i2obscsi.h	2004-05-31 04:06:24.000000000 +0200
@@ -81,7 +81,7 @@
 #if !defined(I2O_BASE_SCSI_HDR)
 #define I2O_BASE_SCSI_HDR
 
-#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD__))
+#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD_kernel__))
 # if (KERN_VERSION < 3)
 #  include    "i386/pci/i2omsg.h"          /* Include the Base Message file */
 # else
diff -ur kfreebsd-headers-4.6/import.old/dev/asr/i2oexec.h kfreebsd-headers-4.6/import/dev/asr/i2oexec.h
--- kfreebsd-headers-4.6/import.old/dev/asr/i2oexec.h	2004-05-31 03:25:20.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/i2oexec.h	2004-05-31 04:06:24.000000000 +0200
@@ -90,7 +90,7 @@
 
 #define I2OEXEC_REV 1_5_4  /* I2OExec header file revision string */
 
-#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD__))
+#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD_kernel__))
 # if (!defined(KERN_VERSION))
 #  include <sys/sysctl.h>
 # endif
diff -ur kfreebsd-headers-4.6/import.old/dev/asr/i2omsg.h kfreebsd-headers-4.6/import/dev/asr/i2omsg.h
--- kfreebsd-headers-4.6/import.old/dev/asr/i2omsg.h	2004-05-31 03:25:20.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/i2omsg.h	2004-05-31 04:06:24.000000000 +0200
@@ -114,7 +114,7 @@
 
 
 
-#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD__))
+#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD_kernel__))
 # if (KERN_VERSION < 3)
 #  include "i386/pci/i2otypes.h"
 # else
diff -ur kfreebsd-headers-4.6/import.old/dev/asr/i2otypes.h kfreebsd-headers-4.6/import/dev/asr/i2otypes.h
--- kfreebsd-headers-4.6/import.old/dev/asr/i2otypes.h	2004-05-31 03:25:20.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/i2otypes.h	2004-05-31 04:06:24.000000000 +0200
@@ -85,7 +85,7 @@
 
 /* include architecture/compiler dependencies */
 
-#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD__))
+#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD_kernel__))
 # if (KERN_VERSION < 3)
 #  include "i386/pci/i2odep.h"
 # else
diff -ur kfreebsd-headers-4.6/import.old/dev/asr/i2outil.h kfreebsd-headers-4.6/import/dev/asr/i2outil.h
--- kfreebsd-headers-4.6/import.old/dev/asr/i2outil.h	2004-05-31 03:25:20.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/i2outil.h	2004-05-31 04:06:24.000000000 +0200
@@ -90,7 +90,7 @@
 
 #define I2OUTIL_REV 1_5_4  /* I2OUtil header file revision string */
 
-#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD__))
+#if ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD_kernel__))
 # if (KERN_VERSION < 3)
 #  include   "i386/pci/i2omsg.h"      /* Include the Base Message file */
 # else
diff -ur kfreebsd-headers-4.6/import.old/dev/asr/osd_defs.h kfreebsd-headers-4.6/import/dev/asr/osd_defs.h
--- kfreebsd-headers-4.6/import.old/dev/asr/osd_defs.h	2004-05-31 03:25:20.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/osd_defs.h	2004-05-31 04:06:24.000000000 +0200
@@ -56,7 +56,7 @@
 # define _DPT_LINUX
 #elif (defined(__bsdi__))
 # define _DPT_BSDI
-#elif (defined(__FreeBSD__))
+#elif (defined(__FreeBSD_kernel__))
 # undef _DPT_FREE_BSD
 # define _DPT_FREE_BSD
 #else
diff -ur kfreebsd-headers-4.6/import.old/dev/asr/osd_util.h kfreebsd-headers-4.6/import/dev/asr/osd_util.h
--- kfreebsd-headers-4.6/import.old/dev/asr/osd_util.h	2004-05-31 03:25:20.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/osd_util.h	2004-05-31 04:06:24.000000000 +0200
@@ -78,7 +78,7 @@
 
 #if (defined(KERNEL) && defined(__bsdi__))
 # include        "i386/isa/dpt_osd_defs.h"
-#elif ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD__))
+#elif ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD_kernel__))
 # if (KERN_VERSION < 3)
 #  include        "i386/isa/dpt_osd_defs.h"
 # else
diff -ur kfreebsd-headers-4.6/import.old/dev/asr/sys_info.h kfreebsd-headers-4.6/import/dev/asr/sys_info.h
--- kfreebsd-headers-4.6/import.old/dev/asr/sys_info.h	2004-05-31 03:25:21.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/asr/sys_info.h	2004-05-31 04:06:24.000000000 +0200
@@ -53,7 +53,7 @@
 
 #if (defined(KERNEL) && defined(__bsdi__))
 # include        "i386/isa/dpt_osd_util.h"
-#elif ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD__))
+#elif ((defined(KERNEL) || defined(_KERNEL)) && defined(__FreeBSD_kernel__))
 # if (KERN_VERSION < 3)
 #  include        "i386/isa/dpt_osd_util.h"
 # else
diff -ur kfreebsd-headers-4.6/import.old/dev/awi/awivar.h kfreebsd-headers-4.6/import/dev/awi/awivar.h
--- kfreebsd-headers-4.6/import.old/dev/awi/awivar.h	2004-05-31 03:25:23.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/awi/awivar.h	2004-05-31 04:06:25.000000000 +0200
@@ -94,7 +94,7 @@
 	struct ethercom		sc_ec;
 	void			*sc_ih; /* interrupt handler */
 #endif
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #if __FreeBSD__ >= 4
 	struct {
 		char	dv_xname[64];	/*XXX*/
diff -ur kfreebsd-headers-4.6/import.old/dev/bktr/bktr_os.h kfreebsd-headers-4.6/import/dev/bktr/bktr_os.h
--- kfreebsd-headers-4.6/import.old/dev/bktr/bktr_os.h	2004-05-31 03:25:25.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/bktr/bktr_os.h	2004-05-31 04:06:25.000000000 +0200
@@ -47,7 +47,7 @@
 /******************************/
 /* *** Memory Allocation  *** */
 /******************************/
-#if (defined(__FreeBSD__) || defined(__bsdi__))
+#if (defined(__FreeBSD_kernel__) || defined(__bsdi__))
 vm_offset_t     get_bktr_mem( int unit, unsigned size );
 #endif
 
@@ -59,7 +59,7 @@
 /************************************/
 /* *** Interrupt Enable/Disable *** */
 /************************************/
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 #define DECLARE_INTR_MASK(s)	intrmask_t s
 #define DISABLE_INTR(s)		s=spltty()
 #define ENABLE_INTR(s)		splx(s)
diff -ur kfreebsd-headers-4.6/import.old/dev/bktr/bktr_reg.h kfreebsd-headers-4.6/import/dev/bktr/bktr_reg.h
--- kfreebsd-headers-4.6/import.old/dev/bktr/bktr_reg.h	2004-05-31 03:25:25.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/bktr/bktr_reg.h	2004-05-31 04:06:25.000000000 +0200
@@ -34,7 +34,7 @@
  *
  */
 
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #  if (__FreeBSD_version >= 310000)
 #    include "smbus.h"
 #  else
@@ -61,7 +61,7 @@
  * Support the older kernel options on FreeBSD and OpenBSD.
  *
  */
-#if defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD_kernel__) || defined(__OpenBSD__)
 #if defined(BROOKTREE_ALLOC_PAGES)
 #define BKTR_ALLOC_PAGES BROOKTREE_ALLOC_PAGES
 #endif
@@ -470,7 +470,7 @@
  * memory mapped structure method only works on 32 bit processors
  * with the right type of endianness.
  */
-#if defined(__NetBSD__) || ( defined(__FreeBSD__) && (__FreeBSD_version >=300000) )
+#if defined(__NetBSD__) || ( defined(__FreeBSD_kernel__) && (__FreeBSD_version >=300000) )
 #define INB(bktr,offset)	bus_space_read_1((bktr)->memt,(bktr)->memh,(offset))
 #define INW(bktr,offset)	bus_space_read_2((bktr)->memt,(bktr)->memh,(offset))
 #define INL(bktr,offset)	bus_space_read_4((bktr)->memt,(bktr)->memh,(offset))
@@ -726,7 +726,7 @@
 typedef int ioctl_cmd_t;
 #endif
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 #if (__FreeBSD_version >= 300000)
 typedef u_long ioctl_cmd_t;
 #endif
diff -ur kfreebsd-headers-4.6/import.old/dev/en/midwayreg.h kfreebsd-headers-4.6/import/dev/en/midwayreg.h
--- kfreebsd-headers-4.6/import.old/dev/en/midwayreg.h	2004-05-31 03:25:30.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/en/midwayreg.h	2004-05-31 04:06:28.000000000 +0200
@@ -8,7 +8,7 @@
  *
  */
 
-#if defined(sparc) || defined(__FreeBSD__)
+#if defined(sparc) || defined(__FreeBSD_kernel__)
 /* XXX: gross.   netbsd/sparc doesn't have machine/bus.h yet. */
 typedef void * bus_space_tag_t;
 typedef u_int32_t pci_chipset_tag_t;
diff -ur kfreebsd-headers-4.6/import.old/dev/en/midwayvar.h kfreebsd-headers-4.6/import/dev/en/midwayvar.h
--- kfreebsd-headers-4.6/import.old/dev/en/midwayvar.h	2004-05-31 03:25:30.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/en/midwayvar.h	2004-05-31 04:06:28.000000000 +0200
@@ -68,7 +68,7 @@
 #define EN_IOCTL_CMDT int
 #endif
 
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 
 #define EN_INTR_TYPE void
 #define EN_INTR_RET(X) return
diff -ur kfreebsd-headers-4.6/import.old/dev/nsp/nspvar.h kfreebsd-headers-4.6/import/dev/nsp/nspvar.h
--- kfreebsd-headers-4.6/import.old/dev/nsp/nspvar.h	2004-05-31 03:25:55.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/nsp/nspvar.h	2004-05-31 04:06:38.000000000 +0200
@@ -52,7 +52,7 @@
 	void *sc_ih;
 #endif	/* __NetBSD__ */
 
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 	bus_space_tag_t sc_iot;
 	bus_space_handle_t sc_ioh;
 	bus_space_tag_t sc_memt;
diff -ur kfreebsd-headers-4.6/import.old/dev/pdq/pdqvar.h kfreebsd-headers-4.6/import/dev/pdq/pdqvar.h
--- kfreebsd-headers-4.6/import.old/dev/pdq/pdqvar.h	2004-05-31 03:25:57.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/pdq/pdqvar.h	2004-05-31 04:06:38.000000000 +0200
@@ -58,7 +58,7 @@
 
 #if defined(PDQTEST)
 #include <pdq_os_test.h>
-#elif defined(__FreeBSD__) || defined(__bsdi__) || defined(__NetBSD__)
+#elif defined(__FreeBSD_kernel__) || defined(__bsdi__) || defined(__NetBSD__)
 
 #include <sys/param.h>
 #include <sys/systm.h>
@@ -77,7 +77,7 @@
 #define	PDQ_OS_PREFIX			"%s%d: "
 #define	PDQ_OS_PREFIX_ARGS		pdq->pdq_os_name, pdq->pdq_unit
 #endif
-#if defined(__FreeBSD__) && BSD >= 199506
+#if defined(__FreeBSD_kernel__) && BSD >= 199506
 #define	PDQ_OS_PAGESIZE			PAGE_SIZE
 #else
 #define	PDQ_OS_PAGESIZE			NBPG
@@ -86,14 +86,14 @@
 #define	PDQ_OS_MEMZERO(p, n)		bzero((caddr_t)(p), (n))
 #if defined(__NetBSD__) && defined(__alpha__)
 #define	PDQ_OS_VA_TO_PA(pdq, p)		(vtophys((vm_offset_t)p) | (pdq->pdq_type == PDQ_DEFTA ? 0 : 0x40000000))
-#elif defined(__FreeBSD__) && defined(__alpha__)
+#elif defined(__FreeBSD_kernel__) && defined(__alpha__)
 #define	PDQ_OS_VA_TO_PA(pdq, p)		(vtophys((vm_offset_t)p) | (pdq->pdq_type == PDQ_DEFTA ? 0 : alpha_XXX_dmamap_or))
 #else
 #define	PDQ_OS_VA_TO_PA(pdq, p)		vtophys(p)
 #endif
 #define	PDQ_OS_MEMALLOC(n)		malloc(n, M_DEVBUF, M_NOWAIT)
 #define	PDQ_OS_MEMFREE(p, n)		free((void *) p, M_DEVBUF)
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #define	PDQ_OS_MEMALLOC_CONTIG(n)	vm_page_alloc_contig(n, 0, 0xffffffff, PAGE_SIZE)
 #define	PDQ_OS_MEMFREE_CONTIG(p, n)	kmem_free(kernel_map, (vm_offset_t) p, n)
 #else
@@ -101,7 +101,7 @@
 #define	PDQ_OS_MEMFREE_CONTIG(p, n)	kmem_free(kernel_map, (vm_offset_t) p, n)
 #endif /* __FreeBSD__ */
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 #include <vm/pmap.h>
 #include <vm/vm_extern.h>
 #include <machine/bus.h>
@@ -203,7 +203,7 @@
     struct device sc_dev;		/* base device */
     void *sc_ih;			/* interrupt vectoring */
     void *sc_ats;			/* shutdown hook */
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
     struct kern_devconf *sc_kdc;	/* freebsd cruft */
 #endif
     struct arpcom sc_ac;
diff -ur kfreebsd-headers-4.6/import.old/dev/snc/dp83932var.h kfreebsd-headers-4.6/import/dev/snc/dp83932var.h
--- kfreebsd-headers-4.6/import.old/dev/snc/dp83932var.h	2004-05-31 03:26:03.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/snc/dp83932var.h	2004-05-31 04:06:40.000000000 +0200
@@ -34,7 +34,7 @@
 #endif
 
 /* NetBSD Emulation */
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #ifndef NBPG
 #define NBPG PAGE_SIZE
 #endif
diff -ur kfreebsd-headers-4.6/import.old/dev/twe/twe_compat.h kfreebsd-headers-4.6/import/dev/twe/twe_compat.h
--- kfreebsd-headers-4.6/import.old/dev/twe/twe_compat.h	2004-05-31 03:26:15.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/twe/twe_compat.h	2004-05-31 04:06:44.000000000 +0200
@@ -30,7 +30,7 @@
  * Portability and compatibility interfaces.
  */
 
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 /******************************************************************************
  * FreeBSD
  */
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/dsbr100io.h kfreebsd-headers-4.6/import/dev/usb/dsbr100io.h
--- kfreebsd-headers-4.6/import.old/dev/usb/dsbr100io.h	2004-05-31 03:26:17.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/dsbr100io.h	2004-05-31 04:06:44.000000000 +0200
@@ -30,11 +30,11 @@
 
 /*  $FreeBSD: src/sys/dev/usb/dsbr100io.h,v 1.1.2.1 2002/03/04 04:01:35 alfred Exp $ */
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <sys/ioccom.h>
 #endif
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #define FM_SET_FREQ	_IOWR('U', 200, int)
 #define FM_GET_FREQ	_IOWR('U', 201, int)
 #define FM_START	_IOWR('U', 202, int)
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/ohcivar.h kfreebsd-headers-4.6/import/dev/usb/ohcivar.h
--- kfreebsd-headers-4.6/import.old/dev/usb/ohcivar.h	2004-05-31 03:26:19.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/ohcivar.h	2004-05-31 04:06:45.000000000 +0200
@@ -79,7 +79,7 @@
 	bus_space_tag_t iot;
 	bus_space_handle_t ioh;
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 	void *ih;
 
 	struct resource *io_res;
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/rio500_usb.h kfreebsd-headers-4.6/import/dev/usb/rio500_usb.h
--- kfreebsd-headers-4.6/import.old/dev/usb/rio500_usb.h	2004-05-31 03:26:19.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/rio500_usb.h	2004-05-31 04:06:45.000000000 +0200
@@ -21,7 +21,7 @@
 
 /*  $FreeBSD: src/sys/dev/usb/rio500_usb.h,v 1.1.2.1 2001/09/27 17:43:05 alfred Exp $ */
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <sys/ioccom.h>
 #ifndef USB_VENDOR_DIAMOND
 #define USB_VENDOR_DIAMOND 0x841
@@ -33,7 +33,7 @@
 
 struct RioCommand
 {
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__)
   u_int16_t  length;
 #else
   short length;
@@ -46,7 +46,7 @@
   int  timeout;
 };
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #define RIO_SEND_COMMAND	_IOWR('U', 200, struct RioCommand)
 #define RIO_RECV_COMMAND	_IOWR('U', 201, struct RioCommand)
 #else
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/uhcivar.h kfreebsd-headers-4.6/import/dev/usb/uhcivar.h
--- kfreebsd-headers-4.6/import.old/dev/usb/uhcivar.h	2004-05-31 03:26:19.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/uhcivar.h	2004-05-31 04:06:45.000000000 +0200
@@ -74,9 +74,9 @@
 	uhci_soft_td_t *stdstart;
 	uhci_soft_td_t *stdend;
 	LIST_ENTRY(uhci_intr_info) list;
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 	struct callout_handle timeout_handle;
-#endif /* defined(__FreeBSD__) */
+#endif /* defined(__FreeBSD_kernel__) */
 #ifdef DIAGNOSTIC
 	int isdone;
 #endif
@@ -130,7 +130,7 @@
 	struct usbd_bus sc_bus;		/* base device */
 	bus_space_tag_t iot;
 	bus_space_handle_t ioh;
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 	void *ih;
 
 	struct resource *io_res;
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/usb.h kfreebsd-headers-4.6/import/dev/usb/usb.h
--- kfreebsd-headers-4.6/import.old/dev/usb/usb.h	2004-05-31 03:26:19.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/usb.h	2004-05-31 04:06:45.000000000 +0200
@@ -52,7 +52,7 @@
 #include <dev/usb/usb_port.h>
 #endif /* _KERNEL */
 
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 #if defined(_KERNEL)
 #include <sys/malloc.h>
 
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/usb_mem.h kfreebsd-headers-4.6/import/dev/usb/usb_mem.h
--- kfreebsd-headers-4.6/import.old/dev/usb/usb_mem.h	2004-05-31 03:26:19.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/usb_mem.h	2004-05-31 04:06:45.000000000 +0200
@@ -57,7 +57,7 @@
 usbd_status	usb_allocmem(usbd_bus_handle,size_t,size_t, usb_dma_t *);
 void		usb_freemem(usbd_bus_handle, usb_dma_t *);
 
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 
 /* 
  * FreeBSD does not have special functions for dma memory, so let's keep it
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/usb_port.h kfreebsd-headers-4.6/import/dev/usb/usb_port.h
--- kfreebsd-headers-4.6/import.old/dev/usb/usb_port.h	2004-05-31 03:26:19.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/usb_port.h	2004-05-31 04:06:45.000000000 +0200
@@ -270,7 +270,7 @@
 #define USB_DO_ATTACH(dev, bdev, parent, args, print, sub) \
 	(config_found_sm(parent, args, print, sub))
 
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 /*
  * FreeBSD
  */
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/usbdi.h kfreebsd-headers-4.6/import/dev/usb/usbdi.h
--- kfreebsd-headers-4.6/import.old/dev/usb/usbdi.h	2004-05-31 03:26:21.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/usbdi.h	2004-05-31 04:06:46.000000000 +0200
@@ -91,7 +91,7 @@
 #define USBD_NO_TIMEOUT 0
 #define USBD_DEFAULT_TIMEOUT 5000 /* ms = 5 s */
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 #define USB_CDEV_MAJOR 108
 #endif
 
@@ -233,7 +233,7 @@
 /* No match */
 #define UMATCH_NONE					 0
 
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 /* FreeBSD needs values less than zero */
 #define UMATCH_VENDOR_PRODUCT_REV			(-10)
 #define UMATCH_VENDOR_PRODUCT				(-20)
@@ -253,7 +253,7 @@
 
 #endif
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 int usbd_driver_load(module_t mod, int what, void *arg);
 #endif
 
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/usbdi_util.h kfreebsd-headers-4.6/import/dev/usb/usbdi_util.h
--- kfreebsd-headers-4.6/import.old/dev/usb/usbdi_util.h	2004-05-31 03:26:21.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/usbdi_util.h	2004-05-31 04:06:46.000000000 +0200
@@ -72,7 +72,7 @@
 #if defined(__NetBSD__) || defined(__OpenBSD__)
 usbd_status	usbd_alloc_report_desc
 	(usbd_interface_handle ifc, void **descp, int *sizep, int mem);
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 usbd_status	usbd_alloc_report_desc
 	(usbd_interface_handle ifc, void **descp, int *sizep, struct malloc_type * mem);
 #endif
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/usbdivar.h kfreebsd-headers-4.6/import/dev/usb/usbdivar.h
--- kfreebsd-headers-4.6/import.old/dev/usb/usbdivar.h	2004-05-31 03:26:21.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/usbdivar.h	2004-05-31 04:06:46.000000000 +0200
@@ -199,7 +199,7 @@
 	void		       *hcpriv; /* private use by the HC driver */
 	int			hcprivint;
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 	struct callout_handle  timo_handle;
 #endif
 };
@@ -251,7 +251,7 @@
 
 #if defined(__NetBSD__)
 #include "locators.h"
-#elif defined(__FreeBSD__) || defined(__OpenBSD__)
+#elif defined(__FreeBSD_kernel__) || defined(__OpenBSD__)
 /* XXX these values are used to statically bind some elements in the USB tree
  * to specific driver instances. This should be somehow emulated in FreeBSD
  * but can be done later on.
diff -ur kfreebsd-headers-4.6/import.old/dev/usb/usbhid.h kfreebsd-headers-4.6/import/dev/usb/usbhid.h
--- kfreebsd-headers-4.6/import.old/dev/usb/usbhid.h	2004-05-31 03:26:21.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/usb/usbhid.h	2004-05-31 04:06:46.000000000 +0200
@@ -54,7 +54,7 @@
 #define UR_GET_PROTOCOL		0x03
 #define UR_SET_PROTOCOL		0x0b
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 #define UPACKED __attribute__ ((packed))
 #endif
 
diff -ur kfreebsd-headers-4.6/import.old/dev/wi/if_wireg.h kfreebsd-headers-4.6/import/dev/wi/if_wireg.h
--- kfreebsd-headers-4.6/import.old/dev/wi/if_wireg.h	2004-05-31 03:26:24.000000000 +0200
+++ kfreebsd-headers-4.6/import/dev/wi/if_wireg.h	2004-05-31 04:06:47.000000000 +0200
@@ -86,7 +86,7 @@
 #ifdef __NetBSD__
 #define OS_STRING_NAME	"NetBSD"
 #endif
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #define OS_STRING_NAME	"FreeBSD"
 #endif
 #ifdef __OpenBSD__
diff -ur kfreebsd-headers-4.6/import.old/i386/machine/i4b_ioctl.h kfreebsd-headers-4.6/import/i386/machine/i4b_ioctl.h
--- kfreebsd-headers-4.6/import.old/i386/machine/i4b_ioctl.h	2004-05-31 03:28:07.000000000 +0200
+++ kfreebsd-headers-4.6/import/i386/machine/i4b_ioctl.h	2004-05-31 04:07:16.000000000 +0200
@@ -36,7 +36,7 @@
 #ifndef _I4B_IOCTL_H_
 #define _I4B_IOCTL_H_
 
-#if defined(__FreeBSD__) && __FreeBSD__ >= 3
+#if defined(__FreeBSD_kernel__) && __FreeBSD__ >= 3
 #ifndef _MACHINE_TYPES_H_
 #include <machine/types.h>
 #endif /* _MACHINE_TYPES_H_ */
diff -ur kfreebsd-headers-4.6/import.old/i386/machine/i4b_isppp.h kfreebsd-headers-4.6/import/i386/machine/i4b_isppp.h
--- kfreebsd-headers-4.6/import.old/i386/machine/i4b_isppp.h	2004-05-31 03:28:07.000000000 +0200
+++ kfreebsd-headers-4.6/import/i386/machine/i4b_isppp.h	2004-05-31 04:07:16.000000000 +0200
@@ -31,7 +31,7 @@
 
 #ifdef SPPP_VJ
 #if !(defined (KERNEL) || defined (_KERNEL))
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #if 0
 /*
  * this is needed on FreeBSD to make /usr/src/usr.bin/kdump and
@@ -123,7 +123,7 @@
 	u_char  confid[IDX_COUNT];	/* id of last configuration request */
 	int	rst_counter[IDX_COUNT];	/* restart counter */
 	int	fail_counter[IDX_COUNT]; /* negotiation failure counter */
-#if defined(__FreeBSD__) && __FreeBSD__ >= 3
+#if defined(__FreeBSD_kernel__) && __FreeBSD__ >= 3
 	struct callout_handle ch[IDX_COUNT]; /* per-proto and if callouts */
 	struct callout_handle pap_my_to_ch; /* PAP needs one more... */
 #endif
@@ -214,7 +214,7 @@
 #if defined(__FreeBSD_version) && __FreeBSD_version >= 300003
 int sppp_ioctl (struct ifnet *ifp, u_long cmd, void *data);
 #else
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 int sppp_ioctl (struct ifnet *ifp, int cmd, void *data);
 #else
 int sppp_ioctl (struct ifnet *ifp, u_long cmd, void *data);
@@ -235,7 +235,7 @@
 #if defined(__FreeBSD_version) && __FreeBSD_version >= 300003
 int isppp_ioctl (struct ifnet *ifp, u_long cmd, void *data);
 #else
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 int isppp_ioctl (struct ifnet *ifp, int cmd, void *data);
 #else
 int isppp_ioctl (struct ifnet *ifp, u_long cmd, void *data);
diff -ur kfreebsd-headers-4.6/import.old/i386/machine/physio_proc.h kfreebsd-headers-4.6/import/i386/machine/physio_proc.h
--- kfreebsd-headers-4.6/import.old/i386/machine/physio_proc.h	2004-05-31 03:28:12.000000000 +0200
+++ kfreebsd-headers-4.6/import/i386/machine/physio_proc.h	2004-05-31 04:07:18.000000000 +0200
@@ -61,7 +61,7 @@
 
 	s = splstatclock();
 	TAILQ_REMOVE(&physio_proc_freet, pp, pp_chain);
-#if !defined(__FreeBSD__) || __FreeBSD_version < 400001
+#if !defined(__FreeBSD_kernel__) || __FreeBSD_version < 400001
 	pp->pp_proc = bp->b_proc;
 #endif
 	TAILQ_INSERT_TAIL(&physio_proc_busyt, pp, pp_chain);
Only in kfreebsd-headers-4.6/import.old/net: if.h.orig
diff -ur kfreebsd-headers-4.6/import.old/net/if_atm.h kfreebsd-headers-4.6/import/net/if_atm.h
--- kfreebsd-headers-4.6/import.old/net/if_atm.h	2004-05-31 03:26:31.000000000 +0200
+++ kfreebsd-headers-4.6/import/net/if_atm.h	2004-05-31 04:06:48.000000000 +0200
@@ -39,7 +39,7 @@
 
 #if defined(__NetBSD__) || defined(__OpenBSD__) || defined(__bsdi__)
 #define RTALLOC1(A,B)		rtalloc1((A),(B))
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 #define RTALLOC1(A,B)		rtalloc1((A),(B),0UL)
 #endif
 
diff -ur kfreebsd-headers-4.6/import.old/net/net_osdep.h kfreebsd-headers-4.6/import/net/net_osdep.h
--- kfreebsd-headers-4.6/import.old/net/net_osdep.h	2004-05-31 03:26:35.000000000 +0200
+++ kfreebsd-headers-4.6/import/net/net_osdep.h	2004-05-31 04:06:49.000000000 +0200
@@ -209,7 +209,7 @@
  *	others: bpfilter.h, NBPFILTER
  *	FreeBSD4: bpf.h, NBPF
  *	solution:
- *		#if defined(__FreeBSD__) && __FreeBSD__ >= 4
+ *		#if defined(__FreeBSD_kernel__) && __FreeBSD__ >= 4
  *		#include "bpf.h"
  *		#define NBPFILTER	NBPF
  *		#else
diff -ur kfreebsd-headers-4.6/import.old/net/zlib.h kfreebsd-headers-4.6/import/net/zlib.h
--- kfreebsd-headers-4.6/import.old/net/zlib.h	2004-05-31 03:26:37.000000000 +0200
+++ kfreebsd-headers-4.6/import/net/zlib.h	2004-05-31 04:06:50.000000000 +0200
@@ -507,7 +507,7 @@
    done by inflate().
 */
 
-#if defined(__FreeBSD__) && defined(_KERNEL)
+#if defined(__FreeBSD_kernel__) && defined(_KERNEL)
 #define inflate       inflate_ppp     /* FreeBSD already has an inflate :-( */
 #endif
 
diff -ur kfreebsd-headers-4.6/import.old/netatm/atm_if.h kfreebsd-headers-4.6/import/netatm/atm_if.h
--- kfreebsd-headers-4.6/import.old/netatm/atm_if.h	2004-05-31 03:26:40.000000000 +0200
+++ kfreebsd-headers-4.6/import/netatm/atm_if.h	2004-05-31 04:06:51.000000000 +0200
@@ -356,7 +356,7 @@
 #if defined(sun)
 #define	DEVICE_LOCK(u)		((u)->cu_savepri = splr((u)->cu_intrpri))
 #endif
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 #define	DEVICE_LOCK(u)		((u)->cu_savepri = splimp())
 #endif
 #define	DEVICE_UNLOCK(u)	((void) splx((u)->cu_savepri))
@@ -379,7 +379,7 @@
 #ifdef sun
 #define	SCHED_ATM	schednetisr(atm_intr) 
 #endif
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #define	NETISR_ATM	AF_ATM
 #define	SCHED_ATM	schednetisr(NETISR_ATM) 
 #endif
diff -ur kfreebsd-headers-4.6/import.old/netatm/atm_var.h kfreebsd-headers-4.6/import/netatm/atm_var.h
--- kfreebsd-headers-4.6/import.old/netatm/atm_var.h	2004-05-31 03:26:43.000000000 +0200
+++ kfreebsd-headers-4.6/import/netatm/atm_var.h	2004-05-31 04:06:51.000000000 +0200
@@ -44,7 +44,7 @@
  * Global variable declarations
  */
 	/* atm_aal5.c */
-#if (defined(__FreeBSD__) && (BSD >= 199506))
+#if (defined(__FreeBSD_kernel__) && (BSD >= 199506))
 extern struct pr_usrreqs	atm_aal5_usrreqs;
 #endif
 
@@ -72,7 +72,7 @@
 extern struct sp_info	atm_attributes_pool;
 
 	/* atm_usrreq.c */
-#if (defined(__FreeBSD__) && (BSD >= 199506))
+#if (defined(__FreeBSD_kernel__) && (BSD >= 199506))
 extern struct pr_usrreqs	atm_dgram_usrreqs;
 #endif
 
@@ -147,7 +147,7 @@
 		atm_nifname __P((char *));
 
 	/* atm_proto.c */
-#if (defined(__FreeBSD__) && (BSD >= 199506))
+#if (defined(__FreeBSD_kernel__) && (BSD >= 199506))
 int		atm_proto_notsupp1 __P((struct socket *));
 int		atm_proto_notsupp2 __P((struct socket *, struct sockaddr *,
 			struct proc *));
@@ -198,7 +198,7 @@
 void		atm_pdu_print __P((KBuffer *, char *));
 
 	/* atm_usrreq.c */
-#if (!(defined(__FreeBSD__) && (BSD >= 199506)))
+#if (!(defined(__FreeBSD_kernel__) && (BSD >= 199506)))
 int		atm_dgram_usrreq __P((struct socket *, int, KBuffer *,
 			KBuffer *, KBuffer *));
 #endif
diff -ur kfreebsd-headers-4.6/import.old/netatm/kern_include.h kfreebsd-headers-4.6/import/netatm/kern_include.h
--- kfreebsd-headers-4.6/import.old/netatm/kern_include.h	2004-05-31 03:26:44.000000000 +0200
+++ kfreebsd-headers-4.6/import/netatm/kern_include.h	2004-05-31 04:06:52.000000000 +0200
@@ -68,7 +68,7 @@
 #include <sundev/mbvar.h>
 #endif
 
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #include <sys/eventhandler.h>
 #include <machine/clock.h>
 #include <vm/vm.h>
diff -ur kfreebsd-headers-4.6/import.old/ntfs/ntfs_inode.h kfreebsd-headers-4.6/import/ntfs/ntfs_inode.h
--- kfreebsd-headers-4.6/import.old/ntfs/ntfs_inode.h	2004-05-31 03:27:14.000000000 +0200
+++ kfreebsd-headers-4.6/import/ntfs/ntfs_inode.h	2004-05-31 04:07:00.000000000 +0200
@@ -29,7 +29,7 @@
  */
 
 /* These flags are kept in i_flag. */
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD_kernel__)
 #define	IN_ACCESS	0x0001	/* Access time update request. */
 #define	IN_CHANGE	0x0002	/* Inode change time update request. */
 #define	IN_UPDATE	0x0004	/* Modification time update request. */
@@ -84,7 +84,7 @@
 #define	FN_VALID	0x0002
 #define	FN_AATTRNAME	0x0004	/* space allocated for f_attrname */
 struct fnode {
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 	struct lock	f_lock;	/* fnode lock >Keep this first< */
 #endif
 	
diff -ur kfreebsd-headers-4.6/import.old/sys/device_port.h kfreebsd-headers-4.6/import/sys/device_port.h
--- kfreebsd-headers-4.6/import.old/sys/device_port.h	2004-05-31 03:27:23.000000000 +0200
+++ kfreebsd-headers-4.6/import/sys/device_port.h	2004-05-31 04:07:02.000000000 +0200
@@ -29,7 +29,7 @@
 
 #if defined(__NetBSD__)
 # include <sys/device.h>
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 # if __FreeBSD_version >= 400001
 #  include <sys/module.h>
 #  include <sys/bus.h>
@@ -47,7 +47,7 @@
 # define DEVPORT_DEVNAME(dev)		(dev).dv_xname
 # define DEVPORT_DEVUNIT(dev)		(dev).dv_unit
 
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD_kernel__)
 /*
  * FreeBSD (compatibility for struct device)
  */
