--- kfreebsd-headers-4.6.old/import/sys/ioctl_compat.h	2004-05-31 03:27:29.000000000 +0200
+++ kfreebsd-headers-4.6/import/sys/ioctl_compat.h	2004-07-29 01:44:09.000000000 +0200
@@ -45,38 +45,6 @@
 #include <sys/ttychars.h>
 #include <sys/ttydev.h>
 
-struct tchars {
-	char	t_intrc;	/* interrupt */
-	char	t_quitc;	/* quit */
-	char	t_startc;	/* start output */
-	char	t_stopc;	/* stop output */
-	char	t_eofc;		/* end-of-file */
-	char	t_brkc;		/* input delimiter (like nl) */
-};
-
-struct ltchars {
-	char	t_suspc;	/* stop process signal */
-	char	t_dsuspc;	/* delayed stop process signal */
-	char	t_rprntc;	/* reprint line */
-	char	t_flushc;	/* flush output (toggles) */
-	char	t_werasc;	/* word erase */
-	char	t_lnextc;	/* literal next character */
-};
-
-/*
- * Structure for TIOCGETP and TIOCSETP ioctls.
- */
-#ifndef _SGTTYB_
-#define	_SGTTYB_
-struct sgttyb {
-	char	sg_ispeed;		/* input speed */
-	char	sg_ospeed;		/* output speed */
-	char	sg_erase;		/* erase character */
-	char	sg_kill;		/* kill character */
-	short	sg_flags;		/* mode flags */
-};
-#endif
-
 #ifdef USE_OLD_TTY
 # undef  TIOCGETD
 # define TIOCGETD	_IOR('t', 0, int)	/* get line discipline */
