diff -ur src.old/sys/sys/blist.h src/sys/sys/blist.h
--- src.old/sys/sys/blist.h	1999-08-28 02:51:33.000000000 +0200
+++ src/sys/sys/blist.h	2004-08-07 01:28:11.000000000 +0200
@@ -30,6 +30,10 @@
 #ifndef _SYS_BLIST_H_
 #define _SYS_BLIST_H_
 
+#include <sys/types.h>
+
+typedef u_int32_t u_daddr_t;
+
 #define LOG2(v)		(((u_daddr_t)(v) >= 0x80000000U) ? 31 : \
 			((u_daddr_t)(v) >= 0x40000000U) ? 30 : \
 			((u_daddr_t)(v) >= 0x20000000U) ? 29 : \
diff -ur src.old/sys/sys/sysctl.h src/sys/sys/sysctl.h
--- src.old/sys/sys/sysctl.h	2004-08-07 01:19:50.000000000 +0200
+++ src/sys/sys/sysctl.h	2004-08-07 01:31:39.000000000 +0200
@@ -389,39 +389,7 @@
 				   as next argument */
 
 
-/* Other sysctl identifiers from <vm/vm_param.h>.  */
-
-/*
- * CTL_VM identifiers
- */
-#define	VM_METER		1	/* struct vmmeter */
-#define	VM_LOADAVG	 	2	/* struct loadavg */
-#define VM_V_FREE_MIN		3	/* cnt.v_free_min */
-#define VM_V_FREE_TARGET	4	/* cnt.v_free_target */
-#define VM_V_FREE_RESERVED	5	/* cnt.v_free_reserved */
-#define VM_V_INACTIVE_TARGET	6	/* cnt.v_inactive_target */
-#define VM_V_CACHE_MIN		7	/* cnt.v_cache_max */
-#define VM_V_CACHE_MAX		8	/* cnt.v_cache_min */
-#define VM_V_PAGEOUT_FREE_MIN	9	/* cnt.v_pageout_free_min */
-#define	VM_PAGEOUT_ALGORITHM	10	/* pageout algorithm */
-#define VM_SWAPPING_ENABLED	11	/* swapping enabled */
-#define	VM_MAXID		12	/* number of valid vm ids */
-
-#define CTL_VM_NAMES { \
-	{ 0, 0 }, \
-	{ "vmmeter", CTLTYPE_STRUCT }, \
-	{ "loadavg", CTLTYPE_STRUCT }, \
-	{ "v_free_min", CTLTYPE_INT }, \
-	{ "v_free_target", CTLTYPE_INT }, \
-	{ "v_free_reserved", CTLTYPE_INT }, \
-	{ "v_inactive_target", CTLTYPE_INT }, \
-	{ "v_cache_min", CTLTYPE_INT }, \
-	{ "v_cache_max", CTLTYPE_INT }, \
-	{ "v_pageout_free_min", CTLTYPE_INT}, \
-	{ "pageout_algorithm", CTLTYPE_INT}, \
-	{ "swapping_enabled", CTLTYPE_INT},\
-}
-
+#include "/usr/src/kfreebsd5-headers/vm/vm_param.h"
 
 /* Other sysctl identifiers from <sys/socket.h>.  */
 
