From 661a052926486d972423ff3c1a06b613b787b233 Mon Sep 17 00:00:00 2001
From: Laurent Fousse <laurent@persil.lateralis.org>
Date: Fri, 17 Dec 2010 15:59:48 +0100
Subject: Include missing header <stdint.h>.

---
 lib/libmd/sha256.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/lib/libmd/sha256.h b/lib/libmd/sha256.h
index ce51787..930a121 100644
--- a/lib/libmd/sha256.h
+++ b/lib/libmd/sha256.h
@@ -30,6 +30,7 @@
 #define _SHA256_H_
 
 #include <sys/types.h>
+#include <stdint.h>
 
 typedef struct SHA256Context {
 	uint32_t state[8];
--·
1.7.2.3

