
Author: rmh
Status: in BTS

diff -ur sysvinit-2.86.ds1.old/debian/initscripts/etc/init.d/mountvirtfs sysvinit-2.86.ds1/debian/initscripts/etc/init.d/mountvirtfs
--- sysvinit-2.86.ds1.old/debian/initscripts/etc/init.d/mountvirtfs	2005-09-06 14:13:10.000000000 +0200
+++ sysvinit-2.86.ds1/debian/initscripts/etc/init.d/mountvirtfs	2005-09-06 14:13:18.000000000 +0200
@@ -67,6 +67,9 @@
 			Linux|GNU)
 				TYPE=proc
 				;;
+			*FreeBSD)
+				TYPE=linprocfs
+				;;
 			*)
 				TYPE=procfs
 				;;
diff -ur sysvinit-2.86.ds1.old/debian/initscripts/etc/init.d/umountfs sysvinit-2.86.ds1/debian/initscripts/etc/init.d/umountfs
--- sysvinit-2.86.ds1.old/debian/initscripts/etc/init.d/umountfs	2005-09-06 14:13:10.000000000 +0200
+++ sysvinit-2.86.ds1/debian/initscripts/etc/init.d/umountfs	2005-09-06 14:13:18.000000000 +0200
@@ -32,7 +32,7 @@
 	esac
 
         case $TYPE in 
-        proc|procfs|devfs|sysfs|usbfs|usbdevfs|devpts)
+        proc|procfs|linprocfs|devfs|sysfs|usbfs|usbdevfs|devpts)
 	    continue # Ignoring non-tmpfs virtual file systems
             ;;
         esac
diff -ur sysvinit-2.86.ds1.old/debian/initscripts/etc/init.d/umountnfs.sh sysvinit-2.86.ds1/debian/initscripts/etc/init.d/umountnfs.sh
--- sysvinit-2.86.ds1.old/debian/initscripts/etc/init.d/umountnfs.sh	2005-09-06 14:13:10.000000000 +0200
+++ sysvinit-2.86.ds1/debian/initscripts/etc/init.d/umountnfs.sh	2005-09-06 14:13:18.000000000 +0200
@@ -63,7 +63,7 @@
 				nfs|nfs4|smbfs|ncp|ncpfs|cifs|coda|ocfs2|gfs)
 					DIRS="$DIR $DIRS"
 					;;
-				proc|procfs|devfs|devpts|usbfs|usbdevfs|sysfs)
+				proc|procfs|linprocfs|devfs|devpts|usbfs|usbdevfs|sysfs)
 					DIRS="$DIR $DIRS"
 					;;
 			esac
