
Author: rmh
Status: in BTS

diff -ur sysvinit-2.86.ds1/src/ifdown.c sysvinit-2.86.ds1.new/src/ifdown.c
--- sysvinit-2.86.ds1/src/ifdown.c	1998-06-02 22:41:47.000000000 +0200
+++ sysvinit-2.86.ds1.new/src/ifdown.c	2005-09-06 14:50:21.000000000 +0200
@@ -29,6 +29,7 @@
  */
 int ifdown(void)
 {
+#ifdef __linux__
 	struct ifreq ifr[MAX_IFS];
 	struct ifconf ifc;
 	int i, fd;
@@ -69,6 +70,7 @@
 		}
 	}
 	close(fd);
+#endif  /* __linux__ */
 
 	return 0;
 }
