--- a/sys/mips/malta/gt_pci.c
+++ b/sys/mips/malta/gt_pci.c
@@ -401,7 +401,22 @@
 	}
 
 	/* Initialize memory and i/o rmans. */
-	device_add_child(dev, "pci", busno);
+	{
+		device_t child;
+		int rid;
+		void *res;
+
+		child = device_add_child(dev, "pci", busno);
+
+		res = bus_alloc_resource(child, SYS_RES_IOPORT, &rid, IO_ICU1, IO_ICU1 + IO_ICUSIZE, IO_ICUSIZE, RF_ACTIVE);
+		if (res == NULL)
+			device_printf(child, "cannot allocate registers\n");
+
+		res = bus_alloc_resource(child, SYS_RES_IOPORT, &rid, IO_ICU2, IO_ICU2 + IO_ICUSIZE, IO_ICUSIZE, RF_ACTIVE);
+		if (res == NULL)
+			device_printf(child, "cannot allocate registers\n");
+	}
+
 	return (bus_generic_attach(dev));
 }
 
