--- a/sys/conf/files.i386
+++ b/sys/conf/files.i386
@@ -46,27 +46,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci				\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/i386-elf.raid.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptmv/i386-elf.raid.o.uu"	\
-	no-implicit-rule
-#
-hptrr_lib.o			optional	hptrr			\
-	dependency	"$S/dev/hptrr/i386-elf.hptrr_lib.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptrr/i386-elf.hptrr_lib.o.uu" \
-	no-implicit-rule
-#
 compat/linprocfs/linprocfs.c	optional linprocfs
 compat/linsysfs/linsysfs.c	optional linsysfs
 compat/linux/linux_emul.c	optional compat_linux
@@ -174,14 +153,6 @@
 dev/glxiic/glxiic.c		optional glxiic
 dev/glxsb/glxsb.c		optional glxsb
 dev/glxsb/glxsb_hash.c		optional glxsb
-dev/hptmv/entry.c		optional hptmv
-dev/hptmv/mv.c			optional hptmv
-dev/hptmv/gui_lib.c		optional hptmv
-dev/hptmv/hptproc.c		optional hptmv
-dev/hptmv/ioctl.c		optional hptmv
-dev/hptrr/hptrr_os_bsd.c	optional hptrr
-dev/hptrr/hptrr_osm_bsd.c	optional hptrr
-dev/hptrr/hptrr_config.c	optional hptrr
 dev/hwpmc/hwpmc_amd.c		optional hwpmc
 dev/hwpmc/hwpmc_intel.c		optional hwpmc
 dev/hwpmc/hwpmc_core.c		optional hwpmc
@@ -214,7 +185,6 @@
 dev/mse/mse.c			optional mse
 dev/mse/mse_isa.c		optional mse isa
 dev/nfe/if_nfe.c		optional nfe pci
-dev/nve/if_nve.c		optional nve pci
 dev/nvram/nvram.c		optional nvram isa
 dev/pcf/pcf_isa.c		optional pcf
 dev/random/nehemiah.c		optional random
--- a/sys/i386/conf/GENERIC
+++ b/sys/i386/conf/GENERIC
@@ -141,8 +141,6 @@
 device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		hptrr		# Highpoint RocketRAID 17xx, 22xx, 23xx, 25xx
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
--- a/sys/modules/Makefile
+++ b/sys/modules/Makefile
@@ -108,8 +108,6 @@
 	hifn \
 	hme \
 	${_hptiop} \
-	${_hptmv} \
-	${_hptrr} \
 	hwpmc \
 	${_i2c} \
 	${_ibcs2} \
@@ -229,7 +227,6 @@
 	ntfs \
 	ntfs_iconv \
 	nullfs \
-	${_nve} \
 	${_nvram} \
 	${_nwfs} \
 	${_nxge} \
@@ -500,8 +497,6 @@
 _dpt=		dpt
 _ex=		ex
 _hptiop=	hptiop
-_hptmv=		hptmv
-_hptrr=		hptrr
 _ichwd=		ichwd
 _ida=		ida
 _iir=		iir
@@ -517,7 +512,6 @@
 _ixgbe=		ixgbe
 _mly=		mly
 _nfe=		nfe
-_nve=		nve
 _nvram=		nvram
 _nxge=		nxge
 _tpm=		tpm
@@ -573,8 +567,6 @@
 _exca=		exca
 _ext2fs=	ext2fs
 _hptiop=	hptiop
-_hptmv=		hptmv
-_hptrr=		hptrr
 _i2c=		i2c
 _ichwd=		ichwd
 _ida=		ida
@@ -605,7 +597,6 @@
 .endif
 _ndis=		ndis
 _nfe=		nfe
-_nve=		nve
 _nvram=		nvram
 _nxge=		nxge
 .if ${MK_CDDL} != "no" || defined(ALL_MODULES)
--- a/sys/amd64/conf/GENERIC
+++ b/sys/amd64/conf/GENERIC
@@ -135,8 +135,6 @@
 #device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		hptrr		# Highpoint RocketRAID 17xx, 22xx, 23xx, 25xx
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
--- a/sys/conf/files.amd64
+++ b/sys/conf/files.amd64
@@ -47,26 +47,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/amd64-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/dev/hptmv/amd64-elf.raid.o.uu" \
-	no-implicit-rule
-hptrr_lib.o			optional	hptrr			\
-	dependency	"$S/dev/hptrr/amd64-elf.hptrr_lib.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptrr/amd64-elf.hptrr_lib.o.uu" \
-	no-implicit-rule
-#
 amd64/acpica/acpi_machdep.c	optional	acpi
 amd64/acpica/acpi_switch.S	optional	acpi
 acpi_wakecode.o			optional	acpi			\
@@ -187,14 +167,6 @@
 dev/fdc/fdc_acpi.c		optional	fdc
 dev/fdc/fdc_isa.c		optional	fdc isa
 dev/fdc/fdc_pccard.c		optional	fdc pccard
-dev/hptmv/entry.c		optional	hptmv
-dev/hptmv/mv.c			optional	hptmv
-dev/hptmv/gui_lib.c		optional	hptmv
-dev/hptmv/hptproc.c		optional	hptmv
-dev/hptmv/ioctl.c		optional	hptmv
-dev/hptrr/hptrr_os_bsd.c	optional	hptrr
-dev/hptrr/hptrr_osm_bsd.c	optional	hptrr
-dev/hptrr/hptrr_config.c	optional	hptrr
 dev/hwpmc/hwpmc_amd.c		optional	hwpmc
 dev/hwpmc/hwpmc_intel.c		optional	hwpmc
 dev/hwpmc/hwpmc_core.c		optional	hwpmc
@@ -206,7 +178,6 @@
 dev/lindev/full.c		optional	lindev
 dev/lindev/lindev.c		optional	lindev
 dev/nfe/if_nfe.c		optional	nfe pci
-dev/nve/if_nve.c		optional	nve pci
 dev/nvram/nvram.c		optional	nvram isa
 dev/sio/sio.c			optional	sio
 dev/sio/sio_isa.c		optional	sio isa
