--- a/src/share/mk/version_gen.awk
+++ b/src/share/mk/version_gen.awk
@@ -41,7 +41,7 @@
 	version_count = 0;
 	current_version = "";
 	stderr = "/dev/stderr";
-	while (getline < vfile) {
+	while (bsd_getline < vfile) {
 		# Strip comments.
 		sub("#.*$", "", $0);
 
--- a/src/usr.bin/file2c/file2c.c
+++ b/src/usr.bin/file2c/file2c.c
@@ -10,6 +10,7 @@
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
+#include <bsd/stdlib.h>
 #include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
--- a/src/usr.bin/file2c/Makefile
+++ b/src/usr.bin/file2c/Makefile
@@ -1,5 +1,6 @@
 # $FreeBSD$
 
 PROG=	file2c
+LDADD= -lbsd
 WARNS?=	6
 .include <bsd.prog.mk>
--- a/src/usr.bin/make/arch.c
+++ b/src/usr.bin/make/arch.c
@@ -88,7 +88,8 @@
  */
 
 #include <sys/param.h>
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
+#include <bsd/string.h>
 #include <sys/types.h>
 #include <ar.h>
 #include <ctype.h>
--- a/src/usr.bin/make/dir.h
+++ b/src/usr.bin/make/dir.h
@@ -43,7 +43,7 @@
 #ifndef dir_h_6002e3b8
 #define	dir_h_6002e3b8
 
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
 #include "hash.h"
 
 struct GNode;
--- a/src/usr.bin/make/job.c
+++ b/src/usr.bin/make/job.c
@@ -101,7 +101,8 @@
  *			    thems as need creatin'
  */
 
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
+#include <bsd/stdlib.h>
 #include <sys/types.h>
 #include <sys/select.h>
 #include <sys/stat.h>
--- a/src/usr.bin/make/main.c
+++ b/src/usr.bin/make/main.c
@@ -65,7 +65,8 @@
 #include <sys/stat.h>
 #include <sys/sysctl.h>
 #include <sys/time.h>
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
+#include <bsd/err.h>
 #include <sys/resource.h>
 #include <sys/utsname.h>
 #include <sys/wait.h>
--- a/src/usr.bin/make/shell.c
+++ b/src/usr.bin/make/shell.c
@@ -40,7 +40,7 @@
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- a/src/usr.bin/make/shell.h
+++ b/src/usr.bin/make/shell.h
@@ -42,7 +42,7 @@
 #ifndef shell_h_6002e3b8
 #define	shell_h_6002e3b8
 
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
 
 #include "str.h"
 #include "util.h"
--- a/src/usr.bin/make/suff.c
+++ b/src/usr.bin/make/suff.c
@@ -90,7 +90,7 @@
  *	    	  	    	if the target had no implicit sources.
  */
 
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
 #include <assert.h>
 #include <string.h>
 #include <stdlib.h>
--- a/src/usr.bin/make/Makefile
+++ b/src/usr.bin/make/Makefile
@@ -7,6 +7,7 @@
 SRCS=	arch.c buf.c cond.c dir.c for.c hash.c hash_tables.c job.c	\
 	lst.c main.c make.c parse.c proc.c shell.c str.c suff.c targ.c	\
 	util.c var.c
+LDFLAGS= -lbsd
 
 WARNS?=	6
 NO_SHARED?=	YES
--- a/src/usr.bin/yacc/test/ftp.tab.c
+++ b/src/usr.bin/yacc/test/ftp.tab.c
@@ -490,10 +490,10 @@
 #include <arpa/telnet.h>
 
 /*
- * getline - a hacked up version of fgets to ignore TELNET escape codes.
+ * bsd_getline - a hacked up version of fgets to ignore TELNET escape codes.
  */
 char *
-getline(s, n, iop)
+bsd_getline(s, n, iop)
 	char *s;
 	register FILE *iop;
 {
@@ -584,7 +584,7 @@
 		case CMD:
 			(void) signal(SIGALRM, toolong);
 			(void) alarm((unsigned) timeout);
-			if (getline(cbuf, sizeof(cbuf)-1, stdin) == NULL) {
+			if (bsd_getline(cbuf, sizeof(cbuf)-1, stdin) == NULL) {
 				reply(221, "You could at least say goodbye.");
 				dologout(0);
 			}
--- a/src/usr.bin/yacc/test/ftp.y
+++ b/src/usr.bin/yacc/test/ftp.y
@@ -750,10 +750,10 @@
 #include <arpa/telnet.h>
 
 /*
- * getline - a hacked up version of fgets to ignore TELNET escape codes.
+ * bsd_getline - a hacked up version of fgets to ignore TELNET escape codes.
  */
 char *
-getline(s, n, iop)
+bsd_getline(s, n, iop)
 	char *s;
 	register FILE *iop;
 {
@@ -844,7 +844,7 @@
 		case CMD:
 			(void) signal(SIGALRM, toolong);
 			(void) alarm((unsigned) timeout);
-			if (getline(cbuf, sizeof(cbuf)-1, stdin) == NULL) {
+			if (bsd_getline(cbuf, sizeof(cbuf)-1, stdin) == NULL) {
 				reply(221, "You could at least say goodbye.");
 				dologout(0);
 			}
--- a/src/usr.sbin/config/config.h
+++ b/src/usr.sbin/config/config.h
@@ -34,7 +34,7 @@
  * Config.
  */
 #include <sys/types.h>
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
 #include <stdlib.h>
 #include <string.h>
 
--- a/src/usr.sbin/config/mkoptions.c
+++ b/src/usr.sbin/config/mkoptions.c
@@ -40,6 +40,7 @@
  * Make all the .h files for the optional entries
  */
 
+#include <bsd/string.h>
 #include <ctype.h>
 #include <err.h>
 #include <stdio.h>
--- a/src/usr.sbin/config/Makefile
+++ b/src/usr.sbin/config/Makefile
@@ -13,7 +13,7 @@
 CFLAGS+= -I. -I${.CURDIR}
 
 DPADD=	${LIBL} ${LIBSBUF}
-LDADD=	-ll -lsbuf
+LDADD=	-ll -lsbuf -lbsd
 
 CLEANFILES+=	kernconf.c
 
--- a/src/usr.sbin/mtree/excludes.c
+++ b/src/usr.sbin/mtree/excludes.c
@@ -32,7 +32,8 @@
 
 #include <sys/types.h>
 #include <sys/time.h>		/* XXX for mtree.h */
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
+#include <bsd/stdio.h>
 
 #include <err.h>
 #include <fnmatch.h>
--- a/src/usr.sbin/mtree/spec.c
+++ b/src/usr.sbin/mtree/spec.c
@@ -35,6 +35,7 @@
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
+#include <bsd/stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <ctype.h>
--- a/src/usr.sbin/mtree/Makefile
+++ b/src/usr.sbin/mtree/Makefile
@@ -7,6 +7,7 @@
 MAN=	mtree.8
 SRCS=	compare.c crc.c create.c excludes.c misc.c mtree.c spec.c verify.c
 SRCS+=	specspec.c
+LDADD= -lbsd
 
 WARNS?=	4
 
