--- a/src/usr.sbin/mtree/specspec.c
+++ b/src/usr.sbin/mtree/specspec.c
@@ -84,8 +84,6 @@
 		printf(" rmd160digest=%s", n->rmd160digest);
 	if (f & F_SHA256)
 		printf(" sha256digest=%s", n->sha256digest);
-	if (f & F_FLAGS)
-		printf(" flags=%s", flags_to_string(n->st_flags));
 	printf("\n");
 }
 
--- a/src/usr.sbin/mtree/Makefile
+++ b/src/usr.sbin/mtree/Makefile
@@ -8,8 +8,4 @@
 SRCS=	compare.c crc.c create.c excludes.c misc.c mtree.c spec.c verify.c
 SRCS+=	specspec.c
 
-CFLAGS+= -DMD5 -DSHA1 -DRMD160 -DSHA256
-DPADD=	${LIBMD}
-LDADD=	-lmd
-
 .include <bsd.prog.mk>
