--- a/src/usr.bin/cksum/Makefile
+++ b/src/usr.bin/cksum/Makefile
@@ -3,7 +3,7 @@
 
 PROG=	cksum
 SRCS=	cksum.c crc.c print.c sum1.c sum2.c crc32.c
-LINKS=	${BINDIR}/cksum ${BINDIR}/sum
+SIMLINKS=	${BINDIR}/cksum ${BINDIR}/sum
 MLINKS=	cksum.1 sum.1
 
 .include <bsd.prog.mk>
--- a/src/usr.bin/lex/Makefile
+++ b/src/usr.bin/lex/Makefile
@@ -10,9 +10,9 @@
 #
 
 PROG=		lex
-LINKS+=	${BINDIR}/lex ${BINDIR}/lex++
-LINKS+=	${BINDIR}/lex ${BINDIR}/flex
-LINKS+=	${BINDIR}/lex ${BINDIR}/flex++
+SIMLINKS+=	${BINDIR}/lex ${BINDIR}/lex++
+SIMLINKS+=	${BINDIR}/lex ${BINDIR}/flex
+SIMLINKS+=	${BINDIR}/lex ${BINDIR}/flex++
 
 SRCS=		scan.c ccl.c dfa.c ecs.c gen.c main.c misc.c nfa.c parse.y \
 		skel.c sym.c tblcmp.c yylex.c
--- a/src/usr.bin/yacc/Makefile
+++ b/src/usr.bin/yacc/Makefile
@@ -6,7 +6,7 @@
 	skeleton.c symtab.c verbose.c warshall.c
 SCRIPTS=yyfix.sh
 MAN=	yacc.1 yyfix.1
-LINKS=	${BINDIR}/yacc ${BINDIR}/byacc
+SIMLINKS=	${BINDIR}/yacc ${BINDIR}/byacc
 MLINKS=	yacc.1 byacc.1
 
 .include <bsd.prog.mk>
