---
 src/usr.bin/make/main.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/src/usr.bin/make/main.c
+++ b/src/usr.bin/make/main.c
@@ -948,11 +948,11 @@
 	 * run-time.
 	 */
 	if (machine == NULL) {
-		static struct utsname utsname;
-
-		if (uname(&utsname) == -1)
-			err(2, "uname");
-		machine = utsname.machine;
+#ifdef MACHINE_ARCH
+		machine = MACHINE_ARCH;
+#else
+		machine = "unknown";
+#endif
 	}
 
 	if ((machine_arch = getenv("MACHINE_ARCH")) == NULL) {
