--- a/src/usr.bin/make/main.c
+++ b/src/usr.bin/make/main.c
@@ -147,6 +147,8 @@
 time_t		now;		/* Time at start of make */
 struct GNode	*DEFAULT;	/* .DEFAULT node */
 
+#include <bsd/getopt.h>		/* optreset */
+
 /**
  * Exit with usage message.
  */
@@ -380,7 +382,7 @@
 		if ((optind < argc) && strcmp(argv[optind], "--") == 0) {
 			found_dd = TRUE;
 		}
-		if ((c = getopt(argc, argv, OPTFLAGS)) == -1) {
+		if ((c = bsd_getopt(argc, argv, OPTFLAGS)) == -1) {
 			break;
 		}
 		switch(c) {
