---
 src/usr.bin/make/main.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/src/usr.bin/make/main.c
+++ b/src/usr.bin/make/main.c
@@ -148,6 +148,8 @@ uint32_t	warn_nocmd;	/* command line no-
 time_t		now;		/* Time at start of make */
 struct GNode	*DEFAULT;	/* .DEFAULT node */
 
+#include <bsd/getopt.h>		/* optreset */
+
 /**
  * Exit with usage message.
  */
@@ -381,7 +383,7 @@ rearg:
 		if ((optind < argc) && strcmp(argv[optind], "--") == 0) {
 			found_dd = TRUE;
 		}
-		if ((c = getopt(argc, argv, OPTFLAGS)) == -1) {
+		if ((c = bsd_getopt(argc, argv, OPTFLAGS)) == -1) {
 			break;
 		}
 		switch(c) {
