--- a/src/sys/sys/sbuf.h
+++ b/src/sys/sys/sbuf.h
@@ -31,7 +31,8 @@
 #ifndef _SYS_SBUF_H_
 #define	_SYS_SBUF_H_
 
-#include <sys/_types.h>
+#include <stdarg.h>
+#include <sys/types.h>
 
 /*
  * Structure definition
@@ -64,10 +65,8 @@
 int		 sbuf_bcpy(struct sbuf *, const void *, size_t);
 int		 sbuf_cat(struct sbuf *, const char *);
 int		 sbuf_cpy(struct sbuf *, const char *);
-int		 sbuf_printf(struct sbuf *, const char *, ...)
-	__printflike(2, 3);
-int		 sbuf_vprintf(struct sbuf *, const char *, __va_list)
-	__printflike(2, 0);
+int		 sbuf_printf(struct sbuf *, const char *, ...);
+int		 sbuf_vprintf(struct sbuf *, const char *, va_list);
 int		 sbuf_putc(struct sbuf *, int);
 int		 sbuf_trim(struct sbuf *);
 int		 sbuf_overflowed(struct sbuf *);
--- a/src/sys/kern/subr_sbuf.c
+++ b/src/sys/kern/subr_sbuf.c
@@ -77,8 +77,8 @@
 #define	SBUF_CLEARFLAG(s, f)	do { (s)->s_flags &= ~(f); } while (0)
 
 #define	SBUF_MINEXTENDSIZE	16		/* Should be power of 2. */
-#define	SBUF_MAXEXTENDSIZE	PAGE_SIZE
-#define	SBUF_MAXEXTENDINCR	PAGE_SIZE
+#define	SBUF_MAXEXTENDSIZE	4096
+#define	SBUF_MAXEXTENDINCR	SBUF_MAXEXTENDSIZE
 
 /*
  * Debugging support
--- a/src/usr.sbin/config/Makefile
+++ b/src/usr.sbin/config/Makefile
@@ -4,7 +4,7 @@
 PROG=	config
 MAN=	config.5 config.8
 SRCS=	config.y main.c lang.l mkmakefile.c mkheaders.c \
-	mkoptions.c y.tab.h kernconf.c
+	mkoptions.c y.tab.h kernconf.c subr_sbuf.c
 
 kernconf.c: kernconf.tmpl
 	file2c 'char kernconfstr[] = {' ',0};' < ${.CURDIR}/kernconf.tmpl > kernconf.c
@@ -13,7 +13,7 @@
 CFLAGS+= -I. -I${.CURDIR}
 
 DPADD=	${LIBL} ${LIBSBUF}
-LDADD=	-ll -lsbuf -lbsd
+LDADD=	-ll -lbsd
 
 CLEANFILES+=	kernconf.c
 
--- a/src/usr.sbin/config/main.c
+++ b/src/usr.sbin/config/main.c
@@ -43,6 +43,8 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <bsd/string.h>
+#include <stdarg.h>
 #include <sys/sbuf.h>
 #include <sys/file.h>
 #include <sys/mman.h>
