---
 sys/sys/mount.h |   19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

--- a/sys/sys/mount.h
+++ b/sys/sys/mount.h
@@ -34,20 +34,16 @@
 #define _SYS_MOUNT_H_
 
 #include <sys/ucred.h>
-#include <sys/queue.h>
-#ifdef _KERNEL
-#include <sys/lock.h>
-#include <sys/lockmgr.h>
-#include <sys/_mutex.h>
-#endif
+#include <stdint.h>
+#include <sys/vfs.h>
+#include <sys/types.h>
+#include <sys/socket.h>
 
 /*
  * NOTE: When changing statfs structure, mount structure, MNT_* flags or
  * MNTK_* flags also update DDB show mount command in vfs_subr.c.
  */
 
-typedef struct fsid { int32_t val[2]; } fsid_t;	/* filesystem id type */
-
 /*
  * File identifier.
  * These are unique per filesystem on a single machine.
@@ -65,6 +61,7 @@ struct fid {
  */
 #define	MFSNAMELEN	16		/* length of type name including null */
 #define	MNAMELEN	88		/* size of on/from name bufs */
+#if 0
 #define	STATFS_VERSION	0x20030518	/* current version number */
 struct statfs {
 	uint32_t f_version;		/* structure version number */
@@ -220,6 +217,7 @@ void          __mnt_vnode_markerfree(str
 } while (0)
 
 #endif /* _KERNEL */
+#endif
 
 /*
  * User specifiable flags, stored in mnt_flag.
@@ -415,6 +413,7 @@ struct nfs_public {
 	char		*np_index;	/* Index file */
 };
 
+#if 0
 /*
  * Filesystem configuration information. One of these exists for each
  * type of filesystem supported by the kernel. These are searched at
@@ -452,6 +451,7 @@ struct ovfsconf {
 	int	vfc_flags;
 };
 #endif
+#endif
 
 /*
  * NB: these flags refer to IMPLEMENTATION properties, not properties of
@@ -517,6 +517,7 @@ struct vfsquery {
 #define VQ_FLAG4000	0x4000	/* placeholder */
 #define VQ_FLAG8000	0x8000	/* placeholder */
 
+#if 0
 #ifdef _KERNEL
 /* Point a sysctl request at a vfsidctl's data. */
 #define VCTLTOREQ(vc, req)						\
@@ -794,5 +795,5 @@ int	getvfsbyname(const char *, struct xv
 __END_DECLS
 
 #endif /* _KERNEL */
-
+#endif
 #endif /* !_SYS_MOUNT_H_ */
