---
 sys/sys/param.h |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/sys/sys/param.h
+++ b/sys/sys/param.h
@@ -38,6 +38,7 @@
 #ifndef _SYS_PARAM_H_
 #define _SYS_PARAM_H_
 
+#if 0
 #include <sys/_null.h>
 
 #define	BSD	199506		/* System version (year & month). */
@@ -77,6 +78,9 @@
  * MAXLOGNAME should be == UT_NAMESIZE+1 (see <utmp.h>)
  */
 #include <sys/syslimits.h>
+#endif
+#include <sys/types.h>
+#include <limits.h>
 
 #define	MAXCOMLEN	19		/* max command name remembered */
 #define	MAXINTERP	32		/* max interpreter file name length */
@@ -89,6 +93,7 @@
 #define MAXHOSTNAMELEN	256		/* max hostname size */
 #define SPECNAMELEN	63		/* max length of devicename */
 
+#if 0
 /* More types and definitions used throughout the kernel. */
 #ifdef _KERNEL
 #include <sys/cdefs.h>
@@ -116,6 +121,7 @@
 #ifndef _KERNEL
 #include <sys/limits.h>
 #endif
+#endif
 
 #ifndef _NO_NAMESPACE_POLLUTION
 
@@ -194,7 +200,9 @@
 #define	PDROP	0x200	/* OR'd with pri to stop re-entry of interlock mutex */
 #define	PBDRY	0x400	/* for PCATCH stop is done on the user boundary */
 
+#if 0
 #define	NZERO	0		/* default "nice" */
+#endif
 
 #define	NBBY	8		/* number of bits in a byte */
 #define	NBPW	sizeof(int)	/* number of bytes per word (integer) */
@@ -263,6 +271,7 @@
 #define	MIN(a,b) (((a)<(b))?(a):(b))
 #define	MAX(a,b) (((a)>(b))?(a):(b))
 
+#if 0
 #ifdef _KERNEL
 /*
  * Basic byte order function prototypes for non-inline functions.
@@ -317,4 +326,5 @@
 #define	member2struct(s, m, x)						\
 	((struct s *)(void *)((char *)(x) - offsetof(struct s, m)))
 
+#endif
 #endif	/* _SYS_PARAM_H_ */
