---
 libdisklabel/Makefile       |   10 ++++++++++
 libdisklabel/getdisklabel.c |    8 ++++++++
 2 files changed, 18 insertions(+)

--- /dev/null
+++ b/libdisklabel/Makefile
@@ -0,0 +1,10 @@
+LIB = disklabel
+LIB_type = static
+LIBSRCS = getdisklabel.c
+
+ALL_CFLAGS = -D_LIBDISKLABEL
+
+INCLUDES =
+
+include ../Makefile.common
+
--- /dev/null
+++ b/libdisklabel/getdisklabel.c
@@ -0,0 +1,8 @@
+#include <stdlib.h>
+#include <sys/disklabel.h>
+
+struct disklabel *getdiskbyname(const char *name)
+{
+	return NULL;
+}
+
