Index: ufsutils/libdisklabel/Makefile
===================================================================
--- /dev/null
+++ ufsutils/libdisklabel/Makefile
@@ -0,0 +1,10 @@
+LIB = disklabel
+LIB_type = static
+LIBSRCS = getdisklabel.c
+
+ALL_CFLAGS = -D_LIBDISKLABEL
+
+INCLUDES =
+
+include ../Makefile.common
+
Index: ufsutils/libdisklabel/getdisklabel.c
===================================================================
--- /dev/null
+++ ufsutils/libdisklabel/getdisklabel.c
@@ -0,0 +1,8 @@
+#include <stdlib.h>
+#include <sys/disklabel.h>
+
+struct disklabel *getdiskbyname(const char *name)
+{
+	return NULL;
+}
+
