---
 sbin/badsect/Makefile |   14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

--- a/sbin/badsect/Makefile
+++ b/sbin/badsect/Makefile
@@ -1,9 +1,13 @@
 #	@(#)Makefile	8.1 (Berkeley) 6/5/93
 # $FreeBSD$
 
-PROG=	badsect
-DPADD=	${LIBUFS}
-LDADD=	-lufs
-MAN=	badsect.8
+PROG = badsect.ufs
+SRCS = badsect.c
+WARNS = 0
+MAN = badsect.8:badsect.ufs.8
+
+INCLUDES = -I../../lib/libufs
+LDADD += -L../../lib/libufs -lufs -lbsd
+
+include ../../Makefile.common
 
-.include <bsd.prog.mk>
