Index: ufsutils/badsect.ufs/Makefile
===================================================================
--- ufsutils.orig/badsect.ufs/Makefile
+++ ufsutils/badsect.ufs/Makefile
@@ -1,9 +1,13 @@
 #	@(#)Makefile	8.1 (Berkeley) 6/5/93
 # $FreeBSD: src/sbin/badsect/Makefile,v 1.8 2003/02/22 21:33:42 johan Exp $
 
-PROG=	badsect
-DPADD=	${LIBUFS}
-LDADD=	-lufs
-MAN=	badsect.8
+PROG = badsect.ufs
+SRCS = badsect.c
+WARNS = 0
+MAN = badsect.8:badsect.ufs.8
+
+INCLUDES = -I../libufs
+LDADD += -L../libufs -lufs
+
+include ../Makefile.common
 
-.include <bsd.prog.mk>
