#DPATCHLEVEL=1
Index: ufsutils/dump.ufs/Makefile
===================================================================
--- ufsutils.orig/dump.ufs/Makefile
+++ ufsutils/dump.ufs/Makefile
@@ -15,9 +15,14 @@
 
 PROG=	dump
 LINKS=	${BINDIR}/dump ${BINDIR}/rdump
-CFLAGS+=-DRDUMP
 SRCS=	itime.c main.c optr.c dumprmt.c tape.c traverse.c unctime.c cache.c
 MAN=	dump.8
-MLINKS=	dump.8 rdump.8
+MLINKS+=dump.8 rdump.8
+
+ALL_CFLAGS = -DRDUMP
+
+LDADD = -L../libufs -lufs -L../libport -lport
+INCLUDES = -I../include -I${CURDIR}/../../libexec/rlogind
+
+include ../Makefile.common
 
-.include <bsd.prog.mk>
