#DPATCHLEVEL=1
Index: ufsutils/dumpfs.ufs/dumpfs.c
===================================================================
--- ufsutils.orig/dumpfs.ufs/dumpfs.c
+++ ufsutils/dumpfs.ufs/dumpfs.c
@@ -62,6 +62,7 @@
 #include <fcntl.h>
 #include <fstab.h>
 #include <libufs.h>
+#include <time.h>
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
Index: ufsutils/dumpfs.ufs/Makefile
===================================================================
--- ufsutils.orig/dumpfs.ufs/Makefile
+++ ufsutils/dumpfs.ufs/Makefile
@@ -1,9 +1,13 @@
 #	@(#)Makefile	8.1 (Berkeley) 6/5/93
 # $FreeBSD: src/sbin/dumpfs/Makefile,v 1.9 2003/11/14 13:13:23 johan Exp $
 
-PROG=	dumpfs
-DPADD=	${LIBUFS}
-LDADD=	-lufs
-MAN=	dumpfs.8
+PROG = dumpfs.ufs
+SRCS = dumpfs.c
+WARNS = 0
+MAN = dumpfs.8:dumpfs.ufs.8
+
+LDADD = -L../libufs/ -lufs
+INCLUDES = -I../libufs
+
+include ../Makefile.common
 
-.include <bsd.prog.mk>
