#DPATCHLEVEL=1
Index: ufsutils/ffsinfo/Makefile
===================================================================
--- ufsutils.orig/ffsinfo/Makefile
+++ ufsutils/ffsinfo/Makefile
@@ -4,16 +4,12 @@
 # $FreeBSD: src/sbin/ffsinfo/Makefile,v 1.7 2004/04/01 20:31:48 johan Exp $
 #
 
-GROWFS= ${.CURDIR}/../growfs
-.PATH: ${GROWFS}
+PROG = ffsinfo
+SRCS = ffsinfo.c ../growfs.ufs/debug.c
+MAN = ffsinfo.8:ffsinfo.8
 
-PROG=	ffsinfo
-SRCS=	ffsinfo.c debug.c
-MAN=	ffsinfo.8
+ALL_CFLAGS = -DFS_DEBUG -I../growfs.ufs -I../libufs
+LDADD = -L../libufs -lufs
+
+include ../Makefile.common
 
-WARNS?=	0
-CFLAGS+=-DFS_DEBUG -I${GROWFS}
-DPADD=	${LIBUFS}
-LDADD=	-lufs
- 
-.include <bsd.prog.mk>               
