---
 sbin/ffsinfo/Makefile |   18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

--- a/sbin/ffsinfo/Makefile
+++ b/sbin/ffsinfo/Makefile
@@ -4,16 +4,12 @@
 # $FreeBSD$
 #
 
-GROWFS= ${.CURDIR}/../growfs
-.PATH: ${GROWFS}
+PROG = ffsinfo
+SRCS = ffsinfo.c ../growfs/debug.c
+MAN = ffsinfo.8:ffsinfo.8
 
-PROG=	ffsinfo
-SRCS=	ffsinfo.c debug.c
-MAN=	ffsinfo.8
+ALL_CFLAGS = -DFS_DEBUG -I../growfs -I../../lib/libufs
+LDADD += -L../../lib/libufs -lufs -lbsd
+
+include ../../Makefile.common
 
-WARNS?=	0
-CFLAGS+=-DFS_DEBUG -I${GROWFS}
-DPADD=	${LIBUFS}
-LDADD=	-lufs
- 
-.include <bsd.prog.mk>               
