Index: ufsutils/growfs.ufs/growfs.c
===================================================================
--- ufsutils.orig/growfs.ufs/growfs.c
+++ ufsutils/growfs.ufs/growfs.c
@@ -51,10 +51,10 @@
 
 /* ********************************************************** INCLUDES ***** */
 #include <sys/param.h>
+#define rounddown(x, y) (((x)/(y))*(y))
 #include <sys/disklabel.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
-#include <sys/disk.h>
 
 #include <stdio.h>
 #include <paths.h>
@@ -1929,10 +1929,12 @@
    int sectorsize;
    off_t mediasize;
 
+#ifdef HAVE_BSD_DISKLABEL
    if (ioctl(fd, DIOCGSECTORSIZE, &sectorsize) == -1)
         err(1,"DIOCGSECTORSIZE");
    if (ioctl(fd, DIOCGMEDIASIZE, &mediasize) == -1)
         err(1,"DIOCGMEDIASIZE");
+#endif
 
    if (sectorsize <= 0)
        errx(1, "bogus sectorsize: %d", sectorsize);
@@ -2288,9 +2290,11 @@
 		}
 		lp->d_checksum=sum;
 
+#if HAVE_BSD_DISKLABEL
 		if (ioctl(fd, DIOCWDINFO, (char *)lp) < 0) {
 			errx(1, "DIOCWDINFO failed");
 		}
+#endif
 	}
 	free(lp);
 
@@ -2314,8 +2318,10 @@
 	if (!lab)
 		errx(1, "malloc failed");
 
+#ifdef HAVE_BSD_DISKLABEL
     if (!ioctl(fd, DIOCGDINFO, (char *)lab))
         return (lab);
+#endif
 
     unlabeled++;
 
Index: ufsutils/growfs.ufs/Makefile
===================================================================
--- ufsutils.orig/growfs.ufs/Makefile
+++ ufsutils/growfs.ufs/Makefile
@@ -6,14 +6,18 @@
 
 #GFSDBG=
 
-PROG=   growfs
+PROG=   growfs.ufs
 SRCS=   growfs.c
-MAN=	growfs.8
+MAN=	growfs.8:growfs.ufs.8
 
-WARNS?=	6
+ifdef GFSDBG
+SRCS +=  debug.c
+ALL_CFLAGS = -DFS_DEBUG
+endif
+ 
+WARNS = 6
+LDADD += -L../libufs -lufs
+INCLUDES =
 
-.if defined(GFSDBG)
-SRCS+=  debug.c
-.endif  
-
-.include <bsd.prog.mk>      
+include ../Makefile.common
+ 
