#DPATCHLEVEL=1
diff -Naur ufsutils-0.0+2004.06.26.orig/mkfs.ufs/mkfs.c ufsutils-0.0+2004.06.26/mkfs.ufs/mkfs.c
--- ufsutils-0.0+2004.06.26.orig/mkfs.ufs/mkfs.c	2005-02-03 08:34:18.000000000 +0100
+++ ufsutils-0.0+2004.06.26/mkfs.ufs/mkfs.c	2005-02-03 08:36:07.000000000 +0100
@@ -737,7 +737,9 @@
 
 	memset(&node, 0, sizeof node);
 	if ((grp = getgrnam("operator")) == NULL)
-		errx(35, "Cannot retrieve operator gid");
+		grp = getgrnam("root")
+	if (grp == NULL)
+		errx(35, "Cannot retrieve operator nor root gid");
 	if (sblock.fs_magic == FS_UFS1_MAGIC) {
 		/*
 		 * initialize the node
