To be merged in the kfreebsd/ directory when 2.11 is the default

--- a/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/Makefile
--- b/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/Makefile
@@ -36,7 +36,7 @@
 # For <fcntl.h>.
 sysdep_routines += sys_open sys_openat open_2
 # For <sys/stat.h>.
-sysdep_routines += sys_fchmodat sys_fstat sys_fstatat sys_lstat sys_mkdirat sys_mkfifoat sys_mknod sys_mknodat sys_nfstat sys_nlstat sys_nstat sys_stat
+sysdep_routines += sys_fchmodat sys_fstat sys_fstatat sys_lstat sys_mkdirat sys_mkfifoat sys_mknod sys_mknodat sys_nfstat sys_nlstat sys_nstat sys_stat lchflags
 # For <sys/statfs.h>.
 sysdep_routines += fstatfs64 statfs64 sys_fstatfs sys_statfs
 # For <stdio.h>
--- a/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/syscalls.list
+++ b/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/syscalls.list
@@ -15,6 +15,7 @@@ -57,6 +59,7 @@
 kldunload		-	kldunload		i:i		kldunload
 kldunloadf		-	kldunloadf		i:ii		kldunloadf
 ktrace			-	ktrace			i:siii		ktrace
+lchflags		-	lchflags		i:si		lchflags
 lchmod			-	lchmod			i:si		__lchmod lchmod
 lchown			-	lchown			i:sii		__lchown lchown
 sys_lio_listio		-	lio_listio		i:ibnP		__syscall_lio_listio
--- a/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/Versions
+++ b/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/Versions
@@ -87,6 +87,11 @@
     kenv;
     sysctlnametomib;
   }
+  GLIBC_2.11 {
+    lchflags;
+    #errlist-compat 94
+    _sys_errlist; sys_errlist; _sys_nerr; sys_nerr;
+  }
   GLIBC_PRIVATE {
     # needed by libpthread.
     __clone; __libc_fork; __libc_sigaction;
--- a/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/errno.h
+++ b/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/errno.h
@@ -155,8 +155,9 @@
 #define	EMULTIHOP	90		/* Multihop attempted */
 #define	ENOLINK		91		/* Link has been severed */
 #define	EPROTO		92		/* Protocol error */
+#define	ENOTCAPABLE	93		/* Capabilities insufficient */
 
-#define	ELAST		92		/* Must be equal largest errno */
+#define	ELAST		93		/* Must be equal largest errno */
 
 # ifndef __ASSEMBLER__
 /* Function to get address of global `errno' variable.  */
