diff -ur src.old/sys/cam/scsi/scsi_all.h src/sys/cam/scsi/scsi_all.h
--- src.old/sys/cam/scsi/scsi_all.h	2004-08-03 08:37:00.000000000 +0200
+++ src/sys/cam/scsi/scsi_all.h	2004-08-03 08:43:46.000000000 +0200
@@ -25,6 +25,8 @@
 #define _SCSI_SCSI_ALL_H 1
 
 #include <sys/cdefs.h>
+#include <sys/types.h>
+#include <stdio.h>
 
 #ifdef _KERNEL
 /*
diff -ur src.old/sys/cam/scsi/scsi_cd.h src/sys/cam/scsi/scsi_cd.h
--- src.old/sys/cam/scsi/scsi_cd.h	2004-08-03 08:37:00.000000000 +0200
+++ src/sys/cam/scsi/scsi_cd.h	2004-08-03 08:43:46.000000000 +0200
@@ -46,6 +46,9 @@
 #ifndef	_SCSI_SCSI_CD_H
 #define _SCSI_SCSI_CD_H 1
 
+#include <sys/types.h>
+#include "scsi_all.h"
+
 /*
  *	Define two bits always in the same place in byte 2 (flag byte)
  */
diff -ur src.old/sys/cam/scsi/scsi_ch.h src/sys/cam/scsi/scsi_ch.h
--- src.old/sys/cam/scsi/scsi_ch.h	2004-08-03 08:37:00.000000000 +0200
+++ src/sys/cam/scsi/scsi_ch.h	2004-08-03 08:43:46.000000000 +0200
@@ -64,6 +64,7 @@
 #define _SCSI_SCSI_CH_H 1
 
 #include <sys/cdefs.h>
+#include <sys/types.h>
 
 /*
  * SCSI command format
diff -ur src.old/sys/cam/scsi/scsi_da.h src/sys/cam/scsi/scsi_da.h
--- src.old/sys/cam/scsi/scsi_da.h	2004-08-03 08:37:00.000000000 +0200
+++ src/sys/cam/scsi/scsi_da.h	2004-08-03 08:43:46.000000000 +0200
@@ -53,6 +53,7 @@
 #define _SCSI_SCSI_DA_H 1
 
 #include <sys/cdefs.h>
+#include <sys/types.h>
 
 struct scsi_rezero_unit
 {
diff -ur src.old/sys/cam/scsi/scsi_low.h src/sys/cam/scsi/scsi_low.h
--- src.old/sys/cam/scsi/scsi_low.h	2004-08-03 08:37:00.000000000 +0200
+++ src/sys/cam/scsi/scsi_low.h	2004-08-03 08:43:46.000000000 +0200
@@ -53,7 +53,7 @@
 #define	SCSI_LOW_INTERFACE_XS
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__
+#ifdef	__FreeBSD_kernel__
 #define	SCSI_LOW_INTERFACE_CAM
 #define	CAM
 #endif	/* __FreeBSD__ */
@@ -64,7 +64,7 @@
 #include <dev/isa/ccbque.h>
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__
+#ifdef	__FreeBSD_kernel__
 #include <sys/device_port.h>
 #include <cam/cam.h>
 #include <cam/cam_ccb.h>
@@ -84,7 +84,7 @@
 #define	SCSI_LOW_BZERO(pt, size)	memset((pt), 0, (size))
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__
+#ifdef	__FreeBSD_kernel__
 #undef	MSG_IDENTIFY
 #define	SCSI_LOW_DEBUGGER(dev)	Debugger((dev))
 #define	SCSI_LOW_DELAY(mu)	DELAY((mu))
@@ -110,7 +110,7 @@
 };
 #endif	/* __NetBSD__ */
 
-#ifdef	__FreeBSD__
+#ifdef	__FreeBSD_kernel__
 typedef	struct scsi_sense_data scsi_low_osdep_sense_data_t;
 
 struct scsi_low_osdep_interface {
