diff -ur src.old/sys/i386/include/kse.h src/sys/i386/include/kse.h
--- src.old/sys/i386/include/kse.h	2002-06-29 19:26:16.000000000 +0200
+++ src/sys/i386/include/kse.h	2004-08-23 04:00:49.000000000 +0200
@@ -33,7 +33,7 @@
 #ifdef _KERNEL
 #include <machine/frame.h>
 #endif
-#include <machine/ucontext.h>
+#include <sys/ucontext.h>
 
 union kse_td_ctx {
 #ifdef _KERNEL
diff -ur src.old/sys/sys/kse.h src/sys/sys/kse.h
--- src.old/sys/sys/kse.h	2004-08-23 03:59:55.000000000 +0200
+++ src/sys/sys/kse.h	2004-08-23 04:00:14.000000000 +0200
@@ -33,10 +33,11 @@
 #ifndef _SYS_KSE_H_
 #define _SYS_KSE_H_
 
-#include <machine/kse.h>
-#include <sys/ucontext.h>
+#include "/usr/src/kfreebsd5-headers/i386/include/kse.h"
+#include <sys/ucontext.h>
 #include <sys/time.h>
 #include <sys/signal.h>
+#include <sys/types.h>
 
 /*
  * This file defines the structures needed for communication between
@@ -58,8 +59,8 @@
 	unsigned int		tm_flags;	/* Thread flags */
 	struct kse_thr_mailbox	*tm_next;	/* Next thread in list */
 	void			*tm_udata;	/* For use by the UTS */
-	uint32_t		tm_uticks;
-	uint32_t		tm_sticks;
+	u_int32_t		tm_uticks;
+	u_int32_t		tm_sticks;
 	siginfo_t		tm_syncsig;
 	int			tm_spare[8];
 };
@@ -75,7 +76,7 @@
 	struct kse_thr_mailbox	*km_curthread;	/* Currently running thread */
 	struct kse_thr_mailbox	*km_completed;	/* Threads back from kernel */
 	sigset_t		km_sigscaught;	/* Caught signals */
-	uint32_t		km_flags;	/* KSE flags */
+	u_int32_t		km_flags;	/* KSE flags */
 	kse_func_t		*km_func;	/* UTS function */
 	stack_t			km_stack;	/* UTS context */
 	void			*km_udata;	/* For use by the UTS */
