diff -ur src.old/sys/sys/mount.h src/sys/sys/mount.h
--- src.old/sys/sys/mount.h	2003-11-12 09:01:40.000000000 +0100
+++ src/sys/sys/mount.h	2004-11-27 18:30:05.000000000 +0100
@@ -65,30 +65,6 @@
 #define	MFSNAMELEN	16		/* length of type name including null */
 #define	MNAMELEN	88		/* size of on/from name bufs */
 #define	STATFS_VERSION	0x20030518	/* current version number */
-struct statfs {
-	uint32_t f_version;		/* structure version number */
-	uint32_t f_type;		/* type of filesystem */
-	uint64_t f_flags;		/* copy of mount exported flags */
-	uint64_t f_bsize;		/* filesystem fragment size */
-	uint64_t f_iosize;		/* optimal transfer block size */
-	uint64_t f_blocks;		/* total data blocks in filesystem */
-	uint64_t f_bfree;		/* free blocks in filesystem */
-	int64_t	 f_bavail;		/* free blocks avail to non-superuser */
-	uint64_t f_files;		/* total file nodes in filesystem */
-	int64_t	 f_ffree;		/* free nodes avail to non-superuser */
-	uint64_t f_syncwrites;		/* count of sync writes since mount */
-	uint64_t f_asyncwrites;		/* count of async writes since mount */
-	uint64_t f_syncreads;		/* count of sync reads since mount */
-	uint64_t f_asyncreads;		/* count of async reads since mount */
-	uint64_t f_spare[10];		/* unused spare */
-	uint32_t f_namemax;		/* maximum filename length */
-	uid_t	  f_owner;		/* user that mounted the filesystem */
-	fsid_t	  f_fsid;		/* filesystem id */
-	char	  f_charspare[80];	    /* spare string space */
-	char	  f_fstypename[MFSNAMELEN]; /* filesystem type name */
-	char	  f_mntfromname[MNAMELEN];  /* mounted filesystem */
-	char	  f_mntonname[MNAMELEN];    /* directory on which mounted */
-};
 
 #ifdef _KERNEL
 #define	OMFSNAMELEN	16	/* length of fs type name, including null */
diff -ur src.old/sys/sys/types.h src/sys/sys/types.h
--- src.old/sys/sys/types.h	2003-03-28 16:27:30.000000000 +0100
+++ src/sys/sys/types.h	2004-11-27 18:31:06.000000000 +0100
@@ -209,11 +209,6 @@
 
 typedef	__segsz_t	segsz_t;	/* segment size (in pages) */
 
-#ifndef _SIZE_T_DECLARED
-typedef	__size_t	size_t;
-#define	_SIZE_T_DECLARED
-#endif
-
 #ifndef _SSIZE_T_DECLARED
 typedef	__ssize_t	ssize_t;
 #define	_SSIZE_T_DECLARED
--- src/sys/net/if.h~	2004-11-27 21:46:23.000000000 +0100
+++ src/sys/net/if.h	2004-11-27 21:49:53.000000000 +0100
@@ -89,8 +89,6 @@
     { { 0 }, name, sizeof(name) - 1, minifs, maxunit, NULL, 0, create, destroy }
 #endif
 
-#if __BSD_VISIBLE
-
 /*
  * Structure used to query names of interface cloners.
  */
@@ -340,8 +338,6 @@
 	struct	sockaddr_storage dstaddr; /* out */
 };
 
-#endif /* __BSD_VISIBLE */
-
 #ifdef _KERNEL
 #ifdef MALLOC_DECLARE
 MALLOC_DECLARE(M_IFADDR);
