--- a/sys/dev/aic7xxx/aicasm/Makefile
+++ b/sys/dev/aic7xxx/aicasm/Makefile
@@ -14,7 +14,7 @@
 SRCS=	${GENHDRS} ${CSRCS} ${YSRCS} ${LSRCS}
 CLEANFILES+= ${GENHDRS} ${YSRCS:R:C/(.*)/\1.output/g}
 DPADD=	${LIBL}
-LDADD=	-ll
+LDADD=	-ll -ldb
 WARNS?=	6
 
 # Correct path for kernel builds
@@ -24,13 +24,6 @@
 DEPENDFILE=	.depend_aicasm
 .endif
 
-.if ${CC} == "icc"
-CFLAGS+=	-restrict
-NOSTDINC=	-X
-.else
-NOSTDINC=	-nostdinc
-.endif
-CFLAGS+= ${NOSTDINC} -I/usr/include -I.
 .ifdef MAKESRCPATH
 CFLAGS+= -I${MAKESRCPATH}
 .endif
--- a/sys/dev/aic7xxx/aicasm/aicasm.h
+++ b/sys/dev/aic7xxx/aicasm/aicasm.h
@@ -42,8 +42,9 @@
  * $FreeBSD$
  */
 
-#ifdef __linux__
-#include "../queue.h"
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
--- a/sys/dev/aic7xxx/aicasm/aicasm_gram.y
+++ b/sys/dev/aic7xxx/aicasm/aicasm_gram.y
@@ -52,8 +52,9 @@
 #include <string.h>
 #include <sysexits.h>
 
-#ifdef __linux__
-#include "../queue.h"
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
@@ -1534,7 +1535,7 @@
 }
 
 static void
-add_macro_arg(const char *argtext, int argnum __unused)
+add_macro_arg(const char *argtext, int argnum)
 {
 	struct macro_arg *marg;
 	int retval;
--- a/sys/dev/aic7xxx/aicasm/aicasm_macro_gram.y
+++ b/sys/dev/aic7xxx/aicasm/aicasm_macro_gram.y
@@ -52,8 +52,9 @@
 #include <string.h>
 #include <sysexits.h>
 
-#ifdef __linux__
-#include "../queue.h"
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
--- a/sys/dev/aic7xxx/aicasm/aicasm_macro_scan.l
+++ b/sys/dev/aic7xxx/aicasm/aicasm_macro_scan.l
@@ -51,8 +51,10 @@
 #include <stdio.h>
 #include <string.h>
 #include <sysexits.h>
-#ifdef __linux__
-#include "../queue.h"
+
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
--- a/sys/dev/aic7xxx/aicasm/aicasm_scan.l
+++ b/sys/dev/aic7xxx/aicasm/aicasm_scan.l
@@ -51,8 +51,10 @@
 #include <stdio.h>
 #include <string.h>
 #include <sysexits.h>
-#ifdef __linux__
-#include "../queue.h"
+
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
--- a/sys/dev/aic7xxx/aicasm/aicasm_symbol.c
+++ b/sys/dev/aic7xxx/aicasm/aicasm_symbol.c
@@ -44,10 +44,11 @@
 
 #include <sys/types.h>
 
-#ifdef __linux__
-#include "aicdb.h"
+#include <sys/param.h>		/* BSD */
+#ifdef BSD
+#include <db.h>			/* BSD native libdb */
 #else
-#include <db.h>
+#include <db_185.h>		/* Sleepycat 1.85 compat */
 #endif
 #include <ctype.h>
 #include <fcntl.h>
--- a/sys/dev/aic7xxx/aicasm/aicasm_symbol.h
+++ b/sys/dev/aic7xxx/aicasm/aicasm_symbol.h
@@ -42,8 +42,9 @@
  * $FreeBSD$
  */
 
-#ifdef __linux__
-#include "../queue.h"
+#include <sys/param.h>		/* __GLIBC__ */
+#ifdef __GLIBC__
+#include <bsd/queue.h>
 #else
 #include <sys/queue.h>
 #endif
