
Status: xargs -J option requested to both upstream and debian.

--- src.orig/sys/conf/kmod.mk
+++ src/sys/conf/kmod.mk
@@ -203,7 +203,7 @@
 	grep -v '^#' < ${EXPORT_SYMS} > export_syms
 .endif
 	awk -f ${SYSDIR}/conf/kmod_syms.awk ${.TARGET} \
-	    export_syms | xargs -J% ${OBJCOPY} % ${.TARGET}
+	    export_syms | xargs -I% ${OBJCOPY} % ${.TARGET}
 .endif
 .endif
 .if !defined(DEBUG_FLAGS) && ${MACHINE_ARCH} == amd64
--- src.orig/sys/contrib/dev/acpica/acpica_prep.sh
+++ src/sys/contrib/dev/acpica/acpica_prep.sh
@@ -58,10 +58,10 @@
 # copy files
 echo copying full dirs
 for i in ${fulldirs}; do
-	find ${wrk} -name ${i} -type d | xargs -J % mv % ${dst}
+	find ${wrk} -name ${i} -type d | xargs -I % mv % ${dst}
 done
 echo copying remaining files
-find ${wrk} -type f | xargs -J % mv % ${dst}
+find ${wrk} -type f | xargs -I % mv % ${dst}
 
 # canonify include paths
 for H in ${src_headers}; do
