---
 sys/i386/i386/pmap.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/sys/i386/i386/pmap.c
+++ b/sys/i386/i386/pmap.c
@@ -218,7 +218,7 @@
 SYSCTL_INT(_vm_pmap, OID_AUTO, pat_works, CTLFLAG_RD, &pat_works, 1,
     "Is page attribute table fully functional?");
 
-static int pg_ps_enabled;
+static int pg_ps_enabled = 1;
 SYSCTL_INT(_vm_pmap, OID_AUTO, pg_ps_enabled, CTLFLAG_RDTUN, &pg_ps_enabled, 0,
     "Are large page mappings enabled?");
 
