--- a/sys/dev/xen/evtchn/evtchn_dev.c
+++ b/sys/dev/xen/evtchn/evtchn_dev.c
@@ -41,7 +41,7 @@
 } evtchn_softc_t;
 
 
-#ifdef linuxcrap
+#if 0
 /* NB. This must be shared amongst drivers if more things go in /dev/xen */
 static devfs_handle_t xen_dev_dir;
 #endif
--- a/sys/xen/evtchn/evtchn_dev.c
+++ b/sys/xen/evtchn/evtchn_dev.c
@@ -39,7 +39,7 @@
 } evtchn_softc_t;
 
 
-#ifdef linuxcrap
+#if 0
 /* NB. This must be shared amongst drivers if more things go in /dev/xen */
 static devfs_handle_t xen_dev_dir;
 #endif
