Index: src/sys/conf/files.i386
===================================================================
--- src.orig/sys/conf/files.i386
+++ src/sys/conf/files.i386
@@ -46,26 +46,5 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci				\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/i386-elf.raid.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptmv/i386-elf.raid.o.uu"	\
-	no-implicit-rule
-#
-hptrr_lib.o			optional	hptrr			\
-	dependency	"$S/dev/hptrr/i386-elf.hptrr_lib.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptrr/i386-elf.hptrr_lib.o.uu" \
-	no-implicit-rule
-#
 compat/linprocfs/linprocfs.c	optional linprocfs
 compat/linsysfs/linsysfs.c	optional linsysfs
@@ -181,14 +145,6 @@
 dev/fe/if_fe_isa.c		optional fe isa
 dev/glxsb/glxsb.c		optional glxsb
 dev/glxsb/glxsb_hash.c		optional glxsb
-dev/hptmv/entry.c		optional hptmv
-dev/hptmv/mv.c			optional hptmv
-dev/hptmv/gui_lib.c		optional hptmv
-dev/hptmv/hptproc.c		optional hptmv
-dev/hptmv/ioctl.c		optional hptmv
-dev/hptrr/hptrr_os_bsd.c	optional hptrr
-dev/hptrr/hptrr_osm_bsd.c	optional hptrr
-dev/hptrr/hptrr_config.c	optional hptrr
 dev/hwpmc/hwpmc_amd.c		optional hwpmc
 dev/hwpmc/hwpmc_pentium.c	optional hwpmc
 dev/hwpmc/hwpmc_piv.c		optional hwpmc
@@ -215,7 +171,6 @@
 dev/mse/mse.c			optional mse
 dev/mse/mse_isa.c		optional mse isa
 dev/nfe/if_nfe.c		optional nfe pci
-dev/nve/if_nve.c		optional nve pci
 dev/nvram/nvram.c		optional nvram isa
 dev/pcf/pcf_isa.c		optional pcf
 dev/random/nehemiah.c		optional random
Index: src/sys/i386/conf/GENERIC
===================================================================
--- src.orig/sys/i386/conf/GENERIC
+++ src/sys/i386/conf/GENERIC
@@ -131,8 +131,6 @@
 device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		hptrr		# Highpoint RocketRAID 17xx, 22xx, 23xx, 25xx
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
Index: src/sys/modules/Makefile
===================================================================
--- src.orig/sys/modules/Makefile
+++ src/sys/modules/Makefile
@@ -100,8 +95,6 @@
 	hifn \
 	hme \
 	${_hptiop} \
-	${_hptmv} \
-	${_hptrr} \
 	hwpmc \
 	${_i2c} \
 	${_ibcs2} \
@@ -200,11 +193,9 @@
 	ntfs_iconv \
 	nullfs \
-	${_nve} \
 	${_nvram} \
 	${_nwfs} \
 	${_nxge} \
-	${_opensolaris} \
 	ohci \
 	${_padlock} \
 	patm \
 	${_pccard} \
@@ -327,7 +317,6 @@
 	${_xe} \
 	xfs \
 	xl \
-	${_zfs} \
 	zlib \
 	zyd
 
@@ -422,8 +406,5 @@
 .if ${MK_NCP} != "no"
 _nwfs=		nwfs
 .endif
-.if ${MK_CDDL} != "no" || defined(ALL_MODULES)
-_opensolaris=	opensolaris
-.endif
 _pccard=	pccard
 _pcfclock=	pcfclock
@@ -447,9 +427,6 @@
 _tmpfs= 	tmpfs
 _wi=		wi
 _xe=		xe
-.if ${MK_ZFS} != "no" || defined(ALL_MODULES)
-_zfs=		zfs
-.endif
 .if ${MACHINE} == "i386"
 _aac=		aac
 _acpi=		acpi
@@ -467,8 +444,6 @@
 _dpt=		dpt
 _ex=		ex
 _hptiop=	hptiop
-_hptmv=		hptmv
-_hptrr=		hptrr
 _ichwd=		ichwd
 _ida=		ida
 _iir=		iir
@@ -481,7 +456,6 @@
 _ixgb=   	ixgb
 _mly=		mly
 _nfe=		nfe
-_nve=		nve
 _nvram=		nvram
 _nxge=		nxge
 _wpi=		wpi
@@ -533,8 +502,6 @@
 _exca=		exca
 _ext2fs=	ext2fs
 _hptiop=	hptiop
-_hptmv=		hptmv
-_hptrr=		hptrr
 _i2c=		i2c
 _ichwd=		ichwd
 _ida=		ida
@@ -554,12 +521,8 @@
 _mly=		mly
 _ndis=		ndis
 _nfe=		nfe
-_nve=		nve
 _nvram=		nvram
 _nxge=		nxge
-.if ${MK_CDDL} != "no" || defined(ALL_MODULES)
-_opensolaris=	opensolaris
-.endif
 .if ${MK_CRYPT} != "no" || defined(ALL_MODULES)
 .if exists(${.CURDIR}/../crypto/via)
 _padlock=	padlock
@@ -573,9 +536,6 @@
 _wi=		wi
 _wpi=		wpi
 _wpifw=		wpifw
-.if ${MK_ZFS} != "no" || defined(ALL_MODULES)
-_zfs=		zfs
-.endif
 .endif
 
 .if ${MACHINE_ARCH} == "ia64"
Index: src/sys/amd64/conf/GENERIC
===================================================================
--- src.orig/sys/amd64/conf/GENERIC
+++ src/sys/amd64/conf/GENERIC
@@ -125,8 +125,6 @@
 device		arcmsr		# Areca SATA II RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		hptrr		# Highpoint RocketRAID 17xx, 22xx, 23xx, 25xx
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
Index: src/sys/conf/files.amd64
===================================================================
--- src.orig/sys/conf/files.amd64
+++ src/sys/conf/files.amd64
@@ -47,26 +47,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/amd64-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/dev/hptmv/amd64-elf.raid.o.uu" \
-	no-implicit-rule
-hptrr_lib.o			optional	hptrr			\
-	dependency	"$S/dev/hptrr/amd64-elf.hptrr_lib.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptrr/amd64-elf.hptrr_lib.o.uu" \
-	no-implicit-rule
-#
 amd64/acpica/OsdEnvironment.c	optional	acpi
 amd64/acpica/acpi_machdep.c	optional	acpi
 amd64/acpica/acpi_wakeup.c	optional	acpi
@@ -173,14 +153,6 @@
 dev/fdc/fdc_acpi.c		optional	fdc
 dev/fdc/fdc_isa.c		optional	fdc isa
 dev/fdc/fdc_pccard.c		optional	fdc pccard
-dev/hptmv/entry.c		optional	hptmv
-dev/hptmv/mv.c			optional	hptmv
-dev/hptmv/gui_lib.c		optional	hptmv
-dev/hptmv/hptproc.c		optional	hptmv
-dev/hptmv/ioctl.c		optional	hptmv
-dev/hptrr/hptrr_os_bsd.c	optional	hptrr
-dev/hptrr/hptrr_osm_bsd.c	optional	hptrr
-dev/hptrr/hptrr_config.c	optional	hptrr
 dev/hwpmc/hwpmc_amd.c		optional	hwpmc
 dev/hwpmc/hwpmc_piv.c		optional	hwpmc
 dev/hwpmc/hwpmc_x86.c		optional	hwpmc
@@ -191,7 +163,6 @@
 dev/kbd/kbd.c			optional	atkbd | sc | ukbd
 dev/mem/memutil.c		optional	mem
 dev/nfe/if_nfe.c		optional	nfe pci
-dev/nve/if_nve.c		optional	nve pci
 dev/nvram/nvram.c		optional	nvram isa
 dev/sio/sio.c			optional	sio
 dev/sio/sio_isa.c		optional	sio isa
