
Status: Rejected in upstream (kern/77355).

--- src.orig/sys/i386/i386/identcpu.c
+++ src/sys/i386/i386/identcpu.c
@@ -634,6 +634,7 @@
 #if defined(I486_CPU)
 	case CPUCLASS_486:
 		printf("486");
+		machine[1] = '4';
 		bzero_vector = i486_bzero;
 		break;
 #endif
@@ -644,6 +645,7 @@
 		       (intmax_t)(tsc_freq + 4999) / 1000000,
 		       (u_int)((tsc_freq + 4999) / 10000) % 100);
 		printf("586");
+		machine[1] = '5';
 		break;
 #endif
 #if defined(I686_CPU)
@@ -653,6 +655,7 @@
 		       (intmax_t)(tsc_freq + 4999) / 1000000,
 		       (u_int)((tsc_freq + 4999) / 10000) % 100);
 		printf("686");
+		machine[1] = '6';
 		break;
 #endif
 	default:
