---
 sys/compat/linprocfs/linprocfs.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/sys/compat/linprocfs/linprocfs.c
+++ b/sys/compat/linprocfs/linprocfs.c
@@ -358,10 +358,6 @@
 
 		/* determine fs type */
 		fstype = mp->mnt_stat.f_fstypename;
-		if (strcmp(fstype, pn->pn_info->pi_name) == 0)
-			mntfrom = fstype = "proc";
-		else if (strcmp(fstype, "procfs") == 0)
-			continue;
 
 		if (strcmp(fstype, "linsysfs") == 0) {
 			sbuf_printf(sb, "/sys %s sysfs %s", mntto,
