Index: src/sys/i386/conf/GENERIC.hints
===================================================================
--- src.orig/sys/i386/conf/GENERIC.hints
+++ src/sys/i386/conf/GENERIC.hints
@@ -18,6 +18,7 @@
 hint.atkbd.0.at="atkbdc"
 hint.atkbd.0.irq="1"
 hint.psm.0.at="atkbdc"
+hint.psm.0.flags="0x100"
 hint.psm.0.irq="12"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
Index: src/sys/amd64/conf/GENERIC.hints
===================================================================
--- src.orig/sys/amd64/conf/GENERIC.hints
+++ src/sys/amd64/conf/GENERIC.hints
@@ -12,6 +12,7 @@
 hint.atkbd.0.at="atkbdc"
 hint.atkbd.0.irq="1"
 hint.psm.0.at="atkbdc"
+hint.psm.0.flags="0x100"
 hint.psm.0.irq="12"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
