
From upstream 8-STABLE (SVN rev 234512).

--- a/sys/fs/tmpfs/tmpfs_vfsops.c
+++ b/sys/fs/tmpfs/tmpfs_vfsops.c
@@ -155,9 +155,6 @@
 		return EOPNOTSUPP;
 	}
 
-	printf("WARNING: TMPFS is considered to be a highly experimental "
-	    "feature in FreeBSD.\n");
-
 	vn_lock(mp->mnt_vnodecovered, LK_SHARED | LK_RETRY);
 	error = VOP_GETATTR(mp->mnt_vnodecovered, &va, mp->mnt_cred);
 	VOP_UNLOCK(mp->mnt_vnodecovered, 0);
