--- src/sys/sys/mount.h~	2002-04-26 02:46:00.000000000 +0200
+++ src/sys/sys/mount.h	2004-11-26 21:16:36.000000000 +0100
@@ -50,7 +50,9 @@
 #include <sys/lock.h>
 #endif
 
-typedef struct fsid { int32_t val[2]; } fsid_t;	/* file system id type */
+#ifndef __USE_BSD
+typedef __fsid_t fsid_t;	/* file system id type */
+#endif
 
 /*
  * File identifier.
--- src/sys/sys/mount.h~	2004-11-26 21:17:08.000000000 +0100
+++ src/sys/sys/mount.h	2004-11-26 21:20:59.000000000 +0100
@@ -478,7 +478,6 @@
 int	statfs __P((const char *, struct statfs *));
 int	unmount __P((const char *, int));
 int	fhopen __P((const struct fhandle *, int));
-int	fhstat __P((const struct fhandle *, struct stat *));
 int	fhstatfs __P((const struct fhandle *, struct statfs *));
 
 /* C library stuff */
--- src/sys/sys/mount.h~	2004-11-26 21:21:25.000000000 +0100
+++ src/sys/sys/mount.h	2004-11-26 21:32:01.000000000 +0100
@@ -78,31 +78,6 @@
 #define	MNAMELEN	72	/* length of buffer for returned name */
 #endif
 
-struct statfs {
-	long	f_spare2;		/* placeholder */
-	long	f_bsize;		/* fundamental file system block size */
-	long	f_iosize;		/* optimal transfer block size */
-	long	f_blocks;		/* total data blocks in file system */
-	long	f_bfree;		/* free blocks in fs */
-	long	f_bavail;		/* free blocks avail to non-superuser */
-	long	f_files;		/* total file nodes in file system */
-	long	f_ffree;		/* free file nodes in fs */
-	fsid_t	f_fsid;			/* file system id */
-	uid_t	f_owner;		/* user that mounted the filesystem */
-	int	f_type;			/* type of filesystem */
-	int	f_flags;		/* copy of mount exported flags */
-	long    f_syncwrites;		/* count of sync writes since mount */
-	long    f_asyncwrites;		/* count of async writes since mount */
-	char	f_fstypename[MFSNAMELEN]; /* fs type name */
-	char	f_mntonname[MNAMELEN];	/* directory on which mounted */
-	long    f_syncreads;		/* count of sync reads since mount */
-	long    f_asyncreads;		/* count of async reads since mount */
-	short	f_spares1;		/* unused spare */
-	char	f_mntfromname[MNAMELEN];/* mounted filesystem */
-	short	f_spares2;		/* unused spare */
-	long    f_spare[2];		/* unused spare */
-};
-
 #ifdef _KERNEL
 /*
  * Structure per mounted file system.  Each mounted file system has an
--- src/sys/sys/signal.h~	2001-04-19 03:38:35.000000000 +0200
+++ src/sys/sys/signal.h	2004-11-26 21:49:03.000000000 +0100
@@ -229,11 +229,6 @@
 
 typedef	__sighandler_t	*sig_t;	/* type of pointer to a signal function */
 
-#ifdef	_BSD_SIZE_T_
-typedef	_BSD_SIZE_T_	size_t;
-#undef	_BSD_SIZE_T_
-#endif
-
 /*
  * Structure used in sigaltstack call.
  */
--- src/sys/sys/types.h~	2001-04-21 16:53:06.000000000 +0200
+++ src/sys/sys/types.h	2004-11-26 21:49:25.000000000 +0100
@@ -135,11 +135,6 @@
 #undef	_BSD_CLOCKID_T_
 #endif
 
-#ifdef	_BSD_SIZE_T_
-typedef	_BSD_SIZE_T_	size_t;
-#undef	_BSD_SIZE_T_
-#endif
-
 #ifdef	_BSD_SSIZE_T_
 typedef	_BSD_SSIZE_T_	ssize_t;
 #undef	_BSD_SSIZE_T_
--- src/sys/i386/machine/ansi.h~	2004-11-27 04:45:40.000000000 +0100
+++ src/sys/i386/machine/ansi.h	2004-11-27 04:45:45.000000000 +0100
@@ -34,8 +34,8 @@
  * $FreeBSD: src/sys/i386/include/ansi.h,v 1.18.2.4 2001/06/03 17:15:54 obrien Exp $
  */
 
-#ifndef _MACHINE_ANSI_H_
-#define	_MACHINE_ANSI_H_
+#ifndef _MACHINE_ANSI_H__
+#define	_MACHINE_ANSI_H__
 
 /*
  * Types which are fundamental to the implementation and must be declared
@@ -54,7 +54,6 @@
 #define	_BSD_SSIZE_T_	int			/* byte count or error */
 #define	_BSD_TIME_T_	long			/* time()... */
 #define	_BSD_TIMER_T_	int			/* timer_gettime()... */
-#define	_BSD_WCHAR_T_	_BSD_CT_RUNE_T_		/* wchar_t (see below) */
 #define	_BSD_WINT_T_	_BSD_CT_RUNE_T_		/* wint_t (see below) */
 
 /*
