diff -ur src.old/sys/cam/cam.h src/sys/cam/cam.h
--- src.old/sys/cam/cam.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/cam/cam.h	2004-09-16 15:11:07.000000000 +0200
@@ -36,6 +36,7 @@
 #endif
 
 #include <sys/cdefs.h>
+#include <sys/types.h>
 
 typedef u_int path_id_t;
 typedef u_int target_id_t;
diff -ur src.old/sys/cam/cam_ccb.h src/sys/cam/cam_ccb.h
--- src.old/sys/cam/cam_ccb.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/cam/cam_ccb.h	2004-09-16 15:11:07.000000000 +0200
@@ -31,7 +31,7 @@
 #ifndef _CAM_CAM_CCB_H
 #define _CAM_CAM_CCB_H 1
 
-#include <sys/queue.h>
+#include "/usr/src/kfreebsd4-headers/sys/queue.h"
 #include <sys/cdefs.h>
 #include <sys/time.h>
 #ifndef _KERNEL
@@ -39,6 +39,7 @@
 #endif
 #include <cam/cam_debug.h>
 #include <cam/scsi/scsi_all.h>
+#include <cam/cam.h>
 
 
 /* General allocation length definitions for CCB structures */
diff -ur src.old/sys/cam/scsi/scsi_low.h src/sys/cam/scsi/scsi_low.h
--- src.old/sys/cam/scsi/scsi_low.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/cam/scsi/scsi_low.h	2004-09-16 15:11:07.000000000 +0200
@@ -44,6 +44,11 @@
 #ifndef	_SCSI_LOW_H_
 #define	_SCSI_LOW_H_
 
+#ifndef NBBY
+#define NBBY 8
+#endif
+#include <errno.h>	/* EIO */
+
 /*================================================
  * Scsi low OSDEP 
  * (All os depend structures should be here!)
@@ -73,7 +78,7 @@
 #include <cam/cam_debug.h>
 
 #include <cam/scsi/scsi_dvcfg.h>
-#include <i386/isa/ccbque.h>
+#include "/usr/src/kfreebsd5-headers/i386/isa/ccbque.h"
 #endif	/* __FreeBSD__ */
 
 /******** functions macro ************************/
@@ -92,6 +97,8 @@
 #define	SCSI_LOW_BZERO(pt, size)	bzero((pt), (size))
 #endif	/* __FreeBSD__ */
 
+typedef struct device           *device_t;
+
 /******** os depend interface structures **********/
 #ifdef	__NetBSD__
 typedef	struct scsipi_sense_data scsi_low_osdep_sense_data_t;
diff -ur src.old/sys/cam/scsi/scsi_pt.h src/sys/cam/scsi/scsi_pt.h
--- src.old/sys/cam/scsi/scsi_pt.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/cam/scsi/scsi_pt.h	2004-09-16 15:11:07.000000000 +0200
@@ -31,6 +31,8 @@
 #ifndef	_SCSI_SCSI_PT_H
 #define _SCSI_SCSI_PT_H 1
 
+#include <sys/types.h>
+
 struct scsi_send_receive
 {
 	u_int8_t opcode;
diff -ur src.old/sys/cam/scsi/scsi_sa.h src/sys/cam/scsi/scsi_sa.h
--- src.old/sys/cam/scsi/scsi_sa.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/cam/scsi/scsi_sa.h	2004-09-16 15:11:07.000000000 +0200
@@ -33,6 +33,7 @@
 #define _SCSI_SCSI_SA_H 1
 
 #include <sys/cdefs.h>
+#include <sys/types.h>
 
 struct scsi_read_block_limits
 {
diff -ur src.old/sys/dev/aac/aac_compat.h src/sys/dev/aac/aac_compat.h
--- src.old/sys/dev/aac/aac_compat.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aac/aac_compat.h	2004-09-16 15:11:07.000000000 +0200
@@ -34,7 +34,7 @@
  * Handle the new/old bio/buf changeover
  */
 
-#if __FreeBSD_version < 500003		/* old buf style */
+#if 1
 #include <sys/buf.h>
 #define FREEBSD_4
 #define bio					buf
diff -ur src.old/sys/dev/aac/aac_ioctl.h src/sys/dev/aac/aac_ioctl.h
--- src.old/sys/dev/aac/aac_ioctl.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aac/aac_ioctl.h	2004-09-16 15:11:07.000000000 +0200
@@ -28,6 +28,9 @@
  *	$FreeBSD: src/sys/dev/aac/aac_ioctl.h,v 1.2.2.2 2001/09/19 19:09:11 scottl Exp $
  */
 
+#include <sys/types.h>
+#include <dev/aac/aacreg.h>
+
 /*
  * Command queue statistics
  */
diff -ur src.old/sys/dev/aac/aac_tables.h src/sys/dev/aac/aac_tables.h
--- src.old/sys/dev/aac/aac_tables.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aac/aac_tables.h	2004-09-16 15:11:07.000000000 +0200
@@ -27,6 +27,8 @@
  *	$FreeBSD: src/sys/dev/aac/aac_tables.h,v 1.1.2.3 2001/12/04 02:38:04 scottl Exp $
  */
 
+#include <dev/aac/aacvar.h>
+
 /*
  * Status codes for block read/write commands, etc.
  *
diff -ur src.old/sys/dev/aac/aacreg.h src/sys/dev/aac/aacreg.h
--- src.old/sys/dev/aac/aacreg.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aac/aacreg.h	2004-09-16 15:11:07.000000000 +0200
@@ -29,6 +29,11 @@
  *	$FreeBSD: src/sys/dev/aac/aacreg.h,v 1.4.2.5 2002/04/30 22:58:38 scottl Exp $
  */
 
+#ifndef _DEV_AAC_AACREG_H
+#define _DEV_AAC_AACREG_H 1
+
+#include <sys/types.h>
+
 /*
  * Data structures defining the interface between the driver and the Adaptec
  * 'FSA' adapters.  Note that many field names and comments here are taken
@@ -1330,3 +1335,5 @@
 #define AAC_DB_INTERRUPTS	(AAC_DB_COMMAND_READY  |	\
 				 AAC_DB_RESPONSE_READY |	\
 				 AAC_DB_PRINTF)
+
+#endif		/* _DEV_AAC_AACREG_H */
diff -ur src.old/sys/dev/aac/aacvar.h src/sys/dev/aac/aacvar.h
--- src.old/sys/dev/aac/aacvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aac/aacvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -29,6 +29,21 @@
  *	$FreeBSD: src/sys/dev/aac/aacvar.h,v 1.4.2.6 2002/04/30 22:58:38 scottl Exp $
  */
 
+#include <dev/aac/aacreg.h>
+#include <sys/queue.h>
+#include "/usr/src/kfreebsd4-headers/sys/disk.h"
+#include <sys/devicestat.h>
+#include <sys/disklabel.h>
+#include <machine/bus_dma.h>
+#include <dev/aac/aac_compat.h>
+#include <dev/aac/aac_ioctl.h>		/* AACQ_COUNT */
+#include <sys/kernel.h>			/* intr_config_hook */
+#include <stddef.h>			/* NULL */
+
+/* Borrowed from sys/bus.h */
+typedef struct device           *device_t;
+typedef struct devclass         *devclass_t;
+
 /*
  * Driver Parameter Definitions
  */
@@ -94,10 +109,6 @@
  * Driver Variable Definitions
  */
 
-#if __FreeBSD_version >= 500005
-# include <sys/taskqueue.h>
-#endif
-
 /*
  * Per-container data structure
  */
@@ -250,7 +261,7 @@
 TAILQ_HEAD(aac_container_tq, aac_container);
 
 /* Define the OS version specific locks */
-#if __FreeBSD_version >= 500005
+#if 0
 #include <sys/lock.h>
 #include <sys/mutex.h>
 typedef struct mtx aac_lock_t;
@@ -264,11 +275,14 @@
 #define AAC_LOCK_RELEASE(l)	simple_unlock(l)
 #endif
 
-#if __FreeBSD_version >= 500005
-#include <sys/selinfo.h>
-#else
 #include <sys/select.h>
-#endif
+#include <sys/event.h>
+
+struct selinfo {
+        pid_t   si_pid;         /* process to be notified */
+        struct  klist si_note;  /* kernel note list */
+        short   si_flags;       /* see below */
+};
 
 /*
  * Per-controller structure.
@@ -341,7 +355,7 @@
 	aac_lock_t		aac_sync_lock;
 
 	/* delayed activity infrastructure */
-#if __FreeBSD_version >= 500005
+#if 0
 	struct task		aac_task_complete;	/* deferred-completion
 							 * task */
 #endif
diff -ur src.old/sys/dev/advansys/advansys.h src/sys/dev/advansys/advansys.h
--- src.old/sys/dev/advansys/advansys.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/advansys/advansys.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,7 @@
 #define _ADVANSYS_H_
 
 #include <dev/advansys/advlib.h>
+#include <sys/systm.h>			/* timeout_t */
 
 struct adv_softc *	adv_alloc(device_t dev, bus_space_tag_t tag,
 				  bus_space_handle_t bsh);
diff -ur src.old/sys/dev/advansys/advlib.h src/sys/dev/advansys/advlib.h
--- src.old/sys/dev/advansys/advlib.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/advansys/advlib.h	2004-09-16 15:11:07.000000000 +0200
@@ -48,6 +48,21 @@
 #define _ADVLIB_H_
 
 #include <sys/queue.h>
+#ifndef SLIST_ENTRY
+#define SLIST_ENTRY(type)                                               \
+struct {                                                                \
+        struct type *sle_next;  /* next element */                      \
+}
+#endif
+#ifndef SLIST_HEAD
+#define SLIST_HEAD(name, type)                                          \
+struct name {                                                           \
+        struct type *slh_first; /* first element */                     \
+}
+#endif
+#include <sys/types.h>
+#include <machine/bus_dma.h>		/* bus_dmamap_t */
+typedef struct device           *device_t;
 
 struct cam_path;
 union ccb;
diff -ur src.old/sys/dev/advansys/advmcode.h src/sys/dev/advansys/advmcode.h
--- src.old/sys/dev/advansys/advmcode.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/advansys/advmcode.h	2004-09-16 15:11:07.000000000 +0200
@@ -14,6 +14,8 @@
  * modification.
  */
 
+#include <sys/types.h>
+
 extern u_int16_t adv_mcode[];
 extern u_int16_t adv_mcode_size;
 extern u_int32_t adv_mcode_chksum;
diff -ur src.old/sys/dev/advansys/adwlib.h src/sys/dev/advansys/adwlib.h
--- src.old/sys/dev/advansys/adwlib.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/advansys/adwlib.h	2004-09-16 15:11:07.000000000 +0200
@@ -47,11 +47,26 @@
 #ifndef _ADWLIB_H_
 #define _ADWLIB_H_
 
-#include "opt_adw.h"
-
 #include <stddef.h>	/* for offsetof */
 
 #include <dev/advansys/adwmcode.h>
+#include <machine/bus_dma.h>		/* bus_dmamap_t */
+#include <sys/queue.h>
+#ifndef SLIST_ENTRY
+#define SLIST_ENTRY(type)                                               \
+struct {                                                                \
+        struct type *sle_next;  /* next element */                      \
+}
+#endif
+#ifndef SLIST_HEAD
+#define SLIST_HEAD(name, type)                                          \
+struct name {                                                           \
+        struct type *slh_first; /* first element */                     \
+}
+#endif
+typedef struct device           *device_t;
+#include <cam/scsi/scsi_all.h>		/* scsi_sense_data */
+#include <cam/cam.h>			/* cam_status */
 
 #define ADW_DEF_MAX_HOST_QNG	253
 #define ADW_DEF_MIN_HOST_QNG	16
diff -ur src.old/sys/dev/advansys/adwmcode.h src/sys/dev/advansys/adwmcode.h
--- src.old/sys/dev/advansys/adwmcode.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/advansys/adwmcode.h	2004-09-16 15:11:07.000000000 +0200
@@ -17,6 +17,8 @@
 #ifndef _ADMCODE_H_
 #define _ADMCODE_H_
 
+#include <sys/types.h>
+
 struct adw_mcode
 {
 	const u_int8_t*	mcode_buf;
diff -ur src.old/sys/dev/advansys/adwvar.h src/sys/dev/advansys/adwvar.h
--- src.old/sys/dev/advansys/adwvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/advansys/adwvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,7 @@
 #define _ADWVAR_H_
 
 #include <dev/advansys/adwlib.h>
+#include <sys/systm.h>			/* timeout_t */
 
 struct adw_softc *	adw_alloc(device_t dev, struct resource *regs,
 				  int regs_type, int regs_id);
diff -ur src.old/sys/dev/aha/ahareg.h src/sys/dev/aha/ahareg.h
--- src.old/sys/dev/aha/ahareg.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aha/ahareg.h	2004-09-16 15:11:07.000000000 +0200
@@ -36,10 +36,21 @@
 #ifndef _AHAREG_H_
 #define _AHAREG_H_
 
-#include "aha.h"
-
 #include <sys/queue.h>
+#ifndef SLIST_ENTRY
+#define SLIST_ENTRY(type)                                               \
+struct {                                                                \
+        struct type *sle_next;  /* next element */                      \
+}
+#endif
+#ifndef SLIST_HEAD
+#define SLIST_HEAD(name, type)                                          \
+struct name {                                                           \
+        struct type *slh_first; /* first element */                     \
+}
+#endif
 #include <cam/scsi/scsi_all.h>
+#include <machine/bus_dma.h>		/* bus_dmamap_t */
 
 #define AHA_MAXTRANSFER_SIZE	 0xffffff	/* limited by 24bit counter */
 #define AHA_NSEG		17	/* The number of dma segments 
diff -ur src.old/sys/dev/ahb/ahbreg.h src/sys/dev/ahb/ahbreg.h
--- src.old/sys/dev/ahb/ahbreg.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/ahb/ahbreg.h	2004-09-16 15:11:07.000000000 +0200
@@ -28,6 +28,23 @@
  * $FreeBSD: src/sys/dev/ahb/ahbreg.h,v 1.2 1999/08/28 00:43:37 peter Exp $
  */
 
+#include <sys/types.h>
+#include <sys/queue.h>
+#ifndef SLIST_ENTRY
+#define SLIST_ENTRY(type)                                               \
+struct {                                                                \
+        struct type *sle_next;  /* next element */                      \
+}
+#endif
+#ifndef SLIST_HEAD
+#define SLIST_HEAD(name, type)                                          \
+struct name {                                                           \
+        struct type *slh_first; /* first element */                     \
+}
+#endif
+#include <cam/scsi/scsi_all.h>		/* scsi_inquiry_data */
+#include <machine/bus_dma.h>		/* bus_dmamap_t */
+
 /* Resource Constatns */
 #define	AHB_NECB	64
 #define AHB_NSEG	32
diff -ur src.old/sys/dev/aic/aicvar.h src/sys/dev/aic/aicvar.h
--- src.old/sys/dev/aic/aicvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aic/aicvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -26,6 +26,18 @@
  * $FreeBSD: src/sys/dev/aic/aicvar.h,v 1.2.2.1 2000/08/08 23:51:23 peter Exp $
  */
 
+#include <sys/types.h>
+#include <machine/bus_at386.h>		/* bus_space_tag_t */
+#include <machine/bus_dma.h>		/* bus_dma_tag_t */
+#include <sys/queue.h>
+#ifndef TAILQ_HEAD
+#define TAILQ_HEAD(name, type)                                          \
+struct name {                                                           \
+        struct type *tqh_first; /* first element */                     \
+        struct type **tqh_last; /* addr of last next element */         \
+}
+#endif
+
 struct aic_transinfo {
 	u_int8_t period;
 	u_int8_t offset;
diff -ur src.old/sys/dev/aic7xxx/aic7xxx.h src/sys/dev/aic7xxx/aic7xxx.h
--- src.old/sys/dev/aic7xxx/aic7xxx.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aic7xxx/aic7xxx.h	2004-09-16 15:11:07.000000000 +0200
@@ -45,8 +45,11 @@
 #ifndef _AIC7XXX_H_
 #define _AIC7XXX_H_
 
-/* Register Definitions */
-#include "aic7xxx_reg.h"
+#include <sys/types.h>
+#include <stdint.h>
+#include <machine/bus_dma.h>		/* bus_dmamap_t */
+#include "/usr/src/kfreebsd4-headers/sys/queue.h"
+#include <dev/aic7xxx/aic7xxx_osm.h>	/* ahc_io_ctx_t */
 
 /************************* Forward Declarations *******************************/
 struct ahc_platform_data;
@@ -381,21 +384,21 @@
  * complete abnormally (non-zero status byte).
  */
 struct status_pkt {
-	uint32_t residual_datacnt;	/* Residual in the current S/G seg */
-	uint32_t residual_sg_ptr;	/* The next S/G for this transfer */
-	uint8_t	 scsi_status;		/* Standard SCSI status byte */
+	u_int32_t residual_datacnt;	/* Residual in the current S/G seg */
+	u_int32_t residual_sg_ptr;	/* The next S/G for this transfer */
+	u_int8_t	 scsi_status;		/* Standard SCSI status byte */
 };
 
 /*
  * Target mode version of the shared data SCB segment.
  */
 struct target_data {
-	uint32_t residual_datacnt;	/* Residual in the current S/G seg */
-	uint32_t residual_sg_ptr;	/* The next S/G for this transfer */
-	uint8_t  scsi_status;		/* SCSI status to give to initiator */
-	uint8_t  target_phases;		/* Bitmap of phases to execute */
-	uint8_t  data_phase;		/* Data-In or Data-Out */
-	uint8_t  initiator_tag;		/* Initiator's transaction tag */
+	u_int32_t residual_datacnt;	/* Residual in the current S/G seg */
+	u_int32_t residual_sg_ptr;	/* The next S/G for this transfer */
+	u_int8_t  scsi_status;		/* SCSI status to give to initiator */
+	u_int8_t  target_phases;		/* Bitmap of phases to execute */
+	u_int8_t  data_phase;		/* Data-In or Data-Out */
+	u_int8_t  initiator_tag;		/* Initiator's transaction tag */
 };
 
 struct hardware_scb {
@@ -406,8 +409,8 @@
 		 * of the cdb payload as seen by the chip and a DMA
 		 * is used to pull it in.
 		 */
-		uint8_t	 cdb[12];
-		uint32_t cdb_ptr;
+		u_int8_t	 cdb[12];
+		u_int32_t cdb_ptr;
 		struct	 status_pkt status;
 		struct	 target_data tdata;
 	} shared_data;
@@ -448,31 +451,31 @@
  *	  and so may point beyond the last valid sg entry for the
  *	  transfer.
  */ 
-/*12*/	uint32_t dataptr;
-/*16*/	uint32_t datacnt;		/*
+/*12*/	u_int32_t dataptr;
+/*16*/	u_int32_t datacnt;		/*
 					 * Byte 3 (numbered from 0) of
 					 * the datacnt is really the
 					 * 4th byte in that data address.
 					 */
-/*20*/	uint32_t sgptr;
+/*20*/	u_int32_t sgptr;
 #define SG_PTR_MASK	0xFFFFFFF8
-/*24*/	uint8_t  control;	/* See SCB_CONTROL in aic7xxx.reg for details */
-/*25*/	uint8_t  scsiid;	/* what to load in the SCSIID register */
-/*26*/	uint8_t  lun;
-/*27*/	uint8_t  tag;			/*
+/*24*/	u_int8_t  control;	/* See SCB_CONTROL in aic7xxx.reg for details */
+/*25*/	u_int8_t  scsiid;	/* what to load in the SCSIID register */
+/*26*/	u_int8_t  lun;
+/*27*/	u_int8_t  tag;			/*
 					 * Index into our kernel SCB array.
 					 * Also used as the tag for tagged I/O
 					 */
-/*28*/	uint8_t  cdb_len;
-/*29*/	uint8_t  scsirate;		/* Value for SCSIRATE register */
-/*30*/	uint8_t  scsioffset;		/* Value for SCSIOFFSET register */
-/*31*/	uint8_t  next;			/*
+/*28*/	u_int8_t  cdb_len;
+/*29*/	u_int8_t  scsirate;		/* Value for SCSIRATE register */
+/*30*/	u_int8_t  scsioffset;		/* Value for SCSIOFFSET register */
+/*31*/	u_int8_t  next;			/*
 					 * Used for threading SCBs in the
 					 * "Waiting for Selection" and
 					 * "Disconnected SCB" lists down
 					 * in the sequencer.
 					 */
-/*32*/	uint8_t  cdb32[32];		/*
+/*32*/	u_int8_t  cdb32[32];		/*
 					 * CDB storage for cdbs of size
 					 * 13->32.  We store them here
 					 * because hardware scbs are
@@ -499,8 +502,8 @@
  * that can support dual address cycles on 32bit PCI busses.
  */
 struct ahc_dma_seg {
-	uint32_t	addr;
-	uint32_t	len;
+	u_int32_t	addr;
+	u_int32_t	len;
 #define	AHC_DMA_LAST_SEG	0x80000000
 #define	AHC_SG_HIGH_ADDR_MASK	0x7F000000
 #define	AHC_SG_LEN_MASK		0x00FFFFFF
@@ -587,9 +590,9 @@
 	bus_addr_t	 sense_busaddr;
 	bus_dma_tag_t	 sg_dmat;	/* dmat for our sg segments */
 	SLIST_HEAD(, sg_map_node) sg_maps;
-	uint8_t	numscbs;
-	uint8_t	maxhscbs;		/* Number of SCBs on the card */
-	uint8_t	init_level;		/*
+	u_int8_t	numscbs;
+	u_int8_t	maxhscbs;		/* Number of SCBs on the card */
+	u_int8_t	init_level;		/*
 					 * How far we've initialized
 					 * this structure.
 					 */
@@ -601,14 +604,14 @@
  * Connection desciptor for select-in requests in target mode.
  */
 struct target_cmd {
-	uint8_t scsiid;		/* Our ID and the initiator's ID */
-	uint8_t identify;	/* Identify message */
-	uint8_t bytes[22];	/* 
+	u_int8_t scsiid;		/* Our ID and the initiator's ID */
+	u_int8_t identify;	/* Identify message */
+	u_int8_t bytes[22];	/* 
 				 * Bytes contains any additional message
 				 * bytes terminated by 0xFF.  The remainder
 				 * is the cdb to execute.
 				 */
-	uint8_t cmd_valid;	/*
+	u_int8_t cmd_valid;	/*
 				 * When a command is complete, the firmware
 				 * will set cmd_valid to all bits set.
 				 * After the host has seen the command,
@@ -619,7 +622,7 @@
 				 * it on aic7880 hardware which only has
 				 * limited direct access to the DMA FIFO.
 				 */
-	uint8_t pad[7];
+	u_int8_t pad[7];
 };
 
 /*
@@ -628,10 +631,10 @@
  */
 #define AHC_TMODE_EVENT_BUFFER_SIZE 8
 struct ahc_tmode_event {
-	uint8_t initiator_id;
-	uint8_t event_type;	/* MSG type or EVENT_TYPE_BUS_RESET */
+	u_int8_t initiator_id;
+	u_int8_t event_type;	/* MSG type or EVENT_TYPE_BUS_RESET */
 #define	EVENT_TYPE_BUS_RESET 0xFF
-	uint8_t event_arg;
+	u_int8_t event_arg;
 };
 
 /*
diff -ur src.old/sys/dev/aic7xxx/aic7xxx_osm.h src/sys/dev/aic7xxx/aic7xxx_osm.h
--- src.old/sys/dev/aic7xxx/aic7xxx_osm.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aic7xxx/aic7xxx_osm.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,11 +37,6 @@
 #ifndef _AIC7XXX_FREEBSD_H_
 #define _AIC7XXX_FREEBSD_H_
 
-#include <opt_aic7xxx.h>	/* for config options */
-#ifndef NPCI
-#include <pci.h>
-#endif
-
 #include <sys/param.h>
 #include <sys/systm.h>
 #include <sys/bus.h>		/* For device_t */
diff -ur src.old/sys/dev/aic7xxx/aicasm/aicasm.h src/sys/dev/aic7xxx/aicasm/aicasm.h
--- src.old/sys/dev/aic7xxx/aicasm/aicasm.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aic7xxx/aicasm/aicasm.h	2004-09-16 15:11:07.000000000 +0200
@@ -45,7 +45,7 @@
 #ifdef __linux__
 #include "../queue.h"
 #else
-#include <sys/queue.h>
+#include "/usr/src/kfreebsd4-headers/sys/queue.h"
 #endif
 
 #ifndef TRUE
diff -ur src.old/sys/dev/aic7xxx/aicasm/aicasm_insformat.h src/sys/dev/aic7xxx/aicasm/aicasm_insformat.h
--- src.old/sys/dev/aic7xxx/aicasm/aicasm_insformat.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aic7xxx/aicasm/aicasm_insformat.h	2004-09-16 15:11:07.000000000 +0200
@@ -42,16 +42,19 @@
  * $FreeBSD: src/sys/dev/aic7xxx/aicasm/aicasm_insformat.h,v 1.3.2.3 2002/04/29 19:36:36 gibbs Exp $
  */
 
+#include <sys/types.h>
+#include "/usr/src/kfreebsd4-headers/sys/queue.h"
+
 struct ins_format1 {
 #if BYTE_ORDER == LITTLE_ENDIAN
-	uint32_t	immediate	: 8,
+	u_int32_t	immediate	: 8,
 			source		: 9,
 			destination	: 9,
 			ret		: 1,
 			opcode		: 4,
 			parity		: 1;
 #else
-	uint32_t	parity		: 1,
+	u_int32_t	parity		: 1,
 			opcode		: 4,
 			ret		: 1,
 			destination	: 9,
@@ -62,14 +65,14 @@
 
 struct ins_format2 {
 #if BYTE_ORDER == LITTLE_ENDIAN
-	uint32_t	shift_control	: 8,
+	u_int32_t	shift_control	: 8,
 			source		: 9,
 			destination	: 9,
 			ret		: 1,
 			opcode		: 4,
 			parity		: 1;
 #else
-	uint32_t	parity		: 1,
+	u_int32_t	parity		: 1,
 			opcode		: 4,
 			ret		: 1,
 			destination	: 9,
@@ -80,13 +83,13 @@
 
 struct ins_format3 {
 #if BYTE_ORDER == LITTLE_ENDIAN
-	uint32_t	immediate	: 8,
+	u_int32_t	immediate	: 8,
 			source		: 9,
 			address		: 10,
 			opcode		: 4,
 			parity		: 1;
 #else
-	uint32_t	parity		: 1,
+	u_int32_t	parity		: 1,
 			opcode		: 4,
 			address		: 10,
 			source		: 9,
@@ -98,8 +101,8 @@
 		struct ins_format1 format1;
 		struct ins_format2 format2;
 		struct ins_format3 format3;
-		uint8_t		   bytes[4];
-		uint32_t	   integer;
+		u_int8_t		   bytes[4];
+		u_int32_t	   integer;
 };
 struct instruction {
 	union	ins_formats format;
diff -ur src.old/sys/dev/aic7xxx/aicasm/aicasm_symbol.h src/sys/dev/aic7xxx/aicasm/aicasm_symbol.h
--- src.old/sys/dev/aic7xxx/aicasm/aicasm_symbol.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/aic7xxx/aicasm/aicasm_symbol.h	2004-09-16 15:11:07.000000000 +0200
@@ -44,8 +44,11 @@
 #ifdef __linux__
 #include "../queue.h"
 #else
-#include <sys/queue.h>
+#include "/usr/src/kfreebsd4-headers/sys/queue.h"
 #endif
+#include <sys/types.h>
+#include <regex.h>		/* regex_t */
+#include <stdio.h>		/* FILE* */
 
 typedef enum {
 	UNINITIALIZED,
diff -ur src.old/sys/dev/amr/amr_compat.h src/sys/dev/amr/amr_compat.h
--- src.old/sys/dev/amr/amr_compat.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/amr/amr_compat.h	2004-09-16 15:11:07.000000000 +0200
@@ -30,7 +30,7 @@
  * Backwards compatibility support.
  */
 
-#if __FreeBSD_version < 500003		/* old buf style */
+#if 1
 # include <sys/buf.h>
 # include <machine/clock.h>
 
diff -ur src.old/sys/dev/amr/amrvar.h src/sys/dev/amr/amrvar.h
--- src.old/sys/dev/amr/amrvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/amr/amrvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -27,7 +27,7 @@
  *      $FreeBSD: src/sys/dev/amr/amrvar.h,v 1.2.2.2 2000/10/28 10:16:59 msmith Exp $
  */
 
-#if __FreeBSD_version >= 500005
+#if 0
 # include <sys/taskqueue.h>
 #endif
 
@@ -185,7 +185,7 @@
     /* misc glue */
     struct intr_config_hook	amr_ich;		/* wait-for-interrupts probe hook */
     struct callout_handle	amr_timeout;		/* periodic status check */
-#if __FreeBSD_version >= 500005
+#if 0
     struct task			amr_task_complete;	/* deferred-completion task */
 #endif
 };
diff -ur src.old/sys/dev/bktr/bktr_reg.h src/sys/dev/bktr/bktr_reg.h
--- src.old/sys/dev/bktr/bktr_reg.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/bktr/bktr_reg.h	2004-09-16 15:11:07.000000000 +0200
@@ -35,7 +35,7 @@
  */
 
 #ifdef __FreeBSD_kernel__
-#  if (__FreeBSD_version >= 310000)
+#  if (1)
 #    include "smbus.h"
 #  else
 #    define NSMBUS 0		/* FreeBSD before 3.1 does not have SMBUS */
@@ -470,7 +470,7 @@
  * memory mapped structure method only works on 32 bit processors
  * with the right type of endianness.
  */
-#if defined(__NetBSD__) || ( defined(__FreeBSD_kernel__) && (__FreeBSD_version >=300000) )
+#if defined(__NetBSD__) || defined(__FreeBSD_kernel__)
 #define INB(bktr,offset)	bus_space_read_1((bktr)->memt,(bktr)->memh,(offset))
 #define INW(bktr,offset)	bus_space_read_2((bktr)->memt,(bktr)->memh,(offset))
 #define INL(bktr,offset)	bus_space_read_4((bktr)->memt,(bktr)->memh,(offset))
@@ -534,11 +534,11 @@
 #endif
 
 #if defined (__FreeBSD__)
-    #if (__FreeBSD_version < 400000)
+    #if 0
     vm_offset_t     phys_base;	/* 2.x Bt848 register physical address */
     pcici_t         tag;	/* 2.x PCI tag, for doing PCI commands */
     #endif
-    #if (__FreeBSD_version >= 400000)
+    #if 1
     int             mem_rid;	/* 4.x resource id */
     struct resource *res_mem;	/* 4.x resource descriptor for registers */
     int             irq_rid;	/* 4.x resource id */
@@ -551,7 +551,7 @@
     dev_t           tunerdev_alias;	/* alias /dev/tuner to /dev/tuner0 */
     dev_t           vbidev_alias;	/* alias /dev/vbi to /dev/vbi0 */
     #endif
-    #if (__FreeBSD_version >= 310000)
+    #if 1
     bus_space_tag_t	memt;	/* Bus space register access functions */
     bus_space_handle_t	memh;	/* Bus space register access functions */
     bus_size_t		obmemsz;/* Size of card (bytes) */
@@ -727,7 +727,7 @@
 #endif
 
 #if defined(__FreeBSD_kernel__)
-#if (__FreeBSD_version >= 300000)
+#if 1
 typedef u_long ioctl_cmd_t;
 #endif
 #endif
diff -ur src.old/sys/dev/ciss/cissvar.h src/sys/dev/ciss/cissvar.h
--- src.old/sys/dev/ciss/cissvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/ciss/cissvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -74,7 +74,7 @@
 /************************************************************************
  * Compatibility with older versions of FreeBSD
  */
-#if __FreeBSD_version < 440001
+#if 0
 #warning testing old-FreeBSD compat
 typedef struct proc	d_thread_t;
 #endif
diff -ur src.old/sys/dev/fxp/if_fxpvar.h src/sys/dev/fxp/if_fxpvar.h
--- src.old/sys/dev/fxp/if_fxpvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/fxp/if_fxpvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -88,7 +88,7 @@
  */
 #define TUNABLE_BUNDLE_MAX 6
 
-#if __FreeBSD_version < 500000
+#if 0
 #define	FXP_LOCK(_sc)
 #define	FXP_UNLOCK(_sc)
 #define mtx_init(a, b, c)
diff -ur src.old/sys/dev/gx/if_gxvar.h src/sys/dev/gx/if_gxvar.h
--- src.old/sys/dev/gx/if_gxvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/gx/if_gxvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -29,7 +29,7 @@
  *	$FreeBSD: src/sys/dev/gx/if_gxvar.h,v 1.1.2.1 2001/11/15 03:09:23 jlemon Exp $
  */
 
-#if __FreeBSD_version < 500000
+#if 1
 #define GX_LOCK(gx)		
 #define GX_UNLOCK(gx)		
 #define mtx_init(a, b, c)
diff -ur src.old/sys/dev/mlx/mlx_compat.h src/sys/dev/mlx/mlx_compat.h
--- src.old/sys/dev/mlx/mlx_compat.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/mlx/mlx_compat.h	2004-09-16 15:11:07.000000000 +0200
@@ -30,7 +30,7 @@
  * Portability and compatibility interfaces.
  */
 
-#if __FreeBSD_version < 500003
+#if 1
 # include <machine/clock.h>
 # define INTR_ENTROPY			0
 
diff -ur src.old/sys/dev/mly/mlyvar.h src/sys/dev/mly/mlyvar.h
--- src.old/sys/dev/mly/mlyvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/mly/mlyvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -52,11 +52,11 @@
  ********************************************************************************
  ********************************************************************************/
 
-#if __FreeBSD_version >= 500005
+#if 0
 # include <sys/taskqueue.h>
 #endif
 
-#if __FreeBSD_version <= 500014
+#if 1
 # include <machine/clock.h>
 # undef offsetof
 # define offsetof(type, field) ((size_t)(&((type *)0)->field))
@@ -224,7 +224,7 @@
     struct cam_sim		*mly_cam_sim[MLY_MAX_CHANNELS];
     int				mly_cam_lowbus;
 
-#if __FreeBSD_version >= 500005
+#if 0
     /* command-completion task */
     struct task		mly_task_complete;	/* deferred-completion task */
 #endif
diff -ur src.old/sys/dev/sound/pcm/sound.h src/sys/dev/sound/pcm/sound.h
--- src.old/sys/dev/sound/pcm/sound.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/sound/pcm/sound.h	2004-09-16 15:11:07.000000000 +0200
@@ -52,7 +52,7 @@
 #include <sys/errno.h>
 #include <sys/malloc.h>
 #include <sys/bus.h>
-#if __FreeBSD_version > 500000
+#if 0
 #include <sys/bio.h>
 #endif
 #include <sys/buf.h>
@@ -73,7 +73,7 @@
 #undef	USING_MUTEX
 #undef	USING_DEVFS
 
-#if __FreeBSD_version > 500000
+#if 0
 #define USING_MUTEX
 #define USING_DEVFS
 #endif
diff -ur src.old/sys/dev/twe/twe_compat.h src/sys/dev/twe/twe_compat.h
--- src.old/sys/dev/twe/twe_compat.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/twe/twe_compat.h	2004-09-16 15:11:07.000000000 +0200
@@ -130,7 +130,7 @@
 #define twe_printf(sc, fmt, args...)	device_printf(sc->twe_dev, fmt , ##args)
 #define twed_printf(twed, fmt, args...)	device_printf(twed->twed_dev, fmt , ##args)
 
-#if __FreeBSD_version < 500003
+#if 1
 # include <machine/clock.h>
 # define INTR_ENTROPY			0
 
diff -ur src.old/sys/dev/wi/if_wivar.h src/sys/dev/wi/if_wivar.h
--- src.old/sys/dev/wi/if_wivar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/dev/wi/if_wivar.h	2004-09-16 15:11:07.000000000 +0200
@@ -160,7 +160,7 @@
 	u_int32_t		wi_icv;
 	int			wi_icv_flag;
 	struct callout_handle	wi_stat_ch;
-#if __FreeBSD_version >= 500000
+#if 0
 	struct mtx		wi_mtx;
 #endif
 	int			wi_nic_type;
@@ -190,7 +190,7 @@
 	u_int8_t	firm_type;
 };
 
-#if __FreeBSD_version < 500000
+#if 1
 /*
  * Various compat hacks/kludges
  */
diff -ur src.old/sys/i386/machine/ansi.h src/sys/i386/machine/ansi.h
--- src.old/sys/i386/machine/ansi.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/ansi.h	2004-09-16 15:11:07.000000000 +0200
@@ -34,8 +34,10 @@
  * $FreeBSD: src/sys/i386/include/ansi.h,v 1.18.2.4 2001/06/03 17:15:54 obrien Exp $
  */
 
-#ifndef _MACHINE_ANSI_H_
-#define	_MACHINE_ANSI_H_
+/* Defining _MACHINE_ANSI_H_ would mess up gcc's stddef.h.  Fortunately,
+   this won't be an issue for kfreebsd5-headers (rmh) */
+#ifndef _MACHINE_ANSI_H_SAFE
+#define	_MACHINE_ANSI_H_SAFE
 
 /*
  * Types which are fundamental to the implementation and must be declared
@@ -104,4 +106,4 @@
  */
 #define __offsetof(type, field) ((size_t)(&((type *)0)->field))
 
-#endif /* !_MACHINE_ANSI_H_ */
+#endif /* !_MACHINE_ANSI_H_SAFE */
diff -ur src.old/sys/i386/machine/atomic.h src/sys/i386/machine/atomic.h
--- src.old/sys/i386/machine/atomic.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/atomic.h	2004-09-16 15:11:07.000000000 +0200
@@ -28,6 +28,8 @@
 #ifndef _MACHINE_ATOMIC_H_
 #define _MACHINE_ATOMIC_H_
 
+#include <sys/types.h>		/* u_##TYPE */
+
 /*
  * Various simple arithmetic on memory which is atomic in the presence
  * of interrupts and multiple processors.
diff -ur src.old/sys/i386/machine/bootinfo.h src/sys/i386/machine/bootinfo.h
--- src.old/sys/i386/machine/bootinfo.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/bootinfo.h	2004-09-16 15:11:07.000000000 +0200
@@ -35,6 +35,8 @@
 #ifndef	_MACHINE_BOOTINFO_H_
 #define	_MACHINE_BOOTINFO_H_
 
+#include <sys/types.h>
+
 /* Only change the version number if you break compatibility. */
 #define	BOOTINFO_VERSION	1
 
diff -ur src.old/sys/i386/machine/bus_at386.h src/sys/i386/machine/bus_at386.h
--- src.old/sys/i386/machine/bus_at386.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/bus_at386.h	2004-09-16 15:11:07.000000000 +0200
@@ -73,6 +73,8 @@
 #define _I386_BUS_AT386_H_
 
 #include <machine/cpufunc.h>
+#include <stddef.h>		/* size_t */
+
 
 /*
  * To remain compatible with NetBSD's interface, default to both memio and
diff -ur src.old/sys/i386/machine/bus_dma.h src/sys/i386/machine/bus_dma.h
--- src.old/sys/i386/machine/bus_dma.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/bus_dma.h	2004-09-16 15:11:07.000000000 +0200
@@ -72,6 +72,8 @@
 #ifndef _I386_BUS_DMA_H_
 #define _I386_BUS_DMA_H_
 
+#include <machine/bus_at386.h>
+
 /*
  * Flags used in various bus DMA methods.
  */
diff -ur src.old/sys/i386/machine/cpufunc.h src/sys/i386/machine/cpufunc.h
--- src.old/sys/i386/machine/cpufunc.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/cpufunc.h	2004-09-16 15:11:07.000000000 +0200
@@ -41,6 +41,8 @@
 #define	_MACHINE_CPUFUNC_H_
 
 #include <sys/cdefs.h>
+#include <sys/io.h>
+#include <sys/types.h>
 
 __BEGIN_DECLS
 #define readb(va)	(*(volatile u_int8_t *) (va))
@@ -140,36 +142,6 @@
 
 #else /* __GNUC >= 2 */
 
-/*
- * The following complications are to get around gcc not having a
- * constraint letter for the range 0..255.  We still put "d" in the
- * constraint because "i" isn't a valid constraint when the port
- * isn't constant.  This only matters for -O0 because otherwise
- * the non-working version gets optimized away.
- * 
- * Use an expression-statement instead of a conditional expression
- * because gcc-2.6.0 would promote the operands of the conditional
- * and produce poor code for "if ((inb(var) & const1) == const2)".
- *
- * The unnecessary test `(port) < 0x10000' is to generate a warning if
- * the `port' has type u_short or smaller.  Such types are pessimal.
- * This actually only works for signed types.  The range check is
- * careful to avoid generating warnings.
- */
-#define	inb(port) __extension__ ({					\
-	u_char	_data;							\
-	if (__builtin_constant_p(port) && ((port) & 0xffff) < 0x100	\
-	    && (port) < 0x10000)					\
-		_data = inbc(port);					\
-	else								\
-		_data = inbv(port);					\
-	_data; })
-
-#define	outb(port, data) (						\
-	__builtin_constant_p(port) && ((port) & 0xffff) < 0x100		\
-	&& (port) < 0x10000						\
-	? outbc(port, data) : outbv(port, data))
-
 static __inline u_char
 inbc(u_int port)
 {
@@ -187,6 +159,7 @@
 
 #endif /* __GNUC <= 2 */
 
+
 static __inline u_char
 inbv(u_int port)
 {
@@ -200,42 +173,6 @@
 	return (data);
 }
 
-static __inline u_int
-inl(u_int port)
-{
-	u_int	data;
-
-	__asm __volatile("inl %%dx,%0" : "=a" (data) : "d" (port));
-	return (data);
-}
-
-static __inline void
-insb(u_int port, void *addr, size_t cnt)
-{
-	__asm __volatile("cld; rep; insb"
-			 : "=D" (addr), "=c" (cnt)
-			 :  "0" (addr),  "1" (cnt), "d" (port)
-			 : "memory");
-}
-
-static __inline void
-insw(u_int port, void *addr, size_t cnt)
-{
-	__asm __volatile("cld; rep; insw"
-			 : "=D" (addr), "=c" (cnt)
-			 :  "0" (addr),  "1" (cnt), "d" (port)
-			 : "memory");
-}
-
-static __inline void
-insl(u_int port, void *addr, size_t cnt)
-{
-	__asm __volatile("cld; rep; insl"
-			 : "=D" (addr), "=c" (cnt)
-			 :  "0" (addr),  "1" (cnt), "d" (port)
-			 : "memory");
-}
-
 static __inline void
 invd(void)
 {
@@ -298,14 +235,6 @@
 
 #endif /* SMP && _KERNEL */
 
-static __inline u_short
-inw(u_int port)
-{
-	u_short	data;
-
-	__asm __volatile("inw %%dx,%0" : "=a" (data) : "d" (port));
-	return (data);
-}
 
 static __inline u_int
 loadandclear(volatile u_int *addr)
@@ -331,47 +260,6 @@
 	__asm __volatile("outb %0,%%dx" : : "a" (al), "d" (port));
 }
 
-static __inline void
-outl(u_int port, u_int data)
-{
-	/*
-	 * outl() and outw() aren't used much so we haven't looked at
-	 * possible micro-optimizations such as the unnecessary
-	 * assignment for them.
-	 */
-	__asm __volatile("outl %0,%%dx" : : "a" (data), "d" (port));
-}
-
-static __inline void
-outsb(u_int port, const void *addr, size_t cnt)
-{
-	__asm __volatile("cld; rep; outsb"
-			 : "=S" (addr), "=c" (cnt)
-			 :  "0" (addr),  "1" (cnt), "d" (port));
-}
-
-static __inline void
-outsw(u_int port, const void *addr, size_t cnt)
-{
-	__asm __volatile("cld; rep; outsw"
-			 : "=S" (addr), "=c" (cnt)
-			 :  "0" (addr),  "1" (cnt), "d" (port));
-}
-
-static __inline void
-outsl(u_int port, const void *addr, size_t cnt)
-{
-	__asm __volatile("cld; rep; outsl"
-			 : "=S" (addr), "=c" (cnt)
-			 :  "0" (addr),  "1" (cnt), "d" (port));
-}
-
-static __inline void
-outw(u_int port, u_short data)
-{
-	__asm __volatile("outw %0,%%dx" : : "a" (data), "d" (port));
-}
-
 static __inline u_int
 rcr2(void)
 {
diff -ur src.old/sys/i386/machine/db_machdep.h src/sys/i386/machine/db_machdep.h
--- src.old/sys/i386/machine/db_machdep.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/db_machdep.h	2004-09-16 15:11:07.000000000 +0200
@@ -32,6 +32,7 @@
 #include <machine/frame.h>
 #include <machine/psl.h>
 #include <machine/trap.h>
+#include <machine/types.h>		/* vm_offset_t */
 
 #define i386_saved_state trapframe
 
diff -ur src.old/sys/i386/machine/dvcfg.h src/sys/i386/machine/dvcfg.h
--- src.old/sys/i386/machine/dvcfg.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/dvcfg.h	2004-09-16 15:11:07.000000000 +0200
@@ -35,6 +35,9 @@
 #ifndef _I386_DVCFG_H_
 #define _I386_DVCFG_H_
 
+#include <sys/cdefs.h>		/* __P */
+#include <sys/types.h>
+
 typedef void *dvcfg_hw_t;
 
 struct dvcfg_hwsel {
diff -ur src.old/sys/i386/machine/frame.h src/sys/i386/machine/frame.h
--- src.old/sys/i386/machine/frame.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/frame.h	2004-09-16 15:11:07.000000000 +0200
@@ -40,6 +40,8 @@
 #ifndef _MACHINE_FRAME_H_
 #define _MACHINE_FRAME_H_ 1
 
+#include <sys/cdefs.h>	  	/* __P */
+
 /*
  * System stack frames.
  */
diff -ur src.old/sys/i386/machine/globaldata.h src/sys/i386/machine/globaldata.h
--- src.old/sys/i386/machine/globaldata.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/globaldata.h	2004-09-16 15:11:07.000000000 +0200
@@ -26,6 +26,10 @@
  * $FreeBSD: src/sys/i386/include/globaldata.h,v 1.11.2.1 2000/05/16 06:58:10 dillon Exp $
  */
 
+#include <sys/types.h>
+#include <machine/tss.h>		/* i386tss */
+#include <machine/segments.h>		/* segment_descriptor */
+
 /*
  * This structure maps out the global data that needs to be kept on a
  * per-cpu basis.  genassym uses this to generate offsets for the assembler
diff -ur src.old/sys/i386/machine/i4b_debug.h src/sys/i386/machine/i4b_debug.h
--- src.old/sys/i386/machine/i4b_debug.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/i4b_debug.h	2004-09-16 15:11:07.000000000 +0200
@@ -33,6 +33,8 @@
  *
  *---------------------------------------------------------------------------*/
 
+#include <sys/types.h>
+
 #if !defined DO_I4B_DEBUG
 # define DO_I4B_DEBUG 1		/* default = include debugging code */
 #elif DO_I4B_DEBUG != 0
diff -ur src.old/sys/i386/machine/i4b_ioctl.h src/sys/i386/machine/i4b_ioctl.h
--- src.old/sys/i386/machine/i4b_ioctl.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/i4b_ioctl.h	2004-09-16 15:11:07.000000000 +0200
@@ -36,7 +36,9 @@
 #ifndef _I4B_IOCTL_H_
 #define _I4B_IOCTL_H_
 
-#if defined(__FreeBSD_kernel__) && __FreeBSD__ >= 3
+#include <sys/types.h>
+
+#if defined(__FreeBSD_kernel__)
 #ifndef _MACHINE_TYPES_H_
 #include <machine/types.h>
 #endif /* _MACHINE_TYPES_H_ */
diff -ur src.old/sys/i386/machine/i4b_isppp.h src/sys/i386/machine/i4b_isppp.h
--- src.old/sys/i386/machine/i4b_isppp.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/i4b_isppp.h	2004-09-16 15:11:07.000000000 +0200
@@ -26,6 +26,9 @@
 #ifndef _I4B_ISPPP_H_
 #define _I4B_ISPPP_H_
 
+#include <net/if_var.h>		/* ifnet, ifqueue */
+#include <sys/callout.h>	/* callout_handle */
+
 #define SPPP_VJ			/* use VJ compression */
 
 
@@ -123,7 +126,7 @@
 	u_char  confid[IDX_COUNT];	/* id of last configuration request */
 	int	rst_counter[IDX_COUNT];	/* restart counter */
 	int	fail_counter[IDX_COUNT]; /* negotiation failure counter */
-#if defined(__FreeBSD_kernel__) && __FreeBSD__ >= 3
+#if defined(__FreeBSD_kernel__)
 	struct callout_handle ch[IDX_COUNT]; /* per-proto and if callouts */
 	struct callout_handle pap_my_to_ch; /* PAP needs one more... */
 #endif
@@ -211,7 +214,7 @@
 void sppp_detach (struct ifnet *ifp);
 void sppp_input (struct ifnet *ifp, struct mbuf *m);
 
-#if defined(__FreeBSD_version) && __FreeBSD_version >= 300003
+#if defined(__FreeBSD_kernel__)
 int sppp_ioctl (struct ifnet *ifp, u_long cmd, void *data);
 #else
 #ifdef __FreeBSD_kernel__
@@ -232,7 +235,7 @@
 void isppp_detach (struct ifnet *ifp);
 void isppp_input (struct ifnet *ifp, struct mbuf *m);
 
-#if defined(__FreeBSD_version) && __FreeBSD_version >= 300003
+#if 1
 int isppp_ioctl (struct ifnet *ifp, u_long cmd, void *data);
 #else
 #ifdef __FreeBSD_kernel__
diff -ur src.old/sys/i386/machine/i4b_rbch_ioctl.h src/sys/i386/machine/i4b_rbch_ioctl.h
--- src.old/sys/i386/machine/i4b_rbch_ioctl.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/i4b_rbch_ioctl.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,8 @@
 #ifndef _I4B_RBCH_IOCTL_H_
 #define _I4B_RBCH_IOCTL_H_
 
+#include <machine/i4b_ioctl.h>		/* TELNO_MAX */
+
 /*---------------------------------------------------------------------------*
  *	instruct the rbch device to dial the given number
  *---------------------------------------------------------------------------*/
diff -ur src.old/sys/i386/machine/i4b_trace.h src/sys/i386/machine/i4b_trace.h
--- src.old/sys/i386/machine/i4b_trace.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/i4b_trace.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,8 @@
 #ifndef _I4B_TRACE_H_
 #define _I4B_TRACE_H_
 
+#include "/usr/src/kfreebsd4-headers/sys/time.h"
+
 /*---------------------------------------------------------------------------*
  *	structure of the header at the beginning of every trace mbuf
  *---------------------------------------------------------------------------*/
diff -ur src.old/sys/i386/machine/iic.h src/sys/i386/machine/iic.h
--- src.old/sys/i386/machine/iic.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/iic.h	2004-09-16 15:11:07.000000000 +0200
@@ -30,6 +30,7 @@
 #define __IIC_H
 
 #include <sys/ioccom.h>
+#include <sys/types.h>
 
 struct iiccmd {
 	u_char slave;
diff -ur src.old/sys/i386/machine/in_cksum.h src/sys/i386/machine/in_cksum.h
--- src.old/sys/i386/machine/in_cksum.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/in_cksum.h	2004-09-16 15:11:07.000000000 +0200
@@ -40,6 +40,8 @@
 #define	_MACHINE_IN_CKSUM_H_	1
 
 #include <sys/cdefs.h>
+#include <sys/types.h>
+#include <netinet/ip.h>		/* ip */
 
 /*
  * It it useful to have an Internet checksum routine which is inlineable
diff -ur src.old/sys/i386/machine/ioctl_bt848.h src/sys/i386/machine/ioctl_bt848.h
--- src.old/sys/i386/machine/ioctl_bt848.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/ioctl_bt848.h	2004-09-16 15:11:07.000000000 +0200
@@ -7,6 +7,8 @@
 #ifndef  _MACHINE_IOCTL_BT848_H_
 #define  _MACHINE_IOCTL_BT848_H_
 
+#include <sys/types.h>
+
 /*
  * frequency sets
  */
diff -ur src.old/sys/i386/machine/ipl.h src/sys/i386/machine/ipl.h
--- src.old/sys/i386/machine/ipl.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/ipl.h	2004-09-16 15:11:07.000000000 +0200
@@ -36,12 +36,6 @@
 #ifndef _MACHINE_IPL_H_
 #define	_MACHINE_IPL_H_
 
-#ifdef APIC_IO
-#include <i386/isa/apic_ipl.h>
-#else
-#include <i386/isa/icu_ipl.h>
-#endif
-
 /*
  * Software interrupt bit numbers in priority order.  The priority only
  * determines which swi will be dispatched next; a higher priority swi
diff -ur src.old/sys/i386/machine/md_var.h src/sys/i386/machine/md_var.h
--- src.old/sys/i386/machine/md_var.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/md_var.h	2004-09-16 15:11:07.000000000 +0200
@@ -32,6 +32,10 @@
 #ifndef _MACHINE_MD_VAR_H_
 #define	_MACHINE_MD_VAR_H_
 
+#include <sys/types.h>
+#include <sys/cdefs.h>				/* __P */
+#include <machine/types.h>			/* vm_offset_t */
+
 /*
  * Miscellaneous machine-dependent declarations.
  */
diff -ur src.old/sys/i386/machine/mpapic.h src/sys/i386/machine/mpapic.h
--- src.old/sys/i386/machine/mpapic.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/mpapic.h	2004-09-16 15:11:07.000000000 +0200
@@ -30,7 +30,10 @@
 
 #include <machine/apic.h>
 
-#include <i386/isa/icu.h>
+/* Sounds like the number of CPUs, so I define it to 1 */
+#ifndef mp_ncpus
+#define mp_ncpus 1
+#endif
 
 /*
  * Size of APIC ID list.
diff -ur src.old/sys/i386/machine/pc/bios.h src/sys/i386/machine/pc/bios.h
--- src.old/sys/i386/machine/pc/bios.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/pc/bios.h	2004-09-16 15:11:07.000000000 +0200
@@ -27,6 +27,9 @@
  * $FreeBSD: src/sys/i386/include/pc/bios.h,v 1.7.2.3 2001/10/21 03:16:56 yokota Exp $
  */
 
+#include <sys/types.h>
+#include <machine/types.h>		/* vm_offset_t */
+
 /* 
  * Signature structure for the BIOS32 Service Directory header 
  */
diff -ur src.old/sys/i386/machine/pc/vesa.h src/sys/i386/machine/pc/vesa.h
--- src.old/sys/i386/machine/pc/vesa.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/pc/vesa.h	2004-09-16 15:11:07.000000000 +0200
@@ -29,6 +29,8 @@
 #ifndef _MACHINE_PC_VESA_H
 #define _MACHINE_PC_VESA_H
 
+#include <sys/types.h>
+
 struct vesa_info
 {
     /* mandatory fields */
diff -ur src.old/sys/i386/machine/pci_cfgreg.h src/sys/i386/machine/pci_cfgreg.h
--- src.old/sys/i386/machine/pci_cfgreg.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/pci_cfgreg.h	2004-09-16 15:11:07.000000000 +0200
@@ -27,6 +27,8 @@
  *
  */
 
+#include <sys/types.h>
+
 #define CONF1_ADDR_PORT    0x0cf8
 #define CONF1_DATA_PORT    0x0cfc
 
diff -ur src.old/sys/i386/machine/physio_proc.h src/sys/i386/machine/physio_proc.h
--- src.old/sys/i386/machine/physio_proc.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/physio_proc.h	2004-09-16 15:11:07.000000000 +0200
@@ -33,7 +33,8 @@
 #ifndef _I386_PHYSIO_PROC_H_
 #define _I386_PHYSIO_PROC_H_
 #include <sys/buf.h>
-#include <sys/queue.h>
+#include "/usr/src/kfreebsd4-headers/sys/queue.h"
+#include <stdio.h>		/* NULL */
 
 struct physio_proc;
 TAILQ_HEAD(physio_proc_head, physio_proc);
@@ -61,7 +62,7 @@
 
 	s = splstatclock();
 	TAILQ_REMOVE(&physio_proc_freet, pp, pp_chain);
-#if !defined(__FreeBSD_kernel__) || __FreeBSD_version < 400001
+#if !defined(__FreeBSD_kernel__)
 	pp->pp_proc = bp->b_proc;
 #endif
 	TAILQ_INSERT_TAIL(&physio_proc_busyt, pp, pp_chain);
diff -ur src.old/sys/i386/machine/profile.h src/sys/i386/machine/profile.h
--- src.old/sys/i386/machine/profile.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/profile.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,8 @@
 #ifndef _MACHINE_PROFILE_H_
 #define	_MACHINE_PROFILE_H_
 
+#include <sys/types.h>
+
 #ifdef _KERNEL
 
 /*
diff -ur src.old/sys/i386/machine/sigframe.h src/sys/i386/machine/sigframe.h
--- src.old/sys/i386/machine/sigframe.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/sigframe.h	2004-09-16 15:11:07.000000000 +0200
@@ -31,6 +31,9 @@
 #ifndef _MACHINE_SIGFRAME_H_
 #define	_MACHINE_SIGFRAME_H_
 
+#include <sys/types.h>			/* register_t */
+#include <signal.h>			/* __sighandler_t */
+
 /*
  * Signal frames, arguments passed to application signal handlers.
  */
@@ -52,43 +55,6 @@
 	register_t	sf_scp;
 
 	register_t	sf_addr;
-
-	/*
-	 * The following arguments are not constrained by the
-	 * function call protocol.
-	 * Applications are not supposed to access these members,
-	 * except using the pointers we provide in the first three
-	 * arguments.
-	 */
-
-	union {
-		__osiginfohandler_t	*sf_action;
-		__sighandler_t		*sf_handler;
-	} sf_ahu;
-
-	/* In the SA_SIGINFO case, sf_arg2 points here. */
-	osiginfo_t	sf_siginfo;
-};
-
-struct sigframe {
-	/*
-	 * The first four members may be used by applications.
-	 *
-	 * NOTE: The 4th argument is undocumented, ill commented
-	 * on and seems to be somewhat BSD "standard".  Handlers
-	 * installed with sigvec may be using it.
-	 */
-	register_t	sf_signum;
-	register_t	sf_siginfo;	/* code or pointer to sf_si */
-	register_t	sf_ucontext;	/* points to sf_uc */
-	register_t	sf_addr;	/* undocumented 4th arg */
-
-	union {
-		__siginfohandler_t	*sf_action;
-		__sighandler_t		*sf_handler;
-	} sf_ahu;
-	ucontext_t	sf_uc;		/* = *sf_ucontext */
-	siginfo_t	sf_si;		/* = *sf_siginfo (SA_SIGINFO case) */
 };
 
 #endif /* !_MACHINE_SIGFRAME_H_ */
diff -ur src.old/sys/i386/machine/signal.h src/sys/i386/machine/signal.h
--- src.old/sys/i386/machine/signal.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/signal.h	2004-09-16 15:11:07.000000000 +0200
@@ -80,46 +80,7 @@
 	int	sc_err;
 };
 
-/*
- * The sequence of the fields/registers in struct sigcontext should match
- * those in mcontext_t.
- */
-struct	sigcontext {
-	sigset_t sc_mask;		/* signal mask to restore */
-	int	sc_onstack;		/* sigstack state to restore */
-	int	sc_gs;			/* machine state (struct trapframe): */
-	int	sc_fs;
-	int	sc_es;
-	int	sc_ds;
-	int	sc_edi;
-	int	sc_esi;
-	int	sc_ebp;
-	int	sc_isp;
-	int	sc_ebx;
-	int	sc_edx;
-	int	sc_ecx;
-	int	sc_eax;
-	int	sc_trapno;
-	int	sc_err;
-	int	sc_eip;
-	int	sc_cs;
-	int	sc_efl;
-	int	sc_esp;
-	int	sc_ss;
-	/*
-	 * XXX FPU state is 27 * 4 bytes h/w, 1 * 4 bytes s/w (probably not
-	 * needed here), or that + 16 * 4 bytes for emulators (probably all
-	 * needed here).  The "spare" bytes are mostly not spare.
-	 */
-	int	sc_fpregs[28];		/* machine state (FPU): */
-	int	sc_spare[17];
-};
-
-#define	sc_sp		sc_esp
-#define	sc_fp		sc_ebp
-#define	sc_pc		sc_eip
-#define	sc_ps		sc_efl
-#define	sc_eflags	sc_efl
+#include <signal.h>			/* sigcontext */
 
 #endif /* !_ANSI_SOURCE && !_POSIX_SOURCE */
 
diff -ur src.old/sys/i386/machine/smb.h src/sys/i386/machine/smb.h
--- src.old/sys/i386/machine/smb.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/smb.h	2004-09-16 15:11:07.000000000 +0200
@@ -30,6 +30,7 @@
 #define __SMB_H
 
 #include <sys/ioccom.h>
+#include <sys/types.h>
 
 struct smbcmd {
 	char cmd;
diff -ur src.old/sys/i386/machine/specialreg.h src/sys/i386/machine/specialreg.h
--- src.old/sys/i386/machine/specialreg.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/specialreg.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,8 @@
 #ifndef _MACHINE_SPECIALREG_H_
 #define	_MACHINE_SPECIALREG_H_
 
+#include <sys/types.h>
+
 /*
  * Bits in 386 special registers:
  */
diff -ur src.old/sys/i386/machine/uc_device.h src/sys/i386/machine/uc_device.h
--- src.old/sys/i386/machine/uc_device.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/uc_device.h	2004-09-16 15:11:07.000000000 +0200
@@ -52,6 +52,8 @@
 #ifndef _I386_MACHINE_UC_DEVICE_H
 #define _I386_MACHINE_UC_DEVICE_H
 
+#include <sys/types.h>
+
 /*
  * Per device structure.  This just happens to resemble the old isa_device
  * but that is by accident.  It is NOT the same.
diff -ur src.old/sys/i386/machine/vm86.h src/sys/i386/machine/vm86.h
--- src.old/sys/i386/machine/vm86.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/i386/machine/vm86.h	2004-09-16 15:11:07.000000000 +0200
@@ -32,6 +32,9 @@
 #ifndef _MACHINE_VM86_H_
 #define _MACHINE_VM86_H_ 1
 
+#include <sys/types.h>
+#include <machine/types.h>		/* vm_offset_t */
+
 /* standard register representation */
 typedef union {
 	u_int	r_ex;
diff -ur src.old/sys/isofs/cd9660/cd9660_mount.h src/sys/isofs/cd9660/cd9660_mount.h
--- src.old/sys/isofs/cd9660/cd9660_mount.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/isofs/cd9660/cd9660_mount.h	2004-09-16 15:11:07.000000000 +0200
@@ -39,6 +39,8 @@
  * $FreeBSD: src/sys/isofs/cd9660/cd9660_mount.h,v 1.3.2.2 2001/03/14 12:03:50 bp Exp $
  */
 
+#include <sys/mount.h>			/* export_args */
+
 /*
  * Arguments to mount ISO 9660 filesystems.
  */
diff -ur src.old/sys/isofs/cd9660/cd9660_node.h src/sys/isofs/cd9660/cd9660_node.h
--- src.old/sys/isofs/cd9660/cd9660_node.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/isofs/cd9660/cd9660_node.h	2004-09-16 15:11:07.000000000 +0200
@@ -39,6 +39,10 @@
  * $FreeBSD: src/sys/isofs/cd9660/cd9660_node.h,v 1.20 1999/12/29 04:54:37 peter Exp $
  */
 
+#include <time.h>			/* __kernel_timespec */
+#include <sys/types.h>
+#include <sys/lock.h>			/* lock */
+
 /*
  * Theoretically, directories can be more than 2Gb in length,
  * however, in practice this seems unlikely. So, we define
@@ -56,7 +60,7 @@
 	uid_t		iso_uid;	/* owner user id */
 	gid_t		iso_gid;	/* owner group id */
 	short		iso_links;	/* links of file */
-	udev_t		iso_rdev;	/* Major/Minor number for special */
+	dev_t		iso_rdev;	/* Major/Minor number for special */
 } ISO_RRIP_INODE;
 
 
diff -ur src.old/sys/isofs/cd9660/cd9660_rrip.h src/sys/isofs/cd9660/cd9660_rrip.h
--- src.old/sys/isofs/cd9660/cd9660_rrip.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/isofs/cd9660/cd9660_rrip.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,7 +38,10 @@
  *	@(#)cd9660_rrip.h	8.2 (Berkeley) 12/5/94
  * $FreeBSD: src/sys/isofs/cd9660/cd9660_rrip.h,v 1.6 1999/08/28 00:46:06 peter Exp $
  */
-  
+
+#include <sys/types.h>  
+#include <isofs/cd9660/iso.h>			/* ISODCL */
+
 typedef struct {
 	char   type			[ISODCL (  0,    1)];
 	u_char length			[ISODCL (  2,    2)]; /* 711 */
diff -ur src.old/sys/isofs/cd9660/iso.h src/sys/isofs/cd9660/iso.h
--- src.old/sys/isofs/cd9660/iso.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/isofs/cd9660/iso.h	2004-09-16 15:11:07.000000000 +0200
@@ -39,6 +39,8 @@
  * $FreeBSD: src/sys/isofs/cd9660/iso.h,v 1.19.2.1 2000/07/08 14:35:56 bp Exp $
  */
 
+#include <sys/types.h>
+
 #define ISODCL(from, to) (to - from + 1)
 
 struct iso_volume_descriptor {
diff -ur src.old/sys/isofs/cd9660/iso_rrip.h src/sys/isofs/cd9660/iso_rrip.h
--- src.old/sys/isofs/cd9660/iso_rrip.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/isofs/cd9660/iso_rrip.h	2004-09-16 15:11:07.000000000 +0200
@@ -39,6 +39,7 @@
  * $FreeBSD: src/sys/isofs/cd9660/iso_rrip.h,v 1.6 1999/08/28 00:46:07 peter Exp $
  */
 
+#include <sys/types.h>				/* daddr_t */
 
 /*
  *	Analyze function flag (similar to RR field bits)
diff -ur src.old/sys/msdosfs/bootsect.h src/sys/msdosfs/bootsect.h
--- src.old/sys/msdosfs/bootsect.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/msdosfs/bootsect.h	2004-09-16 15:11:07.000000000 +0200
@@ -17,6 +17,8 @@
  * October 1992
  */
 
+#include <sys/types.h>
+
 /*
  * Format of a boot sector.  This is the first sector on a DOS floppy disk
  * or the fist sector of a partition on a hard disk.  But, it is not the
diff -ur src.old/sys/msdosfs/bpb.h src/sys/msdosfs/bpb.h
--- src.old/sys/msdosfs/bpb.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/msdosfs/bpb.h	2004-09-16 15:11:07.000000000 +0200
@@ -17,6 +17,8 @@
  * October 1992
  */
 
+#include <sys/types.h>
+
 /*
  * BIOS Parameter Block (BPB) for DOS 3.3
  */
diff -ur src.old/sys/msdosfs/denode.h src/sys/msdosfs/denode.h
--- src.old/sys/msdosfs/denode.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/msdosfs/denode.h	2004-09-16 15:11:07.000000000 +0200
@@ -91,6 +91,9 @@
  *   things.
  */
 
+#include <sys/types.h>
+#include <sys/lock.h>			/* lock */
+
 /*
  * Internal pseudo-offset for (nonexistent) directory entry for the root
  * dir in the root dir
diff -ur src.old/sys/msdosfs/direntry.h src/sys/msdosfs/direntry.h
--- src.old/sys/msdosfs/direntry.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/msdosfs/direntry.h	2004-09-16 15:11:07.000000000 +0200
@@ -48,6 +48,8 @@
  * October 1992
  */
 
+#include <sys/types.h>
+
 /*
  * Structure of a dos directory entry.
  */
diff -ur src.old/sys/msdosfs/msdosfsmount.h src/sys/msdosfs/msdosfsmount.h
--- src.old/sys/msdosfs/msdosfsmount.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/msdosfs/msdosfsmount.h	2004-09-16 15:11:07.000000000 +0200
@@ -51,6 +51,8 @@
 #ifndef _MSDOSFS_MSDOSFSMOUNT_H_
 #define	_MSDOSFS_MSDOSFSMOUNT_H_
 
+#include <sys/mount.h>				/* export_args */
+
 #ifdef _KERNEL
 
 #ifdef MALLOC_DECLARE
diff -ur src.old/sys/net/bpf.h src/sys/net/bpf.h
--- src.old/sys/net/bpf.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/bpf.h	2004-09-16 15:11:07.000000000 +0200
@@ -44,6 +44,8 @@
 #ifndef _NET_BPF_H_
 #define _NET_BPF_H_
 
+#include <sys/types.h>
+
 /* BSD style release date */
 #define	BPF_RELEASE 199606
 
diff -ur src.old/sys/net/bpfdesc.h src/sys/net/bpfdesc.h
--- src.old/sys/net/bpfdesc.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/bpfdesc.h	2004-09-16 15:11:07.000000000 +0200
@@ -43,6 +43,8 @@
 #ifndef _NET_BPFDESC_H_
 #define _NET_BPFDESC_H_
 
+#include <sys/types.h>
+
 #include <sys/callout.h>
 #include <sys/select.h>
 
diff -ur src.old/sys/net/bridge.h src/sys/net/bridge.h
--- src.old/sys/net/bridge.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/bridge.h	2004-09-16 15:11:07.000000000 +0200
@@ -27,6 +27,8 @@
  * $FreeBSD: src/sys/net/bridge.h,v 1.4.2.5 2002/02/15 05:13:58 luigi Exp $
  */
 
+#include <sys/types.h>
+
 extern int do_bridge;
 
 /*
diff -ur src.old/sys/net/hostcache.h src/sys/net/hostcache.h
--- src.old/sys/net/hostcache.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/hostcache.h	2004-09-16 15:11:07.000000000 +0200
@@ -32,6 +32,9 @@
 #ifndef _NET_HOSTCACHE_H
 #define	_NET_HOSTCACHE_H	1
 
+#include <time.h>			/* __kernel_timeval */
+#include <sys/types.h>
+
 /*
  * This file defines the interface between network protocols and
  * the cache of host-specific information maintained by the kernel.
diff -ur src.old/sys/net/if_arc.h src/sys/net/if_arc.h
--- src.old/sys/net/if_arc.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_arc.h	2004-09-16 15:11:07.000000000 +0200
@@ -40,6 +40,9 @@
 #ifndef _NET_IF_ARC_H_
 #define _NET_IF_ARC_H_
 
+#include <sys/types.h>
+#include <net/if_var.h>			/* ifnet */
+
 /*
  * Arcnet address - 1 octets
  * don't know who uses this.
diff -ur src.old/sys/net/if_arp.h src/sys/net/if_arp.h
--- src.old/sys/net/if_arp.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_arp.h	2004-09-16 15:11:07.000000000 +0200
@@ -41,6 +41,7 @@
 #include <sys/types.h>
 /* Define 'struct sockaddr'.  */
 #include <sys/socket.h>
+#include <net/if_var.h>				/* ifnet */
 
 /*
  * Address Resolution Protocol.
@@ -113,7 +114,6 @@
 #define	ATF_PUBL	0x08	/* publish entry (respond for other host) */
 #define	ATF_USETRAILERS	0x10	/* has requested trailers */
 
-#ifdef _KERNEL
 /*
  * Structure shared between the ethernet driver modules and
  * the address resolution code.  For example, each ec_softc or il_softc
@@ -129,6 +129,7 @@
 	void	*ac_netgraph;		/* ng_ether(4) netgraph node info */
 };
 
+#ifdef _KERNEL
 extern u_char	etherbroadcastaddr[6];
 extern u_char	arcbroadcastaddr;
 #endif
diff -ur src.old/sys/net/if_atm.h src/sys/net/if_atm.h
--- src.old/sys/net/if_atm.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_atm.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,8 @@
  * net/if_atm.h
  */
 
+#include <sys/types.h>
+
 #if defined(__NetBSD__) || defined(__OpenBSD__) || defined(__bsdi__)
 #define RTALLOC1(A,B)		rtalloc1((A),(B))
 #elif defined(__FreeBSD_kernel__)
diff -ur src.old/sys/net/if_gif.h src/sys/net/if_gif.h
--- src.old/sys/net/if_gif.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_gif.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,10 @@
 #ifndef _NET_IF_GIF_H_
 #define _NET_IF_GIF_H_
 
+#include <net/if_var.h>			/* ifnet */
+#include "/usr/src/kfreebsd4-headers/sys/queue.h"
+#include <sys/types.h>
+#include <net/route.h>			/* route */
 
 #if defined(_KERNEL) && !defined(_LKM)
 #include "opt_inet.h"
diff -ur src.old/sys/net/if_ieee80211.h src/sys/net/if_ieee80211.h
--- src.old/sys/net/if_ieee80211.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_ieee80211.h	2004-09-16 15:11:07.000000000 +0200
@@ -4,6 +4,10 @@
 #ifndef _NET_IF_IEEE80211_H_
 #define _NET_IF_IEEE80211_H_
 
+#include <sys/types.h>
+#include <net/ethernet.h>		/* ETHER_ADDR_LEN */
+#include <net/if.h>			/* IFNAMSIZ */
+
 /*
  * generic definitions for IEEE 802.11 frames
  */
diff -ur src.old/sys/net/if_llc.h src/sys/net/if_llc.h
--- src.old/sys/net/if_llc.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_llc.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,8 @@
 #ifndef _NET_IF_LLC_H_
 #define _NET_IF_LLC_H_
 
+#include <sys/types.h>
+
 /*
  * IEEE 802.2 Link Level Control headers, for use in conjunction with
  * 802.{3,4,5} media access control methods.
diff -ur src.old/sys/net/if_mib.h src/sys/net/if_mib.h
--- src.old/sys/net/if_mib.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_mib.h	2004-09-16 15:11:07.000000000 +0200
@@ -32,6 +32,8 @@
 #ifndef _NET_IF_MIB_H
 #define	_NET_IF_MIB_H	1
 
+#include <net/if.h>                     /* IFNAMSIZ */
+
 struct ifmibdata {
 	char	ifmd_name[IFNAMSIZ]; /* name of interface */
 	int	ifmd_pcount;	/* number of promiscuous listeners */
diff -ur src.old/sys/net/if_pppvar.h src/sys/net/if_pppvar.h
--- src.old/sys/net/if_pppvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_pppvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -42,6 +42,11 @@
  * $FreeBSD: src/sys/net/if_pppvar.h,v 1.15 2000/01/29 16:56:23 peter Exp $
  */
 
+#include <sys/types.h>
+#include <net/if_var.h>				/* ifnet */
+#include <sys/callout.h>			/* callout */
+#include <net/ppp_defs.h>			/* pppstat */
+
 /*
  * Supported network protocols.  These values are used for
  * indexing sc_npmode.
diff -ur src.old/sys/net/if_slvar.h src/sys/net/if_slvar.h
--- src.old/sys/net/if_slvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_slvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -39,6 +39,8 @@
 #define _NET_IF_SLVAR_H_
 
 #include <sys/callout.h>
+#include <net/if_var.h>				/* ifnet */
+#include <net/slcompress.h>			/* slcompress */
 
 /*
  * Definitions for SLIP interface data structures
diff -ur src.old/sys/net/if_sppp.h src/sys/net/if_sppp.h
--- src.old/sys/net/if_sppp.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_sppp.h	2004-09-16 15:11:07.000000000 +0200
@@ -22,6 +22,8 @@
 #ifndef _NET_IF_SPPP_H_
 #define _NET_IF_SPPP_H_ 1
 
+#include <sys/types.h>
+
 #define IDX_LCP 0		/* idx into state table */
 
 struct slcp {
diff -ur src.old/sys/net/if_stf.h src/sys/net/if_stf.h
--- src.old/sys/net/if_stf.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_stf.h	2004-09-16 15:11:07.000000000 +0200
@@ -33,6 +33,8 @@
 #ifndef _NET_IF_STF_H_
 #define _NET_IF_STF_H_
 
+#include <sys/cdefs.h>
+
 void in_stf_input __P((struct mbuf *, ...));
 
 #endif /* _NET_IF_STF_H_ */
diff -ur src.old/sys/net/if_tap.h src/sys/net/if_tap.h
--- src.old/sys/net/if_tap.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_tap.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,8 @@
 #ifndef _NET_IF_TAP_H_
 #define _NET_IF_TAP_H_
 
+#include <sys/types.h>
+
 /* refer to if_tapvar.h for the softc stuff */
 
 /* maximum receive packet size (hard limit) */
diff -ur src.old/sys/net/if_tapvar.h src/sys/net/if_tapvar.h
--- src.old/sys/net/if_tapvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_tapvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -41,6 +41,11 @@
 #ifndef _NET_IF_TAPVAR_H_
 #define _NET_IF_TAPVAR_H_
 
+#include <sys/types.h>
+#include <net/if_arp.h>					/* arpcom */
+#include <net/ethernet.h>				/* ETHER_ADDR_LEN */
+#include "/usr/src/kfreebsd4-headers/sys/select.h"	/* selinfo */
+
 struct tap_softc {
 	struct arpcom	arpcom;			/* ethernet common data      */
 #define tap_if		arpcom.ac_if
diff -ur src.old/sys/net/if_tun.h src/sys/net/if_tun.h
--- src.old/sys/net/if_tun.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_tun.h	2004-09-16 15:11:07.000000000 +0200
@@ -19,6 +19,8 @@
 #ifndef _NET_IF_TUN_H_
 #define _NET_IF_TUN_H_
 
+#include <sys/types.h>
+
 /* Refer to if_tunvar.h for the softc stuff */
 
 /* Maximum transmit packet size (default) */
diff -ur src.old/sys/net/if_tunvar.h src/sys/net/if_tunvar.h
--- src.old/sys/net/if_tunvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_tunvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -29,6 +29,10 @@
 #ifndef _NET_IF_TUNVAR_H_
 #define _NET_IF_TUNVAR_H_
 
+#include <sys/types.h>
+#include <net/if_var.h>					/* ifnet */
+#include "/usr/src/kfreebsd4-headers/sys/select.h"	/* selinfo */
+
 struct tun_softc {
 	u_short	tun_flags;		/* misc flags */
 #define	TUN_OPEN	0x0001
diff -ur src.old/sys/net/if_vlan_var.h src/sys/net/if_vlan_var.h
--- src.old/sys/net/if_vlan_var.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/if_vlan_var.h	2004-09-16 15:11:07.000000000 +0200
@@ -32,6 +32,10 @@
 #ifndef _NET_IF_VLAN_VAR_H_
 #define	_NET_IF_VLAN_VAR_H_	1
 
+#include <sys/types.h>
+#include <net/ethernet.h>			/* ETHER_ADDR_LEN */
+#include <net/if.h>				/* IFNAMSIZ */
+
 #ifdef _KERNEL
 struct vlan_mc_entry {
 	struct ether_addr		mc_addr;
diff -ur src.old/sys/net/iso88025.h src/sys/net/iso88025.h
--- src.old/sys/net/iso88025.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/iso88025.h	2004-09-16 15:11:07.000000000 +0200
@@ -43,6 +43,8 @@
 #ifndef _NET_ISO88025_H_
 #define _NET_ISO88025_H_
 
+#include <sys/types.h>
+
 /*
  * General ISO 802.5 definitions
  */
diff -ur src.old/sys/net/pfkeyv2.h src/sys/net/pfkeyv2.h
--- src.old/sys/net/pfkeyv2.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/pfkeyv2.h	2004-09-16 15:11:07.000000000 +0200
@@ -39,6 +39,8 @@
 #ifndef _NET_PFKEYV2_H_
 #define _NET_PFKEYV2_H_
 
+#include <sys/types.h>
+
 /*
 This file defines structures and symbols for the PF_KEY Version 2
 key management interface. It was written at the U.S. Naval Research
diff -ur src.old/sys/net/ppp_defs.h src/sys/net/ppp_defs.h
--- src.old/sys/net/ppp_defs.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/ppp_defs.h	2004-09-16 15:11:07.000000000 +0200
@@ -30,6 +30,8 @@
 #ifndef _PPP_DEFS_H_
 #define _PPP_DEFS_H_
 
+#include <sys/types.h>
+
 /*
  * The basic PPP frame.
  */
diff -ur src.old/sys/net/radix.h src/sys/net/radix.h
--- src.old/sys/net/radix.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/radix.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,8 @@
 #ifndef _RADIX_H_
 #define	_RADIX_H_
 
+#include <sys/types.h>
+
 #ifdef MALLOC_DECLARE
 MALLOC_DECLARE(M_RTABLE);
 #endif
diff -ur src.old/sys/net/raw_cb.h src/sys/net/raw_cb.h
--- src.old/sys/net/raw_cb.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/raw_cb.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,7 @@
 #define _NET_RAW_CB_H_
 
 #include <sys/queue.h>
+#include "/usr/src/kfreebsd4-headers/sys/socket.h"	/* sockproto */
 
 /*
  * Raw protocol interface control block.  Used
diff -ur src.old/sys/net/route.h src/sys/net/route.h
--- src.old/sys/net/route.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/route.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,9 @@
 #ifndef _NET_ROUTE_H_
 #define _NET_ROUTE_H_
 
+#include <sys/types.h>
+#include <sys/socket.h>				/* sockaddr */
+
 /*
  * Kernel resident routing tables.
  *
diff -ur src.old/sys/net/slcompress.h src/sys/net/slcompress.h
--- src.old/sys/net/slcompress.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/net/slcompress.h	2004-09-16 15:11:07.000000000 +0200
@@ -40,6 +40,8 @@
 #ifndef _NET_SLCOMPRESS_H_
 #define _NET_SLCOMPRESS_H_
 
+#include <netinet/ip.h>				/* ip */
+
 #define MAX_STATES 16		/* must be > 2 and < 256 */
 #define MAX_HDR 128
 
diff -ur src.old/sys/netatalk/aarp.h src/sys/netatalk/aarp.h
--- src.old/sys/netatalk/aarp.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatalk/aarp.h	2004-09-16 15:11:07.000000000 +0200
@@ -6,6 +6,11 @@
  */
 
 #ifndef _NETATALK_AARP_H_
+
+#include <sys/types.h>
+#include <net/if_arp.h>				/* arphdr */
+#include <netatalk/at.h>			/* at_addr */
+
 /*
  * This structure is used for both phase 1 and 2. Under phase 1
  * the net is not filled in. It is in phase 2. In both cases, the
diff -ur src.old/sys/netatalk/at.h src/sys/netatalk/at.h
--- src.old/sys/netatalk/at.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatalk/at.h	2004-09-16 15:11:07.000000000 +0200
@@ -23,6 +23,9 @@
 
 #ifndef __AT_HEADER__
 #define __AT_HEADER__
+
+#include <sys/types.h>
+
 /*
  * Supported protocols
  */
diff -ur src.old/sys/netatalk/at_extern.h src/sys/netatalk/at_extern.h
--- src.old/sys/netatalk/at_extern.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatalk/at_extern.h	2004-09-16 15:11:07.000000000 +0200
@@ -1,3 +1,6 @@
+#include <sys/cdefs.h>			/* __P */
+#include <sys/types.h>
+
 struct mbuf;
 struct sockaddr_at;
 
diff -ur src.old/sys/netatalk/at_var.h src/sys/netatalk/at_var.h
--- src.old/sys/netatalk/at_var.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatalk/at_var.h	2004-09-16 15:11:07.000000000 +0200
@@ -25,6 +25,11 @@
 
 #ifndef _NETATALK_AT_VAR_H_
 #define _NETATALK_AT_VAR_H_ 1
+
+#include <net/if_var.h>	    			/* ifaddr */
+#include <netatalk/at.h>			/* sockaddr_at */
+#include <sys/callout.h>			/* callout_handle */
+
 /*
  * For phase2, we need to keep not only our address on an interface,
  * but also the legal networks on the interface.
diff -ur src.old/sys/netatalk/ddp.h src/sys/netatalk/ddp.h
--- src.old/sys/netatalk/ddp.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatalk/ddp.h	2004-09-16 15:11:07.000000000 +0200
@@ -23,6 +23,8 @@
 #ifndef _NETATALK_DDP_H_
 #define _NETATALK_DDP_H_ 1
 
+#include <sys/types.h>
+
 /*
  * <-1byte(8bits) ->
  * +---------------+
diff -ur src.old/sys/netatalk/ddp_var.h src/sys/netatalk/ddp_var.h
--- src.old/sys/netatalk/ddp_var.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatalk/ddp_var.h	2004-09-16 15:11:07.000000000 +0200
@@ -7,6 +7,10 @@
 
 #ifndef _NETATALK_DDP_VAR_H_
 #define _NETATALK_DDP_VAR_H_ 1
+
+#include <netatalk/at.h>     			/* sockaddr_at */
+#include <net/route.h>				/* route */
+
 struct ddpcb {
     struct sockaddr_at	ddp_fsat, ddp_lsat;
     struct route	ddp_route;
diff -ur src.old/sys/netatm/atm.h src/sys/netatm/atm.h
--- src.old/sys/netatm/atm.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/atm.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,7 @@
 #ifndef _NETATM_ATM_H
 #define _NETATM_ATM_H
 
+#include <net/if.h>				/* IFNAMSIZ */
 
 /*
  * The definitions in this file are intended to conform to the 
diff -ur src.old/sys/netatm/atm_cm.h src/sys/netatm/atm_cm.h
--- src.old/sys/netatm/atm_cm.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/atm_cm.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,13 @@
 #ifndef _NETATM_ATM_CM_H
 #define _NETATM_ATM_CM_H
 
+#include <sys/types.h>
+#include <netatm/atm_sys.h>                 /* Aal_t */
+#include <netatm/atm.h>			    /* t_atm_aal5 */
+#include <netatm/port.h>		    /* KBuffer */
+#include <netatm/queue.h>		    /* Qelem_t */
+#include <netatm/atm_sys.h>		    /* atm_time */
+
 /*
  * Forward declaration
  */
@@ -45,7 +52,6 @@
 struct atm_attributes;
 
 
-#ifdef ATM_KERNEL
 /*
  * Structure used to define a kernel ATM endpoint service module and its 
  * associated entry points.  An endpoint service is defined as a kernel 
@@ -94,7 +100,6 @@
 				__P((void *, KBuffer *, u_int));
 };
 typedef struct atm_endpoint	Atm_endpoint;
-#endif	/* ATM_KERNEL */
 
 /*
  * Endpoint IDs
@@ -230,7 +235,6 @@
 #define	CMAPI_SSCOP	2		/* Reliable data (SSCOP) */
 
 
-#ifdef ATM_KERNEL
 /*
  * ATM Connection Instance
  *
@@ -252,6 +256,7 @@
 };
 typedef struct atm_connection	Atm_connection;
 
+#ifdef ATM_KERNEL
 /*
  * Connection Flags
  */
diff -ur src.old/sys/netatm/atm_if.h src/sys/netatm/atm_if.h
--- src.old/sys/netatm/atm_if.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/atm_if.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,8 @@
 #ifndef _NETATM_ATM_IF_H
 #define _NETATM_ATM_IF_H
 
+#include <sys/types.h>
+
 /*
  * Handy constants
  */
diff -ur src.old/sys/netatm/atm_ioctl.h src/sys/netatm/atm_ioctl.h
--- src.old/sys/netatm/atm_ioctl.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/atm_ioctl.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,13 @@
 #ifndef _NETATM_ATM_IOCTL_H
 #define _NETATM_ATM_IOCTL_H
 
+#include <sys/types.h>
+#include <net/if.h>				/* IFNAMSIZ */
+#include <netatm/atm.h>				/* Atm_addr */
+#include <netatm/atm_if.h>			/* mac_addr */
+#include <netatm/atm_sap.h>			/* Sap_t */
+#include <netatm/atm_sys.h>			/* Aal_t */
+
 
 /*
  * Structure for PF_ATM configure (AIOCCFG) socket ioctls
diff -ur src.old/sys/netatm/atm_sap.h src/sys/netatm/atm_sap.h
--- src.old/sys/netatm/atm_sap.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/atm_sap.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,8 @@
 #ifndef _NETATM_ATM_SAP_H
 #define _NETATM_ATM_SAP_H
 
+#include <sys/types.h>
+
 /*
  * Service Access Point (SAP)
  *
diff -ur src.old/sys/netatm/atm_sys.h src/sys/netatm/atm_sys.h
--- src.old/sys/netatm/atm_sys.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/atm_sys.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,8 @@
 #ifndef _NETATM_ATM_SYS_H
 #define _NETATM_ATM_SYS_H
 
+#include <sys/types.h>
+
 /*
  * Software Version
  */
@@ -151,7 +153,6 @@
 #define	ATM_ENC_LLC	2		/* LLC encapsulation */
 
 
-#ifdef ATM_KERNEL
 /*
  * ATM timer control block.  Used to schedule a timeout via atm_timeout().
  * This control block will typically be embedded in a processing-specific
@@ -165,6 +166,7 @@
 	struct atm_time	*ti_next;	/* Next on queue */
 };
 
+#ifdef ATM_KERNEL
 /*
  * Timer Flags
  */
diff -ur src.old/sys/netatm/ipatm/ipatm_serv.h src/sys/netatm/ipatm/ipatm_serv.h
--- src.old/sys/netatm/ipatm/ipatm_serv.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/ipatm/ipatm_serv.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,12 @@
 #ifndef _IPATM_IPATM_SERV_H
 #define _IPATM_IPATM_SERV_H
 
+#include <sys/cdefs.h>				/* __P */
+#include <sys/types.h>
+#include <netatm/atm_sys.h>			/* Aal_t */
+#include <netatm/atm.h>				/* Atm_addr */
+#include <netatm/port.h>			/* KBuffer */
+#include <netinet/in.h>				/* in_addr */
 
 /*
  * Structures specifying VCC parameters and pointers to all of the IP 
diff -ur src.old/sys/netatm/port.h src/sys/netatm/port.h
--- src.old/sys/netatm/port.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/port.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,14 @@
 #ifndef _NETATM_PORT_H
 #define	_NETATM_PORT_H
 
+#if (defined(__FreeBSD__) || defined(__FreeBSD_kernel__)) && !defined(BSD)
+#define BSD 199506
+#endif
+
+#ifndef ATM_KERNEL
+#define ATM_KERNEL 1
+#endif
+
 /*
  * Try to ensure that this system is supported
  */
diff -ur src.old/sys/netatm/spans/spans_cls.h src/sys/netatm/spans/spans_cls.h
--- src.old/sys/netatm/spans/spans_cls.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/spans/spans_cls.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,11 @@
 #ifndef _SPANS_SPANSCLS_H
 #define _SPANS_SPANSCLS_H
 
+#include <sys/types.h>
+#include <netatm/atm_cm.h>			/* Atm_connection */
+#include <netatm/ipatm/ipatm_serv.h>		/* arpmap */
+#include <netatm/spans/spans_var.h>		/* spans_addr */
+
 /*
  * Protocol constants
  */
diff -ur src.old/sys/netatm/spans/spans_var.h src/sys/netatm/spans/spans_var.h
--- src.old/sys/netatm/spans/spans_var.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netatm/spans/spans_var.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,6 +38,10 @@
 #ifndef _SPANS_SPANS_VAR_H
 #define _SPANS_SPANS_VAR_H
 
+#include <netatm/atm_sigmgr.h>				/* siginst */
+#include <netatm/atm_vc.h>				/* vccb */
+
+
 #ifdef ATM_KERNEL
 /*
  * Constants to indicate the state of the signalling interface
@@ -99,9 +103,6 @@
 struct spans_vccb {
 	struct vccb	vcp_hdr;	/* Generic VCCB */
 	u_short		sv_retry;	/* Xmit retry count */
-	spans_atm_conn	sv_conn;	/* SPANS connection info */
-	spans_resrc sv_spans_qos;	/* QoS for VCC */
-	spans_aal	sv_spans_aal;	/* AAL for VCC */
 };
 
 #define	sv_type		vcp_hdr.vc_type
diff -ur src.old/sys/netinet/in_systm.h src/sys/netinet/in_systm.h
--- src.old/sys/netinet/in_systm.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/netinet/in_systm.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,8 @@
 #ifndef _NETINET_IN_SYSTM_H_
 #define _NETINET_IN_SYSTM_H_
 
+#include <sys/types.h>
+
 /*
  * Miscellaneous internetwork
  * definitions for kernel.
diff -ur src.old/sys/pccard/pcicvar.h src/sys/pccard/pcicvar.h
--- src.old/sys/pccard/pcicvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/pccard/pcicvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -111,7 +111,7 @@
 void pcic_putb_io(struct pcic_slot *sp, int reg, unsigned char val);
 int pcic_set_memory_offset(device_t bus, device_t child, int rid,
     u_int32_t offset
-#if __FreeBSD_version >= 500000
+#if 0a
     , u_int32_t *deltap
 #endif
     );
diff -ur src.old/sys/pccard/slot.h src/sys/pccard/slot.h
--- src.old/sys/pccard/slot.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/pccard/slot.h	2004-09-16 15:11:07.000000000 +0200
@@ -47,7 +47,7 @@
 #include <machine/bus.h>
 #include <sys/rman.h>
 #include <machine/resource.h>
-#if __FreeBSD_version >= 500000
+#if 0
 #include <sys/selinfo.h>
 #else
 #include <sys/select.h>
diff -ur src.old/sys/sys/buf.h src/sys/sys/buf.h
--- src.old/sys/sys/buf.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/buf.h	2004-09-16 15:11:07.000000000 +0200
@@ -44,6 +44,7 @@
 
 #include <sys/queue.h>
 #include <sys/lock.h>
+#include <machine/param.h>
 
 struct buf;
 struct mount;
diff -ur src.old/sys/sys/callout.h src/sys/sys/callout.h
--- src.old/sys/sys/callout.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/callout.h	2004-09-16 15:11:07.000000000 +0200
@@ -44,7 +44,13 @@
 
 #include <features.h>
 
-#include "/usr/src/kfreebsd-headers/sys/queue.h"
+#include <sys/queue.h>
+#ifndef SLIST_ENTRY
+#define SLIST_ENTRY(type)                                               \
+struct {                                                                \
+        struct type *sle_next;  /* next element */                      \
+}
+#endif
 
 SLIST_HEAD(callout_list, callout);
 TAILQ_HEAD(callout_tailq, callout);
diff -ur src.old/sys/sys/device_port.h src/sys/sys/device_port.h
--- src.old/sys/sys/device_port.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/device_port.h	2004-09-16 15:11:07.000000000 +0200
@@ -30,12 +30,8 @@
 #if defined(__NetBSD__)
 # include <sys/device.h>
 #elif defined(__FreeBSD_kernel__)
-# if __FreeBSD_version >= 400001
 #  include <sys/module.h>
 #  include <sys/bus.h>
-# else
-#  include <sys/device.h>
-# endif
 #endif
 
 /*
@@ -51,7 +47,7 @@
 /*
  * FreeBSD (compatibility for struct device)
  */
-#if __FreeBSD_version >= 400001
+#if 1
 # define DEVPORT_DEVICE			device_t
 # define DEVPORT_DEVNAME(dev)		device_get_name(dev)
 # define DEVPORT_DEVUNIT(dev)		device_get_unit(dev)
@@ -81,7 +77,7 @@
 /*
  * PC-Card device driver (compatibility for struct pccard_devinfo *)
  */
-#if __FreeBSD_version >= 400001
+#if 1
 # define DEVPORT_PDEVICE		device_t
 # define DEVPORT_PDEVUNIT(pdev)		device_get_unit(pdev)
 # define DEVPORT_PDEVFLAGS(pdev)	device_get_flags(pdev)
diff -ur src.old/sys/sys/devicestat.h src/sys/sys/devicestat.h
--- src.old/sys/sys/devicestat.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/devicestat.h	2004-09-16 15:11:07.000000000 +0200
@@ -32,6 +32,10 @@
 #define _DEVICESTAT_H
 
 #include <sys/queue.h>
+#define STAILQ_ENTRY(type)                                              \
+struct {                                                                \
+        struct type *stqe_next; /* next element */                      \
+}
 #include <sys/time.h>
 
 #define DEVSTAT_NAME_LEN  16
diff -ur src.old/sys/sys/elf_common.h src/sys/sys/elf_common.h
--- src.old/sys/sys/elf_common.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/elf_common.h	2004-09-16 15:11:07.000000000 +0200
@@ -29,6 +29,8 @@
 #ifndef _SYS_ELF_COMMON_H_
 #define _SYS_ELF_COMMON_H_ 1
 
+#include <sys/types.h>
+
 /*
  * ELF definitions that are independent of architecture or word size.
  */
diff -ur src.old/sys/sys/elf_generic.h src/sys/sys/elf_generic.h
--- src.old/sys/sys/elf_generic.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/elf_generic.h	2004-09-16 15:11:07.000000000 +0200
@@ -30,13 +30,24 @@
 #define _SYS_ELF_GENERIC_H_ 1
 
 #include <sys/cdefs.h>
+/* GNU __CONCAT apparently wasn't good enough (why?), so I copied this weird
+   hack from FreeBSD <sys/cdefs.h>.  We should probably merge this hack into
+   Glibc, but I don't know the rationale yet.. */
+#define __CONCAT1(x,y)  x ## y
+#undef __CONCAT
+#define __CONCAT(x,y)   __CONCAT1(x,y)
+#include <machine/elf.h>		/* __ELF_WORD_SIZE */
 
 /*
  * Definitions of generic ELF names which relieve applications from
  * needing to know the word size.
  */
 
-#if __ELF_WORD_SIZE != 32 && __ELF_WORD_SIZE != 64
+#if __ELF_WORD_SIZE == 32
+#include <sys/elf32.h>			/* Elf32_Addr */
+#elif __ELF_WORD_SIZE == 64
+#include <sys/elf64.h>			/* Elf64_Addr */
+#else
 #error "__ELF_WORD_SIZE must be defined as 32 or 64"
 #endif
 
diff -ur src.old/sys/sys/event.h src/sys/sys/event.h
--- src.old/sys/sys/event.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/event.h	2004-09-16 15:11:07.000000000 +0200
@@ -111,6 +111,25 @@
  * programs which pull in <sys/proc.h> or <sys/select.h>.
  */
 #include <sys/queue.h> 
+#ifndef SLIST_HEAD
+#define SLIST_HEAD(name, type)                                          \
+struct name {                                                           \
+        struct type *slh_first; /* first element */                     \
+}
+#endif
+#ifndef SLIST_ENTRY
+#define SLIST_ENTRY(type)                                               \
+struct {                                                                \
+        struct type *sle_next;  /* next element */                      \
+}
+#endif
+#ifndef TAILQ_ENTRY
+#define TAILQ_ENTRY(type)                                               \
+struct {                                                                \
+        struct type *tqe_next;  /* next element */                      \
+        struct type **tqe_prev; /* address of previous next element */  \
+}
+#endif
 struct knote;
 SLIST_HEAD(klist, knote);
 
diff -ur src.old/sys/sys/fbio.h src/sys/sys/fbio.h
--- src.old/sys/sys/fbio.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/fbio.h	2004-09-16 15:11:07.000000000 +0200
@@ -46,6 +46,7 @@
 #include <sys/types.h>
 #endif
 #include <sys/ioccom.h>
+#include <machine/types.h>		/* vm_offset_t */
 
 /*
  * Frame buffer ioctls (from Sprite, trimmed to essentials for X11).
diff -ur src.old/sys/sys/filedesc.h src/sys/sys/filedesc.h
--- src.old/sys/sys/filedesc.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/filedesc.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,7 +38,19 @@
 #define _SYS_FILEDESC_H_
 
 #include <sys/types.h>
-#include "/usr/src/kfreebsd-headers/sys/queue.h"
+#include <sys/queue.h>
+#ifndef SLIST_ENTRY
+#define SLIST_ENTRY(type)                                               \
+struct {                                                                \
+        struct type *sle_next;  /* next element */                      \
+}
+#endif
+#ifndef SLIST_HEAD
+#define SLIST_HEAD(name, type)                                          \
+struct name {                                                           \
+        struct type *slh_first; /* first element */                     \
+}
+#endif
 
 /*
  * This structure is used for the management of descriptors.  It may be
diff -ur src.old/sys/sys/inttypes.h src/sys/sys/inttypes.h
--- src.old/sys/sys/inttypes.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/inttypes.h	2004-09-16 15:11:07.000000000 +0200
@@ -1,24 +1 @@
-/*
- * This file is in the public domain.
- * $FreeBSD: src/sys/sys/inttypes.h,v 1.2 1999/08/28 00:51:47 peter Exp $
- */
-
-#ifndef _SYS_INTTYPES_H_
-#define	_SYS_INTTYPES_H_
-
-#include <machine/ansi.h>
-
-typedef	__int8_t	int8_t;
-typedef	__int16_t	int16_t;
-typedef	__int32_t	int32_t;
-typedef	__int64_t	int64_t;
-
-typedef	__uint8_t	uint8_t;
-typedef	__uint16_t	uint16_t;
-typedef	__uint32_t	uint32_t;
-typedef	__uint64_t	uint64_t;
-
-typedef	__intptr_t	intptr_t;
-typedef	__uintptr_t	uintptr_t;
-
-#endif /* !_SYS_INTTYPES_H_ */
+#include <inttypes.h>
diff -ur src.old/sys/sys/mbuf.h src/sys/sys/mbuf.h
--- src.old/sys/sys/mbuf.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/mbuf.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,8 @@
 #ifndef _SYS_MBUF_H_
 #define	_SYS_MBUF_H_
 
+#include <machine/param.h>			/* MSIZE */
+
 /*
  * Mbufs are of a single size, MSIZE (machine/param.h), which
  * includes overhead.  An mbuf may add a single "mbuf cluster" of size
diff -ur src.old/sys/sys/module.h src/sys/sys/module.h
--- src.old/sys/sys/module.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/module.h	2004-09-16 15:11:07.000000000 +0200
@@ -29,6 +29,8 @@
 #ifndef _SYS_MODULE_H_
 #define _SYS_MODULE_H_
 
+#include <sys/types.h>
+
 typedef enum modeventtype {
     MOD_LOAD,
     MOD_UNLOAD,
diff -ur src.old/sys/sys/mount.h src/sys/sys/mount.h
--- src.old/sys/sys/mount.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/mount.h	2004-09-16 15:11:21.000000000 +0200
@@ -45,12 +45,12 @@
 #endif /* !_POSIX_C_SOURCE */
 #endif /* !_KERNEL */
 
-#include <sys/queue.h>
+#include "/usr/src/kfreebsd4-headers/sys/queue.h"
 #ifdef _KERNEL
 #include <sys/lock.h>
 #endif
 
-typedef struct fsid { int32_t val[2]; } fsid_t;	/* file system id type */
+#include <sys/types.h>				/* fsid_t */
 
 /*
  * File identifier.
@@ -58,47 +58,16 @@
  */
 #define	MAXFIDSZ	16
 
-struct fid {
-	u_short		fid_len;		/* length of data in bytes */
-	u_short		fid_reserved;		/* force longword alignment */
-	char		fid_data[MAXFIDSZ];	/* data (variable length) */
-};
-
 /*
  * file system statistics
  */
 
-#define MFSNAMELEN	16	/* length of fs type name, including null */
-#ifdef __i386__
-#define	MNAMELEN	80	/* length of buffer for returned name */
-#endif
-#ifdef __alpha__
-#define	MNAMELEN	72	/* length of buffer for returned name */
-#endif
+#define MFSNAMELEN      16      /* length of fs type name, including null */
 
-struct statfs {
-	long	f_spare2;		/* placeholder */
-	long	f_bsize;		/* fundamental file system block size */
-	long	f_iosize;		/* optimal transfer block size */
-	long	f_blocks;		/* total data blocks in file system */
-	long	f_bfree;		/* free blocks in fs */
-	long	f_bavail;		/* free blocks avail to non-superuser */
-	long	f_files;		/* total file nodes in file system */
-	long	f_ffree;		/* free file nodes in fs */
-	fsid_t	f_fsid;			/* file system id */
-	uid_t	f_owner;		/* user that mounted the filesystem */
-	int	f_type;			/* type of filesystem */
-	int	f_flags;		/* copy of mount exported flags */
-	long    f_syncwrites;		/* count of sync writes since mount */
-	long    f_asyncwrites;		/* count of async writes since mount */
-	char	f_fstypename[MFSNAMELEN]; /* fs type name */
-	char	f_mntonname[MNAMELEN];	/* directory on which mounted */
-	long    f_syncreads;		/* count of sync reads since mount */
-	long    f_asyncreads;		/* count of async reads since mount */
-	short	f_spares1;		/* unused spare */
-	char	f_mntfromname[MNAMELEN];/* mounted filesystem */
-	short	f_spares2;		/* unused spare */
-	long    f_spare[2];		/* unused spare */
+struct fid {
+	u_short		fid_len;		/* length of data in bytes */
+	u_short		fid_reserved;		/* force longword alignment */
+	char		fid_data[MAXFIDSZ];	/* data (variable length) */
 };
 
 #ifdef _KERNEL
@@ -470,14 +439,8 @@
 __BEGIN_DECLS
 int	fstatfs __P((int, struct statfs *));
 int	getfh __P((const char *, fhandle_t *));
-int	getfsstat __P((struct statfs *, long, int));
-int	getmntinfo __P((struct statfs **, int));
-int	mount __P((const char *, const char *, int, void *));
 int	statfs __P((const char *, struct statfs *));
-int	unmount __P((const char *, int));
 int	fhopen __P((const struct fhandle *, int));
-int	fhstat __P((const struct fhandle *, struct stat *));
-int	fhstatfs __P((const struct fhandle *, struct statfs *));
 
 /* C library stuff */
 void	endvfsent __P((void));
diff -ur src.old/sys/sys/sbuf.h src/sys/sys/sbuf.h
--- src.old/sys/sys/sbuf.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/sbuf.h	2004-09-16 15:11:07.000000000 +0200
@@ -62,8 +62,8 @@
 int		 sbuf_bcpy(struct sbuf *, const char *, size_t);
 int		 sbuf_cat(struct sbuf *, const char *);
 int		 sbuf_cpy(struct sbuf *, const char *);
-int		 sbuf_printf(struct sbuf *, const char *, ...) __printflike(2, 3);
-int		 sbuf_vprintf(struct sbuf *, const char *, _BSD_VA_LIST_) __printflike(2, 0);
+int		 sbuf_printf(struct sbuf *, const char *, ...);
+int		 sbuf_vprintf(struct sbuf *, const char *, _BSD_VA_LIST_);
 int		 sbuf_putc(struct sbuf *, int);
 int		 sbuf_trim(struct sbuf *);
 int		 sbuf_overflowed(struct sbuf *);
diff -ur src.old/sys/sys/signalvar.h src/sys/sys/signalvar.h
--- src.old/sys/sys/signalvar.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/signalvar.h	2004-09-16 15:11:07.000000000 +0200
@@ -40,6 +40,7 @@
 #include <sys/signal.h>
 #include <sys/proc.h>
 #include <machine/smp.h>
+#include <machine/signal.h>		/* osigcontext */
 
 /*
  * Kernel signal definitions and data structures,
@@ -65,8 +66,7 @@
 	sigset_t ps_usertramp;		/* SunOS compat; libc sigtramp XXX */
 };
 
-#if 0
-
+#if !defined(_ANSI_SOURCE) && !defined(_POSIX_SOURCE)
 /*
  * Compatibility.
  */
diff -ur src.old/sys/sys/socket.h src/sys/sys/socket.h
--- src.old/sys/sys/socket.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/socket.h	2004-09-16 15:11:07.000000000 +0200
@@ -37,6 +37,8 @@
 #ifndef _SYS_SOCKET_H_
 #define	_SYS_SOCKET_H_
 
+#include <sys/types.h>
+
 #include <machine/ansi.h>
 #define _NO_NAMESPACE_POLLUTION
 #include <machine/param.h>
diff -ur src.old/sys/sys/systm.h src/sys/sys/systm.h
--- src.old/sys/sys/systm.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/systm.h	2004-09-16 15:11:07.000000000 +0200
@@ -45,6 +45,10 @@
 #include <machine/atomic.h>
 #include <machine/cpufunc.h>
 #include <sys/callout.h>
+#include <machine/types.h>	/* vm_offset_t */
+#include <sys/types.h>		/* dev_t */
+#include <sys/sysmacros.h>	/* major, minor, makedev */
+#include <machine/ansi.h>	/* _BSD_VA_LIST_ */
 
 extern int securelevel;		/* system security level (see init(8)) */
 
@@ -115,20 +119,20 @@
 void	init_param1 __P((void));
 void	init_param2 __P((int physpages));
 void	tablefull __P((const char *));
-int	addlog __P((const char *, ...)) __printflike(1, 2);
+int	addlog __P((const char *, ...));
 int	kvprintf __P((char const *, void (*)(int, void*), void *, int,
-		      _BSD_VA_LIST_)) __printflike(1, 0);
-int	log __P((int, const char *, ...)) __printflike(2, 3);
+		      _BSD_VA_LIST_));
+int	log __P((int, const char *, ...));
 void	logwakeup __P((void));
 void	log_console __P((struct uio *));
-int	printf __P((const char *, ...)) __printflike(1, 2);
-int	snprintf __P((char *, size_t, const char *, ...)) __printflike(3, 4);
-int	sprintf __P((char *buf, const char *, ...)) __printflike(2, 3);
-int	uprintf __P((const char *, ...)) __printflike(1, 2);
-int	vprintf __P((const char *, _BSD_VA_LIST_)) __printflike(1, 0);
-int	vsnprintf __P((char *, size_t, const char *, _BSD_VA_LIST_)) __printflike(3, 0);
-int     vsprintf __P((char *buf, const char *, _BSD_VA_LIST_)) __printflike(2, 0);
-int	ttyprintf __P((struct tty *, const char *, ...)) __printflike(2, 3);
+int	printf __P((const char *, ...));
+int	snprintf __P((char *, size_t, const char *, ...));
+int	sprintf __P((char *buf, const char *, ...));
+int	uprintf __P((const char *, ...));
+int	vprintf __P((const char *, _BSD_VA_LIST_));
+int	vsnprintf __P((char *, size_t, const char *, _BSD_VA_LIST_));
+int     vsprintf __P((char *buf, const char *, _BSD_VA_LIST_));
+int	ttyprintf __P((struct tty *, const char *, ...));
 int	sscanf __P((const char *, char const *, ...));
 int	vsscanf __P((const char *, char const *, _BSD_VA_LIST_));
 long	strtol __P((const char *, char **, int));
@@ -338,12 +342,9 @@
  * Common `dev_t' stuff are declared here to avoid #include poisoning
  */
 
-int major(dev_t x);
-int minor(dev_t x);
-dev_t makedev(int x, int y);
-udev_t dev2udev(dev_t x);
-dev_t udev2dev(udev_t x, int b);
-int uminor(udev_t dev);
-int umajor(udev_t dev);
-udev_t makeudev(int x, int y);
+dev_t dev2udev(dev_t x);
+dev_t udev2dev(dev_t x, int b);
+int uminor(dev_t dev);
+int umajor(dev_t dev);
+dev_t makeudev(int x, int y);
 #endif /* !_SYS_SYSTM_H_ */
diff -ur src.old/sys/sys/time.h src/sys/sys/time.h
--- src.old/sys/sys/time.h	2004-09-16 15:11:01.000000000 +0200
+++ src/sys/sys/time.h	2004-09-16 15:11:07.000000000 +0200
@@ -38,22 +38,10 @@
 #define _SYS_TIME_H_
 
 #include <sys/types.h>
-
-/*
- * Structure returned by gettimeofday(2) system call,
- * and used in other calls.
- */
-struct timeval {
-	long	tv_sec;		/* seconds */
-	long	tv_usec;	/* and microseconds */
-};
+#include <time.h>
 
 #ifndef _TIMESPEC_DECLARED
 #define _TIMESPEC_DECLARED
-struct timespec {
-	time_t	tv_sec;		/* seconds */
-	long	tv_nsec;	/* and nanoseconds */
-};
 #endif
 
 #define	TIMEVAL_TO_TIMESPEC(tv, ts)					\
@@ -67,7 +55,7 @@
 		(tv)->tv_usec = (ts)->tv_nsec / 1000;			\
 	} while (0)
 
-struct timezone {
+struct __kernel_timezone {
 	int	tz_minuteswest;	/* minutes west of Greenwich */
 	int	tz_dsttime;	/* type of dst correction */
 };
@@ -152,8 +140,8 @@
 	u_int32_t		tc_offset_sec;
 	u_int32_t		tc_offset_micro;
 	u_int64_t		tc_offset_nano;
-	struct timeval		tc_microtime;
-	struct timespec		tc_nanotime;
+	struct __kernel_timeval		tc_microtime;
+	struct __kernel_timespec		tc_nanotime;
 	struct timecounter	*tc_avail;
 	struct timecounter	*tc_other;
 	struct timecounter	*tc_tweak;
@@ -236,11 +224,6 @@
 #define	ITIMER_VIRTUAL	1
 #define	ITIMER_PROF	2
 
-struct	itimerval {
-	struct	timeval it_interval;	/* timer interval */
-	struct	timeval it_value;	/* current value */
-};
-
 /*
  * Getkerninfo clock information structure
  */
@@ -291,14 +274,14 @@
 #include <sys/cdefs.h>
 
 __BEGIN_DECLS
-int	adjtime __P((const struct timeval *, struct timeval *));
-int	futimes __P((int, const struct timeval *));
-int	getitimer __P((int, struct itimerval *));
-int	gettimeofday __P((struct timeval *, struct timezone *));
-int	lutimes __P((const char *, const struct timeval *));
-int	setitimer __P((int, const struct itimerval *, struct itimerval *));
-int	settimeofday __P((const struct timeval *, const struct timezone *));
-int	utimes __P((const char *, const struct timeval *));
+int	adjtime __P((const struct __kernel_timeval *, struct __kernel_timeval *));
+int	futimes __P((int, const struct __kernel_timeval *));
+int	getitimer __P((int, struct __kernel_itimerval *));
+int	gettimeofday __P((struct __kernel_timeval *, struct __kernel_timezone *));
+int	lutimes __P((const char *, const struct __kernel_timeval *));
+int	setitimer __P((int, const struct __kernel_itimerval *, struct __kernel_itimerval *));
+int	settimeofday __P((const struct __kernel_timeval *, const struct __kernel_timezone *));
+int	utimes __P((const char *, const struct __kernel_timeval *));
 __END_DECLS
 
 #endif /* !_KERNEL */
