*** orig/sys/rtprio.h	Wed Dec 29 05:24:46 1999
--- import/sys/rtprio.h	Fri Sep  6 01:13:46 2002
***************
*** 1,3 ****
--- 1,22 ----
+ /* Header file for manipulating realtime priority.  FreeBSD version.
+    Copyright (C) 2002 Free Software Foundation, Inc.
+    This file is part of the GNU C Library.
+ 
+    The GNU C Library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+ 
+    The GNU C Library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+ 
+    You should have received a copy of the GNU Lesser General Public
+    License along with the GNU C Library; if not, write to the Free
+    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+    02111-1307 USA.  */
+ 
  /*
   * Copyright (c) 1994, Henrik Vestergaard Draboel
   * All rights reserved.
***************
*** 34,39 ****
--- 53,62 ----
  #ifndef _SYS_RTPRIO_H_
  #define _SYS_RTPRIO_H_
  
+ #include <features.h>
+ 
+ #include <sys/types.h>
+ 
  /*
   * Process realtime-priority specifications to rtprio.
   */
***************
*** 63,80 ****
  #define RTP_LOOKUP		0
  #define RTP_SET			1
  
- #ifndef LOCORE
  struct rtprio {
  	u_short type;
  	u_short prio;
  };
- #endif
- 
- #ifndef _KERNEL
- #include <sys/cdefs.h>
  
  __BEGIN_DECLS
! int	rtprio __P((int, pid_t, struct rtprio *));
  __END_DECLS
! #endif	/* !_KERNEL */
  #endif	/* !_SYS_RTPRIO_H_ */
--- 86,105 ----
  #define RTP_LOOKUP		0
  #define RTP_SET			1
  
  struct rtprio {
  	u_short type;
  	u_short prio;
  };
  
  __BEGIN_DECLS
! 
! extern int rtprio (int __function, __pid_t __pid, struct rtprio *__rtp)
!      __THROW;
! 
! #ifdef _LIBC
! extern int __rtprio (int __function, __pid_t __pid, struct rtprio *__rtp);
! #endif
! 
  __END_DECLS
! 
  #endif	/* !_SYS_RTPRIO_H_ */
