---
 sys/sys/syslimits.h |    9 ---------
 1 file changed, 9 deletions(-)

--- a/sys/sys/syslimits.h
+++ b/sys/sys/syslimits.h
@@ -33,15 +33,6 @@
 #ifndef _SYS_SYSLIMITS_H_
 #define _SYS_SYSLIMITS_H_
 
-#if !defined(_KERNEL) && !defined(_LIMITS_H_) && !defined(_SYS_PARAM_H_)
-#ifndef _SYS_CDEFS_H
-#error this file needs sys/cdefs.h as a prerequisite
-#endif
-#if defined(__GNUC__)
-#warning "No user-serviceable parts inside."
-#endif
-#endif
-
 /*
  * Do not add any new variables here.  (See the comment at the end of
  * the file for why.)
