---
 sys/ufs/ufs/quota.h |    1 -
 1 file changed, 1 deletion(-)

--- a/sys/ufs/ufs/quota.h
+++ b/sys/ufs/ufs/quota.h
@@ -237,7 +237,6 @@
 int	setquota(struct thread *, struct mount *, u_long, int, void *);
 int	setuse(struct thread *, struct mount *, u_long, int, void *);
 int	getquotasize(struct thread *, struct mount *, u_long, int, void *);
-vfs_quotactl_t ufs_quotactl;
 
 #ifdef SOFTUPDATES
 int	quotaref(struct vnode *, struct dquot **);
