diff -ur src/sys/vm.old/vm_param.h src/sys/vm/vm_param.h
--- src/sys/vm.old/vm_param.h	2004-08-12 01:11:00.000000000 +0200
+++ src/sys/vm/vm_param.h	2004-08-12 01:10:15.000000000 +0200
@@ -71,6 +71,7 @@
 #ifndef	_VM_PARAM_
 #define	_VM_PARAM_
 
+#include <sys/types.h>
 #include <machine/vmparam.h>
 
 /*
@@ -110,7 +111,7 @@
  */
 #define XSWDEV_VERSION	1
 struct xswdev {
-	u_int	xsw_version;
+	unsigned int	xsw_version;
 	dev_t	xsw_dev;
 	int	xsw_flags;
 	int	xsw_nblks;
