--- src/sys/sys/mount.h~	2004-11-30 01:27:13.000000000 +0100
+++ src/sys/sys/mount.h	2004-11-30 01:39:31.000000000 +0100
@@ -357,7 +357,7 @@
 #define	VFCF_LOOPBACK	0x00100000	/* aliases some other mounted FS */
 #define	VFCF_UNICODE	0x00200000	/* stores file names as Unicode*/
 
-typedef uint32_t fsctlop_t;
+typedef u_int32_t fsctlop_t;
 
 struct vfsidctl {
 	int		vc_vers;	/* should be VFSIDCTL_VERS1 (below) */
