diff -ur src.old/sys/i386/include/_types.h src/sys/i386/include/_types.h
--- src.old/sys/i386/include/_types.h	2004-03-20 21:41:40.000000000 +0100
+++ src/sys/i386/include/_types.h	2005-01-13 05:38:54.000000000 +0100
@@ -39,6 +39,7 @@
 #ifndef _MACHINE__TYPES_H_
 #define	_MACHINE__TYPES_H_
 
+#ifndef __GLIBC__
 /*
  * Basic types upon which most other types are built.
  */
@@ -63,6 +64,7 @@
 /* LONGLONG */
 typedef	unsigned long long	__uint64_t;
 #endif
+#endif /* __GLIBC__ */
 
 /*
  * Standard type definitions.
@@ -74,7 +76,11 @@
 typedef	double		__float_t;
 typedef	__int32_t	__intfptr_t;
 typedef	__int64_t	__intmax_t;
+#ifndef __GLIBC__
 typedef	__int32_t	__intptr_t;
+typedef	__int32_t	__ssize_t;		/* byte count or error */
+typedef	__uint32_t	__size_t;		/* sizeof() */
+#endif
 typedef	__int32_t	__int_fast8_t;
 typedef	__int32_t	__int_fast16_t;
 typedef	__int32_t	__int_fast32_t;
@@ -86,7 +92,6 @@
 typedef	__int32_t	__ptrdiff_t;		/* ptr1 - ptr2 */
 typedef	__int32_t	__register_t;
 typedef	__int32_t	__segsz_t;		/* segment size (in pages) */
-typedef	__uint32_t	__size_t;		/* sizeof() */
 typedef	__int32_t	__ssize_t;		/* byte count or error */
 typedef	__int32_t	__time_t;		/* time()... */
 typedef	__uint32_t	__uintfptr_t;
