---
 sys/conf/Makefile.mips |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/sys/conf/Makefile.mips
+++ b/sys/conf/Makefile.mips
@@ -71,7 +71,7 @@
 # XXX hardcoded kernel entry point
 ASM_CFLAGS+=${CFLAGS} -D_LOCORE -DLOCORE
 
-KERNEL_EXTRA=trampoline
+#KERNEL_EXTRA=trampoline
 trampoline: ${KERNEL_KO}.tramp.bin
 ${KERNEL_KO}.tramp.bin: ${KERNEL_KO} $S/$M/$M/elf_trampoline.c \
 	$S/$M/$M/inckern.S 
