--- bsd-airtools/dstumbler/configure~	2006-03-29 15:37:24.000000000 +0200
+++ bsd-airtools/dstumbler/configure	2006-03-29 15:39:18.000000000 +0200
@@ -33,6 +33,12 @@
     ldflags="-L/usr/local/lib -lncurses"
     curses="ncurses"
     cursesincl="/usr/include/ncurses.h" ;;
+  "GNU/kFreeBSD"*)
+    hostos="freebsd"; echo $hostos
+    cflags="-Wall"
+    ldflags="-lncurses -lbsd"
+    curses="ncurses"
+    cursesincl="/usr/include/ncurses.h" ;;
   *)
     hostos="unsupported"; echo $hostos;
     echo
