
Patch from upstream (1.155:1.158).  Thanks to Petr for spotting this.

--- src.old/usr.bin/make/main.c	2006-02-12 17:43:38.000000000 +0100
+++ src/usr.bin/make/main.c	2006-02-12 17:44:22.000000000 +0100
@@ -235,14 +235,14 @@
 static Boolean
 ReadMakefile(const char p[])
 {
-	char *fname;			/* makefile to read */
+	char *fname, *fnamesave;	/* makefile to read */
 	FILE *stream;
 	char *name, path[MAXPATHLEN];
 	char *MAKEFILE;
 	int setMAKEFILE;
 
 	/* XXX - remove this once constification is done */
-	fname = estrdup(p);
+	fnamesave = fname = estrdup(p);
 
 	if (!strcmp(fname, "-")) {
 		Parse_File("(stdin)", stdin);
@@ -297,8 +297,10 @@
 		name = Path_FindFile(fname, &parseIncPath);
 		if (!name)
 			name = Path_FindFile(fname, &sysIncPath);
-		if (!name || !(stream = fopen(name, "r")))
+		if (!name || !(stream = fopen(name, "r"))) {
+			free(fnamesave);
 			return (FALSE);
+		}
 		MAKEFILE = fname = name;
 		/*
 		 * set the MAKEFILE variable desired by System V fans -- the
@@ -309,8 +311,8 @@
 		if (setMAKEFILE)
 			Var_SetGlobal("MAKEFILE", MAKEFILE);
 		Parse_File(fname, stream);
-		fclose(stream);
 	}
+	free(fnamesave);
 	return (TRUE);
 }
 
