diff -ur ../foo.old/src/share/mk/bsd.incs.mk ./src/share/mk/bsd.incs.mk
--- ../foo.old/src/share/mk/bsd.incs.mk	2005-12-17 12:45:38.000000000 +0100
+++ ./src/share/mk/bsd.incs.mk	2005-12-17 12:49:33.000000000 +0100
@@ -39,7 +39,7 @@
 .endif
 installincludes: _${group}INS_${header:T}
 _${group}INS_${header:T}: ${header}
-	${INSTALL} -C -o ${${group}OWN_${.ALLSRC:T}} \
+	${INSTALL} -o ${${group}OWN_${.ALLSRC:T}} \
 	    -g ${${group}GRP_${.ALLSRC:T}} -m ${${group}MODE_${.ALLSRC:T}} \
 	    ${.ALLSRC} \
 	    ${DESTDIR}${${group}DIR_${.ALLSRC:T}}/${${group}NAME_${.ALLSRC:T}}
@@ -51,10 +51,10 @@
 installincludes: _${group}INS
 _${group}INS: ${_${group}INCS}
 .if defined(${group}NAME)
-	${INSTALL} -C -o ${${group}OWN} -g ${${group}GRP} -m ${${group}MODE} \
+	${INSTALL} -o ${${group}OWN} -g ${${group}GRP} -m ${${group}MODE} \
 	    ${.ALLSRC} ${DESTDIR}${${group}DIR}/${${group}NAME}
 .else
-	${INSTALL} -C -o ${${group}OWN} -g ${${group}GRP} -m ${${group}MODE} \
+	${INSTALL} -o ${${group}OWN} -g ${${group}GRP} -m ${${group}MODE} \
 	    ${.ALLSRC} ${DESTDIR}${${group}DIR}
 .endif
 .endif
diff -ur ../foo.old/src/share/mk/bsd.info.mk ./src/share/mk/bsd.info.mk
--- ../foo.old/src/share/mk/bsd.info.mk	2005-12-17 12:45:38.000000000 +0100
+++ ./src/share/mk/bsd.info.mk	2005-12-17 12:50:33.000000000 +0100
@@ -75,7 +75,7 @@
 MAKEINFOFLAGS+=	--no-split # simplify some things, e.g., compression
 SRCDIR?=	${.CURDIR}
 INFODIRFILE?=   dir
-INSTALLINFO?=   install-info
+INSTALLINFO?=   `which ginstall-info || which install-info`
 INSTALLINFOFLAGS+=--quiet
 INFOSECTION?=   Miscellaneous
 ICOMPRESS_CMD?=	${COMPRESS_CMD}
@@ -145,10 +145,8 @@
 .for x in ${INFO}
 INSTALLINFODIRS+= ${x:S/$/-install/}
 ${x:S/$/-install/}:
-	${INSTALLINFO} ${INSTALLINFOFLAGS} \
-	    --defsection=${INFOSECTION} \
-	    --defentry=${INFOENTRY_${x}} \
-	    ${x}.info ${DESTDIR}${INFODIR}/${INFODIRFILE}
+	mkdir -p ${DESTDIR}${INFODIR}
+	install -m 644 ${x}.info ${DESTDIR}${INFODIR}/
 .endfor
 
 .PHONY: ${INSTALLINFODIRS}
diff -ur ../foo.old/src/share/mk/bsd.lib.mk ./src/share/mk/bsd.lib.mk
--- ../foo.old/src/share/mk/bsd.lib.mk	2005-12-17 12:45:38.000000000 +0100
+++ ./src/share/mk/bsd.lib.mk	2005-12-17 12:49:11.000000000 +0100
@@ -109,7 +109,7 @@
 lib${LIB}.a: ${OBJS} ${STATICOBJS}
 	@${ECHO} building static ${LIB} library
 	@rm -f ${.TARGET}
-	@${AR} cq ${.TARGET} `lorder ${OBJS} ${STATICOBJS} | tsort -q` ${ARADD}
+	@${AR} cq ${.TARGET} `lorder ${OBJS} ${STATICOBJS} | tsort` ${ARADD}
 	${RANLIB} ${.TARGET}
 .endif
 
@@ -122,7 +122,7 @@
 lib${LIB}_p.a: ${POBJS}
 	@${ECHO} building profiled ${LIB} library
 	@rm -f ${.TARGET}
-	@${AR} cq ${.TARGET} `lorder ${POBJS} | tsort -q` ${ARADD}
+	@${AR} cq ${.TARGET} `lorder ${POBJS} | tsort` ${ARADD}
 	${RANLIB} ${.TARGET}
 .endif
 
@@ -138,11 +138,11 @@
 	@${ECHO} building shared library ${SHLIB_NAME}
 	@rm -f ${.TARGET} ${SHLIB_LINK}
 .if defined(SHLIB_LINK)
-	@ln -fs ${.TARGET} ${SHLIB_LINK}
+	@ln -fs ${SHLIBDIR}${.TARGET} ${SHLIB_LINK}
 .endif
 	@${CC} ${LDFLAGS} -shared -Wl,-x \
 	    -o ${.TARGET} -Wl,-soname,${SONAME} \
-	    `lorder ${SOBJS} | tsort -q` ${LDADD}
+	    `lorder ${SOBJS} | tsort` ${LDADD}
 .endif
 
 .if defined(INSTALL_PIC_ARCHIVE) && defined(LIB) && !empty(LIB)
@@ -208,11 +208,11 @@
 .ORDER: beforeinstall _libinstall
 _libinstall:
 .if defined(LIB) && !empty(LIB) && !defined(NO_INSTALLLIB)
-	${INSTALL} -C -o ${LIBOWN} -g ${LIBGRP} -m ${LIBMODE} \
+	${INSTALL} -o ${LIBOWN} -g ${LIBGRP} -m ${LIBMODE} \
 	    ${_INSTALLFLAGS} lib${LIB}.a ${DESTDIR}${LIBDIR}
 .endif
 .if !defined(NO_PROFILE) && defined(LIB) && !empty(LIB)
-	${INSTALL} -C -o ${LIBOWN} -g ${LIBGRP} -m ${LIBMODE} \
+	${INSTALL} -o ${LIBOWN} -g ${LIBGRP} -m ${LIBMODE} \
 	    ${_INSTALLFLAGS} lib${LIB}_p.a ${DESTDIR}${LIBDIR}
 .endif
 .if defined(SHLIB_NAME)
diff -ur ../foo.old/src/share/mk/bsd.man.mk ./src/share/mk/bsd.man.mk
--- ../foo.old/src/share/mk/bsd.man.mk	2005-01-06 12:12:43.000000000 +0100
+++ ./src/share/mk/bsd.man.mk	2005-12-17 12:49:11.000000000 +0100
@@ -211,12 +211,12 @@
 			[123]) echo "warn: empty MLINK: $$1 $$2 $$3"; break;; \
 		esac; \
 		name=$$1; shift; sect=$$1; shift; \
-		l=${DESTDIR}${MANDIR}$${sect}${MANSUBDIR}/$$name; \
+		l=${MANDIR}$${sect}${MANSUBDIR}/$$name; \
 		name=$$1; shift; sect=$$1; shift; \
 		t=${DESTDIR}${MANDIR}$${sect}${MANSUBDIR}/$$name; \
 		${ECHO} $${t}${ZEXT} -\> $${l}${ZEXT}; \
 		rm -f $${t} $${t}${MCOMPRESS_EXT}; \
-		ln $${l}${ZEXT} $${t}${ZEXT}; \
+		ln -s $${l}${ZEXT} $${t}${ZEXT}; \
 	done
 .if defined(MANBUILDCAT) && !empty(MANBUILDCAT)
 	@set ${MLINKS:C/\.([^.]*)$/.\1 \1/}; \
@@ -226,12 +226,12 @@
 			[123]) echo "warn: empty MLINK: $$1 $$2 $$3"; break;; \
 		esac; \
 		name=$$1; shift; sect=$$1; shift; \
-		l=${DESTDIR}${CATDIR}$${sect}${MANSUBDIR}/$$name; \
+		l=${CATDIR}$${sect}${MANSUBDIR}/$$name; \
 		name=$$1; shift; sect=$$1; shift; \
 		t=${DESTDIR}${CATDIR}$${sect}${MANSUBDIR}/$$name; \
 		${ECHO} $${t}${ZEXT} -\> $${l}${ZEXT}; \
 		rm -f $${t} $${t}${MCOMPRESS_EXT}; \
-		ln $${l}${ZEXT} $${t}${ZEXT}; \
+		ln -s $${l}${ZEXT} $${t}${ZEXT}; \
 	done
 .endif
 .endif
diff -ur ../foo.old/src/share/mk/bsd.own.mk ./src/share/mk/bsd.own.mk
--- ../foo.old/src/share/mk/bsd.own.mk	2005-04-11 09:13:29.000000000 +0200
+++ ./src/share/mk/bsd.own.mk	2005-12-17 12:49:11.000000000 +0100
@@ -18,7 +18,7 @@
 #
 # BINOWN	Binary owner. [root]
 #
-# BINGRP	Binary group. [wheel]
+# BINGRP	Binary group. [root]
 #
 # BINMODE	Binary mode. [555]
 #
@@ -56,7 +56,7 @@
 #
 # SHAREOWN	ASCII text file owner. [root]
 #
-# SHAREGRP	ASCII text file group. [wheel]
+# SHAREGRP	ASCII text file group. [root]
 #
 # SHAREMODE	ASCII text file mode. [${NOBINMODE}]
 #
@@ -106,7 +106,7 @@
 
 # Binaries
 BINOWN?=	root
-BINGRP?=	wheel
+BINGRP?=	root
 BINMODE?=	555
 NOBINMODE?=	444
 
@@ -132,7 +132,7 @@
 # Share files
 SHAREDIR?=	/usr/share
 SHAREOWN?=	root
-SHAREGRP?=	wheel
+SHAREGRP?=	root
 SHAREMODE?=	${NOBINMODE}
 
 MANDIR?=	${SHAREDIR}/man/man
diff -ur ../foo.old/src/usr.bin/make/Makefile ./src/usr.bin/make/Makefile
--- ../foo.old/src/usr.bin/make/Makefile	2005-12-17 12:46:35.000000000 +0100
+++ ./src/usr.bin/make/Makefile	2005-12-17 12:49:11.000000000 +0100
@@ -10,7 +10,6 @@
 
 NO_WERROR=
 WARNS?=	6
-NO_SHARED?=	YES
 
 CFLAGS+=-DMAKE_VERSION=\"5200408120\"
 .if defined(_UPGRADING)
diff -ur ../foo.old/src/usr.bin/make/pathnames.h ./src/usr.bin/make/pathnames.h
--- ../foo.old/src/usr.bin/make/pathnames.h	2005-03-11 14:02:38.000000000 +0100
+++ ./src/usr.bin/make/pathnames.h	2005-12-17 12:49:11.000000000 +0100
@@ -50,7 +50,7 @@
 #endif /* ! PATH_DEFSYSMK */
 
 #ifndef PATH_DEFSYSPATH
-#define	PATH_DEFSYSPATH	"/usr/share/mk"
+#define	PATH_DEFSYSPATH	"/usr/share/freebsd6-buildutils/mk"
 #endif /* ! PATH_DEFSYSPATH */
 
 #endif /* pathnames_h_235b888a */
