diff -ur src/usr.bin/make/job.c src/usr.bin/make/job.c
--- src/usr.bin/make/job.c	2005-12-17 12:46:35.000000000 +0100
+++ src/usr.bin/make/job.c	2005-12-17 12:59:30.000000000 +0100
@@ -304,7 +304,7 @@
  * The maximum number of jobs that may run. This is initialize from the
  * -j argument for the leading make and from the FIFO for sub-makes.
  */
-static int	maxJobs;
+int	maxJobs;
 
 static int	nJobs;		/* The number of children currently running */
 
diff -ur src/usr.bin/make/main.c src/usr.bin/make/main.c
--- src/usr.bin/make/main.c	2005-12-17 12:46:35.000000000 +0100
+++ src/usr.bin/make/main.c	2005-12-17 12:59:30.000000000 +0100
@@ -329,12 +329,11 @@
 static void
 MainParseArgs(int argc, char **argv)
 {
-	int c;
+	int c, optind;
 	Boolean	found_dd = FALSE;
 
 rearg:
 	optind = 1;	/* since we're called more than once */
-	optreset = 1;
 #define OPTFLAGS "ABC:D:E:I:PSV:Xd:ef:ij:km:nqrstvx:"
 	for (;;) {
 		if ((optind < argc) && strcmp(argv[optind], "--") == 0) {
