diff -ur ../foo.old/ports/Mk/bsd.port.mk ./ports/Mk/bsd.port.mk
--- ../foo.old/ports/Mk/bsd.port.mk	2005-11-08 10:02:51.000000000 +0100
+++ ./ports/Mk/bsd.port.mk	2005-12-17 00:07:52.000000000 +0100
@@ -473,7 +473,7 @@
 #				  files.
 #				  Default: ${.CURDIR}
 # PORTSDIR		- The root of the ports tree.
-#				  Default: /usr/ports
+#				  Default: /usr/share/freebsd6-buildutils/ports
 # DISTDIR		- Where to search for and store copies of original sources
 #				  Default: ${PORTSDIR}/distfiles
 # PACKAGES		- A top level directory where all packages go (rather than
@@ -960,9 +960,9 @@
 .endif
 .endif
 
-AWK?=		/usr/bin/awk
+AWK?=		freebsd-awk
 BASENAME?=	/usr/bin/basename
-BRANDELF?=	/usr/bin/brandelf
+BRANDELF?=	freebsd-brandelf
 .if exists(/usr/bin/bzip2)
 BZCAT?=		/usr/bin/bzcat
 BZIP2_CMD?=	/usr/bin/bzip2
@@ -972,56 +972,56 @@
 BZIP2DEPENDS=	yes
 .endif
 CAT?=		/bin/cat
-CHGRP?=		/usr/bin/chgrp
+CHGRP?=		chgrp
 CHMOD?=		/bin/chmod
-CHOWN?=		/usr/sbin/chown
+CHOWN?=		chown
 COMM?=		/usr/bin/comm
 CP?=		/bin/cp
-CPIO?=		/usr/bin/cpio
+CPIO?=		cpio
 CUT?=		/usr/bin/cut
-DC?=		/usr/bin/dc
+DC?=		dc
 DIALOG?=	/usr/bin/dialog
 DIFF?=		/usr/bin/diff
 DIRNAME?=	/usr/bin/dirname
-EGREP?=		/usr/bin/egrep
-EXPR?=		/bin/expr
+EGREP?=		egrep
+EXPR?=		expr
 FALSE?=		false				# Shell builtin
 FILE?=		/usr/bin/file
 FIND?=		/usr/bin/find
 FMT?=		/usr/bin/fmt
-GREP?=		/usr/bin/grep
-GUNZIP_CMD?=	/usr/bin/gunzip -f
-GZCAT?=		/usr/bin/gzcat
+GREP?=		grep
+GUNZIP_CMD?=	gunzip -f
+GZCAT?=		zcat
 GZIP?=		-9
-GZIP_CMD?=	/usr/bin/gzip -nf ${GZIP}
+GZIP_CMD?=	gzip -nf ${GZIP}
 HEAD?=		/usr/bin/head
 ID?=		/usr/bin/id
-IDENT?=		/usr/bin/ident
+IDENT?=		ident
 LDCONFIG?=	/sbin/ldconfig
 LN?=		/bin/ln
 LS?=		/bin/ls
 MKDIR?=		/bin/mkdir -p
-MKTEMP?=	/usr/bin/mktemp
+MKTEMP?=	mktemp
 MV?=		/bin/mv
 OBJCOPY?=	/usr/bin/objcopy
 OBJDUMP?=	/usr/bin/objdump
 PASTE?=		/usr/bin/paste
 PAX?=		/bin/pax
 PRINTF?=	/usr/bin/printf
-REALPATH?=	/bin/realpath
+REALPATH?=	realpath
 RM?=		/bin/rm
 RMDIR?=		/bin/rmdir
-SED?=		/usr/bin/sed
+SED?=		sed
 SETENV?=	/usr/bin/env
 SH?=		/bin/sh
 SORT?=		/usr/bin/sort
 STRIP_CMD?=	/usr/bin/strip
-SU_CMD?=	/usr/bin/su root -c
+SU_CMD?=	su root -c
 TAIL?=		/usr/bin/tail
 TEST?=		test				# Shell builtin
 TR?=		/usr/bin/tr
 TRUE?=		true				# Shell builtin
-UNAME?=		/usr/bin/uname
+UNAME?=		uname
 .if exists(/usr/bin/unzip)
 UNZIP_CMD?=	/usr/bin/unzip
 .else
@@ -1030,7 +1030,7 @@
 .endif
 WHICH?=		/usr/bin/which
 XARGS?=		/usr/bin/xargs
-YACC?=		/usr/bin/yacc
+YACC?=		freebsd-yacc
 
 # ECHO is defined in /usr/share/mk/sys.mk, which can either be "echo",
 # or "true" if the make flag -s is given.  Use ECHO_CMD where you mean
@@ -1045,7 +1045,7 @@
 
 # Get the architecture
 .if !defined(ARCH)
-ARCH!=	${UNAME} -p
+ARCH!= ${UNAME} -m
 .endif
 
 # Kludge for pre-3.0 systems
@@ -1198,7 +1198,7 @@
 # These need to be absolute since we don't know how deep in the ports
 # tree we are and thus can't go relative.  They can, of course, be overridden
 # by individual Makefiles or local system make configuration.
-PORTSDIR?=		/usr/ports
+PORTSDIR?=		/usr/share/freebsd6-buildutils/ports
 LOCALBASE?=		${DESTDIR}/usr/local
 X11BASE?=		${DESTDIR}/usr/X11R6
 LINUXBASE?=		${DESTDIR}/compat/linux
@@ -1836,15 +1836,7 @@
 # Miscellaneous overridable commands:
 GMAKE?=			gmake
 XMKMF?=			xmkmf -a
-.if exists(/sbin/md5)
-MD5?=			/sbin/md5
-.elif exists(/bin/md5)
-MD5?=			/bin/md5
-.elif exists(/usr/bin/md5)
-MD5?=			/usr/bin/md5
-.else
-MD5?=			md5
-.endif
+MD5?=			md5sum
 .if exists(/sbin/sha256)
 SHA256?=		/sbin/sha256
 .elif exists(${LOCALBASE}/sbin/sha256)
@@ -1969,7 +1961,7 @@
 MTREE_FILE=	/etc/mtree/BSD.local.dist
 .endif
 .endif
-MTREE_CMD?=	/usr/sbin/mtree
+MTREE_CMD?=	/usr/bin/freebsd-mtree
 MTREE_ARGS?=	-U ${MTREE_FOLLOWS_SYMLINKS} -f ${MTREE_FILE} -d -e -p
 
 # Determine whether or not we can use rootly owner/group functions.
diff -ur ../foo.old/ports/Mk/bsd.port.subdir.mk ./ports/Mk/bsd.port.subdir.mk
--- ../foo.old/ports/Mk/bsd.port.subdir.mk	2005-11-08 10:02:51.000000000 +0100
+++ ./ports/Mk/bsd.port.subdir.mk	2005-12-16 23:59:10.000000000 +0100
@@ -48,10 +48,10 @@
 
 .if !defined(NOPRECIOUSMAKEVARS)
 .if !defined(ARCH)
-ARCH!=	/usr/bin/uname -p
+ARCH!=	/bin/uname -m
 .endif
 .if !defined(OSREL)
-OSREL!=	/usr/bin/uname -r | sed -e 's/[-(].*//'
+OSREL!=	/bin/uname -r | sed -e 's/[-(].*//'
 .endif
 .if !defined(OSVERSION)
 .if exists(/sbin/sysctl)
@@ -61,7 +61,7 @@
 .endif
 .endif
 .if !defined(PORTOBJFORMAT)
-PORTOBJFORMAT!= test -x /usr/bin/objformat && /usr/bin/objformat || echo aout
+PORTOBJFORMAT!= which objformat >/dev/null && objformat || echo aout
 .endif
 .endif
 
@@ -73,11 +73,11 @@
 .else
 PKG_INFO?=	/usr/sbin/pkg_info
 .endif
-SED?=		/usr/bin/sed
+SED?=		/bin/sed
 PKGINSTALLVER!=	${PKG_INFO} -P 2>/dev/null | ${SED} -e 's/.*: //'
 
 .if !defined(OPSYS)
-OPSYS!=	/usr/bin/uname -s
+OPSYS!=	/bin/uname -s
 .endif
 
 ECHO_MSG?=	echo
@@ -255,7 +255,7 @@
 .if (${OPSYS} == "NetBSD")
 PORTSDIR ?= /usr/opt
 .else
-PORTSDIR ?= /usr/ports
+PORTSDIR ?= /usr/share/freebsd6-buildutils/ports
 .endif
 TEMPLATES ?= ${PORTSDIR}/Templates
 .if defined(PORTSTOP)
diff -ur ../foo.old/src/share/mk/bsd.port.mk ./src/share/mk/bsd.port.mk
--- ../foo.old/src/share/mk/bsd.port.mk	2004-07-02 22:47:18.000000000 +0200
+++ ./src/share/mk/bsd.port.mk	2005-12-16 23:59:10.000000000 +0100
@@ -1,6 +1,6 @@
 # $FreeBSD: ./src/share/mk/bsd.port.mk,v 1.307 2004/07/02 20:47:18 eik Exp $
 
-PORTSDIR?=	/usr/ports
+PORTSDIR?=	/usr/share/freebsd6-buildutils/ports
 BSDPORTMK?=	${PORTSDIR}/Mk/bsd.port.mk
 
 .include <bsd.own.mk>
diff -ur ../foo.old/src/share/mk/bsd.port.subdir.mk ./src/share/mk/bsd.port.subdir.mk
--- ../foo.old/src/share/mk/bsd.port.subdir.mk	2004-07-02 22:47:18.000000000 +0200
+++ ./src/share/mk/bsd.port.subdir.mk	2005-12-16 23:59:10.000000000 +0100
@@ -1,6 +1,6 @@
 # $FreeBSD: ./src/share/mk/bsd.port.subdir.mk,v 1.31 2004/07/02 20:47:18 eik Exp $
 
-PORTSDIR?=	/usr/ports
+PORTSDIR?=	/usr/share/freebsd6-buildutils/ports
 BSDPORTSUBDIRMK?=	${PORTSDIR}/Mk/bsd.port.subdir.mk
 
 .include "${BSDPORTSUBDIRMK}"
