--- src/usr.bin/make/main.c~	2006-02-13 20:44:23.000000000 +0100
+++ src/usr.bin/make/main.c	2006-02-13 20:46:55.000000000 +0100@
@@ -136,6 +136,8 @@
 time_t		now;		/* Time at start of make */
 struct GNode	*DEFAULT;	/* .DEFAULT node */
 
+#include <bsd/getopt.h>		/* optreset */
+
 /**
  * Exit with usage message.
  */
@@ -358,7 +358,7 @@
 		if ((optind < argc) && strcmp(argv[optind], "--") == 0) {
 			found_dd = TRUE;
 		}
-		if ((c = getopt(argc, argv, OPTFLAGS)) == -1) {
+		if ((c = bsd_getopt(argc, argv, OPTFLAGS)) == -1) {
 			break;
 		}
 		switch(c) {
