diff -ur -x debian a/cddl/compat/opensolaris/misc/mnttab.c b/cddl/compat/opensolaris/misc/mnttab.c
--- a/cddl/compat/opensolaris/misc/mnttab.c	2008-11-17 15:49:29.000000000 -0500
+++ b/cddl/compat/opensolaris/misc/mnttab.c	2010-07-27 20:07:02.656479850 -0400
@@ -89,7 +89,7 @@
 }
 
 int
-getmntany(FILE *fd __unused, struct mnttab *mgetp, struct mnttab *mrefp)
+getmntany(FILE *fd, struct mnttab *mgetp, struct mnttab *mrefp)
 {
 	static struct statfs *sfs = NULL;
 	static char mntopts[MNTMAXSTR];
diff -ur -x debian a/cddl/compat/opensolaris/misc/zmount.c b/cddl/compat/opensolaris/misc/zmount.c
--- a/cddl/compat/opensolaris/misc/zmount.c	2008-11-17 15:49:29.000000000 -0500
+++ b/cddl/compat/opensolaris/misc/zmount.c	2010-07-27 20:07:02.659058030 -0400
@@ -92,6 +92,9 @@
 	iov = NULL;
 	iovlen = 0;
 	build_iovec(&iov, &iovlen, "fstype", fstype, (size_t)-1);
+#ifndef __DECONST
+#define __DECONST(type, var)	((type)(uintptr_t)(const void *)(var))
+#endif
 	build_iovec(&iov, &iovlen, "fspath", __DECONST(char *, dir),
 	    (size_t)-1);
 	build_iovec(&iov, &iovlen, "from", __DECONST(char *, spec), (size_t)-1);
