diff -ur -x debian a/sys/cddl/compat/opensolaris/sys/types.h b/sys/cddl/compat/opensolaris/sys/types.h
--- a/sys/cddl/compat/opensolaris/sys/types.h	2008-11-17 15:49:29.000000000 -0500
+++ b/sys/cddl/compat/opensolaris/sys/types.h	2010-07-27 20:10:36.439355449 -0400
@@ -33,7 +33,7 @@
  * This is a bag of dirty hacks to keep things compiling.
  */
 
-#include <sys/stdint.h>
+#include <stdint.h>
 #include_next <sys/types.h>
 
 #define	MAXNAMELEN	256
diff -ur -x debian a/sys/cddl/contrib/opensolaris/common/avl/avl.c b/sys/cddl/contrib/opensolaris/common/avl/avl.c
--- a/sys/cddl/contrib/opensolaris/common/avl/avl.c	2008-11-17 15:49:29.000000000 -0500
+++ b/sys/cddl/contrib/opensolaris/common/avl/avl.c	2010-07-27 20:10:36.441379999 -0400
@@ -92,7 +92,7 @@
 
 #include <sys/types.h>
 #include <sys/param.h>
-#include <sys/stdint.h>
+#include <stdint.h>
 #include <sys/debug.h>
 #include <sys/avl.h>
 
diff -ur -x debian a/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/sys/zfs_context.h b/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/sys/zfs_context.h
--- a/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/sys/zfs_context.h	2009-08-17 05:03:47.000000000 -0400
+++ b/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/sys/zfs_context.h	2010-07-27 20:10:36.442398029 -0400
@@ -33,7 +33,7 @@
 #endif
 
 #include <sys/param.h>
-#include <sys/stdint.h>
+#include <stdint.h>
 #include <sys/note.h>
 #include <sys/kernel.h>
 #include <sys/debug.h>
