diff -ur -x debian freebsd-libs-8.0.old/cddl/lib/libzfs/Makefile freebsd-libs-8.0/cddl/lib/libzfs/Makefile
--- freebsd-libs-8.0.old/cddl/lib/libzfs/Makefile	2008-11-17 15:49:29.000000000 -0500
+++ freebsd-libs-8.0/cddl/lib/libzfs/Makefile	2010-07-27 20:11:17.801799169 -0400
@@ -47,5 +47,6 @@
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/lib/libnvpair
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/lib/libuutil/common
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/lib/libzfs/common
+CFLAGS+= -I${.CURDIR}/../../../lib/libgeom
 
 .include <bsd.lib.mk>
diff -ur -x debian freebsd-libs-8.0.old/cddl/sbin/zfs/Makefile freebsd-libs-8.0/cddl/sbin/zfs/Makefile
--- freebsd-libs-8.0.old/cddl/sbin/zfs/Makefile	2008-03-27 19:21:25.000000000 -0400
+++ freebsd-libs-8.0/cddl/sbin/zfs/Makefile	2010-07-27 20:11:17.802817408 -0400
@@ -21,7 +21,13 @@
 
 DPADD=	${LIBZFS} ${LIBGEOM} ${LIBBSDXML} ${LIBSBUF} \
 	${LIBM} ${LIBNVPAIR} ${LIBUUTIL} ${LIBUTIL}
-LDADD=	-lzfs -lgeom -lbsdxml -lsbuf \
+LDADD=	-lzfs -lgeom -lexpat -lsbuf \
 	-lm -lnvpair -luutil -lutil
 
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzfs \
+	-L${.CURDIR}/../../../cddl/lib/libnvpair \
+	-L${.CURDIR}/../../../cddl/lib/libuutil \
+	-L${.CURDIR}/../../../lib/libgeom \
+	-L${.CURDIR}/../../../lib/libsbuf
+
 .include <bsd.prog.mk>
diff -ur -x debian freebsd-libs-8.0.old/cddl/sbin/zpool/Makefile freebsd-libs-8.0/cddl/sbin/zpool/Makefile
--- freebsd-libs-8.0.old/cddl/sbin/zpool/Makefile	2008-11-17 15:49:29.000000000 -0500
+++ freebsd-libs-8.0/cddl/sbin/zpool/Makefile	2010-07-27 20:11:17.803821828 -0400
@@ -23,7 +23,12 @@
 
 DPADD=	${LIBAVL} ${LIBZFS} ${LIBGEOM} ${LIBBSDXML} ${LIBSBUF} \
 	${LIBM} ${LIBNVPAIR} ${LIBUUTIL} ${LIBUTIL}
-LDADD=	-lavl -lzfs -lgeom -lbsdxml -lsbuf \
+LDADD=	-lavl -lzfs -lgeom -lexpat -lsbuf \
 	-lm -lnvpair -luutil -lutil
 
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzfs \
+	-L${.CURDIR}/../../../cddl/lib/libnvpair \
+	-L${.CURDIR}/../../../cddl/lib/libuutil
+
+
 .include <bsd.prog.mk>
