--- a/cddl/lib/libzfs/Makefile
+++ b/cddl/lib/libzfs/Makefile
@@ -7,7 +7,10 @@
 
 LIB=	zfs
 DPADD=	${LIBUTIL}
-LDADD=	-lutil
+LDADD=	-lm -lbsd -lgeom -lnvpair -luutil
+
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libnvpair \
+	-L${.CURDIR}/../../../cddl/lib/libuutil \
 
 SRCS=	deviceid.c \
 	fsshare.c \
--- a/cddl/lib/libuutil/Makefile	2010-07-29 03:23:17.000000000 +0200
+++ b/cddl/lib/libuutil/Makefile	2010-07-29 03:24:01.000000000 +0200
@@ -22,4 +22,6 @@
 CFLAGS+= -I${.CURDIR}/../../../cddl/compat/opensolaris/include
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/head
 
+LDADD= -lrt -lbsd -lpthread
+
 .include <bsd.lib.mk>
diff -ur -x debian a/cddl/sbin/zfs/Makefile b/cddl/sbin/zfs/Makefile
--- a/cddl/sbin/zfs/Makefile	2008-03-27 19:21:25.000000000 -0400
+++ b/cddl/sbin/zfs/Makefile	2010-07-27 20:11:17.802817408 -0400
@@ -19,9 +19,12 @@
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/uts/common/fs/zfs
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/uts/common/sys
 
-DPADD=	${LIBZFS} ${LIBGEOM} ${LIBBSDXML} ${LIBSBUF} \
-	${LIBM} ${LIBNVPAIR} ${LIBUUTIL} ${LIBUTIL}
-LDADD=	-lzfs -lgeom -lbsdxml -lsbuf \
-	-lm -lnvpair -luutil -lutil
+DPADD=	${LIBZFS} ${LIBGEOM} \
+	${LIBM} ${LIBNVPAIR} ${LIBUUTIL}
+LDADD=	-lzfs -lgeom -lnvpair -luutil -lbsd
+
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzfs \
+       -L${.CURDIR}/../../../cddl/lib/libnvpair \
+       -L${.CURDIR}/../../../cddl/lib/libuutil 
 
 .include <bsd.prog.mk>
diff -ur -x debian a/cddl/sbin/zpool/Makefile b/cddl/sbin/zpool/Makefile
--- a/cddl/sbin/zpool/Makefile	2008-11-17 15:49:29.000000000 -0500
+++ b/cddl/sbin/zpool/Makefile	2010-07-27 20:11:17.803821828 -0400
@@ -21,9 +21,10 @@
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/uts/common/fs/zfs
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/uts/common/sys
 
-DPADD=	${LIBAVL} ${LIBZFS} ${LIBGEOM} ${LIBBSDXML} ${LIBSBUF} \
-	${LIBM} ${LIBNVPAIR} ${LIBUUTIL} ${LIBUTIL}
-LDADD=	-lavl -lzfs -lgeom -lbsdxml -lsbuf \
-	-lm -lnvpair -luutil -lutil
+DPADD=	${LIBZFS} ${LIBGEOM} ${LIBNVPAIR} ${LIBUUTIL}
+LDADD=	-lzfs -lgeom -lnvpair -luutil -lbsd
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzfs \
+	-L${.CURDIR}/../../../cddl/lib/libnvpair \
+	-L${.CURDIR}/../../../cddl/lib/libuutil
 
 .include <bsd.prog.mk>

Index: zfsutils-8.1-patch/cddl/usr.sbin/zdb/Makefile
===================================================================
--- zfsutils-8.1-patch.orig/cddl/usr.sbin/zdb/Makefile	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/cddl/usr.sbin/zdb/Makefile	2011-04-06 17:17:47.000000000 -0400
@@ -18,7 +18,13 @@
 
 DPADD=	${LIBAVL} ${LIBGEOM} ${LIBM} ${LIBNVPAIR} ${LIBPTHREAD} ${LIBUMEM} \
 	${LIBUUTIL} ${LIBZ} ${LIBZFS} ${LIBZPOOL}
-LDADD=	-lavl -lgeom -lm -lnvpair -lpthread -lumem -luutil -lz -lzfs -lzpool
+LDADD=	-lgeom -lm -lnvpair -lpthread -lumem -luutil -lz -lzfs -lzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libumem
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libnvpair
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libuutil
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzfs
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libavl  ${.CURDIR}/../../../cddl/lib/libavl/libavl.a
 
 CSTD=	c99
 
Index: zfsutils-8.1-patch/cddl/usr.bin/zinject/Makefile
===================================================================
--- zfsutils-8.1-patch.orig/cddl/usr.bin/zinject/Makefile	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/cddl/usr.bin/zinject/Makefile	2011-04-06 17:17:47.000000000 -0400
@@ -17,6 +17,12 @@
 
 DPADD=	${LIBAVL} ${LIBGEOM} ${LIBM} ${LIBNVPAIR} ${LIBUMEM} ${LIBUUTIL} \
 	${LIBZFS} ${LIBZPOOL} ${LIBUUTIL}
-LDADD=	-lavl -lgeom -lm -lnvpair -lumem -luutil -lzfs -lzpool
+LDADD=	-lgeom -lm -lnvpair -lumem -luutil -lzfs -lzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libumem
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libnvpair
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libuutil
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzfs
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libavl  ${.CURDIR}/../../../cddl/lib/libavl/libavl.a
 
 .include <bsd.prog.mk>
Index: zfsutils-8.1-patch/cddl/usr.bin/ztest/Makefile
===================================================================
--- zfsutils-8.1-patch.orig/cddl/usr.bin/ztest/Makefile	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/cddl/usr.bin/ztest/Makefile	2011-04-06 17:21:32.000000000 -0400
@@ -14,7 +14,11 @@
 CFLAGS+= -I${.CURDIR}/../../lib/libumem
 
 DPADD=	${LIBM} ${LIBNVPAIR} ${LIBUMEM} ${LIBZPOOL} \
 	${LIBPTHREAD} ${LIBZ} ${LIBAVL}
-LDADD=	-lm -lnvpair -lumem -lzpool -lpthread -lz -lavl
+LDADD=	-lm -lnvpair -lumem -lzpool -lpthread -lz -lrt -ldl -lbsd
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libumem
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libnvpair
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libavl  ${.CURDIR}/../../../cddl/lib/libavl/libavl.a
 
 CSTD=	c99
 
