--- a/sys/cddl/compat/opensolaris/rpc/xdr.h
+++ b/sys/cddl/compat/opensolaris/rpc/xdr.h
@@ -34,6 +34,26 @@
 
 #ifndef _KERNEL
 
+/* Copied from the FreeBSD version of <rpc/xdr.h> */
+#ifdef __GLIBC__
+/*
+ * These are XDR control operators
+ */
+
+#define        XDR_GET_BYTES_AVAIL     1
+#define        XDR_PEEK                2
+#define        XDR_SKIPBYTES           3
+
+struct xdr_bytesrec {
+	bool_t xc_is_last_record;
+	size_t xc_num_avail;
+};
+
+typedef struct xdr_bytesrec xdr_bytesrec;
+
+#define xdr_control(xdrs, req, op)	XDR_CONTROL(xdrs, req, op)
+#endif
+
 #include <assert.h>
 
 /*
@@ -80,10 +80,16 @@
 }
 
 #undef XDR_CONTROL
+#ifdef __GLIBC__
+#define	XDR_CONTROL(xdrs, req, op)					\
+	(xdrmem_control((xdrs), (req), (op)))
+
+#else
 #define	XDR_CONTROL(xdrs, req, op)					\
 	(((xdrs)->x_ops->x_control == NULL) ?				\
 	    xdrmem_control((xdrs), (req), (op)) :			\
 	    (*(xdrs)->x_ops->x_control)(xdrs, req, op))   
+#endif
 
 #endif	/* !_KERNEL */
 
