Index: zfsutils-8.1-patch/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/arc.c
===================================================================
--- zfsutils-8.1-patch.orig/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/arc.c	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/arc.c	2011-04-06 17:17:47.000000000 -0400
@@ -141,6 +141,9 @@
 extern uint64_t zfs_write_limit_max;
 extern kmutex_t zfs_write_limit_lock;
 
+
+#define __unused __attribute__((unused))
+
 #define	ARC_REDUCE_DNLC_PERCENT	3
 uint_t arc_reduce_dnlc_percent = ARC_REDUCE_DNLC_PERCENT;

