Index: zfsutils-8.1-patch/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/arc.c
===================================================================
--- zfsutils-8.1-patch.orig/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/arc.c	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/sys/cddl/contrib/opensolaris/uts/common/fs/zfs/arc.c	2011-04-06 17:17:47.000000000 -0400
@@ -141,6 +141,9 @@
 extern uint64_t zfs_write_limit_max;
 extern kmutex_t zfs_write_limit_lock;
 
+
+#define __unused __attribute__((unused))
+
 #define	ARC_REDUCE_DNLC_PERCENT	3
 uint_t arc_reduce_dnlc_percent = ARC_REDUCE_DNLC_PERCENT;
 
Index: zfsutils-8.1-patch/cddl/usr.sbin/zdb/Makefile
===================================================================
--- zfsutils-8.1-patch.orig/cddl/usr.sbin/zdb/Makefile	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/cddl/usr.sbin/zdb/Makefile	2011-04-06 17:17:47.000000000 -0400
@@ -18,10 +18,19 @@
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/uts/common/sys
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/head
 CFLAGS+= -I${.CURDIR}/../../lib/libumem
+CFLAGS+= -D'alloca(x)=__builtin_alloca(x)'
+
 
 DPADD=	${LIBAVL} ${LIBGEOM} ${LIBM} ${LIBNVPAIR} ${LIBPTHREAD} ${LIBUMEM} \
 	${LIBUUTIL} ${LIBZ} ${LIBZFS} ${LIBZPOOL}
-LDADD=	-lavl -lgeom -lm -lnvpair -lpthread -lumem -luutil -lz -lzfs -lzpool
+
+LDADD=	-lgeom -lm -lnvpair -lpthread -lumem -luutil -lz -lzfs -lzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libumem
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libnvpair
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libuutil
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzfs
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libavl  ${.CURDIR}/../../../cddl/lib/libavl/libavl.a
 
 CSTD=	c99
 
Index: zfsutils-8.1-patch/cddl/usr.bin/zinject/Makefile
===================================================================
--- zfsutils-8.1-patch.orig/cddl/usr.bin/zinject/Makefile	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/cddl/usr.bin/zinject/Makefile	2011-04-06 17:17:47.000000000 -0400
@@ -17,9 +17,16 @@
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/uts/common
 CFLAGS+= -I${.CURDIR}/../../contrib/opensolaris/head
 CFLAGS+= -I${.CURDIR}/../../lib/libumem
+CFLAGS+= -D'alloca(x)=__builtin_alloca(x)'
 
 DPADD=	${LIBAVL} ${LIBGEOM} ${LIBM} ${LIBNVPAIR} ${LIBUMEM} ${LIBUUTIL} \
 	${LIBZFS} ${LIBZPOOL} ${LIBUUTIL}
-LDADD=	-lavl -lgeom -lm -lnvpair -lumem -luutil -lzfs -lzpool
+LDADD=	-lgeom -lm -lnvpair -lumem -luutil -lzfs -lzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libumem
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libnvpair
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libuutil
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzfs
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libavl  ${.CURDIR}/../../../cddl/lib/libavl/libavl.a
 
 .include <bsd.prog.mk>
Index: zfsutils-8.1-patch/cddl/usr.bin/ztest/Makefile
===================================================================
--- zfsutils-8.1-patch.orig/cddl/usr.bin/ztest/Makefile	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/cddl/usr.bin/ztest/Makefile	2011-04-06 17:21:32.000000000 -0400
@@ -14,10 +14,15 @@
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/uts/common
 CFLAGS+= -I${.CURDIR}/../../contrib/opensolaris/head
 CFLAGS+= -I${.CURDIR}/../../lib/libumem
+CFLAGS+= -D'alloca(x)=__builtin_alloca(x)'
 
 DPADD=	${LIBM} ${LIBNVPAIR} ${LIBUMEM} ${LIBZPOOL} \
 	${LIBPTHREAD} ${LIBZ} ${LIBAVL}
-LDADD=	-lm -lnvpair -lumem -lzpool -lpthread -lz -lavl
+LDADD=	-lm -lnvpair -lumem -lzpool -lpthread -lz -lrt -ldl -lbsd
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libzpool
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libumem
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libnvpair
+LDADD+= -L${.CURDIR}/../../../cddl/lib/libavl  ${.CURDIR}/../../../cddl/lib/libavl/libavl.a
 
 CSTD=	c99
 
Index: zfsutils-8.1-patch/cddl/lib/libzpool/Makefile
===================================================================
--- zfsutils-8.1-patch.orig/cddl/lib/libzpool/Makefile	2011-04-06 17:17:31.000000000 -0400
+++ zfsutils-8.1-patch/cddl/lib/libzpool/Makefile	2011-04-06 17:17:47.000000000 -0400
@@ -51,6 +51,7 @@
 CFLAGS+=	-I${.CURDIR}/../../../lib/libpthread/thread
 CFLAGS+=	-I${.CURDIR}/../../../lib/libpthread/sys
 CFLAGS+=	-I${.CURDIR}/../../../lib/libthr/arch/${MACHINE_ARCH}/include
+CFLAGS+=	-D'alloca(x)=__builtin_alloca(x)'
 
 DPADD=		${LIBPTHREAD} ${LIBZ}
 LDADD=		-lpthread -lz
