*** orig/alpha/machine/types.h	Wed Mar 21 11:50:58 2001
--- import/alpha/machine/types.h	Fri Sep  6 01:36:03 2002
***************
*** 1,3 ****
--- 1,21 ----
+ /* Copyright (C) 2002 Free Software Foundation, Inc.
+    This file is part of the GNU C Library.
+ 
+    The GNU C Library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+ 
+    The GNU C Library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+ 
+    You should have received a copy of the GNU Lesser General Public
+    License along with the GNU C Library; if not, write to the Free
+    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+    02111-1307 USA.  */
+ 
  /* $FreeBSD: src/sys/alpha/include/types.h,v 1.11.2.2 2001/03/21 10:50:58 peter Exp $ */
  /* From: NetBSD: types.h,v 1.8 1997/04/06 08:47:45 cgd Exp */
  
***************
*** 40,45 ****
--- 58,64 ----
  #define	_MACHTYPES_H_
  
  #include <sys/cdefs.h>
+ #include <sys/types.h>
  
  #if !defined(_ANSI_SOURCE) && !defined(_POSIX_SOURCE)
  typedef struct _physadr {
***************
*** 56,63 ****
  typedef	unsigned long	vm_pindex_t;
  typedef	unsigned long	vm_size_t;
  
! typedef __int64_t		register_t;
! typedef __uint64_t		u_register_t;
  
  #ifdef _KERNEL
  typedef	long		intfptr_t;
--- 75,83 ----
  typedef	unsigned long	vm_pindex_t;
  typedef	unsigned long	vm_size_t;
  
! typedef	int64_t		segsz_t;	/* segment size */
! 
! typedef	u_int32_t	fixpt_t;	/* fixed point number */
  
  #ifdef _KERNEL
  typedef	long		intfptr_t;
