diff -ur kfreebsd-headers-4.6/import.old/sys/callout.h kfreebsd-headers-4.6/import/sys/callout.h
--- kfreebsd-headers-4.6/import.old/sys/callout.h	2004-05-31 03:27:21.000000000 +0200
+++ kfreebsd-headers-4.6/import/sys/callout.h	2004-06-03 00:24:33.000000000 +0200
@@ -62,7 +62,7 @@
 
 #include <features.h>
 
-#include <sys/queue.h>
+#include "/usr/src/kfreebsd-headers/sys/queue.h"
 
 SLIST_HEAD(callout_list, callout);
 TAILQ_HEAD(callout_tailq, callout);
diff -ur kfreebsd-headers-4.6/import.old/sys/event.h kfreebsd-headers-4.6/import/sys/event.h
--- kfreebsd-headers-4.6/import.old/sys/event.h	2004-05-31 03:27:26.000000000 +0200
+++ kfreebsd-headers-4.6/import/sys/event.h	2004-06-03 00:26:29.000000000 +0200
@@ -29,6 +29,8 @@
 #ifndef _SYS_EVENT_H_
 #define _SYS_EVENT_H_
 
+#include <stdint.h>
+
 #define EVFILT_READ		(-1)
 #define EVFILT_WRITE		(-2)
 #define EVFILT_AIO		(-3)	/* attached to aio requests */
diff -ur kfreebsd-headers-4.6/import.old/sys/filedesc.h kfreebsd-headers-4.6/import/sys/filedesc.h
--- kfreebsd-headers-4.6/import.old/sys/filedesc.h	2004-05-31 03:27:27.000000000 +0200
+++ kfreebsd-headers-4.6/import/sys/filedesc.h	2004-06-03 00:25:14.000000000 +0200
@@ -56,7 +56,7 @@
 #define _SYS_FILEDESC_H_
 
 #include <sys/types.h>
-#include <sys/queue.h>
+#include "/usr/src/kfreebsd-headers/sys/queue.h"
 
 /*
  * This structure is used for the management of descriptors.  It may be
