Author: aurel32
Description: also build arch-all packages

diff -Nru wanna-build-2005.11.15/buildd wanna-build-2005.11.15+aurel32/buildd
--- wanna-build-2005.11.15/buildd	2005-11-15 09:28:42.000000000 +0100
+++ wanna-build-2005.11.15+aurel32/buildd	2005-11-16 00:43:41.000000000 +0100
@@ -466,7 +466,7 @@
 
 	my @sbuild_args = ( 'nice', '-n', "$conf::nice_level", 'sbuild',
 			    '--batch', "--stats-dir=$main::HOME/stats",
-			    "--dist=$dist" );
+			    "--dist=$dist", "--arch-all");
 	my $sbuild_gb = '--auto-give-back';
 	if ($conf::sshcmd) {
 		$sbuild_gb .= "=";
