*** orig/net/if.h	Sun Feb 10 00:02:39 2002
--- import/net/if.h	Fri Sep  6 01:13:15 2002
***************
*** 1,3 ****
--- 1,22 ----
+ /* net/if.h -- declarations for inquiring about network interfaces
+    Copyright (C) 1997-1999, 2000-2002 Free Software Foundation, Inc.
+    This file is part of the GNU C Library.
+ 
+    The GNU C Library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+ 
+    The GNU C Library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+ 
+    You should have received a copy of the GNU Lesser General Public
+    License along with the GNU C Library; if not, write to the Free
+    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+    02111-1307 USA.  */
+ 
  /*
   * Copyright (c) 1982, 1986, 1989, 1993
   *	The Regents of the University of California.  All rights reserved.
***************
*** 39,44 ****
--- 58,68 ----
  
  #include <sys/queue.h>
  
+ #include <features.h>
+ 
+ #include <sys/types.h>
+ #include <sys/socket.h>
+ 
  /*
   * <net/if.h> does not depend on <sys/time.h> on most other systems.  This
   * helps userland compatability.  (struct timeval ifi_lastchange)
***************
*** 223,228 ****
--- 247,253 ----
  		struct	sockaddr ifru_addr;
  		struct	sockaddr ifru_dstaddr;
  		struct	sockaddr ifru_broadaddr;
+ 		struct	sockaddr ifru_netmask;
  		short	ifru_flags[2];
  		int	ifru_metric;
  		int	ifru_mtu;
***************
*** 234,239 ****
--- 259,265 ----
  #define	ifr_addr	ifr_ifru.ifru_addr	/* address */
  #define	ifr_dstaddr	ifr_ifru.ifru_dstaddr	/* other end of p-to-p link */
  #define	ifr_broadaddr	ifr_ifru.ifru_broadaddr	/* broadcast address */
+ #define	ifr_netmask	ifr_ifru.ifru_netmask	/* interface net mask */
  #define	ifr_flags	ifr_ifru.ifru_flags[0]	/* flags */
  #define	ifr_prevflags	ifr_ifru.ifru_flags[1]	/* flags */
  #define	ifr_metric	ifr_ifru.ifru_metric	/* metric */
