--- perl-5.8.4.orig/ext/DynaLoader/hints/gnukfreebsd.pl
+++ perl-5.8.4/ext/DynaLoader/hints/gnukfreebsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/DynaLoader/hints/gnuknetbsd.pl
+++ perl-5.8.4/ext/DynaLoader/hints/gnuknetbsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/NDBM_File/hints/gnukfreebsd.pl
+++ perl-5.8.4/ext/NDBM_File/hints/gnukfreebsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/NDBM_File/hints/gnuknetbsd.pl
+++ perl-5.8.4/ext/NDBM_File/hints/gnuknetbsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/ODBM_File/hints/gnukfreebsd.pl
+++ perl-5.8.4/ext/ODBM_File/hints/gnukfreebsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/ODBM_File/hints/gnuknetbsd.pl
+++ perl-5.8.4/ext/ODBM_File/hints/gnuknetbsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/POSIX/hints/gnukfreebsd.pl
+++ perl-5.8.4/ext/POSIX/hints/gnukfreebsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/POSIX/hints/gnuknetbsd.pl
+++ perl-5.8.4/ext/POSIX/hints/gnuknetbsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/Storable/hints/gnukfreebsd.pl
+++ perl-5.8.4/ext/Storable/hints/gnukfreebsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/ext/Storable/hints/gnuknetbsd.pl
+++ perl-5.8.4/ext/Storable/hints/gnuknetbsd.pl
@@ -0,0 +1,2 @@
+# just use linux hints
+do './hints/linux.pl';
--- perl-5.8.4.orig/hints/gnukfreebsd.sh
+++ perl-5.8.4/hints/gnukfreebsd.sh
@@ -0,0 +1,10 @@
+#! /bin/sh
+
+# Support for GNU/kFreeBSD (kfreebsd-gnu)
+# (GNU userland/libc with kernel of FreeBSD)
+
+. ./hints/linux.sh
+
+# Configure sets these where $osname = linux
+ccdlflags='-Wl,-E'
+lddlflags='-shared'
--- perl-5.8.4.orig/hints/gnuknetbsd.sh
+++ perl-5.8.4/hints/gnuknetbsd.sh
@@ -0,0 +1,10 @@
+#! /bin/sh
+
+# Support for GNU/kNetBSD (knetbsd-gnu)
+# (GNU userland/libc with kernel of NetBSD)
+
+. ./hints/linux.sh
+
+# Configure sets these where $osname = linux
+ccdlflags='-Wl,-E'
+lddlflags='-shared'
