--- linuxthreads/old_pthread_atfork.c~	2005-12-22 13:26:01.000000000 +0000
+++ linuxthreads/old_pthread_atfork.c	2005-12-22 13:26:01.000000000 +0000
@@ -19,7 +19,7 @@
 
 #include <shlib-compat.h>
 
-#if SHLIB_COMPAT (libpthread, GLIBC_2_0, GLIBC_2_3)
+#if SHLIB_COMPAT (libpthread, GLIBC_2_0, GLIBC_2_3_2)
 # define __pthread_atfork __dyn_pthread_atfork
 # include "pthread_atfork.c"
 # undef __pthread_atfork
