Index: programs/xload/get_load.c
===================================================================
RCS file: /cvs/xorg/xc/programs/xload/get_load.c,v
retrieving revision 1.2
diff -u -r1.2 get_load.c
--- programs/xload/get_load.c	23 Apr 2004 19:54:57 -0000	1.2
+++ programs/xload/get_load.c	18 Jul 2005 19:17:46 -0000
@@ -406,7 +406,7 @@
 }
 #else /* not KVM_ROUTINES */
 
-#ifdef linux
+#if defined(linux) || defined(__FreeBSD_kernel__)
 
 void InitLoadPoint()
 {
Index: util/memleak/Imakefile
===================================================================
RCS file: /cvs/xorg/xc/util/memleak/Imakefile,v
retrieving revision 1.2
diff -u -r1.2 Imakefile
--- util/memleak/Imakefile	23 Apr 2004 19:55:10 -0000	1.2
+++ util/memleak/Imakefile	18 Jul 2005 19:17:46 -0000
@@ -42,7 +42,7 @@
 GRA_SRCS = getreti386.c stackbottom.c
 #endif
 
-#ifdef LinuxArchitecture
+#if defined(LinuxArchitecture) || defined(__GLIBC__)
 #ifdef i386Architecture
 #define HasGetReturnAddress	YES
 #define TopOfStack 0xbffff800
