Index: config/cf/Imake.cf
===================================================================
RCS file: /cvs/xorg/xc/config/cf/Imake.cf,v
retrieving revision 1.9
diff -u -r1.9 Imake.cf
--- config/cf/Imake.cf	21 May 2005 07:46:37 -0000	1.9
+++ config/cf/Imake.cf	18 Jul 2005 19:17:19 -0000
@@ -756,11 +756,13 @@
 #undef __MINGW32__
 #endif /* CYGWIN */
 
-#ifdef linux 
+#if defined(linux) || defined(__GLIBC__)
 # define MacroIncludeFile <linux.cf>
 # define MacroFile linux.cf
 # undef linux
-# define LinuxArchitecture
+# ifdef __linux__
+#  define LinuxArchitecture
+# endif
 # ifdef i386
 #  ifndef i386Architecture
 #   define i386Architecture
@@ -985,7 +987,7 @@
 #   define AMD64Architecture
 #  endif
 # endif
-#endif /* linux */
+#endif /* linux || __GLIBC__ */
 
 #if (defined(__Lynx__) || defined(Lynx)) && (defined(i386) || defined(__i386__) || defined(__x86__) || defined(__powerpc__) || defined(__sparc__) || defined(sparc))
 # define MacroIncludeFile <lynx.cf>
@@ -1065,9 +1067,7 @@
 #endif /* minix */
 
 /* Systems with GNU libc and userland */
-#if defined(__GNU__) || defined(__GNU_userland__) || \
-  defined(__GNU_KFreeBSD__) || \
-  defined(__GNU_KNetBSD__) || defined(__GNU_KOpenBSD__)
+#if defined(__GNU__)
 #define GNUArchitecture
 #define MacroIncludeFile <gnu.cf>
 #define MacroFile gnu.cf
Index: config/cf/linux.cf
===================================================================
RCS file: /cvs/xorg/xc/config/cf/linux.cf,v
retrieving revision 1.29
diff -u -r1.29 linux.cf
--- config/cf/linux.cf	3 Jul 2005 00:06:01 -0000	1.29
+++ config/cf/linux.cf	18 Jul 2005 19:17:20 -0000
@@ -164,17 +164,18 @@
 #   define HasXdmAuth			YES
 #   define HasLatex			YES
 /* at present, glide2 only works on i386 */
-#   ifdef i386Architecture
+#   if defined(__linux__) && defined(i386Architecture)
 #    define HasGlide2			YES
 #    define Glide2IncDir		/usr/include/glide
-#   endif /* i386Architecture */
+#   endif /* __linux__ && i386Architecture */
 /* at present, glide3 only works on alpha, amd64, i386 and ia64 */
-#   if defined(i386Architecture) || defined(AlphaArchitecture) || \
-       defined(ia64Architecture) || defined(AMD64Architecture)
+#   if defined(__linux__) && (defined(i386Architecture) || \
+       defined(AlphaArchitecture) || defined(ia64Architecture) || \
+       defined(AMD64Architecture))
 #    define HasGlide3			YES
 #    define Glide3IncDir		/usr/include/glide3
-#   endif /* i386Architecture || AlphaArchitecture || ia64Architecture ||
-             AMD64Architecture */
+#   endif /* __linux__ && (i386Architecture || AlphaArchitecture || \
+             ia64Architecture || AMD64Architecture) */
 /* extended instruction set support */
 #   ifdef i386Architecture
 #   define HasX86Support		YES
@@ -278,22 +279,23 @@
 
 /* <linux/input.h> support mainly for USB support */
 #ifndef HasLinuxInput
-# if OSMajorVersion > 2 || (OSMajorVersion == 2 && OSMinorVersion >= 4)
+# if defined(__linux__) && (OSMajorVersion > 2 || (OSMajorVersion == 2 && OSMinorVersion >= 4))
 #  define HasLinuxInput YES
 # else
 #  define HasLinuxInput NO
 # endif
 #endif
 
-/* <linux/joystick.h> appeared in 2.1.45 (officially) */
 #ifndef JoystickSupport
 # define JoystickSupport	NO	/* joystick driver is broken */
 #endif
+
+/* <linux/joystick.h> appeared in 2.1.45 (officially) */
 #ifndef HasLinuxSupport
-# if JoystickSupport || \
+# if defined(__linux__) && (JoystickSupport || \
      (OSMajorVersion > 2) || \
      ((OSMajorVersion == 2) && (OSMinorVersion > 1)) || \
-     ((OSMajorVersion == 2) && (OSMinorVersion == 1) && (OSTeenyVersion >= 45))
+     ((OSMajorVersion == 2) && (OSMinorVersion == 1) && (OSTeenyVersion >= 45)))
 #  define HasLinuxJoystick	YES
 # else
 #  define HasLinuxJoystick	NO
@@ -318,9 +320,9 @@
 #endif
 
 #ifndef XF86INT10_BUILD
-# if defined (i386Architecture)
+# if defined(__linux__) && defined(i386Architecture)
 #  define XF86INT10_BUILD	X86VM
-# elif defined (AMD64Architecture)
+# elif defined(__linux__) && defined(AMD64Architecture)
 #  define XF86INT10_BUILD	X86EMU_OS
 # else
 #  define XF86INT10_BUILD	X86EMU_GENERIC
@@ -353,7 +355,9 @@
 #  elif defined(HPArchitecture)
 #    define BuildXF86DRI	YES
 #  elif defined(i386Architecture)
-#    if OSMajorVersion > 2 || (OSMajorVersion == 2 && OSMinorVersion >= 2)
+#    if (defined(__linux__) && (OSMajorVersion > 2 || (OSMajorVersion == 2 && \
+        OSMinorVersion >= 2))) || \
+        defined(__FreeBSD_kernel__)
 #      define BuildXF86DRI	YES
 #    else
 #      define BuildXF86DRI	NO
@@ -458,7 +462,7 @@
 # define BuildXF86RushExt	YES
 #endif
 
-#ifndef LinuxFBDevSupport
+#if defined(__linux__) && !defined(LinuxFBDevSupport)
 # define LinuxFBDevSupport	YES
 #endif
 
@@ -788,7 +792,11 @@
 #endif
 
 #ifndef XF86OSCardDrivers
-# define XF86OSCardDrivers	fbdev v4l sisusb
+# ifdef __linux__
+#  define XF86OSCardDrivers	fbdev v4l sisusb
+# else
+#  define XF86OSCardDrivers	/**/
+# endif
 #endif
 
 #ifndef OSXInputDrivers
@@ -965,7 +973,11 @@
 #endif /* AMD64Architecture */
 
 #ifndef StandardDefines
-# define StandardDefines	-Dlinux LinuxMachineDefines LinuxSourceDefines
+# ifdef __linux__
+#  define StandardDefines	-Dlinux LinuxMachineDefines LinuxSourceDefines
+# else
+#  define StandardDefines	LinuxMachineDefines LinuxSourceDefines
+# endif
 #endif
 
 #if UseElfFormat
Index: config/cf/xfree86.cf
===================================================================
RCS file: /cvs/xorg/xc/config/cf/xfree86.cf,v
retrieving revision 1.13
diff -u -r1.13 xfree86.cf
--- config/cf/xfree86.cf	5 Jul 2005 19:06:50 -0000	1.13
+++ config/cf/xfree86.cf	18 Jul 2005 19:17:21 -0000
@@ -1747,7 +1747,8 @@
 # if SystemV || SystemV4 || \
     (defined(LinuxArchitecture) && !defined(Mc68020Architecture)) || \
     defined(i386BsdArchitecture) || defined(LynxOSArchitecture) || \
-    defined(OS2Architecture) || defined(GNUMachArchitecture)
+    defined(OS2Architecture) || defined(GNUMachArchitecture) || \
+    (defined(KFreeBSDArchitecture) && defined(i386Architecture))
 #   define BuildScanpci		YES
 # else
 #   define BuildScanpci		NO
@@ -1853,9 +1854,9 @@
 #endif
 
 #ifndef XFree86ConsoleDefines
-# if defined(i386BsdArchitecture) || defined(AlphaBsdArchitecture)
+# if defined(i386BsdArchitecture) || defined(AlphaBsdArchitecture) || (defined(KFreeBSDArchitecture) && defined(i386Architecture))
 #  define XFree86ConsoleDefines	-DPCCONS_SUPPORT -DSYSCONS_SUPPORT -DPCVT_SUPPORT
-# elif defined(FreeBSDArchitecture)
+# elif defined(KFreeBSDArchitecture)
 #  define XFree86ConsoleDefines -DSYSCONS_SUPPORT
 # else
 #  define XFree86ConsoleDefines	/**/
Index: config/cf/xorg.cf
===================================================================
RCS file: /cvs/xorg/xc/config/cf/xorg.cf,v
retrieving revision 1.50
diff -u -r1.50 xorg.cf
--- config/cf/xorg.cf	6 Jul 2005 10:13:48 -0000	1.50
+++ config/cf/xorg.cf	18 Jul 2005 19:17:21 -0000
@@ -1285,7 +1285,8 @@
 # if SystemV || SystemV4 || \
     (defined(LinuxArchitecture) && !defined(Mc68020Architecture)) || \
     defined(i386BsdArchitecture) || defined(LynxOSArchitecture) || \
-    defined(OS2Architecture) || defined(GNUMachArchitecture)
+    defined(OS2Architecture) || defined(GNUMachArchitecture) || \
+    (defined(KFreeBSDArchitecture) && defined(i386Architecture))
 #   define BuildScanpci		YES
 # else
 #   define BuildScanpci		NO
@@ -1392,9 +1393,9 @@
 #endif
 
 #ifndef XFree86ConsoleDefines
-# if defined(i386BsdArchitecture) || defined(AlphaBsdArchitecture)
+# if defined(i386BsdArchitecture) || defined(AlphaBsdArchitecture) || (defined(KFreeBSDArchitecture) && defined(i386Architecture))
 #  define XFree86ConsoleDefines	-DPCCONS_SUPPORT -DSYSCONS_SUPPORT -DPCVT_SUPPORT
-# elif defined(FreeBSDArchitecture)
+# elif defined(KFreeBSDArchitecture)
 #  define XFree86ConsoleDefines -DSYSCONS_SUPPORT
 # else
 #  define XFree86ConsoleDefines	/**/
