diff -Nurd src/usr.bin/lex/flex.skl src/usr.bin/lex/flex.skl
--- src/usr.bin/lex/flex.skl	2005-12-23 15:35:47.000000000 +0100
+++ src/usr.bin/lex/flex.skl	2005-12-23 15:35:57.000000000 +0100
@@ -263,7 +263,7 @@
 %*
 
 static void *yy_flex_alloc YY_PROTO(( yy_size_t ));
-static void *yy_flex_realloc YY_PROTO(( void *, yy_size_t )) __unused;
+static void *yy_flex_realloc YY_PROTO(( void *, yy_size_t ));
 static void yy_flex_free YY_PROTO(( void * ));
 
 #define yy_new_buffer yy_create_buffer
diff -Nurd src/usr.bin/make/dir.c src/usr.bin/make/dir.c
--- src/usr.bin/make/dir.c	2005-12-23 15:35:49.000000000 +0100
+++ src/usr.bin/make/dir.c	2005-12-23 15:35:57.000000000 +0100
@@ -539,7 +539,7 @@
  *-----------------------------------------------------------------------
  */
 static int
-DirPrintWord(void *word, void *dummy __unused)
+DirPrintWord(void *word, void *dummy)
 {
     DEBUGF(DIR, ("%s ", (char *) word));
 
@@ -1223,7 +1223,7 @@
 }
 
 static int
-DirPrintDir (void *p, void *dummy __unused)
+DirPrintDir (void *p, void *dummy)
 {
     printf ("%s ", ((Path *) p)->name);
 
diff -Nurd src/usr.bin/make/main.c src/usr.bin/make/main.c
--- src/usr.bin/make/main.c	2005-12-23 15:35:56.000000000 +0100
+++ src/usr.bin/make/main.c	2005-12-23 15:35:57.000000000 +0100
@@ -422,7 +422,7 @@
 }
 
 static void
-catch_child(int sig __unused)
+catch_child(int sig)
 {
 }
 
@@ -938,7 +938,7 @@
  *	lots
  */
 static Boolean
-ReadMakefile(void *p, void *q __unused)
+ReadMakefile(void *p, void *q)
 {
 	char *fname;			/* makefile to read */
 	FILE *stream;
diff -Nurd src/usr.bin/make/parse.c src/usr.bin/make/parse.c
--- src/usr.bin/make/parse.c	2005-12-23 15:35:49.000000000 +0100
+++ src/usr.bin/make/parse.c	2005-12-23 15:35:57.000000000 +0100
@@ -575,7 +575,7 @@
  *-----------------------------------------------------------------------
  */
 static int
-ParseFindMain(void *gnp, void *dummy __unused)
+ParseFindMain(void *gnp, void *dummy)
 {
     GNode   	  *gn = (GNode *) gnp;
     if ((gn->type & (OP_NOTMAIN|OP_USE|OP_EXEC|OP_TRANSFORM)) == 0) {
@@ -621,7 +621,7 @@
  *-----------------------------------------------------------------------
  */
 static int
-ParseClearPath(void *path, void *dummy __unused)
+ParseClearPath(void *path, void *dummy)
 {
     Dir_ClearPath((Lst) path);
     return (0);
diff -Nurd src/usr.bin/make/suff.c src/usr.bin/make/suff.c
--- src/usr.bin/make/suff.c	2005-12-23 15:35:50.000000000 +0100
+++ src/usr.bin/make/suff.c	2005-12-23 15:35:57.000000000 +0100
@@ -625,7 +625,7 @@
  *-----------------------------------------------------------------------
  */
 int
-Suff_EndTransform(void *gnp, void *dummy __unused)
+Suff_EndTransform(void *gnp, void *dummy)
 {
     GNode *gn = (GNode *) gnp;
 
@@ -2277,14 +2277,14 @@
 /********************* DEBUGGING FUNCTIONS **********************/
 
 static int
-SuffPrintName(void *s, void *dummy __unused)
+SuffPrintName(void *s, void *dummy)
 {
     printf ("`%s' ", ((Suff *) s)->name);
     return (0);
 }
 
 static int
-SuffPrintSuff (void *sp, void *dummy __unused)
+SuffPrintSuff (void *sp, void *dummy)
 {
     Suff    *s = (Suff *) sp;
     int	    flags;
@@ -2328,7 +2328,7 @@
 }
 
 static int
-SuffPrintTrans (void *tp, void *dummy __unused)
+SuffPrintTrans (void *tp, void *dummy)
 {
     GNode   *t = (GNode *) tp;
 
diff -Nurd src/usr.bin/make/targ.c src/usr.bin/make/targ.c
--- src/usr.bin/make/targ.c	2005-12-23 15:35:50.000000000 +0100
+++ src/usr.bin/make/targ.c	2005-12-23 15:35:57.000000000 +0100
@@ -418,7 +418,7 @@
 
 
 int
-Targ_PrintCmd (void *cmd, void *dummy __unused)
+Targ_PrintCmd (void *cmd, void *dummy)
 {
     printf ("\t%s\n", (char *) cmd);
     return (0);
@@ -585,7 +585,7 @@
  *-----------------------------------------------------------------------
  */
 static int
-TargPrintOnlySrc(void *gnp, void *dummy __unused)
+TargPrintOnlySrc(void *gnp, void *dummy)
 {
     GNode   	  *gn = (GNode *) gnp;
     if (OP_NOP(gn->type))
diff -Nurd src/usr.bin/make/util.c src/usr.bin/make/util.c
--- src/usr.bin/make/util.c	2005-12-23 15:35:50.000000000 +0100
+++ src/usr.bin/make/util.c	2005-12-23 15:35:57.000000000 +0100
@@ -277,7 +277,7 @@
  * 	Print the address of a node, used as an interative function.
  */
 int
-PrintAddr(void *a, void *b __unused)
+PrintAddr(void *a, void *b)
 {
     printf("%p ", a);
     return 0;
diff -Nurd src/usr.bin/make/var.c src/usr.bin/make/var.c
--- src/usr.bin/make/var.c	2005-12-23 15:35:50.000000000 +0100
+++ src/usr.bin/make/var.c	2005-12-23 15:35:57.000000000 +0100
@@ -1944,7 +1944,7 @@
 
 /****************** PRINT DEBUGGING INFO *****************/
 static int
-VarPrintVar (void *vp, void *dummy __unused)
+VarPrintVar (void *vp, void *dummy)
 {
     Var    *v = (Var *) vp;
     printf ("%-16s = %s\n", v->name, (char *) Buf_GetAll(v->val, (int *)NULL));
diff -Nurd src/usr.bin/make/var_modify.c src/usr.bin/make/var_modify.c
--- src/usr.bin/make/var_modify.c	2005-12-23 15:35:50.000000000 +0100
+++ src/usr.bin/make/var_modify.c	2005-12-23 15:35:57.000000000 +0100
@@ -65,7 +65,7 @@
  *-----------------------------------------------------------------------
  */
 Boolean
-VarHead (const char *word, Boolean addSpace, Buffer buf, void *dummy __unused)
+VarHead (const char *word, Boolean addSpace, Buffer buf, void *dummy)
 {
     char *slash;
     char *buffer;
@@ -110,7 +110,7 @@
  *-----------------------------------------------------------------------
  */
 Boolean
-VarTail (const char *word, Boolean addSpace, Buffer buf, void *dummy __unused)
+VarTail (const char *word, Boolean addSpace, Buffer buf, void *dummy)
 {
     const char *slash;
 
@@ -142,7 +142,7 @@
  *-----------------------------------------------------------------------
  */
 Boolean
-VarSuffix (const char *word, Boolean addSpace, Buffer buf, void *dummy __unused)
+VarSuffix (const char *word, Boolean addSpace, Buffer buf, void *dummy)
 {
     const char *dot;
 
@@ -173,7 +173,7 @@
  *-----------------------------------------------------------------------
  */
 Boolean
-VarRoot (const char *word, Boolean addSpace, Buffer buf, void *dummy __unused)
+VarRoot (const char *word, Boolean addSpace, Buffer buf, void *dummy)
 {
     char *buffer;
     char *dot;
diff -Nurd src/usr.bin/yacc/main.c src/usr.bin/yacc/main.c
--- src/usr.bin/yacc/main.c	2005-12-23 15:35:56.000000000 +0100
+++ src/usr.bin/yacc/main.c	2005-12-23 15:35:57.000000000 +0100
@@ -130,7 +130,7 @@
 
 static void
 onintr(signo)
-	int signo __unused;
+	int signo;
 {
     sigdie = 1;
     done(1);
diff -Nurd src/usr.bin/yacc/skeleton.c src/usr.bin/yacc/skeleton.c
--- src/usr.bin/yacc/skeleton.c	2005-12-23 15:35:52.000000000 +0100
+++ src/usr.bin/yacc/skeleton.c	2005-12-23 15:35:57.000000000 +0100
@@ -60,9 +60,6 @@
 {
     "#include <stdlib.h>",
     "#ifndef lint",
-    "#ifdef __unused",
-    "__unused",
-    "#endif",
     "static char const ",
     "yyrcsid[] = \"$FreeBSD: src/usr.bin/yacc/skeleton.c,v 1.37 2003/02/12 18:03:55 davidc Exp $\";",
     "#endif",
diff -Nurd src/usr.sbin/mtree/compare.c src/usr.sbin/mtree/compare.c
--- src/usr.sbin/mtree/compare.c	2005-12-23 15:35:56.000000000 +0100
+++ src/usr.sbin/mtree/compare.c	2005-12-23 15:35:57.000000000 +0100
@@ -68,7 +68,7 @@
 	}
 
 int
-compare(char *name __unused, NODE *s, FTSENT *p)
+compare(char *name, NODE *s, FTSENT *p)
 {
 	struct timeval tv[2];
 	uint32_t val;
