diff -ur src.old/usr.bin/cksum/cksum.c src/usr.bin/cksum/cksum.c
--- src.old/usr.bin/cksum/cksum.c	2003-03-14 00:32:28.000000000 +0100
+++ src/usr.bin/cksum/cksum.c	2004-11-26 00:00:46.000000000 +0100
@@ -47,7 +47,9 @@
 #endif /* not lint */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/cksum/cksum.c,v 1.17 2003/03/13 23:32:28 robert Exp $");
+#endif
 
 #include <sys/types.h>
 
diff -ur src.old/usr.bin/cksum/crc.c src/usr.bin/cksum/crc.c
--- src.old/usr.bin/cksum/crc.c	2003-03-14 00:32:28.000000000 +0100
+++ src/usr.bin/cksum/crc.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,9 @@
 #endif
 #endif /* not lint */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/cksum/crc.c,v 1.8 2003/03/13 23:32:28 robert Exp $");
+#endif
 
 #include <sys/types.h>
 
diff -ur src.old/usr.bin/cksum/crc32.c src/usr.bin/cksum/crc32.c
--- src.old/usr.bin/cksum/crc32.c	2003-03-14 00:32:28.000000000 +0100
+++ src/usr.bin/cksum/crc32.c	2004-11-26 00:00:46.000000000 +0100
@@ -12,7 +12,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/cksum/crc32.c,v 1.9 2003/03/13 23:32:28 robert Exp $");
+#endif
 
 #include <sys/types.h>
 
diff -ur src.old/usr.bin/cksum/print.c src/usr.bin/cksum/print.c
--- src.old/usr.bin/cksum/print.c	2003-03-14 00:32:28.000000000 +0100
+++ src/usr.bin/cksum/print.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 #endif /* not lint */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/cksum/print.c,v 1.7 2003/03/13 23:32:28 robert Exp $");
+#endif
 
 #include <sys/types.h>
 
diff -ur src.old/usr.bin/cksum/sum1.c src/usr.bin/cksum/sum1.c
--- src.old/usr.bin/cksum/sum1.c	2003-03-14 00:32:28.000000000 +0100
+++ src/usr.bin/cksum/sum1.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 #endif /* not lint */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/cksum/sum1.c,v 1.8 2003/03/13 23:32:28 robert Exp $");
+#endif
 
 #include <sys/types.h>
 
diff -ur src.old/usr.bin/cksum/sum2.c src/usr.bin/cksum/sum2.c
--- src.old/usr.bin/cksum/sum2.c	2003-03-14 00:32:28.000000000 +0100
+++ src/usr.bin/cksum/sum2.c	2004-11-26 00:00:46.000000000 +0100
@@ -37,7 +37,9 @@
 #endif
 #endif /* not lint */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/cksum/sum2.c,v 1.8 2003/03/13 23:32:28 robert Exp $");
+#endif
 
 #include <sys/types.h>
 
diff -ur src.old/usr.bin/file2c/file2c.c src/usr.bin/file2c/file2c.c
--- src.old/usr.bin/file2c/file2c.c	2002-06-30 07:21:18.000000000 +0200
+++ src/usr.bin/file2c/file2c.c	2004-11-26 00:00:46.000000000 +0100
@@ -8,7 +8,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/file2c/file2c.c,v 1.7 2002/06/30 05:21:18 obrien Exp $");
 
 #include <stdio.h>
 
diff -ur src.old/usr.bin/lex/ccl.c src/usr.bin/lex/ccl.c
--- src.old/usr.bin/lex/ccl.c	2002-06-30 07:25:03.000000000 +0200
+++ src/usr.bin/lex/ccl.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/ccl.c,v 2.9 93/09/16 20:32:14 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/ccl.c,v 1.6 2002/06/30 05:25:03 obrien Exp $");
+#endif
 
 #include "flexdef.h"
 
diff -ur src.old/usr.bin/lex/dfa.c src/usr.bin/lex/dfa.c
--- src.old/usr.bin/lex/dfa.c	2002-06-30 07:25:03.000000000 +0200
+++ src/usr.bin/lex/dfa.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/dfa.c,v 2.26 95/04/20 13:53:14 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/dfa.c,v 1.6 2002/06/30 05:25:03 obrien Exp $");
+#endif
 
 #include "flexdef.h"
 
diff -ur src.old/usr.bin/lex/ecs.c src/usr.bin/lex/ecs.c
--- src.old/usr.bin/lex/ecs.c	2002-06-30 07:25:03.000000000 +0200
+++ src/usr.bin/lex/ecs.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/ecs.c,v 2.9 93/12/07 10:18:20 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/ecs.c,v 1.6 2002/06/30 05:25:03 obrien Exp $");
+#endif
 
 #include "flexdef.h"
 
diff -ur src.old/usr.bin/lex/flex.skl src/usr.bin/lex/flex.skl
--- src.old/usr.bin/lex/flex.skl	2004-01-06 20:03:44.000000000 +0100
+++ src/usr.bin/lex/flex.skl	2004-11-26 00:00:46.000000000 +0100
@@ -263,7 +263,7 @@
 %*
 
 static void *yy_flex_alloc YY_PROTO(( yy_size_t ));
-static void *yy_flex_realloc YY_PROTO(( void *, yy_size_t )) __unused;
+static void *yy_flex_realloc YY_PROTO(( void *, yy_size_t ));
 static void yy_flex_free YY_PROTO(( void * ));
 
 #define yy_new_buffer yy_create_buffer
diff -ur src.old/usr.bin/lex/gen.c src/usr.bin/lex/gen.c
--- src.old/usr.bin/lex/gen.c	2004-01-06 19:54:55.000000000 +0100
+++ src/usr.bin/lex/gen.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/gen.c,v 2.56 96/05/25 20:43:38 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/gen.c,v 1.7 2004/01/06 18:54:55 nectar Exp $");
+#endif
 
 #include "flexdef.h"
 
diff -ur src.old/usr.bin/lex/initscan.c src/usr.bin/lex/initscan.c
--- src.old/usr.bin/lex/initscan.c	2002-08-25 15:22:44.000000000 +0200
+++ src/usr.bin/lex/initscan.c	2004-11-26 00:00:46.000000000 +0100
@@ -5,7 +5,9 @@
  * $Header: /home/daffy/u0/vern/flex/RCS/flex.skl,v 2.91 96/09/10 16:58:48 vern Exp $
  */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/initscan.c,v 1.9 2002/08/25 13:22:44 charnier Exp $");
+#endif
 
 #define FLEX_SCANNER
 #define YY_FLEX_MAJOR_VERSION 2
diff -ur src.old/usr.bin/lex/main.c src/usr.bin/lex/main.c
--- src.old/usr.bin/lex/main.c	2002-06-30 07:25:04.000000000 +0200
+++ src/usr.bin/lex/main.c	2004-11-26 00:00:46.000000000 +0100
@@ -34,7 +34,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/main.c,v 2.64 96/05/25 20:42:42 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/main.c,v 1.8 2002/06/30 05:25:04 obrien Exp $");
+#endif
 
 
 #include "flexdef.h"
diff -ur src.old/usr.bin/lex/misc.c src/usr.bin/lex/misc.c
--- src.old/usr.bin/lex/misc.c	2002-06-30 07:25:04.000000000 +0200
+++ src/usr.bin/lex/misc.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/misc.c,v 2.47 95/04/28 11:39:39 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/misc.c,v 1.6 2002/06/30 05:25:04 obrien Exp $");
+#endif
 
 #include "flexdef.h"
 
diff -ur src.old/usr.bin/lex/nfa.c src/usr.bin/lex/nfa.c
--- src.old/usr.bin/lex/nfa.c	2002-06-30 07:25:04.000000000 +0200
+++ src/usr.bin/lex/nfa.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/nfa.c,v 2.17 95/03/04 16:11:42 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/nfa.c,v 1.6 2002/06/30 05:25:04 obrien Exp $");
+#endif
 
 #include "flexdef.h"
 
diff -ur src.old/usr.bin/lex/sym.c src/usr.bin/lex/sym.c
--- src.old/usr.bin/lex/sym.c	2002-06-30 07:25:04.000000000 +0200
+++ src/usr.bin/lex/sym.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/sym.c,v 2.19 95/03/04 16:11:04 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/sym.c,v 1.6 2002/06/30 05:25:04 obrien Exp $");
+#endif
 
 #include "flexdef.h"
 
diff -ur src.old/usr.bin/lex/tblcmp.c src/usr.bin/lex/tblcmp.c
--- src.old/usr.bin/lex/tblcmp.c	2002-06-30 07:25:04.000000000 +0200
+++ src/usr.bin/lex/tblcmp.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/tblcmp.c,v 2.11 94/11/05 17:08:28 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/tblcmp.c,v 1.6 2002/06/30 05:25:04 obrien Exp $");
+#endif
 
 #include "flexdef.h"
 
diff -ur src.old/usr.bin/lex/yylex.c src/usr.bin/lex/yylex.c
--- src.old/usr.bin/lex/yylex.c	2002-06-30 07:25:04.000000000 +0200
+++ src/usr.bin/lex/yylex.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/yylex.c,v 2.13 95/03/04 16:10:41 vern Exp $ */
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/lex/yylex.c,v 1.6 2002/06/30 05:25:04 obrien Exp $");
+#endif
 
 #include <ctype.h>
 #include "flexdef.h"
diff -ur src.old/usr.bin/make/arch.c src/usr.bin/make/arch.c
--- src.old/usr.bin/make/arch.c	2002-10-24 01:16:42.000000000 +0200
+++ src/usr.bin/make/arch.c	2004-11-26 00:00:46.000000000 +0100
@@ -39,7 +39,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/arch.c,v 1.33 2002/10/23 23:16:42 jmallett Exp $");
+#endif
 
 /*-
  * arch.c --
diff -ur src.old/usr.bin/make/buf.c src/usr.bin/make/buf.c
--- src.old/usr.bin/make/buf.c	2002-10-09 05:42:09.000000000 +0200
+++ src/usr.bin/make/buf.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/buf.c,v 1.18 2002/10/09 03:42:09 jmallett Exp $");
+#endif
 
 /*-
  * buf.c --
diff -ur src.old/usr.bin/make/compat.c src/usr.bin/make/compat.c
--- src.old/usr.bin/make/compat.c	2004-07-29 16:29:23.000000000 +0200
+++ src/usr.bin/make/compat.c	2004-11-26 00:01:02.000000000 +0100
@@ -40,7 +40,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/make/compat.c,v 1.37 2004/07/29 14:29:23 harti Exp $");
 
 /*-
  * compat.c --
diff -ur src.old/usr.bin/make/cond.c src/usr.bin/make/cond.c
--- src.old/usr.bin/make/cond.c	2004-07-20 09:42:06.000000000 +0200
+++ src/usr.bin/make/cond.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/make/cond.c,v 1.28 2004/07/20 07:42:06 harti Exp $");
 
 /*-
  * cond.c --
diff -ur src.old/usr.bin/make/dir.c src/usr.bin/make/dir.c
--- src.old/usr.bin/make/dir.c	2003-10-02 23:30:30.000000000 +0200
+++ src/usr.bin/make/dir.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/dir.c,v 1.32 2003/10/02 21:30:30 ru Exp $");
+#endif
 
 /*-
  * dir.c --
@@ -540,7 +542,7 @@
  *-----------------------------------------------------------------------
  */
 static int
-DirPrintWord(void *word, void *dummy __unused)
+DirPrintWord(void *word, void *dummy)
 {
     DEBUGF(DIR, ("%s ", (char *) word));
 
@@ -1224,7 +1226,7 @@
 }
 
 static int
-DirPrintDir (void *p, void *dummy __unused)
+DirPrintDir (void *p, void *dummy)
 {
     printf ("%s ", ((Path *) p)->name);
 
diff -ur src.old/usr.bin/make/for.c src/usr.bin/make/for.c
--- src.old/usr.bin/make/for.c	2004-03-10 22:51:06.000000000 +0100
+++ src/usr.bin/make/for.c	2004-11-26 00:00:46.000000000 +0100
@@ -37,7 +37,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/for.c,v 1.21 2004/03/10 21:51:06 ru Exp $");
+#endif
 
 /*-
  * for.c --
diff -ur src.old/usr.bin/make/hash.c src/usr.bin/make/hash.c
--- src.old/usr.bin/make/hash.c	2002-10-09 05:42:10.000000000 +0200
+++ src/usr.bin/make/hash.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/hash.c,v 1.18 2002/10/09 03:42:10 jmallett Exp $");
+#endif
 
 /* hash.c --
  *
diff -ur src.old/usr.bin/make/job.c src/usr.bin/make/job.c
--- src.old/usr.bin/make/job.c	2004-07-29 16:29:23.000000000 +0200
+++ src/usr.bin/make/job.c	2004-11-26 00:01:16.000000000 +0100
@@ -40,7 +40,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/make/job.c,v 1.50 2004/07/29 14:29:23 harti Exp $");
 
 #ifndef OLD_JOKE
 #define	OLD_JOKE 0
diff -ur src.old/usr.bin/make/lst.lib/lstAppend.c src/usr.bin/make/lst.lib/lstAppend.c
--- src.old/usr.bin/make/lst.lib/lstAppend.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstAppend.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstAppend.c,v 1.10 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstAtEnd.c src/usr.bin/make/lst.lib/lstAtEnd.c
--- src.old/usr.bin/make/lst.lib/lstAtEnd.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstAtEnd.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstAtEnd.c,v 1.9 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstAtFront.c src/usr.bin/make/lst.lib/lstAtFront.c
--- src.old/usr.bin/make/lst.lib/lstAtFront.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstAtFront.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstAtFront.c,v 1.9 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstClose.c src/usr.bin/make/lst.lib/lstClose.c
--- src.old/usr.bin/make/lst.lib/lstClose.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstClose.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstClose.c,v 1.8 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstConcat.c src/usr.bin/make/lst.lib/lstConcat.c
--- src.old/usr.bin/make/lst.lib/lstConcat.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstConcat.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstConcat.c,v 1.11 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstDatum.c src/usr.bin/make/lst.lib/lstDatum.c
--- src.old/usr.bin/make/lst.lib/lstDatum.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstDatum.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstDatum.c,v 1.9 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstDeQueue.c src/usr.bin/make/lst.lib/lstDeQueue.c
--- src.old/usr.bin/make/lst.lib/lstDeQueue.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstDeQueue.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstDeQueue.c,v 1.10 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstDestroy.c src/usr.bin/make/lst.lib/lstDestroy.c
--- src.old/usr.bin/make/lst.lib/lstDestroy.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstDestroy.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstDestroy.c,v 1.12 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstDupl.c src/usr.bin/make/lst.lib/lstDupl.c
--- src.old/usr.bin/make/lst.lib/lstDupl.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstDupl.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstDupl.c,v 1.12 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstEnQueue.c src/usr.bin/make/lst.lib/lstEnQueue.c
--- src.old/usr.bin/make/lst.lib/lstEnQueue.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstEnQueue.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstEnQueue.c,v 1.9 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstFind.c src/usr.bin/make/lst.lib/lstFind.c
--- src.old/usr.bin/make/lst.lib/lstFind.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstFind.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstFind.c,v 1.11 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstFindFrom.c src/usr.bin/make/lst.lib/lstFindFrom.c
--- src.old/usr.bin/make/lst.lib/lstFindFrom.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstFindFrom.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstFindFrom.c,v 1.12 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstFirst.c src/usr.bin/make/lst.lib/lstFirst.c
--- src.old/usr.bin/make/lst.lib/lstFirst.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstFirst.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstFirst.c,v 1.8 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstForEach.c src/usr.bin/make/lst.lib/lstForEach.c
--- src.old/usr.bin/make/lst.lib/lstForEach.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstForEach.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstForEach.c,v 1.10 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstForEachFrom.c src/usr.bin/make/lst.lib/lstForEachFrom.c
--- src.old/usr.bin/make/lst.lib/lstForEachFrom.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstForEachFrom.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstForEachFrom.c,v 1.12 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstInit.c src/usr.bin/make/lst.lib/lstInit.c
--- src.old/usr.bin/make/lst.lib/lstInit.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstInit.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstInit.c,v 1.9 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstInsert.c src/usr.bin/make/lst.lib/lstInsert.c
--- src.old/usr.bin/make/lst.lib/lstInsert.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstInsert.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstInsert.c,v 1.10 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstIsAtEnd.c src/usr.bin/make/lst.lib/lstIsAtEnd.c
--- src.old/usr.bin/make/lst.lib/lstIsAtEnd.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstIsAtEnd.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstIsAtEnd.c,v 1.7 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstIsEmpty.c src/usr.bin/make/lst.lib/lstIsEmpty.c
--- src.old/usr.bin/make/lst.lib/lstIsEmpty.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstIsEmpty.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstIsEmpty.c,v 1.8 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstLast.c src/usr.bin/make/lst.lib/lstLast.c
--- src.old/usr.bin/make/lst.lib/lstLast.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstLast.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstLast.c,v 1.8 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstMember.c src/usr.bin/make/lst.lib/lstMember.c
--- src.old/usr.bin/make/lst.lib/lstMember.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstMember.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstMember.c,v 1.10 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstNext.c src/usr.bin/make/lst.lib/lstNext.c
--- src.old/usr.bin/make/lst.lib/lstNext.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstNext.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstNext.c,v 1.9 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstOpen.c src/usr.bin/make/lst.lib/lstOpen.c
--- src.old/usr.bin/make/lst.lib/lstOpen.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstOpen.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstOpen.c,v 1.8 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstRemove.c src/usr.bin/make/lst.lib/lstRemove.c
--- src.old/usr.bin/make/lst.lib/lstRemove.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstRemove.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstRemove.c,v 1.10 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstReplace.c src/usr.bin/make/lst.lib/lstReplace.c
--- src.old/usr.bin/make/lst.lib/lstReplace.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstReplace.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstReplace.c,v 1.9 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/lst.lib/lstSucc.c src/usr.bin/make/lst.lib/lstSucc.c
--- src.old/usr.bin/make/lst.lib/lstSucc.c	2002-10-09 04:00:22.000000000 +0200
+++ src/usr.bin/make/lst.lib/lstSucc.c	2004-11-26 00:00:46.000000000 +0100
@@ -38,7 +38,9 @@
 
 #ifndef lint
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/lst.lib/lstSucc.c,v 1.8 2002/10/09 02:00:22 jmallett Exp $");
+#endif
 #endif /* not lint */
 
 /*-
diff -ur src.old/usr.bin/make/main.c src/usr.bin/make/main.c
--- src.old/usr.bin/make/main.c	2004-08-12 13:49:55.000000000 +0200
+++ src/usr.bin/make/main.c	2004-11-26 00:01:32.000000000 +0100
@@ -46,7 +46,6 @@
 #endif
 #endif /* not lint */
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/make/main.c,v 1.91 2004/08/12 11:49:55 harti Exp $");
 
 /*-
  * main.c --
@@ -425,7 +424,7 @@
 }
 
 static void
-catch_child(int sig __unused)
+catch_child(int sig)
 {
 }
 
@@ -935,7 +934,7 @@
  *	lots
  */
 static Boolean
-ReadMakefile(void *p, void *q __unused)
+ReadMakefile(void *p, void *q)
 {
 	char *fname;			/* makefile to read */
 	FILE *stream;
diff -ur src.old/usr.bin/make/make.c src/usr.bin/make/make.c
--- src.old/usr.bin/make/make.c	2002-10-24 01:16:43.000000000 +0200
+++ src/usr.bin/make/make.c	2004-11-26 00:00:46.000000000 +0100
@@ -39,7 +39,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/make.c,v 1.24 2002/10/23 23:16:43 jmallett Exp $");
+#endif
 
 /*-
  * make.c --
diff -ur src.old/usr.bin/make/parse.c src/usr.bin/make/parse.c
--- src.old/usr.bin/make/parse.c	2004-07-22 13:12:01.000000000 +0200
+++ src/usr.bin/make/parse.c	2004-11-26 00:00:46.000000000 +0100
@@ -39,7 +39,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/make/parse.c,v 1.57 2004/07/22 11:12:01 harti Exp $");
 
 /*-
  * parse.c --
@@ -576,7 +575,7 @@
  *-----------------------------------------------------------------------
  */
 static int
-ParseFindMain(void *gnp, void *dummy __unused)
+ParseFindMain(void *gnp, void *dummy)
 {
     GNode   	  *gn = (GNode *) gnp;
     if ((gn->type & (OP_NOTMAIN|OP_USE|OP_EXEC|OP_TRANSFORM)) == 0) {
@@ -622,7 +621,7 @@
  *-----------------------------------------------------------------------
  */
 static int
-ParseClearPath(void *path, void *dummy __unused)
+ParseClearPath(void *path, void *dummy)
 {
     Dir_ClearPath((Lst) path);
     return (0);
diff -ur src.old/usr.bin/make/str.c src/usr.bin/make/str.c
--- src.old/usr.bin/make/str.c	2004-01-25 13:28:36.000000000 +0100
+++ src/usr.bin/make/str.c	2004-11-26 00:00:46.000000000 +0100
@@ -39,7 +39,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/str.c,v 1.29 2004/01/25 12:28:36 ru Exp $");
+#endif
 
 #include "make.h"
 
diff -ur src.old/usr.bin/make/suff.c src/usr.bin/make/suff.c
--- src.old/usr.bin/make/suff.c	2004-04-12 22:05:11.000000000 +0200
+++ src/usr.bin/make/suff.c	2004-11-26 00:00:46.000000000 +0100
@@ -39,7 +39,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/make/suff.c,v 1.29 2004/04/12 20:05:11 ru Exp $");
 
 /*-
  * suff.c --
@@ -626,7 +625,7 @@
  *-----------------------------------------------------------------------
  */
 int
-Suff_EndTransform(void *gnp, void *dummy __unused)
+Suff_EndTransform(void *gnp, void *dummy)
 {
     GNode *gn = (GNode *) gnp;
 
@@ -2278,14 +2277,14 @@
 /********************* DEBUGGING FUNCTIONS **********************/
 
 static int
-SuffPrintName(void *s, void *dummy __unused)
+SuffPrintName(void *s, void *dummy)
 {
     printf ("`%s' ", ((Suff *) s)->name);
     return (0);
 }
 
 static int
-SuffPrintSuff (void *sp, void *dummy __unused)
+SuffPrintSuff (void *sp, void *dummy)
 {
     Suff    *s = (Suff *) sp;
     int	    flags;
@@ -2329,7 +2328,7 @@
 }
 
 static int
-SuffPrintTrans (void *tp, void *dummy __unused)
+SuffPrintTrans (void *tp, void *dummy)
 {
     GNode   *t = (GNode *) tp;
 
diff -ur src.old/usr.bin/make/targ.c src/usr.bin/make/targ.c
--- src.old/usr.bin/make/targ.c	2002-10-24 01:16:43.000000000 +0200
+++ src/usr.bin/make/targ.c	2004-11-26 00:00:46.000000000 +0100
@@ -39,7 +39,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/targ.c,v 1.26 2002/10/23 23:16:43 jmallett Exp $");
+#endif
 
 /*-
  * targ.c --
@@ -419,7 +421,7 @@
 
 
 int
-Targ_PrintCmd (void *cmd, void *dummy __unused)
+Targ_PrintCmd (void *cmd, void *dummy)
 {
     printf ("\t%s\n", (char *) cmd);
     return (0);
@@ -586,7 +588,7 @@
  *-----------------------------------------------------------------------
  */
 static int
-TargPrintOnlySrc(void *gnp, void *dummy __unused)
+TargPrintOnlySrc(void *gnp, void *dummy)
 {
     GNode   	  *gn = (GNode *) gnp;
     if (OP_NOP(gn->type))
diff -ur src.old/usr.bin/make/util.c src/usr.bin/make/util.c
--- src.old/usr.bin/make/util.c	2002-10-10 21:27:48.000000000 +0200
+++ src/usr.bin/make/util.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/util.c,v 1.12 2002/10/10 19:27:48 jmallett Exp $");
+#endif
 
 /*-
  * util.c --
@@ -278,7 +280,7 @@
  * 	Print the address of a node, used as an interative function.
  */
 int
-PrintAddr(void *a, void *b __unused)
+PrintAddr(void *a, void *b)
 {
     printf("%p ", a);
     return 0;
diff -ur src.old/usr.bin/make/var.c src/usr.bin/make/var.c
--- src.old/usr.bin/make/var.c	2004-08-12 13:49:55.000000000 +0200
+++ src/usr.bin/make/var.c	2004-11-26 00:00:46.000000000 +0100
@@ -39,7 +39,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/make/var.c,v 1.48 2004/08/12 11:49:55 harti Exp $");
 
 /*-
  * var.c --
@@ -1945,7 +1944,7 @@
 
 /****************** PRINT DEBUGGING INFO *****************/
 static int
-VarPrintVar (void *vp, void *dummy __unused)
+VarPrintVar (void *vp, void *dummy)
 {
     Var    *v = (Var *) vp;
     printf ("%-16s = %s\n", v->name, (char *) Buf_GetAll(v->val, (int *)NULL));
diff -ur src.old/usr.bin/make/var_modify.c src/usr.bin/make/var_modify.c
--- src.old/usr.bin/make/var_modify.c	2002-10-29 13:11:56.000000000 +0100
+++ src/usr.bin/make/var_modify.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/make/var_modify.c,v 1.2 2002/10/29 12:11:56 jmallett Exp $");
+#endif
 
 #include    <ctype.h>
 #include    <sys/types.h>
@@ -66,7 +68,7 @@
  *-----------------------------------------------------------------------
  */
 Boolean
-VarHead (const char *word, Boolean addSpace, Buffer buf, void *dummy __unused)
+VarHead (const char *word, Boolean addSpace, Buffer buf, void *dummy)
 {
     char *slash;
     char *buffer;
@@ -111,7 +113,7 @@
  *-----------------------------------------------------------------------
  */
 Boolean
-VarTail (const char *word, Boolean addSpace, Buffer buf, void *dummy __unused)
+VarTail (const char *word, Boolean addSpace, Buffer buf, void *dummy)
 {
     const char *slash;
 
@@ -143,7 +145,7 @@
  *-----------------------------------------------------------------------
  */
 Boolean
-VarSuffix (const char *word, Boolean addSpace, Buffer buf, void *dummy __unused)
+VarSuffix (const char *word, Boolean addSpace, Buffer buf, void *dummy)
 {
     const char *dot;
 
@@ -174,7 +176,7 @@
  *-----------------------------------------------------------------------
  */
 Boolean
-VarRoot (const char *word, Boolean addSpace, Buffer buf, void *dummy __unused)
+VarRoot (const char *word, Boolean addSpace, Buffer buf, void *dummy)
 {
     char *buffer;
     char *dot;
diff -ur src.old/usr.bin/yacc/closure.c src/usr.bin/yacc/closure.c
--- src.old/usr.bin/yacc/closure.c	2002-04-09 13:39:05.000000000 +0200
+++ src/usr.bin/yacc/closure.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/closure.c,v 1.12 2002/04/09 11:39:05 ru Exp $");
+#endif
 
 #include <stdlib.h>
 #include "defs.h"
diff -ur src.old/usr.bin/yacc/error.c src/usr.bin/yacc/error.c
--- src.old/usr.bin/yacc/error.c	2002-04-09 13:39:05.000000000 +0200
+++ src/usr.bin/yacc/error.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,9 @@
 #endif
 #endif
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/error.c,v 1.15 2002/04/09 11:39:05 ru Exp $");
+#endif
 
 /* routines for printing error messages  */
 
diff -ur src.old/usr.bin/yacc/lalr.c src/usr.bin/yacc/lalr.c
--- src.old/usr.bin/yacc/lalr.c	2002-06-11 13:27:20.000000000 +0200
+++ src/usr.bin/yacc/lalr.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/lalr.c,v 1.15 2002/06/11 11:27:20 robert Exp $");
+#endif
 
 #include <limits.h>
 #include <stdlib.h>
diff -ur src.old/usr.bin/yacc/lr0.c src/usr.bin/yacc/lr0.c
--- src.old/usr.bin/yacc/lr0.c	2002-06-11 13:27:20.000000000 +0200
+++ src/usr.bin/yacc/lr0.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/lr0.c,v 1.13 2002/06/11 11:27:20 robert Exp $");
+#endif
 
 #include <limits.h>
 #include <stdlib.h>
diff -ur src.old/usr.bin/yacc/main.c src/usr.bin/yacc/main.c
--- src.old/usr.bin/yacc/main.c	2004-03-05 03:05:56.000000000 +0100
+++ src/usr.bin/yacc/main.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/main.c,v 1.22 2004/03/05 02:05:56 wes Exp $");
+#endif
 
 #include <paths.h>
 #include <signal.h>
@@ -128,7 +130,7 @@
 
 static void
 onintr(signo)
-	int signo __unused;
+	int signo;
 {
     sigdie = 1;
     done(1);
diff -ur src.old/usr.bin/yacc/mkpar.c src/usr.bin/yacc/mkpar.c
--- src.old/usr.bin/yacc/mkpar.c	2002-04-09 13:39:05.000000000 +0200
+++ src/usr.bin/yacc/mkpar.c	2004-11-26 00:00:46.000000000 +0100
@@ -40,7 +40,9 @@
 #endif
 #endif
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/mkpar.c,v 1.19 2002/04/09 11:39:05 ru Exp $");
+#endif
 
 #include <stdlib.h>
 #include "defs.h"
diff -ur src.old/usr.bin/yacc/output.c src/usr.bin/yacc/output.c
--- src.old/usr.bin/yacc/output.c	2002-06-11 13:27:20.000000000 +0200
+++ src/usr.bin/yacc/output.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/output.c,v 1.28 2002/06/11 11:27:20 robert Exp $");
+#endif
 
 #include <limits.h>
 #include <stdlib.h>
diff -ur src.old/usr.bin/yacc/reader.c src/usr.bin/yacc/reader.c
--- src.old/usr.bin/yacc/reader.c	2002-08-25 15:23:09.000000000 +0200
+++ src/usr.bin/yacc/reader.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/reader.c,v 1.19 2002/08/25 13:23:09 charnier Exp $");
+#endif
 
 #include <limits.h>
 #include <stdlib.h>
diff -ur src.old/usr.bin/yacc/skeleton.c src/usr.bin/yacc/skeleton.c
--- src.old/usr.bin/yacc/skeleton.c	2003-02-12 19:03:55.000000000 +0100
+++ src/usr.bin/yacc/skeleton.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/skeleton.c,v 1.37 2003/02/12 18:03:55 davidc Exp $");
+#endif
 
 #include "defs.h"
 
@@ -61,9 +63,6 @@
 {
     "#include <stdlib.h>",
     "#ifndef lint",
-    "#ifdef __unused",
-    "__unused",
-    "#endif",
     "static char const ",
     "yyrcsid[] = \"$FreeBSD: src/usr.bin/yacc/skeleton.c,v 1.37 2003/02/12 18:03:55 davidc Exp $\";",
     "#endif",
diff -ur src.old/usr.bin/yacc/symtab.c src/usr.bin/yacc/symtab.c
--- src.old/usr.bin/yacc/symtab.c	2002-04-09 13:39:05.000000000 +0200
+++ src/usr.bin/yacc/symtab.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/symtab.c,v 1.12 2002/04/09 11:39:05 ru Exp $");
+#endif
 
 #include <stdlib.h>
 #include <string.h>
diff -ur src.old/usr.bin/yacc/verbose.c src/usr.bin/yacc/verbose.c
--- src.old/usr.bin/yacc/verbose.c	2002-04-09 13:39:05.000000000 +0200
+++ src/usr.bin/yacc/verbose.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/verbose.c,v 1.13 2002/04/09 11:39:05 ru Exp $");
+#endif
 
 #include <stdlib.h>
 #include "defs.h"
diff -ur src.old/usr.bin/yacc/warshall.c src/usr.bin/yacc/warshall.c
--- src.old/usr.bin/yacc/warshall.c	2002-04-09 13:39:05.000000000 +0200
+++ src/usr.bin/yacc/warshall.c	2004-11-26 00:00:46.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.bin/yacc/warshall.c,v 1.12 2002/04/09 11:39:05 ru Exp $");
+#endif
 
 #include "defs.h"
 
diff -ur src.old/usr.sbin/mtree/compare.c src/usr.sbin/mtree/compare.c
--- src.old/usr.sbin/mtree/compare.c	2004-01-11 20:25:56.000000000 +0100
+++ src/usr.sbin/mtree/compare.c	2004-11-26 00:00:46.000000000 +0100
@@ -33,7 +33,9 @@
 #endif /* not lint */
 #endif
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.sbin/mtree/compare.c,v 1.33 2004/01/11 19:25:56 phk Exp $");
+#endif
 
 #include <sys/param.h>
 #include <sys/stat.h>
@@ -69,7 +71,7 @@
 	}
 
 int
-compare(char *name __unused, NODE *s, FTSENT *p)
+compare(char *name, NODE *s, FTSENT *p)
 {
 	struct timeval tv[2];
 	uint32_t val;
diff -ur src.old/usr.sbin/mtree/create.c src/usr.sbin/mtree/create.c
--- src.old/usr.sbin/mtree/create.c	2004-01-11 20:38:48.000000000 +0100
+++ src/usr.sbin/mtree/create.c	2004-11-26 00:00:46.000000000 +0100
@@ -33,7 +33,9 @@
 #endif /* not lint */
 #endif
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.sbin/mtree/create.c,v 1.36 2004/01/11 19:38:48 phk Exp $");
+#endif
 
 #include <sys/param.h>
 #include <sys/stat.h>
diff -ur src.old/usr.sbin/mtree/excludes.c src/usr.sbin/mtree/excludes.c
--- src.old/usr.sbin/mtree/excludes.c	2003-10-21 10:27:05.000000000 +0200
+++ src/usr.sbin/mtree/excludes.c	2004-11-26 00:00:46.000000000 +0100
@@ -28,7 +28,9 @@
  */
 
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.sbin/mtree/excludes.c,v 1.8 2003/10/21 08:27:05 phk Exp $");
+#endif
 
 #include <sys/types.h>
 #include <sys/time.h>		/* XXX for mtree.h */
diff -ur src.old/usr.sbin/mtree/misc.c src/usr.sbin/mtree/misc.c
--- src.old/usr.sbin/mtree/misc.c	2004-01-11 20:25:56.000000000 +0100
+++ src/usr.sbin/mtree/misc.c	2004-11-26 00:00:46.000000000 +0100
@@ -33,7 +33,9 @@
 #endif /*not lint */
 #endif
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.sbin/mtree/misc.c,v 1.15 2004/01/11 19:25:56 phk Exp $");
+#endif
 
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur src.old/usr.sbin/mtree/mtree.c src/usr.sbin/mtree/mtree.c
--- src.old/usr.sbin/mtree/mtree.c	2004-06-04 21:29:28.000000000 +0200
+++ src/usr.sbin/mtree/mtree.c	2004-11-26 00:00:46.000000000 +0100
@@ -39,7 +39,6 @@
 #endif /* not lint */
 #endif
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.sbin/mtree/mtree.c,v 1.29 2004/06/04 19:29:28 ru Exp $");
 
 #include <sys/param.h>
 #include <sys/stat.h>
diff -ur src.old/usr.sbin/mtree/spec.c src/usr.sbin/mtree/spec.c
--- src.old/usr.sbin/mtree/spec.c	2003-11-19 16:28:21.000000000 +0100
+++ src/usr.sbin/mtree/spec.c	2004-11-26 00:00:46.000000000 +0100
@@ -33,7 +33,9 @@
 #endif /* not lint */
 #endif
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.sbin/mtree/spec.c,v 1.21 2003/11/19 15:28:21 phk Exp $");
+#endif
 
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur src.old/usr.sbin/mtree/specspec.c src/usr.sbin/mtree/specspec.c
--- src.old/usr.sbin/mtree/specspec.c	2004-08-07 12:04:29.000000000 +0200
+++ src/usr.sbin/mtree/specspec.c	2004-11-26 00:00:46.000000000 +0100
@@ -25,7 +25,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.sbin/mtree/specspec.c,v 1.5 2004/08/07 10:04:29 simokawa Exp $");
 
 #include <err.h>
 #include <grp.h>
diff -ur src.old/usr.sbin/mtree/verify.c src/usr.sbin/mtree/verify.c
--- src.old/usr.sbin/mtree/verify.c	2004-01-11 20:25:56.000000000 +0100
+++ src/usr.sbin/mtree/verify.c	2004-11-26 00:00:46.000000000 +0100
@@ -33,7 +33,9 @@
 #endif /* not lint */
 #endif
 #include <sys/cdefs.h>
+#ifdef __FBSDID
 __FBSDID("$FreeBSD: src/usr.sbin/mtree/verify.c,v 1.23 2004/01/11 19:25:56 phk Exp $");
+#endif
 
 #include <sys/param.h>
 #include <sys/stat.h>
--- src/usr.bin/brandelf/brandelf.c~	2002-06-02 22:05:57.000000000 +0200
+++ src/usr.bin/brandelf/brandelf.c	2004-12-27 21:45:33.000000000 +0100
@@ -28,7 +28,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/usr.bin/brandelf/brandelf.c,v 1.23 2002/06/02 20:05:57 schweikh Exp $");
 
 #include <sys/types.h>
 #include <sys/elf_common.h>
