diff -Nurd src/usr.sbin/btxld/btxld.c src/usr.sbin/btxld/btxld.c
--- src/usr.sbin/btxld/btxld.c	2005-12-23 15:35:53.000000000 +0100
+++ src/usr.sbin/btxld/btxld.c	2005-12-23 15:35:58.000000000 +0100
@@ -30,7 +30,6 @@
 #endif /* not lint */
 
 #include <sys/param.h>
-#include <sys/endian.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
 
diff -Nurd src/usr.sbin/btxld/elfh.c src/usr.sbin/btxld/elfh.c
--- src/usr.sbin/btxld/elfh.c	2005-12-23 15:35:58.000000000 +0100
+++ src/usr.sbin/btxld/elfh.c	2005-12-23 15:35:58.000000000 +0100
@@ -27,7 +27,6 @@
  */
 
 #include <sys/types.h>
-#include <sys/endian.h>
 
 #include <stddef.h>
 #include <stdint.h>
diff -Nurd src/usr.sbin/kgzip/elfhdr.c src/usr.sbin/kgzip/elfhdr.c
--- src/usr.sbin/kgzip/elfhdr.c	2005-12-23 15:35:54.000000000 +0100
+++ src/usr.sbin/kgzip/elfhdr.c	2005-12-23 15:35:58.000000000 +0100
@@ -26,7 +26,6 @@
  * $FreeBSD: src/usr.sbin/kgzip/elfhdr.c,v 1.4 2004/06/22 17:05:39 obrien Exp $
  */
 
-#include <sys/endian.h>
 #include <stddef.h>
 #include "elfhdr.h"
 
diff -Nurd src/usr.sbin/kgzip/kgzld.c src/usr.sbin/kgzip/kgzld.c
--- src/usr.sbin/kgzip/kgzld.c	2005-12-23 15:35:54.000000000 +0100
+++ src/usr.sbin/kgzip/kgzld.c	2005-12-23 15:35:58.000000000 +0100
@@ -27,7 +27,6 @@
  */
 
 #include <sys/types.h>
-#include <sys/endian.h>
 #include <sys/wait.h>
 
 #include <err.h>
