diff -Nurd src/usr.bin/awk/Makefile src/usr.bin/awk/Makefile
--- src/usr.bin/awk/Makefile	2005-12-23 15:35:46.000000000 +0100
+++ src/usr.bin/awk/Makefile	2005-12-23 15:35:58.000000000 +0100
@@ -11,7 +11,7 @@
 DPADD=	${LIBM}
 LDADD=	-lm
 
-LINKS=	${BINDIR}/nawk ${BINDIR}/awk
+SIMLINKS=	${BINDIR}/nawk ${BINDIR}/awk
 MLINKS=	nawk.1 awk.1
 
 CLEANFILES= maketab proctab.c ytab.h
diff -Nurd src/usr.bin/cksum/Makefile src/usr.bin/cksum/Makefile
--- src/usr.bin/cksum/Makefile	2005-12-23 15:35:46.000000000 +0100
+++ src/usr.bin/cksum/Makefile	2005-12-23 15:35:58.000000000 +0100
@@ -3,7 +3,7 @@
 
 PROG=	cksum
 SRCS=	cksum.c crc.c print.c sum1.c sum2.c crc32.c
-LINKS=	${BINDIR}/cksum ${BINDIR}/sum
+SIMLINKS=	${BINDIR}/cksum ${BINDIR}/sum
 MLINKS=	cksum.1 sum.1
 
 .include <bsd.prog.mk>
diff -Nurd src/usr.bin/lex/Makefile src/usr.bin/lex/Makefile
--- src/usr.bin/lex/Makefile	2005-12-23 15:35:47.000000000 +0100
+++ src/usr.bin/lex/Makefile	2005-12-23 15:35:58.000000000 +0100
@@ -10,9 +10,9 @@
 #
 
 PROG=		lex
-LINKS+=	${BINDIR}/lex ${BINDIR}/lex++
-LINKS+=	${BINDIR}/lex ${BINDIR}/flex
-LINKS+=	${BINDIR}/lex ${BINDIR}/flex++
+SIMLINKS+=	${BINDIR}/lex ${BINDIR}/lex++
+SIMLINKS+=	${BINDIR}/lex ${BINDIR}/flex
+SIMLINKS+=	${BINDIR}/lex ${BINDIR}/flex++
 
 SRCS=		scan.c ccl.c dfa.c ecs.c gen.c main.c misc.c nfa.c parse.y \
 		skel.c sym.c tblcmp.c yylex.c
diff -Nurd src/usr.bin/yacc/Makefile src/usr.bin/yacc/Makefile
--- src/usr.bin/yacc/Makefile	2005-12-23 15:35:52.000000000 +0100
+++ src/usr.bin/yacc/Makefile	2005-12-23 15:35:58.000000000 +0100
@@ -6,7 +6,7 @@
 	skeleton.c symtab.c verbose.c warshall.c
 SCRIPTS=yyfix.sh
 MAN=	yacc.1 yyfix.1
-LINKS=	${BINDIR}/yacc ${BINDIR}/byacc
+SIMLINKS=	${BINDIR}/yacc ${BINDIR}/byacc
 MLINKS=	yacc.1 byacc.1
 
 .include <bsd.prog.mk>
