diff -Nurd src/usr.bin/make/job.c src/usr.bin/make/job.c
--- src/usr.bin/make/job.c	2005-12-23 15:35:49.000000000 +0100
+++ src/usr.bin/make/job.c	2005-12-23 15:35:59.000000000 +0100
@@ -228,7 +228,7 @@
                	*shellName;	      	      	  /* last component of shell */
 
 
-static int  	maxJobs;    	/* The most children we can run at once */
+int  	maxJobs;    	/* The most children we can run at once */
 static int  	maxLocal;    	/* The most local ones we can have */
 STATIC int     	nJobs;	    	/* The number of children currently running */
 STATIC int	nLocal;    	/* The number of local children */
