diff -ur ports.old/Mk/bsd.port.mk ports/Mk/bsd.port.mk
--- ports.old/Mk/bsd.port.mk	2004-10-07 19:59:18.000000000 +0200
+++ ports/Mk/bsd.port.mk	2004-12-28 00:47:51.000000000 +0100
@@ -440,7 +440,7 @@
 # MASTERDIR		- Where the port finds patches, package files, etc.  Define
 #				  this is you have two or more ports that share most of the
 #				  files (default: ${.CURDIR}).
-# PORTSDIR		- The root of the ports tree. (default: /usr/ports)
+# PORTSDIR		- The root of the ports tree. (default: /usr/share/freebsd5-buildutils/ports)
 # DISTDIR 		- Where to search for and store copies of original sources
 #				  (default: ${PORTSDIR}/distfiles).
 # PACKAGES		- A top level directory where all packages go (rather than
@@ -1002,7 +1002,7 @@
 # These need to be absolute since we don't know how deep in the ports
 # tree we are and thus can't go relative.  They can, of course, be overridden
 # by individual Makefiles or local system make configuration.
-PORTSDIR?=		/usr/ports
+PORTSDIR?=		/usr/share/freebsd5-buildutils/ports
 LOCALBASE?=		${DESTDIR}/usr/local
 X11BASE?=		${DESTDIR}/usr/X11R6
 LINUXBASE?=		${DESTDIR}/compat/linux
diff -ur ports.old/Mk/bsd.port.subdir.mk ports/Mk/bsd.port.subdir.mk
--- ports.old/Mk/bsd.port.subdir.mk	2004-07-14 10:18:16.000000000 +0200
+++ ports/Mk/bsd.port.subdir.mk	2004-12-28 00:48:04.000000000 +0100
@@ -248,7 +248,7 @@
 .if (${OPSYS} == "NetBSD")
 PORTSDIR ?= /usr/opt
 .else
-PORTSDIR ?= /usr/ports
+PORTSDIR ?= /usr/share/freebsd5-buildutils/ports
 .endif
 TEMPLATES ?= ${PORTSDIR}/Templates
 .if defined(PORTSTOP)
diff -ur src.old/share/mk/bsd.port.mk src.old/share/mk/bsd.port.mk
--- src.old/share/mk/bsd.port.mk	2004-07-02 22:47:18.000000000 +0200
+++ src/share/mk/bsd.port.mk	2004-12-28 00:47:40.000000000 +0100
@@ -1,6 +1,6 @@
 # $FreeBSD: src/share/mk/bsd.port.mk,v 1.307 2004/07/02 20:47:18 eik Exp $
 
-PORTSDIR?=	/usr/ports
+PORTSDIR?=	/usr/share/freebsd5-buildutils/ports
 BSDPORTMK?=	${PORTSDIR}/Mk/bsd.port.mk
 
 .include <bsd.own.mk>
diff -ur src.old/share/mk/bsd.port.subdir.mk src/share/mk/bsd.port.subdir.mk
--- src.old/share/mk/bsd.port.subdir.mk	2004-07-02 22:47:18.000000000 +0200
+++ src/share/mk/bsd.port.subdir.mk	2004-12-28 00:47:40.000000000 +0100
@@ -1,6 +1,6 @@
 # $FreeBSD: src/share/mk/bsd.port.subdir.mk,v 1.31 2004/07/02 20:47:18 eik Exp $
 
-PORTSDIR?=	/usr/ports
+PORTSDIR?=	/usr/share/freebsd5-buildutils/ports
 BSDPORTSUBDIRMK?=	${PORTSDIR}/Mk/bsd.port.subdir.mk
 
 .include "${BSDPORTSUBDIRMK}"
