Index: sys/kern/kern_fork.c
===================================================================
--- sys/kern/kern_fork.c.orig
+++ sys/kern/kern_fork.c
@@ -485,7 +485,11 @@
 		p2->p_sigacts = newsigacts;
 	}
 	if (flags & RFLINUXTHPN) 
-	        p2->p_sigparent = SIGUSR1;
+	{
+		int sig;
+		sig = RFTHPNSIGNUM(flags);
+	        p2->p_sigparent = sig ? sig : SIGUSR1;
+	}
 	else
 	        p2->p_sigparent = SIGCHLD;
 
Index: sys/sys/unistd.h
===================================================================
--- sys/sys/unistd.h.orig
+++ sys/sys/unistd.h
@@ -181,6 +181,10 @@
 #define	RFHIGHPID	(1<<18)	/* use a pid higher then 10 (idleproc) */
 #define	RFPPWAIT	(1<<31)	/* parent sleeps until child exits (vfork) */
 #define	RFKERNELONLY	(RFSTOPPED | RFHIGHPID | RFPPWAIT)
+#define RFTHPNSHIFT	24	/* reserve bits 24-30 */
+#define RFTHPNMASK	0x7F    /* for compatibility with linuxthreads/clone()   */
+				/* allow to specify  "clone exit parent notification" signal */
+#define RFTHPNSIGNUM(flags)	(((flags) >> RFTHPNSHIFT) & RFTHPNMASK)
 
 #endif /* __BSD_VISIBLE */
 
