
Status: in upstream BTS

Index: src/sys/conf/Makefile.amd64
===================================================================
--- src/sys/conf/Makefile.amd64.orig
+++ src/sys/conf/Makefile.amd64
@@ -38,6 +38,8 @@
 
 MKMODULESENV+= MACHINE=amd64
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: src/sys/conf/Makefile.arm
===================================================================
--- src/sys/conf/Makefile.arm.orig
+++ src/sys/conf/Makefile.arm
@@ -109,6 +109,8 @@
 	rm ${KERNEL_KO}.tmp.gz ${KERNEL_KO}.tramp.noheader opt_kernname.h \
 	inflate-tramp.o tmphack.S
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: src/sys/conf/Makefile.i386
===================================================================
--- src/sys/conf/Makefile.i386.orig
+++ src/sys/conf/Makefile.i386
@@ -34,6 +34,8 @@
 
 MKMODULESENV+=	MACHINE=${MACHINE}
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: src/sys/conf/Makefile.ia64
===================================================================
--- src/sys/conf/Makefile.ia64.orig
+++ src/sys/conf/Makefile.ia64
@@ -36,6 +36,8 @@
 
 ASM_CFLAGS= -x assembler-with-cpp -Wa,-x -DLOCORE ${CFLAGS}
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: src/sys/conf/Makefile.pc98
===================================================================
--- src/sys/conf/Makefile.pc98.orig
+++ src/sys/conf/Makefile.pc98
@@ -34,6 +34,8 @@
 
 MKMODULESENV+=	MACHINE=${MACHINE}
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: src/sys/conf/Makefile.powerpc
===================================================================
--- src/sys/conf/Makefile.powerpc.orig
+++ src/sys/conf/Makefile.powerpc
@@ -37,6 +37,8 @@
 CFLAGS+=	-fno-omit-frame-pointer
 .endif
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: src/sys/conf/Makefile.sparc64
===================================================================
--- src/sys/conf/Makefile.sparc64.orig
+++ src/sys/conf/Makefile.sparc64
@@ -32,6 +32,8 @@
 
 MDOBJS=	exception.o interrupt.o
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: src/usr.sbin/config/mkmakefile.c
===================================================================
--- src/usr.sbin/config/mkmakefile.c.orig
+++ src/usr.sbin/config/mkmakefile.c
@@ -70,6 +70,7 @@
 static void do_rules(FILE *);
 static void do_xxfiles(char *, FILE *);
 static void do_objs(FILE *);
+static void do_without_modules(FILE *);
 static void do_before_depend(FILE *);
 static int opteq(const char *, const char *);
 static void read_files(void);
@@ -142,6 +143,8 @@
 		}
 		if (eq(line, "%BEFORE_DEPEND\n"))
 			do_before_depend(ofp);
+		else if (eq(line, "%WITHOUT_MODULES\n"))
+			do_without_modules(ofp);
 		else if (eq(line, "%OBJS\n"))
 			do_objs(ofp);
 		else if (strncmp(line, "%FILES.", 7) == 0)
@@ -622,6 +625,28 @@
 }
 
 static void
+do_without_modules(FILE *fp)
+{
+	struct device *dp;
+	int lpos, len;
+	char *cp, *sp;
+
+	fprintf(fp, "MKMODULESENV+= WITHOUT_MODULES=\"");
+	lpos = 34;
+	STAILQ_FOREACH(dp, &dtab, d_next) {
+		sp = dp->d_name;
+		cp = sp + (len = strlen(sp)) - 1;
+		if (len + lpos > 72) {
+			lpos = 8;
+			fprintf(fp, "\\\n\t");
+		}
+		fprintf(fp, "%s ", sp);
+		lpos += len + 1;
+	}
+	fprintf(fp, "\"\n");
+}
+
+static void
 do_xxfiles(char *tag, FILE *fp)
 {
 	struct file_list *tp;
