
Status: in upstream BTS

Index: sys/conf/Makefile.amd64
===================================================================
--- sys/conf/Makefile.amd64.orig	2005-10-28 21:04:03.000000000 +0200
+++ sys/conf/Makefile.amd64	2006-06-24 17:33:28.000000000 +0200
@@ -38,6 +38,8 @@
 
 MKMODULESENV+= MACHINE=amd64
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: sys/conf/Makefile.arm
===================================================================
--- sys/conf/Makefile.arm.orig	2005-10-28 21:04:03.000000000 +0200
+++ sys/conf/Makefile.arm	2006-06-24 17:33:28.000000000 +0200
@@ -101,6 +101,8 @@
 	rm ${KERNEL_KO}.tmp.gz ${KERNEL_KO}.tramp.noheader opt_kernname.h \
 	inflate-tramp.o
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: sys/conf/Makefile.i386
===================================================================
--- sys/conf/Makefile.i386.orig	2005-10-28 21:04:03.000000000 +0200
+++ sys/conf/Makefile.i386	2006-06-24 17:33:28.000000000 +0200
@@ -32,6 +32,8 @@
 
 MKMODULESENV+= MACHINE=i386
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: sys/conf/Makefile.ia64
===================================================================
--- sys/conf/Makefile.ia64.orig	2005-10-28 21:04:03.000000000 +0200
+++ sys/conf/Makefile.ia64	2006-06-24 17:33:28.000000000 +0200
@@ -38,6 +38,8 @@
 
 ASM_CFLAGS= -x assembler-with-cpp -Wa,-x -DLOCORE ${CFLAGS}
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: sys/conf/Makefile.pc98
===================================================================
--- sys/conf/Makefile.pc98.orig	2005-10-28 21:04:03.000000000 +0200
+++ sys/conf/Makefile.pc98	2006-06-24 17:33:28.000000000 +0200
@@ -32,6 +32,8 @@
 
 MKMODULESENV+=	MACHINE=pc98
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: sys/conf/Makefile.powerpc
===================================================================
--- sys/conf/Makefile.powerpc.orig	2005-10-28 21:04:03.000000000 +0200
+++ sys/conf/Makefile.powerpc	2006-06-24 17:33:28.000000000 +0200
@@ -37,6 +37,8 @@
 CFLAGS+=	-fno-omit-frame-pointer
 .endif
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: sys/conf/Makefile.sparc64
===================================================================
--- sys/conf/Makefile.sparc64.orig	2005-10-28 21:04:03.000000000 +0200
+++ sys/conf/Makefile.sparc64	2006-06-24 17:33:28.000000000 +0200
@@ -32,6 +32,8 @@
 
 MDOBJS=	exception.o interrupt.o
 
+%WITHOUT_MODULES
+
 %BEFORE_DEPEND
 
 %OBJS
Index: usr.sbin/config/mkmakefile.c
===================================================================
--- usr.sbin/config/mkmakefile.c.orig	2004-08-31 01:03:56.000000000 +0200
+++ usr.sbin/config/mkmakefile.c	2006-06-24 17:33:28.000000000 +0200
@@ -70,6 +70,7 @@
 static void do_rules(FILE *);
 static void do_xxfiles(char *, FILE *);
 static void do_objs(FILE *);
+static void do_without_modules(FILE *);
 static void do_before_depend(FILE *);
 static int opteq(const char *, const char *);
 static void read_files(void);
@@ -150,6 +151,8 @@
 		}
 		if (eq(line, "%BEFORE_DEPEND\n"))
 			do_before_depend(ofp);
+		else if (eq(line, "%WITHOUT_MODULES\n"))
+			do_without_modules(ofp);
 		else if (eq(line, "%OBJS\n"))
 			do_objs(ofp);
 		else if (strncmp(line, "%FILES.", 7) == 0)
@@ -618,6 +621,28 @@
 }
 
 static void
+do_without_modules(FILE *fp)
+{
+	struct device *dp;
+	int lpos, len;
+	char *cp, *sp;
+
+	fprintf(fp, "MKMODULESENV+= WITHOUT_MODULES=\"");
+	lpos = 34;
+	STAILQ_FOREACH(dp, &dtab, d_next) {
+		sp = dp->d_name;
+		cp = sp + (len = strlen(sp)) - 1;
+		if (len + lpos > 72) {
+			lpos = 8;
+			fprintf(fp, "\\\n\t");
+		}
+		fprintf(fp, "%s ", sp);
+		lpos += len + 1;
+	}
+	fprintf(fp, "\"\n");
+}
+
+static void
 do_xxfiles(char *tag, FILE *fp)
 {
 	struct file_list *tp;
