Index: sys/netinet/ip.h
===================================================================
--- sys/netinet/ip.h.orig	2005-01-07 02:45:44.000000000 +0100
+++ sys/netinet/ip.h	2006-06-24 17:38:11.000000000 +0200
@@ -65,7 +65,7 @@
 	u_char	ip_p;			/* protocol */
 	u_short	ip_sum;			/* checksum */
 	struct	in_addr ip_src,ip_dst;	/* source and dest address */
-} __packed;
+} __attribute__ ((packed));
 
 #ifdef CTASSERT
 CTASSERT(sizeof (struct ip) == 20);
