Index: sys/sys/soundcard.h
===================================================================
--- sys/sys/soundcard.h.orig	2005-12-30 20:55:52.000000000 +0100
+++ sys/sys/soundcard.h	2006-06-24 17:38:44.000000000 +0200
@@ -809,6 +813,8 @@
     /* Has a coprocessor, sometimes it's a DSP but usually not */
 #define DSP_CAP_TRIGGER		0x00001000 /* Supports SETTRIGGER */
 #define DSP_CAP_MMAP 0x00002000 /* Supports mmap() */
+#define DSP_CAP_MULTI		0x00004000 /* Supports multiple open */
+#define DSP_CAP_BIND		0x00008000 /* Channel binding to front/rear/cneter/lfe */
 
 /*
  * What do these function do ?
