
 MAXCPU is needed in support.s

Index: sys/amd64/amd64/genassym.c
===================================================================
--- sys/amd64/amd64/genassym.c.orig	2006-06-24 20:18:36.612344000 +0200
+++ sys/amd64/amd64/genassym.c	2006-06-24 20:20:33.000000000 +0200
@@ -88,6 +88,7 @@
 ASSYM(V_INTR, offsetof(struct vmmeter, v_intr));
 ASSYM(KSTACK_PAGES, KSTACK_PAGES);
 ASSYM(PAGE_SIZE, PAGE_SIZE);
+ASSYM(MAXCPU, MAXCPU);
 ASSYM(NPTEPG, NPTEPG);
 ASSYM(NPDEPG, NPDEPG);
 ASSYM(addr_PTmap, addr_PTmap);
Index: sys/i386/i386/genassym.c
===================================================================
--- sys/i386/i386/genassym.c.orig	2006-06-24 20:18:44.297911000 +0200
+++ sys/i386/i386/genassym.c	2006-06-24 20:20:19.000000000 +0200
@@ -97,6 +97,7 @@
 /* ASSYM(UPAGES, UPAGES);*/
 ASSYM(KSTACK_PAGES, KSTACK_PAGES);
 ASSYM(PAGE_SIZE, PAGE_SIZE);
+ASSYM(MAXCPU, MAXCPU);
 ASSYM(NPTEPG, NPTEPG);
 ASSYM(NPDEPG, NPDEPG);
 ASSYM(NPDEPTD, NPDEPTD);
