Index: sys/conf/newvers.sh
===================================================================
--- sys/conf/newvers.sh.orig	2006-06-24 17:10:55.000000000 +0200
+++ sys/conf/newvers.sh	2006-06-24 17:38:49.000000000 +0200
@@ -89,7 +89,7 @@
 cat << EOF > vers.c
 $COPYRIGHT
 #define SCCSSTR "@(#)${VERSION} #${v}: ${t}"
-#define VERSTR "${VERSION} #${v}: ${t}\\n    ${u}@${h}:${d}\\n"
+#define VERSTR "#${v} ${t}\\n"
 #define RELSTR "${RELEASE}"
 
 char sccs[sizeof(SCCSSTR) > 128 ? sizeof(SCCSSTR) : 128] = SCCSSTR;
