Index: src/sys/conf/newvers.sh
===================================================================
--- src/sys/conf/newvers.sh.orig
+++ src/sys/conf/newvers.sh
@@ -105,7 +105,7 @@
 cat << EOF > vers.c
 $COPYRIGHT
 #define SCCSSTR "@(#)${VERSION} #${v}${svn}: ${t}"
-#define VERSTR "${VERSION} #${v}${svn}: ${t}\\n    ${u}@${h}:${d}\\n"
+#define VERSTR "#${v} ${t}\\n"
 #define RELSTR "${RELEASE}"
 
 char sccs[sizeof(SCCSSTR) > 128 ? sizeof(SCCSSTR) : 128] = SCCSSTR;
