Index: sys/conf/files
===================================================================
--- sys/conf/files.orig
+++ sys/conf/files
@@ -469,17 +469,6 @@
 dev/ata/atapi-cd.c		optional atapicd
 dev/ata/atapi-fd.c		optional atapifd
 dev/ata/atapi-tape.c		optional atapist
-dev/ath/ah_osdep.c optional ath_hal \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/ath_rate/amrr/amrr.c	optional ath_rate_amrr
-dev/ath/ath_rate/onoe/onoe.c	optional ath_rate_onoe \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/ath_rate/sample/sample.c	optional ath_rate_sample \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/if_ath.c		optional ath \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
-dev/ath/if_ath_pci.c		optional ath pci \
-	compile-with "${NORMAL_C} -I$S/dev/ath"
 dev/awi/am79c930.c		optional awi
 dev/awi/awi.c			optional awi
 dev/awi/if_awi_pccard.c		optional awi pccard
Index: sys/conf/files.i386
===================================================================
--- sys/conf/files.i386.orig
+++ sys/conf/files.i386
@@ -46,47 +46,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-trlld.o				optional	oltr			\
-	dependency	"$S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/oltr/i386-elf.trlld.o.uu"	\
-	no-implicit-rule
-#
-hal.o				optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/i386-elf.hal.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/ath/public/i386-elf.hal.o.uu" \
-	no-implicit-rule
-opt_ah.h			optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/i386-elf.opt_ah.h"	\
-	compile-with	"rm -f opt_ah.h; cp $S/contrib/dev/ath/public/i386-elf.opt_ah.h opt_ah.h" \
-	no-obj no-implicit-rule before-depend				\
-	clean		"opt_ah.h"
-#
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/i386/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
-	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptrr_lib.o			optional	hptrr			\
-	dependency	"$S/dev/hptrr/i386-elf.hptrr_lib.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptrr/i386-elf.hptrr_lib.o.uu" \
-	no-implicit-rule
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/i386-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/dev/hptmv/i386-elf.raid.o.uu" \
-	no-implicit-rule
-#
-rr232x_lib.o			optional	rr232x			\
-	dependency	"$S/dev/rr232x/i386-elf.rr232x_lib.o.uu"	\
-	compile-with	"uudecode < $S/dev/rr232x/i386-elf.rr232x_lib.o.uu" \
-	no-implicit-rule
-#
 #
 compat/linprocfs/linprocfs.c	optional linprocfs
 compat/linsysfs/linsysfs.c	optional linsysfs
@@ -184,14 +143,6 @@
 dev/fdc/fdc_isa.c		optional fdc isa
 dev/fdc/fdc_pccard.c		optional fdc pccard
 dev/fe/if_fe_isa.c		optional fe isa
-dev/hptmv/entry.c		optional hptmv
-dev/hptmv/mv.c			optional hptmv
-dev/hptmv/gui_lib.c		optional hptmv
-dev/hptmv/hptproc.c		optional hptmv
-dev/hptmv/ioctl.c		optional hptmv
-dev/hptrr/hptrr_os_bsd.c	optional hptrr
-dev/hptrr/hptrr_osm_bsd.c	optional hptrr
-dev/hptrr/hptrr_config.c	optional hptrr
 dev/hwpmc/hwpmc_amd.c		optional hwpmc
 dev/hwpmc/hwpmc_pentium.c	optional hwpmc
 dev/hwpmc/hwpmc_piv.c		optional hwpmc
@@ -218,13 +169,9 @@
 dev/mse/mse.c			optional mse
 dev/mse/mse_isa.c		optional mse isa
 dev/nfe/if_nfe.c		optional nfe pci
-dev/nve/if_nve.c		optional nve pci
 dev/nvram/nvram.c		optional nvram isa
 dev/pcf/pcf_isa.c		optional pcf
 dev/random/nehemiah.c		optional random
-dev/rr232x/os_bsd.c		optional rr232x
-dev/rr232x/osm_bsd.c		optional rr232x
-dev/rr232x/rr232x_config.c	optional rr232x
 dev/sbni/if_sbni.c		optional sbni
 dev/sbni/if_sbni_isa.c		optional sbni isa
 dev/sbni/if_sbni_pci.c		optional sbni pci
Index: sys/conf/kern.pre.mk
===================================================================
--- sys/conf/kern.pre.mk.orig
+++ sys/conf/kern.pre.mk
@@ -67,9 +67,6 @@
 # ... and the same for pf
 INCLUDES+= -I$S/contrib/pf
 
-# ... and the same for Atheros HAL
-INCLUDES+= -I$S/dev/ath
-
 # ... and the same for the NgATM stuff
 INCLUDES+= -I$S/contrib/ngatm
 
Index: sys/i386/conf/GENERIC
===================================================================
--- sys/i386/conf/GENERIC.orig
+++ sys/i386/conf/GENERIC
@@ -128,9 +128,6 @@
 device		asr		# DPT SmartRAID V, VI and Adaptec SCSI RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		hptrr		# Highpoint RocketRAID 17xx, 22xx, 23xx, 25xx
-device		rr232x		# Highpoint RocketRAID 232x
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
Index: sys/modules/Makefile
===================================================================
--- sys/modules/Makefile.orig
+++ sys/modules/Makefile
@@ -25,11 +25,6 @@
 	${_arl} \
 	${_asr} \
 	ata \
-	${_ath} \
-	${_ath_hal} \
-	${_ath_rate_amrr} \
-	${_ath_rate_onoe} \
-	${_ath_rate_sample} \
 	aue \
 	${_auxio} \
 	${_awi} \
@@ -95,8 +90,6 @@
 	hifn \
 	hme \
 	${_hptiop} \
-	${_hptmv} \
-	${_hptrr} \
 	hwpmc \
 	${_i2c} \
 	${_ibcs2} \
@@ -189,10 +182,8 @@
 	ntfs_iconv \
 	${_nxge} \
 	nullfs \
-	${_nve} \
 	${_nvram} \
 	${_nwfs} \
-	${_oltr} \
 	${_padlock} \
 	patm \
 	${_pccard} \
@@ -221,7 +212,6 @@
 	reiserfs \
 	rl \
 	rp \
-	${_rr232x} \
 	rue \
 	rum \
 	${_s3} \
@@ -314,7 +304,6 @@
 	${_xe} \
 	xfs \
 	xl \
-	${_zfs} \
 	zlib \
 	zyd
 
@@ -364,11 +353,6 @@
 _apm=		apm
 _ar=		ar
 _arcnet=	arcnet
-_ath=		ath
-_ath_hal=	ath_hal
-_ath_rate_amrr=	ath_rate_amrr
-_ath_rate_onoe=	ath_rate_onoe
-_ath_rate_sample=ath_rate_sample
 _awi=		awi
 _bktr=		bktr
 _cardbus=	cardbus
@@ -407,7 +391,6 @@
 .if ${MK_NCP} != "no"
 _nwfs=		nwfs
 .endif
-_oltr=		oltr
 _pccard=	pccard
 _pcfclock=	pcfclock
 _pecoff=	pecoff
@@ -428,9 +411,6 @@
 _tmpfs= 	tmpfs
 _wi=		wi
 _xe=		xe
-.if ${MK_ZFS} != "no" || defined(ALL_MODULES)
-_zfs=		zfs
-.endif
 .if ${MACHINE} == "i386"
 _aac=		aac
 _acpi=		acpi
@@ -447,8 +427,6 @@
 _dpt=		dpt
 _ex=		ex
 _hptiop=	hptiop
-_hptmv=		hptmv
-_hptrr=		hptrr
 _ichwd=		ichwd
 _ida=		ida
 # Disabled due to non-MPSAFEty in 7.0
@@ -463,7 +441,6 @@
 _ixgb=   	ixgb
 _mly=		mly
 _nfe=		nfe
-_nve=		nve
 _nvram=		nvram
 _nxge=		nxge
 _wpi=		wpi
@@ -473,7 +450,6 @@
 _padlock=	padlock
 .endif
 .endif
-_rr232x=	rr232x
 _s3=		s3
 _twa=		twa
 _vesa=		vesa
@@ -491,11 +467,6 @@
 _agp=		agp
 _an=		an
 _arcmsr=	arcmsr
-_ath=		ath
-_ath_hal=	ath_hal
-_ath_rate_amrr=	ath_rate_amrr
-_ath_rate_onoe=	ath_rate_onoe
-_ath_rate_sample=ath_rate_sample
 _cardbus=	cardbus
 _cbb=		cbb
 _ciss=		ciss
@@ -508,8 +479,6 @@
 _exca=		exca
 _ext2fs=	ext2fs
 _hptiop=	hptiop
-_hptmv=		hptmv
-_hptrr=		hptrr
 _i2c=		i2c
 _ichwd=		ichwd
 _ida=		ida
@@ -527,11 +496,9 @@
 _mly=		mly
 _ndis=		ndis
 _nfe=		nfe
-_nve=		nve
 _nvram=		nvram
 _nxge=		nxge
 _pccard=	pccard
-_rr232x=	rr232x
 _safe=		safe
 _scsi_low=	scsi_low
 _smbfs=		smbfs
@@ -543,9 +510,6 @@
 _wi=		wi
 _wpi=		wpi
 _wpifw=		wpifw
-.if ${MK_ZFS} != "no" || defined(ALL_MODULES)
-_zfs=		zfs
-.endif
 .endif
 
 .if ${MACHINE_ARCH} == "ia64"
@@ -586,21 +550,11 @@
 
 .if ${MACHINE_ARCH} == "powerpc"
 _an=		an
-_ath=		ath
-_ath_hal=	ath_hal
-_ath_rate_amrr=	ath_rate_amrr
-_ath_rate_onoe=	ath_rate_onoe
-_ath_rate_sample=ath_rate_sample
 _nvram=		powermac_nvram
 _smbfs=		smbfs
 .endif
 
 .if ${MACHINE_ARCH} == "sparc64"
-_ath=		ath
-_ath_hal=	ath_hal
-_ath_rate_amrr=	ath_rate_amrr
-_ath_rate_onoe=	ath_rate_onoe
-_ath_rate_sample=ath_rate_sample
 _auxio=		auxio
 _em=		em
 _i2c=		i2c
Index: sys/amd64/conf/GENERIC
===================================================================
--- sys/amd64/conf/GENERIC.orig
+++ sys/amd64/conf/GENERIC
@@ -121,9 +121,6 @@
 device		arcmsr		# Areca SATA II RAID
 device		ciss		# Compaq Smart RAID 5*
 device		dpt		# DPT Smartcache III, IV - See NOTES for options
-device		hptmv		# Highpoint RocketRAID 182x
-device		hptrr		# Highpoint RocketRAID 17xx, 22xx, 23xx, 25xx
-device		rr232x		# Highpoint RocketRAID 232x
 device		iir		# Intel Integrated RAID
 device		ips		# IBM (Adaptec) ServeRAID
 device		mly		# Mylex AcceleRAID/eXtremeRAID
Index: sys/conf/files.amd64
===================================================================
--- sys/conf/files.amd64.orig
+++ sys/conf/files.amd64
@@ -47,41 +47,6 @@
 	no-obj no-implicit-rule before-depend				\
 	clean		"ukbdmap.h"
 #
-hal.o				optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/x86_64-elf.hal.o.uu"	\
-	compile-with	"uudecode < $S/contrib/dev/ath/public/x86_64-elf.hal.o.uu" \
-	no-implicit-rule
-opt_ah.h			optional	ath_hal			\
-	dependency	"$S/contrib/dev/ath/public/x86_64-elf.opt_ah.h"	\
-	compile-with	"rm -f opt_ah.h; cp $S/contrib/dev/ath/public/x86_64-elf.opt_ah.h opt_ah.h" \
-	no-obj no-implicit-rule before-depend				\
-	clean		"opt_ah.h"
-#
-nvenetlib.o			optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu"	\
-	compile-with	"uudecode $S/contrib/dev/nve/amd64/nvenetlib.o.bz2.uu ; bzip2 -df nvenetlib.o.bz2" \
- 	no-implicit-rule
-#
-os+%DIKED-nve.h		optional	nve pci			\
-	dependency	"$S/contrib/dev/nve/os.h"			\
-	compile-with	"sed -e 's/^.*#include.*phy\.h.*$$//' $S/contrib/dev/nve/os.h > os+%DIKED-nve.h" \
-	no-implicit-rule no-obj before-depend				\
-	clean		"os+%DIKED-nve.h"
-#
-hptmvraid.o			optional	hptmv			\
-	dependency	"$S/dev/hptmv/amd64-elf.raid.o.uu"	\
-	compile-with	"uudecode < $S/dev/hptmv/amd64-elf.raid.o.uu" \
-	no-implicit-rule
-#
-hptrr_lib.o			optional	hptrr			\
-	dependency	"$S/dev/hptrr/amd64-elf.hptrr_lib.o.uu"		\
-	compile-with	"uudecode < $S/dev/hptrr/amd64-elf.hptrr_lib.o.uu" \
-	no-implicit-rule
-#
-rr232x_lib.o		optional	rr232x				\
-	dependency	"$S/dev/rr232x/amd64-elf.rr232x_lib.o.uu"	\
-	compile-with	"uudecode < $S/dev/rr232x/amd64-elf.rr232x_lib.o.uu" \
-	no-implicit-rule
 #
 amd64/acpica/OsdEnvironment.c	optional	acpi
 amd64/acpica/acpi_machdep.c	optional	acpi
@@ -179,25 +144,13 @@
 dev/fdc/fdc_acpi.c		optional	fdc
 dev/fdc/fdc_isa.c		optional	fdc isa
 dev/fdc/fdc_pccard.c		optional	fdc pccard
-dev/hptmv/entry.c		optional	hptmv
-dev/hptmv/mv.c			optional	hptmv
-dev/hptmv/gui_lib.c		optional	hptmv
-dev/hptmv/hptproc.c		optional	hptmv
-dev/hptmv/ioctl.c		optional	hptmv
-dev/hptrr/hptrr_os_bsd.c	optional	hptrr
-dev/hptrr/hptrr_osm_bsd.c	optional	hptrr
-dev/hptrr/hptrr_config.c	optional	hptrr
 dev/hwpmc/hwpmc_amd.c		optional	hwpmc
 dev/hwpmc/hwpmc_piv.c		optional	hwpmc
 dev/hwpmc/hwpmc_x86.c		optional	hwpmc
 dev/kbd/kbd.c			optional	atkbd | sc | ukbd
 dev/mem/memutil.c		optional	mem
 dev/nfe/if_nfe.c		optional	nfe pci
-dev/nve/if_nve.c		optional	nve pci
 dev/nvram/nvram.c		optional	nvram isa
-dev/rr232x/os_bsd.c		optional	rr232x
-dev/rr232x/osm_bsd.c		optional	rr232x
-dev/rr232x/rr232x_config.c	optional	rr232x
 dev/sio/sio.c			optional	sio
 dev/sio/sio_isa.c		optional	sio isa
 dev/sio/sio_pccard.c		optional	sio pccard
