
Status: Rejected in upstream (kern/77355).

Index: sys/i386/i386/identcpu.c
===================================================================
--- sys/i386/i386/identcpu.c.orig	2007-12-09 13:17:58.684647000 +0100
+++ sys/i386/i386/identcpu.c	2007-12-09 13:24:47.000000000 +0100
@@ -643,6 +643,7 @@
 #if defined(I486_CPU)
 	case CPUCLASS_486:
 		printf("486");
+		machine[1] = '4';
 		bzero_vector = i486_bzero;
 		break;
 #endif
@@ -653,6 +654,7 @@
 		       (intmax_t)(tsc_freq + 4999) / 1000000,
 		       (u_int)((tsc_freq + 4999) / 10000) % 100);
 		printf("586");
+		machine[1] = '5';
 		break;
 #endif
 #if defined(I686_CPU)
@@ -662,6 +664,7 @@
 		       (intmax_t)(tsc_freq + 4999) / 1000000,
 		       (u_int)((tsc_freq + 4999) / 10000) % 100);
 		printf("686");
+		machine[1] = '6';
 		break;
 #endif
 	default:
