Index: src/sys/compat/linprocfs/linprocfs.c
===================================================================
--- src.orig/sys/compat/linprocfs/linprocfs.c
+++ src/sys/compat/linprocfs/linprocfs.c
@@ -336,10 +336,6 @@
 
 		/* determine fs type */
 		fstype = mp->mnt_stat.f_fstypename;
-		if (strcmp(fstype, pn->pn_info->pi_name) == 0)
-			mntfrom = fstype = "proc";
-		else if (strcmp(fstype, "procfs") == 0)
-			continue;
 
 		if (strcmp(fstype, "linsysfs") == 0) {
 			sbuf_printf(sb, "/sys %s sysfs %s", mntto,
