Index: sys/compat/linprocfs/linprocfs.c
===================================================================
--- sys/compat/linprocfs/linprocfs.c.orig	2007-12-09 13:17:58.550924000 +0100
+++ sys/compat/linprocfs/linprocfs.c	2007-12-09 13:24:47.000000000 +0100
@@ -336,10 +336,6 @@
 
 		/* determine fs type */
 		fstype = mp->mnt_stat.f_fstypename;
-		if (strcmp(fstype, pn->pn_info->pi_name) == 0)
-			mntfrom = fstype = "proc";
-		else if (strcmp(fstype, "procfs") == 0)
-			continue;
 
 		if (strcmp(fstype, "linsysfs") == 0) {
 			sbuf_printf(sb, "/sys %s sysfs %s", mntto,
