Index: sys/i386/conf/GENERIC.hints
===================================================================
--- sys/i386/conf/GENERIC.hints.orig	2007-12-09 13:17:58.365646000 +0100
+++ sys/i386/conf/GENERIC.hints	2007-12-09 13:24:47.000000000 +0100
@@ -27,6 +27,7 @@
 hint.atkbd.0.irq="1"
 hint.psm.0.at="atkbdc"
 hint.psm.0.irq="12"
+hint.psm.0.flags="0x100"
 hint.vga.0.at="isa"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
Index: sys/amd64/conf/GENERIC.hints
===================================================================
--- sys/amd64/conf/GENERIC.hints.orig	2007-12-09 13:17:58.415001000 +0100
+++ sys/amd64/conf/GENERIC.hints	2007-12-09 13:24:47.000000000 +0100
@@ -13,6 +13,7 @@
 hint.atkbd.0.irq="1"
 hint.psm.0.at="atkbdc"
 hint.psm.0.irq="12"
+hint.psm.0.flags="0x100"
 hint.vga.0.at="isa"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
