--- sys/i386/conf/GENERIC.hints.orig	
+++ sys/i386/conf/GENERIC.hints
@@ -27,6 +27,7 @@
 hint.atkbd.0.irq="1"
 hint.psm.0.at="atkbdc"
 hint.psm.0.irq="12"
+hint.psm.0.flags="0x100"
 hint.vga.0.at="isa"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
--- sys.old/amd64/conf/GENERIC.hints	2006-02-27 10:04:33.000000000 +0100
+++ sys/amd64/conf/GENERIC.hints	2006-02-27 10:19:52.000000000 +0100
@@ -14,6 +14,7 @@
 hint.atkbd.0.flags="0x1"
 hint.psm.0.at="atkbdc"
 hint.psm.0.irq="12"
+hint.psm.0.flags="0x100"
 hint.vga.0.at="isa"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
