Index: sys/i386/conf/GENERIC.hints
===================================================================
--- sys/i386/conf/GENERIC.hints.orig	2005-01-14 07:54:59.000000000 +0100
+++ sys/i386/conf/GENERIC.hints	2006-06-24 17:43:10.000000000 +0200
@@ -27,6 +27,7 @@
 hint.atkbd.0.irq="1"
 hint.psm.0.at="atkbdc"
 hint.psm.0.irq="12"
+hint.psm.0.flags="0x100"
 hint.vga.0.at="isa"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
Index: sys/amd64/conf/GENERIC.hints
===================================================================
--- sys/amd64/conf/GENERIC.hints.orig	2004-07-23 00:28:34.000000000 +0200
+++ sys/amd64/conf/GENERIC.hints	2006-06-24 17:43:10.000000000 +0200
@@ -14,6 +14,7 @@
 hint.atkbd.0.flags="0x1"
 hint.psm.0.at="atkbdc"
 hint.psm.0.irq="12"
+hint.psm.0.flags="0x100"
 hint.vga.0.at="isa"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
