Index: src/sys/i386/conf/GENERIC.hints
===================================================================
--- src/sys/i386/conf/GENERIC.hints.orig
+++ src/sys/i386/conf/GENERIC.hints
@@ -27,6 +27,7 @@
 hint.atkbd.0.irq="1"
 hint.psm.0.at="atkbdc"
 hint.psm.0.irq="12"
+hint.psm.0.flags="0x100"
 hint.vga.0.at="isa"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
Index: src/sys/amd64/conf/GENERIC.hints
===================================================================
--- src/sys/amd64/conf/GENERIC.hints.orig
+++ src/sys/amd64/conf/GENERIC.hints
@@ -13,6 +13,7 @@
 hint.atkbd.0.irq="1"
 hint.psm.0.at="atkbdc"
 hint.psm.0.irq="12"
+hint.psm.0.flags="0x100"
 hint.vga.0.at="isa"
 hint.sc.0.at="isa"
 hint.sc.0.flags="0x100"
