--- sys/conf/ldscript.amd64.orig	2007-01-01 23:54:36.000000000 +0100
+++ sys/conf/ldscript.amd64	2007-01-01 23:55:05.000000000 +0100
@@ -1,5 +1,5 @@
 /* $FreeBSD: src/sys/conf/ldscript.amd64,v 1.9 2004/05/29 01:09:00 tjr Exp $ */
-OUTPUT_FORMAT("elf64-x86-64", "elf64-x86-64", "elf64-x86-64")
+OUTPUT_FORMAT("elf64-x86-64-freebsd", "elf64-x86-64-freebsd", "elf64-x86-64-freebsd")
 OUTPUT_ARCH(i386:x86-64)
 ENTRY(btext)
 SEARCH_DIR("/usr/lib");
