diff -ur sys.orig/modules/cxgb/Makefile sys/modules/cxgb/Makefile
--- sys.orig/modules/cxgb/Makefile	2007-05-30 18:42:07.000000000 +0200
+++ sys/modules/cxgb/Makefile	2007-05-30 20:02:23.000000000 +0200
@@ -16,7 +16,7 @@
 .if ${MACHINE_ARCH} != "ia64"
 # ld is broken on ia64
 t3fw-4.1.0.bin: ${CXGB}/t3fw-4.1.0.bin.gz.uu
-	uudecode -p < ${CXGB}/t3fw-4.1.0.bin.gz.uu \
+	uudecode -o- < ${CXGB}/t3fw-4.1.0.bin.gz.uu \
 	| gzip -dc > ${.TARGET}
 
 FIRMWS= t3fw-4.0.0.bin:t3fw400
diff -ur sys.orig/modules/ipwfw/ipw_bss/Makefile sys/modules/ipwfw/ipw_bss/Makefile
--- sys.orig/modules/ipwfw/ipw_bss/Makefile	2007-05-30 18:15:13.000000000 +0200
+++ sys/modules/ipwfw/ipw_bss/Makefile	2007-05-30 20:03:15.000000000 +0200
@@ -9,7 +9,7 @@
 CLEANFILES=	ipw_bss
 
 ipw_bss: ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
diff -ur sys.orig/modules/ipwfw/ipw_ibss/Makefile sys/modules/ipwfw/ipw_ibss/Makefile
--- sys.orig/modules/ipwfw/ipw_ibss/Makefile	2007-05-30 18:15:13.000000000 +0200
+++ sys/modules/ipwfw/ipw_ibss/Makefile	2007-05-30 20:03:23.000000000 +0200
@@ -9,7 +9,7 @@
 CLEANFILES=	ipw_ibss
 
 ipw_ibss: ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-i.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-i.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-i.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
diff -ur sys.orig/modules/ipwfw/ipw_monitor/Makefile sys/modules/ipwfw/ipw_monitor/Makefile
--- sys.orig/modules/ipwfw/ipw_monitor/Makefile	2007-05-30 18:15:13.000000000 +0200
+++ sys/modules/ipwfw/ipw_monitor/Makefile	2007-05-30 20:03:35.000000000 +0200
@@ -9,7 +9,7 @@
 CLEANFILES=	ipw_monitor
 
 ipw_monitor: ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-p.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-p.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-p.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
diff -ur sys.orig/modules/iwifw/iwi_bss/Makefile sys/modules/iwifw/iwi_bss/Makefile
--- sys.orig/modules/iwifw/iwi_bss/Makefile	2007-05-30 18:15:14.000000000 +0200
+++ sys/modules/iwifw/iwi_bss/Makefile	2007-05-30 20:03:56.000000000 +0200
@@ -9,7 +9,7 @@
 CLEANFILES=	iwi_bss
 
 iwi_bss: ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-bss.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-bss.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-bss.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
diff -ur sys.orig/modules/iwifw/iwi_ibss/Makefile sys/modules/iwifw/iwi_ibss/Makefile
--- sys.orig/modules/iwifw/iwi_ibss/Makefile	2007-05-30 18:15:14.000000000 +0200
+++ sys/modules/iwifw/iwi_ibss/Makefile	2007-05-30 20:04:05.000000000 +0200
@@ -9,7 +9,7 @@
 CLEANFILES=	iwi_fw
 
 iwi_ibss: ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-ibss.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-ibss.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-ibss.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
diff -ur sys.orig/modules/iwifw/iwi_monitor/Makefile sys/modules/iwifw/iwi_monitor/Makefile
--- sys.orig/modules/iwifw/iwi_monitor/Makefile	2007-05-30 18:15:14.000000000 +0200
+++ sys/modules/iwifw/iwi_monitor/Makefile	2007-05-30 20:04:17.000000000 +0200
@@ -9,7 +9,7 @@
 CLEANFILES=	iwi_monitor
 
 iwi_monitor: ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-sniffer.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-sniffer.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-sniffer.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
diff -ur sys.orig/modules/mxge/mxge_eth_z8e/Makefile sys/modules/mxge/mxge_eth_z8e/Makefile
--- sys.orig/modules/mxge/mxge_eth_z8e/Makefile	2007-05-30 18:15:15.000000000 +0200
+++ sys/modules/mxge/mxge_eth_z8e/Makefile	2007-05-30 20:02:39.000000000 +0200
@@ -7,7 +7,7 @@
 CLEANFILES+= eth_z8e.dat
 
 eth_z8e.dat: ${MXGE}/eth_z8e.dat.gz.uu
-	uudecode -p < ${MXGE}/eth_z8e.dat.gz.uu \
+	uudecode -o- < ${MXGE}/eth_z8e.dat.gz.uu \
 		| gzip -dc > ${.TARGET}
 
 .include <bsd.kmod.mk>
diff -ur sys.orig/modules/mxge/mxge_ethp_z8e/Makefile sys/modules/mxge/mxge_ethp_z8e/Makefile
--- sys.orig/modules/mxge/mxge_ethp_z8e/Makefile	2007-05-30 18:15:15.000000000 +0200
+++ sys/modules/mxge/mxge_ethp_z8e/Makefile	2007-05-30 20:02:48.000000000 +0200
@@ -7,7 +7,7 @@
 CLEANFILES+= ethp_z8e.dat
 
 ethp_z8e.dat: ${MXGE}/ethp_z8e.dat.gz.uu
-	uudecode -p < ${MXGE}/ethp_z8e.dat.gz.uu \
+	uudecode -o- < ${MXGE}/ethp_z8e.dat.gz.uu \
 		| gzip -dc > ${.TARGET}
 
 .include <bsd.kmod.mk>
