Index: sys/modules/cxgb/Makefile
===================================================================
--- sys/modules/cxgb/Makefile.orig
+++ sys/modules/cxgb/Makefile
@@ -18,21 +18,21 @@
 .if ${MACHINE_ARCH} != "ia64"
 # ld is broken on ia64
 t3fw-4.7.0.bin: ${CXGB}/t3fw-4.7.0.bin.gz.uu
-	uudecode -p < ${CXGB}/t3fw-4.7.0.bin.gz.uu \
+	uudecode -o- < ${CXGB}/t3fw-4.7.0.bin.gz.uu \
 	| gzip -dc > ${.TARGET}
 
 FIRMWS= t3fw-4.7.0.bin:t3fw470
 CLEANFILES+= t3fw-4.7.0.bin
 
 t3b_protocol_sram-1.1.0.bin: ${CXGB}/t3b_protocol_sram-1.1.0.bin.gz.uu
-	uudecode -p < ${CXGB}/t3b_protocol_sram-1.1.0.bin.gz.uu \
+	uudecode -o- < ${CXGB}/t3b_protocol_sram-1.1.0.bin.gz.uu \
 	| gzip -dc > ${.TARGET}
 
 FIRMWS+= t3b_protocol_sram-1.1.0.bin:t3bps110
 CLEANFILES+= t3b_protocol_sram-1.1.0.bin
 
 t3b_tp_eeprom-1.1.0.bin: ${CXGB}/t3b_tp_eeprom-1.1.0.bin.gz.uu
-	uudecode -p < ${CXGB}/t3b_tp_eeprom-1.1.0.bin.gz.uu \
+	uudecode -o- < ${CXGB}/t3b_tp_eeprom-1.1.0.bin.gz.uu \
 	| gzip -dc > ${.TARGET}
 
 FIRMWS+= t3b_tp_eeprom-1.1.0.bin:t3btpe110
Index: sys/modules/ipwfw/ipw_bss/Makefile
===================================================================
--- sys/modules/ipwfw/ipw_bss/Makefile.orig
+++ sys/modules/ipwfw/ipw_bss/Makefile
@@ -9,7 +9,7 @@
 CLEANFILES=	ipw_bss
 
 ipw_bss: ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
Index: sys/modules/ipwfw/ipw_ibss/Makefile
===================================================================
--- sys/modules/ipwfw/ipw_ibss/Makefile.orig
+++ sys/modules/ipwfw/ipw_ibss/Makefile
@@ -9,7 +9,7 @@
 CLEANFILES=	ipw_ibss
 
 ipw_ibss: ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-i.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-i.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-i.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
Index: sys/modules/ipwfw/ipw_monitor/Makefile
===================================================================
--- sys/modules/ipwfw/ipw_monitor/Makefile.orig
+++ sys/modules/ipwfw/ipw_monitor/Makefile
@@ -9,7 +9,7 @@
 CLEANFILES=	ipw_monitor
 
 ipw_monitor: ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-p.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-p.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/ipw/ipw2100-1.3-p.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
Index: sys/modules/iwifw/iwi_bss/Makefile
===================================================================
--- sys/modules/iwifw/iwi_bss/Makefile.orig
+++ sys/modules/iwifw/iwi_bss/Makefile
@@ -9,7 +9,7 @@
 CLEANFILES=	iwi_bss
 
 iwi_bss: ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-bss.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-bss.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-bss.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
Index: sys/modules/iwifw/iwi_ibss/Makefile
===================================================================
--- sys/modules/iwifw/iwi_ibss/Makefile.orig
+++ sys/modules/iwifw/iwi_ibss/Makefile
@@ -9,7 +9,7 @@
 CLEANFILES=	iwi_fw
 
 iwi_ibss: ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-ibss.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-ibss.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-ibss.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
Index: sys/modules/iwifw/iwi_monitor/Makefile
===================================================================
--- sys/modules/iwifw/iwi_monitor/Makefile.orig
+++ sys/modules/iwifw/iwi_monitor/Makefile
@@ -9,7 +9,7 @@
 CLEANFILES=	iwi_monitor
 
 iwi_monitor: ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-sniffer.fw.uu
-	uudecode -p ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-sniffer.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../../contrib/dev/iwi/ipw2200-sniffer.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
 
Index: sys/modules/wpifw/Makefile
===================================================================
--- sys/modules/wpifw/Makefile.orig
+++ sys/modules/wpifw/Makefile
@@ -9,6 +9,6 @@
 CLEANFILES=	iwlwifi-3945-2.14.4.fw
 
 iwlwifi-3945-2.14.4.fw: ${.CURDIR}/../../contrib/dev/wpi/iwlwifi-3945-2.14.4.fw.uu
-	uudecode -p ${.CURDIR}/../../contrib/dev/wpi/iwlwifi-3945-2.14.4.fw.uu > ${.TARGET}
+	uudecode -o- ${.CURDIR}/../../contrib/dev/wpi/iwlwifi-3945-2.14.4.fw.uu > ${.TARGET}
 
 .include <bsd.kmod.mk>
