Index: sys/amd64/conf/GENERIC
===================================================================
--- sys/amd64/conf/GENERIC.orig
+++ sys/amd64/conf/GENERIC
@@ -18,13 +18,8 @@
 #
 # $FreeBSD: src/sys/amd64/conf/GENERIC,v 1.484.2.1 2007/10/11 06:20:26 kensmith Exp $
 
-cpu		HAMMER
-ident		GENERIC
-
 # To statically compile in device wiring instead of /boot/device.hints
-#hints		"GENERIC.hints"		# Default places to look for devices.
-
-makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
+hints		"GENERIC.hints"		# Default places to look for devices.
 
 options 	SCHED_4BSD		# 4BSD scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
@@ -44,6 +39,8 @@
 options 	MSDOSFS			# MSDOS Filesystem
 options 	CD9660			# ISO 9660 Filesystem
 options 	PROCFS			# Process filesystem (requires PSEUDOFS)
+options 	LINPROCFS
+options 	COMPAT_LINUX32
 options 	PSEUDOFS		# Pseudo-filesystem framework
 options 	GEOM_PART_GPT		# GUID Partition Tables.
 options 	GEOM_LABEL		# Provides labelization
@@ -52,7 +49,7 @@
 options 	COMPAT_FREEBSD4		# Compatible with FreeBSD4
 options 	COMPAT_FREEBSD5		# Compatible with FreeBSD5
 options 	COMPAT_FREEBSD6		# Compatible with FreeBSD6
-options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
+options 	SCSI_DELAY=2000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	SYSVSHM			# SYSV-style shared memory
 options         SHMMAXPGS=4096
@@ -79,9 +76,6 @@
 device		acpi
 device		pci
 
-# Floppy drives
-device		fdc
-
 # ATA and ATAPI devices
 device		ata
 device		atadisk		# ATA disk drives
@@ -89,6 +83,8 @@
 device		atapicd		# ATAPI CDROM drives
 device		atapifd		# ATAPI floppy drives
 device		atapist		# ATAPI tape drives
+device		atapicam	# emulate ATAPI devices as SCSI via CAM
+				# needs CAM to be present (scbus && pass)
 options 	ATA_STATIC_ID	# Static device numbering
 
 # SCSI Controllers
@@ -171,10 +167,6 @@
 # Parallel port
 device		ppc
 device		ppbus		# Parallel port bus (required)
-device		lpt		# Printer
-device		plip		# TCP/IP over parallel
-device		ppi		# Parallel port interface device
-#device		vpo		# Requires scbus and da
 
 # If you've got a "dumb" serial or parallel PCI card that is
 # supported by the puc(4) glue driver, uncomment the following
@@ -236,9 +228,6 @@
 device		wlan_scan_ap	# 802.11 AP mode scanning
 device		wlan_scan_sta	# 802.11 STA mode scanning
 device		an		# Aironet 4500/4800 802.11 wireless NICs.
-device		ath		# Atheros pci/cardbus NIC's
-device		ath_hal		# Atheros HAL (Hardware Access Layer)
-device		ath_rate_sample	# SampleRate tx rate control for ath
 device		awi		# BayStack 660 and others
 device		ral		# Ralink Technology RT2500 wireless NICs.
 device		wi		# WaveLAN/Intersil/Symbol 802.11 wireless NICs.
@@ -247,13 +236,8 @@
 device		loop		# Network loopback
 device		random		# Entropy device
 device		ether		# Ethernet support
-device		sl		# Kernel SLIP
-device		ppp		# Kernel PPP
-device		tun		# Packet tunnel.
 device		pty		# Pseudo-ttys (telnet etc)
 device		md		# Memory "disks"
-device		gif		# IPv6 and IPv4 tunneling
-device		faith		# IPv6-to-IPv4 relaying (translation)
 device		firmware	# firmware assist module
 
 # The `bpf' device enables the Berkeley Packet Filter.
@@ -270,12 +254,8 @@
 device		ugen		# Generic
 device		uhid		# "Human Interface Devices"
 device		ukbd		# Keyboard
-device		ulpt		# Printer
 device		umass		# Disks/Mass storage - Requires scbus and da
-device		ums		# Mouse
 device		ural		# Ralink Technology RT2500USB wireless NICs
-device		urio		# Diamond Rio 500 MP3 player
-device		uscanner	# Scanners
 # USB Ethernet, requires miibus
 device		aue		# ADMtek USB Ethernet
 device		axe		# ASIX Electronics USB Ethernet
@@ -291,3 +271,11 @@
 device		fwip		# IP over FireWire (RFC 2734,3146)
 device		dcons		# Dumb console driver
 device		dcons_crom	# Configuration ROM for dcons
+
+# Alternate queueing
+options		ALTQ
+options		ALTQ_CBQ	# Class Bases Queuing (CBQ)
+options		ALTQ_RED	# Random Early Detection (RED)
+options		ALTQ_RIO	# RED In/Out
+options		ALTQ_HFSC	# Hierarchical Packet Scheduler (HFSC)
+options		ALTQ_PRIQ	# Priority Queuing (PRIQ)
Index: sys/i386/conf/GENERIC
===================================================================
--- sys/i386/conf/GENERIC.orig
+++ sys/i386/conf/GENERIC
@@ -18,15 +18,8 @@
 #
 # $FreeBSD: src/sys/i386/conf/GENERIC,v 1.474.2.1 2007/10/11 06:20:26 kensmith Exp $
 
-cpu		I486_CPU
-cpu		I586_CPU
-cpu		I686_CPU
-ident		GENERIC
-
 # To statically compile in device wiring instead of /boot/device.hints
-#hints		"GENERIC.hints"		# Default places to look for devices.
-
-makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
+hints		"GENERIC.hints"		# Default places to look for devices.
 
 options 	SCHED_4BSD		# 4BSD scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
@@ -45,6 +38,8 @@
 options 	MSDOSFS			# MSDOS Filesystem
 options 	CD9660			# ISO 9660 Filesystem
 options 	PROCFS			# Process filesystem (requires PSEUDOFS)
+options		LINPROCFS
+options		COMPAT_LINUX		# Needed by LINPROCFS
 options 	PSEUDOFS		# Pseudo-filesystem framework
 options 	GEOM_PART_GPT		# GUID Partition Tables.
 options 	GEOM_LABEL		# Provides labelization
@@ -52,7 +47,7 @@
 options 	COMPAT_FREEBSD4		# Compatible with FreeBSD4
 options 	COMPAT_FREEBSD5		# Compatible with FreeBSD5
 options 	COMPAT_FREEBSD6		# Compatible with FreeBSD6
-options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
+options 	SCSI_DELAY=2000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	SYSVSHM			# SYSV-style shared memory
 options         SHMMAXPGS=4096
@@ -80,9 +75,6 @@
 device		eisa
 device		pci
 
-# Floppy drives
-device		fdc
-
 # ATA and ATAPI devices
 device		ata
 device		atadisk		# ATA disk drives
@@ -90,6 +82,8 @@
 device		atapicd		# ATAPI CDROM drives
 device		atapifd		# ATAPI floppy drives
 device		atapist		# ATAPI tape drives
+device		atapicam	# emulate ATAPI devices as SCSI via CAM
+				# needs CAM to be present (scbus && pass)
 options 	ATA_STATIC_ID	# Static device numbering
 
 # SCSI Controllers
@@ -182,10 +176,6 @@
 # Parallel port
 device		ppc
 device		ppbus		# Parallel port bus (required)
-device		lpt		# Printer
-device		plip		# TCP/IP over parallel
-device		ppi		# Parallel port interface device
-#device		vpo		# Requires scbus and da
 
 # If you've got a "dumb" serial or parallel PCI card that is
 # supported by the puc(4) glue driver, uncomment the following
@@ -249,9 +239,6 @@
 device		wlan_scan_ap	# 802.11 AP mode scanning
 device		wlan_scan_sta	# 802.11 STA mode scanning
 device		an		# Aironet 4500/4800 802.11 wireless NICs.
-device		ath		# Atheros pci/cardbus NIC's
-device		ath_hal		# Atheros HAL (Hardware Access Layer)
-device		ath_rate_sample	# SampleRate tx rate control for ath
 device		awi		# BayStack 660 and others
 device		ral		# Ralink Technology RT2500 wireless NICs.
 device		wi		# WaveLAN/Intersil/Symbol 802.11 wireless NICs.
@@ -261,13 +248,8 @@
 device		loop		# Network loopback
 device		random		# Entropy device
 device		ether		# Ethernet support
-device		sl		# Kernel SLIP
-device		ppp		# Kernel PPP
-device		tun		# Packet tunnel.
 device		pty		# Pseudo-ttys (telnet etc)
 device		md		# Memory "disks"
-device		gif		# IPv6 and IPv4 tunneling
-device		faith		# IPv6-to-IPv4 relaying (translation)
 device		firmware	# firmware assist module
 
 # The `bpf' device enables the Berkeley Packet Filter.
@@ -284,13 +266,9 @@
 device		ugen		# Generic
 device		uhid		# "Human Interface Devices"
 device		ukbd		# Keyboard
-device		ulpt		# Printer
 device		umass		# Disks/Mass storage - Requires scbus and da
-device		ums		# Mouse
 device		ural		# Ralink Technology RT2500USB wireless NICs
 device		rum		# Ralink Technology RT2501USB wireless NICs
-device		urio		# Diamond Rio 500 MP3 player
-device		uscanner	# Scanners
 # USB Ethernet, requires miibus
 device		aue		# ADMtek USB Ethernet
 device		axe		# ASIX Electronics USB Ethernet
@@ -306,3 +284,11 @@
 device		fwip		# IP over FireWire (RFC 2734,3146)
 device		dcons		# Dumb console driver
 device		dcons_crom	# Configuration ROM for dcons
+
+# Alternate queueing
+options		ALTQ
+options		ALTQ_CBQ	# Class Bases Queuing (CBQ)
+options		ALTQ_RED	# Random Early Detection (RED)
+options		ALTQ_RIO	# RED In/Out
+options		ALTQ_HFSC	# Hierarchical Packet Scheduler (HFSC)
+options		ALTQ_PRIQ	# Priority Queuing (PRIQ)
