Index: src/sys/amd64/conf/GENERIC
===================================================================
--- src.orig/sys/amd64/conf/GENERIC
+++ src/sys/amd64/conf/GENERIC
@@ -18,13 +18,13 @@
 #
 # $FreeBSD: src/sys/amd64/conf/GENERIC,v 1.484.2.14 2008/08/29 18:54:35 jhb Exp $
 
-cpu		HAMMER
-ident		GENERIC
+#cpu		HAMMER
+#dent		GENERIC
 
 # To statically compile in device wiring instead of /boot/device.hints
-#hints		"GENERIC.hints"		# Default places to look for devices.
+hints		"GENERIC.hints"		# Default places to look for devices.
 
-makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
+#makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
 
 options 	SCHED_ULE		# ULE scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
@@ -45,6 +45,8 @@
 options 	MSDOSFS			# MSDOS Filesystem
 options 	CD9660			# ISO 9660 Filesystem
 options 	PROCFS			# Process filesystem (requires PSEUDOFS)
+options 	LINPROCFS
+options 	COMPAT_LINUX32
 options 	PSEUDOFS		# Pseudo-filesystem framework
 options 	GEOM_PART_GPT		# GUID Partition Tables.
 options 	GEOM_LABEL		# Provides labelization
@@ -53,7 +55,7 @@
 options 	COMPAT_FREEBSD4		# Compatible with FreeBSD4
 options 	COMPAT_FREEBSD5		# Compatible with FreeBSD5
 options 	COMPAT_FREEBSD6		# Compatible with FreeBSD6
-options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
+options 	SCSI_DELAY=2000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	STACK			# stack(9) support
 options 	SYSVSHM			# SYSV-style shared memory
@@ -74,7 +76,7 @@
 #options 	KDTRACE_HOOKS		# Kernel DTrace hooks
 
 # Make an SMP-capable kernel by default
-options 	SMP			# Symmetric MultiProcessor Kernel
+#options 	SMP			# Symmetric MultiProcessor Kernel
 
 # CPU frequency control
 device		cpufreq
@@ -83,9 +85,6 @@
 device		acpi
 device		pci
 
-# Floppy drives
-device		fdc
-
 # ATA and ATAPI devices
 device		ata
 device		atadisk		# ATA disk drives
@@ -93,6 +92,8 @@
 device		atapicd		# ATAPI CDROM drives
 device		atapifd		# ATAPI floppy drives
 device		atapist		# ATAPI tape drives
+device		atapicam	# emulate ATAPI devices as SCSI via CAM
+				# needs CAM to be present (scbus && pass)
 options 	ATA_STATIC_ID	# Static device numbering
 
 # SCSI Controllers
@@ -175,10 +176,6 @@
 # Parallel port
 device		ppc
 device		ppbus		# Parallel port bus (required)
-device		lpt		# Printer
-device		plip		# TCP/IP over parallel
-device		ppi		# Parallel port interface device
-#device		vpo		# Requires scbus and da
 
 # If you've got a "dumb" serial or parallel PCI card that is
 # supported by the puc(4) glue driver, uncomment the following
@@ -252,13 +249,8 @@
 device		loop		# Network loopback
 device		random		# Entropy device
 device		ether		# Ethernet support
-device		sl		# Kernel SLIP
-device		ppp		# Kernel PPP
-device		tun		# Packet tunnel.
 device		pty		# Pseudo-ttys (telnet etc)
 device		md		# Memory "disks"
-device		gif		# IPv6 and IPv4 tunneling
-device		faith		# IPv6-to-IPv4 relaying (translation)
 device		firmware	# firmware assist module
 
 # The `bpf' device enables the Berkeley Packet Filter.
@@ -275,12 +267,9 @@
 device		ugen		# Generic
 device		uhid		# "Human Interface Devices"
 device		ukbd		# Keyboard
-device		ulpt		# Printer
 device		umass		# Disks/Mass storage - Requires scbus and da
 device		ums		# Mouse
 device		ural		# Ralink Technology RT2500USB wireless NICs
-device		urio		# Diamond Rio 500 MP3 player
-device		uscanner	# Scanners
 # USB Serial devices
 device		ucom		# Generic com ttys
 device		uark		# Technologies ARK3116 based serial adapters
@@ -307,3 +296,11 @@
 device		fwip		# IP over FireWire (RFC 2734,3146)
 device		dcons		# Dumb console driver
 device		dcons_crom	# Configuration ROM for dcons
+
+# Alternate queueing
+options		ALTQ
+options		ALTQ_CBQ	# Class Bases Queuing (CBQ)
+options		ALTQ_RED	# Random Early Detection (RED)
+options		ALTQ_RIO	# RED In/Out
+options		ALTQ_HFSC	# Hierarchical Packet Scheduler (HFSC)
+options		ALTQ_PRIQ	# Priority Queuing (PRIQ)
Index: src/sys/i386/conf/GENERIC
===================================================================
--- src.orig/sys/i386/conf/GENERIC
+++ src/sys/i386/conf/GENERIC
@@ -18,15 +18,15 @@
 #
 # $FreeBSD: src/sys/i386/conf/GENERIC,v 1.474.2.14 2008/08/29 18:54:35 jhb Exp $
 
-cpu		I486_CPU
-cpu		I586_CPU
-cpu		I686_CPU
-ident		GENERIC
+#cpu		I486_CPU
+#cpu		I586_CPU
+#cpu		I686_CPU
+#ident		GENERIC
 
 # To statically compile in device wiring instead of /boot/device.hints
-#hints		"GENERIC.hints"		# Default places to look for devices.
+hints		"GENERIC.hints"		# Default places to look for devices.
 
-makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
+#makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
 
 options 	SCHED_ULE		# ULE scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
@@ -46,6 +46,8 @@
 options 	MSDOSFS			# MSDOS Filesystem
 options 	CD9660			# ISO 9660 Filesystem
 options 	PROCFS			# Process filesystem (requires PSEUDOFS)
+options		LINPROCFS
+options		COMPAT_LINUX		# Needed by LINPROCFS
 options 	PSEUDOFS		# Pseudo-filesystem framework
 options 	GEOM_PART_GPT		# GUID Partition Tables.
 options 	GEOM_LABEL		# Provides labelization
@@ -53,7 +55,7 @@
 options 	COMPAT_FREEBSD4		# Compatible with FreeBSD4
 options 	COMPAT_FREEBSD5		# Compatible with FreeBSD5
 options 	COMPAT_FREEBSD6		# Compatible with FreeBSD6
-options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
+options 	SCSI_DELAY=2000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	STACK			# stack(9) support
 options 	SYSVSHM			# SYSV-style shared memory
@@ -73,7 +75,7 @@
 #options 	KDTRACE_HOOKS		# Kernel DTrace hooks
 
 # To make an SMP kernel, the next two lines are needed
-options 	SMP			# Symmetric MultiProcessor Kernel
+#options 	SMP			# Symmetric MultiProcessor Kernel
 device		apic			# I/O APIC
 
 # CPU frequency control
@@ -83,9 +85,6 @@
 device		eisa
 device		pci
 
-# Floppy drives
-device		fdc
-
 # ATA and ATAPI devices
 device		ata
 device		atadisk		# ATA disk drives
@@ -93,6 +92,8 @@
 device		atapicd		# ATAPI CDROM drives
 device		atapifd		# ATAPI floppy drives
 device		atapist		# ATAPI tape drives
+device		atapicam	# emulate ATAPI devices as SCSI via CAM
+				# needs CAM to be present (scbus && pass)
 options 	ATA_STATIC_ID	# Static device numbering
 
 # SCSI Controllers
@@ -185,10 +186,6 @@
 # Parallel port
 device		ppc
 device		ppbus		# Parallel port bus (required)
-device		lpt		# Printer
-device		plip		# TCP/IP over parallel
-device		ppi		# Parallel port interface device
-#device		vpo		# Requires scbus and da
 
 # If you've got a "dumb" serial or parallel PCI card that is
 # supported by the puc(4) glue driver, uncomment the following
@@ -265,13 +262,8 @@
 device		loop		# Network loopback
 device		random		# Entropy device
 device		ether		# Ethernet support
-device		sl		# Kernel SLIP
-device		ppp		# Kernel PPP
-device		tun		# Packet tunnel.
 device		pty		# Pseudo-ttys (telnet etc)
 device		md		# Memory "disks"
-device		gif		# IPv6 and IPv4 tunneling
-device		faith		# IPv6-to-IPv4 relaying (translation)
 device		firmware	# firmware assist module
 
 # The `bpf' device enables the Berkeley Packet Filter.
@@ -288,13 +280,10 @@
 device		ugen		# Generic
 device		uhid		# "Human Interface Devices"
 device		ukbd		# Keyboard
-device		ulpt		# Printer
 device		umass		# Disks/Mass storage - Requires scbus and da
 device		ums		# Mouse
 device		ural		# Ralink Technology RT2500USB wireless NICs
 device		rum		# Ralink Technology RT2501USB wireless NICs
-device		urio		# Diamond Rio 500 MP3 player
-device		uscanner	# Scanners
 # USB Serial devices
 device		ucom		# Generic com ttys
 device		uark		# Technologies ARK3116 based serial adapters
@@ -321,3 +310,11 @@
 device		fwip		# IP over FireWire (RFC 2734,3146)
 device		dcons		# Dumb console driver
 device		dcons_crom	# Configuration ROM for dcons
+
+# Alternate queueing
+options		ALTQ
+options		ALTQ_CBQ	# Class Bases Queuing (CBQ)
+options		ALTQ_RED	# Random Early Detection (RED)
+options		ALTQ_RIO	# RED In/Out
+options		ALTQ_HFSC	# Hierarchical Packet Scheduler (HFSC)
+options		ALTQ_PRIQ	# Priority Queuing (PRIQ)
