*** orig/sys/procfs.h	Sat Aug 28 02:51:56 1999
--- import/sys/procfs.h	Fri Sep  6 01:13:45 2002
***************
*** 1,3 ****
--- 1,21 ----
+ /* Copyright (C) 2002 Free Software Foundation, Inc.
+    This file is part of the GNU C Library.
+ 
+    The GNU C Library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+ 
+    The GNU C Library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+ 
+    You should have received a copy of the GNU Lesser General Public
+    License along with the GNU C Library; if not, write to the Free
+    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+    02111-1307 USA.  */
+ 
  /*-
   * Copyright (c) 1998 John D. Polstra.
   * All rights reserved.
***************
*** 65,72 ****
      gregset_t	pr_reg;		/* General purpose registers (1) */
  } prstatus_t;
  
- typedef fpregset_t prfpregset_t;
- 
  #define PRARGSZ		80	/* Maximum argument bytes saved */
  
  #define PRPSINFO_VERSION	1	/* Current version of prpsinfo_t */
--- 83,88 ----
***************
*** 78,81 ****
--- 94,108 ----
      char	pr_psargs[PRARGSZ+1];	/* Arguments, null terminated (1) */
  } prpsinfo_t;
  
+ /* Addresses.  */
+ typedef void *psaddr_t;
+ 
+ /* Register sets.  */
+ typedef gregset_t prgregset_t;
+ typedef fpregset_t prfpregset_t;
+ 
+ /* We don't have any differences between processes and threads,
+    therefore have only one PID type.  */
+ typedef __pid_t lwpid_t;
+ 
  #endif /* _SYS_PROCFS_H_ */
