diff -ur src.old/sbin/mount/mount.c src/sbin/mount/mount.c
--- src.old/sbin/mount/mount.c	2006-03-19 20:56:08.000000000 +0100
+++ src/sbin/mount/mount.c	2006-03-19 20:56:47.000000000 +0100
@@ -137,7 +137,7 @@
 	options = NULL;
 	vfslist = NULL;
 	vfstype = "ufs";
-	while ((ch = getopt(argc, argv, "adF:fo:prwt:uv")) != -1)
+	while ((ch = getopt(argc, argv, "adfo:prwt:uv")) != -1)
 		switch (ch) {
 		case 'a':
 			all = 1;
@@ -145,9 +145,6 @@
 		case 'd':
 			debug = 1;
 			break;
-		case 'F':
-			setfstab(optarg);
-			break;
 		case 'f':
 			init_flags |= MNT_FORCE;
 			break;
@@ -707,7 +704,7 @@
 {
 
 	(void)fprintf(stderr, "%s\n%s\n%s\n",
-"usage: mount [-adfpruvw] [-F fstab] [-o options] [-t ufs | external_type]",
+"usage: mount [-adfpruvw] [-o options] [-t ufs | external_type]",
 "       mount [-dfpruvw] special | node",
 "       mount [-dfpruvw] [-o options] [-t ufs | external_type] special node");
 	exit(1);
diff -ur src.old/sbin/umount/umount.c src/sbin/umount/umount.c
--- src.old/sbin/umount/umount.c	2006-03-19 20:56:08.000000000 +0100
+++ src/sbin/umount/umount.c	2006-03-19 20:57:21.000000000 +0100
@@ -94,7 +94,7 @@
 	sync();
 
 	all = errs = 0;
-	while ((ch = getopt(argc, argv, "AaF:fh:t:v")) != -1)
+	while ((ch = getopt(argc, argv, "Aafh:t:v")) != -1)
 		switch (ch) {
 		case 'A':
 			all = 2;
@@ -102,9 +102,6 @@
 		case 'a':
 			all = 1;
 			break;
-		case 'F':
-			setfstab(optarg);
-			break;
 		case 'f':
 			fflag = MNT_FORCE;
 			break;
@@ -160,7 +157,7 @@
 		break;
 	case 1:
 		if (setfsent() == 0)
-			err(1, "%s", getfstab());
+			err(1, "/etc/fstab");
 		errs = umountall(typelist);
 		break;
 	case 0:
@@ -577,6 +574,6 @@
 
 	(void)fprintf(stderr, "%s\n%s\n",
 	    "usage: umount [-fv] special | node | fsid",
-	    "       umount -a | -A [-F fstab] [-fv] [-h host] [-t type]");
+	    "       umount -a | -A [-fv] [-h host] [-t type]");
 	exit(1);
 }
