---
 usr.sbin/kbdcontrol/kbdcontrol.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/usr.sbin/kbdcontrol/kbdcontrol.c
+++ b/usr.sbin/kbdcontrol/kbdcontrol.c
@@ -136,7 +136,7 @@ void		set_functionkey(char *keynumstr, c
 void		set_keyboard(char *device);
 void		set_keyrates(char *opt);
 void		show_kbd_info(void);
-void		usage(void) __dead2;
+void		usage(void);
 
 char *
 nextarg(int ac, char **av, int *indp, int oc)
