--- src/sbin/mount/mount.c~	2006-03-19 20:51:12.000000000 +0100
+++ src/sbin/mount/mount.c	2006-03-19 20:52:07.000000000 +0100
@@ -465,10 +465,9 @@
 			exit(mount_ufs(argc, (char * const *) argv));
 
 		/* Go find an executable. */
-		execvP(execname, _PATH_SYSPATH, (char * const *)argv);
+		execvp(execname, (char * const *)argv);
 		if (errno == ENOENT) {
-			warn("exec mount_%s not found in %s", vfstype,
-			    _PATH_SYSPATH);
+			warn("exec mount_%s not found", vfstype);
 		}
 		exit(1);
 		/* NOTREACHED */
