Index: src/sbin/mount/mount.c
===================================================================
--- src.orig/sbin/mount/mount.c
+++ src/sbin/mount/mount.c
@@ -159,10 +159,9 @@
 		exit (1);
 	case 0:					/* Child. */
 		/* Go find an executable. */
-		execvP(execname, _PATH_SYSPATH, argv);
+		execvp(execname, argv);
 		if (errno == ENOENT) {
-			warn("exec %s not found in %s", execname,
-			    _PATH_SYSPATH);
+			warn("exec %s not found", execname);
 		}
 		exit(1);
 	default:				/* Parent. */
