--- src/sbin/umount/umount.c~	2006-03-18 14:22:12.000000000 +0100
+++ src/sbin/umount/umount.c	2006-03-18 14:22:55.000000000 +0100
@@ -197,9 +197,6 @@
 		    strcmp(fs->fs_type, FSTAB_RO) &&
 		    strcmp(fs->fs_type, FSTAB_RQ))
 			continue;
-		/* Ignore unknown file system types. */
-		if (getvfsbyname(fs->fs_vfstype, &vfc) == -1)
-			continue;
 		if (checkvfsname(fs->fs_vfstype, typelist))
 			continue;
 
