--- a/sbin/kldconfig/Makefile
+++ b/sbin/kldconfig/Makefile
@@ -29,4 +29,6 @@
 PROG=		kldconfig
 MAN=		kldconfig.8
 
+LDADD=		-lbsd
+
 .include <bsd.prog.mk>
--- a/sbin/kldload/Makefile
+++ b/sbin/kldload/Makefile
@@ -29,4 +29,6 @@
 PROG=	kldload
 MAN=	kldload.8
 
+LDADD= -lbsd
+
 .include <bsd.prog.mk>
--- a/sbin/kldunload/Makefile
+++ b/sbin/kldunload/Makefile
@@ -29,4 +29,6 @@
 PROG=	kldunload
 MAN=	kldunload.8
 
+LDADD+= -lbsd
+
 .include <bsd.prog.mk>
