--- src/sbin/umount/umount.c~	2006-03-18 14:21:27.000000000 +0100
+++ src/sbin/umount/umount.c	2006-03-18 14:22:12.000000000 +0100
@@ -45,6 +45,7 @@
 #include <sys/mount.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
+#include <sys/syscall.h>
 
 #include <netdb.h>
 #include <rpc/rpc.h>
@@ -349,7 +350,7 @@
 	if (!namematch(ai))
 		return (1);
 
-		if (unmount(sfs->f_mntonname, fflag) != 0) {
+		if (syscall (SYS_unmount, sfs->f_mntonname, fflag) != 0) {
 			warn("unmount of %s failed", sfs->f_mntonname);
 			return (1);
 		}
