--- src/sbin/dmesg/dmesg.c~	2006-03-30 12:40:23.000000000 +0200
+++ src/sbin/dmesg/dmesg.c	2006-03-30 12:41:26.000000000 +0200
@@ -82,7 +82,7 @@
 	all = 0;
 	(void) setlocale(LC_CTYPE, "");
 	memf = nlistf = NULL;
-	while ((ch = getopt(argc, argv, "aM:N:")) != -1)
+	while ((ch = getopt(argc, argv, "aM:N:s:n:")) != -1)
 		switch(ch) {
 		case 'a':
 			all++;
@@ -93,6 +93,9 @@
 		case 'N':
 			nlistf = optarg;
 			break;
+		case 's':
+		case 'n':
+			break;
 		case '?':
 		default:
 			usage();
--- src/sbin/swapon/swapon.c~	2006-03-30 12:40:23.000000000 +0200
+++ src/sbin/swapon/swapon.c	2006-03-30 12:43:08.000000000 +0200
@@ -80,7 +80,7 @@
 	orig_prog = which_prog;
 	
 	doall = 0;
-	while ((ch = getopt(argc, argv, "AadlhksU")) != -1) {
+	while ((ch = getopt(argc, argv, "AadlhksUv")) != -1) {
 		switch(ch) {
 		case 'A':
 			if (which_prog == SWAPCTL) {
@@ -122,6 +122,8 @@
 				usage();
 			}
 			break;
+		case 'v':
+			break;
 		case '?':
 		default:
 			usage();
