--- src/sbin/swapon/swapon.c~	2006-03-18 14:30:31.000000000 +0100
+++ src/sbin/swapon/swapon.c	2006-03-18 15:57:02.000000000 +0100
@@ -167,7 +167,7 @@
 static int
 swap_on_off(char *name, int doingall)
 {
-	if ((which_prog == SWAPOFF ? swapoff(name) : swapon(name)) == -1) {
+	if ((which_prog == SWAPOFF ? syscall (SYS_swapoff, name) : syscall (SYS_swapon, name)) == -1) {
 		switch (errno) {
 		case EBUSY:
 			if (!doingall)
@@ -209,69 +209,6 @@
 static void
 swaplist(int lflag, int sflag, int hflag)
 {
-	size_t mibsize, size;
-	struct xswdev xsw;
-	int hlen, mib[16], n, pagesize;
-	long blocksize;
-	long long total = 0;
-	long long used = 0;
-	long long tmp_total;
-	long long tmp_used;
-	
-	pagesize = getpagesize();
-	switch(hflag) {
-	case 'K':
-	    blocksize = 1024;
-	    hlen = 10;
-	    break;
-	case 'M':
-	    blocksize = 1024 * 1024;
-	    hlen = 10;
-	    break;
-	default:
-	    getbsize(&hlen, &blocksize);
-	    break;
-	}
-	
-	mibsize = sizeof mib / sizeof mib[0];
-	if (sysctlnametomib("vm.swap_info", mib, &mibsize) == -1)
-		err(1, "sysctlnametomib()");
-	
-	if (lflag) {
-		char buf[32];
-		snprintf(buf, sizeof(buf), "%ld-blocks", blocksize);
-		printf("%-13s %*s %*s\n",
-		    "Device:", 
-		    hlen, buf,
-		    hlen, "Used:");
-	}
-	
-	for (n = 0; ; ++n) {
-		mib[mibsize] = n;
-		size = sizeof xsw;
-		if (sysctl(mib, mibsize + 1, &xsw, &size, NULL, 0) == -1)
-			break;
-		if (xsw.xsw_version != XSWDEV_VERSION)
-			errx(1, "xswdev version mismatch");
-		
-		tmp_total = (long long)xsw.xsw_nblks * pagesize / blocksize;
-		tmp_used  = (long long)xsw.xsw_used * pagesize / blocksize;
-		total += tmp_total;
-		used  += tmp_used;
-		if (lflag) {
-			printf("/dev/%-8s %*lld %*lld\n", 
-			    devname(xsw.xsw_dev, S_IFCHR),
-			    hlen, tmp_total,
-			    hlen, tmp_used);
-		}
-	}
-	if (errno != ENOENT)
-		err(1, "sysctl()");
-	
-	if (sflag) {
-		printf("Total:        %*lld %*lld\n",
-		       hlen, total,
-		       hlen, used);
-	}
+  fprintf (stderr, "%s\n", strerror (ENOSYS));
 }
 
--- src/sbin/swapon/swapon.c~	2006-03-18 15:57:38.000000000 +0100
+++ src/sbin/swapon/swapon.c	2006-03-18 15:58:06.000000000 +0100
@@ -53,6 +53,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <sys/syscall.h>
 
 static void usage(void);
 static int swap_on_off(char *name, int ignoreebusy);
