---
 sbin/mount_msdosfs/mount_msdosfs.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/sbin/mount_msdosfs/mount_msdosfs.c
+++ b/sbin/mount_msdosfs/mount_msdosfs.c
@@ -49,7 +49,7 @@ static const char rcsid[] =
 #include <pwd.h>
 #include <stdio.h>
 /* must be after stdio to declare fparseln */
-#include <libutil.h>
+#include <sys/iconv.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sysexits.h>
@@ -60,7 +60,7 @@ static const char rcsid[] =
 static gid_t	a_gid(char *);
 static uid_t	a_uid(char *);
 static mode_t	a_mask(char *);
-static void	usage(void) __dead2;
+static void	usage(void);
 static int	set_charset(struct iovec **iov, int *iovlen, const char *, const char *);
 
 int
