Index: src/sbin/sysctl/sysctl.c
===================================================================
--- src.orig/sbin/sysctl/sysctl.c
+++ src/sbin/sysctl/sysctl.c
@@ -339,6 +339,7 @@
 static int
 S_loadavg(int l2, void *p)
 {
+#if 0
 	struct loadavg *tv = (struct loadavg*)p;
 
 	if (l2 != sizeof(*tv)) {
@@ -349,6 +350,7 @@
 		(double)tv->ldavg[0]/(double)tv->fscale,
 		(double)tv->ldavg[1]/(double)tv->fscale,
 		(double)tv->ldavg[2]/(double)tv->fscale);
+#endif	
 	return (0);
 }
 
