Index: src/sbin/ifconfig/Makefile
===================================================================
--- src.orig/sbin/ifconfig/Makefile
+++ src/sbin/ifconfig/Makefile
@@ -20,18 +20,18 @@
 SRCS+=	af_atalk.c		# AppleTalk support
 
 SRCS+=	ifclone.c		# clone device support
-SRCS+=	ifmac.c			# MAC support
+#SRCS+=	ifmac.c			# MAC support
 SRCS+=	ifmedia.c		# SIOC[GS]IFMEDIA support
 SRCS+=	ifvlan.c		# SIOC[GS]ETVLAN support
 SRCS+=	ifieee80211.c		# SIOC[GS]IEEE80211 support
-SRCS+=	ifgre.c			# GRE keys etc
+#SRCS+=	ifgre.c			# GRE keys etc
 
-SRCS+=	ifcarp.c		# SIOC[GS]VH support
-SRCS+=	ifgroup.c		# ...
-SRCS+=	ifpfsync.c		# pfsync(4) support
+#SRCS+=	ifcarp.c		# SIOC[GS]VH support
+#SRCS+=	ifgroup.c		# ...
+#SRCS+=	ifpfsync.c		# pfsync(4) support
 
-SRCS+=	ifbridge.c		# bridge support
-SRCS+=	iflagg.c		# lagg support
+#SRCS+=	ifbridge.c		# bridge support
+#SRCS+=	iflagg.c		# lagg support
 
 .if ${MK_IPX_SUPPORT} != "no" && !defined(RELEASE_CRUNCH)
 SRCS+=	af_ipx.c		# IPX support
Index: src/sbin/ifconfig/ifcarp.c
===================================================================
--- src.orig/sbin/ifconfig/ifcarp.c
+++ src/sbin/ifconfig/ifcarp.c
@@ -43,6 +43,7 @@
 #include <ctype.h>
 #include <stdio.h>
 #include <string.h>
+#include <bsd/string.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <err.h>
Index: src/sbin/ifconfig/ifconfig.c
===================================================================
--- src.orig/sbin/ifconfig/ifconfig.c
+++ src/sbin/ifconfig/ifconfig.c
@@ -44,10 +44,14 @@
 #include <sys/param.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <sys/types.h>
 #include <sys/sysctl.h>
 #include <sys/time.h>
 #include <sys/module.h>
 #include <sys/linker.h>
+#include <sys/sockio.h>
+#include <bsd/string.h>
+#include <netinet/ether.h>
 
 #include <net/ethernet.h>
 #include <net/if.h>
@@ -78,6 +78,13 @@
 
 #include "ifconfig.h"
 
+#ifndef IFCAP_WOL
+#define IFCAP_WOL_UCAST         0x00800 /* wake on any unicast frame */   
+#define IFCAP_WOL_MCAST         0x01000 /* wake on any multicast frame */
+#define IFCAP_WOL_MAGIC         0x02000 /* wake on any Magic Packet */   
+#define IFCAP_WOL       (IFCAP_WOL_UCAST | IFCAP_WOL_MCAST | IFCAP_WOL_MAGIC)
+#endif
+
 /*
  * Since "struct ifreq" is composed of various union members, callers
  * should pay special attention to interprete the value.
Index: src/sbin/ifconfig/af_inet6.c
===================================================================
--- src.orig/sbin/ifconfig/af_inet6.c
+++ src/sbin/ifconfig/af_inet6.c
@@ -42,7 +42,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <ifaddrs.h>
 
 #include <arpa/inet.h>
 
@@ -53,6 +52,8 @@
 #include <netdb.h>
 
 #include <netinet6/nd6.h>	/* Define ND6_INFINITE_LIFETIME */
+#include <time.h>
+#include <ifaddrs.h>
 
 #include "ifconfig.h"
 
Index: src/sbin/ifconfig/ifclone.c
===================================================================
--- src.orig/sbin/ifconfig/ifclone.c
+++ src/sbin/ifconfig/ifclone.c
@@ -41,6 +41,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <bsd/string.h>
 #include <unistd.h>
 
 #include "ifconfig.h"
Index: src/sbin/ifconfig/af_link.c
===================================================================
--- src.orig/sbin/ifconfig/af_link.c
+++ src/sbin/ifconfig/af_link.c
@@ -46,6 +46,7 @@
 #include <net/if_dl.h>
 #include <net/if_types.h>
 #include <net/ethernet.h>
+#include <netinet/ether.h>
 
 #include "ifconfig.h"
 
Index: src/sbin/ifconfig/ifieee80211.c
===================================================================
--- src.orig/sbin/ifconfig/ifieee80211.c
+++ src/sbin/ifconfig/ifieee80211.c
@@ -87,6 +87,9 @@
 #include <fcntl.h>
 #include <inttypes.h>
 #include <stdio.h>
+#include <stdint.h>
+#include <netinet/ether.h>
+#include <bsd/string.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
Index: src/sbin/route/route.c
===================================================================
--- src.orig/sbin/route/route.c
+++ src/sbin/route/route.c
@@ -63,10 +63,12 @@
 #include <paths.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 #include <string.h>
 #include <sysexits.h>
 #include <unistd.h>
 #include <ifaddrs.h>
+#include <time.h>
 
 struct keytab {
 	char	*kt_cp;
@@ -134,7 +136,7 @@
 	if (argc < 2)
 		usage((char *)NULL);
 
-	while ((ch = getopt(argc, argv, "nqdtv")) != -1)
+	while ((ch = getopt(argc, argv, "+nqdtv")) != -1)
 		switch(ch) {
 		case 'n':
 			nflag = 1;
@@ -821,7 +821,7 @@
 			else if (IN_CLASSC(addr))
 				mask = IN_CLASSC_NET;
 			else if (IN_MULTICAST(addr))
-				mask = IN_CLASSD_NET;
+				mask = 0xf0000000;
 			else
 				mask = 0xffffffff;
 		}
