--- src/sbin/mount_std/mount_std.c~	2006-03-18 17:20:33.000000000 +0100
+++ src/sbin/mount_std/mount_std.c	2006-03-18 17:21:52.000000000 +0100
@@ -55,6 +55,8 @@
 #include <sysexits.h>
 #include <unistd.h>
 
+#include <sys/syscall.h>
+
 #include "mntopts.h"
 
 static struct mntopt mopts[] = {
@@ -136,7 +138,7 @@
 	 * FreeBSD-5.0.
 	 */
 	signal(SIGSYS, catchsig);
-	error = nmount(iov, 4, mntflags);
+	error = syscall (SYS_nmount, iov, 4, mntflags);
 	signal(SIGSYS, SIG_DFL);
 
 	/*
@@ -145,7 +147,7 @@
 	 * or the user didn't recompile his kernel.
 	 */
 	if (error && (errno == EOPNOTSUPP || errno == ENOSYS || caughtsig))
-		error = mount(fsname, mntpath, mntflags, NULL);
+		error = syscall (SYS_mount, fsname, mntpath, mntflags, NULL);
 
 	if (error)
 		err(EX_OSERR, NULL);
