diff -ur src.old/sbin/ifconfig/Makefile src/sbin/ifconfig/Makefile
--- src.old/sbin/ifconfig/Makefile	2006-05-04 16:40:04.000000000 +0200
+++ src/sbin/ifconfig/Makefile	2006-05-04 16:44:05.000000000 +0200
@@ -18,16 +18,16 @@
 SRCS+=	af_atalk.c		# AppleTalk support
 
 SRCS+=	ifclone.c		# clone device support
-SRCS+=	ifmac.c			# MAC support
+#SRCS+=	ifmac.c			# MAC support
 SRCS+=	ifmedia.c		# SIOC[GS]IFMEDIA support
 SRCS+=	ifvlan.c		# SIOC[GS]ETVLAN support
 SRCS+=	ifieee80211.c		# SIOC[GS]IEEE80211 support
 
-SRCS+=	ifcarp.c		# SIOC[GS]VH support
-SRCS+=	ifpfsync.c		# pfsync(4) support
+#SRCS+=	ifcarp.c		# SIOC[GS]VH support
+#SRCS+=	ifpfsync.c		# pfsync(4) support
 
-SRCS+=	ifbridge.c		# bridge support
+#SRCS+=	ifbridge.c		# bridge support
-SRCS+=	iflagg.c		# lagg support
+#SRCS+=	iflagg.c		# lagg support
 
 .if !defined(RELEASE_CRUNCH)
 SRCS+=	af_ipx.c		# IPX support
diff -ur src.old/sbin/ifconfig/ifcarp.c src/sbin/ifconfig/ifcarp.c
--- src.old/sbin/ifconfig/ifcarp.c	2006-05-04 16:40:04.000000000 +0200
+++ src/sbin/ifconfig/ifcarp.c	2006-05-04 16:44:05.000000000 +0200
@@ -43,6 +43,7 @@
 #include <ctype.h>
 #include <stdio.h>
 #include <string.h>
+#include <bsd/string.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <err.h>
diff -ur src.old/sbin/ifconfig/ifconfig.c src/sbin/ifconfig/ifconfig.c
--- src.old/sbin/ifconfig/ifconfig.c	2006-05-04 16:40:04.000000000 +0200
+++ src/sbin/ifconfig/ifconfig.c	2006-05-04 16:44:05.000000000 +0200
@@ -44,10 +44,14 @@
 #include <sys/param.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <sys/types.h>
 #include <sys/sysctl.h>
 #include <sys/time.h>
 #include <sys/module.h>
 #include <sys/linker.h>
+#include <sys/sockio.h>
+#include <bsd/string.h>
+#include <netinet/ether.h>
 
 #include <net/ethernet.h>
 #include <net/if.h>
diff -u src/sbin/ifconfig/af_inet6.c src/sbin/ifconfig/af_inet6.c
--- src/sbin/ifconfig/af_inet6.c	2005-06-16 21:37:09.000000000 +0200
+++ src/sbin/ifconfig/af_inet6.c	2006-08-25 21:22:49.000000000 +0200
@@ -43,7 +43,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <ifaddrs.h>
 
 #include <arpa/inet.h>
 
@@ -54,6 +53,8 @@
 #include <netdb.h>
 
 #include <netinet6/nd6.h>	/* Define ND6_INFINITE_LIFETIME */
+#include <time.h>
+#include <ifaddrs.h>
 
 #include "ifconfig.h"
 
diff -u src/sbin/ifconfig/ifclone.c src/sbin/ifconfig/ifclone.c
--- src/sbin/ifconfig/ifclone.c	2004-12-08 20:18:07.000000000 +0100
+++ src/sbin/ifconfig/ifclone.c	2006-08-25 21:21:56.000000000 +0200
@@ -41,6 +41,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <bsd/string.h>
 #include <unistd.h>
 
 #include "ifconfig.h"
diff -u src/sbin/ifconfig/af_link.c src/sbin/ifconfig/af_link.c
--- src/sbin/ifconfig/af_link.c	2004-12-31 20:46:27.000000000 +0100
+++ src/sbin/ifconfig/af_link.c	2006-08-25 21:20:02.000000000 +0200
@@ -46,6 +46,7 @@
 #include <net/if_dl.h>
 #include <net/if_types.h>
 #include <net/ethernet.h>
+#include <netinet/ether.h>
 
 #include "ifconfig.h"
 
diff -ur src.old/sbin/ifconfig/ifieee80211.c src/sbin/ifconfig/ifieee80211.c
--- src.old/sbin/ifconfig/ifieee80211.c	2006-05-04 16:40:04.000000000 +0200
+++ src/sbin/ifconfig/ifieee80211.c	2006-05-04 16:44:05.000000000 +0200
@@ -84,6 +84,9 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <stdio.h>
+#include <stdint.h>
+#include <netinet/ether.h>
+#include <bsd/string.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
diff -ur src.old/sbin/route/route.c src/sbin/route/route.c
--- src.old/sbin/route/route.c	2006-05-04 16:40:04.000000000 +0200
+++ src/sbin/route/route.c	2006-05-04 16:44:09.000000000 +0200
@@ -63,10 +64,12 @@
 #include <paths.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 #include <string.h>
 #include <sysexits.h>
 #include <unistd.h>
 #include <ifaddrs.h>
+#include <time.h>
 
 struct keytab {
 	char	*kt_cp;
@@ -134,7 +134,7 @@
 	if (argc < 2)
 		usage((char *)NULL);
 
-	while ((ch = getopt(argc, argv, "nqdtv")) != -1)
+	while ((ch = getopt(argc, argv, "+nqdtv")) != -1)
		switch(ch) {
		case 'n':
			nflag = 1;
