diff -ur src.old/sbin/mount_msdosfs/mount_msdosfs.c src/sbin/mount_msdosfs/mount_msdosfs.c
--- src.old/sbin/mount_msdosfs/mount_msdosfs.c	2006-05-04 17:14:14.000000000 +0200
+++ src/sbin/mount_msdosfs/mount_msdosfs.c	2006-05-04 17:17:32.000000000 +0200
@@ -51,7 +51,7 @@
 #include <pwd.h>
 #include <stdio.h>
 /* must be after stdio to declare fparseln */
-#include <libutil.h>
+#include <sys/iconv.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sysexits.h>
diff -ur src.old/sbin/mount_ntfs/mount_ntfs.c src/sbin/mount_ntfs/mount_ntfs.c
--- src.old/sbin/mount_ntfs/mount_ntfs.c	2006-05-04 17:14:14.000000000 +0200
+++ src/sbin/mount_ntfs/mount_ntfs.c	2006-05-04 17:17:32.000000000 +0200
@@ -38,6 +38,7 @@
 #include <sys/mount.h>
 #include <sys/stat.h>
 #include <sys/module.h>
+#include <iconv.h>
 #include <sys/iconv.h>
 #include <sys/linker.h>
 #include <fs/ntfs/ntfsmount.h>
@@ -50,7 +51,6 @@
 #include <string.h>
 #include <sysexits.h>
 #include <unistd.h>
-#include <libutil.h>
 
 #include "mntopts.h"
 
