---
 sbin/route/route.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

Index: b/sbin/route/route.c
===================================================================
--- a/sbin/route/route.c
+++ b/sbin/route/route.c
@@ -63,10 +63,13 @@
 #include <paths.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 #include <string.h>
 #include <sysexits.h>
 #include <unistd.h>
 #include <ifaddrs.h>
+#include <time.h>
+#include "../../lib/linkaddr.h"

 struct keytab {
 	char	*kt_cp;
@@ -110,7 +112,7 @@
 int	prefixlen();
 extern	char *iso_ntoa();
 
-void usage(const char *) __dead2;
+void usage(const char *);
 
 void
 usage(cp)
@@ -134,7 +136,7 @@
 	if (argc < 2)
 		usage((char *)NULL);
 
-	while ((ch = getopt(argc, argv, "nqdtv")) != -1)
+	while ((ch = getopt(argc, argv, "+nqdtv")) != -1)
 		switch(ch) {
 		case 'n':
 			nflag = 1;
@@ -821,7 +823,7 @@
 	else if (IN_CLASSC(addr))
 		mask = IN_CLASSC_NET;
 	else if (IN_MULTICAST(addr))
-		mask = IN_CLASSD_NET;
+		mask = 0xf0000000;
 	else
 		mask = 0xffffffff;

--- a/sbin/route/Makefile
+++ b/sbin/route/Makefile
@@ -7,6 +7,7 @@
 MAN=	route.8
 SRCS=	route.c keywords.h
 WARNS?=	0
+LDADD= ../../lib/linkaddr.o
 CLEANFILES+=keywords.h _keywords.tmp
 
 CFLAGS+= -DNS
