---
 sbin/route/route.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

Index: freebsd-utils/sbin/route/route.c
===================================================================
--- freebsd-utils.orig/sbin/route/route.c	2009-04-14 23:05:36.000000000 +0000
+++ freebsd-utils/sbin/route/route.c	2009-11-14 10:24:14.000000000 +0000
@@ -63,10 +63,13 @@
 #include <paths.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 #include <string.h>
 #include <sysexits.h>
 #include <unistd.h>
 #include <ifaddrs.h>
+#include <time.h>
+#include "../../lib/linkaddr.h"
 
 struct keytab {
 	char	*kt_cp;
@@ -110,7 +113,7 @@
 int	prefixlen();
 extern	char *iso_ntoa();
 
-void usage(const char *) __dead2;
+void usage(const char *);
 
 void
 usage(cp)
@@ -134,7 +137,7 @@
 	if (argc < 2)
 		usage((char *)NULL);
 
-	while ((ch = getopt(argc, argv, "nqdtv")) != -1)
+	while ((ch = getopt(argc, argv, "+nqdtv")) != -1)
 		switch(ch) {
 		case 'n':
 			nflag = 1;
Index: freebsd-utils/sbin/route/Makefile
===================================================================
--- freebsd-utils.orig/sbin/route/Makefile	2006-07-27 13:26:29.000000000 +0000
+++ freebsd-utils/sbin/route/Makefile	2009-11-14 10:24:14.000000000 +0000
@@ -7,6 +7,7 @@
 MAN=	route.8
 SRCS=	route.c keywords.h
 WARNS?=	0
+LDADD= ../../lib/linkaddr.o
 CLEANFILES+=keywords.h _keywords.tmp
 
 CFLAGS+= -DNS
