Index: src/contrib/pf/authpf/authpf.c
===================================================================
--- src.orig/contrib/pf/authpf/authpf.c
+++ src/contrib/pf/authpf/authpf.c
@@ -35,7 +35,8 @@
 #ifdef __FreeBSD_kernel__
 #include <inttypes.h>
 #endif
-#include <login_cap.h>
+//#include <login_cap.h>
+#include <bsd/string.h>
 #include <pwd.h>
 #include <signal.h>
 #include <stdio.h>
@@ -92,8 +93,9 @@
 	gid_t		 gid;
 	uid_t		 uid;
 	char		*shell;
+#if 0
 	login_cap_t	*lc;
-
+#endif
 	config = fopen(PATH_CONFFILE, "r");
 	if (config == NULL) {
 		syslog(LOG_ERR, "can not open %s (%m)", PATH_CONFFILE);
@@ -139,15 +141,16 @@
 		syslog(LOG_ERR, "cannot find user for uid %u", uid);
 		goto die;
 	}
-
+#if 0
 	if ((lc = login_getclass(pw->pw_class)) != NULL)
 		shell = (char *)login_getcapstr(lc, "shell", pw->pw_shell,
 		    pw->pw_shell);
 	else
+#endif
 		shell = pw->pw_shell;
-
+#if 0
 	login_close(lc);
-
+#endif
 	if (strcmp(shell, PATH_AUTHPF_SHELL)) {
 		syslog(LOG_ERR, "wrong shell for user %s, uid %u",
 		    pw->pw_name, pw->pw_uid);
@@ -317,7 +320,9 @@
 	while (1) {
 		printf("\r\nHello %s. ", luser);
 		printf("You are authenticated from host \"%s\"\r\n", ipsrc);
+#if 0
 		setproctitle("%s@%s", luser, ipsrc);
+#endif
 		print_message(PATH_MESSAGE);
 		while (1) {
 			sleep(10);
Index: src/contrib/pf/pfctl/parse.y
===================================================================
--- src.orig/contrib/pf/pfctl/parse.y
+++ src/contrib/pf/pfctl/parse.y
@@ -56,7 +56,7 @@
 #include <limits.h>
 #include <pwd.h>
 #include <grp.h>
-#include <md5.h>
+#include <bsd/md5.h>
 
 #include "pfctl_parser.h"
 #include "pfctl.h"
@@ -401,6 +401,7 @@
 	} v;
 	int lineno;
 } YYSTYPE;
+#define YYSTYPE_IS_DECLARED 1
 
 #define DYNIF_MULTIADDR(addr) ((addr).type == PF_ADDR_DYNIFTL && \
 	(!((addr).iflags & PFI_AFLAG_NOALIAS) ||		 \
Index: src/contrib/pf/pfctl/pf_print_state.c
===================================================================
--- src.orig/contrib/pf/pfctl/pf_print_state.c
+++ src/contrib/pf/pfctl/pf_print_state.c
@@ -34,7 +34,21 @@
 
 #include <sys/types.h>
 #include <sys/socket.h>
+
+#ifdef __GLIBC__
+#include <stdint.h>
+#include <endian.h>
+#include <byteswap.h>
+# if __BYTE_ORDER == __LITTLE_ENDIAN
+# define be64toh(x) bswap_64(x)
+# elif __BYTE_ORDER == __BIG_ENDIAN
+# define be64toh(x) ((uint64_t)(x))
+# else
+# error Unknown endian
+# endif
+#else
 #include <sys/endian.h>
+#endif
 #include <net/if.h>
 #define TCPSTATES
 #include <netinet/tcp_fsm.h>
Index: src/contrib/pf/pfctl/pfctl.h
===================================================================
--- src.orig/contrib/pf/pfctl/pfctl.h
+++ src/contrib/pf/pfctl/pfctl.h
@@ -36,6 +36,12 @@
 
 enum pfctl_show { PFCTL_SHOW_RULES, PFCTL_SHOW_LABELS, PFCTL_SHOW_NOTHING };
 
+#include <bsd/string.h>
+#include <bsd/random.h>
+#define __BSD_VISIBLE 1
+#define __XSI_VISIBLE 1
+#include <sys/limits.h>
+
 enum {	PFRB_TABLES = 1, PFRB_TSTATS, PFRB_ADDRS, PFRB_ASTATS,
 	PFRB_IFACES, PFRB_TRANS, PFRB_MAX };
 struct pfr_buffer {
Index: src/contrib/pf/pfctl/pfctl_parser.c
===================================================================
--- src.orig/contrib/pf/pfctl/pfctl_parser.c
+++ src/contrib/pf/pfctl/pfctl_parser.c
@@ -42,7 +42,7 @@
 #include <netinet/in.h>
 #include <netinet/in_systm.h>
 #include <netinet/ip.h>
-#include <netinet/ip_icmp.h>
+//#include <netinet/ip_icmp.h>
 #include <netinet/icmp6.h>
 #include <net/pfvar.h>
 #include <arpa/inet.h>
@@ -59,6 +59,14 @@
 #include <ifaddrs.h>
 #include <unistd.h>
 
+#include <bsd/ip_icmp.h>
+#include <bsd/bsd.h>
+
+/* RFC2292 decls */
+#define ICMP6_MEMBERSHIP_QUERY          130     /* group membership query */
+#define ICMP6_MEMBERSHIP_REPORT         131     /* group membership report */
+#define ICMP6_MEMBERSHIP_REDUCTION      132     /* group membership termination */
+#define ICMP6_DST_UNREACH_NOTNEIGHBOR 2 /* not a neighbor(obsolete) */
 #include "pfctl_parser.h"
 #include "pfctl.h"
 
Index: src/sbin/pfctl/Makefile
===================================================================
--- src.orig/sbin/pfctl/Makefile
+++ src/sbin/pfctl/Makefile
@@ -21,7 +21,7 @@
 
 YFLAGS=
 
-LDADD+=	-lm -lmd
+LDADD+=	-lm
 DPADD+=	${LIBM} ${LIBMD}
 
 .include <bsd.prog.mk>
Index: src/usr.sbin/authpf/Makefile
===================================================================
--- src.orig/usr.sbin/authpf/Makefile
+++ src/usr.sbin/authpf/Makefile
@@ -16,7 +16,7 @@
 # XXX ALTQ:
 CFLAGS+= -DENABLE_ALTQ
 
-LDADD+= -lm -lmd -lutil
+LDADD+= -lm -lutil
 DPADD+= ${LIBM} ${LIBMD} ${LIBUTIL}
 
 .include <bsd.prog.mk>
