--- freebsd-utils-8.0.orig/usr.sbin/acpi/acpiconf/acpiconf.c  (revision 200600)
+++ freebsd-utils-8.0/usr.sbin/acpi/acpiconfacpiconf.c  (working copy)
@@ -35,10 +35,13 @@
 #include <sys/ioctl.h>
 #include <sysexits.h>
 #include <unistd.h>
+#include <stdlib.h>
 
 #include <dev/acpica/acpiio.h>
 
+#ifndef __GLIBC__
 #include <contrib/dev/acpica/include/acpi.h>
+#endif
 
 #define ACPIDEV                "/dev/acpi"
 

