---
 usr.sbin/acpi/acpiconf/acpiconf.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/usr.sbin/acpi/acpiconf/acpiconf.c
+++ b/usr.sbin/acpi/acpiconf/acpiconf.c
@@ -35,10 +35,13 @@
 #include <sys/ioctl.h>
 #include <sysexits.h>
 #include <unistd.h>
+#include <stdlib.h>
 
 #include <dev/acpica/acpiio.h>
 
+#ifndef __GLIBC__
 #include <contrib/dev/acpica/include/acpi.h>
+#endif
 
 #define ACPIDEV		"/dev/acpi"
 
