--- a/sbin/devfs/extern.h
+++ b/sbin/devfs/extern.h
@@ -50,7 +50,7 @@
 uint16_t eatonum(const char *);
 size_t efgetln(FILE *, char **);
 void tokenize(const char *, int *, char ***);
-void usage(void) __dead2;
+void usage(void);
 
 extern int mpfd;		/* Mount-point file descriptor. */
 
--- a/sbin/devfs/Makefile
+++ b/sbin/devfs/Makefile
@@ -3,5 +3,6 @@
 PROG=	devfs
 SRCS=	devfs.c rule.c
 MAN=	devfs.8
+LDADD+= -lbsd
 
 .include <bsd.prog.mk>
--- a/sbin/devfs/rule.c
+++ b/sbin/devfs/rule.c
@@ -85,8 +85,8 @@
 	int ch;
 
 	setprogname("devfs rule");
-	optreset = optind = 1;
-	while ((ch = getopt(ac, av, "s:")) != -1)
+	optind = 1;
+	while ((ch = getopt(ac, av, "+s:")) != -1)
 		switch (ch) {
 		case 's':
 			in_rsnum = eatonum(optarg);
--- a/sbin/devfs/devfs.c
+++ b/sbin/devfs/devfs.c
@@ -62,7 +62,7 @@
 	int ch;
 
 	mountpt = NULL;
-	while ((ch = getopt(ac, av, "m:")) != -1)
+	while ((ch = getopt(ac, av, "+m:")) != -1)
 		switch (ch) {
 		case 'm':
 			mountpt = optarg;
