Index: freebsd-utils/sbin/camcontrol/camcontrol.c
===================================================================
--- freebsd-utils.orig/sbin/camcontrol/camcontrol.c	2010-05-08 16:06:54.000000000 +0000
+++ freebsd-utils/sbin/camcontrol/camcontrol.c	2010-06-29 07:55:38.000000000 +0000
@@ -30,7 +30,7 @@
 __FBSDID("$FreeBSD$");
 
 #include <sys/ioctl.h>
-#include <sys/stdint.h>
+#include <stdint.h>
 #include <sys/types.h>
 #include <sys/endian.h>
 
@@ -123,9 +123,9 @@
 };
 
 #ifndef MINIMALISTIC
-static const char scsicmd_opts[] = "a:c:dfi:o:r";
-static const char readdefect_opts[] = "f:GP";
-static const char negotiate_opts[] = "acD:M:O:qR:T:UW:";
+static const char scsicmd_opts[] = "+a:c:dfi:o:r";
+static const char readdefect_opts[] = "+f:GP";
+static const char negotiate_opts[] = "+acD:M:O:qR:T:UW:";
 #endif
 
 struct camcontrol_opts option_table[] = {
@@ -4705,7 +4705,6 @@
 	 * the arguments again.
 	 */
 	optind = optstart;
-	optreset = 1;
 
 	switch(cmdlist) {
 #ifndef MINIMALISTIC
Index: freebsd-utils/sbin/camcontrol/modeedit.c
===================================================================
--- freebsd-utils.orig/sbin/camcontrol/modeedit.c	2004-01-22 07:23:36.000000000 +0000
+++ freebsd-utils/sbin/camcontrol/modeedit.c	2010-06-29 07:55:00.000000000 +0000
@@ -140,7 +140,7 @@
 
 
 static void
-editentry_create(void *hook __unused, int letter, void *arg, int count,
+editentry_create(void *hook, int letter, void *arg, int count,
 		 char *name)
 {
 	struct editentry *newentry;	/* Buffer to hold new entry. */
@@ -162,7 +162,7 @@
 }
 
 static void
-editentry_update(void *hook __unused, int letter, void *arg, int count,
+editentry_update(void *hook, int letter, void *arg, int count,
 		 char *name)
 {
 	struct editentry *dest;		/* Buffer to hold entry to update. */
@@ -190,7 +190,7 @@
 }
 
 static int
-editentry_save(void *hook __unused, char *name)
+editentry_save(void *hook, char *name)
 {
 	struct editentry *src;		/* Entry value to save. */
 
Index: freebsd-utils/sbin/camcontrol/util.c
===================================================================
--- freebsd-utils.orig/sbin/camcontrol/util.c	2003-08-05 09:19:07.000000000 +0000
+++ freebsd-utils/sbin/camcontrol/util.c	2010-06-29 07:55:00.000000000 +0000
@@ -46,7 +46,7 @@
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
-#include <sys/stdint.h>
+#include <stdint.h>
 #include <sys/types.h>
 
 #include <stdlib.h>
@@ -107,7 +107,7 @@
 /* arg_put: "put argument" callback
  */
 void
-arg_put(void *hook __unused, int letter, void *arg, int count, char *name)
+arg_put(void *hook, int letter, void *arg, int count, char *name)
 {
 	if (verbose && name && *name)
 		printf("%s:  ", name);
