---
 usr.sbin/arp/arp.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/usr.sbin/arp/arp.c
+++ b/usr.sbin/arp/arp.c
@@ -65,6 +65,7 @@
 
 #include <netinet/in.h>
 #include <netinet/if_ether.h>
+#include <netinet/ether.h>
 
 #include <arpa/inet.h>
 
@@ -656,8 +657,8 @@
  * Nuke an arp entry
  */
 static void
-nuke_entry(struct sockaddr_dl *sdl __attribute__((unused)),
-	struct sockaddr_inarp *addr, struct rt_msghdr *rtm __attribute__((unused)))
+nuke_entry(struct sockaddr_dl *sdl,
+	struct sockaddr_inarp *addr, struct rt_msghdr *rtm)
 {
 	char ip[20];
 
