
Support for PCI sound devices.

--- /dev/null
+++ b/etc/devd/pci.conf
@@ -0,0 +1,1266 @@
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x4005";
+	match "device" "0x4000";
+	action "kldload snd_als4000";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x13f6";
+	match "device" "0x0100";
+	action "kldload snd_cmi";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x13f6";
+	match "device" "0x0101";
+	action "kldload snd_cmi";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x13f6";
+	match "device" "0x0111";
+	action "kldload snd_cmi";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x13f6";
+	match "device" "0x0112";
+	action "kldload snd_cmi";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x0d8c";
+	match "device" "0x0103";
+	action "kldload snd_cmi";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1013";
+	match "device" "0x6005";
+	action "kldload snd_cs4281";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1013";
+	match "device" "0x6001";
+	action "kldload snd_csa";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1013";
+	match "device" "0x6003";
+	action "kldload snd_csa";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1013";
+	match "device" "0x6004";
+	action "kldload snd_csa";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x0004";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x000d";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x0005";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x0008";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x0020";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x0006";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x000a";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x000c";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x0010";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1073";
+	match "device" "0x0012";
+	action "kldload snd_ds1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1102";
+	match "device" "0x0002";
+	action "kldload snd_emu10k1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1102";
+	match "device" "0x0004";
+	action "kldload snd_emu10k1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1102";
+	match "device" "0x0008";
+	action "kldload snd_emu10k1";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1274";
+	match "device" "0x5000";
+	action "kldload snd_es137x";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1274";
+	match "device" "0x1371";
+	action "kldload snd_es137x";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x3274";
+	match "device" "0x1371";
+	action "kldload snd_es137x";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1274";
+	match "device" "0x5880";
+	action "kldload snd_es137x";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1102";
+	match "device" "0x8938";
+	action "kldload snd_es137x";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1319";
+	match "device" "0x0801";
+	action "kldload snd_fm801";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1319";
+	match "device" "0x0802";
+	action "kldload snd_fm801";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x1c20";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x1d20";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x2668";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x269a";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x27d8";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x284b";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x293e";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x3a3e";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x3a6e";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x3b56";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x3b57";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0x811b";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x026c";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0371";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x03e4";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x03f0";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x044a";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x044b";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x055c";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x055d";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0774";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0775";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0776";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0777";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x07fc";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x07fd";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0ac0";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0ac1";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0ac2";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0ac3";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0d94";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0d95";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0d96";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0x0d97";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0x437b";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0x4383";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0x793b";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0x7919";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0x960f";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa00";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa08";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa10";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa18";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa20";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa28";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa30";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa38";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa40";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xaa48";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1002";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x17f3";
+	match "device" "0x3010";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1106";
+	match "device" "0x3288";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1106";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1039";
+	match "device" "0x7502";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1039";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10b9";
+	match "device" "0x5461";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10b9";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x8086";
+	match "device" "0xd600";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x30b5";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x30a2";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x30aa";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x30b0";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x3013";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x3010";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x30a5";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x2801";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0x2802";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x103c";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1028";
+	match "device" "0x01f9";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1028";
+	match "device" "0x01cc";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1028";
+	match "device" "0x0227";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1028";
+	match "device" "0x0228";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1028";
+	match "device" "0x01c9";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1028";
+	match "device" "0x01d7";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1028";
+	match "device" "0x01da";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1028";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1558";
+	match "device" "0x0900";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1558";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1025";
+	match "device" "0x010f";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1025";
+	match "device" "0x0127";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1025";
+	match "device" "0x012f";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1025";
+	match "device" "0x0133";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1025";
+	match "device" "0x0110";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1025";
+	match "device" "0x011b";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1025";
+	match "device" "0x011f";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1025";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x1153";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x1263";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x1263";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x1323";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x1338";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x1339";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x13c2";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x1971";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x1993";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x817f";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x81cb";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x81cb";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x81e7";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x81ec";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0x8234";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10de";
+	match "device" "0xcb84";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1043";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1014";
+	match "device" "0x02f6";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1014";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x17aa";
+	match "device" "0x2066";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x17aa";
+	match "device" "0x384e";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x17aa";
+	match "device" "0x1015";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x17aa";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x144d";
+	match "device" "0xc027";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x144d";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x161f";
+	match "device" "0x203d";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x161f";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x106b";
+	match "device" "0x00a1";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x104d";
+	match "device" "0x81cc";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x104d";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1854";
+	match "device" "0x0018";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1854";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1734";
+	match "device" "0x10b8";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1734";
+	match "device" "0x10cd";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1734";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10cf";
+	match "device" "0x1326";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10cf";
+	match "device" "0x142d";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10cf";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1179";
+	match "device" "0x0001";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1179";
+	match "device" "0xff01";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1179";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1462";
+	match "device" "0x0349";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1462";
+	match "device" "0x034a";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1462";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1458";
+	match "device" "0xa022";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1458";
+	match "device" "0xffff";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1584";
+	match "device" "0x9075";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1584";
+	match "device" "0x9080";
+	action "kldload snd_hda";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x125d";
+	match "device" "0x1988";
+	action "kldload snd_maestro3";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x125d";
+	match "device" "0x199a";
+	action "kldload snd_maestro3";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1285";
+	match "device" "0x0100";
+	action "kldload snd_maestro";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x125d";
+	match "device" "0x1968";
+	action "kldload snd_maestro";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x125d";
+	match "device" "0x1978";
+	action "kldload snd_maestro";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10c8";
+	match "device" "0x8005";
+	action "kldload snd_neomagic";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10c8";
+	match "device" "0x8006";
+	action "kldload snd_neomagic";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x125d";
+	match "device" "0x1969";
+	action "kldload snd_solo";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1023";
+	match "device" "0x2000";
+	action "kldload snd_t4dwave";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1023";
+	match "device" "0x2001";
+	action "kldload snd_t4dwave";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x10b9";
+	match "device" "0x5451";
+	action "kldload snd_t4dwave";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1039";
+	match "device" "0x7018";
+	action "kldload snd_t4dwave";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1106";
+	match "device" "0x3059";
+	action "kldload snd_via8233";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x1106";
+	match "device" "0x3058";
+	action "kldload snd_via82c686";
+};
+
+nomatch 10 {
+	match "bus" "pci[0-9]+";
+	match "vendor" "0x5333";
+	match "device" "0xca00";
+	action "kldload snd_vibes";
+};
