--- a/usr.sbin/rpc.lockd/Makefile
+++ b/usr.sbin/rpc.lockd/Makefile
@@ -5,7 +5,7 @@
 MAN=	rpc.lockd.8
 MLINKS=	rpc.lockd.8 lockd.8
 SRCS=	kern.c nlm_prot_svc.c lockd.c lock_proc.c lockd_lock.c
-SRCS+=	nlm_prot_xdr.c sm_inter_xdr.c
+SRCS+=	nlm_prot_xdr.c sm_inter_xdr.c nlm_syscall.c
 
 CFLAGS+= -I. -I${.CURDIR}/../../include/rpcsvc
 WARNS?= 3
--- /dev/null
+++ b/usr.sbin/rpc.lockd/nlm_syscall.c
@@ -0,0 +1,11 @@
+#include <sys/syscall.h>
+
+#ifndef SYS_nlm_syscall
+#define SYS_nlm_syscall		154
+#endif
+
+int
+nlm_syscall (int a, int b, int c, char **d)
+{
+  return syscall (SYS_nlm_syscall, a, b, c, d);
+}
