diff -Nurd freebsd-libs-7.1.orig/lib/libcam/camlib.c freebsd-libs-7.1/lib/libcam/camlib.c
--- freebsd-libs-7.1.orig/lib/libcam/camlib.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libcam/camlib.c	2009-04-07 11:07:26 +0200
@@ -24,7 +24,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libcam/camlib.c,v 1.16.2.1.2.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/types.h>
 #include <sys/param.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libcam/scsi_cmdparse.c freebsd-libs-7.1/lib/libcam/scsi_cmdparse.c
--- freebsd-libs-7.1.orig/lib/libcam/scsi_cmdparse.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libcam/scsi_cmdparse.c	2009-04-07 11:07:26 +0200
@@ -35,7 +35,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libcam/scsi_cmdparse.c,v 1.7.32.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/types.h>
 
diff -Nurd freebsd-libs-7.1.orig/lib/libdevstat/devstat.c freebsd-libs-7.1/lib/libdevstat/devstat.c
--- freebsd-libs-7.1.orig/lib/libdevstat/devstat.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libdevstat/devstat.c	2009-04-07 11:07:26 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libdevstat/devstat.c,v 1.29.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/types.h>
 #include <sys/sysctl.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libgeom/geom_util.c freebsd-libs-7.1/lib/libgeom/geom_util.c
--- freebsd-libs-7.1.orig/lib/libgeom/geom_util.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libgeom/geom_util.c	2009-04-07 11:07:26 +0200
@@ -25,7 +25,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libgeom/geom_util.c,v 1.1.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/disk.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libipx/ipx_addr.c freebsd-libs-7.1/lib/libipx/ipx_addr.c
--- freebsd-libs-7.1.orig/lib/libipx/ipx_addr.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libipx/ipx_addr.c	2009-04-07 11:07:26 +0200
@@ -31,7 +31,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libipx/ipx_addr.c,v 1.9.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 static char sccsid[] = "@(#)ipx_addr.c";
diff -Nurd freebsd-libs-7.1.orig/lib/libipx/ipx_ntoa.c freebsd-libs-7.1/lib/libipx/ipx_ntoa.c
--- freebsd-libs-7.1.orig/lib/libipx/ipx_ntoa.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libipx/ipx_ntoa.c	2009-04-07 11:07:26 +0200
@@ -28,7 +28,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libipx/ipx_ntoa.c,v 1.7.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 static char sccsid[] = "@(#)ipx_ntoa.c";
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_amd64.c freebsd-libs-7.1/lib/libkvm/kvm_amd64.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_amd64.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_amd64.c	2009-04-07 11:07:26 +0200
@@ -32,7 +32,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_amd64.c,v 1.22.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 #if 0
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_arm.c freebsd-libs-7.1/lib/libkvm/kvm_arm.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_arm.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_arm.c	2009-04-07 11:07:26 +0200
@@ -36,7 +36,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_arm.c,v 1.4.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/elf32.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm.c freebsd-libs-7.1/lib/libkvm/kvm.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm.c	2009-04-07 11:07:26 +0200
@@ -32,7 +32,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm.c,v 1.31.2.1.2.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 #if 0
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_cptime.c freebsd-libs-7.1/lib/libkvm/kvm_cptime.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_cptime.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_cptime.c	2009-04-07 11:07:26 +0200
@@ -29,7 +29,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_cptime.c,v 1.1.2.1.2.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/pcpu.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_file.c freebsd-libs-7.1/lib/libkvm/kvm_file.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_file.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_file.c	2009-04-07 11:07:26 +0200
@@ -28,7 +28,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_file.c,v 1.18.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 #if 0
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_getloadavg.c freebsd-libs-7.1/lib/libkvm/kvm_getloadavg.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_getloadavg.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_getloadavg.c	2009-04-07 11:07:26 +0200
@@ -28,7 +28,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_getloadavg.c,v 1.10.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 #if 0
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_getswapinfo.c freebsd-libs-7.1/lib/libkvm/kvm_getswapinfo.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_getswapinfo.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_getswapinfo.c	2009-04-07 11:07:26 +0200
@@ -25,7 +25,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_getswapinfo.c,v 1.27.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/time.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_i386.c freebsd-libs-7.1/lib/libkvm/kvm_i386.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_i386.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_i386.c	2009-04-07 11:07:26 +0200
@@ -32,7 +32,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_i386.c,v 1.21.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 #if 0
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_minidump_amd64.c freebsd-libs-7.1/lib/libkvm/kvm_minidump_amd64.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_minidump_amd64.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_minidump_amd64.c	2009-04-07 11:07:26 +0200
@@ -24,7 +24,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_minidump_amd64.c,v 1.2.8.1 2008/11/25 02:59:29 kensmith Exp $");
 
 /*
  * AMD64 machine dependent routines for kvm and minidumps. 
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_minidump_i386.c freebsd-libs-7.1/lib/libkvm/kvm_minidump_i386.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_minidump_i386.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_minidump_i386.c	2009-04-07 11:07:26 +0200
@@ -24,7 +24,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_minidump_i386.c,v 1.2.8.1 2008/11/25 02:59:29 kensmith Exp $");
 
 /*
  * AMD64 machine dependent routines for kvm and minidumps. 
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_pcpu.c freebsd-libs-7.1/lib/libkvm/kvm_pcpu.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_pcpu.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_pcpu.c	2009-04-07 11:07:26 +0200
@@ -29,7 +29,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_pcpu.c,v 1.1.2.1.2.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/pcpu.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_powerpc.c freebsd-libs-7.1/lib/libkvm/kvm_powerpc.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_powerpc.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_powerpc.c	2009-04-07 11:07:26 +0200
@@ -36,7 +36,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_powerpc.c,v 1.3.18.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_proc.c freebsd-libs-7.1/lib/libkvm/kvm_proc.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_proc.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_proc.c	2009-04-07 11:07:26 +0200
@@ -38,7 +38,6 @@
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_proc.c,v 1.94.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 /*
  * Proc traversal interface for kvm.  ps and w are (probably) the exclusive
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_sparc64.c freebsd-libs-7.1/lib/libkvm/kvm_sparc64.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_sparc64.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_sparc64.c	2009-04-07 11:07:26 +0200
@@ -34,7 +34,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_sparc64.c,v 1.9.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 #if 0
diff -Nurd freebsd-libs-7.1.orig/lib/libkvm/kvm_sparc.c freebsd-libs-7.1/lib/libkvm/kvm_sparc.c
--- freebsd-libs-7.1.orig/lib/libkvm/kvm_sparc.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libkvm/kvm_sparc.c	2009-04-07 11:07:26 +0200
@@ -32,7 +32,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libkvm/kvm_sparc.c,v 1.7.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #if defined(LIBC_SCCS) && !defined(lint)
 #if 0
diff -Nurd freebsd-libs-7.1.orig/lib/libusbhid/data.c freebsd-libs-7.1/lib/libusbhid/data.c
--- freebsd-libs-7.1.orig/lib/libusbhid/data.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libusbhid/data.c	2009-04-07 11:07:26 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libusbhid/data.c,v 1.6.30.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <assert.h>
 #include <stdlib.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libusbhid/descr.c freebsd-libs-7.1/lib/libusbhid/descr.c
--- freebsd-libs-7.1.orig/lib/libusbhid/descr.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libusbhid/descr.c	2009-04-07 11:07:26 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libusbhid/descr.c,v 1.8.30.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/types.h>
 
diff -Nurd freebsd-libs-7.1.orig/lib/libusbhid/parse.c freebsd-libs-7.1/lib/libusbhid/parse.c
--- freebsd-libs-7.1.orig/lib/libusbhid/parse.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libusbhid/parse.c	2009-04-07 11:07:26 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libusbhid/parse.c,v 1.10.26.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <assert.h>
 #include <stdlib.h>
diff -Nurd freebsd-libs-7.1.orig/lib/libusbhid/usage.c freebsd-libs-7.1/lib/libusbhid/usage.c
--- freebsd-libs-7.1.orig/lib/libusbhid/usage.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/lib/libusbhid/usage.c	2009-04-07 11:07:26 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libusbhid/usage.c,v 1.8.30.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <assert.h>
 #include <ctype.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/cam.c freebsd-libs-7.1/sys/cam/cam.c
--- freebsd-libs-7.1.orig/sys/cam/cam.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/cam.c	2009-04-07 11:07:27 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/cam.c,v 1.11.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #ifdef _KERNEL
diff -Nurd freebsd-libs-7.1.orig/sys/cam/cam_periph.c freebsd-libs-7.1/sys/cam/cam_periph.c
--- freebsd-libs-7.1.orig/sys/cam/cam_periph.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/cam_periph.c	2009-04-07 11:07:27 +0200
@@ -28,7 +28,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/cam_periph.c,v 1.69.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/systm.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/cam_queue.c freebsd-libs-7.1/sys/cam/cam_queue.c
--- freebsd-libs-7.1.orig/sys/cam/cam_queue.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/cam_queue.c	2009-04-07 11:07:27 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/cam_queue.c,v 1.9.18.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/systm.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/cam_sim.c freebsd-libs-7.1/sys/cam/cam_sim.c
--- freebsd-libs-7.1.orig/sys/cam/cam_sim.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/cam_sim.c	2009-04-07 11:07:27 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/cam_sim.c,v 1.11.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/systm.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/cam_xpt.c freebsd-libs-7.1/sys/cam/cam_xpt.c
--- freebsd-libs-7.1.orig/sys/cam/cam_xpt.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/cam_xpt.c	2009-04-07 11:07:27 +0200
@@ -28,7 +28,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/cam_xpt.c,v 1.190.2.6.2.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/bus.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_all.c freebsd-libs-7.1/sys/cam/scsi/scsi_all.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_all.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_all.c	2009-04-07 11:07:26 +0200
@@ -28,7 +28,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_all.c,v 1.51.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_cd.c freebsd-libs-7.1/sys/cam/scsi/scsi_cd.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_cd.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_cd.c	2009-04-07 11:07:26 +0200
@@ -46,7 +46,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_cd.c,v 1.102.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include "opt_cd.h"
 
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_ch.c freebsd-libs-7.1/sys/cam/scsi/scsi_ch.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_ch.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_ch.c	2009-04-07 11:07:26 +0200
@@ -68,7 +68,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_ch.c,v 1.46.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/queue.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_da.c freebsd-libs-7.1/sys/cam/scsi/scsi_da.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_da.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_da.c	2009-04-07 11:07:26 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_da.c,v 1.224.2.2.2.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_low.c freebsd-libs-7.1/sys/cam/scsi/scsi_low.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_low.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_low.c	2009-04-07 11:07:26 +0200
@@ -2,7 +2,6 @@
 /*	$NetBSD$	*/
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_low.c,v 1.29.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #define	SCSI_LOW_STATICS
 #define	SCSI_LOW_DEBUG
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_low_pisa.c freebsd-libs-7.1/sys/cam/scsi/scsi_low_pisa.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_low_pisa.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_low_pisa.c	2009-04-07 11:07:26 +0200
@@ -32,7 +32,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_low_pisa.c,v 1.8.18.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #ifdef	__NetBSD__
 #include <sys/param.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_pass.c freebsd-libs-7.1/sys/cam/scsi/scsi_pass.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_pass.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_pass.c	2009-04-07 11:07:26 +0200
@@ -26,7 +26,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_pass.c,v 1.48.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/systm.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_pt.c freebsd-libs-7.1/sys/cam/scsi/scsi_pt.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_pt.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_pt.c	2009-04-07 11:07:26 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_pt.c,v 1.47.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/queue.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_sa.c freebsd-libs-7.1/sys/cam/scsi/scsi_sa.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_sa.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_sa.c	2009-04-07 11:07:27 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_sa.c,v 1.113.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/queue.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_ses.c freebsd-libs-7.1/sys/cam/scsi/scsi_ses.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_ses.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_ses.c	2009-04-07 11:07:27 +0200
@@ -25,7 +25,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_ses.c,v 1.35.2.1.2.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/queue.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_sg.c freebsd-libs-7.1/sys/cam/scsi/scsi_sg.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_sg.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_sg.c	2009-04-07 11:07:27 +0200
@@ -30,7 +30,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_sg.c,v 1.9.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/systm.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_targ_bh.c freebsd-libs-7.1/sys/cam/scsi/scsi_targ_bh.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_targ_bh.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_targ_bh.c	2009-04-07 11:07:27 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_targ_bh.c,v 1.25.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 #include <sys/queue.h>
diff -Nurd freebsd-libs-7.1.orig/sys/cam/scsi/scsi_target.c freebsd-libs-7.1/sys/cam/scsi/scsi_target.c
--- freebsd-libs-7.1.orig/sys/cam/scsi/scsi_target.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/cam/scsi/scsi_target.c	2009-04-07 11:07:27 +0200
@@ -28,7 +28,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/cam/scsi/scsi_target.c,v 1.73.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 
 #include <sys/param.h>
diff -Nurd freebsd-libs-7.1.orig/sys/kern/subr_sbuf.c freebsd-libs-7.1/sys/kern/subr_sbuf.c
--- freebsd-libs-7.1.orig/sys/kern/subr_sbuf.c	2008-11-25 03:59:29 +0100
+++ freebsd-libs-7.1/sys/kern/subr_sbuf.c	2009-04-07 11:07:27 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/kern/subr_sbuf.c,v 1.30.6.1 2008/11/25 02:59:29 kensmith Exp $");
 
 #include <sys/param.h>
 
