diff -ur lib.old/libcam/Makefile lib/libcam/Makefile
--- lib.old/libcam/Makefile	2006-04-19 09:40:00.000000000 +0200
+++ lib/libcam/Makefile	2006-04-19 09:40:19.000000000 +0200
@@ -5,8 +5,7 @@
 SRCS=		camlib.c scsi_cmdparse.c scsi_all.c scsi_da.c scsi_sa.c cam.c
 INCS=		camlib.h
 
-DPADD=		${LIBSBUF}
-LDADD+=		-lsbuf
+LDADD+=		-L../libsbuf/ -lsbuf
 
 MAN=		cam.3 cam_cdbparse.3
 
@@ -34,9 +33,11 @@
 		cam_cdbparse.3 csio_encode_visit.3 \
 		cam_cdbparse.3 buff_encode_visit.3
 
-.PATH:		${.CURDIR}/../../sys/cam/scsi ${.CURDIR}/../../sys/cam
+KERNEL=		${.CURDIR}/../kernel
 
-SDIR=		${.CURDIR}/../../sys
+.PATH:		${KERNEL}/sys/cam/scsi ${KERNEL}/sys/cam
+
+SDIR=		${KERNEL}/sys
 CFLAGS+=	-I${.CURDIR} -I${SDIR}
 
 .include <bsd.lib.mk>
diff -ur lib.old/libcam/camlib.h lib/libcam/camlib.h
--- lib.old/libcam/camlib.h	2006-04-19 09:40:00.000000000 +0200
+++ lib/libcam/camlib.h	2006-04-19 09:38:34.000000000 +0200
@@ -150,13 +150,11 @@
 /*
  * Buffer encoding/decoding routines, from the old SCSI library.
  */
-int csio_decode(struct ccb_scsiio *csio, const char *fmt, ...)
-		__printflike(2, 3);
+int csio_decode(struct ccb_scsiio *csio, const char *fmt, ...);
 int csio_decode_visit(struct ccb_scsiio *csio, const char *fmt,
 		      void (*arg_put)(void *, int, void *, int, char *),
 		      void *puthook);
-int buff_decode(u_int8_t *buff, size_t len, const char *fmt, ...)
-		__printflike(3, 4);
+int buff_decode(u_int8_t *buff, size_t len, const char *fmt, ...);
 int buff_decode_visit(u_int8_t *buff, size_t len, const char *fmt,
 		      void (*arg_put)(void *, int, void *, int, char *),
 		      void *puthook);
@@ -168,8 +166,7 @@
 		     int timeout, const char *cmd_spec,
 		     int (*arg_get)(void *hook, char *field_name),
 		     void *gethook);
-int csio_encode(struct ccb_scsiio *csio, const char *fmt, ...)
-		__printflike(2, 3);
+int csio_encode(struct ccb_scsiio *csio, const char *fmt, ...);
 int buff_encode_visit(u_int8_t *buff, size_t len, const char *fmt,
 		      int (*arg_get)(void *hook, char *field_name),
 		      void *gethook);
