--- a/sys/cam/cam.c
+++ b/sys/cam/cam.c
@@ -36,6 +36,7 @@
 #include <sys/sysctl.h>
 #else /* _KERNEL */
 #include <stdlib.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <string.h>
 #include <camlib.h>
--- a/sys/cam/scsi/scsi_all.c
+++ b/sys/cam/scsi/scsi_all.c
@@ -43,6 +43,7 @@
 #include <sys/sysctl.h>
 #else
 #include <errno.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
