diff -ur lib.old/libsbuf/Makefile lib/libsbuf/Makefile
--- lib.old/libsbuf/Makefile	2003-08-18 17:25:39.000000000 +0200
+++ lib/libsbuf/Makefile	2005-08-13 11:06:10.000000000 +0200
@@ -1,7 +1,8 @@
 # $FreeBSD: src/lib/libsbuf/Makefile,v 1.5 2003/08/18 15:25:39 obrien Exp $
 
 LIB=	sbuf
-SHLIBDIR?= /lib
+SHLIBDIR?= /usr/lib
+SHLIB_MAJOR= 0
 SRCS=	subr_sbuf.c 
 WARNS?= 2
 
diff -ur lib.old/libsbuf/subr_sbuf.c lib/libsbuf/subr_sbuf.c
--- lib.old/libsbuf/subr_sbuf.c	2005-08-12 17:30:02.000000000 +0200
+++ lib/libsbuf/subr_sbuf.c	2005-08-13 11:06:10.000000000 +0200
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/sys/kern/subr_sbuf.c,v 1.28 2004/07/09 11:37:44 des Exp $");
 
 #include <sys/param.h>
 
