--- a/lib/libsbuf/Makefile
+++ b/lib/libsbuf/Makefile
@@ -4,6 +4,7 @@
 SHLIBDIR?= /lib
 SRCS=	subr_sbuf.c 
 WARNS?= 2
+CFLAGS+= -I${.CURDIR}/../../sys
 
 .PATH:	${.CURDIR}/../../sys/kern
 
--- a/sys/sys/sbuf.h
+++ b/sys/sys/sbuf.h
@@ -31,7 +31,8 @@
 #ifndef _SYS_SBUF_H_
 #define	_SYS_SBUF_H_
 
-#include <sys/_types.h>
+#include <sys/types.h>
+#include <stdarg.h>
 
 struct sbuf;
 struct sbuf_drain_data;
