--- a/lib/libsbuf/Makefile
+++ b/lib/libsbuf/Makefile
@@ -3,6 +3,7 @@
 LIB=		sbuf
 SHLIBDIR?= 	/lib
 SRCS=		subr_sbuf.c 
+CFLAGS+=	-I${.CURDIR}/../../sys
 
 SHLIB_MAJOR	= 6
 
--- a/sys/sys/sbuf.h
+++ b/sys/sys/sbuf.h
@@ -31,7 +31,8 @@
 #ifndef _SYS_SBUF_H_
 #define	_SYS_SBUF_H_
 
-#include <sys/_types.h>
+#include <sys/types.h>
+#include <stdarg.h>
 
 struct sbuf;
 typedef int (sbuf_drain_func)(void *, const char *, int);
