diff -ur lib.old/libkvm/Makefile lib/libkvm/Makefile
--- lib.old/libkvm/Makefile	2006-04-19 09:43:10.000000000 +0200
+++ lib/libkvm/Makefile	2006-04-19 09:42:15.000000000 +0200
@@ -2,8 +2,7 @@
 # $FreeBSD: src/lib/libkvm/Makefile,v 1.14 2003/08/18 15:25:38 obrien Exp $
 
 LIB=	kvm
-SHLIBDIR?= /lib
-CFLAGS+=-DLIBC_SCCS -I${.CURDIR}
+SHLIBDIR?= /lib
 SRCS=	kvm.c kvm_${MACHINE_ARCH}.c kvm_file.c kvm_getloadavg.c \
 	kvm_getswapinfo.c kvm_proc.c
 INCS=	kvm.h
diff -ur lib.old/libkvm/kvm.c lib/libkvm/kvm.c
--- lib.old/libkvm/kvm.c	2006-04-19 09:43:10.000000000 +0200
+++ lib/libkvm/kvm.c	2006-04-19 09:38:34.000000000 +0200
@@ -50,6 +50,7 @@
 #include <sys/stat.h>
 #include <sys/sysctl.h>
 #include <sys/linker.h>
+#include <sys/sysctl.h>
 
 #include <vm/vm.h>
 #include <vm/vm_param.h>
@@ -62,11 +63,14 @@
 #include <limits.h>
 #include <nlist.h>
 #include <paths.h>
+#include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
+
+
 #include "kvm_private.h"
 
 /* from src/lib/libc/gen/nlist.c */
diff -ur lib.old/libkvm/kvm_file.c lib/libkvm/kvm_file.c
--- lib.old/libkvm/kvm_file.c	2006-04-19 09:43:10.000000000 +0200
+++ lib/libkvm/kvm_file.c	2006-04-19 09:38:34.000000000 +0200
@@ -63,7 +63,7 @@
 #include <sys/sysctl.h>
 
 #include <limits.h>
-#include <ndbm.h>
+#include <gdbm-ndbm.h>
 #include <paths.h>
 
 #include "kvm_private.h"
@@ -80,6 +80,7 @@
 	int op, arg, nfiles;
 	long filehead_o;
 {
+#if 0
 	int buflen = kd->arglen, n = 0;
 	struct file *fp;
 	char *where = kd->argspc;
@@ -117,6 +118,8 @@
 		return (0);
 	}
 	return (nfiles);
+#endif
+	return 0;
 }
 
 char *
@@ -125,6 +128,7 @@
 	int op, arg;
 	int *cnt;
 {
+#if 0
 	int mib[2], st, nfiles;
 	size_t size;
 	struct file *fp, *fplim;
@@ -188,4 +192,6 @@
 	}
 	*cnt = nfiles;
 	return (kd->argspc);
+#endif
+	return 0;
 }
diff -ur lib.old/libkvm/kvm_getloadavg.c lib/libkvm/kvm_getloadavg.c
--- lib.old/libkvm/kvm_getloadavg.c	2006-04-19 09:43:10.000000000 +0200
+++ lib/libkvm/kvm_getloadavg.c	2006-04-19 09:38:34.000000000 +0200
@@ -47,6 +47,7 @@
 #include <limits.h>
 #include <nlist.h>
 #include <kvm.h>
+#include <sys/_types.h>
 
 #include "kvm_private.h"
 
@@ -58,6 +59,11 @@
 	{ "" },
 };
 
+struct loadavg {
+	__fixpt_t       ldavg[3];
+	long            fscale;
+};
+
 /*
  * kvm_getloadavg() -- Get system load averages, from live or dead kernels.
  *
diff -ur lib.old/libkvm/kvm_getswapinfo.c lib/libkvm/kvm_getswapinfo.c
--- lib.old/libkvm/kvm_getswapinfo.c	2006-04-19 09:43:10.000000000 +0200
+++ lib/libkvm/kvm_getswapinfo.c	2006-04-19 09:38:34.000000000 +0200
@@ -150,8 +150,10 @@
 			swap_ary[unswdev].ksw_total = ttl;
 			swap_ary[unswdev].ksw_used = xsd.xsw_used;
 			swap_ary[unswdev].ksw_flags = xsd.xsw_flags;
+#if 0
 			GETSWDEVNAME(xsd.xsw_dev, swap_ary[unswdev].ksw_devname,
 			     flags);
+#endif
 		}
 		tot.ksw_total += ttl;
 		tot.ksw_used += xsd.xsw_used;
diff -ur lib.old/libkvm/kvm_private.h lib/libkvm/kvm_private.h
--- lib.old/libkvm/kvm_private.h	2006-04-19 09:43:10.000000000 +0200
+++ lib/libkvm/kvm_private.h	2006-04-19 09:38:34.000000000 +0200
@@ -71,7 +71,7 @@
  * Functions used internally by kvm, but across kvm modules.
  */
 void	 _kvm_err(kvm_t *kd, const char *program, const char *fmt, ...)
-	    __printflike(3, 4);
+	    __attribute__((format(printf,3,4)));
 void	 _kvm_freeprocs(kvm_t *kd);
 void	 _kvm_freevtop(kvm_t *);
 int	 _kvm_initvtop(kvm_t *);
@@ -79,5 +79,5 @@
 void	*_kvm_malloc(kvm_t *kd, size_t);
 void	*_kvm_realloc(kvm_t *kd, void *, size_t);
 void	 _kvm_syserr (kvm_t *kd, const char *program, const char *fmt, ...)
-	    __printflike(3, 4);
+	    __attribute__((format(printf,3,4)));
 int	 _kvm_uvatop(kvm_t *, const struct proc *, u_long, u_long *);
diff -ur lib.old/libkvm/kvm_proc.c lib/libkvm/kvm_proc.c
--- lib.old/libkvm/kvm_proc.c	2006-04-19 09:43:10.000000000 +0200
+++ lib/libkvm/kvm_proc.c	2006-04-19 09:38:34.000000000 +0200
@@ -93,6 +93,7 @@
 	struct kinfo_proc *bp;
 	int maxcnt;
 {
+#if 0
 	int cnt = 0;
 	struct kinfo_proc kinfo_proc, *kp;
 	struct pgrp pgrp;
@@ -413,6 +414,8 @@
 		++cnt;
 	}
 	return (cnt);
+#endif
+	return -1;
 }
 
 /*
