---
 lib/libkiconv/Makefile |    1 +
 lib/libkiconv/quirks.h |    1 +
 2 files changed, 2 insertions(+)

--- a/lib/libkiconv/Makefile
+++ b/lib/libkiconv/Makefile
@@ -14,5 +14,6 @@
 		kiconv.3 kiconv_add_xlat16_table.3
 
 CFLAGS+=	-I${.CURDIR}/../../sys
+LDADD=	-ldl -lfreebsd
 
 .include <bsd.lib.mk>
--- a/lib/libkiconv/quirks.h
+++ b/lib/libkiconv/quirks.h
@@ -25,6 +25,7 @@
  *
  * $FreeBSD: src/lib/libkiconv/quirks.h,v 1.1.28.1 2008/11/25 02:59:29 kensmith Exp $
  */
+#include <stdint.h>
 
 #ifndef _KICONV_QUIRKS_H_
 #define _KICONV_QUIRKS_H_
