---
 lib/libusbhid/Makefile |    1 +
 1 file changed, 1 insertion(+)

Index: freebsd-libs/lib/libusbhid/Makefile
===================================================================
--- freebsd-libs.orig/lib/libusbhid/Makefile	2009-07-19 17:25:24.000000000 +0000
+++ freebsd-libs/lib/libusbhid/Makefile	2009-11-13 13:45:33.000000000 +0000
@@ -4,6 +4,7 @@
 LIB=	usbhid
 MAN=	usbhid.3
 
+LDADD= -lbsd
 SHLIB_MAJOR= 4
 
 MLINKS=	usbhid.3 libusbhid.3 usbhid.3 hid_get_report_desc.3 \
