---
 lib/libusbhid/Makefile |    1 +
 1 file changed, 1 insertion(+)

--- a/lib/libusbhid/Makefile
+++ b/lib/libusbhid/Makefile
@@ -4,6 +4,7 @@
 LIB=	usbhid
 MAN=	usbhid.3
 
+LDADD= -lbsd
 SHLIB_MAJOR= 4
 
 MLINKS=	usbhid.3 libusbhid.3 usbhid.3 hid_get_report_desc.3 \
--- a/lib/libusbhid/parse.c
+++ b/lib/libusbhid/parse.c
@@ -143,7 +143,7 @@
  *	hid_start_parse
  *------------------------------------------------------------------------*/
 hid_data_t
-hid_start_parse(report_desc_t d, int kindset, int id __unused)
+hid_start_parse(report_desc_t d, int kindset, int id)
 {
 	struct hid_data *s;
 
