Index: freebsd-libs/lib/libnetgraph/netgraph.h
===================================================================
--- freebsd-libs.orig/lib/libnetgraph/netgraph.h	2009-11-16 20:42:38.000000000 +0000
+++ freebsd-libs/lib/libnetgraph/netgraph.h	2009-11-16 20:42:39.000000000 +0000
@@ -43,14 +43,15 @@
 #ifndef _NETGRAPH_H_
 #define _NETGRAPH_H_
 
+#include <stdint.h>
 #include <sys/types.h>
 #include <netgraph/ng_message.h>
 
 __BEGIN_DECLS
 int	NgMkSockNode(const char *, int *, int *);
-int	NgNameNode(int, const char *, const char *, ...) __printflike(3, 4);
+int	NgNameNode(int, const char *, const char *, ...);
 int	NgSendMsg(int, const char *, int, int, const void *, size_t);
-int	NgSendAsciiMsg(int, const char *, const char *, ...) __printflike(3, 4);
+int	NgSendAsciiMsg(int, const char *, const char *, ...);
 int	NgSendReplyMsg(int, const char *,
 		const struct ng_mesg *, const void *, size_t);
 int	NgRecvMsg(int, struct ng_mesg *, size_t, char *);
Index: freebsd-libs/lib/libnetgraph/sock.c
===================================================================
--- freebsd-libs.orig/lib/libnetgraph/sock.c	2009-11-16 20:42:38.000000000 +0000
+++ freebsd-libs/lib/libnetgraph/sock.c	2009-11-16 20:42:39.000000000 +0000
@@ -44,6 +44,8 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <stdarg.h>
+#include <stdint.h>
+
 #include <netgraph/ng_message.h>
 #include <netgraph/ng_socket.h>
 
Index: freebsd-libs/lib/libnetgraph/msg.c
===================================================================
--- freebsd-libs.orig/lib/libnetgraph/msg.c	2009-11-16 20:42:38.000000000 +0000
+++ freebsd-libs/lib/libnetgraph/msg.c	2009-11-16 20:42:39.000000000 +0000
@@ -44,12 +44,15 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <stdarg.h>
+#include <stdint.h>
 #include <netgraph/ng_message.h>
 #include <netgraph/ng_socket.h>
 
 #include "netgraph.h"
 #include "internal.h"
 
+#define INFTIM          (-1)
+
 /* Next message token value */
 static int	gMsgId;
 
Index: freebsd-libs/lib/libnetgraph/Makefile
===================================================================
--- freebsd-libs.orig/lib/libnetgraph/Makefile	2009-11-16 20:43:39.000000000 +0000
+++ freebsd-libs/lib/libnetgraph/Makefile	2009-11-16 20:44:22.000000000 +0000
@@ -4,6 +4,7 @@
 LIB=		netgraph
 WARNS?=		3
 MAN=		netgraph.3
+LDADD= -lbsd
 
 SHLIB_MAJOR=	4
 
