--- a/lib/libusb/usb.h
+++ b/lib/libusb/usb.h
@@ -28,7 +28,7 @@
 #define	_LIBUSB20_COMPAT_01_H_
 
 #include <sys/param.h>
-#include <sys/endian.h>
+#include <endian.h>
 
 #include <stdint.h>
 
--- a/lib/libusb/Makefile
+++ b/lib/libusb/Makefile
@@ -15,6 +15,7 @@
 MAN=		libusb.3 libusb20.3
 MKLINT=		no
 NOGCCERROR=
+LDADD+=		-lpthread -lrt -lbsd
 
 MLINKS+=	libusb.3 usb.3
 
--- a/lib/libusb/libusb.h
+++ b/lib/libusb/libusb.h
@@ -29,6 +29,7 @@
 
 #include <sys/time.h>
 #include <sys/types.h>
+#include <stdint.h>
 
 #ifdef __cplusplus
 extern	"C" {
--- a/lib/libusb/libusb20_desc.h
+++ b/lib/libusb/libusb20_desc.h
@@ -45,6 +45,8 @@
 #ifndef _LIBUSB20_DESC_H_
 #define	_LIBUSB20_DESC_H_
 
+#include <stdint.h>
+
 #ifdef __cplusplus
 extern	"C" {
 #endif
