--- a/lib/libusb/usb.h
+++ b/lib/libusb/usb.h
@@ -28,7 +28,7 @@
 #define	_LIBUSB20_COMPAT_01_H_
 
 #include <sys/param.h>
-#include <sys/endian.h>
+#include <endian.h>
 
 #include <stdint.h>
 
--- a/lib/libusb/Makefile
+++ b/lib/libusb/Makefile
@@ -15,6 +15,7 @@
 MAN=		libusb.3 libusb20.3
 MKLINT=		no
 NOGCCERROR=
+LDADD+=		-lpthread -lrt -lbsd
 
 MLINKS+=	libusb.3 usb.3
 
--- a/lib/libusb/libusb20.c
+++ b/lib/libusb/libusb20.c
@@ -1167,7 +1167,7 @@
 {
 	struct libusb20_backend *pbe;
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 	pbe = libusb20_be_alloc(&libusb20_ugen20_backend);
 #else
 	pbe = NULL;
